<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <title>HOWARD README</title>
  <style>
    html {
      color: #1a1a1a;
      background-color: #fdfdfd;
    }
    body {
      margin: 0 auto;
      max-width: 1000px;
      padding-left: 50px;
      padding-right: 50px;
      padding-top: 50px;
      padding-bottom: 50px;
      hyphens: auto;
      overflow-wrap: break-word;
      text-rendering: optimizeLegibility;
      font-kerning: normal;
    }
    @media (max-width: 600px) {
      body {
        font-size: 0.9em;
        padding: 12px;
      }
      h1 {
        font-size: 1.8em;
      }
    }
    @media print {
      html {
        background-color: white;
      }
      body {
        background-color: transparent;
        color: black;
        font-size: 12pt;
      }
      p, h2, h3 {
        orphans: 3;
        widows: 3;
      }
      h2, h3, h4 {
        page-break-after: avoid;
      }
    }
    p {
      margin: 1em 0;
    }
    a {
      color: #1a1a1a;
    }
    a:visited {
      color: #1a1a1a;
    }
    img {
      max-width: 100%;
    }
    svg {
      height: auto;
      max-width: 100%;
    }
    h1, h2, h3, h4, h5, h6 {
      margin-top: 1.4em;
    }
    h5, h6 {
      font-size: 1em;
      font-style: italic;
    }
    h6 {
      font-weight: normal;
    }
    ol, ul {
      padding-left: 1.7em;
      margin-top: 1em;
    }
    li > ol, li > ul {
      margin-top: 0;
    }
    blockquote {
      margin: 1em 0 1em 1.7em;
      padding-left: 1em;
      border-left: 2px solid #e6e6e6;
      color: #606060;
    }
    code {
      font-family: Menlo, Monaco, Consolas, 'Lucida Console', monospace;
      font-size: 85%;
      margin: 0;
      hyphens: manual;
    }
    pre {
      margin: 1em 0;
      overflow: auto;
    }
    pre code {
      padding: 0;
      overflow: visible;
      overflow-wrap: normal;
    }
    .sourceCode {
     background-color: transparent;
     overflow: visible;
    }
    hr {
      background-color: #1a1a1a;
      border: none;
      height: 1px;
      margin: 1em 0;
    }
    table {
      margin: 1em 0;
      border-collapse: collapse;
      width: 100%;
      overflow-x: auto;
      display: block;
      font-variant-numeric: lining-nums tabular-nums;
    }
    table caption {
      margin-bottom: 0.75em;
    }
    tbody {
      margin-top: 0.5em;
      border-top: 1px solid #1a1a1a;
      border-bottom: 1px solid #1a1a1a;
    }
    th {
      border-top: 1px solid #1a1a1a;
      padding: 0.25em 0.5em 0.25em 0.5em;
    }
    td {
      padding: 0.125em 0.5em 0.25em 0.5em;
    }
    header {
      margin-bottom: 4em;
      text-align: center;
    }
    #TOC li {
      list-style: none;
    }
    #TOC ul {
      padding-left: 1.3em;
    }
    #TOC > ul {
      padding-left: 0;
    }
    #TOC a:not(:hover) {
      text-decoration: none;
    }
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    /* The extra [class] is a hack that increases specificity enough to
       override a similar rule in reveal.js */
    ul.task-list[class]{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      font-size: inherit;
      width: 0.8em;
      margin: 0 0.8em 0.2em -1.6em;
      vertical-align: middle;
    }
    .display.math{display: block; text-align: center; margin: 0.5rem auto;}
  </style>
</head>
<body>
<header id="title-block-header">
<h1 class="title">HOWARD README</h1>
</header>
<h1 data-number="1" id="howard"><span
class="header-section-number">1</span> HOWARD</h1>
<figure>
<img src="images/icon.png" title="HOWARD Graphical User Interface"
alt="HOWARD Graphical User Interface" />
<figcaption aria-hidden="true">HOWARD Graphical User
Interface</figcaption>
</figure>
<p>Highly Open Workflow for Annotation &amp; Ranking toward genomic
variant Discovery</p>
<p>HOWARD annotates and prioritizes genetic variations, calculates and
normalizes annotations, translates files in multiple formats (e.g.&#xA0;vcf,
tsv, parquet) and generates variants statistics.</p>
<p>HOWARD annotation is mainly based on a build-in Parquet annotation
method, and external tools such as BCFTOOLS, ANNOVAR, snpEff, Exomiser
and Splice (see docs, automatically downloaded if needed). Parquet
annotation uses annotation database in VCF or BED format, in mutliple
file format: Parquet/duckdb, VCF, BED, TSV, CSV, TBL, JSON.</p>
<p>HOWARD calculation processes variants information to calculate new
information, such as: harmonizes allele frequency (VAF), extracts Nomen
(transcript, cNomen, pNomen&#x2026;) from HGVS fields with an optional list of
personalized transcripts, generates VaRank format barcode.</p>
<p>HOWARD prioritization algorithm uses profiles to flag variants (as
passed or filtered), calculate a prioritization score, and automatically
generate a comment for each variants (example: &#x2018;polymorphism identified
in dbSNP. associated to Lung Cancer. Found in ClinVar
database&#x2019;).Prioritization profiles are defined in a configuration file.
A profile is defined as a list of annotation/value, using wildcards and
comparison options (contains, lower than, greater than, equal&#x2026;).
Annotations fields may be quality values (usually from callers, such as
&#x2018;GQ&#x2019;, &#x2018;DP&#x2019;) or other annotations fields provided by annotations tools,
such as HOWARD itself (example: COSMIC, Clinvar, 1000genomes, PolyPhen,
SIFT). Multiple profiles can be used simultaneously, which is useful to
define multiple validation/prioritization levels (example: &#x2018;standard&#x2019;,
&#x2018;stringent&#x2019;, &#x2018;rare variants&#x2019;, &#x2018;low allele frequency&#x2019;).</p>
<p>HOWARD translates VCF format into multiple formats (e.g.&#xA0;VCF, TSV,
Parquet), by sorting variants using specific fields (example :
&#x2018;prioritization score&#x2019;, &#x2018;allele frequency&#x2019;, &#x2018;gene symbol&#x2019;),
including/excluding annotations/fields, including/excluding variants,
adding fixed columns.</p>
<p>HOWARD generates statistics files with a specific algorithm, snpEff
and BCFTOOLS.</p>
<p>HOWARD is multithreaded through the number of variants and by
database (data-scaling).</p>
<p>HOWARD is able to add plugins for further analyses.</p>
<h2 data-number="1.1" id="table-of-contents"><span
class="header-section-number">1.1</span> Table of contents</h2>
<ul>
<li><a href="#installation">Installation</a>
<ul>
<li><a href="#python">Python</a></li>
<li><a href="#docker">Docker</a></li>
<li><a href="#databases">Databases</a></li>
<li><a href="#configuration">Configuration</a></li>
</ul></li>
<li><a href="#tools">Tools</a>
<ul>
<li><a href="#parameters">Parameters</a></li>
<li><a href="#stats">Stats</a></li>
<li><a href="#convert">Convert</a></li>
<li><a href="#query">Query</a></li>
<li><a href="#annotation">Annotation</a></li>
<li><a href="#calculation">Calculation</a></li>
<li><a href="#prioritization">Prioritization</a></li>
<li><a href="#hgvs-annotation">HGVS annotation</a></li>
<li><a href="#process">Process</a></li>
</ul></li>
<li><a href="#documentation">Documentation</a></li>
<li><a href="#contact">Contact</a></li>
</ul>
<h1 data-number="2" id="installation"><span
class="header-section-number">2</span> Installation</h1>
<p>HOWARD can be installed using <a href="#python">Python</a>, and a <a
href="#docker">Docker</a> installation provides a CLI (Command Line
Interface) with all external tools and useful databases. <a
href="#databases">Databases</a> can be automatically downloaded, or
home-made generated (created or downloaded).</p>
<h2 data-number="2.1" id="python"><span
class="header-section-number">2.1</span> Python</h2>
<p>Install HOWARD using Python Pip tool, and run HOWARD for help
options:</p>
<pre><code>conda create --name=howard python=3.10
conda activate howard
python -m pip install -e .
howard --help</code></pre>
<pre><code>usage: howard [-h] {query,stats,convert,hgvs,annotation,calculation,prioritization,process,databases,gui,help,genebe,minimalize} ...

HOWARD:0.11.0
Highly Open Workflow for Annotation &amp; Ranking toward genomic variant Discovery
HOWARD annotates and prioritizes genetic variations, calculates and normalizes annotations, convert on multiple formats, query variations and generates statistics

Shared arguments:
  -h, --help            show this help message and exit

Tools:
  {query,stats,convert,hgvs,annotation,calculation,prioritization,process,databases,gui,help,genebe,minimalize}
    query               Query genetic variations file in SQL format.
    stats               Statistics on genetic variations file.
    convert             Convert genetic variations file to another format.
    hgvs                HGVS annotation (HUGO internation nomenclature) using refGene, genome and transcripts list.
    annotation          Annotation of genetic variations file using databases/files and tools.
    calculation         Calculation operations on genetic variations file and genotype information.
    prioritization      Prioritization of genetic variations based on annotations criteria (profiles).
    process             Full genetic variations process: annotation, calculation, prioritization, format, query, filter...
    databases           Download databases and needed files for howard and associated tools
    gui                 Graphical User Interface tools
    help                Help tools
    genebe              GeneBe annotation using REST API
    minimalize          Minimalize a VCF file, such as removing INFO/Tags or samples

Usage examples:
   howard process --input=tests/data/example.vcf.gz --output=/tmp/example.annotated.vcf.gz --param=config/param.json 
   howard annotation --input=tests/data/example.vcf.gz --output=/tmp/example.howard.vcf.gz --annotations=&#39;tests/databases/annotations/current/hg19/dbnsfp42a.parquet,tests/databases/annotations/current/hg19/gnomad211_genome.parquet&#39; 
   howard calculation --input=tests/data/example.full.vcf --output=/tmp/example.calculation.tsv --calculations=&#39;vartype&#39; 
   howard prioritization --input=tests/data/example.vcf.gz --output=/tmp/example.prioritized.vcf.gz --prioritization_config=config/prioritization_profiles.json --prioritizations=&#39;default,GERMLINE&#39; 
   howard query --input=tests/data/example.vcf.gz --explode_infos --query=&#39;SELECT &quot;#CHROM&quot;, POS, REF, ALT, &quot;DP&quot;, &quot;CLNSIG&quot;, sample2, sample3 FROM variants WHERE &quot;DP&quot; &gt;= 50 OR &quot;CLNSIG&quot; NOT NULL ORDER BY &quot;CLNSIG&quot; DESC, &quot;DP&quot; DESC&#39; 
   howard stats --input=tests/data/example.vcf.gz 
   howard convert --input=tests/data/example.vcf.gz --output=/tmp/example.tsv --explode_infos &amp;&amp; cat /tmp/example.tsv </code></pre>
<p>Install HOWARD Graphical User Interface using Python Pip tool with
supplementary packages, and run as a tool:</p>
<pre><code>python -m pip install -r requirements-gui.txt
howard gui</code></pre>
<figure>
<img src="images/howard-gui.png" title="HOWARD Graphical User Interface"
alt="HOWARD Graphical User Interface" />
<figcaption aria-hidden="true">HOWARD Graphical User
Interface</figcaption>
</figure>
<h2 data-number="2.2" id="docker"><span
class="header-section-number">2.2</span> Docker</h2>
<p>In order to build, setup and create a persitent CLI (running
container with all useful external tools such as <a
href="https://samtools.github.io/bcftools/">BCFTools</a>, <a
href="https://pcingola.github.io/SnpEff/">snpEff</a>, <a
href="https://annovar.openbioinformatics.org/">Annovar</a>, <a
href="https://www.sanger.ac.uk/tool/exomiser/">Exomiser</a>),
docker-compose command build images and launch services as
containers.</p>
<pre><code>docker-compose up -d</code></pre>
<p>A setup container (HOWARD-setup) will download useful databases (take
a while). To avoid databases download (see <a
href="#databases">Databases section</a> to download manually), just
start:</p>
<pre><code>docker-compose up -d HOWARD-CLI</code></pre>
<p>A Command Line Interface container (HOWARD-CLI) is started with host
data and databases folders mounted (by default in ~/howard folder,
i.e.&#xA0;<code>~/howard/data:/data</code> and
<code>~/howard/databases:/databases</code>). Let&#x2019;s play within Docker
HOWARD-CLI service!</p>
<pre><code>docker exec -ti HOWARD-CLI bash
howard --help</code></pre>
<details>
<summary>
More details
</summary>
<p>Docker HOWARD-CLI container (Command Line Interface) can be used to
execute commands.</p>
<blockquote>
<p>Example: Query of an existing VCF</p>
<pre><code>docker exec HOWARD-CLI \
   howard query \
      --input=/tool/tests/data/example.vcf.gz \
      --query=&#39;SELECT * FROM variants&#39;</code></pre>
</blockquote>
<blockquote>
<p>Example: VCF annotation using HOWARD-CLI (snpEff and ANNOVAR
databases will be automatically downloaded), and query list of genes
with HGVS</p>
<pre><code>docker exec --workdir=/tool HOWARD-CLI \
   howard process \
      --config=config/config.json \
      --param=config/param.json \
      --input=tests/data/example.vcf.gz \
      --output=/tmp/example.process.tsv \
      --explode_infos \
      --query=&#39;SELECT &quot;NOMEN&quot;, &quot;PZFlag&quot;, &quot;PZScore&quot;, &quot;PZComment&quot; FROM variants ORDER BY &quot;PZScore&quot; DESC&#39;</code></pre>
</blockquote>
</details>
<h2 data-number="2.3" id="databases"><span
class="header-section-number">2.3</span> Databases</h2>
<p>Multiple databases can be automatically downloaded with databases
tool, such as:</p>
<table>
<colgroup>
<col style="width: 19%" />
<col style="width: 80%" />
</colgroup>
<thead>
<tr class="header">
<th>database</th>
<th>description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><a href="https://genome.ucsc.edu/cgi-bin/hgGateway">Genome</a></td>
<td>Genome Reference Consortium Human</td>
</tr>
<tr class="even">
<td><a
href="https://annovar.openbioinformatics.org/en/latest/">Annovar</a></td>
<td>ANNOVAR is an efficient software tool to utilize update-to-date
information to functionally annotate genetic variants detected from
diverse genomes</td>
</tr>
<tr class="odd">
<td><a href="https://pcingola.github.io/SnpEff/">snpEff</a></td>
<td>Genetic variant annotation, and functional effect prediction
toolbox</td>
</tr>
<tr class="even">
<td><a href="https://www.ncbi.nlm.nih.gov/refseq/">refSeq</a></td>
<td>A comprehensive, integrated, non-redundant, well-annotated set of
reference sequences including genomic, transcript, and protein</td>
</tr>
<tr class="odd">
<td><a href="https://www.ncbi.nlm.nih.gov/snp/">dbSNP</a></td>
<td>dbSNP contains human single nucleotide variations, microsatellites,
and small-scale insertions and deletions along with publication,
population frequency, molecular consequence, and genomic and RefSeq
mapping information for both common variations and clinical
mutations</td>
</tr>
<tr class="even">
<td><a
href="https://sites.google.com/site/jpopgen/dbNSFP">dbNSFP</a></td>
<td>dbNSFP is a database developed for functional prediction and
annotation of all potential non-synonymous single-nucleotide variants
(nsSNVs) in the human genome</td>
</tr>
<tr class="odd">
<td><a
href="https://github.com/google-deepmind/alphamissense">AlphaMissense</a></td>
<td>AlphaMissense model implementation</td>
</tr>
<tr class="even">
<td><a href="https://www.sanger.ac.uk/tool/exomiser/">Exomiser</a></td>
<td>The Exomiser is a Java program that finds potential disease-causing
variants from whole-exome or whole-genome sequencing data</td>
</tr>
</tbody>
</table>
<details>
<summary>
More details
</summary>
<blockquote>
<p>Example: Download Multiple databases in the same time for assembly
&#x2018;hg19&#x2019; (can take a while)</p>
<pre><code>howard databases \
   --assembly=hg19 \
   --download-genomes=~/howard/databases/genomes/current --download-genomes-provider=UCSC --download-genomes-contig-regex=&#39;chr[0-9XYM]+$&#39; \
   --download-annovar=~/howard/databases/annovar/current --download-annovar-files=&#39;refGene,cosmic70,nci60&#39; \
   --download-snpeff=~/howard/databases/snpeff/current \
   --download-refseq=~/howard/databases/refseq/current --download-refseq-format-file=&#39;ncbiRefSeq.txt&#39; \
   --download-dbnsfp=~/howard/databases/dbnsfp/current --download-dbnsfp-release=&#39;4.4a&#39; --download-dbnsfp-subdatabases \
   --download-alphamissense=~/howard/databases/alphamissense/current \
   --download-exomiser=~/howard/databases/exomiser/current \
   --download-dbsnp=~/howard/databases/dbsnp/current --download-dbsnp-vcf \
   --threads=8</code></pre>
</blockquote>
<p>See <a href="docs/help.html#databases-tool">HOWARD Help Databases
tool</a> for more information.</p>
<p>Databases can be home-made generated, starting with a existing
annotation file, especially using <a href="#convert">HOWARD convert</a>
tool. These files need to contain specific fields (depending on the
annotation type):</p>
<ul>
<li>variant annotation: &#x2018;#CHROM&#x2019;, &#x2018;POS&#x2019;, &#x2018;ALT&#x2019;, &#x2018;REF&#x2019;</li>
<li>region annotation: &#x2018;#CHROM&#x2019;, &#x2018;START&#x2019;, &#x2018;STOP&#x2019;</li>
</ul>
<p>Each database annotation file is associated with a &#x2018;header&#x2019; file
(&#x2018;.hdr&#x2019;), in VCF header format, to describe annotations within the
database.</p>
</details>
<h2 data-number="2.4" id="configuration"><span
class="header-section-number">2.4</span> Configuration</h2>
<p>HOWARD Configuration JSON file defined default configuration
regarding resources (e.g.&#xA0;threads, memory), settings (e.g.&#xA0;verbosity,
temporary files), default folders (e.g.&#xA0;for databases) and paths to
external tools.</p>
<p>See <a href="docs/help.config.html">HOWARD Configuration JSON</a> for
more information.</p>
<h1 data-number="3" id="tools"><span
class="header-section-number">3</span> Tools</h1>
<p>HOWARD annotates and prioritizes genetic variations, calculates and
normalizes annotations, convert on multiple formats, query variations
and generates statistics. These tools require options or a <a
href="help.param.html">Parameters JSON</a> file.</p>
<h2 data-number="3.1" id="parameters"><span
class="header-section-number">3.1</span> Parameters</h2>
<p>HOWARD Parameters JSON file defined parameters to process
annotations, prioritization, calculations, convertions and queries. Use
this parameters file to configure tools, instead of options or as a main
configuration (options will replace parameters in JSON file).</p>
<p>See <a href="docs/help.param.html">HOWARD Parameters JSON</a> for
more information.</p>
<h2 data-number="3.2" id="stats"><span
class="header-section-number">3.2</span> Stats</h2>
<p>Statistics on genetic variations, such as: number of variants, number
of samples, statistics by chromosome, genotypes by samples, annotations.
Theses statsitics can be applied to VCF files and all database
annotation files.</p>
<details>
<summary>
More details
</summary>
<blockquote>
<p>Example: Show example VCF statistics and brief overview</p>
<pre><code>howard stats \
   --input=tests/data/example.vcf.gz</code></pre>
</blockquote>
<p>See <a href="docs/help.html#stats-tool">HOWARD Help Stats tool</a>
for more information.</p>
</details>
<h2 data-number="3.3" id="convert"><span
class="header-section-number">3.3</span> Convert</h2>
<p>Convert genetic variations file to another format. Multiple format
are available, such as usual and official VCF format, but also other
formats such as TSV, CSV, TBL, JSON and Parquet/duckDB. These formats
need a header &#x2018;.hdr&#x2019; file to take advantage of the power of howard
(especially through INFO/tag definition), and using howard convert tool
automatically generate header file fo futher use (otherwise, an default
&#x2018;.hdr&#x2019; file is generated).</p>
<details>
<summary>
More details
</summary>
<blockquote>
<p>Example: Translate VCF into TSV, export INFO/tags into columns, and
show output file</p>
<pre><code>howard convert \
   --input=tests/data/example.vcf.gz \
   --explode_infos \
   --output=/tmp/example.tsv
cat /tmp/example.tsv</code></pre>
</blockquote>
<p>See <a href="docs/help.html#convert-tool">HOWARD Help Convert
tool</a> for more options.</p>
</details>
<h2 data-number="3.4" id="query"><span
class="header-section-number">3.4</span> Query</h2>
<p>Query genetic variations in SQL format. Data can be loaded into
&#x2018;variants&#x2019; table from various formats (e.g.&#xA0;VCF, TSV, Parquet&#x2026;). Using
&#x2018;explode&#x2019; option allows querying on INFO/tag annotations. SQL query can
also use external data within the request, such as a Parquet
file(s).</p>
<details>
<summary>
More details
</summary>
<blockquote>
<p>Example: Select variants in VCF with INFO Tags criterions</p>
<pre><code>howard query \
   --input=tests/data/example.vcf.gz \
   --explode_infos \
   --query=&#39;SELECT &quot;#CHROM&quot;, POS, REF, ALT, DP, CLNSIG, sample2, sample3 
            FROM variants 
            WHERE DP &gt;= 50 OR CLNSIG NOT NULL 
            ORDER BY CLNSIG DESC, DP DESC&#39;</code></pre>
</blockquote>
<p>See <a href="docs/help.html#query-tool">HOWARD Help Query tool</a>
for more options.</p>
</details>
<h2 data-number="3.5" id="annotation"><span
class="header-section-number">3.5</span> Annotation</h2>
<p>Annotation is mainly based on a build-in Parquet annotation method,
using database format such as Parquet, duckdb, VCF, BED, TSV, JSON.
External annotation tools are also available, such as BCFTOOLS, Annovar,
snpEff, Exomiser and Splice. It uses available databases and homemade
databases. Annovar and snpEff databases are automatically downloaded
(see <a href="docs/help.html#databases-tool">HOWARD Help Databases
tool</a>). All annotation parameters are defined in <a
href="docs/help.param.html">HOWARD Parameters JSON</a> file.</p>
<p>Quick annotation allows to annotates by simply listing annotation
databases, or defining external tools keywords. These annotations can be
combined.</p>
<details>
<summary>
More details
</summary>
<blockquote>
<p>Example: VCF annotation with Parquet and VCF databases, output as VCF
format</p>
<pre><code>howard annotation \
   --input=tests/data/example.vcf.gz \
   --annotations=&#39;tests/databases/annotations/current/hg19/dbnsfp42a.parquet,tests/databases/annotations/current/hg19/cosmic70.vcf.gz&#39; \
   --output=/tmp/example.howard.vcf.gz</code></pre>
</blockquote>
<blockquote>
<p>Example: VCF annotation with external tools (Annovar refGene and
snpEff databases), output as TSV format</p>
<pre><code>howard annotation \
   --input=tests/data/example.vcf.gz \
   --annotations=&#39;annovar:refGene,snpeff&#39; \
   --output=/tmp/example.howard.tsv</code></pre>
</blockquote>
<p>See <a href="docs/help.html#annotation-tool">HOWARD Help Annotation
tool</a> for more options.</p>
</details>
<h2 data-number="3.6" id="calculation"><span
class="header-section-number">3.6</span> Calculation</h2>
<p>Calculation processes variants information to generate new
information, such as: identify variation type (VarType), harmonizes
allele frequency (VAF) and calculate sttistics (VAF_stats), extracts
Nomen (transcript, cNomen, pNomen&#x2026;) from an HGVS field (e.g.&#xA0;snpEff,
Annovar) with an optional list of personalized transcripts, generates
VaRank format barcode, identify trio inheritance.</p>
<details>
<summary>
More details
</summary>
<blockquote>
<p>Example: Identify variant types and generate a table of variant type
count</p>
<pre><code>howard calculation \
   --input=tests/data/example.full.vcf \
   --calculations=&#39;vartype&#39; \
   --output=/tmp/example.calculation.tsv

howard query \
   --input=/tmp/example.calculation.tsv \
   --explode_infos \
   --query=&#39;SELECT
               &quot;VARTYPE&quot; AS &#39;VariantType&#39;,
               count(*) AS &#39;Count&#39;
            FROM variants
            GROUP BY &quot;VARTYPE&quot;
            ORDER BY count DESC&#39;</code></pre>
<pre><code>  VariantType  Count
0         BND      7
1         DUP      6
2         INS      5
3         SNV      4
4         CNV      3
5         DEL      3
6         INV      3
7      MOSAIC      2
8       INDEL      2
9         MNV      1</code></pre>
</blockquote>
<p>See <a href="docs/help.html#calculation-tool">HOWARD Help Calculation
tool</a> for more options.</p>
</details>
<h2 data-number="3.7" id="prioritization"><span
class="header-section-number">3.7</span> Prioritization</h2>
<p>Prioritization algorithm uses profiles to flag variants (as passed or
filtered), calculate a prioritization score, and automatically generate
a comment for each variants (example: &#x2018;polymorphism identified in dbSNP.
associated to Lung Cancer. Found in ClinVar database&#x2019;). Prioritization
profiles are defined in a configuration file in JSON format. A profile
is defined as a list of annotation/value, using wildcards and comparison
options (contains, lower than, greater than, equal&#x2026;). Annotations fields
may be quality values (usually from callers, such as &#x2018;DP&#x2019;) or other
annotations fields provided by annotations tools, such as HOWARD itself
(example: COSMIC, Clinvar, 1000genomes, PolyPhen, SIFT).</p>
<p>Multiple profiles can be used simultaneously, which is useful to
define multiple validation/prioritization levels (e.g.&#xA0;&#x2018;standard&#x2019;,
&#x2018;stringent&#x2019;, &#x2018;rare variants&#x2019;). Prioritization score can be calculated
following multiple mode, either &#x2018;HOWARD&#x2019; (incremental) or &#x2018;VaRank&#x2019;
(maximum). Prioritization fields can be selected (PZScore, PZFlag,
PZComment, PZTags, PZInfos).</p>
<details>
<summary>
More details
</summary>
<blockquote>
<p>Example: Prioritize variants from criteria on INFO annotations for
profiles &#x2018;default&#x2019; and &#x2018;GERMLINE&#x2019; (from &#x2018;prioritization_profiles.json&#x2019;
profiles configuration), export prioritization tags, and query variants
passing filters</p>
<pre><code>howard prioritization \
   --input=tests/data/example.vcf.gz \
   --prioritizations=config/prioritization_profiles.json \
   --profiles=&#39;default,GERMLINE&#39; \
   --default_profile=&#39;default&#39; \
   --pzfields=&#39;PZFlag,PZScore,PZComment,PZTags,PZInfos&#39; \
   --prioritization_score_mode=&#39;HOWARD&#39; \
   --output=/tmp/example.prioritized.vcf.gz

howard query \
   --input=/tmp/example.prioritized.vcf.gz \
   --explode_infos \
   --query=&quot;SELECT \&quot;#CHROM\&quot;, POS, ALT, REF, \&quot;PZFlag\&quot;, \&quot;PZScore\&quot;, \&quot;PZTags\&quot;, \&quot;DP\&quot;, \&quot;CLNSIG\&quot; \
            FROM variants \
            WHERE \&quot;PZScore\&quot; &gt; 0 \
              AND \&quot;PZFlag\&quot; == &#39;PASS&#39; \
            ORDER BY \&quot;PZScore\&quot; DESC&quot;</code></pre>
<pre><code>  #CHROM       POS ALT REF PZFlag  PZScore                                             PZTags     DP      CLNSIG
0   chr1     28736   C   A   PASS       15  PZFlag#PASS|PZScore#15|PZComment#Described on ...    NaN  pathogenic
1   chr1     69101   G   A   PASS        5  PZFlag#PASS|PZScore#5|PZComment#DP higher than 50   50.0        None
2   chr7  55249063   A   G   PASS        5  PZFlag#PASS|PZScore#5|PZComment#DP higher than 50  125.0        None</code></pre>
</blockquote>
<p>See <a href="docs/help.html#prioritization-tool">HOWARD Help
Prioritization tool</a> for more options.</p>
</details>
<h2 data-number="3.8" id="hgvs-annotation"><span
class="header-section-number">3.8</span> HGVS Annotation</h2>
<p>HOWARD annotates variants with HGVS annotation using HUGO HGVS
internation Sequence Variant Nomenclature (http://varnomen.hgvs.org/).
Annotation refere to refGene and genome to generate HGVS nomenclature
for all available transcripts. This annotation add &#x2018;hgvs&#x2019; field into VCF
INFO column of a VCF file. Several options are available, to add gene,
exon and protein information, to generate a &#x201C;full format&#x201D; detailed
annotation, to choose codon format.</p>
<p>See <a href="docs/help.html#hgvs-tool">HOWARD Help HGVS tool</a> for
more options.</p>
<details>
<summary>
More details
</summary>
<blockquote>
<p>Example: HGVS annotation with quick options</p>
<pre><code>howard hgvs \
   --input=tests/data/example.vcf.gz \
   --output=/tmp/example.process.tsv \
   --hgvs=full_format,use_exon

howard query \
   --input=/tmp/example.process.tsv \
   --explode_infos \
   --query=&quot;SELECT hgvs \
            FROM variants &quot;</code></pre>
<pre><code>                                                hgvs
0                     WASH7P:NR_024540.1:n.50+585T&gt;G
1     FAM138A:NR_026818.1:exon3:n.597T&gt;G:p.Tyr199Asp
2  OR4F5:NM_001005484.2:NP_001005484.2:exon3:c.74...
3  LINC01128:NR_047526.1:n.287+3767A&gt;G,LINC01128:...
4  LINC01128:NR_047526.1:n.287+3768A&gt;G,LINC01128:...
5  LINC01128:NR_047526.1:n.287+3769A&gt;G,LINC01128:...
6  EGFR:NM_001346897.2:NP_001333826.1:exon19:c.22...</code></pre>
</blockquote>
</details>
<h2 data-number="3.9" id="process"><span
class="header-section-number">3.9</span> Process</h2>
<p>HOWARD process tool manage genetic variations to:</p>
<ul>
<li>annotates genetic variants with multiple annotation databases/files
and tools</li>
<li>calculates and normalizes annotations</li>
<li>prioritizes variants with profiles (list of citeria) to calculate
scores and flags</li>
<li>annotates genetic variants with HGVS nomenclature</li>
<li>translates into various formats</li>
<li>query genetic variants and annotations</li>
<li>generates variants statistics</li>
</ul>
<p>This process tool combines all other tools to pipe them in a uniq
command, through available options or a parameters file in JSON format
(see <a href="docs/help.param.html">HOWARD Parameters JSON</a>
file).</p>
<p>See <a href="docs/help.html#process-tool">HOWARD Help Process
tool</a> tool for more information.</p>
<details>
<summary>
More details
</summary>
<blockquote>
<p>Example: Full process command with options (HGVS, annotation,
calculation and prioritization)</p>
<pre><code>howard process \
   --input=tests/data/example.vcf.gz \
   --output=/tmp/example.process.tsv \
   --hgvs=full_format,use_exon \
   --annotations=&quot;tests/databases/annotations/current/hg19/avsnp150.parquet,tests/databases/annotations/current/hg19/dbnsfp42a.parquet,tests/databases/annotations/current/hg19/gnomad211_genome.parquet,bcftools:tests/databases/annotations/current/hg19/cosmic70.vcf.gz,snpeff,annovar:refGene&quot; \
   --calculations=&quot;vartype,snpeff_hgvs,VAF,NOMEN&quot; \
   --prioritizations=&quot;config/prioritization_profiles.json&quot; \
   --explode_infos \
   --query=&quot;SELECT NOMEN, PZFlag, PZScore, snpeff_hgvs \
            FROM variants \
            ORDER BY PZScore DESC&quot;</code></pre>
<pre><code>                                            NOMEN    PZFlag  PZScore                                        snpeff_hgvs
0                    WASH7P:NR_024540:n.50+585T&gt;G      PASS       15  MIR1302-2:NR_036051.1:n.-1630A&gt;C,MIR1302-9:NR_...
1     OR4F5:NP_001005484:exon3:c.74A&gt;G:p.Glu25Gly      PASS        5       OR4F5:NM_001005484.1:exon1:c.11A&gt;G:p.Glu4Gly
2  EGFR:NM_001346897:exon19:c.2226G&gt;A:p.Gln742Gln      PASS        5  EGFR:NM_005228.5:exon20:c.2361G&gt;A:p.Gln787Gln,...
3               LINC01128:NR_047526:n.287+3767A&gt;G      PASS        0  LINC01128:NR_047519.1:exon2:n.287+3767A&gt;G,LINC...
4               LINC01128:NR_047526:n.287+3768A&gt;G      PASS        0  LINC01128:NR_047519.1:exon2:n.287+3768A&gt;G,LINC...
5               LINC01128:NR_047526:n.287+3769A&gt;G      PASS        0  LINC01128:NR_047519.1:exon2:n.287+3769A&gt;G,LINC...
6    FAM138A:NR_026818:exon3:n.597T&gt;G:p.Tyr199Asp  FILTERED     -100  MIR1302-2:NR_036051.1:n.*4641A&gt;C,MIR1302-9:NR_...</code></pre>
</blockquote>
</details>
<h1 data-number="4" id="documentation"><span
class="header-section-number">4</span> Documentation</h1>
<p><a href="docs/user_guide.html">HOWARD User Guide</a> is available to
assist users for particular commands, such as software installation,
databases download, annotation command, and so on.</p>
<p><a href="docs/tips.html">HOWARD Tips</a> proposes some additional
advices to handle HOWARD for particular use cases.</p>
<p><a href="docs/options.html">HOWARD Help</a> describes options of all
HOWARD tools. All information are also available for each tool using
<code>--help</code> option.</p>
<p><a href="docs/help.config.html">HOWARD Configuration JSON</a>
describes configuration JSON file structure and options.</p>
<p><a href="docs/help.param.html">HOWARD Parameters JSON</a> describes
parameters JSON file structure and options.</p>
<p><a href="docs/help.param.databases.html">HOWARD Parameters Databases
JSON</a> describes configuration JSON file for databases download and
convert.</p>
<p><a href="plugins/README.html">HOWARD Plugins</a> describes how to
create HOWARD plugins.</p>
<p><a href="docs/pdoc/index.html">HOWARD Package</a> describes HOWARD
Package, Classes and Functions.</p>
<h1 data-number="5" id="contact"><span
class="header-section-number">5</span> Contact</h1>
<p><a
href="https://www.chru-strasbourg.fr/service/bioinformatique-medicale-appliquee-au-diagnostic-unite-de/">Medical
Bioinformatics applied to Diagnosis Lab</a> @ Strasbourg Univerty
Hospital</p>
<p><a
href="bioinfo@chru-strasbourg.fr">bioinfo@chru-strasbourg.fr</a></p>
<p><a href="https://github.com/bioinfo-chru-strasbourg">GitHub</a></p>
</body>
</html>
