<!doctype html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="generator" content="pdoc 14.5.1"/>
    <title>howard.objects.hgvs API documentation</title>

    <style>/*! * Bootstrap Reboot v5.0.0 (https://getbootstrap.com/) * Copyright 2011-2021 The Bootstrap Authors * Copyright 2011-2021 Twitter, Inc. * Licensed under MIT (https://github.com/twbs/bootstrap/blob/main/LICENSE) * Forked from Normalize.css, licensed MIT (https://github.com/necolas/normalize.css/blob/master/LICENSE.md) */*,::after,::before{box-sizing:border-box}@media (prefers-reduced-motion:no-preference){:root{scroll-behavior:smooth}}body{margin:0;font-family:system-ui,-apple-system,"Segoe UI",Roboto,"Helvetica Neue",Arial,"Noto Sans","Liberation Sans",sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol","Noto Color Emoji";font-size:1rem;font-weight:400;line-height:1.5;color:#212529;background-color:#fff;-webkit-text-size-adjust:100%;-webkit-tap-highlight-color:transparent}hr{margin:1rem 0;color:inherit;background-color:currentColor;border:0;opacity:.25}hr:not([size]){height:1px}h1,h2,h3,h4,h5,h6{margin-top:0;margin-bottom:.5rem;font-weight:500;line-height:1.2}h1{font-size:calc(1.375rem + 1.5vw)}@media (min-width:1200px){h1{font-size:2.5rem}}h2{font-size:calc(1.325rem + .9vw)}@media (min-width:1200px){h2{font-size:2rem}}h3{font-size:calc(1.3rem + .6vw)}@media (min-width:1200px){h3{font-size:1.75rem}}h4{font-size:calc(1.275rem + .3vw)}@media (min-width:1200px){h4{font-size:1.5rem}}h5{font-size:1.25rem}h6{font-size:1rem}p{margin-top:0;margin-bottom:1rem}abbr[data-bs-original-title],abbr[title]{-webkit-text-decoration:underline dotted;text-decoration:underline dotted;cursor:help;-webkit-text-decoration-skip-ink:none;text-decoration-skip-ink:none}address{margin-bottom:1rem;font-style:normal;line-height:inherit}ol,ul{padding-left:2rem}dl,ol,ul{margin-top:0;margin-bottom:1rem}ol ol,ol ul,ul ol,ul ul{margin-bottom:0}dt{font-weight:700}dd{margin-bottom:.5rem;margin-left:0}blockquote{margin:0 0 1rem}b,strong{font-weight:bolder}small{font-size:.875em}mark{padding:.2em;background-color:#fcf8e3}sub,sup{position:relative;font-size:.75em;line-height:0;vertical-align:baseline}sub{bottom:-.25em}sup{top:-.5em}a{color:#0d6efd;text-decoration:underline}a:hover{color:#0a58ca}a:not([href]):not([class]),a:not([href]):not([class]):hover{color:inherit;text-decoration:none}code,kbd,pre,samp{font-family:SFMono-Regular,Menlo,Monaco,Consolas,"Liberation Mono","Courier New",monospace;font-size:1em;direction:ltr;unicode-bidi:bidi-override}pre{display:block;margin-top:0;margin-bottom:1rem;overflow:auto;font-size:.875em}pre code{font-size:inherit;color:inherit;word-break:normal}code{font-size:.875em;color:#d63384;word-wrap:break-word}a>code{color:inherit}kbd{padding:.2rem .4rem;font-size:.875em;color:#fff;background-color:#212529;border-radius:.2rem}kbd kbd{padding:0;font-size:1em;font-weight:700}figure{margin:0 0 1rem}img,svg{vertical-align:middle}table{caption-side:bottom;border-collapse:collapse}caption{padding-top:.5rem;padding-bottom:.5rem;color:#6c757d;text-align:left}th{text-align:inherit;text-align:-webkit-match-parent}tbody,td,tfoot,th,thead,tr{border-color:inherit;border-style:solid;border-width:0}label{display:inline-block}button{border-radius:0}button:focus:not(:focus-visible){outline:0}button,input,optgroup,select,textarea{margin:0;font-family:inherit;font-size:inherit;line-height:inherit}button,select{text-transform:none}[role=button]{cursor:pointer}select{word-wrap:normal}select:disabled{opacity:1}[list]::-webkit-calendar-picker-indicator{display:none}[type=button],[type=reset],[type=submit],button{-webkit-appearance:button}[type=button]:not(:disabled),[type=reset]:not(:disabled),[type=submit]:not(:disabled),button:not(:disabled){cursor:pointer}::-moz-focus-inner{padding:0;border-style:none}textarea{resize:vertical}fieldset{min-width:0;padding:0;margin:0;border:0}legend{float:left;width:100%;padding:0;margin-bottom:.5rem;font-size:calc(1.275rem + .3vw);line-height:inherit}@media (min-width:1200px){legend{font-size:1.5rem}}legend+*{clear:left}::-webkit-datetime-edit-day-field,::-webkit-datetime-edit-fields-wrapper,::-webkit-datetime-edit-hour-field,::-webkit-datetime-edit-minute,::-webkit-datetime-edit-month-field,::-webkit-datetime-edit-text,::-webkit-datetime-edit-year-field{padding:0}::-webkit-inner-spin-button{height:auto}[type=search]{outline-offset:-2px;-webkit-appearance:textfield}::-webkit-search-decoration{-webkit-appearance:none}::-webkit-color-swatch-wrapper{padding:0}::file-selector-button{font:inherit}::-webkit-file-upload-button{font:inherit;-webkit-appearance:button}output{display:inline-block}iframe{border:0}summary{display:list-item;cursor:pointer}progress{vertical-align:baseline}[hidden]{display:none!important}</style>
    <style>/*! syntax-highlighting.css */pre{line-height:125%;}span.linenos{color:inherit; background-color:transparent; padding-left:5px; padding-right:20px;}.pdoc-code .hll{background-color:#ffffcc}.pdoc-code{background:#f8f8f8;}.pdoc-code .c{color:#3D7B7B; font-style:italic}.pdoc-code .err{border:1px solid #FF0000}.pdoc-code .k{color:#008000; font-weight:bold}.pdoc-code .o{color:#666666}.pdoc-code .ch{color:#3D7B7B; font-style:italic}.pdoc-code .cm{color:#3D7B7B; font-style:italic}.pdoc-code .cp{color:#9C6500}.pdoc-code .cpf{color:#3D7B7B; font-style:italic}.pdoc-code .c1{color:#3D7B7B; font-style:italic}.pdoc-code .cs{color:#3D7B7B; font-style:italic}.pdoc-code .gd{color:#A00000}.pdoc-code .ge{font-style:italic}.pdoc-code .gr{color:#E40000}.pdoc-code .gh{color:#000080; font-weight:bold}.pdoc-code .gi{color:#008400}.pdoc-code .go{color:#717171}.pdoc-code .gp{color:#000080; font-weight:bold}.pdoc-code .gs{font-weight:bold}.pdoc-code .gu{color:#800080; font-weight:bold}.pdoc-code .gt{color:#0044DD}.pdoc-code .kc{color:#008000; font-weight:bold}.pdoc-code .kd{color:#008000; font-weight:bold}.pdoc-code .kn{color:#008000; font-weight:bold}.pdoc-code .kp{color:#008000}.pdoc-code .kr{color:#008000; font-weight:bold}.pdoc-code .kt{color:#B00040}.pdoc-code .m{color:#666666}.pdoc-code .s{color:#BA2121}.pdoc-code .na{color:#687822}.pdoc-code .nb{color:#008000}.pdoc-code .nc{color:#0000FF; font-weight:bold}.pdoc-code .no{color:#880000}.pdoc-code .nd{color:#AA22FF}.pdoc-code .ni{color:#717171; font-weight:bold}.pdoc-code .ne{color:#CB3F38; font-weight:bold}.pdoc-code .nf{color:#0000FF}.pdoc-code .nl{color:#767600}.pdoc-code .nn{color:#0000FF; font-weight:bold}.pdoc-code .nt{color:#008000; font-weight:bold}.pdoc-code .nv{color:#19177C}.pdoc-code .ow{color:#AA22FF; font-weight:bold}.pdoc-code .w{color:#bbbbbb}.pdoc-code .mb{color:#666666}.pdoc-code .mf{color:#666666}.pdoc-code .mh{color:#666666}.pdoc-code .mi{color:#666666}.pdoc-code .mo{color:#666666}.pdoc-code .sa{color:#BA2121}.pdoc-code .sb{color:#BA2121}.pdoc-code .sc{color:#BA2121}.pdoc-code .dl{color:#BA2121}.pdoc-code .sd{color:#BA2121; font-style:italic}.pdoc-code .s2{color:#BA2121}.pdoc-code .se{color:#AA5D1F; font-weight:bold}.pdoc-code .sh{color:#BA2121}.pdoc-code .si{color:#A45A77; font-weight:bold}.pdoc-code .sx{color:#008000}.pdoc-code .sr{color:#A45A77}.pdoc-code .s1{color:#BA2121}.pdoc-code .ss{color:#19177C}.pdoc-code .bp{color:#008000}.pdoc-code .fm{color:#0000FF}.pdoc-code .vc{color:#19177C}.pdoc-code .vg{color:#19177C}.pdoc-code .vi{color:#19177C}.pdoc-code .vm{color:#19177C}.pdoc-code .il{color:#666666}</style>
    <style>/*! theme.css */:root{--pdoc-background:#fff;}.pdoc{--text:#212529;--muted:#6c757d;--link:#3660a5;--link-hover:#1659c5;--code:#f8f8f8;--active:#fff598;--accent:#eee;--accent2:#c1c1c1;--nav-hover:rgba(255, 255, 255, 0.5);--name:#0066BB;--def:#008800;--annotation:#007020;}</style>
    <style>/*! layout.css */html, body{width:100%;height:100%;}html, main{scroll-behavior:smooth;}body{background-color:var(--pdoc-background);}@media (max-width:769px){#navtoggle{cursor:pointer;position:absolute;width:50px;height:40px;top:1rem;right:1rem;border-color:var(--text);color:var(--text);display:flex;opacity:0.8;z-index:999;}#navtoggle:hover{opacity:1;}#togglestate + div{display:none;}#togglestate:checked + div{display:inherit;}main, header{padding:2rem 3vw;}header + main{margin-top:-3rem;}.git-button{display:none !important;}nav input[type="search"]{max-width:77%;}nav input[type="search"]:first-child{margin-top:-6px;}nav input[type="search"]:valid ~ *{display:none !important;}}@media (min-width:770px){:root{--sidebar-width:clamp(12.5rem, 28vw, 22rem);}nav{position:fixed;overflow:auto;height:100vh;width:var(--sidebar-width);}main, header{padding:3rem 2rem 3rem calc(var(--sidebar-width) + 3rem);width:calc(54rem + var(--sidebar-width));max-width:100%;}header + main{margin-top:-4rem;}#navtoggle{display:none;}}#togglestate{position:absolute;height:0;opacity:0;}nav.pdoc{--pad:clamp(0.5rem, 2vw, 1.75rem);--indent:1.5rem;background-color:var(--accent);border-right:1px solid var(--accent2);box-shadow:0 0 20px rgba(50, 50, 50, .2) inset;padding:0 0 0 var(--pad);overflow-wrap:anywhere;scrollbar-width:thin; scrollbar-color:var(--accent2) transparent; z-index:1}nav.pdoc::-webkit-scrollbar{width:.4rem; }nav.pdoc::-webkit-scrollbar-thumb{background-color:var(--accent2); }nav.pdoc > div{padding:var(--pad) 0;}nav.pdoc .module-list-button{display:inline-flex;align-items:center;color:var(--text);border-color:var(--muted);margin-bottom:1rem;}nav.pdoc .module-list-button:hover{border-color:var(--text);}nav.pdoc input[type=search]{display:block;outline-offset:0;width:calc(100% - var(--pad));}nav.pdoc .logo{max-width:calc(100% - var(--pad));max-height:35vh;display:block;margin:0 auto 1rem;transform:translate(calc(-.5 * var(--pad)), 0);}nav.pdoc ul{list-style:none;padding-left:0;}nav.pdoc > div > ul{margin-left:calc(0px - var(--pad));}nav.pdoc li a{padding:.2rem 0 .2rem calc(var(--pad) + var(--indent));}nav.pdoc > div > ul > li > a{padding-left:var(--pad);}nav.pdoc li{transition:all 100ms;}nav.pdoc li:hover{background-color:var(--nav-hover);}nav.pdoc a, nav.pdoc a:hover{color:var(--text);}nav.pdoc a{display:block;}nav.pdoc > h2:first-of-type{margin-top:1.5rem;}nav.pdoc .class:before{content:"class ";color:var(--muted);}nav.pdoc .function:after{content:"()";color:var(--muted);}nav.pdoc footer:before{content:"";display:block;width:calc(100% - var(--pad));border-top:solid var(--accent2) 1px;margin-top:1.5rem;padding-top:.5rem;}nav.pdoc footer{font-size:small;}</style>
    <style>/*! content.css */.pdoc{color:var(--text);box-sizing:border-box;line-height:1.5;background:none;}.pdoc .pdoc-button{cursor:pointer;display:inline-block;border:solid black 1px;border-radius:2px;font-size:.75rem;padding:calc(0.5em - 1px) 1em;transition:100ms all;}.pdoc .pdoc-alert{padding:1rem 1rem 1rem calc(1.5rem + 24px);border:1px solid transparent;border-radius:.25rem;background-repeat:no-repeat;background-position:1rem center;margin-bottom:1rem;}.pdoc .pdoc-alert > *:last-child{margin-bottom:0;}.pdoc .pdoc-alert-note {color:#084298;background-color:#cfe2ff;border-color:#b6d4fe;background-image:url("data:image/svg+xml,%3Csvg%20xmlns%3D%22http%3A//www.w3.org/2000/svg%22%20width%3D%2224%22%20height%3D%2224%22%20fill%3D%22%23084298%22%20viewBox%3D%220%200%2016%2016%22%3E%3Cpath%20d%3D%22M8%2016A8%208%200%201%200%208%200a8%208%200%200%200%200%2016zm.93-9.412-1%204.705c-.07.34.029.533.304.533.194%200%20.487-.07.686-.246l-.088.416c-.287.346-.92.598-1.465.598-.703%200-1.002-.422-.808-1.319l.738-3.468c.064-.293.006-.399-.287-.47l-.451-.081.082-.381%202.29-.287zM8%205.5a1%201%200%201%201%200-2%201%201%200%200%201%200%202z%22/%3E%3C/svg%3E");}.pdoc .pdoc-alert-warning{color:#664d03;background-color:#fff3cd;border-color:#ffecb5;background-image:url("data:image/svg+xml,%3Csvg%20xmlns%3D%22http%3A//www.w3.org/2000/svg%22%20width%3D%2224%22%20height%3D%2224%22%20fill%3D%22%23664d03%22%20viewBox%3D%220%200%2016%2016%22%3E%3Cpath%20d%3D%22M8.982%201.566a1.13%201.13%200%200%200-1.96%200L.165%2013.233c-.457.778.091%201.767.98%201.767h13.713c.889%200%201.438-.99.98-1.767L8.982%201.566zM8%205c.535%200%20.954.462.9.995l-.35%203.507a.552.552%200%200%201-1.1%200L7.1%205.995A.905.905%200%200%201%208%205zm.002%206a1%201%200%201%201%200%202%201%201%200%200%201%200-2z%22/%3E%3C/svg%3E");}.pdoc .pdoc-alert-danger{color:#842029;background-color:#f8d7da;border-color:#f5c2c7;background-image:url("data:image/svg+xml,%3Csvg%20xmlns%3D%22http%3A//www.w3.org/2000/svg%22%20width%3D%2224%22%20height%3D%2224%22%20fill%3D%22%23842029%22%20viewBox%3D%220%200%2016%2016%22%3E%3Cpath%20d%3D%22M5.52.359A.5.5%200%200%201%206%200h4a.5.5%200%200%201%20.474.658L8.694%206H12.5a.5.5%200%200%201%20.395.807l-7%209a.5.5%200%200%201-.873-.454L6.823%209.5H3.5a.5.5%200%200%201-.48-.641l2.5-8.5z%22/%3E%3C/svg%3E");}.pdoc .visually-hidden{position:absolute !important;width:1px !important;height:1px !important;padding:0 !important;margin:-1px !important;overflow:hidden !important;clip:rect(0, 0, 0, 0) !important;white-space:nowrap !important;border:0 !important;}.pdoc h1, .pdoc h2, .pdoc h3{font-weight:300;margin:.3em 0;padding:.2em 0;}.pdoc > section:not(.module-info) h1{font-size:1.5rem;font-weight:500;}.pdoc > section:not(.module-info) h2{font-size:1.4rem;font-weight:500;}.pdoc > section:not(.module-info) h3{font-size:1.3rem;font-weight:500;}.pdoc > section:not(.module-info) h4{font-size:1.2rem;}.pdoc > section:not(.module-info) h5{font-size:1.1rem;}.pdoc a{text-decoration:none;color:var(--link);}.pdoc a:hover{color:var(--link-hover);}.pdoc blockquote{margin-left:2rem;}.pdoc pre{border-top:1px solid var(--accent2);border-bottom:1px solid var(--accent2);margin-top:0;margin-bottom:1em;padding:.5rem 0 .5rem .5rem;overflow-x:auto;background-color:var(--code);}.pdoc code{color:var(--text);padding:.2em .4em;margin:0;font-size:85%;background-color:var(--accent);border-radius:6px;}.pdoc a > code{color:inherit;}.pdoc pre > code{display:inline-block;font-size:inherit;background:none;border:none;padding:0;}.pdoc > section:not(.module-info){margin-bottom:1.5rem;}.pdoc .modulename{margin-top:0;font-weight:bold;}.pdoc .modulename a{color:var(--link);transition:100ms all;}.pdoc .git-button{float:right;border:solid var(--link) 1px;}.pdoc .git-button:hover{background-color:var(--link);color:var(--pdoc-background);}.view-source-toggle-state,.view-source-toggle-state ~ .pdoc-code{display:none;}.view-source-toggle-state:checked ~ .pdoc-code{display:block;}.view-source-button{display:inline-block;float:right;font-size:.75rem;line-height:1.5rem;color:var(--muted);padding:0 .4rem 0 1.3rem;cursor:pointer;text-indent:-2px;}.view-source-button > span{visibility:hidden;}.module-info .view-source-button{float:none;display:flex;justify-content:flex-end;margin:-1.2rem .4rem -.2rem 0;}.view-source-button::before{position:absolute;content:"View Source";display:list-item;list-style-type:disclosure-closed;}.view-source-toggle-state:checked ~ .attr .view-source-button::before,.view-source-toggle-state:checked ~ .view-source-button::before{list-style-type:disclosure-open;}.pdoc .docstring{margin-bottom:1.5rem;}.pdoc section:not(.module-info) .docstring{margin-left:clamp(0rem, 5vw - 2rem, 1rem);}.pdoc .docstring .pdoc-code{margin-left:1em;margin-right:1em;}.pdoc h1:target,.pdoc h2:target,.pdoc h3:target,.pdoc h4:target,.pdoc h5:target,.pdoc h6:target,.pdoc .pdoc-code > pre > span:target{background-color:var(--active);box-shadow:-1rem 0 0 0 var(--active);}.pdoc .pdoc-code > pre > span:target{display:block;}.pdoc div:target > .attr,.pdoc section:target > .attr,.pdoc dd:target > a{background-color:var(--active);}.pdoc *{scroll-margin:2rem;}.pdoc .pdoc-code .linenos{user-select:none;}.pdoc .attr:hover{filter:contrast(0.95);}.pdoc section, .pdoc .classattr{position:relative;}.pdoc .headerlink{--width:clamp(1rem, 3vw, 2rem);position:absolute;top:0;left:calc(0rem - var(--width));transition:all 100ms ease-in-out;opacity:0;}.pdoc .headerlink::before{content:"#";display:block;text-align:center;width:var(--width);height:2.3rem;line-height:2.3rem;font-size:1.5rem;}.pdoc .attr:hover ~ .headerlink,.pdoc *:target > .headerlink,.pdoc .headerlink:hover{opacity:1;}.pdoc .attr{display:block;margin:.5rem 0 .5rem;padding:.4rem .4rem .4rem 1rem;background-color:var(--accent);overflow-x:auto;}.pdoc .classattr{margin-left:2rem;}.pdoc .name{color:var(--name);font-weight:bold;}.pdoc .def{color:var(--def);font-weight:bold;}.pdoc .signature{background-color:transparent;}.pdoc .param, .pdoc .return-annotation{white-space:pre;}.pdoc .signature.multiline .param{display:block;}.pdoc .signature.condensed .param{display:inline-block;}.pdoc .annotation{color:var(--annotation);}.pdoc .view-value-toggle-state,.pdoc .view-value-toggle-state ~ .default_value{display:none;}.pdoc .view-value-toggle-state:checked ~ .default_value{display:inherit;}.pdoc .view-value-button{font-size:.5rem;vertical-align:middle;border-style:dashed;margin-top:-0.1rem;}.pdoc .view-value-button:hover{background:white;}.pdoc .view-value-button::before{content:"show";text-align:center;width:2.2em;display:inline-block;}.pdoc .view-value-toggle-state:checked ~ .view-value-button::before{content:"hide";}.pdoc .inherited{margin-left:2rem;}.pdoc .inherited dt{font-weight:700;}.pdoc .inherited dt, .pdoc .inherited dd{display:inline;margin-left:0;margin-bottom:.5rem;}.pdoc .inherited dd:not(:last-child):after{content:", ";}.pdoc .inherited .class:before{content:"class ";}.pdoc .inherited .function a:after{content:"()";}.pdoc .search-result .docstring{overflow:auto;max-height:25vh;}.pdoc .search-result.focused > .attr{background-color:var(--active);}.pdoc .attribution{margin-top:2rem;display:block;opacity:0.5;transition:all 200ms;filter:grayscale(100%);}.pdoc .attribution:hover{opacity:1;filter:grayscale(0%);}.pdoc .attribution img{margin-left:5px;height:35px;vertical-align:middle;width:70px;transition:all 200ms;}.pdoc table{display:block;width:max-content;max-width:100%;overflow:auto;margin-bottom:1rem;}.pdoc table th{font-weight:600;}.pdoc table th, .pdoc table td{padding:6px 13px;border:1px solid var(--accent2);}</style>
    <style>/*! custom.css */</style></head>
<body>
    <nav class="pdoc">
        <label id="navtoggle" for="togglestate" class="pdoc-button"><svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 30 30'><path stroke-linecap='round' stroke="currentColor" stroke-miterlimit='10' stroke-width='2' d='M4 7h22M4 15h22M4 23h22'/></svg></label>
        <input id="togglestate" type="checkbox" aria-hidden="true" tabindex="-1">
        <div>            <a class="pdoc-button module-list-button" href="../objects.html">
<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-box-arrow-in-left" viewBox="0 0 16 16">
  <path fill-rule="evenodd" d="M10 3.5a.5.5 0 0 0-.5-.5h-8a.5.5 0 0 0-.5.5v9a.5.5 0 0 0 .5.5h8a.5.5 0 0 0 .5-.5v-2a.5.5 0 0 1 1 0v2A1.5 1.5 0 0 1 9.5 14h-8A1.5 1.5 0 0 1 0 12.5v-9A1.5 1.5 0 0 1 1.5 2h8A1.5 1.5 0 0 1 11 3.5v2a.5.5 0 0 1-1 0v-2z"/>
  <path fill-rule="evenodd" d="M4.146 8.354a.5.5 0 0 1 0-.708l3-3a.5.5 0 1 1 .708.708L5.707 7.5H14.5a.5.5 0 0 1 0 1H5.707l2.147 2.146a.5.5 0 0 1-.708.708l-3-3z"/>
</svg>                &nbsp;howard.objects</a>


            <input type="search" placeholder="Search..." role="searchbox" aria-label="search"
                   pattern=".+" required>

            <h2>Contents</h2>
            <ul>
  <li><a href="#a-genomic-range">A genomic range:</a></li>
  <li><a href="#a-cdna-range">A cDNA range:</a></li>
  <li><a href="#a-cdna-coordinate">A cDNA coordinate:</a></li>
  <li><a href="#primatives">Primatives:</a></li>
</ul>



            <h2>API Documentation</h2>
                <ul class="memberlist">
            <li>
                    <a class="variable" href="#CHROM_PREFIX">CHROM_PREFIX</a>
            </li>
            <li>
                    <a class="variable" href="#CODON_1">CODON_1</a>
            </li>
            <li>
                    <a class="variable" href="#CODON_3">CODON_3</a>
            </li>
            <li>
                    <a class="variable" href="#CODON_FULL">CODON_FULL</a>
            </li>
            <li>
                    <a class="variable" href="#NUCLEOTIDE_TRANSLATE">NUCLEOTIDE_TRANSLATE</a>
            </li>
            <li>
                    <a class="class" href="#HGVSRegex">HGVSRegex</a>
                            <ul class="memberlist">
                        <li>
                                <a class="variable" href="#HGVSRegex.BASE">BASE</a>
                        </li>
                        <li>
                                <a class="variable" href="#HGVSRegex.BASES">BASES</a>
                        </li>
                        <li>
                                <a class="variable" href="#HGVSRegex.DNA_REF">DNA_REF</a>
                        </li>
                        <li>
                                <a class="variable" href="#HGVSRegex.DNA_ALT">DNA_ALT</a>
                        </li>
                        <li>
                                <a class="variable" href="#HGVSRegex.EQUAL">EQUAL</a>
                        </li>
                        <li>
                                <a class="variable" href="#HGVSRegex.SUB">SUB</a>
                        </li>
                        <li>
                                <a class="variable" href="#HGVSRegex.INS">INS</a>
                        </li>
                        <li>
                                <a class="variable" href="#HGVSRegex.DEL">DEL</a>
                        </li>
                        <li>
                                <a class="variable" href="#HGVSRegex.DUP">DUP</a>
                        </li>
                        <li>
                                <a class="variable" href="#HGVSRegex.COORD_START">COORD_START</a>
                        </li>
                        <li>
                                <a class="variable" href="#HGVSRegex.COORD_END">COORD_END</a>
                        </li>
                        <li>
                                <a class="variable" href="#HGVSRegex.COORD_RANGE">COORD_RANGE</a>
                        </li>
                        <li>
                                <a class="variable" href="#HGVSRegex.CDNA_COORD">CDNA_COORD</a>
                        </li>
                        <li>
                                <a class="variable" href="#HGVSRegex.CDNA_START">CDNA_START</a>
                        </li>
                        <li>
                                <a class="variable" href="#HGVSRegex.CDNA_END">CDNA_END</a>
                        </li>
                        <li>
                                <a class="variable" href="#HGVSRegex.CDNA_RANGE">CDNA_RANGE</a>
                        </li>
                        <li>
                                <a class="variable" href="#HGVSRegex.CDNA_ALLELE">CDNA_ALLELE</a>
                        </li>
                        <li>
                                <a class="variable" href="#HGVSRegex.CDNA_ALLELE_REGEXES">CDNA_ALLELE_REGEXES</a>
                        </li>
                        <li>
                                <a class="variable" href="#HGVSRegex.PEP">PEP</a>
                        </li>
                        <li>
                                <a class="variable" href="#HGVSRegex.PEP_REF">PEP_REF</a>
                        </li>
                        <li>
                                <a class="variable" href="#HGVSRegex.PEP_REF2">PEP_REF2</a>
                        </li>
                        <li>
                                <a class="variable" href="#HGVSRegex.PEP_ALT">PEP_ALT</a>
                        </li>
                        <li>
                                <a class="variable" href="#HGVSRegex.PEP_EXTRA">PEP_EXTRA</a>
                        </li>
                        <li>
                                <a class="variable" href="#HGVSRegex.PEP_ALLELE">PEP_ALLELE</a>
                        </li>
                        <li>
                                <a class="variable" href="#HGVSRegex.PEP_ALLELE_REGEXES">PEP_ALLELE_REGEXES</a>
                        </li>
                        <li>
                                <a class="variable" href="#HGVSRegex.GENOMIC_ALLELE">GENOMIC_ALLELE</a>
                        </li>
                        <li>
                                <a class="variable" href="#HGVSRegex.GENOMIC_ALLELE_REGEXES">GENOMIC_ALLELE_REGEXES</a>
                        </li>
                </ul>

            </li>
            <li>
                    <a class="variable" href="#REFSEQ_PREFIXES">REFSEQ_PREFIXES</a>
            </li>
            <li>
                    <a class="variable" href="#REFSEQ_PREFIX_LOOKUP">REFSEQ_PREFIX_LOOKUP</a>
            </li>
            <li>
                    <a class="function" href="#get_refseq_type">get_refseq_type</a>
            </li>
            <li>
                    <a class="class" href="#InvalidHGVSName">InvalidHGVSName</a>
                            <ul class="memberlist">
                        <li>
                                <a class="function" href="#InvalidHGVSName.__init__">InvalidHGVSName</a>
                        </li>
                        <li>
                                <a class="variable" href="#InvalidHGVSName.name">name</a>
                        </li>
                        <li>
                                <a class="variable" href="#InvalidHGVSName.part">part</a>
                        </li>
                        <li>
                                <a class="variable" href="#InvalidHGVSName.reason">reason</a>
                        </li>
                </ul>

            </li>
            <li>
                    <a class="class" href="#HGVSName">HGVSName</a>
                            <ul class="memberlist">
                        <li>
                                <a class="function" href="#HGVSName.__init__">HGVSName</a>
                        </li>
                        <li>
                                <a class="variable" href="#HGVSName.name">name</a>
                        </li>
                        <li>
                                <a class="variable" href="#HGVSName.prefix">prefix</a>
                        </li>
                        <li>
                                <a class="variable" href="#HGVSName.chrom">chrom</a>
                        </li>
                        <li>
                                <a class="variable" href="#HGVSName.transcript">transcript</a>
                        </li>
                        <li>
                                <a class="variable" href="#HGVSName.transcript_protein">transcript_protein</a>
                        </li>
                        <li>
                                <a class="variable" href="#HGVSName.gene">gene</a>
                        </li>
                        <li>
                                <a class="variable" href="#HGVSName.exon">exon</a>
                        </li>
                        <li>
                                <a class="variable" href="#HGVSName.kind">kind</a>
                        </li>
                        <li>
                                <a class="variable" href="#HGVSName.mutation_type">mutation_type</a>
                        </li>
                        <li>
                                <a class="variable" href="#HGVSName.start">start</a>
                        </li>
                        <li>
                                <a class="variable" href="#HGVSName.end">end</a>
                        </li>
                        <li>
                                <a class="variable" href="#HGVSName.ref_allele">ref_allele</a>
                        </li>
                        <li>
                                <a class="variable" href="#HGVSName.ref2_allele">ref2_allele</a>
                        </li>
                        <li>
                                <a class="variable" href="#HGVSName.alt_allele">alt_allele</a>
                        </li>
                        <li>
                                <a class="variable" href="#HGVSName.cdna_start">cdna_start</a>
                        </li>
                        <li>
                                <a class="variable" href="#HGVSName.cdna_end">cdna_end</a>
                        </li>
                        <li>
                                <a class="variable" href="#HGVSName.pep_extra">pep_extra</a>
                        </li>
                        <li>
                                <a class="function" href="#HGVSName.parse">parse</a>
                        </li>
                        <li>
                                <a class="function" href="#HGVSName.parse_prefix">parse_prefix</a>
                        </li>
                        <li>
                                <a class="function" href="#HGVSName.parse_allele">parse_allele</a>
                        </li>
                        <li>
                                <a class="function" href="#HGVSName.parse_cdna">parse_cdna</a>
                        </li>
                        <li>
                                <a class="function" href="#HGVSName.parse_protein">parse_protein</a>
                        </li>
                        <li>
                                <a class="function" href="#HGVSName.parse_genome">parse_genome</a>
                        </li>
                        <li>
                                <a class="function" href="#HGVSName.format">format</a>
                        </li>
                        <li>
                                <a class="function" href="#HGVSName.format_prefix">format_prefix</a>
                        </li>
                        <li>
                                <a class="function" href="#HGVSName.format_cdna_coords">format_cdna_coords</a>
                        </li>
                        <li>
                                <a class="function" href="#HGVSName.format_dna_allele">format_dna_allele</a>
                        </li>
                        <li>
                                <a class="function" href="#HGVSName.format_cdna">format_cdna</a>
                        </li>
                        <li>
                                <a class="function" href="#HGVSName.format_protein">format_protein</a>
                        </li>
                        <li>
                                <a class="function" href="#HGVSName.format_coords">format_coords</a>
                        </li>
                        <li>
                                <a class="function" href="#HGVSName.format_genome">format_genome</a>
                        </li>
                        <li>
                                <a class="function" href="#HGVSName.get_raw_coords">get_raw_coords</a>
                        </li>
                        <li>
                                <a class="function" href="#HGVSName.get_ref_coords">get_ref_coords</a>
                        </li>
                        <li>
                                <a class="function" href="#HGVSName.get_vcf_coords">get_vcf_coords</a>
                        </li>
                        <li>
                                <a class="function" href="#HGVSName.get_ref_alt">get_ref_alt</a>
                        </li>
                </ul>

            </li>
    </ul>



        <a class="attribution" title="pdoc: Python API documentation generator" href="https://pdoc.dev" target="_blank">
            built with <span class="visually-hidden">pdoc</span><img
                alt="pdoc logo"
                src="data:image/svg+xml,%3Csvg%20xmlns%3D%22http%3A//www.w3.org/2000/svg%22%20role%3D%22img%22%20aria-label%3D%22pdoc%20logo%22%20width%3D%22300%22%20height%3D%22150%22%20viewBox%3D%22-1%200%2060%2030%22%3E%3Ctitle%3Epdoc%3C/title%3E%3Cpath%20d%3D%22M29.621%2021.293c-.011-.273-.214-.475-.511-.481a.5.5%200%200%200-.489.503l-.044%201.393c-.097.551-.695%201.215-1.566%201.704-.577.428-1.306.486-2.193.182-1.426-.617-2.467-1.654-3.304-2.487l-.173-.172a3.43%203.43%200%200%200-.365-.306.49.49%200%200%200-.286-.196c-1.718-1.06-4.931-1.47-7.353.191l-.219.15c-1.707%201.187-3.413%202.131-4.328%201.03-.02-.027-.49-.685-.141-1.763.233-.721.546-2.408.772-4.076.042-.09.067-.187.046-.288.166-1.347.277-2.625.241-3.351%201.378-1.008%202.271-2.586%202.271-4.362%200-.976-.272-1.935-.788-2.774-.057-.094-.122-.18-.184-.268.033-.167.052-.339.052-.516%200-1.477-1.202-2.679-2.679-2.679-.791%200-1.496.352-1.987.9a6.3%206.3%200%200%200-1.001.029c-.492-.564-1.207-.929-2.012-.929-1.477%200-2.679%201.202-2.679%202.679A2.65%202.65%200%200%200%20.97%206.554c-.383.747-.595%201.572-.595%202.41%200%202.311%201.507%204.29%203.635%205.107-.037.699-.147%202.27-.423%203.294l-.137.461c-.622%202.042-2.515%208.257%201.727%2010.643%201.614.908%203.06%201.248%204.317%201.248%202.665%200%204.492-1.524%205.322-2.401%201.476-1.559%202.886-1.854%206.491.82%201.877%201.393%203.514%201.753%204.861%201.068%202.223-1.713%202.811-3.867%203.399-6.374.077-.846.056-1.469.054-1.537zm-4.835%204.313c-.054.305-.156.586-.242.629-.034-.007-.131-.022-.307-.157-.145-.111-.314-.478-.456-.908.221.121.432.25.675.355.115.039.219.051.33.081zm-2.251-1.238c-.05.33-.158.648-.252.694-.022.001-.125-.018-.307-.157-.217-.166-.488-.906-.639-1.573.358.344.754.693%201.198%201.036zm-3.887-2.337c-.006-.116-.018-.231-.041-.342.635.145%201.189.368%201.599.625.097.231.166.481.174.642-.03.049-.055.101-.067.158-.046.013-.128.026-.298.004-.278-.037-.901-.57-1.367-1.087zm-1.127-.497c.116.306.176.625.12.71-.019.014-.117.045-.345.016-.206-.027-.604-.332-.986-.695.41-.051.816-.056%201.211-.031zm-4.535%201.535c.209.22.379.47.358.598-.006.041-.088.138-.351.234-.144.055-.539-.063-.979-.259a11.66%2011.66%200%200%200%20.972-.573zm.983-.664c.359-.237.738-.418%201.126-.554.25.237.479.548.457.694-.006.042-.087.138-.351.235-.174.064-.694-.105-1.232-.375zm-3.381%201.794c-.022.145-.061.29-.149.401-.133.166-.358.248-.69.251h-.002c-.133%200-.306-.26-.45-.621.417.091.854.07%201.291-.031zm-2.066-8.077a4.78%204.78%200%200%201-.775-.584c.172-.115.505-.254.88-.378l-.105.962zm-.331%202.302a10.32%2010.32%200%200%201-.828-.502c.202-.143.576-.328.984-.49l-.156.992zm-.45%202.157l-.701-.403c.214-.115.536-.249.891-.376a11.57%2011.57%200%200%201-.19.779zm-.181%201.716c.064.398.194.702.298.893-.194-.051-.435-.162-.736-.398.061-.119.224-.3.438-.495zM8.87%204.141c0%20.152-.123.276-.276.276s-.275-.124-.275-.276.123-.276.276-.276.275.124.275.276zm-.735-.389a1.15%201.15%200%200%200-.314.783%201.16%201.16%200%200%200%201.162%201.162c.457%200%20.842-.27%201.032-.653.026.117.042.238.042.362a1.68%201.68%200%200%201-1.679%201.679%201.68%201.68%200%200%201-1.679-1.679c0-.843.626-1.535%201.436-1.654zM5.059%205.406A1.68%201.68%200%200%201%203.38%207.085a1.68%201.68%200%200%201-1.679-1.679c0-.037.009-.072.011-.109.21.3.541.508.935.508a1.16%201.16%200%200%200%201.162-1.162%201.14%201.14%200%200%200-.474-.912c.015%200%20.03-.005.045-.005.926.001%201.679.754%201.679%201.68zM3.198%204.141c0%20.152-.123.276-.276.276s-.275-.124-.275-.276.123-.276.276-.276.275.124.275.276zM1.375%208.964c0-.52.103-1.035.288-1.52.466.394%201.06.64%201.717.64%201.144%200%202.116-.725%202.499-1.738.383%201.012%201.355%201.738%202.499%201.738.867%200%201.631-.421%202.121-1.062.307.605.478%201.267.478%201.942%200%202.486-2.153%204.51-4.801%204.51s-4.801-2.023-4.801-4.51zm24.342%2019.349c-.985.498-2.267.168-3.813-.979-3.073-2.281-5.453-3.199-7.813-.705-1.315%201.391-4.163%203.365-8.423.97-3.174-1.786-2.239-6.266-1.261-9.479l.146-.492c.276-1.02.395-2.457.444-3.268a6.11%206.11%200%200%200%201.18.115%206.01%206.01%200%200%200%202.536-.562l-.006.175c-.802.215-1.848.612-2.021%201.25-.079.295.021.601.274.837.219.203.415.364.598.501-.667.304-1.243.698-1.311%201.179-.02.144-.022.507.393.787.213.144.395.26.564.365-1.285.521-1.361.96-1.381%201.126-.018.142-.011.496.427.746l.854.489c-.473.389-.971.914-.999%201.429-.018.278.095.532.316.713.675.556%201.231.721%201.653.721.059%200%20.104-.014.158-.02.207.707.641%201.64%201.513%201.64h.013c.8-.008%201.236-.345%201.462-.626.173-.216.268-.457.325-.692.424.195.93.374%201.372.374.151%200%20.294-.021.423-.068.732-.27.944-.704.993-1.021.009-.061.003-.119.002-.179.266.086.538.147.789.147.15%200%20.294-.021.423-.069.542-.2.797-.489.914-.754.237.147.478.258.704.288.106.014.205.021.296.021.356%200%20.595-.101.767-.229.438.435%201.094.992%201.656%201.067.106.014.205.021.296.021a1.56%201.56%200%200%200%20.323-.035c.17.575.453%201.289.866%201.605.358.273.665.362.914.362a.99.99%200%200%200%20.421-.093%201.03%201.03%200%200%200%20.245-.164c.168.428.39.846.68%201.068.358.273.665.362.913.362a.99.99%200%200%200%20.421-.093c.317-.148.512-.448.639-.762.251.157.495.257.726.257.127%200%20.25-.024.37-.071.427-.17.706-.617.841-1.314.022-.015.047-.022.068-.038.067-.051.133-.104.196-.159-.443%201.486-1.107%202.761-2.086%203.257zM8.66%209.925a.5.5%200%201%200-1%200c0%20.653-.818%201.205-1.787%201.205s-1.787-.552-1.787-1.205a.5.5%200%201%200-1%200c0%201.216%201.25%202.205%202.787%202.205s2.787-.989%202.787-2.205zm4.4%2015.965l-.208.097c-2.661%201.258-4.708%201.436-6.086.527-1.542-1.017-1.88-3.19-1.844-4.198a.4.4%200%200%200-.385-.414c-.242-.029-.406.164-.414.385-.046%201.249.367%203.686%202.202%204.896.708.467%201.547.7%202.51.7%201.248%200%202.706-.392%204.362-1.174l.185-.086a.4.4%200%200%200%20.205-.527c-.089-.204-.326-.291-.527-.206zM9.547%202.292c.093.077.205.114.317.114a.5.5%200%200%200%20.318-.886L8.817.397a.5.5%200%200%200-.703.068.5.5%200%200%200%20.069.703l1.364%201.124zm-7.661-.065c.086%200%20.173-.022.253-.068l1.523-.893a.5.5%200%200%200-.506-.863l-1.523.892a.5.5%200%200%200-.179.685c.094.158.261.247.432.247z%22%20transform%3D%22matrix%28-1%200%200%201%2058%200%29%22%20fill%3D%22%233bb300%22/%3E%3Cpath%20d%3D%22M.3%2021.86V10.18q0-.46.02-.68.04-.22.18-.5.28-.54%201.34-.54%201.06%200%201.42.28.38.26.44.78.76-1.04%202.38-1.04%201.64%200%203.1%201.54%201.46%201.54%201.46%203.58%200%202.04-1.46%203.58-1.44%201.54-3.08%201.54-1.64%200-2.38-.92v4.04q0%20.46-.04.68-.02.22-.18.5-.14.3-.5.42-.36.12-.98.12-.62%200-1-.12-.36-.12-.52-.4-.14-.28-.18-.5-.02-.22-.02-.68zm3.96-9.42q-.46.54-.46%201.18%200%20.64.46%201.18.48.52%201.2.52.74%200%201.24-.52.52-.52.52-1.18%200-.66-.48-1.18-.48-.54-1.26-.54-.76%200-1.22.54zm14.741-8.36q.16-.3.54-.42.38-.12%201-.12.64%200%201.02.12.38.12.52.42.16.3.18.54.04.22.04.68v11.94q0%20.46-.04.7-.02.22-.18.5-.3.54-1.7.54-1.38%200-1.54-.98-.84.96-2.34.96-1.8%200-3.28-1.56-1.48-1.58-1.48-3.66%200-2.1%201.48-3.68%201.5-1.58%203.28-1.58%201.48%200%202.3%201v-4.2q0-.46.02-.68.04-.24.18-.52zm-3.24%2010.86q.52.54%201.26.54.74%200%201.22-.54.5-.54.5-1.18%200-.66-.48-1.22-.46-.56-1.26-.56-.8%200-1.28.56-.48.54-.48%201.2%200%20.66.52%201.2zm7.833-1.2q0-2.4%201.68-3.96%201.68-1.56%203.84-1.56%202.16%200%203.82%201.56%201.66%201.54%201.66%203.94%200%201.66-.86%202.96-.86%201.28-2.1%201.9-1.22.6-2.54.6-1.32%200-2.56-.64-1.24-.66-2.1-1.92-.84-1.28-.84-2.88zm4.18%201.44q.64.48%201.3.48.66%200%201.32-.5.66-.5.66-1.48%200-.98-.62-1.46-.62-.48-1.34-.48-.72%200-1.34.5-.62.5-.62%201.48%200%20.96.64%201.46zm11.412-1.44q0%20.84.56%201.32.56.46%201.18.46.64%200%201.18-.36.56-.38.9-.38.6%200%201.46%201.06.46.58.46%201.04%200%20.76-1.1%201.42-1.14.8-2.8.8-1.86%200-3.58-1.34-.82-.64-1.34-1.7-.52-1.08-.52-2.36%200-1.3.52-2.34.52-1.06%201.34-1.7%201.66-1.32%203.54-1.32.76%200%201.48.22.72.2%201.06.4l.32.2q.36.24.56.38.52.4.52.92%200%20.5-.42%201.14-.72%201.1-1.38%201.1-.38%200-1.08-.44-.36-.34-1.04-.34-.66%200-1.24.48-.58.48-.58%201.34z%22%20fill%3D%22green%22/%3E%3C/svg%3E"/>
        </a>
</div>
    </nav>
    <main class="pdoc">
            <section class="module-info">
                    <h1 class="modulename">
<a href="./../../howard.html">howard</a><wbr>.<a href="./../objects.html">objects</a><wbr>.hgvs    </h1>

                        <div class="docstring"><p>HGVS language currently implemented.</p>

<p>HGVS = ALLELE
     | PREFIX_NAME : ALLELE</p>

<p>PREFIX_NAME = TRANSCRIPT
            | TRANSCRIPT '(' GENE ')'</p>

<p>TRANSCRIPT = TRANSCRIPT_NAME
           | TRANSCRIPT_NAME '.' TRANSCRIPT_VERSION</p>

<p>TRANSCRIPT_VERSION = NUMBER</p>

<p>ALLELE = 'c.' CDNA_ALLELE    # cDNA
       | 'g.' GENOMIC_ALLELE # genomic
       | 'm.' MIT_ALLELE     # mitochondrial sequence
       | 'n.' NC_ALLELE      # non-coding RNA reference sequence
       | 'r.' RNA_ALLELE     # RNA sequence (like r.76a>u)
       | 'p.' PROTEIN_ALLELE # protein sequence (like  p.Lys76Asn)</p>

<p>NC_ALLELE =
RNA_ALLELE =
CDNA_ALLELE = CDNA_COORD SINGLE_BASE_CHANGE
            | CDNA_COORD_RANGE MULTI_BASE_CHANGE</p>

<p>GENOMIC_ALLELE =
MIT_ALLELE = COORD SINGLE_BASE_CHANGE
           | COORD_RANGE MULTI_BASE_CHANGE</p>

<p>SINGLE_BASE_CHANGE = CDNA_ALLELE = CDNA_COORD BASE '='        # no change
                   | CDNA_COORD BASE '>' BASE                 # substitution
                   | CDNA_COORD 'ins' BASE                    # 1bp insertion
                   | CDNA_COORD 'del' BASE                    # 1bp deletion
                   | CDNA_COORD 'dup' BASE                    # 1bp duplication
                   | CDNA_COORD 'ins'                         # 1bp insertion
                   | CDNA_COORD 'del'                         # 1bp deletion
                   | CDNA_COORD 'dup'                         # 1bp duplication
                   | CDNA_COORD 'del' BASE 'ins' BASE         # 1bp indel
                   | CDNA_COORD 'delins' BASE                 # 1bp indel</p>

<p>MULTI_BASE_CHANGE = COORD_RANGE 'del' BASES             # deletion
                  | COORD_RANGE 'ins' BASES             # insertion
                  | COORD_RANGE 'dup' BASES             # duplication
                  | COORD_RANGE 'del'                   # deletion
                  | COORD_RANGE 'dup'                   # duplication
                  | COORD_RANGE 'del' BASES 'ins' BASES # indel
                  | COORD_RANGE 'delins' BASES          # indel</p>

<p>AMINO1 = [GAVLIMFWPSTCYNQDEKRH]</p>

<p>AMINO3 = 'Gly' | 'Ala' | 'Val' | 'Leu' | 'Ile' | 'Met' | 'Phe' | 'Trp' | 'Pro'
       | 'Ser' | 'Thr' | 'Cys' | 'Tyr' | 'Asn' | 'Gln' | 'Asp' | 'Glu' | 'Lys'
       | 'Arg' | 'His'</p>

<p>PROTEIN_ALLELE = AMINO3 COORD '='               # no peptide change
               | AMINO1 COORD '='               # no peptide change
               | AMINO3 COORD AMINO3 PEP_EXTRA  # peptide change
               | AMINO1 COORD AMINO1 PEP_EXTRA  # peptide change
               | AMINO3 COORD '_' AMINO3 COORD PEP_EXTRA        # indel
               | AMINO1 COORD '_' AMINO1 COORD PEP_EXTRA        # indel
               | AMINO3 COORD '_' AMINO3 COORD PEP_EXTRA AMINO3 # indel
               | AMINO1 COORD '_' AMINO1 COORD PEP_EXTRA AMINO1 # indel</p>

<h1 id="a-genomic-range">A genomic range:</h1>

<p>COORD_RANGE = COORD '_' COORD</p>

<h1 id="a-cdna-range">A cDNA range:</h1>

<p>CDNA_COORD_RANGE = CDNA_COORD '_' CDNA_COORD</p>

<h1 id="a-cdna-coordinate">A cDNA coordinate:</h1>

<p>CDNA_COORD = COORD_PREFIX COORD
           | COORD_PREFIX COORD OFFSET_PREFIX OFFSET
COORD_PREFIX = '' | '-' | '*'
COORD = NUMBER
OFFSET_PREFIX = '-' | '+'
OFFSET = NUMBER</p>

<h1 id="primatives">Primatives:</h1>

<p>NUMBER = "\d+"
BASE = [ACGT]
BASES = BASE+</p>
</div>

                        <input id="mod-hgvs-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">

                        <label class="view-source-button" for="mod-hgvs-view-source"><span>View Source</span></label>

                        <div class="pdoc-code codehilite"><pre><span></span><span id="L-1"><a href="#L-1"><span class="linenos">   1</span></a><span class="sa">r</span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-2"><a href="#L-2"><span class="linenos">   2</span></a><span class="sd">HGVS language currently implemented.</span>
</span><span id="L-3"><a href="#L-3"><span class="linenos">   3</span></a>
</span><span id="L-4"><a href="#L-4"><span class="linenos">   4</span></a><span class="sd">HGVS = ALLELE</span>
</span><span id="L-5"><a href="#L-5"><span class="linenos">   5</span></a><span class="sd">     | PREFIX_NAME : ALLELE</span>
</span><span id="L-6"><a href="#L-6"><span class="linenos">   6</span></a>
</span><span id="L-7"><a href="#L-7"><span class="linenos">   7</span></a><span class="sd">PREFIX_NAME = TRANSCRIPT</span>
</span><span id="L-8"><a href="#L-8"><span class="linenos">   8</span></a><span class="sd">            | TRANSCRIPT &#39;(&#39; GENE &#39;)&#39;</span>
</span><span id="L-9"><a href="#L-9"><span class="linenos">   9</span></a>
</span><span id="L-10"><a href="#L-10"><span class="linenos">  10</span></a><span class="sd">TRANSCRIPT = TRANSCRIPT_NAME</span>
</span><span id="L-11"><a href="#L-11"><span class="linenos">  11</span></a><span class="sd">           | TRANSCRIPT_NAME &#39;.&#39; TRANSCRIPT_VERSION</span>
</span><span id="L-12"><a href="#L-12"><span class="linenos">  12</span></a>
</span><span id="L-13"><a href="#L-13"><span class="linenos">  13</span></a><span class="sd">TRANSCRIPT_VERSION = NUMBER</span>
</span><span id="L-14"><a href="#L-14"><span class="linenos">  14</span></a>
</span><span id="L-15"><a href="#L-15"><span class="linenos">  15</span></a><span class="sd">ALLELE = &#39;c.&#39; CDNA_ALLELE    # cDNA</span>
</span><span id="L-16"><a href="#L-16"><span class="linenos">  16</span></a><span class="sd">       | &#39;g.&#39; GENOMIC_ALLELE # genomic</span>
</span><span id="L-17"><a href="#L-17"><span class="linenos">  17</span></a><span class="sd">       | &#39;m.&#39; MIT_ALLELE     # mitochondrial sequence</span>
</span><span id="L-18"><a href="#L-18"><span class="linenos">  18</span></a><span class="sd">       | &#39;n.&#39; NC_ALLELE      # non-coding RNA reference sequence</span>
</span><span id="L-19"><a href="#L-19"><span class="linenos">  19</span></a><span class="sd">       | &#39;r.&#39; RNA_ALLELE     # RNA sequence (like r.76a&gt;u)</span>
</span><span id="L-20"><a href="#L-20"><span class="linenos">  20</span></a><span class="sd">       | &#39;p.&#39; PROTEIN_ALLELE # protein sequence (like  p.Lys76Asn)</span>
</span><span id="L-21"><a href="#L-21"><span class="linenos">  21</span></a>
</span><span id="L-22"><a href="#L-22"><span class="linenos">  22</span></a><span class="sd">NC_ALLELE =</span>
</span><span id="L-23"><a href="#L-23"><span class="linenos">  23</span></a><span class="sd">RNA_ALLELE =</span>
</span><span id="L-24"><a href="#L-24"><span class="linenos">  24</span></a><span class="sd">CDNA_ALLELE = CDNA_COORD SINGLE_BASE_CHANGE</span>
</span><span id="L-25"><a href="#L-25"><span class="linenos">  25</span></a><span class="sd">            | CDNA_COORD_RANGE MULTI_BASE_CHANGE</span>
</span><span id="L-26"><a href="#L-26"><span class="linenos">  26</span></a>
</span><span id="L-27"><a href="#L-27"><span class="linenos">  27</span></a><span class="sd">GENOMIC_ALLELE =</span>
</span><span id="L-28"><a href="#L-28"><span class="linenos">  28</span></a><span class="sd">MIT_ALLELE = COORD SINGLE_BASE_CHANGE</span>
</span><span id="L-29"><a href="#L-29"><span class="linenos">  29</span></a><span class="sd">           | COORD_RANGE MULTI_BASE_CHANGE</span>
</span><span id="L-30"><a href="#L-30"><span class="linenos">  30</span></a>
</span><span id="L-31"><a href="#L-31"><span class="linenos">  31</span></a><span class="sd">SINGLE_BASE_CHANGE = CDNA_ALLELE = CDNA_COORD BASE &#39;=&#39;        # no change</span>
</span><span id="L-32"><a href="#L-32"><span class="linenos">  32</span></a><span class="sd">                   | CDNA_COORD BASE &#39;&gt;&#39; BASE                 # substitution</span>
</span><span id="L-33"><a href="#L-33"><span class="linenos">  33</span></a><span class="sd">                   | CDNA_COORD &#39;ins&#39; BASE                    # 1bp insertion</span>
</span><span id="L-34"><a href="#L-34"><span class="linenos">  34</span></a><span class="sd">                   | CDNA_COORD &#39;del&#39; BASE                    # 1bp deletion</span>
</span><span id="L-35"><a href="#L-35"><span class="linenos">  35</span></a><span class="sd">                   | CDNA_COORD &#39;dup&#39; BASE                    # 1bp duplication</span>
</span><span id="L-36"><a href="#L-36"><span class="linenos">  36</span></a><span class="sd">                   | CDNA_COORD &#39;ins&#39;                         # 1bp insertion</span>
</span><span id="L-37"><a href="#L-37"><span class="linenos">  37</span></a><span class="sd">                   | CDNA_COORD &#39;del&#39;                         # 1bp deletion</span>
</span><span id="L-38"><a href="#L-38"><span class="linenos">  38</span></a><span class="sd">                   | CDNA_COORD &#39;dup&#39;                         # 1bp duplication</span>
</span><span id="L-39"><a href="#L-39"><span class="linenos">  39</span></a><span class="sd">                   | CDNA_COORD &#39;del&#39; BASE &#39;ins&#39; BASE         # 1bp indel</span>
</span><span id="L-40"><a href="#L-40"><span class="linenos">  40</span></a><span class="sd">                   | CDNA_COORD &#39;delins&#39; BASE                 # 1bp indel</span>
</span><span id="L-41"><a href="#L-41"><span class="linenos">  41</span></a>
</span><span id="L-42"><a href="#L-42"><span class="linenos">  42</span></a><span class="sd">MULTI_BASE_CHANGE = COORD_RANGE &#39;del&#39; BASES             # deletion</span>
</span><span id="L-43"><a href="#L-43"><span class="linenos">  43</span></a><span class="sd">                  | COORD_RANGE &#39;ins&#39; BASES             # insertion</span>
</span><span id="L-44"><a href="#L-44"><span class="linenos">  44</span></a><span class="sd">                  | COORD_RANGE &#39;dup&#39; BASES             # duplication</span>
</span><span id="L-45"><a href="#L-45"><span class="linenos">  45</span></a><span class="sd">                  | COORD_RANGE &#39;del&#39;                   # deletion</span>
</span><span id="L-46"><a href="#L-46"><span class="linenos">  46</span></a><span class="sd">                  | COORD_RANGE &#39;dup&#39;                   # duplication</span>
</span><span id="L-47"><a href="#L-47"><span class="linenos">  47</span></a><span class="sd">                  | COORD_RANGE &#39;del&#39; BASES &#39;ins&#39; BASES # indel</span>
</span><span id="L-48"><a href="#L-48"><span class="linenos">  48</span></a><span class="sd">                  | COORD_RANGE &#39;delins&#39; BASES          # indel</span>
</span><span id="L-49"><a href="#L-49"><span class="linenos">  49</span></a>
</span><span id="L-50"><a href="#L-50"><span class="linenos">  50</span></a>
</span><span id="L-51"><a href="#L-51"><span class="linenos">  51</span></a><span class="sd">AMINO1 = [GAVLIMFWPSTCYNQDEKRH]</span>
</span><span id="L-52"><a href="#L-52"><span class="linenos">  52</span></a>
</span><span id="L-53"><a href="#L-53"><span class="linenos">  53</span></a><span class="sd">AMINO3 = &#39;Gly&#39; | &#39;Ala&#39; | &#39;Val&#39; | &#39;Leu&#39; | &#39;Ile&#39; | &#39;Met&#39; | &#39;Phe&#39; | &#39;Trp&#39; | &#39;Pro&#39;</span>
</span><span id="L-54"><a href="#L-54"><span class="linenos">  54</span></a><span class="sd">       | &#39;Ser&#39; | &#39;Thr&#39; | &#39;Cys&#39; | &#39;Tyr&#39; | &#39;Asn&#39; | &#39;Gln&#39; | &#39;Asp&#39; | &#39;Glu&#39; | &#39;Lys&#39;</span>
</span><span id="L-55"><a href="#L-55"><span class="linenos">  55</span></a><span class="sd">       | &#39;Arg&#39; | &#39;His&#39;</span>
</span><span id="L-56"><a href="#L-56"><span class="linenos">  56</span></a>
</span><span id="L-57"><a href="#L-57"><span class="linenos">  57</span></a><span class="sd">PROTEIN_ALLELE = AMINO3 COORD &#39;=&#39;               # no peptide change</span>
</span><span id="L-58"><a href="#L-58"><span class="linenos">  58</span></a><span class="sd">               | AMINO1 COORD &#39;=&#39;               # no peptide change</span>
</span><span id="L-59"><a href="#L-59"><span class="linenos">  59</span></a><span class="sd">               | AMINO3 COORD AMINO3 PEP_EXTRA  # peptide change</span>
</span><span id="L-60"><a href="#L-60"><span class="linenos">  60</span></a><span class="sd">               | AMINO1 COORD AMINO1 PEP_EXTRA  # peptide change</span>
</span><span id="L-61"><a href="#L-61"><span class="linenos">  61</span></a><span class="sd">               | AMINO3 COORD &#39;_&#39; AMINO3 COORD PEP_EXTRA        # indel</span>
</span><span id="L-62"><a href="#L-62"><span class="linenos">  62</span></a><span class="sd">               | AMINO1 COORD &#39;_&#39; AMINO1 COORD PEP_EXTRA        # indel</span>
</span><span id="L-63"><a href="#L-63"><span class="linenos">  63</span></a><span class="sd">               | AMINO3 COORD &#39;_&#39; AMINO3 COORD PEP_EXTRA AMINO3 # indel</span>
</span><span id="L-64"><a href="#L-64"><span class="linenos">  64</span></a><span class="sd">               | AMINO1 COORD &#39;_&#39; AMINO1 COORD PEP_EXTRA AMINO1 # indel</span>
</span><span id="L-65"><a href="#L-65"><span class="linenos">  65</span></a>
</span><span id="L-66"><a href="#L-66"><span class="linenos">  66</span></a><span class="sd"># A genomic range:</span>
</span><span id="L-67"><a href="#L-67"><span class="linenos">  67</span></a><span class="sd">COORD_RANGE = COORD &#39;_&#39; COORD</span>
</span><span id="L-68"><a href="#L-68"><span class="linenos">  68</span></a>
</span><span id="L-69"><a href="#L-69"><span class="linenos">  69</span></a><span class="sd"># A cDNA range:</span>
</span><span id="L-70"><a href="#L-70"><span class="linenos">  70</span></a><span class="sd">CDNA_COORD_RANGE = CDNA_COORD &#39;_&#39; CDNA_COORD</span>
</span><span id="L-71"><a href="#L-71"><span class="linenos">  71</span></a>
</span><span id="L-72"><a href="#L-72"><span class="linenos">  72</span></a><span class="sd"># A cDNA coordinate:</span>
</span><span id="L-73"><a href="#L-73"><span class="linenos">  73</span></a><span class="sd">CDNA_COORD = COORD_PREFIX COORD</span>
</span><span id="L-74"><a href="#L-74"><span class="linenos">  74</span></a><span class="sd">           | COORD_PREFIX COORD OFFSET_PREFIX OFFSET</span>
</span><span id="L-75"><a href="#L-75"><span class="linenos">  75</span></a><span class="sd">COORD_PREFIX = &#39;&#39; | &#39;-&#39; | &#39;*&#39;</span>
</span><span id="L-76"><a href="#L-76"><span class="linenos">  76</span></a><span class="sd">COORD = NUMBER</span>
</span><span id="L-77"><a href="#L-77"><span class="linenos">  77</span></a><span class="sd">OFFSET_PREFIX = &#39;-&#39; | &#39;+&#39;</span>
</span><span id="L-78"><a href="#L-78"><span class="linenos">  78</span></a><span class="sd">OFFSET = NUMBER</span>
</span><span id="L-79"><a href="#L-79"><span class="linenos">  79</span></a>
</span><span id="L-80"><a href="#L-80"><span class="linenos">  80</span></a><span class="sd"># Primatives:</span>
</span><span id="L-81"><a href="#L-81"><span class="linenos">  81</span></a><span class="sd">NUMBER = &quot;\d+&quot;</span>
</span><span id="L-82"><a href="#L-82"><span class="linenos">  82</span></a><span class="sd">BASE = [ACGT]</span>
</span><span id="L-83"><a href="#L-83"><span class="linenos">  83</span></a><span class="sd">BASES = BASE+</span>
</span><span id="L-84"><a href="#L-84"><span class="linenos">  84</span></a>
</span><span id="L-85"><a href="#L-85"><span class="linenos">  85</span></a><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-86"><a href="#L-86"><span class="linenos">  86</span></a>
</span><span id="L-87"><a href="#L-87"><span class="linenos">  87</span></a><span class="kn">import</span> <span class="nn">re</span>
</span><span id="L-88"><a href="#L-88"><span class="linenos">  88</span></a>
</span><span id="L-89"><a href="#L-89"><span class="linenos">  89</span></a><span class="kn">from</span> <span class="nn">.cdna</span> <span class="kn">import</span> <span class="n">CDNACoord</span>
</span><span id="L-90"><a href="#L-90"><span class="linenos">  90</span></a><span class="kn">from</span> <span class="nn">.variant</span> <span class="kn">import</span> <span class="n">revcomp</span>
</span><span id="L-91"><a href="#L-91"><span class="linenos">  91</span></a>
</span><span id="L-92"><a href="#L-92"><span class="linenos">  92</span></a><span class="n">CHROM_PREFIX</span> <span class="o">=</span> <span class="s2">&quot;chr&quot;</span>
</span><span id="L-93"><a href="#L-93"><span class="linenos">  93</span></a>
</span><span id="L-94"><a href="#L-94"><span class="linenos">  94</span></a>
</span><span id="L-95"><a href="#L-95"><span class="linenos">  95</span></a><span class="n">CODON_1</span> <span class="o">=</span> <span class="p">{</span>
</span><span id="L-96"><a href="#L-96"><span class="linenos">  96</span></a>    <span class="s2">&quot;TTT&quot;</span><span class="p">:</span> <span class="s2">&quot;F&quot;</span><span class="p">,</span>
</span><span id="L-97"><a href="#L-97"><span class="linenos">  97</span></a>    <span class="s2">&quot;TTC&quot;</span><span class="p">:</span> <span class="s2">&quot;F&quot;</span><span class="p">,</span>
</span><span id="L-98"><a href="#L-98"><span class="linenos">  98</span></a>    <span class="s2">&quot;TCT&quot;</span><span class="p">:</span> <span class="s2">&quot;S&quot;</span><span class="p">,</span>
</span><span id="L-99"><a href="#L-99"><span class="linenos">  99</span></a>    <span class="s2">&quot;TCC&quot;</span><span class="p">:</span> <span class="s2">&quot;S&quot;</span><span class="p">,</span>
</span><span id="L-100"><a href="#L-100"><span class="linenos"> 100</span></a>    <span class="s2">&quot;TAT&quot;</span><span class="p">:</span> <span class="s2">&quot;Y&quot;</span><span class="p">,</span>
</span><span id="L-101"><a href="#L-101"><span class="linenos"> 101</span></a>    <span class="s2">&quot;TAC&quot;</span><span class="p">:</span> <span class="s2">&quot;Y&quot;</span><span class="p">,</span>
</span><span id="L-102"><a href="#L-102"><span class="linenos"> 102</span></a>    <span class="s2">&quot;TGT&quot;</span><span class="p">:</span> <span class="s2">&quot;C&quot;</span><span class="p">,</span>
</span><span id="L-103"><a href="#L-103"><span class="linenos"> 103</span></a>    <span class="s2">&quot;TGC&quot;</span><span class="p">:</span> <span class="s2">&quot;C&quot;</span><span class="p">,</span>
</span><span id="L-104"><a href="#L-104"><span class="linenos"> 104</span></a>    <span class="s2">&quot;TTA&quot;</span><span class="p">:</span> <span class="s2">&quot;L&quot;</span><span class="p">,</span>
</span><span id="L-105"><a href="#L-105"><span class="linenos"> 105</span></a>    <span class="s2">&quot;TCA&quot;</span><span class="p">:</span> <span class="s2">&quot;S&quot;</span><span class="p">,</span>
</span><span id="L-106"><a href="#L-106"><span class="linenos"> 106</span></a>    <span class="s2">&quot;TAA&quot;</span><span class="p">:</span> <span class="s2">&quot;*&quot;</span><span class="p">,</span>
</span><span id="L-107"><a href="#L-107"><span class="linenos"> 107</span></a>    <span class="s2">&quot;TGA&quot;</span><span class="p">:</span> <span class="s2">&quot;*&quot;</span><span class="p">,</span>
</span><span id="L-108"><a href="#L-108"><span class="linenos"> 108</span></a>    <span class="s2">&quot;TTG&quot;</span><span class="p">:</span> <span class="s2">&quot;L&quot;</span><span class="p">,</span>
</span><span id="L-109"><a href="#L-109"><span class="linenos"> 109</span></a>    <span class="s2">&quot;TCG&quot;</span><span class="p">:</span> <span class="s2">&quot;S&quot;</span><span class="p">,</span>
</span><span id="L-110"><a href="#L-110"><span class="linenos"> 110</span></a>    <span class="s2">&quot;TAG&quot;</span><span class="p">:</span> <span class="s2">&quot;*&quot;</span><span class="p">,</span>
</span><span id="L-111"><a href="#L-111"><span class="linenos"> 111</span></a>    <span class="s2">&quot;TGG&quot;</span><span class="p">:</span> <span class="s2">&quot;W&quot;</span><span class="p">,</span>
</span><span id="L-112"><a href="#L-112"><span class="linenos"> 112</span></a>    <span class="s2">&quot;CTT&quot;</span><span class="p">:</span> <span class="s2">&quot;L&quot;</span><span class="p">,</span>
</span><span id="L-113"><a href="#L-113"><span class="linenos"> 113</span></a>    <span class="s2">&quot;CTC&quot;</span><span class="p">:</span> <span class="s2">&quot;L&quot;</span><span class="p">,</span>
</span><span id="L-114"><a href="#L-114"><span class="linenos"> 114</span></a>    <span class="s2">&quot;CCT&quot;</span><span class="p">:</span> <span class="s2">&quot;P&quot;</span><span class="p">,</span>
</span><span id="L-115"><a href="#L-115"><span class="linenos"> 115</span></a>    <span class="s2">&quot;CCC&quot;</span><span class="p">:</span> <span class="s2">&quot;P&quot;</span><span class="p">,</span>
</span><span id="L-116"><a href="#L-116"><span class="linenos"> 116</span></a>    <span class="s2">&quot;CAT&quot;</span><span class="p">:</span> <span class="s2">&quot;H&quot;</span><span class="p">,</span>
</span><span id="L-117"><a href="#L-117"><span class="linenos"> 117</span></a>    <span class="s2">&quot;CAC&quot;</span><span class="p">:</span> <span class="s2">&quot;H&quot;</span><span class="p">,</span>
</span><span id="L-118"><a href="#L-118"><span class="linenos"> 118</span></a>    <span class="s2">&quot;CGT&quot;</span><span class="p">:</span> <span class="s2">&quot;R&quot;</span><span class="p">,</span>
</span><span id="L-119"><a href="#L-119"><span class="linenos"> 119</span></a>    <span class="s2">&quot;CGC&quot;</span><span class="p">:</span> <span class="s2">&quot;R&quot;</span><span class="p">,</span>
</span><span id="L-120"><a href="#L-120"><span class="linenos"> 120</span></a>    <span class="s2">&quot;CTA&quot;</span><span class="p">:</span> <span class="s2">&quot;L&quot;</span><span class="p">,</span>
</span><span id="L-121"><a href="#L-121"><span class="linenos"> 121</span></a>    <span class="s2">&quot;CTG&quot;</span><span class="p">:</span> <span class="s2">&quot;L&quot;</span><span class="p">,</span>
</span><span id="L-122"><a href="#L-122"><span class="linenos"> 122</span></a>    <span class="s2">&quot;CCA&quot;</span><span class="p">:</span> <span class="s2">&quot;P&quot;</span><span class="p">,</span>
</span><span id="L-123"><a href="#L-123"><span class="linenos"> 123</span></a>    <span class="s2">&quot;CCG&quot;</span><span class="p">:</span> <span class="s2">&quot;P&quot;</span><span class="p">,</span>
</span><span id="L-124"><a href="#L-124"><span class="linenos"> 124</span></a>    <span class="s2">&quot;CAA&quot;</span><span class="p">:</span> <span class="s2">&quot;Q&quot;</span><span class="p">,</span>
</span><span id="L-125"><a href="#L-125"><span class="linenos"> 125</span></a>    <span class="s2">&quot;CAG&quot;</span><span class="p">:</span> <span class="s2">&quot;Q&quot;</span><span class="p">,</span>
</span><span id="L-126"><a href="#L-126"><span class="linenos"> 126</span></a>    <span class="s2">&quot;CGA&quot;</span><span class="p">:</span> <span class="s2">&quot;R&quot;</span><span class="p">,</span>
</span><span id="L-127"><a href="#L-127"><span class="linenos"> 127</span></a>    <span class="s2">&quot;CGG&quot;</span><span class="p">:</span> <span class="s2">&quot;R&quot;</span><span class="p">,</span>
</span><span id="L-128"><a href="#L-128"><span class="linenos"> 128</span></a>    <span class="s2">&quot;ATT&quot;</span><span class="p">:</span> <span class="s2">&quot;I&quot;</span><span class="p">,</span>
</span><span id="L-129"><a href="#L-129"><span class="linenos"> 129</span></a>    <span class="s2">&quot;ATC&quot;</span><span class="p">:</span> <span class="s2">&quot;I&quot;</span><span class="p">,</span>
</span><span id="L-130"><a href="#L-130"><span class="linenos"> 130</span></a>    <span class="s2">&quot;ACT&quot;</span><span class="p">:</span> <span class="s2">&quot;T&quot;</span><span class="p">,</span>
</span><span id="L-131"><a href="#L-131"><span class="linenos"> 131</span></a>    <span class="s2">&quot;ACC&quot;</span><span class="p">:</span> <span class="s2">&quot;T&quot;</span><span class="p">,</span>
</span><span id="L-132"><a href="#L-132"><span class="linenos"> 132</span></a>    <span class="s2">&quot;AAT&quot;</span><span class="p">:</span> <span class="s2">&quot;N&quot;</span><span class="p">,</span>
</span><span id="L-133"><a href="#L-133"><span class="linenos"> 133</span></a>    <span class="s2">&quot;AAC&quot;</span><span class="p">:</span> <span class="s2">&quot;N&quot;</span><span class="p">,</span>
</span><span id="L-134"><a href="#L-134"><span class="linenos"> 134</span></a>    <span class="s2">&quot;AGT&quot;</span><span class="p">:</span> <span class="s2">&quot;S&quot;</span><span class="p">,</span>
</span><span id="L-135"><a href="#L-135"><span class="linenos"> 135</span></a>    <span class="s2">&quot;AGC&quot;</span><span class="p">:</span> <span class="s2">&quot;S&quot;</span><span class="p">,</span>
</span><span id="L-136"><a href="#L-136"><span class="linenos"> 136</span></a>    <span class="s2">&quot;ATA&quot;</span><span class="p">:</span> <span class="s2">&quot;I&quot;</span><span class="p">,</span>
</span><span id="L-137"><a href="#L-137"><span class="linenos"> 137</span></a>    <span class="s2">&quot;ACA&quot;</span><span class="p">:</span> <span class="s2">&quot;T&quot;</span><span class="p">,</span>
</span><span id="L-138"><a href="#L-138"><span class="linenos"> 138</span></a>    <span class="s2">&quot;AAA&quot;</span><span class="p">:</span> <span class="s2">&quot;K&quot;</span><span class="p">,</span>
</span><span id="L-139"><a href="#L-139"><span class="linenos"> 139</span></a>    <span class="s2">&quot;AGA&quot;</span><span class="p">:</span> <span class="s2">&quot;R&quot;</span><span class="p">,</span>
</span><span id="L-140"><a href="#L-140"><span class="linenos"> 140</span></a>    <span class="s2">&quot;ATG&quot;</span><span class="p">:</span> <span class="s2">&quot;M&quot;</span><span class="p">,</span>
</span><span id="L-141"><a href="#L-141"><span class="linenos"> 141</span></a>    <span class="s2">&quot;ACG&quot;</span><span class="p">:</span> <span class="s2">&quot;T&quot;</span><span class="p">,</span>
</span><span id="L-142"><a href="#L-142"><span class="linenos"> 142</span></a>    <span class="s2">&quot;AAG&quot;</span><span class="p">:</span> <span class="s2">&quot;K&quot;</span><span class="p">,</span>
</span><span id="L-143"><a href="#L-143"><span class="linenos"> 143</span></a>    <span class="s2">&quot;AGG&quot;</span><span class="p">:</span> <span class="s2">&quot;R&quot;</span><span class="p">,</span>
</span><span id="L-144"><a href="#L-144"><span class="linenos"> 144</span></a>    <span class="s2">&quot;GTT&quot;</span><span class="p">:</span> <span class="s2">&quot;V&quot;</span><span class="p">,</span>
</span><span id="L-145"><a href="#L-145"><span class="linenos"> 145</span></a>    <span class="s2">&quot;GTC&quot;</span><span class="p">:</span> <span class="s2">&quot;V&quot;</span><span class="p">,</span>
</span><span id="L-146"><a href="#L-146"><span class="linenos"> 146</span></a>    <span class="s2">&quot;GCT&quot;</span><span class="p">:</span> <span class="s2">&quot;A&quot;</span><span class="p">,</span>
</span><span id="L-147"><a href="#L-147"><span class="linenos"> 147</span></a>    <span class="s2">&quot;GCC&quot;</span><span class="p">:</span> <span class="s2">&quot;A&quot;</span><span class="p">,</span>
</span><span id="L-148"><a href="#L-148"><span class="linenos"> 148</span></a>    <span class="s2">&quot;GAT&quot;</span><span class="p">:</span> <span class="s2">&quot;D&quot;</span><span class="p">,</span>
</span><span id="L-149"><a href="#L-149"><span class="linenos"> 149</span></a>    <span class="s2">&quot;GAC&quot;</span><span class="p">:</span> <span class="s2">&quot;D&quot;</span><span class="p">,</span>
</span><span id="L-150"><a href="#L-150"><span class="linenos"> 150</span></a>    <span class="s2">&quot;GGT&quot;</span><span class="p">:</span> <span class="s2">&quot;G&quot;</span><span class="p">,</span>
</span><span id="L-151"><a href="#L-151"><span class="linenos"> 151</span></a>    <span class="s2">&quot;GGC&quot;</span><span class="p">:</span> <span class="s2">&quot;G&quot;</span><span class="p">,</span>
</span><span id="L-152"><a href="#L-152"><span class="linenos"> 152</span></a>    <span class="s2">&quot;GTA&quot;</span><span class="p">:</span> <span class="s2">&quot;V&quot;</span><span class="p">,</span>
</span><span id="L-153"><a href="#L-153"><span class="linenos"> 153</span></a>    <span class="s2">&quot;GTG&quot;</span><span class="p">:</span> <span class="s2">&quot;V&quot;</span><span class="p">,</span>
</span><span id="L-154"><a href="#L-154"><span class="linenos"> 154</span></a>    <span class="s2">&quot;GCA&quot;</span><span class="p">:</span> <span class="s2">&quot;A&quot;</span><span class="p">,</span>
</span><span id="L-155"><a href="#L-155"><span class="linenos"> 155</span></a>    <span class="s2">&quot;GCG&quot;</span><span class="p">:</span> <span class="s2">&quot;A&quot;</span><span class="p">,</span>
</span><span id="L-156"><a href="#L-156"><span class="linenos"> 156</span></a>    <span class="s2">&quot;GAA&quot;</span><span class="p">:</span> <span class="s2">&quot;E&quot;</span><span class="p">,</span>
</span><span id="L-157"><a href="#L-157"><span class="linenos"> 157</span></a>    <span class="s2">&quot;GAG&quot;</span><span class="p">:</span> <span class="s2">&quot;E&quot;</span><span class="p">,</span>
</span><span id="L-158"><a href="#L-158"><span class="linenos"> 158</span></a>    <span class="s2">&quot;GGA&quot;</span><span class="p">:</span> <span class="s2">&quot;G&quot;</span><span class="p">,</span>
</span><span id="L-159"><a href="#L-159"><span class="linenos"> 159</span></a>    <span class="s2">&quot;GGG&quot;</span><span class="p">:</span> <span class="s2">&quot;G&quot;</span><span class="p">,</span>
</span><span id="L-160"><a href="#L-160"><span class="linenos"> 160</span></a><span class="p">}</span>
</span><span id="L-161"><a href="#L-161"><span class="linenos"> 161</span></a><span class="n">CODON_3</span> <span class="o">=</span> <span class="p">{</span>
</span><span id="L-162"><a href="#L-162"><span class="linenos"> 162</span></a>    <span class="s2">&quot;TTT&quot;</span><span class="p">:</span> <span class="s2">&quot;Phe&quot;</span><span class="p">,</span>
</span><span id="L-163"><a href="#L-163"><span class="linenos"> 163</span></a>    <span class="s2">&quot;TTC&quot;</span><span class="p">:</span> <span class="s2">&quot;Phe&quot;</span><span class="p">,</span>
</span><span id="L-164"><a href="#L-164"><span class="linenos"> 164</span></a>    <span class="s2">&quot;TCT&quot;</span><span class="p">:</span> <span class="s2">&quot;Ser&quot;</span><span class="p">,</span>
</span><span id="L-165"><a href="#L-165"><span class="linenos"> 165</span></a>    <span class="s2">&quot;TCC&quot;</span><span class="p">:</span> <span class="s2">&quot;Ser&quot;</span><span class="p">,</span>
</span><span id="L-166"><a href="#L-166"><span class="linenos"> 166</span></a>    <span class="s2">&quot;TAT&quot;</span><span class="p">:</span> <span class="s2">&quot;Tyr&quot;</span><span class="p">,</span>
</span><span id="L-167"><a href="#L-167"><span class="linenos"> 167</span></a>    <span class="s2">&quot;TAC&quot;</span><span class="p">:</span> <span class="s2">&quot;Tyr&quot;</span><span class="p">,</span>
</span><span id="L-168"><a href="#L-168"><span class="linenos"> 168</span></a>    <span class="s2">&quot;TGT&quot;</span><span class="p">:</span> <span class="s2">&quot;Cys&quot;</span><span class="p">,</span>
</span><span id="L-169"><a href="#L-169"><span class="linenos"> 169</span></a>    <span class="s2">&quot;TGC&quot;</span><span class="p">:</span> <span class="s2">&quot;Cys&quot;</span><span class="p">,</span>
</span><span id="L-170"><a href="#L-170"><span class="linenos"> 170</span></a>    <span class="s2">&quot;TTA&quot;</span><span class="p">:</span> <span class="s2">&quot;Leu&quot;</span><span class="p">,</span>
</span><span id="L-171"><a href="#L-171"><span class="linenos"> 171</span></a>    <span class="s2">&quot;TCA&quot;</span><span class="p">:</span> <span class="s2">&quot;Ser&quot;</span><span class="p">,</span>
</span><span id="L-172"><a href="#L-172"><span class="linenos"> 172</span></a>    <span class="s2">&quot;TAA&quot;</span><span class="p">:</span> <span class="s2">&quot;*&quot;</span><span class="p">,</span>
</span><span id="L-173"><a href="#L-173"><span class="linenos"> 173</span></a>    <span class="s2">&quot;TGA&quot;</span><span class="p">:</span> <span class="s2">&quot;*&quot;</span><span class="p">,</span>
</span><span id="L-174"><a href="#L-174"><span class="linenos"> 174</span></a>    <span class="s2">&quot;TTG&quot;</span><span class="p">:</span> <span class="s2">&quot;Leu&quot;</span><span class="p">,</span>
</span><span id="L-175"><a href="#L-175"><span class="linenos"> 175</span></a>    <span class="s2">&quot;TCG&quot;</span><span class="p">:</span> <span class="s2">&quot;Ser&quot;</span><span class="p">,</span>
</span><span id="L-176"><a href="#L-176"><span class="linenos"> 176</span></a>    <span class="s2">&quot;TAG&quot;</span><span class="p">:</span> <span class="s2">&quot;*&quot;</span><span class="p">,</span>
</span><span id="L-177"><a href="#L-177"><span class="linenos"> 177</span></a>    <span class="s2">&quot;TGG&quot;</span><span class="p">:</span> <span class="s2">&quot;Trp&quot;</span><span class="p">,</span>
</span><span id="L-178"><a href="#L-178"><span class="linenos"> 178</span></a>    <span class="s2">&quot;CTT&quot;</span><span class="p">:</span> <span class="s2">&quot;Leu&quot;</span><span class="p">,</span>
</span><span id="L-179"><a href="#L-179"><span class="linenos"> 179</span></a>    <span class="s2">&quot;CTC&quot;</span><span class="p">:</span> <span class="s2">&quot;Leu&quot;</span><span class="p">,</span>
</span><span id="L-180"><a href="#L-180"><span class="linenos"> 180</span></a>    <span class="s2">&quot;CCT&quot;</span><span class="p">:</span> <span class="s2">&quot;Pro&quot;</span><span class="p">,</span>
</span><span id="L-181"><a href="#L-181"><span class="linenos"> 181</span></a>    <span class="s2">&quot;CCC&quot;</span><span class="p">:</span> <span class="s2">&quot;Pro&quot;</span><span class="p">,</span>
</span><span id="L-182"><a href="#L-182"><span class="linenos"> 182</span></a>    <span class="s2">&quot;CAT&quot;</span><span class="p">:</span> <span class="s2">&quot;His&quot;</span><span class="p">,</span>
</span><span id="L-183"><a href="#L-183"><span class="linenos"> 183</span></a>    <span class="s2">&quot;CAC&quot;</span><span class="p">:</span> <span class="s2">&quot;His&quot;</span><span class="p">,</span>
</span><span id="L-184"><a href="#L-184"><span class="linenos"> 184</span></a>    <span class="s2">&quot;CGT&quot;</span><span class="p">:</span> <span class="s2">&quot;Arg&quot;</span><span class="p">,</span>
</span><span id="L-185"><a href="#L-185"><span class="linenos"> 185</span></a>    <span class="s2">&quot;CGC&quot;</span><span class="p">:</span> <span class="s2">&quot;Arg&quot;</span><span class="p">,</span>
</span><span id="L-186"><a href="#L-186"><span class="linenos"> 186</span></a>    <span class="s2">&quot;CTA&quot;</span><span class="p">:</span> <span class="s2">&quot;Leu&quot;</span><span class="p">,</span>
</span><span id="L-187"><a href="#L-187"><span class="linenos"> 187</span></a>    <span class="s2">&quot;CTG&quot;</span><span class="p">:</span> <span class="s2">&quot;Leu&quot;</span><span class="p">,</span>
</span><span id="L-188"><a href="#L-188"><span class="linenos"> 188</span></a>    <span class="s2">&quot;CCA&quot;</span><span class="p">:</span> <span class="s2">&quot;Pro&quot;</span><span class="p">,</span>
</span><span id="L-189"><a href="#L-189"><span class="linenos"> 189</span></a>    <span class="s2">&quot;CCG&quot;</span><span class="p">:</span> <span class="s2">&quot;Pro&quot;</span><span class="p">,</span>
</span><span id="L-190"><a href="#L-190"><span class="linenos"> 190</span></a>    <span class="s2">&quot;CAA&quot;</span><span class="p">:</span> <span class="s2">&quot;Gln&quot;</span><span class="p">,</span>
</span><span id="L-191"><a href="#L-191"><span class="linenos"> 191</span></a>    <span class="s2">&quot;CAG&quot;</span><span class="p">:</span> <span class="s2">&quot;Gln&quot;</span><span class="p">,</span>
</span><span id="L-192"><a href="#L-192"><span class="linenos"> 192</span></a>    <span class="s2">&quot;CGA&quot;</span><span class="p">:</span> <span class="s2">&quot;Arg&quot;</span><span class="p">,</span>
</span><span id="L-193"><a href="#L-193"><span class="linenos"> 193</span></a>    <span class="s2">&quot;CGG&quot;</span><span class="p">:</span> <span class="s2">&quot;Arg&quot;</span><span class="p">,</span>
</span><span id="L-194"><a href="#L-194"><span class="linenos"> 194</span></a>    <span class="s2">&quot;ATT&quot;</span><span class="p">:</span> <span class="s2">&quot;Ile&quot;</span><span class="p">,</span>
</span><span id="L-195"><a href="#L-195"><span class="linenos"> 195</span></a>    <span class="s2">&quot;ATC&quot;</span><span class="p">:</span> <span class="s2">&quot;Ile&quot;</span><span class="p">,</span>
</span><span id="L-196"><a href="#L-196"><span class="linenos"> 196</span></a>    <span class="s2">&quot;ACT&quot;</span><span class="p">:</span> <span class="s2">&quot;Thr&quot;</span><span class="p">,</span>
</span><span id="L-197"><a href="#L-197"><span class="linenos"> 197</span></a>    <span class="s2">&quot;ACC&quot;</span><span class="p">:</span> <span class="s2">&quot;Thr&quot;</span><span class="p">,</span>
</span><span id="L-198"><a href="#L-198"><span class="linenos"> 198</span></a>    <span class="s2">&quot;AAT&quot;</span><span class="p">:</span> <span class="s2">&quot;Asn&quot;</span><span class="p">,</span>
</span><span id="L-199"><a href="#L-199"><span class="linenos"> 199</span></a>    <span class="s2">&quot;AAC&quot;</span><span class="p">:</span> <span class="s2">&quot;Asn&quot;</span><span class="p">,</span>
</span><span id="L-200"><a href="#L-200"><span class="linenos"> 200</span></a>    <span class="s2">&quot;AGT&quot;</span><span class="p">:</span> <span class="s2">&quot;Ser&quot;</span><span class="p">,</span>
</span><span id="L-201"><a href="#L-201"><span class="linenos"> 201</span></a>    <span class="s2">&quot;AGC&quot;</span><span class="p">:</span> <span class="s2">&quot;Ser&quot;</span><span class="p">,</span>
</span><span id="L-202"><a href="#L-202"><span class="linenos"> 202</span></a>    <span class="s2">&quot;ATA&quot;</span><span class="p">:</span> <span class="s2">&quot;Ile&quot;</span><span class="p">,</span>
</span><span id="L-203"><a href="#L-203"><span class="linenos"> 203</span></a>    <span class="s2">&quot;ACA&quot;</span><span class="p">:</span> <span class="s2">&quot;Thr&quot;</span><span class="p">,</span>
</span><span id="L-204"><a href="#L-204"><span class="linenos"> 204</span></a>    <span class="s2">&quot;AAA&quot;</span><span class="p">:</span> <span class="s2">&quot;Lys&quot;</span><span class="p">,</span>
</span><span id="L-205"><a href="#L-205"><span class="linenos"> 205</span></a>    <span class="s2">&quot;AGA&quot;</span><span class="p">:</span> <span class="s2">&quot;Arg&quot;</span><span class="p">,</span>
</span><span id="L-206"><a href="#L-206"><span class="linenos"> 206</span></a>    <span class="s2">&quot;ATG&quot;</span><span class="p">:</span> <span class="s2">&quot;Met&quot;</span><span class="p">,</span>
</span><span id="L-207"><a href="#L-207"><span class="linenos"> 207</span></a>    <span class="s2">&quot;ACG&quot;</span><span class="p">:</span> <span class="s2">&quot;Thr&quot;</span><span class="p">,</span>
</span><span id="L-208"><a href="#L-208"><span class="linenos"> 208</span></a>    <span class="s2">&quot;AAG&quot;</span><span class="p">:</span> <span class="s2">&quot;Lys&quot;</span><span class="p">,</span>
</span><span id="L-209"><a href="#L-209"><span class="linenos"> 209</span></a>    <span class="s2">&quot;AGG&quot;</span><span class="p">:</span> <span class="s2">&quot;Arg&quot;</span><span class="p">,</span>
</span><span id="L-210"><a href="#L-210"><span class="linenos"> 210</span></a>    <span class="s2">&quot;GTT&quot;</span><span class="p">:</span> <span class="s2">&quot;Val&quot;</span><span class="p">,</span>
</span><span id="L-211"><a href="#L-211"><span class="linenos"> 211</span></a>    <span class="s2">&quot;GTC&quot;</span><span class="p">:</span> <span class="s2">&quot;Val&quot;</span><span class="p">,</span>
</span><span id="L-212"><a href="#L-212"><span class="linenos"> 212</span></a>    <span class="s2">&quot;GCT&quot;</span><span class="p">:</span> <span class="s2">&quot;Ala&quot;</span><span class="p">,</span>
</span><span id="L-213"><a href="#L-213"><span class="linenos"> 213</span></a>    <span class="s2">&quot;GCC&quot;</span><span class="p">:</span> <span class="s2">&quot;Ala&quot;</span><span class="p">,</span>
</span><span id="L-214"><a href="#L-214"><span class="linenos"> 214</span></a>    <span class="s2">&quot;GAT&quot;</span><span class="p">:</span> <span class="s2">&quot;Asp&quot;</span><span class="p">,</span>
</span><span id="L-215"><a href="#L-215"><span class="linenos"> 215</span></a>    <span class="s2">&quot;GAC&quot;</span><span class="p">:</span> <span class="s2">&quot;Asp&quot;</span><span class="p">,</span>
</span><span id="L-216"><a href="#L-216"><span class="linenos"> 216</span></a>    <span class="s2">&quot;GGT&quot;</span><span class="p">:</span> <span class="s2">&quot;Gly&quot;</span><span class="p">,</span>
</span><span id="L-217"><a href="#L-217"><span class="linenos"> 217</span></a>    <span class="s2">&quot;GGC&quot;</span><span class="p">:</span> <span class="s2">&quot;Gly&quot;</span><span class="p">,</span>
</span><span id="L-218"><a href="#L-218"><span class="linenos"> 218</span></a>    <span class="s2">&quot;GTA&quot;</span><span class="p">:</span> <span class="s2">&quot;Val&quot;</span><span class="p">,</span>
</span><span id="L-219"><a href="#L-219"><span class="linenos"> 219</span></a>    <span class="s2">&quot;GTG&quot;</span><span class="p">:</span> <span class="s2">&quot;Val&quot;</span><span class="p">,</span>
</span><span id="L-220"><a href="#L-220"><span class="linenos"> 220</span></a>    <span class="s2">&quot;GCA&quot;</span><span class="p">:</span> <span class="s2">&quot;Ala&quot;</span><span class="p">,</span>
</span><span id="L-221"><a href="#L-221"><span class="linenos"> 221</span></a>    <span class="s2">&quot;GCG&quot;</span><span class="p">:</span> <span class="s2">&quot;Ala&quot;</span><span class="p">,</span>
</span><span id="L-222"><a href="#L-222"><span class="linenos"> 222</span></a>    <span class="s2">&quot;GAA&quot;</span><span class="p">:</span> <span class="s2">&quot;Glu&quot;</span><span class="p">,</span>
</span><span id="L-223"><a href="#L-223"><span class="linenos"> 223</span></a>    <span class="s2">&quot;GAG&quot;</span><span class="p">:</span> <span class="s2">&quot;Glu&quot;</span><span class="p">,</span>
</span><span id="L-224"><a href="#L-224"><span class="linenos"> 224</span></a>    <span class="s2">&quot;GGA&quot;</span><span class="p">:</span> <span class="s2">&quot;Gly&quot;</span><span class="p">,</span>
</span><span id="L-225"><a href="#L-225"><span class="linenos"> 225</span></a>    <span class="s2">&quot;GGG&quot;</span><span class="p">:</span> <span class="s2">&quot;Gly&quot;</span><span class="p">,</span>
</span><span id="L-226"><a href="#L-226"><span class="linenos"> 226</span></a><span class="p">}</span>
</span><span id="L-227"><a href="#L-227"><span class="linenos"> 227</span></a><span class="n">CODON_FULL</span> <span class="o">=</span> <span class="p">{</span>
</span><span id="L-228"><a href="#L-228"><span class="linenos"> 228</span></a>    <span class="s2">&quot;TTT&quot;</span><span class="p">:</span> <span class="s2">&quot;Phenylalanine&quot;</span><span class="p">,</span>
</span><span id="L-229"><a href="#L-229"><span class="linenos"> 229</span></a>    <span class="s2">&quot;TTC&quot;</span><span class="p">:</span> <span class="s2">&quot;Phenylalanine&quot;</span><span class="p">,</span>
</span><span id="L-230"><a href="#L-230"><span class="linenos"> 230</span></a>    <span class="s2">&quot;TCT&quot;</span><span class="p">:</span> <span class="s2">&quot;Serine&quot;</span><span class="p">,</span>
</span><span id="L-231"><a href="#L-231"><span class="linenos"> 231</span></a>    <span class="s2">&quot;TCC&quot;</span><span class="p">:</span> <span class="s2">&quot;Serine&quot;</span><span class="p">,</span>
</span><span id="L-232"><a href="#L-232"><span class="linenos"> 232</span></a>    <span class="s2">&quot;TAT&quot;</span><span class="p">:</span> <span class="s2">&quot;Tyrosine&quot;</span><span class="p">,</span>
</span><span id="L-233"><a href="#L-233"><span class="linenos"> 233</span></a>    <span class="s2">&quot;TAC&quot;</span><span class="p">:</span> <span class="s2">&quot;Tyrosine&quot;</span><span class="p">,</span>
</span><span id="L-234"><a href="#L-234"><span class="linenos"> 234</span></a>    <span class="s2">&quot;TGT&quot;</span><span class="p">:</span> <span class="s2">&quot;Cysteine&quot;</span><span class="p">,</span>
</span><span id="L-235"><a href="#L-235"><span class="linenos"> 235</span></a>    <span class="s2">&quot;TGC&quot;</span><span class="p">:</span> <span class="s2">&quot;Cysteine&quot;</span><span class="p">,</span>
</span><span id="L-236"><a href="#L-236"><span class="linenos"> 236</span></a>    <span class="s2">&quot;TTA&quot;</span><span class="p">:</span> <span class="s2">&quot;Leucine&quot;</span><span class="p">,</span>
</span><span id="L-237"><a href="#L-237"><span class="linenos"> 237</span></a>    <span class="s2">&quot;TCA&quot;</span><span class="p">:</span> <span class="s2">&quot;Serine&quot;</span><span class="p">,</span>
</span><span id="L-238"><a href="#L-238"><span class="linenos"> 238</span></a>    <span class="s2">&quot;TAA&quot;</span><span class="p">:</span> <span class="s2">&quot;Stop&quot;</span><span class="p">,</span>
</span><span id="L-239"><a href="#L-239"><span class="linenos"> 239</span></a>    <span class="s2">&quot;TGA&quot;</span><span class="p">:</span> <span class="s2">&quot;Stop&quot;</span><span class="p">,</span>
</span><span id="L-240"><a href="#L-240"><span class="linenos"> 240</span></a>    <span class="s2">&quot;TTG&quot;</span><span class="p">:</span> <span class="s2">&quot;Leucine&quot;</span><span class="p">,</span>
</span><span id="L-241"><a href="#L-241"><span class="linenos"> 241</span></a>    <span class="s2">&quot;TCG&quot;</span><span class="p">:</span> <span class="s2">&quot;Serine&quot;</span><span class="p">,</span>
</span><span id="L-242"><a href="#L-242"><span class="linenos"> 242</span></a>    <span class="s2">&quot;TAG&quot;</span><span class="p">:</span> <span class="s2">&quot;Stop&quot;</span><span class="p">,</span>
</span><span id="L-243"><a href="#L-243"><span class="linenos"> 243</span></a>    <span class="s2">&quot;TGG&quot;</span><span class="p">:</span> <span class="s2">&quot;Tryptophan&quot;</span><span class="p">,</span>
</span><span id="L-244"><a href="#L-244"><span class="linenos"> 244</span></a>    <span class="s2">&quot;CTT&quot;</span><span class="p">:</span> <span class="s2">&quot;Leucine&quot;</span><span class="p">,</span>
</span><span id="L-245"><a href="#L-245"><span class="linenos"> 245</span></a>    <span class="s2">&quot;CTC&quot;</span><span class="p">:</span> <span class="s2">&quot;Leucine&quot;</span><span class="p">,</span>
</span><span id="L-246"><a href="#L-246"><span class="linenos"> 246</span></a>    <span class="s2">&quot;CCT&quot;</span><span class="p">:</span> <span class="s2">&quot;Proline&quot;</span><span class="p">,</span>
</span><span id="L-247"><a href="#L-247"><span class="linenos"> 247</span></a>    <span class="s2">&quot;CCC&quot;</span><span class="p">:</span> <span class="s2">&quot;Proline&quot;</span><span class="p">,</span>
</span><span id="L-248"><a href="#L-248"><span class="linenos"> 248</span></a>    <span class="s2">&quot;CAT&quot;</span><span class="p">:</span> <span class="s2">&quot;Histidine&quot;</span><span class="p">,</span>
</span><span id="L-249"><a href="#L-249"><span class="linenos"> 249</span></a>    <span class="s2">&quot;CAC&quot;</span><span class="p">:</span> <span class="s2">&quot;Histidine&quot;</span><span class="p">,</span>
</span><span id="L-250"><a href="#L-250"><span class="linenos"> 250</span></a>    <span class="s2">&quot;CGT&quot;</span><span class="p">:</span> <span class="s2">&quot;Arginine&quot;</span><span class="p">,</span>
</span><span id="L-251"><a href="#L-251"><span class="linenos"> 251</span></a>    <span class="s2">&quot;CGC&quot;</span><span class="p">:</span> <span class="s2">&quot;Arginine&quot;</span><span class="p">,</span>
</span><span id="L-252"><a href="#L-252"><span class="linenos"> 252</span></a>    <span class="s2">&quot;CTA&quot;</span><span class="p">:</span> <span class="s2">&quot;Leucine&quot;</span><span class="p">,</span>
</span><span id="L-253"><a href="#L-253"><span class="linenos"> 253</span></a>    <span class="s2">&quot;CTG&quot;</span><span class="p">:</span> <span class="s2">&quot;Leucine&quot;</span><span class="p">,</span>
</span><span id="L-254"><a href="#L-254"><span class="linenos"> 254</span></a>    <span class="s2">&quot;CCA&quot;</span><span class="p">:</span> <span class="s2">&quot;Proline&quot;</span><span class="p">,</span>
</span><span id="L-255"><a href="#L-255"><span class="linenos"> 255</span></a>    <span class="s2">&quot;CCG&quot;</span><span class="p">:</span> <span class="s2">&quot;Proline&quot;</span><span class="p">,</span>
</span><span id="L-256"><a href="#L-256"><span class="linenos"> 256</span></a>    <span class="s2">&quot;CAA&quot;</span><span class="p">:</span> <span class="s2">&quot;Glutamine&quot;</span><span class="p">,</span>
</span><span id="L-257"><a href="#L-257"><span class="linenos"> 257</span></a>    <span class="s2">&quot;CAG&quot;</span><span class="p">:</span> <span class="s2">&quot;Glutamine&quot;</span><span class="p">,</span>
</span><span id="L-258"><a href="#L-258"><span class="linenos"> 258</span></a>    <span class="s2">&quot;CGA&quot;</span><span class="p">:</span> <span class="s2">&quot;Arginine&quot;</span><span class="p">,</span>
</span><span id="L-259"><a href="#L-259"><span class="linenos"> 259</span></a>    <span class="s2">&quot;CGG&quot;</span><span class="p">:</span> <span class="s2">&quot;Arginine&quot;</span><span class="p">,</span>
</span><span id="L-260"><a href="#L-260"><span class="linenos"> 260</span></a>    <span class="s2">&quot;ATT&quot;</span><span class="p">:</span> <span class="s2">&quot;Isoleucine&quot;</span><span class="p">,</span>
</span><span id="L-261"><a href="#L-261"><span class="linenos"> 261</span></a>    <span class="s2">&quot;ATC&quot;</span><span class="p">:</span> <span class="s2">&quot;Isoleucine&quot;</span><span class="p">,</span>
</span><span id="L-262"><a href="#L-262"><span class="linenos"> 262</span></a>    <span class="s2">&quot;ACT&quot;</span><span class="p">:</span> <span class="s2">&quot;Threonine&quot;</span><span class="p">,</span>
</span><span id="L-263"><a href="#L-263"><span class="linenos"> 263</span></a>    <span class="s2">&quot;ACC&quot;</span><span class="p">:</span> <span class="s2">&quot;Threonine&quot;</span><span class="p">,</span>
</span><span id="L-264"><a href="#L-264"><span class="linenos"> 264</span></a>    <span class="s2">&quot;AAT&quot;</span><span class="p">:</span> <span class="s2">&quot;Asparagine&quot;</span><span class="p">,</span>
</span><span id="L-265"><a href="#L-265"><span class="linenos"> 265</span></a>    <span class="s2">&quot;AAC&quot;</span><span class="p">:</span> <span class="s2">&quot;Asparagine&quot;</span><span class="p">,</span>
</span><span id="L-266"><a href="#L-266"><span class="linenos"> 266</span></a>    <span class="s2">&quot;AGT&quot;</span><span class="p">:</span> <span class="s2">&quot;Serine&quot;</span><span class="p">,</span>
</span><span id="L-267"><a href="#L-267"><span class="linenos"> 267</span></a>    <span class="s2">&quot;AGC&quot;</span><span class="p">:</span> <span class="s2">&quot;Serine&quot;</span><span class="p">,</span>
</span><span id="L-268"><a href="#L-268"><span class="linenos"> 268</span></a>    <span class="s2">&quot;ATA&quot;</span><span class="p">:</span> <span class="s2">&quot;Isoleucine&quot;</span><span class="p">,</span>
</span><span id="L-269"><a href="#L-269"><span class="linenos"> 269</span></a>    <span class="s2">&quot;ACA&quot;</span><span class="p">:</span> <span class="s2">&quot;Threonine&quot;</span><span class="p">,</span>
</span><span id="L-270"><a href="#L-270"><span class="linenos"> 270</span></a>    <span class="s2">&quot;AAA&quot;</span><span class="p">:</span> <span class="s2">&quot;Lysine&quot;</span><span class="p">,</span>
</span><span id="L-271"><a href="#L-271"><span class="linenos"> 271</span></a>    <span class="s2">&quot;AGA&quot;</span><span class="p">:</span> <span class="s2">&quot;Arginine&quot;</span><span class="p">,</span>
</span><span id="L-272"><a href="#L-272"><span class="linenos"> 272</span></a>    <span class="s2">&quot;ATG&quot;</span><span class="p">:</span> <span class="s2">&quot;Methionine&quot;</span><span class="p">,</span>
</span><span id="L-273"><a href="#L-273"><span class="linenos"> 273</span></a>    <span class="s2">&quot;ACG&quot;</span><span class="p">:</span> <span class="s2">&quot;Threonine&quot;</span><span class="p">,</span>
</span><span id="L-274"><a href="#L-274"><span class="linenos"> 274</span></a>    <span class="s2">&quot;AAG&quot;</span><span class="p">:</span> <span class="s2">&quot;Lysine&quot;</span><span class="p">,</span>
</span><span id="L-275"><a href="#L-275"><span class="linenos"> 275</span></a>    <span class="s2">&quot;AGG&quot;</span><span class="p">:</span> <span class="s2">&quot;Arginine&quot;</span><span class="p">,</span>
</span><span id="L-276"><a href="#L-276"><span class="linenos"> 276</span></a>    <span class="s2">&quot;GTT&quot;</span><span class="p">:</span> <span class="s2">&quot;Valine&quot;</span><span class="p">,</span>
</span><span id="L-277"><a href="#L-277"><span class="linenos"> 277</span></a>    <span class="s2">&quot;GTC&quot;</span><span class="p">:</span> <span class="s2">&quot;Valine&quot;</span><span class="p">,</span>
</span><span id="L-278"><a href="#L-278"><span class="linenos"> 278</span></a>    <span class="s2">&quot;GCT&quot;</span><span class="p">:</span> <span class="s2">&quot;Alanine&quot;</span><span class="p">,</span>
</span><span id="L-279"><a href="#L-279"><span class="linenos"> 279</span></a>    <span class="s2">&quot;GCC&quot;</span><span class="p">:</span> <span class="s2">&quot;Alanine&quot;</span><span class="p">,</span>
</span><span id="L-280"><a href="#L-280"><span class="linenos"> 280</span></a>    <span class="s2">&quot;GAT&quot;</span><span class="p">:</span> <span class="s2">&quot;Aspartic acid&quot;</span><span class="p">,</span>
</span><span id="L-281"><a href="#L-281"><span class="linenos"> 281</span></a>    <span class="s2">&quot;GAC&quot;</span><span class="p">:</span> <span class="s2">&quot;Aspartic acid&quot;</span><span class="p">,</span>
</span><span id="L-282"><a href="#L-282"><span class="linenos"> 282</span></a>    <span class="s2">&quot;GGT&quot;</span><span class="p">:</span> <span class="s2">&quot;Glycine&quot;</span><span class="p">,</span>
</span><span id="L-283"><a href="#L-283"><span class="linenos"> 283</span></a>    <span class="s2">&quot;GGC&quot;</span><span class="p">:</span> <span class="s2">&quot;Glycine&quot;</span><span class="p">,</span>
</span><span id="L-284"><a href="#L-284"><span class="linenos"> 284</span></a>    <span class="s2">&quot;GTA&quot;</span><span class="p">:</span> <span class="s2">&quot;Valine&quot;</span><span class="p">,</span>
</span><span id="L-285"><a href="#L-285"><span class="linenos"> 285</span></a>    <span class="s2">&quot;GTG&quot;</span><span class="p">:</span> <span class="s2">&quot;Valine&quot;</span><span class="p">,</span>
</span><span id="L-286"><a href="#L-286"><span class="linenos"> 286</span></a>    <span class="s2">&quot;GCA&quot;</span><span class="p">:</span> <span class="s2">&quot;Alanine&quot;</span><span class="p">,</span>
</span><span id="L-287"><a href="#L-287"><span class="linenos"> 287</span></a>    <span class="s2">&quot;GCG&quot;</span><span class="p">:</span> <span class="s2">&quot;Alanine&quot;</span><span class="p">,</span>
</span><span id="L-288"><a href="#L-288"><span class="linenos"> 288</span></a>    <span class="s2">&quot;GAA&quot;</span><span class="p">:</span> <span class="s2">&quot;Glutamic acid&quot;</span><span class="p">,</span>
</span><span id="L-289"><a href="#L-289"><span class="linenos"> 289</span></a>    <span class="s2">&quot;GAG&quot;</span><span class="p">:</span> <span class="s2">&quot;Glutamic acid&quot;</span><span class="p">,</span>
</span><span id="L-290"><a href="#L-290"><span class="linenos"> 290</span></a>    <span class="s2">&quot;GGA&quot;</span><span class="p">:</span> <span class="s2">&quot;Glycine&quot;</span><span class="p">,</span>
</span><span id="L-291"><a href="#L-291"><span class="linenos"> 291</span></a>    <span class="s2">&quot;GGG&quot;</span><span class="p">:</span> <span class="s2">&quot;Glycine&quot;</span><span class="p">,</span>
</span><span id="L-292"><a href="#L-292"><span class="linenos"> 292</span></a><span class="p">}</span>
</span><span id="L-293"><a href="#L-293"><span class="linenos"> 293</span></a>
</span><span id="L-294"><a href="#L-294"><span class="linenos"> 294</span></a><span class="n">NUCLEOTIDE_TRANSLATE</span> <span class="o">=</span> <span class="p">{</span>
</span><span id="L-295"><a href="#L-295"><span class="linenos"> 295</span></a>    <span class="s2">&quot;T&quot;</span><span class="p">:</span> <span class="s2">&quot;A&quot;</span><span class="p">,</span>
</span><span id="L-296"><a href="#L-296"><span class="linenos"> 296</span></a>    <span class="s2">&quot;A&quot;</span><span class="p">:</span> <span class="s2">&quot;T&quot;</span><span class="p">,</span>
</span><span id="L-297"><a href="#L-297"><span class="linenos"> 297</span></a>    <span class="s2">&quot;G&quot;</span><span class="p">:</span> <span class="s2">&quot;C&quot;</span><span class="p">,</span>
</span><span id="L-298"><a href="#L-298"><span class="linenos"> 298</span></a>    <span class="s2">&quot;C&quot;</span><span class="p">:</span> <span class="s2">&quot;G&quot;</span><span class="p">,</span>
</span><span id="L-299"><a href="#L-299"><span class="linenos"> 299</span></a><span class="p">}</span>
</span><span id="L-300"><a href="#L-300"><span class="linenos"> 300</span></a>
</span><span id="L-301"><a href="#L-301"><span class="linenos"> 301</span></a>
</span><span id="L-302"><a href="#L-302"><span class="linenos"> 302</span></a><span class="c1"># The HGVSRegex class is used for working with regular expressions related to the Human Genome</span>
</span><span id="L-303"><a href="#L-303"><span class="linenos"> 303</span></a><span class="c1"># Variation Society (HGVS) nomenclature.</span>
</span><span id="L-304"><a href="#L-304"><span class="linenos"> 304</span></a><span class="k">class</span> <span class="nc">HGVSRegex</span><span class="p">(</span><span class="nb">object</span><span class="p">):</span>
</span><span id="L-305"><a href="#L-305"><span class="linenos"> 305</span></a><span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-306"><a href="#L-306"><span class="linenos"> 306</span></a><span class="sd">    All regular expression for HGVS names.</span>
</span><span id="L-307"><a href="#L-307"><span class="linenos"> 307</span></a><span class="sd">    &quot;&quot;&quot;</span>
</span><span id="L-308"><a href="#L-308"><span class="linenos"> 308</span></a>
</span><span id="L-309"><a href="#L-309"><span class="linenos"> 309</span></a>    <span class="c1"># DNA syntax</span>
</span><span id="L-310"><a href="#L-310"><span class="linenos"> 310</span></a>    <span class="c1"># http://www.hgvs.org/mutnomen/standards.html#nucleotide</span>
</span><span id="L-311"><a href="#L-311"><span class="linenos"> 311</span></a>    <span class="n">BASE</span> <span class="o">=</span> <span class="sa">r</span><span class="s2">&quot;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]|\d+&quot;</span>
</span><span id="L-312"><a href="#L-312"><span class="linenos"> 312</span></a>    <span class="n">BASES</span> <span class="o">=</span> <span class="sa">r</span><span class="s2">&quot;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\d+&quot;</span>
</span><span id="L-313"><a href="#L-313"><span class="linenos"> 313</span></a>    <span class="n">DNA_REF</span> <span class="o">=</span> <span class="s2">&quot;(?P&lt;ref&gt;&quot;</span> <span class="o">+</span> <span class="n">BASES</span> <span class="o">+</span> <span class="s2">&quot;)&quot;</span>
</span><span id="L-314"><a href="#L-314"><span class="linenos"> 314</span></a>    <span class="n">DNA_ALT</span> <span class="o">=</span> <span class="s2">&quot;(?P&lt;alt&gt;&quot;</span> <span class="o">+</span> <span class="n">BASES</span> <span class="o">+</span> <span class="s2">&quot;)&quot;</span>
</span><span id="L-315"><a href="#L-315"><span class="linenos"> 315</span></a>
</span><span id="L-316"><a href="#L-316"><span class="linenos"> 316</span></a>    <span class="c1"># Mutation types</span>
</span><span id="L-317"><a href="#L-317"><span class="linenos"> 317</span></a>    <span class="n">EQUAL</span> <span class="o">=</span> <span class="s2">&quot;(?P&lt;mutation_type&gt;=)&quot;</span>
</span><span id="L-318"><a href="#L-318"><span class="linenos"> 318</span></a>    <span class="n">SUB</span> <span class="o">=</span> <span class="s2">&quot;(?P&lt;mutation_type&gt;&gt;)&quot;</span>
</span><span id="L-319"><a href="#L-319"><span class="linenos"> 319</span></a>    <span class="n">INS</span> <span class="o">=</span> <span class="s2">&quot;(?P&lt;mutation_type&gt;ins)&quot;</span>
</span><span id="L-320"><a href="#L-320"><span class="linenos"> 320</span></a>    <span class="n">DEL</span> <span class="o">=</span> <span class="s2">&quot;(?P&lt;mutation_type&gt;del)&quot;</span>
</span><span id="L-321"><a href="#L-321"><span class="linenos"> 321</span></a>    <span class="n">DUP</span> <span class="o">=</span> <span class="s2">&quot;(?P&lt;mutation_type&gt;dup)&quot;</span>
</span><span id="L-322"><a href="#L-322"><span class="linenos"> 322</span></a>
</span><span id="L-323"><a href="#L-323"><span class="linenos"> 323</span></a>    <span class="c1"># Simple coordinate syntax</span>
</span><span id="L-324"><a href="#L-324"><span class="linenos"> 324</span></a>    <span class="n">COORD_START</span> <span class="o">=</span> <span class="sa">r</span><span class="s2">&quot;(?P&lt;start&gt;\d+)&quot;</span>
</span><span id="L-325"><a href="#L-325"><span class="linenos"> 325</span></a>    <span class="n">COORD_END</span> <span class="o">=</span> <span class="sa">r</span><span class="s2">&quot;(?P&lt;end&gt;\d+)&quot;</span>
</span><span id="L-326"><a href="#L-326"><span class="linenos"> 326</span></a>    <span class="n">COORD_RANGE</span> <span class="o">=</span> <span class="n">COORD_START</span> <span class="o">+</span> <span class="s2">&quot;_&quot;</span> <span class="o">+</span> <span class="n">COORD_END</span>
</span><span id="L-327"><a href="#L-327"><span class="linenos"> 327</span></a>
</span><span id="L-328"><a href="#L-328"><span class="linenos"> 328</span></a>    <span class="c1"># cDNA coordinate syntax</span>
</span><span id="L-329"><a href="#L-329"><span class="linenos"> 329</span></a>    <span class="n">CDNA_COORD</span> <span class="o">=</span> <span class="p">(</span>
</span><span id="L-330"><a href="#L-330"><span class="linenos"> 330</span></a>        <span class="sa">r</span><span class="s2">&quot;(?P&lt;coord_prefix&gt;|-|\*)(?P&lt;coord&gt;\d+)&quot;</span>
</span><span id="L-331"><a href="#L-331"><span class="linenos"> 331</span></a>        <span class="sa">r</span><span class="s2">&quot;((?P&lt;offset_prefix&gt;-|\+)(?P&lt;offset&gt;\d+))?&quot;</span>
</span><span id="L-332"><a href="#L-332"><span class="linenos"> 332</span></a>    <span class="p">)</span>
</span><span id="L-333"><a href="#L-333"><span class="linenos"> 333</span></a>    <span class="n">CDNA_START</span> <span class="o">=</span> <span class="p">(</span>
</span><span id="L-334"><a href="#L-334"><span class="linenos"> 334</span></a>        <span class="sa">r</span><span class="s2">&quot;(?P&lt;start&gt;(?P&lt;start_coord_prefix&gt;|-|\*)(?P&lt;start_coord&gt;\d+)&quot;</span>
</span><span id="L-335"><a href="#L-335"><span class="linenos"> 335</span></a>        <span class="sa">r</span><span class="s2">&quot;((?P&lt;start_offset_prefix&gt;-|\+)(?P&lt;start_offset&gt;\d+))?)&quot;</span>
</span><span id="L-336"><a href="#L-336"><span class="linenos"> 336</span></a>    <span class="p">)</span>
</span><span id="L-337"><a href="#L-337"><span class="linenos"> 337</span></a>    <span class="n">CDNA_END</span> <span class="o">=</span> <span class="p">(</span>
</span><span id="L-338"><a href="#L-338"><span class="linenos"> 338</span></a>        <span class="sa">r</span><span class="s2">&quot;(?P&lt;end&gt;(?P&lt;end_coord_prefix&gt;|-|\*)(?P&lt;end_coord&gt;\d+)&quot;</span>
</span><span id="L-339"><a href="#L-339"><span class="linenos"> 339</span></a>        <span class="sa">r</span><span class="s2">&quot;((?P&lt;end_offset_prefix&gt;-|\+)(?P&lt;end_offset&gt;\d+))?)&quot;</span>
</span><span id="L-340"><a href="#L-340"><span class="linenos"> 340</span></a>    <span class="p">)</span>
</span><span id="L-341"><a href="#L-341"><span class="linenos"> 341</span></a>    <span class="n">CDNA_RANGE</span> <span class="o">=</span> <span class="n">CDNA_START</span> <span class="o">+</span> <span class="s2">&quot;_&quot;</span> <span class="o">+</span> <span class="n">CDNA_END</span>
</span><span id="L-342"><a href="#L-342"><span class="linenos"> 342</span></a>
</span><span id="L-343"><a href="#L-343"><span class="linenos"> 343</span></a>    <span class="c1"># cDNA allele syntax</span>
</span><span id="L-344"><a href="#L-344"><span class="linenos"> 344</span></a>    <span class="n">CDNA_ALLELE</span> <span class="o">=</span> <span class="p">[</span>
</span><span id="L-345"><a href="#L-345"><span class="linenos"> 345</span></a>        <span class="c1"># No change</span>
</span><span id="L-346"><a href="#L-346"><span class="linenos"> 346</span></a>        <span class="n">CDNA_START</span> <span class="o">+</span> <span class="n">EQUAL</span><span class="p">,</span>
</span><span id="L-347"><a href="#L-347"><span class="linenos"> 347</span></a>        <span class="n">CDNA_START</span> <span class="o">+</span> <span class="n">DNA_REF</span> <span class="o">+</span> <span class="n">EQUAL</span><span class="p">,</span>
</span><span id="L-348"><a href="#L-348"><span class="linenos"> 348</span></a>        <span class="c1"># Substitution</span>
</span><span id="L-349"><a href="#L-349"><span class="linenos"> 349</span></a>        <span class="n">CDNA_START</span> <span class="o">+</span> <span class="n">DNA_REF</span> <span class="o">+</span> <span class="n">SUB</span> <span class="o">+</span> <span class="n">DNA_ALT</span><span class="p">,</span>
</span><span id="L-350"><a href="#L-350"><span class="linenos"> 350</span></a>        <span class="c1"># 1bp insertion, deletion, duplication</span>
</span><span id="L-351"><a href="#L-351"><span class="linenos"> 351</span></a>        <span class="n">CDNA_START</span> <span class="o">+</span> <span class="n">INS</span> <span class="o">+</span> <span class="n">DNA_ALT</span><span class="p">,</span>
</span><span id="L-352"><a href="#L-352"><span class="linenos"> 352</span></a>        <span class="n">CDNA_START</span> <span class="o">+</span> <span class="n">DEL</span> <span class="o">+</span> <span class="n">DNA_REF</span><span class="p">,</span>
</span><span id="L-353"><a href="#L-353"><span class="linenos"> 353</span></a>        <span class="n">CDNA_START</span> <span class="o">+</span> <span class="n">DUP</span> <span class="o">+</span> <span class="n">DNA_REF</span><span class="p">,</span>
</span><span id="L-354"><a href="#L-354"><span class="linenos"> 354</span></a>        <span class="n">CDNA_START</span> <span class="o">+</span> <span class="n">DEL</span><span class="p">,</span>
</span><span id="L-355"><a href="#L-355"><span class="linenos"> 355</span></a>        <span class="n">CDNA_START</span> <span class="o">+</span> <span class="n">DUP</span><span class="p">,</span>
</span><span id="L-356"><a href="#L-356"><span class="linenos"> 356</span></a>        <span class="c1"># Insertion, deletion, duplication</span>
</span><span id="L-357"><a href="#L-357"><span class="linenos"> 357</span></a>        <span class="n">COORD_RANGE</span> <span class="o">+</span> <span class="n">EQUAL</span><span class="p">,</span>
</span><span id="L-358"><a href="#L-358"><span class="linenos"> 358</span></a>        <span class="n">CDNA_RANGE</span> <span class="o">+</span> <span class="n">INS</span> <span class="o">+</span> <span class="n">DNA_ALT</span><span class="p">,</span>
</span><span id="L-359"><a href="#L-359"><span class="linenos"> 359</span></a>        <span class="n">CDNA_RANGE</span> <span class="o">+</span> <span class="n">DEL</span> <span class="o">+</span> <span class="n">DNA_REF</span><span class="p">,</span>
</span><span id="L-360"><a href="#L-360"><span class="linenos"> 360</span></a>        <span class="n">CDNA_RANGE</span> <span class="o">+</span> <span class="n">DUP</span> <span class="o">+</span> <span class="n">DNA_REF</span><span class="p">,</span>
</span><span id="L-361"><a href="#L-361"><span class="linenos"> 361</span></a>        <span class="n">CDNA_RANGE</span> <span class="o">+</span> <span class="n">DEL</span><span class="p">,</span>
</span><span id="L-362"><a href="#L-362"><span class="linenos"> 362</span></a>        <span class="n">CDNA_RANGE</span> <span class="o">+</span> <span class="n">DUP</span><span class="p">,</span>
</span><span id="L-363"><a href="#L-363"><span class="linenos"> 363</span></a>        <span class="c1"># Indels</span>
</span><span id="L-364"><a href="#L-364"><span class="linenos"> 364</span></a>        <span class="s2">&quot;(?P&lt;delins&gt;&quot;</span> <span class="o">+</span> <span class="n">CDNA_START</span> <span class="o">+</span> <span class="s2">&quot;del&quot;</span> <span class="o">+</span> <span class="n">DNA_REF</span> <span class="o">+</span> <span class="s2">&quot;ins&quot;</span> <span class="o">+</span> <span class="n">DNA_ALT</span> <span class="o">+</span> <span class="s2">&quot;)&quot;</span><span class="p">,</span>
</span><span id="L-365"><a href="#L-365"><span class="linenos"> 365</span></a>        <span class="s2">&quot;(?P&lt;delins&gt;&quot;</span> <span class="o">+</span> <span class="n">CDNA_RANGE</span> <span class="o">+</span> <span class="s2">&quot;del&quot;</span> <span class="o">+</span> <span class="n">DNA_REF</span> <span class="o">+</span> <span class="s2">&quot;ins&quot;</span> <span class="o">+</span> <span class="n">DNA_ALT</span> <span class="o">+</span> <span class="s2">&quot;)&quot;</span><span class="p">,</span>
</span><span id="L-366"><a href="#L-366"><span class="linenos"> 366</span></a>        <span class="s2">&quot;(?P&lt;delins&gt;&quot;</span> <span class="o">+</span> <span class="n">CDNA_START</span> <span class="o">+</span> <span class="s2">&quot;delins&quot;</span> <span class="o">+</span> <span class="n">DNA_ALT</span> <span class="o">+</span> <span class="s2">&quot;)&quot;</span><span class="p">,</span>
</span><span id="L-367"><a href="#L-367"><span class="linenos"> 367</span></a>        <span class="s2">&quot;(?P&lt;delins&gt;&quot;</span> <span class="o">+</span> <span class="n">CDNA_RANGE</span> <span class="o">+</span> <span class="s2">&quot;delins&quot;</span> <span class="o">+</span> <span class="n">DNA_ALT</span> <span class="o">+</span> <span class="s2">&quot;)&quot;</span><span class="p">,</span>
</span><span id="L-368"><a href="#L-368"><span class="linenos"> 368</span></a>    <span class="p">]</span>
</span><span id="L-369"><a href="#L-369"><span class="linenos"> 369</span></a>
</span><span id="L-370"><a href="#L-370"><span class="linenos"> 370</span></a>    <span class="n">CDNA_ALLELE_REGEXES</span> <span class="o">=</span> <span class="p">[</span><span class="n">re</span><span class="o">.</span><span class="n">compile</span><span class="p">(</span><span class="s2">&quot;^&quot;</span> <span class="o">+</span> <span class="n">regex</span> <span class="o">+</span> <span class="s2">&quot;$&quot;</span><span class="p">)</span> <span class="k">for</span> <span class="n">regex</span> <span class="ow">in</span> <span class="n">CDNA_ALLELE</span><span class="p">]</span>
</span><span id="L-371"><a href="#L-371"><span class="linenos"> 371</span></a>
</span><span id="L-372"><a href="#L-372"><span class="linenos"> 372</span></a>    <span class="c1"># Peptide syntax</span>
</span><span id="L-373"><a href="#L-373"><span class="linenos"> 373</span></a>    <span class="n">PEP</span> <span class="o">=</span> <span class="s2">&quot;([A-Z]([a-z]</span><span class="si">{2}</span><span class="s2">))+&quot;</span>
</span><span id="L-374"><a href="#L-374"><span class="linenos"> 374</span></a>    <span class="n">PEP_REF</span> <span class="o">=</span> <span class="s2">&quot;(?P&lt;ref&gt;&quot;</span> <span class="o">+</span> <span class="n">PEP</span> <span class="o">+</span> <span class="s2">&quot;)&quot;</span>
</span><span id="L-375"><a href="#L-375"><span class="linenos"> 375</span></a>    <span class="n">PEP_REF2</span> <span class="o">=</span> <span class="s2">&quot;(?P&lt;ref2&gt;&quot;</span> <span class="o">+</span> <span class="n">PEP</span> <span class="o">+</span> <span class="s2">&quot;)&quot;</span>
</span><span id="L-376"><a href="#L-376"><span class="linenos"> 376</span></a>    <span class="n">PEP_ALT</span> <span class="o">=</span> <span class="s2">&quot;(?P&lt;alt&gt;&quot;</span> <span class="o">+</span> <span class="n">PEP</span> <span class="o">+</span> <span class="s2">&quot;)&quot;</span>
</span><span id="L-377"><a href="#L-377"><span class="linenos"> 377</span></a>
</span><span id="L-378"><a href="#L-378"><span class="linenos"> 378</span></a>    <span class="n">PEP_EXTRA</span> <span class="o">=</span> <span class="sa">r</span><span class="s2">&quot;(?P&lt;extra&gt;(|=|\?)(|fs))&quot;</span>
</span><span id="L-379"><a href="#L-379"><span class="linenos"> 379</span></a>
</span><span id="L-380"><a href="#L-380"><span class="linenos"> 380</span></a>    <span class="c1"># Peptide allele syntax</span>
</span><span id="L-381"><a href="#L-381"><span class="linenos"> 381</span></a>    <span class="n">PEP_ALLELE</span> <span class="o">=</span> <span class="p">[</span>
</span><span id="L-382"><a href="#L-382"><span class="linenos"> 382</span></a>        <span class="c1"># No peptide change</span>
</span><span id="L-383"><a href="#L-383"><span class="linenos"> 383</span></a>        <span class="c1"># Example: Glu1161=</span>
</span><span id="L-384"><a href="#L-384"><span class="linenos"> 384</span></a>        <span class="n">PEP_REF</span> <span class="o">+</span> <span class="n">COORD_START</span> <span class="o">+</span> <span class="n">PEP_EXTRA</span><span class="p">,</span>
</span><span id="L-385"><a href="#L-385"><span class="linenos"> 385</span></a>        <span class="c1"># Peptide change</span>
</span><span id="L-386"><a href="#L-386"><span class="linenos"> 386</span></a>        <span class="c1"># Example: Glu1161Ser</span>
</span><span id="L-387"><a href="#L-387"><span class="linenos"> 387</span></a>        <span class="n">PEP_REF</span> <span class="o">+</span> <span class="n">COORD_START</span> <span class="o">+</span> <span class="n">PEP_ALT</span> <span class="o">+</span> <span class="n">PEP_EXTRA</span><span class="p">,</span>
</span><span id="L-388"><a href="#L-388"><span class="linenos"> 388</span></a>        <span class="c1"># Peptide indel</span>
</span><span id="L-389"><a href="#L-389"><span class="linenos"> 389</span></a>        <span class="c1"># Example: Glu1161_Ser1164?fs</span>
</span><span id="L-390"><a href="#L-390"><span class="linenos"> 390</span></a>        <span class="s2">&quot;(?P&lt;delins&gt;&quot;</span>
</span><span id="L-391"><a href="#L-391"><span class="linenos"> 391</span></a>        <span class="o">+</span> <span class="n">PEP_REF</span>
</span><span id="L-392"><a href="#L-392"><span class="linenos"> 392</span></a>        <span class="o">+</span> <span class="n">COORD_START</span>
</span><span id="L-393"><a href="#L-393"><span class="linenos"> 393</span></a>        <span class="o">+</span> <span class="s2">&quot;_&quot;</span>
</span><span id="L-394"><a href="#L-394"><span class="linenos"> 394</span></a>        <span class="o">+</span> <span class="n">PEP_REF2</span>
</span><span id="L-395"><a href="#L-395"><span class="linenos"> 395</span></a>        <span class="o">+</span> <span class="n">COORD_END</span>
</span><span id="L-396"><a href="#L-396"><span class="linenos"> 396</span></a>        <span class="o">+</span> <span class="n">PEP_EXTRA</span>
</span><span id="L-397"><a href="#L-397"><span class="linenos"> 397</span></a>        <span class="o">+</span> <span class="s2">&quot;)&quot;</span><span class="p">,</span>
</span><span id="L-398"><a href="#L-398"><span class="linenos"> 398</span></a>        <span class="s2">&quot;(?P&lt;delins&gt;&quot;</span>
</span><span id="L-399"><a href="#L-399"><span class="linenos"> 399</span></a>        <span class="o">+</span> <span class="n">PEP_REF</span>
</span><span id="L-400"><a href="#L-400"><span class="linenos"> 400</span></a>        <span class="o">+</span> <span class="n">COORD_START</span>
</span><span id="L-401"><a href="#L-401"><span class="linenos"> 401</span></a>        <span class="o">+</span> <span class="s2">&quot;_&quot;</span>
</span><span id="L-402"><a href="#L-402"><span class="linenos"> 402</span></a>        <span class="o">+</span> <span class="n">PEP_REF2</span>
</span><span id="L-403"><a href="#L-403"><span class="linenos"> 403</span></a>        <span class="o">+</span> <span class="n">COORD_END</span>
</span><span id="L-404"><a href="#L-404"><span class="linenos"> 404</span></a>        <span class="o">+</span> <span class="n">PEP_ALT</span>
</span><span id="L-405"><a href="#L-405"><span class="linenos"> 405</span></a>        <span class="o">+</span> <span class="n">PEP_EXTRA</span>
</span><span id="L-406"><a href="#L-406"><span class="linenos"> 406</span></a>        <span class="o">+</span> <span class="s2">&quot;)&quot;</span><span class="p">,</span>
</span><span id="L-407"><a href="#L-407"><span class="linenos"> 407</span></a>    <span class="p">]</span>
</span><span id="L-408"><a href="#L-408"><span class="linenos"> 408</span></a>
</span><span id="L-409"><a href="#L-409"><span class="linenos"> 409</span></a>    <span class="n">PEP_ALLELE_REGEXES</span> <span class="o">=</span> <span class="p">[</span><span class="n">re</span><span class="o">.</span><span class="n">compile</span><span class="p">(</span><span class="s2">&quot;^&quot;</span> <span class="o">+</span> <span class="n">regex</span> <span class="o">+</span> <span class="s2">&quot;$&quot;</span><span class="p">)</span> <span class="k">for</span> <span class="n">regex</span> <span class="ow">in</span> <span class="n">PEP_ALLELE</span><span class="p">]</span>
</span><span id="L-410"><a href="#L-410"><span class="linenos"> 410</span></a>
</span><span id="L-411"><a href="#L-411"><span class="linenos"> 411</span></a>    <span class="c1"># Genomic allele syntax</span>
</span><span id="L-412"><a href="#L-412"><span class="linenos"> 412</span></a>    <span class="n">GENOMIC_ALLELE</span> <span class="o">=</span> <span class="p">[</span>
</span><span id="L-413"><a href="#L-413"><span class="linenos"> 413</span></a>        <span class="c1"># No change</span>
</span><span id="L-414"><a href="#L-414"><span class="linenos"> 414</span></a>        <span class="n">COORD_START</span> <span class="o">+</span> <span class="n">EQUAL</span><span class="p">,</span>
</span><span id="L-415"><a href="#L-415"><span class="linenos"> 415</span></a>        <span class="n">COORD_START</span> <span class="o">+</span> <span class="n">DNA_REF</span> <span class="o">+</span> <span class="n">EQUAL</span><span class="p">,</span>
</span><span id="L-416"><a href="#L-416"><span class="linenos"> 416</span></a>        <span class="c1"># Substitution</span>
</span><span id="L-417"><a href="#L-417"><span class="linenos"> 417</span></a>        <span class="n">COORD_START</span> <span class="o">+</span> <span class="n">DNA_REF</span> <span class="o">+</span> <span class="n">SUB</span> <span class="o">+</span> <span class="n">DNA_ALT</span><span class="p">,</span>
</span><span id="L-418"><a href="#L-418"><span class="linenos"> 418</span></a>        <span class="c1"># 1bp insertion, deletion, duplication</span>
</span><span id="L-419"><a href="#L-419"><span class="linenos"> 419</span></a>        <span class="n">COORD_START</span> <span class="o">+</span> <span class="n">INS</span> <span class="o">+</span> <span class="n">DNA_ALT</span><span class="p">,</span>
</span><span id="L-420"><a href="#L-420"><span class="linenos"> 420</span></a>        <span class="n">COORD_START</span> <span class="o">+</span> <span class="n">DEL</span> <span class="o">+</span> <span class="n">DNA_REF</span><span class="p">,</span>
</span><span id="L-421"><a href="#L-421"><span class="linenos"> 421</span></a>        <span class="n">COORD_START</span> <span class="o">+</span> <span class="n">DUP</span> <span class="o">+</span> <span class="n">DNA_REF</span><span class="p">,</span>
</span><span id="L-422"><a href="#L-422"><span class="linenos"> 422</span></a>        <span class="n">COORD_START</span> <span class="o">+</span> <span class="n">DEL</span><span class="p">,</span>
</span><span id="L-423"><a href="#L-423"><span class="linenos"> 423</span></a>        <span class="n">COORD_START</span> <span class="o">+</span> <span class="n">DUP</span><span class="p">,</span>
</span><span id="L-424"><a href="#L-424"><span class="linenos"> 424</span></a>        <span class="c1"># Insertion, deletion, duplication</span>
</span><span id="L-425"><a href="#L-425"><span class="linenos"> 425</span></a>        <span class="n">COORD_RANGE</span> <span class="o">+</span> <span class="n">EQUAL</span><span class="p">,</span>
</span><span id="L-426"><a href="#L-426"><span class="linenos"> 426</span></a>        <span class="n">COORD_RANGE</span> <span class="o">+</span> <span class="n">INS</span> <span class="o">+</span> <span class="n">DNA_ALT</span><span class="p">,</span>
</span><span id="L-427"><a href="#L-427"><span class="linenos"> 427</span></a>        <span class="n">COORD_RANGE</span> <span class="o">+</span> <span class="n">DEL</span> <span class="o">+</span> <span class="n">DNA_REF</span><span class="p">,</span>
</span><span id="L-428"><a href="#L-428"><span class="linenos"> 428</span></a>        <span class="n">COORD_RANGE</span> <span class="o">+</span> <span class="n">DUP</span> <span class="o">+</span> <span class="n">DNA_REF</span><span class="p">,</span>
</span><span id="L-429"><a href="#L-429"><span class="linenos"> 429</span></a>        <span class="n">COORD_RANGE</span> <span class="o">+</span> <span class="n">DEL</span><span class="p">,</span>
</span><span id="L-430"><a href="#L-430"><span class="linenos"> 430</span></a>        <span class="n">COORD_RANGE</span> <span class="o">+</span> <span class="n">DUP</span><span class="p">,</span>
</span><span id="L-431"><a href="#L-431"><span class="linenos"> 431</span></a>        <span class="c1"># Indels</span>
</span><span id="L-432"><a href="#L-432"><span class="linenos"> 432</span></a>        <span class="s2">&quot;(?P&lt;delins&gt;&quot;</span> <span class="o">+</span> <span class="n">COORD_START</span> <span class="o">+</span> <span class="s2">&quot;del&quot;</span> <span class="o">+</span> <span class="n">DNA_REF</span> <span class="o">+</span> <span class="s2">&quot;ins&quot;</span> <span class="o">+</span> <span class="n">DNA_ALT</span> <span class="o">+</span> <span class="s2">&quot;)&quot;</span><span class="p">,</span>
</span><span id="L-433"><a href="#L-433"><span class="linenos"> 433</span></a>        <span class="s2">&quot;(?P&lt;delins&gt;&quot;</span> <span class="o">+</span> <span class="n">COORD_RANGE</span> <span class="o">+</span> <span class="s2">&quot;del&quot;</span> <span class="o">+</span> <span class="n">DNA_REF</span> <span class="o">+</span> <span class="s2">&quot;ins&quot;</span> <span class="o">+</span> <span class="n">DNA_ALT</span> <span class="o">+</span> <span class="s2">&quot;)&quot;</span><span class="p">,</span>
</span><span id="L-434"><a href="#L-434"><span class="linenos"> 434</span></a>        <span class="s2">&quot;(?P&lt;delins&gt;&quot;</span> <span class="o">+</span> <span class="n">COORD_START</span> <span class="o">+</span> <span class="s2">&quot;delins&quot;</span> <span class="o">+</span> <span class="n">DNA_ALT</span> <span class="o">+</span> <span class="s2">&quot;)&quot;</span><span class="p">,</span>
</span><span id="L-435"><a href="#L-435"><span class="linenos"> 435</span></a>        <span class="s2">&quot;(?P&lt;delins&gt;&quot;</span> <span class="o">+</span> <span class="n">COORD_RANGE</span> <span class="o">+</span> <span class="s2">&quot;delins&quot;</span> <span class="o">+</span> <span class="n">DNA_ALT</span> <span class="o">+</span> <span class="s2">&quot;)&quot;</span><span class="p">,</span>
</span><span id="L-436"><a href="#L-436"><span class="linenos"> 436</span></a>    <span class="p">]</span>
</span><span id="L-437"><a href="#L-437"><span class="linenos"> 437</span></a>
</span><span id="L-438"><a href="#L-438"><span class="linenos"> 438</span></a>    <span class="n">GENOMIC_ALLELE_REGEXES</span> <span class="o">=</span> <span class="p">[</span><span class="n">re</span><span class="o">.</span><span class="n">compile</span><span class="p">(</span><span class="s2">&quot;^&quot;</span> <span class="o">+</span> <span class="n">regex</span> <span class="o">+</span> <span class="s2">&quot;$&quot;</span><span class="p">)</span> <span class="k">for</span> <span class="n">regex</span> <span class="ow">in</span> <span class="n">GENOMIC_ALLELE</span><span class="p">]</span>
</span><span id="L-439"><a href="#L-439"><span class="linenos"> 439</span></a>
</span><span id="L-440"><a href="#L-440"><span class="linenos"> 440</span></a>
</span><span id="L-441"><a href="#L-441"><span class="linenos"> 441</span></a><span class="c1"># The RefSeq standard for naming contigs/transcripts/proteins:</span>
</span><span id="L-442"><a href="#L-442"><span class="linenos"> 442</span></a><span class="c1"># http://www.ncbi.nlm.nih.gov/books/NBK21091/table/ch18.T.refseq_accession_numbers_and_mole/?report=objectonly  # nopep8</span>
</span><span id="L-443"><a href="#L-443"><span class="linenos"> 443</span></a><span class="n">REFSEQ_PREFIXES</span> <span class="o">=</span> <span class="p">[</span>
</span><span id="L-444"><a href="#L-444"><span class="linenos"> 444</span></a>    <span class="p">(</span><span class="s2">&quot;AC_&quot;</span><span class="p">,</span> <span class="s2">&quot;genomic&quot;</span><span class="p">,</span> <span class="s2">&quot;Complete genomic molecule, usually alternate assembly&quot;</span><span class="p">),</span>
</span><span id="L-445"><a href="#L-445"><span class="linenos"> 445</span></a>    <span class="p">(</span><span class="s2">&quot;NC_&quot;</span><span class="p">,</span> <span class="s2">&quot;genomic&quot;</span><span class="p">,</span> <span class="s2">&quot;Complete genomic molecule, usually reference assembly&quot;</span><span class="p">),</span>
</span><span id="L-446"><a href="#L-446"><span class="linenos"> 446</span></a>    <span class="p">(</span><span class="s2">&quot;NG_&quot;</span><span class="p">,</span> <span class="s2">&quot;genomic&quot;</span><span class="p">,</span> <span class="s2">&quot;Incomplete genomic region&quot;</span><span class="p">),</span>
</span><span id="L-447"><a href="#L-447"><span class="linenos"> 447</span></a>    <span class="p">(</span><span class="s2">&quot;NT_&quot;</span><span class="p">,</span> <span class="s2">&quot;genomic&quot;</span><span class="p">,</span> <span class="s2">&quot;Contig or scaffold, clone-based or WGS&quot;</span><span class="p">),</span>
</span><span id="L-448"><a href="#L-448"><span class="linenos"> 448</span></a>    <span class="p">(</span><span class="s2">&quot;NW_&quot;</span><span class="p">,</span> <span class="s2">&quot;genomic&quot;</span><span class="p">,</span> <span class="s2">&quot;Contig or scaffold, primarily WGS&quot;</span><span class="p">),</span>
</span><span id="L-449"><a href="#L-449"><span class="linenos"> 449</span></a>    <span class="p">(</span><span class="s2">&quot;NS_&quot;</span><span class="p">,</span> <span class="s2">&quot;genomic&quot;</span><span class="p">,</span> <span class="s2">&quot;Environmental sequence&quot;</span><span class="p">),</span>
</span><span id="L-450"><a href="#L-450"><span class="linenos"> 450</span></a>    <span class="p">(</span><span class="s2">&quot;NZ_&quot;</span><span class="p">,</span> <span class="s2">&quot;genomic&quot;</span><span class="p">,</span> <span class="s2">&quot;Unfinished WGS&quot;</span><span class="p">),</span>
</span><span id="L-451"><a href="#L-451"><span class="linenos"> 451</span></a>    <span class="p">(</span><span class="s2">&quot;NM_&quot;</span><span class="p">,</span> <span class="s2">&quot;mRNA&quot;</span><span class="p">,</span> <span class="s2">&quot;&quot;</span><span class="p">),</span>
</span><span id="L-452"><a href="#L-452"><span class="linenos"> 452</span></a>    <span class="p">(</span><span class="s2">&quot;NR_&quot;</span><span class="p">,</span> <span class="s2">&quot;RNA&quot;</span><span class="p">,</span> <span class="s2">&quot;&quot;</span><span class="p">),</span>
</span><span id="L-453"><a href="#L-453"><span class="linenos"> 453</span></a>    <span class="p">(</span><span class="s2">&quot;XM_&quot;</span><span class="p">,</span> <span class="s2">&quot;mRNA&quot;</span><span class="p">,</span> <span class="s2">&quot;Predicted model&quot;</span><span class="p">),</span>
</span><span id="L-454"><a href="#L-454"><span class="linenos"> 454</span></a>    <span class="p">(</span><span class="s2">&quot;XR_&quot;</span><span class="p">,</span> <span class="s2">&quot;RNA&quot;</span><span class="p">,</span> <span class="s2">&quot;Predicted model&quot;</span><span class="p">),</span>
</span><span id="L-455"><a href="#L-455"><span class="linenos"> 455</span></a>    <span class="p">(</span><span class="s2">&quot;AP_&quot;</span><span class="p">,</span> <span class="s2">&quot;Protein&quot;</span><span class="p">,</span> <span class="s2">&quot;Annotated on AC_ alternate assembly&quot;</span><span class="p">),</span>
</span><span id="L-456"><a href="#L-456"><span class="linenos"> 456</span></a>    <span class="p">(</span><span class="s2">&quot;NP_&quot;</span><span class="p">,</span> <span class="s2">&quot;Protein&quot;</span><span class="p">,</span> <span class="s2">&quot;Associated with an NM_ or NC_ accession&quot;</span><span class="p">),</span>
</span><span id="L-457"><a href="#L-457"><span class="linenos"> 457</span></a>    <span class="p">(</span><span class="s2">&quot;YP_&quot;</span><span class="p">,</span> <span class="s2">&quot;Protein&quot;</span><span class="p">,</span> <span class="s2">&quot;&quot;</span><span class="p">),</span>
</span><span id="L-458"><a href="#L-458"><span class="linenos"> 458</span></a>    <span class="p">(</span><span class="s2">&quot;XP_&quot;</span><span class="p">,</span> <span class="s2">&quot;Protein&quot;</span><span class="p">,</span> <span class="s2">&quot;Predicted model, associated with an XM_ accession&quot;</span><span class="p">),</span>
</span><span id="L-459"><a href="#L-459"><span class="linenos"> 459</span></a>    <span class="p">(</span><span class="s2">&quot;ZP_&quot;</span><span class="p">,</span> <span class="s2">&quot;Protein&quot;</span><span class="p">,</span> <span class="s2">&quot;Predicted model, annotated on NZ_ genomic records&quot;</span><span class="p">),</span>
</span><span id="L-460"><a href="#L-460"><span class="linenos"> 460</span></a><span class="p">]</span>
</span><span id="L-461"><a href="#L-461"><span class="linenos"> 461</span></a>
</span><span id="L-462"><a href="#L-462"><span class="linenos"> 462</span></a>
</span><span id="L-463"><a href="#L-463"><span class="linenos"> 463</span></a><span class="n">REFSEQ_PREFIX_LOOKUP</span> <span class="o">=</span> <span class="nb">dict</span><span class="p">(</span>
</span><span id="L-464"><a href="#L-464"><span class="linenos"> 464</span></a>    <span class="p">(</span><span class="n">prefix</span><span class="p">,</span> <span class="p">(</span><span class="n">kind</span><span class="p">,</span> <span class="n">description</span><span class="p">))</span> <span class="k">for</span> <span class="n">prefix</span><span class="p">,</span> <span class="n">kind</span><span class="p">,</span> <span class="n">description</span> <span class="ow">in</span> <span class="n">REFSEQ_PREFIXES</span>
</span><span id="L-465"><a href="#L-465"><span class="linenos"> 465</span></a><span class="p">)</span>
</span><span id="L-466"><a href="#L-466"><span class="linenos"> 466</span></a>
</span><span id="L-467"><a href="#L-467"><span class="linenos"> 467</span></a>
</span><span id="L-468"><a href="#L-468"><span class="linenos"> 468</span></a><span class="k">def</span> <span class="nf">get_refseq_type</span><span class="p">(</span><span class="n">name</span><span class="p">:</span> <span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
</span><span id="L-469"><a href="#L-469"><span class="linenos"> 469</span></a><span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-470"><a href="#L-470"><span class="linenos"> 470</span></a><span class="sd">    The `get_refseq_type` function returns the RefSeq type for a given RefSeq name.</span>
</span><span id="L-471"><a href="#L-471"><span class="linenos"> 471</span></a>
</span><span id="L-472"><a href="#L-472"><span class="linenos"> 472</span></a><span class="sd">    :param name: The `name` parameter is a string representing a RefSeq name</span>
</span><span id="L-473"><a href="#L-473"><span class="linenos"> 473</span></a><span class="sd">    :type name: str</span>
</span><span id="L-474"><a href="#L-474"><span class="linenos"> 474</span></a><span class="sd">    :return: The function `get_refseq_type` returns the RefSeq type for a given RefSeq name.</span>
</span><span id="L-475"><a href="#L-475"><span class="linenos"> 475</span></a><span class="sd">    &quot;&quot;&quot;</span>
</span><span id="L-476"><a href="#L-476"><span class="linenos"> 476</span></a>
</span><span id="L-477"><a href="#L-477"><span class="linenos"> 477</span></a>    <span class="n">prefix</span> <span class="o">=</span> <span class="n">name</span><span class="p">[:</span><span class="mi">3</span><span class="p">]</span>
</span><span id="L-478"><a href="#L-478"><span class="linenos"> 478</span></a>    <span class="k">return</span> <span class="n">REFSEQ_PREFIX_LOOKUP</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="n">prefix</span><span class="p">,</span> <span class="p">(</span><span class="kc">None</span><span class="p">,</span> <span class="s2">&quot;&quot;</span><span class="p">))[</span><span class="mi">0</span><span class="p">]</span>
</span><span id="L-479"><a href="#L-479"><span class="linenos"> 479</span></a>
</span><span id="L-480"><a href="#L-480"><span class="linenos"> 480</span></a>
</span><span id="L-481"><a href="#L-481"><span class="linenos"> 481</span></a><span class="c1"># The InvalidHGVSName class is a subclass of ValueError and is used to represent an error when an</span>
</span><span id="L-482"><a href="#L-482"><span class="linenos"> 482</span></a><span class="c1"># invalid HGVS name is encountered.</span>
</span><span id="L-483"><a href="#L-483"><span class="linenos"> 483</span></a><span class="k">class</span> <span class="nc">InvalidHGVSName</span><span class="p">(</span><span class="ne">ValueError</span><span class="p">):</span>
</span><span id="L-484"><a href="#L-484"><span class="linenos"> 484</span></a>
</span><span id="L-485"><a href="#L-485"><span class="linenos"> 485</span></a>    <span class="k">def</span> <span class="fm">__init__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">name</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span><span class="p">,</span> <span class="n">part</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;name&quot;</span><span class="p">,</span> <span class="n">reason</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
</span><span id="L-486"><a href="#L-486"><span class="linenos"> 486</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-487"><a href="#L-487"><span class="linenos"> 487</span></a><span class="sd">        The function initializes an InvalidHGVSName object with a message, name, part, and reason.</span>
</span><span id="L-488"><a href="#L-488"><span class="linenos"> 488</span></a>
</span><span id="L-489"><a href="#L-489"><span class="linenos"> 489</span></a><span class="sd">        :param name: The name parameter is a string that represents the invalid HGVS name. It is the</span>
</span><span id="L-490"><a href="#L-490"><span class="linenos"> 490</span></a><span class="sd">        name that is considered invalid and does not meet the required criteria</span>
</span><span id="L-491"><a href="#L-491"><span class="linenos"> 491</span></a><span class="sd">        :type name: str</span>
</span><span id="L-492"><a href="#L-492"><span class="linenos"> 492</span></a><span class="sd">        :param part: The &quot;part&quot; parameter represents the part of the HGVS (Human Genome Variation</span>
</span><span id="L-493"><a href="#L-493"><span class="linenos"> 493</span></a><span class="sd">        Society) name that is invalid. It is used to provide more specific information about the error</span>
</span><span id="L-494"><a href="#L-494"><span class="linenos"> 494</span></a><span class="sd">        that occurred, defaults to name</span>
</span><span id="L-495"><a href="#L-495"><span class="linenos"> 495</span></a><span class="sd">        :type part: str (optional)</span>
</span><span id="L-496"><a href="#L-496"><span class="linenos"> 496</span></a><span class="sd">        :param reason: The &quot;reason&quot; parameter is an optional argument that provides additional</span>
</span><span id="L-497"><a href="#L-497"><span class="linenos"> 497</span></a><span class="sd">        information or context for why the HGVS name is considered invalid. It can be used to provide</span>
</span><span id="L-498"><a href="#L-498"><span class="linenos"> 498</span></a><span class="sd">        specific details about the error or to explain why the name does not meet the required criteria</span>
</span><span id="L-499"><a href="#L-499"><span class="linenos"> 499</span></a><span class="sd">        :type reason: str</span>
</span><span id="L-500"><a href="#L-500"><span class="linenos"> 500</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="L-501"><a href="#L-501"><span class="linenos"> 501</span></a>
</span><span id="L-502"><a href="#L-502"><span class="linenos"> 502</span></a>        <span class="k">if</span> <span class="n">name</span><span class="p">:</span>
</span><span id="L-503"><a href="#L-503"><span class="linenos"> 503</span></a>            <span class="n">message</span> <span class="o">=</span> <span class="s1">&#39;Invalid HGVS </span><span class="si">%s</span><span class="s1"> &quot;</span><span class="si">%s</span><span class="s1">&quot;&#39;</span> <span class="o">%</span> <span class="p">(</span><span class="n">part</span><span class="p">,</span> <span class="n">name</span><span class="p">)</span>
</span><span id="L-504"><a href="#L-504"><span class="linenos"> 504</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="L-505"><a href="#L-505"><span class="linenos"> 505</span></a>            <span class="n">message</span> <span class="o">=</span> <span class="s2">&quot;Invalid HGVS </span><span class="si">%s</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="n">part</span>
</span><span id="L-506"><a href="#L-506"><span class="linenos"> 506</span></a>        <span class="k">if</span> <span class="n">reason</span><span class="p">:</span>
</span><span id="L-507"><a href="#L-507"><span class="linenos"> 507</span></a>            <span class="n">message</span> <span class="o">+=</span> <span class="s2">&quot;: &quot;</span> <span class="o">+</span> <span class="n">reason</span>
</span><span id="L-508"><a href="#L-508"><span class="linenos"> 508</span></a>        <span class="nb">super</span><span class="p">(</span><span class="n">InvalidHGVSName</span><span class="p">,</span> <span class="bp">self</span><span class="p">)</span><span class="o">.</span><span class="fm">__init__</span><span class="p">(</span><span class="n">message</span><span class="p">)</span>
</span><span id="L-509"><a href="#L-509"><span class="linenos"> 509</span></a>
</span><span id="L-510"><a href="#L-510"><span class="linenos"> 510</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">name</span> <span class="o">=</span> <span class="n">name</span>
</span><span id="L-511"><a href="#L-511"><span class="linenos"> 511</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">part</span> <span class="o">=</span> <span class="n">part</span>
</span><span id="L-512"><a href="#L-512"><span class="linenos"> 512</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">reason</span> <span class="o">=</span> <span class="n">reason</span>
</span><span id="L-513"><a href="#L-513"><span class="linenos"> 513</span></a>
</span><span id="L-514"><a href="#L-514"><span class="linenos"> 514</span></a>
</span><span id="L-515"><a href="#L-515"><span class="linenos"> 515</span></a><span class="c1"># The HGVSName class is a Python class for handling HGVS (Human Genome Variation Society) names.</span>
</span><span id="L-516"><a href="#L-516"><span class="linenos"> 516</span></a><span class="k">class</span> <span class="nc">HGVSName</span><span class="p">(</span><span class="nb">object</span><span class="p">):</span>
</span><span id="L-517"><a href="#L-517"><span class="linenos"> 517</span></a><span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-518"><a href="#L-518"><span class="linenos"> 518</span></a><span class="sd">    Represents a HGVS variant name.</span>
</span><span id="L-519"><a href="#L-519"><span class="linenos"> 519</span></a><span class="sd">    &quot;&quot;&quot;</span>
</span><span id="L-520"><a href="#L-520"><span class="linenos"> 520</span></a>
</span><span id="L-521"><a href="#L-521"><span class="linenos"> 521</span></a>    <span class="k">def</span> <span class="fm">__init__</span><span class="p">(</span>
</span><span id="L-522"><a href="#L-522"><span class="linenos"> 522</span></a>        <span class="bp">self</span><span class="p">,</span>
</span><span id="L-523"><a href="#L-523"><span class="linenos"> 523</span></a>        <span class="n">name</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span><span class="p">,</span>
</span><span id="L-524"><a href="#L-524"><span class="linenos"> 524</span></a>        <span class="n">prefix</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span><span class="p">,</span>
</span><span id="L-525"><a href="#L-525"><span class="linenos"> 525</span></a>        <span class="n">chrom</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span><span class="p">,</span>
</span><span id="L-526"><a href="#L-526"><span class="linenos"> 526</span></a>        <span class="n">transcript</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span><span class="p">,</span>
</span><span id="L-527"><a href="#L-527"><span class="linenos"> 527</span></a>        <span class="n">transcript_protein</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
</span><span id="L-528"><a href="#L-528"><span class="linenos"> 528</span></a>        <span class="n">gene</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span><span class="p">,</span>
</span><span id="L-529"><a href="#L-529"><span class="linenos"> 529</span></a>        <span class="n">exon</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
</span><span id="L-530"><a href="#L-530"><span class="linenos"> 530</span></a>        <span class="n">kind</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span><span class="p">,</span>
</span><span id="L-531"><a href="#L-531"><span class="linenos"> 531</span></a>        <span class="n">mutation_type</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
</span><span id="L-532"><a href="#L-532"><span class="linenos"> 532</span></a>        <span class="n">start</span><span class="p">:</span> <span class="nb">int</span> <span class="o">=</span> <span class="mi">0</span><span class="p">,</span>
</span><span id="L-533"><a href="#L-533"><span class="linenos"> 533</span></a>        <span class="n">end</span><span class="p">:</span> <span class="nb">int</span> <span class="o">=</span> <span class="mi">0</span><span class="p">,</span>
</span><span id="L-534"><a href="#L-534"><span class="linenos"> 534</span></a>        <span class="n">ref_allele</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span><span class="p">,</span>
</span><span id="L-535"><a href="#L-535"><span class="linenos"> 535</span></a>        <span class="n">ref2_allele</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span><span class="p">,</span>
</span><span id="L-536"><a href="#L-536"><span class="linenos"> 536</span></a>        <span class="n">alt_allele</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span><span class="p">,</span>
</span><span id="L-537"><a href="#L-537"><span class="linenos"> 537</span></a>        <span class="n">cdna_start</span><span class="p">:</span> <span class="nb">int</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
</span><span id="L-538"><a href="#L-538"><span class="linenos"> 538</span></a>        <span class="n">cdna_end</span><span class="p">:</span> <span class="nb">int</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
</span><span id="L-539"><a href="#L-539"><span class="linenos"> 539</span></a>        <span class="n">pep_extra</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span><span class="p">,</span>
</span><span id="L-540"><a href="#L-540"><span class="linenos"> 540</span></a>    <span class="p">):</span>
</span><span id="L-541"><a href="#L-541"><span class="linenos"> 541</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-542"><a href="#L-542"><span class="linenos"> 542</span></a><span class="sd">        The function is a constructor that initializes various attributes of an object and parses a</span>
</span><span id="L-543"><a href="#L-543"><span class="linenos"> 543</span></a><span class="sd">        given name to populate those attributes.</span>
</span><span id="L-544"><a href="#L-544"><span class="linenos"> 544</span></a>
</span><span id="L-545"><a href="#L-545"><span class="linenos"> 545</span></a><span class="sd">        :param name: The full HGVS name of the variant</span>
</span><span id="L-546"><a href="#L-546"><span class="linenos"> 546</span></a><span class="sd">        :type name: str</span>
</span><span id="L-547"><a href="#L-547"><span class="linenos"> 547</span></a><span class="sd">        :param prefix: The `prefix` parameter is a string that is used as a prefix for the HGVS name. It</span>
</span><span id="L-548"><a href="#L-548"><span class="linenos"> 548</span></a><span class="sd">        can be used to indicate additional information or context about the variant</span>
</span><span id="L-549"><a href="#L-549"><span class="linenos"> 549</span></a><span class="sd">        :type prefix: str</span>
</span><span id="L-550"><a href="#L-550"><span class="linenos"> 550</span></a><span class="sd">        :param chrom: The `chrom` parameter represents the chromosome where the mutation occurs. It is a</span>
</span><span id="L-551"><a href="#L-551"><span class="linenos"> 551</span></a><span class="sd">        string that specifies the chromosome number or identifier</span>
</span><span id="L-552"><a href="#L-552"><span class="linenos"> 552</span></a><span class="sd">        :type chrom: str</span>
</span><span id="L-553"><a href="#L-553"><span class="linenos"> 553</span></a><span class="sd">        :param transcript: The `transcript` parameter represents the transcript ID or name associated</span>
</span><span id="L-554"><a href="#L-554"><span class="linenos"> 554</span></a><span class="sd">        with the mutation. It is used to specify the specific transcript in which the mutation occurs</span>
</span><span id="L-555"><a href="#L-555"><span class="linenos"> 555</span></a><span class="sd">        :type transcript: str</span>
</span><span id="L-556"><a href="#L-556"><span class="linenos"> 556</span></a><span class="sd">        :param transcript_protein: The `transcript_protein` parameter is used to store information about</span>
</span><span id="L-557"><a href="#L-557"><span class="linenos"> 557</span></a><span class="sd">        the protein associated with the transcript. It can be used to specify the protein variant or</span>
</span><span id="L-558"><a href="#L-558"><span class="linenos"> 558</span></a><span class="sd">        isoform that is affected by the mutation</span>
</span><span id="L-559"><a href="#L-559"><span class="linenos"> 559</span></a><span class="sd">        :type transcript_protein: str</span>
</span><span id="L-560"><a href="#L-560"><span class="linenos"> 560</span></a><span class="sd">        :param gene: The &quot;gene&quot; parameter represents the gene associated with the variant. It is a</span>
</span><span id="L-561"><a href="#L-561"><span class="linenos"> 561</span></a><span class="sd">        string that specifies the gene name or identifier</span>
</span><span id="L-562"><a href="#L-562"><span class="linenos"> 562</span></a><span class="sd">        :type gene: str</span>
</span><span id="L-563"><a href="#L-563"><span class="linenos"> 563</span></a><span class="sd">        :param exon: The `exon` parameter represents the exon number or range in which the mutation</span>
</span><span id="L-564"><a href="#L-564"><span class="linenos"> 564</span></a><span class="sd">        occurs. It is used to specify the location of the mutation within the transcript</span>
</span><span id="L-565"><a href="#L-565"><span class="linenos"> 565</span></a><span class="sd">        :type exon: str</span>
</span><span id="L-566"><a href="#L-566"><span class="linenos"> 566</span></a><span class="sd">        :param kind: The &quot;kind&quot; parameter is used to specify the type of variant or mutation. It can be</span>
</span><span id="L-567"><a href="#L-567"><span class="linenos"> 567</span></a><span class="sd">        a string that represents the kind of mutation, such as &quot;substitution&quot;, &quot;deletion&quot;, &quot;insertion&quot;,</span>
</span><span id="L-568"><a href="#L-568"><span class="linenos"> 568</span></a><span class="sd">        etc. This parameter helps to categorize and describe the type of mutation being represented by</span>
</span><span id="L-569"><a href="#L-569"><span class="linenos"> 569</span></a><span class="sd">        the</span>
</span><span id="L-570"><a href="#L-570"><span class="linenos"> 570</span></a><span class="sd">        :type kind: str</span>
</span><span id="L-571"><a href="#L-571"><span class="linenos"> 571</span></a><span class="sd">        :param mutation_type: The `mutation_type` parameter is used to specify the type of mutation. It</span>
</span><span id="L-572"><a href="#L-572"><span class="linenos"> 572</span></a><span class="sd">        can be a string that represents the type of mutation, such as &quot;SNP&quot; (single nucleotide</span>
</span><span id="L-573"><a href="#L-573"><span class="linenos"> 573</span></a><span class="sd">        polymorphism), &quot;DEL&quot; (deletion), &quot;INS&quot; (insertion), etc</span>
</span><span id="L-574"><a href="#L-574"><span class="linenos"> 574</span></a><span class="sd">        :type mutation_type: str</span>
</span><span id="L-575"><a href="#L-575"><span class="linenos"> 575</span></a><span class="sd">        :param start: The `start` parameter represents the starting position of the mutation or variant</span>
</span><span id="L-576"><a href="#L-576"><span class="linenos"> 576</span></a><span class="sd">        in the genomic sequence. It is an integer value that indicates the position of the mutation or</span>
</span><span id="L-577"><a href="#L-577"><span class="linenos"> 577</span></a><span class="sd">        variant on the genomic sequence. If not provided, it defaults to 0, defaults to 0</span>
</span><span id="L-578"><a href="#L-578"><span class="linenos"> 578</span></a><span class="sd">        :type start: int (optional)</span>
</span><span id="L-579"><a href="#L-579"><span class="linenos"> 579</span></a><span class="sd">        :param end: The &quot;end&quot; parameter represents the end position of the mutation or variant. It is an</span>
</span><span id="L-580"><a href="#L-580"><span class="linenos"> 580</span></a><span class="sd">        integer value that indicates the position of the mutation or variant on the genomic sequence,</span>
</span><span id="L-581"><a href="#L-581"><span class="linenos"> 581</span></a><span class="sd">        defaults to 0</span>
</span><span id="L-582"><a href="#L-582"><span class="linenos"> 582</span></a><span class="sd">        :type end: int (optional)</span>
</span><span id="L-583"><a href="#L-583"><span class="linenos"> 583</span></a><span class="sd">        :param ref_allele: The `ref_allele` parameter represents the reference allele in a genetic</span>
</span><span id="L-584"><a href="#L-584"><span class="linenos"> 584</span></a><span class="sd">        mutation. It is the allele that is present in the reference genome at a specific position</span>
</span><span id="L-585"><a href="#L-585"><span class="linenos"> 585</span></a><span class="sd">        :type ref_allele: str</span>
</span><span id="L-586"><a href="#L-586"><span class="linenos"> 586</span></a><span class="sd">        :param ref2_allele: The `ref2_allele` parameter represents the reference allele at the end of a</span>
</span><span id="L-587"><a href="#L-587"><span class="linenos"> 587</span></a><span class="sd">        peptide indel. In the context of genetic mutations, an indel refers to the insertion or deletion</span>
</span><span id="L-588"><a href="#L-588"><span class="linenos"> 588</span></a><span class="sd">        of nucleotides in a DNA sequence. The `ref2_allele` specifically represents the reference allele</span>
</span><span id="L-589"><a href="#L-589"><span class="linenos"> 589</span></a><span class="sd">        that is</span>
</span><span id="L-590"><a href="#L-590"><span class="linenos"> 590</span></a><span class="sd">        :type ref2_allele: str</span>
</span><span id="L-591"><a href="#L-591"><span class="linenos"> 591</span></a><span class="sd">        :param alt_allele: The `alt_allele` parameter represents the alternate allele in a genetic</span>
</span><span id="L-592"><a href="#L-592"><span class="linenos"> 592</span></a><span class="sd">        mutation. In genetics, an allele is one of the possible forms of a gene. In the context of this</span>
</span><span id="L-593"><a href="#L-593"><span class="linenos"> 593</span></a><span class="sd">        code, `alt_allele` is used to store the alternate allele that is present in a mutation</span>
</span><span id="L-594"><a href="#L-594"><span class="linenos"> 594</span></a><span class="sd">        :type alt_allele: str</span>
</span><span id="L-595"><a href="#L-595"><span class="linenos"> 595</span></a><span class="sd">        :param cdna_start: The `cdna_start` parameter is used to specify the start position of the</span>
</span><span id="L-596"><a href="#L-596"><span class="linenos"> 596</span></a><span class="sd">        mutation in the cDNA sequence. It is an optional parameter and if not provided, it will be set</span>
</span><span id="L-597"><a href="#L-597"><span class="linenos"> 597</span></a><span class="sd">        to a default value of `CDNACoord()`</span>
</span><span id="L-598"><a href="#L-598"><span class="linenos"> 598</span></a><span class="sd">        :type cdna_start: int</span>
</span><span id="L-599"><a href="#L-599"><span class="linenos"> 599</span></a><span class="sd">        :param cdna_end: The `cdna_end` parameter is used to store the end coordinate of the cDNA</span>
</span><span id="L-600"><a href="#L-600"><span class="linenos"> 600</span></a><span class="sd">        (complementary DNA) sequence. It is an optional parameter and if not provided, it will be</span>
</span><span id="L-601"><a href="#L-601"><span class="linenos"> 601</span></a><span class="sd">        initialized as a `CDNACoord` object. The `CDNACoord` object is likely a</span>
</span><span id="L-602"><a href="#L-602"><span class="linenos"> 602</span></a><span class="sd">        :type cdna_end: int</span>
</span><span id="L-603"><a href="#L-603"><span class="linenos"> 603</span></a><span class="sd">        :param pep_extra: The `pep_extra` parameter is a string that represents any additional</span>
</span><span id="L-604"><a href="#L-604"><span class="linenos"> 604</span></a><span class="sd">        information related to the protein. It is used in the context of protein-specific fields</span>
</span><span id="L-605"><a href="#L-605"><span class="linenos"> 605</span></a><span class="sd">        :type pep_extra: str</span>
</span><span id="L-606"><a href="#L-606"><span class="linenos"> 606</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="L-607"><a href="#L-607"><span class="linenos"> 607</span></a>
</span><span id="L-608"><a href="#L-608"><span class="linenos"> 608</span></a>        <span class="c1"># Full HGVS name.</span>
</span><span id="L-609"><a href="#L-609"><span class="linenos"> 609</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">name</span> <span class="o">=</span> <span class="n">name</span>
</span><span id="L-610"><a href="#L-610"><span class="linenos"> 610</span></a>
</span><span id="L-611"><a href="#L-611"><span class="linenos"> 611</span></a>        <span class="c1"># Name parts.</span>
</span><span id="L-612"><a href="#L-612"><span class="linenos"> 612</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">prefix</span> <span class="o">=</span> <span class="n">prefix</span>
</span><span id="L-613"><a href="#L-613"><span class="linenos"> 613</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">chrom</span> <span class="o">=</span> <span class="n">chrom</span>
</span><span id="L-614"><a href="#L-614"><span class="linenos"> 614</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span> <span class="o">=</span> <span class="n">transcript</span>
</span><span id="L-615"><a href="#L-615"><span class="linenos"> 615</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">transcript_protein</span> <span class="o">=</span> <span class="n">transcript_protein</span>
</span><span id="L-616"><a href="#L-616"><span class="linenos"> 616</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">gene</span> <span class="o">=</span> <span class="n">gene</span>
</span><span id="L-617"><a href="#L-617"><span class="linenos"> 617</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">exon</span> <span class="o">=</span> <span class="n">exon</span>
</span><span id="L-618"><a href="#L-618"><span class="linenos"> 618</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">kind</span> <span class="o">=</span> <span class="n">kind</span>
</span><span id="L-619"><a href="#L-619"><span class="linenos"> 619</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">=</span> <span class="n">mutation_type</span>
</span><span id="L-620"><a href="#L-620"><span class="linenos"> 620</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">start</span> <span class="o">=</span> <span class="n">start</span>
</span><span id="L-621"><a href="#L-621"><span class="linenos"> 621</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">end</span> <span class="o">=</span> <span class="n">end</span>
</span><span id="L-622"><a href="#L-622"><span class="linenos"> 622</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">=</span> <span class="n">ref_allele</span>  <span class="c1"># reference allele</span>
</span><span id="L-623"><a href="#L-623"><span class="linenos"> 623</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">ref2_allele</span> <span class="o">=</span> <span class="n">ref2_allele</span>  <span class="c1"># reference allele at end of pep indel</span>
</span><span id="L-624"><a href="#L-624"><span class="linenos"> 624</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span> <span class="o">=</span> <span class="n">alt_allele</span>  <span class="c1"># alternate allele</span>
</span><span id="L-625"><a href="#L-625"><span class="linenos"> 625</span></a>
</span><span id="L-626"><a href="#L-626"><span class="linenos"> 626</span></a>        <span class="c1"># cDNA-specific fields</span>
</span><span id="L-627"><a href="#L-627"><span class="linenos"> 627</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">cdna_start</span> <span class="o">=</span> <span class="n">cdna_start</span> <span class="k">if</span> <span class="n">cdna_start</span> <span class="k">else</span> <span class="n">CDNACoord</span><span class="p">()</span>
</span><span id="L-628"><a href="#L-628"><span class="linenos"> 628</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">cdna_end</span> <span class="o">=</span> <span class="n">cdna_end</span> <span class="k">if</span> <span class="n">cdna_end</span> <span class="k">else</span> <span class="n">CDNACoord</span><span class="p">()</span>
</span><span id="L-629"><a href="#L-629"><span class="linenos"> 629</span></a>
</span><span id="L-630"><a href="#L-630"><span class="linenos"> 630</span></a>        <span class="c1"># Protein-specific fields</span>
</span><span id="L-631"><a href="#L-631"><span class="linenos"> 631</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">pep_extra</span> <span class="o">=</span> <span class="n">pep_extra</span>
</span><span id="L-632"><a href="#L-632"><span class="linenos"> 632</span></a>
</span><span id="L-633"><a href="#L-633"><span class="linenos"> 633</span></a>        <span class="k">if</span> <span class="n">name</span><span class="p">:</span>
</span><span id="L-634"><a href="#L-634"><span class="linenos"> 634</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">parse</span><span class="p">(</span><span class="n">name</span><span class="p">)</span>
</span><span id="L-635"><a href="#L-635"><span class="linenos"> 635</span></a>
</span><span id="L-636"><a href="#L-636"><span class="linenos"> 636</span></a>    <span class="k">def</span> <span class="nf">parse</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">name</span><span class="p">:</span> <span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
</span><span id="L-637"><a href="#L-637"><span class="linenos"> 637</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-638"><a href="#L-638"><span class="linenos"> 638</span></a><span class="sd">        The `parse` function is used to split an HGVS name into a prefix and allele, and then validate</span>
</span><span id="L-639"><a href="#L-639"><span class="linenos"> 639</span></a><span class="sd">        the parsed components.</span>
</span><span id="L-640"><a href="#L-640"><span class="linenos"> 640</span></a>
</span><span id="L-641"><a href="#L-641"><span class="linenos"> 641</span></a><span class="sd">        :param name: The `name` parameter is a string that represents an HGVS name. It is the input to</span>
</span><span id="L-642"><a href="#L-642"><span class="linenos"> 642</span></a><span class="sd">        the `parse` function and is used to parse the HGVS name by splitting it into a prefix and allele</span>
</span><span id="L-643"><a href="#L-643"><span class="linenos"> 643</span></a><span class="sd">        :type name: str</span>
</span><span id="L-644"><a href="#L-644"><span class="linenos"> 644</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="L-645"><a href="#L-645"><span class="linenos"> 645</span></a>
</span><span id="L-646"><a href="#L-646"><span class="linenos"> 646</span></a>        <span class="c1"># Does HGVS name have transcript/gene prefix?</span>
</span><span id="L-647"><a href="#L-647"><span class="linenos"> 647</span></a>        <span class="k">if</span> <span class="s2">&quot;:&quot;</span> <span class="ow">in</span> <span class="n">name</span><span class="p">:</span>
</span><span id="L-648"><a href="#L-648"><span class="linenos"> 648</span></a>            <span class="n">prefix</span><span class="p">,</span> <span class="n">allele</span> <span class="o">=</span> <span class="n">name</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot;:&quot;</span><span class="p">,</span> <span class="mi">1</span><span class="p">)</span>
</span><span id="L-649"><a href="#L-649"><span class="linenos"> 649</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="L-650"><a href="#L-650"><span class="linenos"> 650</span></a>            <span class="n">prefix</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span>
</span><span id="L-651"><a href="#L-651"><span class="linenos"> 651</span></a>            <span class="n">allele</span> <span class="o">=</span> <span class="n">name</span>
</span><span id="L-652"><a href="#L-652"><span class="linenos"> 652</span></a>
</span><span id="L-653"><a href="#L-653"><span class="linenos"> 653</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">name</span> <span class="o">=</span> <span class="n">name</span>
</span><span id="L-654"><a href="#L-654"><span class="linenos"> 654</span></a>
</span><span id="L-655"><a href="#L-655"><span class="linenos"> 655</span></a>        <span class="c1"># Parse prefix and allele.</span>
</span><span id="L-656"><a href="#L-656"><span class="linenos"> 656</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">parse_allele</span><span class="p">(</span><span class="n">allele</span><span class="p">)</span>
</span><span id="L-657"><a href="#L-657"><span class="linenos"> 657</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">parse_prefix</span><span class="p">(</span><span class="n">prefix</span><span class="p">)</span>
</span><span id="L-658"><a href="#L-658"><span class="linenos"> 658</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">_validate</span><span class="p">()</span>
</span><span id="L-659"><a href="#L-659"><span class="linenos"> 659</span></a>
</span><span id="L-660"><a href="#L-660"><span class="linenos"> 660</span></a>    <span class="k">def</span> <span class="nf">parse_prefix</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">prefix</span><span class="p">:</span> <span class="nb">str</span><span class="p">):</span>
</span><span id="L-661"><a href="#L-661"><span class="linenos"> 661</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-662"><a href="#L-662"><span class="linenos"> 662</span></a><span class="sd">        The `parse_prefix` function is used to parse a HGVS prefix (gene/transcript/chromosome) and</span>
</span><span id="L-663"><a href="#L-663"><span class="linenos"> 663</span></a><span class="sd">        assign the parsed values to the corresponding attributes of the object.</span>
</span><span id="L-664"><a href="#L-664"><span class="linenos"> 664</span></a>
</span><span id="L-665"><a href="#L-665"><span class="linenos"> 665</span></a><span class="sd">        :param prefix: The `prefix` parameter is a string that represents a HGVS prefix, which can be a</span>
</span><span id="L-666"><a href="#L-666"><span class="linenos"> 666</span></a><span class="sd">        gene, transcript, or chromosome identifier. It is used to determine the type of prefix and</span>
</span><span id="L-667"><a href="#L-667"><span class="linenos"> 667</span></a><span class="sd">        assign the parsed values to the corresponding attributes of the object</span>
</span><span id="L-668"><a href="#L-668"><span class="linenos"> 668</span></a><span class="sd">        :type prefix: str</span>
</span><span id="L-669"><a href="#L-669"><span class="linenos"> 669</span></a><span class="sd">        :return: The function `parse_prefix` returns the parsed values for the transcript and gene</span>
</span><span id="L-670"><a href="#L-670"><span class="linenos"> 670</span></a><span class="sd">        attributes, or sets the chrom or gene attributes based on the given prefix.</span>
</span><span id="L-671"><a href="#L-671"><span class="linenos"> 671</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="L-672"><a href="#L-672"><span class="linenos"> 672</span></a>
</span><span id="L-673"><a href="#L-673"><span class="linenos"> 673</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">prefix</span> <span class="o">=</span> <span class="n">prefix</span>
</span><span id="L-674"><a href="#L-674"><span class="linenos"> 674</span></a>
</span><span id="L-675"><a href="#L-675"><span class="linenos"> 675</span></a>        <span class="c1"># No prefix.</span>
</span><span id="L-676"><a href="#L-676"><span class="linenos"> 676</span></a>        <span class="k">if</span> <span class="n">prefix</span> <span class="o">==</span> <span class="s2">&quot;&quot;</span><span class="p">:</span>
</span><span id="L-677"><a href="#L-677"><span class="linenos"> 677</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">chrom</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span>
</span><span id="L-678"><a href="#L-678"><span class="linenos"> 678</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span>
</span><span id="L-679"><a href="#L-679"><span class="linenos"> 679</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">gene</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span>
</span><span id="L-680"><a href="#L-680"><span class="linenos"> 680</span></a>            <span class="k">return</span>
</span><span id="L-681"><a href="#L-681"><span class="linenos"> 681</span></a>
</span><span id="L-682"><a href="#L-682"><span class="linenos"> 682</span></a>        <span class="c1"># Transcript and gene given with parens.</span>
</span><span id="L-683"><a href="#L-683"><span class="linenos"> 683</span></a>        <span class="c1"># example: NM_007294.3(BRCA1):c.2207A&gt;C</span>
</span><span id="L-684"><a href="#L-684"><span class="linenos"> 684</span></a>        <span class="n">match</span> <span class="o">=</span> <span class="n">re</span><span class="o">.</span><span class="n">match</span><span class="p">(</span><span class="sa">r</span><span class="s2">&quot;^(?P&lt;transcript&gt;[^(]+)\((?P&lt;gene&gt;[^)]+)\)$&quot;</span><span class="p">,</span> <span class="n">prefix</span><span class="p">)</span>
</span><span id="L-685"><a href="#L-685"><span class="linenos"> 685</span></a>        <span class="k">if</span> <span class="n">match</span><span class="p">:</span>
</span><span id="L-686"><a href="#L-686"><span class="linenos"> 686</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span> <span class="o">=</span> <span class="n">match</span><span class="o">.</span><span class="n">group</span><span class="p">(</span><span class="s2">&quot;transcript&quot;</span><span class="p">)</span>
</span><span id="L-687"><a href="#L-687"><span class="linenos"> 687</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">gene</span> <span class="o">=</span> <span class="n">match</span><span class="o">.</span><span class="n">group</span><span class="p">(</span><span class="s2">&quot;gene&quot;</span><span class="p">)</span>
</span><span id="L-688"><a href="#L-688"><span class="linenos"> 688</span></a>            <span class="k">return</span>
</span><span id="L-689"><a href="#L-689"><span class="linenos"> 689</span></a>
</span><span id="L-690"><a href="#L-690"><span class="linenos"> 690</span></a>        <span class="c1"># Transcript and gene given with braces.</span>
</span><span id="L-691"><a href="#L-691"><span class="linenos"> 691</span></a>        <span class="c1"># example: BRCA1{NM_007294.3}:c.2207A&gt;C</span>
</span><span id="L-692"><a href="#L-692"><span class="linenos"> 692</span></a>        <span class="n">match</span> <span class="o">=</span> <span class="n">re</span><span class="o">.</span><span class="n">match</span><span class="p">(</span><span class="sa">r</span><span class="s2">&quot;^(?P&lt;gene&gt;[^{]+){(?P&lt;transcript&gt;[^}]+)}$&quot;</span><span class="p">,</span> <span class="n">prefix</span><span class="p">)</span>
</span><span id="L-693"><a href="#L-693"><span class="linenos"> 693</span></a>        <span class="k">if</span> <span class="n">match</span><span class="p">:</span>
</span><span id="L-694"><a href="#L-694"><span class="linenos"> 694</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span> <span class="o">=</span> <span class="n">match</span><span class="o">.</span><span class="n">group</span><span class="p">(</span><span class="s2">&quot;transcript&quot;</span><span class="p">)</span>
</span><span id="L-695"><a href="#L-695"><span class="linenos"> 695</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">gene</span> <span class="o">=</span> <span class="n">match</span><span class="o">.</span><span class="n">group</span><span class="p">(</span><span class="s2">&quot;gene&quot;</span><span class="p">)</span>
</span><span id="L-696"><a href="#L-696"><span class="linenos"> 696</span></a>            <span class="k">return</span>
</span><span id="L-697"><a href="#L-697"><span class="linenos"> 697</span></a>
</span><span id="L-698"><a href="#L-698"><span class="linenos"> 698</span></a>        <span class="c1"># Determine using Ensembl type.</span>
</span><span id="L-699"><a href="#L-699"><span class="linenos"> 699</span></a>        <span class="k">if</span> <span class="n">prefix</span><span class="o">.</span><span class="n">startswith</span><span class="p">(</span><span class="s2">&quot;ENST&quot;</span><span class="p">):</span>
</span><span id="L-700"><a href="#L-700"><span class="linenos"> 700</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span> <span class="o">=</span> <span class="n">prefix</span>
</span><span id="L-701"><a href="#L-701"><span class="linenos"> 701</span></a>            <span class="k">return</span>
</span><span id="L-702"><a href="#L-702"><span class="linenos"> 702</span></a>
</span><span id="L-703"><a href="#L-703"><span class="linenos"> 703</span></a>        <span class="c1"># Determine using LRG type.</span>
</span><span id="L-704"><a href="#L-704"><span class="linenos"> 704</span></a>        <span class="k">if</span> <span class="n">prefix</span><span class="o">.</span><span class="n">startswith</span><span class="p">(</span><span class="s2">&quot;LRG_&quot;</span><span class="p">):</span>
</span><span id="L-705"><a href="#L-705"><span class="linenos"> 705</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span> <span class="o">=</span> <span class="n">prefix</span>
</span><span id="L-706"><a href="#L-706"><span class="linenos"> 706</span></a>            <span class="k">return</span>
</span><span id="L-707"><a href="#L-707"><span class="linenos"> 707</span></a>
</span><span id="L-708"><a href="#L-708"><span class="linenos"> 708</span></a>        <span class="c1"># Determine using refseq type.</span>
</span><span id="L-709"><a href="#L-709"><span class="linenos"> 709</span></a>        <span class="n">refseq_type</span> <span class="o">=</span> <span class="n">get_refseq_type</span><span class="p">(</span><span class="n">prefix</span><span class="p">)</span>
</span><span id="L-710"><a href="#L-710"><span class="linenos"> 710</span></a>        <span class="k">if</span> <span class="n">refseq_type</span> <span class="ow">in</span> <span class="p">(</span><span class="s2">&quot;mRNA&quot;</span><span class="p">,</span> <span class="s2">&quot;RNA&quot;</span><span class="p">):</span>
</span><span id="L-711"><a href="#L-711"><span class="linenos"> 711</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span> <span class="o">=</span> <span class="n">prefix</span>
</span><span id="L-712"><a href="#L-712"><span class="linenos"> 712</span></a>            <span class="k">return</span>
</span><span id="L-713"><a href="#L-713"><span class="linenos"> 713</span></a>
</span><span id="L-714"><a href="#L-714"><span class="linenos"> 714</span></a>        <span class="c1"># Determine using refseq type.</span>
</span><span id="L-715"><a href="#L-715"><span class="linenos"> 715</span></a>        <span class="k">if</span> <span class="n">prefix</span><span class="o">.</span><span class="n">startswith</span><span class="p">(</span><span class="n">CHROM_PREFIX</span><span class="p">)</span> <span class="ow">or</span> <span class="n">refseq_type</span> <span class="o">==</span> <span class="s2">&quot;genomic&quot;</span><span class="p">:</span>
</span><span id="L-716"><a href="#L-716"><span class="linenos"> 716</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">chrom</span> <span class="o">=</span> <span class="n">prefix</span>
</span><span id="L-717"><a href="#L-717"><span class="linenos"> 717</span></a>            <span class="k">return</span>
</span><span id="L-718"><a href="#L-718"><span class="linenos"> 718</span></a>
</span><span id="L-719"><a href="#L-719"><span class="linenos"> 719</span></a>        <span class="c1"># Assume gene name.</span>
</span><span id="L-720"><a href="#L-720"><span class="linenos"> 720</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">gene</span> <span class="o">=</span> <span class="n">prefix</span>
</span><span id="L-721"><a href="#L-721"><span class="linenos"> 721</span></a>
</span><span id="L-722"><a href="#L-722"><span class="linenos"> 722</span></a>    <span class="k">def</span> <span class="nf">parse_allele</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">allele</span><span class="p">:</span> <span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
</span><span id="L-723"><a href="#L-723"><span class="linenos"> 723</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-724"><a href="#L-724"><span class="linenos"> 724</span></a><span class="sd">        The function `parse_allele` parses a HGVS allele description and determines the kind of HGVS</span>
</span><span id="L-725"><a href="#L-725"><span class="linenos"> 725</span></a><span class="sd">        name (c., p., g., etc.) and the mutation type.</span>
</span><span id="L-726"><a href="#L-726"><span class="linenos"> 726</span></a>
</span><span id="L-727"><a href="#L-727"><span class="linenos"> 727</span></a><span class="sd">        Some examples include:</span>
</span><span id="L-728"><a href="#L-728"><span class="linenos"> 728</span></a><span class="sd">          cDNA substitution: c.101A&gt;C,</span>
</span><span id="L-729"><a href="#L-729"><span class="linenos"> 729</span></a><span class="sd">          cDNA indel: c.3428delCinsTA, c.1000_1003delATG, c.1000_1001insATG</span>
</span><span id="L-730"><a href="#L-730"><span class="linenos"> 730</span></a><span class="sd">          No protein change: p.Glu1161=</span>
</span><span id="L-731"><a href="#L-731"><span class="linenos"> 731</span></a><span class="sd">          Protein change: p.Glu1161Ser</span>
</span><span id="L-732"><a href="#L-732"><span class="linenos"> 732</span></a><span class="sd">          Protein frameshift: p.Glu1161_Ser1164?fs</span>
</span><span id="L-733"><a href="#L-733"><span class="linenos"> 733</span></a><span class="sd">          Genomic substitution: g.1000100A&gt;T</span>
</span><span id="L-734"><a href="#L-734"><span class="linenos"> 734</span></a><span class="sd">          Genomic indel: g.1000100_1000102delATG</span>
</span><span id="L-735"><a href="#L-735"><span class="linenos"> 735</span></a>
</span><span id="L-736"><a href="#L-736"><span class="linenos"> 736</span></a><span class="sd">        :param allele: The `allele` parameter is a string that represents a HGVS allele description. It</span>
</span><span id="L-737"><a href="#L-737"><span class="linenos"> 737</span></a><span class="sd">        can contain various types of mutations, such as cDNA substitutions, cDNA indels, protein</span>
</span><span id="L-738"><a href="#L-738"><span class="linenos"> 738</span></a><span class="sd">        changes, protein frameshifts, genomic substitutions, and genomic indels. The purpose of the</span>
</span><span id="L-739"><a href="#L-739"><span class="linenos"> 739</span></a><span class="sd">        `parse_allele`</span>
</span><span id="L-740"><a href="#L-740"><span class="linenos"> 740</span></a><span class="sd">        :type allele: str</span>
</span><span id="L-741"><a href="#L-741"><span class="linenos"> 741</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="L-742"><a href="#L-742"><span class="linenos"> 742</span></a>
</span><span id="L-743"><a href="#L-743"><span class="linenos"> 743</span></a>        <span class="k">if</span> <span class="s2">&quot;.&quot;</span> <span class="ow">not</span> <span class="ow">in</span> <span class="n">allele</span><span class="p">:</span>
</span><span id="L-744"><a href="#L-744"><span class="linenos"> 744</span></a>            <span class="k">raise</span> <span class="n">InvalidHGVSName</span><span class="p">(</span>
</span><span id="L-745"><a href="#L-745"><span class="linenos"> 745</span></a>                <span class="n">allele</span><span class="p">,</span> <span class="s2">&quot;allele&quot;</span><span class="p">,</span> <span class="s1">&#39;expected kind &quot;c.&quot;, &quot;p.&quot;, &quot;g.&quot;, etc&#39;</span>
</span><span id="L-746"><a href="#L-746"><span class="linenos"> 746</span></a>            <span class="p">)</span>
</span><span id="L-747"><a href="#L-747"><span class="linenos"> 747</span></a>
</span><span id="L-748"><a href="#L-748"><span class="linenos"> 748</span></a>        <span class="c1"># Determine HGVS name kind.</span>
</span><span id="L-749"><a href="#L-749"><span class="linenos"> 749</span></a>        <span class="n">kind</span><span class="p">,</span> <span class="n">details</span> <span class="o">=</span> <span class="n">allele</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot;.&quot;</span><span class="p">,</span> <span class="mi">1</span><span class="p">)</span>
</span><span id="L-750"><a href="#L-750"><span class="linenos"> 750</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">kind</span> <span class="o">=</span> <span class="n">kind</span>
</span><span id="L-751"><a href="#L-751"><span class="linenos"> 751</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">=</span> <span class="kc">None</span>
</span><span id="L-752"><a href="#L-752"><span class="linenos"> 752</span></a>
</span><span id="L-753"><a href="#L-753"><span class="linenos"> 753</span></a>        <span class="k">if</span> <span class="n">kind</span> <span class="ow">in</span> <span class="p">(</span><span class="s2">&quot;c&quot;</span><span class="p">,</span> <span class="s2">&quot;n&quot;</span><span class="p">):</span>
</span><span id="L-754"><a href="#L-754"><span class="linenos"> 754</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">parse_cdna</span><span class="p">(</span><span class="n">details</span><span class="p">)</span>
</span><span id="L-755"><a href="#L-755"><span class="linenos"> 755</span></a>            <span class="k">if</span> <span class="n">kind</span> <span class="o">==</span> <span class="s2">&quot;n&quot;</span><span class="p">:</span>  <span class="c1"># Ensure no 3&#39;UTR or 5&#39;UTR coords in non-coding</span>
</span><span id="L-756"><a href="#L-756"><span class="linenos"> 756</span></a>                <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">cdna_start</span><span class="o">.</span><span class="n">coord</span> <span class="o">&lt;</span> <span class="mi">0</span><span class="p">:</span>
</span><span id="L-757"><a href="#L-757"><span class="linenos"> 757</span></a>                    <span class="k">raise</span> <span class="n">InvalidHGVSName</span><span class="p">(</span>
</span><span id="L-758"><a href="#L-758"><span class="linenos"> 758</span></a>                        <span class="n">allele</span><span class="p">,</span>
</span><span id="L-759"><a href="#L-759"><span class="linenos"> 759</span></a>                        <span class="s2">&quot;allele&quot;</span><span class="p">,</span>
</span><span id="L-760"><a href="#L-760"><span class="linenos"> 760</span></a>                        <span class="s2">&quot;Non-coding transcript cannot contain negative (5&#39;UTR) coordinates&quot;</span><span class="p">,</span>
</span><span id="L-761"><a href="#L-761"><span class="linenos"> 761</span></a>                    <span class="p">)</span>
</span><span id="L-762"><a href="#L-762"><span class="linenos"> 762</span></a>                <span class="k">if</span> <span class="p">(</span>
</span><span id="L-763"><a href="#L-763"><span class="linenos"> 763</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">cdna_start</span><span class="o">.</span><span class="n">landmark</span> <span class="o">==</span> <span class="s2">&quot;cdna_stop&quot;</span>
</span><span id="L-764"><a href="#L-764"><span class="linenos"> 764</span></a>                    <span class="ow">or</span> <span class="bp">self</span><span class="o">.</span><span class="n">cdna_end</span>
</span><span id="L-765"><a href="#L-765"><span class="linenos"> 765</span></a>                    <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">cdna_end</span><span class="o">.</span><span class="n">landmark</span> <span class="o">==</span> <span class="s2">&quot;cdna_stop&quot;</span>
</span><span id="L-766"><a href="#L-766"><span class="linenos"> 766</span></a>                <span class="p">):</span>
</span><span id="L-767"><a href="#L-767"><span class="linenos"> 767</span></a>                    <span class="k">raise</span> <span class="n">InvalidHGVSName</span><span class="p">(</span>
</span><span id="L-768"><a href="#L-768"><span class="linenos"> 768</span></a>                        <span class="n">allele</span><span class="p">,</span>
</span><span id="L-769"><a href="#L-769"><span class="linenos"> 769</span></a>                        <span class="s2">&quot;allele&quot;</span><span class="p">,</span>
</span><span id="L-770"><a href="#L-770"><span class="linenos"> 770</span></a>                        <span class="s2">&quot;Non-coding transcript cannot contain &#39;*&#39; (3&#39;UTR) coordinates&quot;</span><span class="p">,</span>
</span><span id="L-771"><a href="#L-771"><span class="linenos"> 771</span></a>                    <span class="p">)</span>
</span><span id="L-772"><a href="#L-772"><span class="linenos"> 772</span></a>        <span class="k">elif</span> <span class="n">kind</span> <span class="o">==</span> <span class="s2">&quot;p&quot;</span><span class="p">:</span>
</span><span id="L-773"><a href="#L-773"><span class="linenos"> 773</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">parse_protein</span><span class="p">(</span><span class="n">details</span><span class="p">)</span>
</span><span id="L-774"><a href="#L-774"><span class="linenos"> 774</span></a>        <span class="k">elif</span> <span class="n">kind</span> <span class="ow">in</span> <span class="p">(</span><span class="s2">&quot;g&quot;</span><span class="p">,</span> <span class="s2">&quot;m&quot;</span><span class="p">):</span>
</span><span id="L-775"><a href="#L-775"><span class="linenos"> 775</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">parse_genome</span><span class="p">(</span><span class="n">details</span><span class="p">)</span>
</span><span id="L-776"><a href="#L-776"><span class="linenos"> 776</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="L-777"><a href="#L-777"><span class="linenos"> 777</span></a>            <span class="k">raise</span> <span class="ne">NotImplementedError</span><span class="p">(</span><span class="s2">&quot;unknown kind: </span><span class="si">%s</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="n">allele</span><span class="p">)</span>
</span><span id="L-778"><a href="#L-778"><span class="linenos"> 778</span></a>
</span><span id="L-779"><a href="#L-779"><span class="linenos"> 779</span></a>    <span class="k">def</span> <span class="nf">parse_cdna</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">details</span><span class="p">:</span> <span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
</span><span id="L-780"><a href="#L-780"><span class="linenos"> 780</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-781"><a href="#L-781"><span class="linenos"> 781</span></a><span class="sd">        The function `parse_cdna` is used to parse a HGVS cDNA name and extract information such as</span>
</span><span id="L-782"><a href="#L-782"><span class="linenos"> 782</span></a><span class="sd">        mutation type, coordinates, and alleles.</span>
</span><span id="L-783"><a href="#L-783"><span class="linenos"> 783</span></a>
</span><span id="L-784"><a href="#L-784"><span class="linenos"> 784</span></a><span class="sd">        Some examples include:</span>
</span><span id="L-785"><a href="#L-785"><span class="linenos"> 785</span></a><span class="sd">          Substitution: 101A&gt;C,</span>
</span><span id="L-786"><a href="#L-786"><span class="linenos"> 786</span></a><span class="sd">          Indel: 3428delCinsTA, 1000_1003delATG, 1000_1001insATG</span>
</span><span id="L-787"><a href="#L-787"><span class="linenos"> 787</span></a>
</span><span id="L-788"><a href="#L-788"><span class="linenos"> 788</span></a><span class="sd">        :param details: The `details` parameter is a string that represents a HGVS cDNA name. It</span>
</span><span id="L-789"><a href="#L-789"><span class="linenos"> 789</span></a><span class="sd">        contains information about a genetic mutation, such as a substitution or an indel, along with</span>
</span><span id="L-790"><a href="#L-790"><span class="linenos"> 790</span></a><span class="sd">        the specific coordinates and alleles involved in the mutation</span>
</span><span id="L-791"><a href="#L-791"><span class="linenos"> 791</span></a><span class="sd">        :type details: str</span>
</span><span id="L-792"><a href="#L-792"><span class="linenos"> 792</span></a><span class="sd">        :return: None.</span>
</span><span id="L-793"><a href="#L-793"><span class="linenos"> 793</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="L-794"><a href="#L-794"><span class="linenos"> 794</span></a>
</span><span id="L-795"><a href="#L-795"><span class="linenos"> 795</span></a>        <span class="k">for</span> <span class="n">regex</span> <span class="ow">in</span> <span class="n">HGVSRegex</span><span class="o">.</span><span class="n">CDNA_ALLELE_REGEXES</span><span class="p">:</span>
</span><span id="L-796"><a href="#L-796"><span class="linenos"> 796</span></a>            <span class="n">match</span> <span class="o">=</span> <span class="n">re</span><span class="o">.</span><span class="n">match</span><span class="p">(</span><span class="n">regex</span><span class="p">,</span> <span class="n">details</span><span class="p">)</span>
</span><span id="L-797"><a href="#L-797"><span class="linenos"> 797</span></a>            <span class="k">if</span> <span class="n">match</span><span class="p">:</span>
</span><span id="L-798"><a href="#L-798"><span class="linenos"> 798</span></a>                <span class="n">groups</span> <span class="o">=</span> <span class="n">match</span><span class="o">.</span><span class="n">groupdict</span><span class="p">()</span>
</span><span id="L-799"><a href="#L-799"><span class="linenos"> 799</span></a>
</span><span id="L-800"><a href="#L-800"><span class="linenos"> 800</span></a>                <span class="c1"># Parse mutation type.</span>
</span><span id="L-801"><a href="#L-801"><span class="linenos"> 801</span></a>                <span class="k">if</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;delins&quot;</span><span class="p">):</span>
</span><span id="L-802"><a href="#L-802"><span class="linenos"> 802</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">=</span> <span class="s2">&quot;delins&quot;</span>
</span><span id="L-803"><a href="#L-803"><span class="linenos"> 803</span></a>                <span class="k">else</span><span class="p">:</span>
</span><span id="L-804"><a href="#L-804"><span class="linenos"> 804</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">=</span> <span class="n">groups</span><span class="p">[</span><span class="s2">&quot;mutation_type&quot;</span><span class="p">]</span>
</span><span id="L-805"><a href="#L-805"><span class="linenos"> 805</span></a>
</span><span id="L-806"><a href="#L-806"><span class="linenos"> 806</span></a>                <span class="c1"># Parse coordinates.</span>
</span><span id="L-807"><a href="#L-807"><span class="linenos"> 807</span></a>                <span class="bp">self</span><span class="o">.</span><span class="n">cdna_start</span> <span class="o">=</span> <span class="n">CDNACoord</span><span class="p">(</span><span class="n">string</span><span class="o">=</span><span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;start&quot;</span><span class="p">))</span>
</span><span id="L-808"><a href="#L-808"><span class="linenos"> 808</span></a>                <span class="k">if</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;end&quot;</span><span class="p">):</span>
</span><span id="L-809"><a href="#L-809"><span class="linenos"> 809</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">cdna_end</span> <span class="o">=</span> <span class="n">CDNACoord</span><span class="p">(</span><span class="n">string</span><span class="o">=</span><span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;end&quot;</span><span class="p">))</span>
</span><span id="L-810"><a href="#L-810"><span class="linenos"> 810</span></a>                <span class="k">else</span><span class="p">:</span>
</span><span id="L-811"><a href="#L-811"><span class="linenos"> 811</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">cdna_end</span> <span class="o">=</span> <span class="n">CDNACoord</span><span class="p">(</span><span class="n">string</span><span class="o">=</span><span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;start&quot;</span><span class="p">))</span>
</span><span id="L-812"><a href="#L-812"><span class="linenos"> 812</span></a>
</span><span id="L-813"><a href="#L-813"><span class="linenos"> 813</span></a>                <span class="c1"># Parse alleles.</span>
</span><span id="L-814"><a href="#L-814"><span class="linenos"> 814</span></a>                <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">=</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;ref&quot;</span><span class="p">,</span> <span class="s2">&quot;&quot;</span><span class="p">)</span>
</span><span id="L-815"><a href="#L-815"><span class="linenos"> 815</span></a>                <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span> <span class="o">=</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;alt&quot;</span><span class="p">,</span> <span class="s2">&quot;&quot;</span><span class="p">)</span>
</span><span id="L-816"><a href="#L-816"><span class="linenos"> 816</span></a>
</span><span id="L-817"><a href="#L-817"><span class="linenos"> 817</span></a>                <span class="c1"># Convert numerical allelles.</span>
</span><span id="L-818"><a href="#L-818"><span class="linenos"> 818</span></a>                <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span><span class="o">.</span><span class="n">isdigit</span><span class="p">():</span>
</span><span id="L-819"><a href="#L-819"><span class="linenos"> 819</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">=</span> <span class="s2">&quot;N&quot;</span> <span class="o">*</span> <span class="nb">int</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span><span class="p">)</span>
</span><span id="L-820"><a href="#L-820"><span class="linenos"> 820</span></a>                <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span><span class="o">.</span><span class="n">isdigit</span><span class="p">():</span>
</span><span id="L-821"><a href="#L-821"><span class="linenos"> 821</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span> <span class="o">=</span> <span class="s2">&quot;N&quot;</span> <span class="o">*</span> <span class="nb">int</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span><span class="p">)</span>
</span><span id="L-822"><a href="#L-822"><span class="linenos"> 822</span></a>
</span><span id="L-823"><a href="#L-823"><span class="linenos"> 823</span></a>                <span class="c1"># Convert duplication alleles.</span>
</span><span id="L-824"><a href="#L-824"><span class="linenos"> 824</span></a>                <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">==</span> <span class="s2">&quot;dup&quot;</span><span class="p">:</span>
</span><span id="L-825"><a href="#L-825"><span class="linenos"> 825</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">*</span> <span class="mi">2</span>
</span><span id="L-826"><a href="#L-826"><span class="linenos"> 826</span></a>
</span><span id="L-827"><a href="#L-827"><span class="linenos"> 827</span></a>                <span class="c1"># Convert no match alleles.</span>
</span><span id="L-828"><a href="#L-828"><span class="linenos"> 828</span></a>                <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">==</span> <span class="s2">&quot;=&quot;</span><span class="p">:</span>
</span><span id="L-829"><a href="#L-829"><span class="linenos"> 829</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span>
</span><span id="L-830"><a href="#L-830"><span class="linenos"> 830</span></a>                <span class="k">return</span>
</span><span id="L-831"><a href="#L-831"><span class="linenos"> 831</span></a>
</span><span id="L-832"><a href="#L-832"><span class="linenos"> 832</span></a>        <span class="k">raise</span> <span class="n">InvalidHGVSName</span><span class="p">(</span><span class="n">details</span><span class="p">,</span> <span class="s2">&quot;cDNA allele&quot;</span><span class="p">)</span>
</span><span id="L-833"><a href="#L-833"><span class="linenos"> 833</span></a>
</span><span id="L-834"><a href="#L-834"><span class="linenos"> 834</span></a>    <span class="k">def</span> <span class="nf">parse_protein</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">details</span><span class="p">:</span> <span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
</span><span id="L-835"><a href="#L-835"><span class="linenos"> 835</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-836"><a href="#L-836"><span class="linenos"> 836</span></a><span class="sd">        The function `parse_protein` is used to parse a HGVS protein name and extract information such</span>
</span><span id="L-837"><a href="#L-837"><span class="linenos"> 837</span></a><span class="sd">        as mutation type, coordinates, alleles, and additional details.</span>
</span><span id="L-838"><a href="#L-838"><span class="linenos"> 838</span></a>
</span><span id="L-839"><a href="#L-839"><span class="linenos"> 839</span></a><span class="sd">        Some examples include:</span>
</span><span id="L-840"><a href="#L-840"><span class="linenos"> 840</span></a><span class="sd">          No change: Glu1161=</span>
</span><span id="L-841"><a href="#L-841"><span class="linenos"> 841</span></a><span class="sd">          Change: Glu1161Ser</span>
</span><span id="L-842"><a href="#L-842"><span class="linenos"> 842</span></a><span class="sd">          Frameshift: Glu1161_Ser1164?fs</span>
</span><span id="L-843"><a href="#L-843"><span class="linenos"> 843</span></a>
</span><span id="L-844"><a href="#L-844"><span class="linenos"> 844</span></a><span class="sd">        :param details: The `details` parameter is a string that represents a HGVS protein name. It</span>
</span><span id="L-845"><a href="#L-845"><span class="linenos"> 845</span></a><span class="sd">        contains information about a protein mutation, such as the amino acid change and the position of</span>
</span><span id="L-846"><a href="#L-846"><span class="linenos"> 846</span></a><span class="sd">        the mutation</span>
</span><span id="L-847"><a href="#L-847"><span class="linenos"> 847</span></a><span class="sd">        :type details: str</span>
</span><span id="L-848"><a href="#L-848"><span class="linenos"> 848</span></a><span class="sd">        :return: The method `parse_protein` does not return anything. It updates the instance variables</span>
</span><span id="L-849"><a href="#L-849"><span class="linenos"> 849</span></a><span class="sd">        of the object it is called on.</span>
</span><span id="L-850"><a href="#L-850"><span class="linenos"> 850</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="L-851"><a href="#L-851"><span class="linenos"> 851</span></a>
</span><span id="L-852"><a href="#L-852"><span class="linenos"> 852</span></a>        <span class="k">for</span> <span class="n">regex</span> <span class="ow">in</span> <span class="n">HGVSRegex</span><span class="o">.</span><span class="n">PEP_ALLELE_REGEXES</span><span class="p">:</span>
</span><span id="L-853"><a href="#L-853"><span class="linenos"> 853</span></a>            <span class="n">match</span> <span class="o">=</span> <span class="n">re</span><span class="o">.</span><span class="n">match</span><span class="p">(</span><span class="n">regex</span><span class="p">,</span> <span class="n">details</span><span class="p">)</span>
</span><span id="L-854"><a href="#L-854"><span class="linenos"> 854</span></a>            <span class="k">if</span> <span class="n">match</span><span class="p">:</span>
</span><span id="L-855"><a href="#L-855"><span class="linenos"> 855</span></a>                <span class="n">groups</span> <span class="o">=</span> <span class="n">match</span><span class="o">.</span><span class="n">groupdict</span><span class="p">()</span>
</span><span id="L-856"><a href="#L-856"><span class="linenos"> 856</span></a>
</span><span id="L-857"><a href="#L-857"><span class="linenos"> 857</span></a>                <span class="c1"># Parse mutation type.</span>
</span><span id="L-858"><a href="#L-858"><span class="linenos"> 858</span></a>                <span class="k">if</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;delins&quot;</span><span class="p">):</span>
</span><span id="L-859"><a href="#L-859"><span class="linenos"> 859</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">=</span> <span class="s2">&quot;delins&quot;</span>
</span><span id="L-860"><a href="#L-860"><span class="linenos"> 860</span></a>                <span class="k">else</span><span class="p">:</span>
</span><span id="L-861"><a href="#L-861"><span class="linenos"> 861</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">=</span> <span class="s2">&quot;&gt;&quot;</span>
</span><span id="L-862"><a href="#L-862"><span class="linenos"> 862</span></a>
</span><span id="L-863"><a href="#L-863"><span class="linenos"> 863</span></a>                <span class="c1"># Parse coordinates.</span>
</span><span id="L-864"><a href="#L-864"><span class="linenos"> 864</span></a>                <span class="bp">self</span><span class="o">.</span><span class="n">start</span> <span class="o">=</span> <span class="nb">int</span><span class="p">(</span><span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;start&quot;</span><span class="p">))</span>
</span><span id="L-865"><a href="#L-865"><span class="linenos"> 865</span></a>                <span class="k">if</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;end&quot;</span><span class="p">):</span>
</span><span id="L-866"><a href="#L-866"><span class="linenos"> 866</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">end</span> <span class="o">=</span> <span class="nb">int</span><span class="p">(</span><span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;end&quot;</span><span class="p">))</span>
</span><span id="L-867"><a href="#L-867"><span class="linenos"> 867</span></a>                <span class="k">else</span><span class="p">:</span>
</span><span id="L-868"><a href="#L-868"><span class="linenos"> 868</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">end</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">start</span>
</span><span id="L-869"><a href="#L-869"><span class="linenos"> 869</span></a>
</span><span id="L-870"><a href="#L-870"><span class="linenos"> 870</span></a>                <span class="c1"># Parse alleles.</span>
</span><span id="L-871"><a href="#L-871"><span class="linenos"> 871</span></a>                <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">=</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;ref&quot;</span><span class="p">,</span> <span class="s2">&quot;&quot;</span><span class="p">)</span>
</span><span id="L-872"><a href="#L-872"><span class="linenos"> 872</span></a>                <span class="k">if</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;ref2&quot;</span><span class="p">):</span>
</span><span id="L-873"><a href="#L-873"><span class="linenos"> 873</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">ref2_allele</span> <span class="o">=</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;ref2&quot;</span><span class="p">)</span>
</span><span id="L-874"><a href="#L-874"><span class="linenos"> 874</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span> <span class="o">=</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;alt&quot;</span><span class="p">,</span> <span class="s2">&quot;&quot;</span><span class="p">)</span>
</span><span id="L-875"><a href="#L-875"><span class="linenos"> 875</span></a>                <span class="k">else</span><span class="p">:</span>
</span><span id="L-876"><a href="#L-876"><span class="linenos"> 876</span></a>                    <span class="c1"># If alt is not given, assume matching with ref</span>
</span><span id="L-877"><a href="#L-877"><span class="linenos"> 877</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">ref2_allele</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span>
</span><span id="L-878"><a href="#L-878"><span class="linenos"> 878</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span> <span class="o">=</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;alt&quot;</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span><span class="p">)</span>
</span><span id="L-879"><a href="#L-879"><span class="linenos"> 879</span></a>
</span><span id="L-880"><a href="#L-880"><span class="linenos"> 880</span></a>                <span class="bp">self</span><span class="o">.</span><span class="n">pep_extra</span> <span class="o">=</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;extra&quot;</span><span class="p">)</span>
</span><span id="L-881"><a href="#L-881"><span class="linenos"> 881</span></a>                <span class="k">return</span>
</span><span id="L-882"><a href="#L-882"><span class="linenos"> 882</span></a>
</span><span id="L-883"><a href="#L-883"><span class="linenos"> 883</span></a>        <span class="k">raise</span> <span class="n">InvalidHGVSName</span><span class="p">(</span><span class="n">details</span><span class="p">,</span> <span class="s2">&quot;protein allele&quot;</span><span class="p">)</span>
</span><span id="L-884"><a href="#L-884"><span class="linenos"> 884</span></a>
</span><span id="L-885"><a href="#L-885"><span class="linenos"> 885</span></a>    <span class="k">def</span> <span class="nf">parse_genome</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">details</span><span class="p">:</span> <span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
</span><span id="L-886"><a href="#L-886"><span class="linenos"> 886</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-887"><a href="#L-887"><span class="linenos"> 887</span></a><span class="sd">        The function `parse_genome` is used to parse a HGVS genomic name and extract information such as</span>
</span><span id="L-888"><a href="#L-888"><span class="linenos"> 888</span></a><span class="sd">        mutation type, coordinates, and alleles.</span>
</span><span id="L-889"><a href="#L-889"><span class="linenos"> 889</span></a>
</span><span id="L-890"><a href="#L-890"><span class="linenos"> 890</span></a><span class="sd">        Some examples include:</span>
</span><span id="L-891"><a href="#L-891"><span class="linenos"> 891</span></a><span class="sd">          Substitution: 1000100A&gt;T</span>
</span><span id="L-892"><a href="#L-892"><span class="linenos"> 892</span></a><span class="sd">          Indel: 1000100_1000102delATG</span>
</span><span id="L-893"><a href="#L-893"><span class="linenos"> 893</span></a>
</span><span id="L-894"><a href="#L-894"><span class="linenos"> 894</span></a><span class="sd">        :param details: The `details` parameter is a string that represents a HGVS genomic name. It</span>
</span><span id="L-895"><a href="#L-895"><span class="linenos"> 895</span></a><span class="sd">        contains information about a genomic mutation, such as a substitution or an indel</span>
</span><span id="L-896"><a href="#L-896"><span class="linenos"> 896</span></a><span class="sd">        :type details: str</span>
</span><span id="L-897"><a href="#L-897"><span class="linenos"> 897</span></a><span class="sd">        :return: None.</span>
</span><span id="L-898"><a href="#L-898"><span class="linenos"> 898</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="L-899"><a href="#L-899"><span class="linenos"> 899</span></a>
</span><span id="L-900"><a href="#L-900"><span class="linenos"> 900</span></a>        <span class="k">for</span> <span class="n">regex</span> <span class="ow">in</span> <span class="n">HGVSRegex</span><span class="o">.</span><span class="n">GENOMIC_ALLELE_REGEXES</span><span class="p">:</span>
</span><span id="L-901"><a href="#L-901"><span class="linenos"> 901</span></a>            <span class="n">match</span> <span class="o">=</span> <span class="n">re</span><span class="o">.</span><span class="n">match</span><span class="p">(</span><span class="n">regex</span><span class="p">,</span> <span class="n">details</span><span class="p">)</span>
</span><span id="L-902"><a href="#L-902"><span class="linenos"> 902</span></a>            <span class="k">if</span> <span class="n">match</span><span class="p">:</span>
</span><span id="L-903"><a href="#L-903"><span class="linenos"> 903</span></a>                <span class="n">groups</span> <span class="o">=</span> <span class="n">match</span><span class="o">.</span><span class="n">groupdict</span><span class="p">()</span>
</span><span id="L-904"><a href="#L-904"><span class="linenos"> 904</span></a>
</span><span id="L-905"><a href="#L-905"><span class="linenos"> 905</span></a>                <span class="c1"># Parse mutation type.</span>
</span><span id="L-906"><a href="#L-906"><span class="linenos"> 906</span></a>                <span class="k">if</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;delins&quot;</span><span class="p">):</span>
</span><span id="L-907"><a href="#L-907"><span class="linenos"> 907</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">=</span> <span class="s2">&quot;delins&quot;</span>
</span><span id="L-908"><a href="#L-908"><span class="linenos"> 908</span></a>                <span class="k">else</span><span class="p">:</span>
</span><span id="L-909"><a href="#L-909"><span class="linenos"> 909</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">=</span> <span class="n">groups</span><span class="p">[</span><span class="s2">&quot;mutation_type&quot;</span><span class="p">]</span>
</span><span id="L-910"><a href="#L-910"><span class="linenos"> 910</span></a>
</span><span id="L-911"><a href="#L-911"><span class="linenos"> 911</span></a>                <span class="c1"># Parse coordinates.</span>
</span><span id="L-912"><a href="#L-912"><span class="linenos"> 912</span></a>                <span class="bp">self</span><span class="o">.</span><span class="n">start</span> <span class="o">=</span> <span class="nb">int</span><span class="p">(</span><span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;start&quot;</span><span class="p">))</span>
</span><span id="L-913"><a href="#L-913"><span class="linenos"> 913</span></a>                <span class="k">if</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;end&quot;</span><span class="p">):</span>
</span><span id="L-914"><a href="#L-914"><span class="linenos"> 914</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">end</span> <span class="o">=</span> <span class="nb">int</span><span class="p">(</span><span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;end&quot;</span><span class="p">))</span>
</span><span id="L-915"><a href="#L-915"><span class="linenos"> 915</span></a>                <span class="k">else</span><span class="p">:</span>
</span><span id="L-916"><a href="#L-916"><span class="linenos"> 916</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">end</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">start</span>
</span><span id="L-917"><a href="#L-917"><span class="linenos"> 917</span></a>
</span><span id="L-918"><a href="#L-918"><span class="linenos"> 918</span></a>                <span class="c1"># Parse alleles.</span>
</span><span id="L-919"><a href="#L-919"><span class="linenos"> 919</span></a>                <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">=</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;ref&quot;</span><span class="p">,</span> <span class="s2">&quot;&quot;</span><span class="p">)</span>
</span><span id="L-920"><a href="#L-920"><span class="linenos"> 920</span></a>                <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span> <span class="o">=</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;alt&quot;</span><span class="p">,</span> <span class="s2">&quot;&quot;</span><span class="p">)</span>
</span><span id="L-921"><a href="#L-921"><span class="linenos"> 921</span></a>
</span><span id="L-922"><a href="#L-922"><span class="linenos"> 922</span></a>                <span class="c1"># Convert numerical alleles.</span>
</span><span id="L-923"><a href="#L-923"><span class="linenos"> 923</span></a>                <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span><span class="o">.</span><span class="n">isdigit</span><span class="p">():</span>
</span><span id="L-924"><a href="#L-924"><span class="linenos"> 924</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">=</span> <span class="s2">&quot;N&quot;</span> <span class="o">*</span> <span class="nb">int</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span><span class="p">)</span>
</span><span id="L-925"><a href="#L-925"><span class="linenos"> 925</span></a>                <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span><span class="o">.</span><span class="n">isdigit</span><span class="p">():</span>
</span><span id="L-926"><a href="#L-926"><span class="linenos"> 926</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span> <span class="o">=</span> <span class="s2">&quot;N&quot;</span> <span class="o">*</span> <span class="nb">int</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span><span class="p">)</span>
</span><span id="L-927"><a href="#L-927"><span class="linenos"> 927</span></a>
</span><span id="L-928"><a href="#L-928"><span class="linenos"> 928</span></a>                <span class="c1"># Convert duplication alleles.</span>
</span><span id="L-929"><a href="#L-929"><span class="linenos"> 929</span></a>                <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">==</span> <span class="s2">&quot;dup&quot;</span><span class="p">:</span>
</span><span id="L-930"><a href="#L-930"><span class="linenos"> 930</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">*</span> <span class="mi">2</span>
</span><span id="L-931"><a href="#L-931"><span class="linenos"> 931</span></a>
</span><span id="L-932"><a href="#L-932"><span class="linenos"> 932</span></a>                <span class="c1"># Convert no match alleles.</span>
</span><span id="L-933"><a href="#L-933"><span class="linenos"> 933</span></a>                <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">==</span> <span class="s2">&quot;=&quot;</span><span class="p">:</span>
</span><span id="L-934"><a href="#L-934"><span class="linenos"> 934</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span>
</span><span id="L-935"><a href="#L-935"><span class="linenos"> 935</span></a>                <span class="k">return</span>
</span><span id="L-936"><a href="#L-936"><span class="linenos"> 936</span></a>
</span><span id="L-937"><a href="#L-937"><span class="linenos"> 937</span></a>        <span class="k">raise</span> <span class="n">InvalidHGVSName</span><span class="p">(</span><span class="n">details</span><span class="p">,</span> <span class="s2">&quot;genomic allele&quot;</span><span class="p">)</span>
</span><span id="L-938"><a href="#L-938"><span class="linenos"> 938</span></a>
</span><span id="L-939"><a href="#L-939"><span class="linenos"> 939</span></a>    <span class="k">def</span> <span class="nf">_validate</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
</span><span id="L-940"><a href="#L-940"><span class="linenos"> 940</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-941"><a href="#L-941"><span class="linenos"> 941</span></a><span class="sd">        The function checks for internal inconsistencies in the representation of coordinates.</span>
</span><span id="L-942"><a href="#L-942"><span class="linenos"> 942</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="L-943"><a href="#L-943"><span class="linenos"> 943</span></a>
</span><span id="L-944"><a href="#L-944"><span class="linenos"> 944</span></a>        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">start</span> <span class="o">&gt;</span> <span class="bp">self</span><span class="o">.</span><span class="n">end</span><span class="p">:</span>
</span><span id="L-945"><a href="#L-945"><span class="linenos"> 945</span></a>            <span class="k">raise</span> <span class="n">InvalidHGVSName</span><span class="p">(</span><span class="n">reason</span><span class="o">=</span><span class="s2">&quot;Coordinates are nonincreasing&quot;</span><span class="p">)</span>
</span><span id="L-946"><a href="#L-946"><span class="linenos"> 946</span></a>
</span><span id="L-947"><a href="#L-947"><span class="linenos"> 947</span></a>    <span class="k">def</span> <span class="fm">__repr__</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
</span><span id="L-948"><a href="#L-948"><span class="linenos"> 948</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-949"><a href="#L-949"><span class="linenos"> 949</span></a><span class="sd">        The function returns a string representation of an HGVSName object.</span>
</span><span id="L-950"><a href="#L-950"><span class="linenos"> 950</span></a><span class="sd">        :return: The `__repr__` method is returning a string representation of the object. If the</span>
</span><span id="L-951"><a href="#L-951"><span class="linenos"> 951</span></a><span class="sd">        `format` method is implemented, it will return a string in the format</span>
</span><span id="L-952"><a href="#L-952"><span class="linenos"> 952</span></a><span class="sd">        &quot;HGVSName(&#39;formatted_string&#39;)&quot;. If the `format` method is not implemented, it will return a</span>
</span><span id="L-953"><a href="#L-953"><span class="linenos"> 953</span></a><span class="sd">        string in the format &quot;HGVSName(&#39;name&#39;)&quot;.</span>
</span><span id="L-954"><a href="#L-954"><span class="linenos"> 954</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="L-955"><a href="#L-955"><span class="linenos"> 955</span></a>
</span><span id="L-956"><a href="#L-956"><span class="linenos"> 956</span></a>        <span class="k">try</span><span class="p">:</span>
</span><span id="L-957"><a href="#L-957"><span class="linenos"> 957</span></a>            <span class="k">return</span> <span class="s2">&quot;HGVSName(&#39;</span><span class="si">%s</span><span class="s2">&#39;)&quot;</span> <span class="o">%</span> <span class="bp">self</span><span class="o">.</span><span class="n">format</span><span class="p">()</span>
</span><span id="L-958"><a href="#L-958"><span class="linenos"> 958</span></a>        <span class="k">except</span> <span class="ne">NotImplementedError</span><span class="p">:</span>
</span><span id="L-959"><a href="#L-959"><span class="linenos"> 959</span></a>            <span class="k">return</span> <span class="s2">&quot;HGVSName(&#39;</span><span class="si">%s</span><span class="s2">&#39;)&quot;</span> <span class="o">%</span> <span class="bp">self</span><span class="o">.</span><span class="n">name</span>
</span><span id="L-960"><a href="#L-960"><span class="linenos"> 960</span></a>
</span><span id="L-961"><a href="#L-961"><span class="linenos"> 961</span></a>    <span class="k">def</span> <span class="nf">__unicode__</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
</span><span id="L-962"><a href="#L-962"><span class="linenos"> 962</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-963"><a href="#L-963"><span class="linenos"> 963</span></a><span class="sd">        The function returns a formatted string representation of the object.</span>
</span><span id="L-964"><a href="#L-964"><span class="linenos"> 964</span></a><span class="sd">        :return: The `__unicode__` method is returning the result of the `format()` method.</span>
</span><span id="L-965"><a href="#L-965"><span class="linenos"> 965</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="L-966"><a href="#L-966"><span class="linenos"> 966</span></a>
</span><span id="L-967"><a href="#L-967"><span class="linenos"> 967</span></a>        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">format</span><span class="p">()</span>
</span><span id="L-968"><a href="#L-968"><span class="linenos"> 968</span></a>
</span><span id="L-969"><a href="#L-969"><span class="linenos"> 969</span></a>    <span class="k">def</span> <span class="nf">format</span><span class="p">(</span>
</span><span id="L-970"><a href="#L-970"><span class="linenos"> 970</span></a>        <span class="bp">self</span><span class="p">,</span>
</span><span id="L-971"><a href="#L-971"><span class="linenos"> 971</span></a>        <span class="n">use_prefix</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span>
</span><span id="L-972"><a href="#L-972"><span class="linenos"> 972</span></a>        <span class="n">use_gene</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span>
</span><span id="L-973"><a href="#L-973"><span class="linenos"> 973</span></a>        <span class="n">use_exon</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
</span><span id="L-974"><a href="#L-974"><span class="linenos"> 974</span></a>        <span class="n">use_protein</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
</span><span id="L-975"><a href="#L-975"><span class="linenos"> 975</span></a>        <span class="n">full_format</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
</span><span id="L-976"><a href="#L-976"><span class="linenos"> 976</span></a>        <span class="n">use_version</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
</span><span id="L-977"><a href="#L-977"><span class="linenos"> 977</span></a>    <span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
</span><span id="L-978"><a href="#L-978"><span class="linenos"> 978</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-979"><a href="#L-979"><span class="linenos"> 979</span></a><span class="sd">        The `format` function generates a HGVS name as a string based on various formatting options.</span>
</span><span id="L-980"><a href="#L-980"><span class="linenos"> 980</span></a>
</span><span id="L-981"><a href="#L-981"><span class="linenos"> 981</span></a><span class="sd">        :param use_prefix: A boolean indicating whether to include the prefix in the HGVS name. If set</span>
</span><span id="L-982"><a href="#L-982"><span class="linenos"> 982</span></a><span class="sd">        to True, the prefix will be included in the HGVS name. If set to False, the prefix will be</span>
</span><span id="L-983"><a href="#L-983"><span class="linenos"> 983</span></a><span class="sd">        excluded. The default value is True, defaults to True</span>
</span><span id="L-984"><a href="#L-984"><span class="linenos"> 984</span></a><span class="sd">        :type use_prefix: bool (optional)</span>
</span><span id="L-985"><a href="#L-985"><span class="linenos"> 985</span></a><span class="sd">        :param use_gene: A boolean indicating whether to include the gene name in the HGVS name. If set</span>
</span><span id="L-986"><a href="#L-986"><span class="linenos"> 986</span></a><span class="sd">        to True, the gene name will be included in the HGVS name. If set to False, the gene name will</span>
</span><span id="L-987"><a href="#L-987"><span class="linenos"> 987</span></a><span class="sd">        not be included. The default value is True, defaults to True</span>
</span><span id="L-988"><a href="#L-988"><span class="linenos"> 988</span></a><span class="sd">        :type use_gene: bool (optional)</span>
</span><span id="L-989"><a href="#L-989"><span class="linenos"> 989</span></a><span class="sd">        :param use_exon: A boolean indicating whether to include exon information in the HGVS name. If</span>
</span><span id="L-990"><a href="#L-990"><span class="linenos"> 990</span></a><span class="sd">        set to True, exon information will be included in the HGVS name. If set to False, exon</span>
</span><span id="L-991"><a href="#L-991"><span class="linenos"> 991</span></a><span class="sd">        information will not be included, defaults to False</span>
</span><span id="L-992"><a href="#L-992"><span class="linenos"> 992</span></a><span class="sd">        :type use_exon: bool (optional)</span>
</span><span id="L-993"><a href="#L-993"><span class="linenos"> 993</span></a><span class="sd">        :param use_protein: A boolean indicating whether to include the protein change in the HGVS name.</span>
</span><span id="L-994"><a href="#L-994"><span class="linenos"> 994</span></a><span class="sd">        If set to True, the protein change will be included in the HGVS name. If set to False, the</span>
</span><span id="L-995"><a href="#L-995"><span class="linenos"> 995</span></a><span class="sd">        protein change will not be included, defaults to False</span>
</span><span id="L-996"><a href="#L-996"><span class="linenos"> 996</span></a><span class="sd">        :type use_protein: bool (optional)</span>
</span><span id="L-997"><a href="#L-997"><span class="linenos"> 997</span></a><span class="sd">        :param full_format: A boolean parameter that determines whether the full format of the allele</span>
</span><span id="L-998"><a href="#L-998"><span class="linenos"> 998</span></a><span class="sd">        should be included in the output. If set to True, and if the allele is not a protein variant,</span>
</span><span id="L-999"><a href="#L-999"><span class="linenos"> 999</span></a><span class="sd">        the allele will be appended with &#39;:p.&#39; followed by the formatted protein variant, defaults to</span>
</span><span id="L-1000"><a href="#L-1000"><span class="linenos">1000</span></a><span class="sd">        False (optional)</span>
</span><span id="L-1001"><a href="#L-1001"><span class="linenos">1001</span></a><span class="sd">        :param use_version: A boolean parameter that determines whether to include the version number in</span>
</span><span id="L-1002"><a href="#L-1002"><span class="linenos">1002</span></a><span class="sd">        the formatted HGVS name. If set to True, the version number will be included in the output. If</span>
</span><span id="L-1003"><a href="#L-1003"><span class="linenos">1003</span></a><span class="sd">        set to False, the version number will not be included, defaults to False</span>
</span><span id="L-1004"><a href="#L-1004"><span class="linenos">1004</span></a><span class="sd">        :type use_version: bool (optional)</span>
</span><span id="L-1005"><a href="#L-1005"><span class="linenos">1005</span></a><span class="sd">        :return: a HGVS name as a string.</span>
</span><span id="L-1006"><a href="#L-1006"><span class="linenos">1006</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="L-1007"><a href="#L-1007"><span class="linenos">1007</span></a>
</span><span id="L-1008"><a href="#L-1008"><span class="linenos">1008</span></a>        <span class="k">if</span> <span class="n">use_protein</span> <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">format_protein</span><span class="p">():</span>
</span><span id="L-1009"><a href="#L-1009"><span class="linenos">1009</span></a>            <span class="n">allele</span> <span class="o">=</span> <span class="s2">&quot;p.&quot;</span> <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">format_protein</span><span class="p">()</span>
</span><span id="L-1010"><a href="#L-1010"><span class="linenos">1010</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="L-1011"><a href="#L-1011"><span class="linenos">1011</span></a>            <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">kind</span> <span class="ow">in</span> <span class="p">(</span><span class="s2">&quot;c&quot;</span><span class="p">,</span> <span class="s2">&quot;n&quot;</span><span class="p">):</span>
</span><span id="L-1012"><a href="#L-1012"><span class="linenos">1012</span></a>                <span class="n">allele</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">kind</span> <span class="o">+</span> <span class="s2">&quot;.&quot;</span> <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">format_cdna</span><span class="p">()</span>
</span><span id="L-1013"><a href="#L-1013"><span class="linenos">1013</span></a>            <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">kind</span> <span class="o">==</span> <span class="s2">&quot;p&quot;</span><span class="p">:</span>
</span><span id="L-1014"><a href="#L-1014"><span class="linenos">1014</span></a>                <span class="n">allele</span> <span class="o">=</span> <span class="s2">&quot;p.&quot;</span> <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">format_protein</span><span class="p">()</span>
</span><span id="L-1015"><a href="#L-1015"><span class="linenos">1015</span></a>            <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">kind</span> <span class="ow">in</span> <span class="p">(</span><span class="s2">&quot;g&quot;</span><span class="p">,</span> <span class="s2">&quot;m&quot;</span><span class="p">):</span>
</span><span id="L-1016"><a href="#L-1016"><span class="linenos">1016</span></a>                <span class="n">allele</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">kind</span> <span class="o">+</span> <span class="s2">&quot;.&quot;</span> <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">format_genome</span><span class="p">()</span>
</span><span id="L-1017"><a href="#L-1017"><span class="linenos">1017</span></a>            <span class="k">else</span><span class="p">:</span>
</span><span id="L-1018"><a href="#L-1018"><span class="linenos">1018</span></a>                <span class="k">raise</span> <span class="ne">NotImplementedError</span><span class="p">(</span><span class="s2">&quot;not implemented: &#39;</span><span class="si">%s</span><span class="s2">&#39;&quot;</span> <span class="o">%</span> <span class="bp">self</span><span class="o">.</span><span class="n">kind</span><span class="p">)</span>
</span><span id="L-1019"><a href="#L-1019"><span class="linenos">1019</span></a>
</span><span id="L-1020"><a href="#L-1020"><span class="linenos">1020</span></a>        <span class="k">if</span> <span class="n">full_format</span> <span class="ow">and</span> <span class="ow">not</span> <span class="n">use_protein</span> <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">format_protein</span><span class="p">():</span>
</span><span id="L-1021"><a href="#L-1021"><span class="linenos">1021</span></a>            <span class="n">allele</span> <span class="o">+=</span> <span class="s2">&quot;:p.&quot;</span> <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">format_protein</span><span class="p">()</span>
</span><span id="L-1022"><a href="#L-1022"><span class="linenos">1022</span></a>
</span><span id="L-1023"><a href="#L-1023"><span class="linenos">1023</span></a>        <span class="n">prefix</span> <span class="o">=</span> <span class="p">(</span>
</span><span id="L-1024"><a href="#L-1024"><span class="linenos">1024</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">format_prefix</span><span class="p">(</span>
</span><span id="L-1025"><a href="#L-1025"><span class="linenos">1025</span></a>                <span class="n">use_gene</span><span class="o">=</span><span class="n">use_gene</span><span class="p">,</span>
</span><span id="L-1026"><a href="#L-1026"><span class="linenos">1026</span></a>                <span class="n">use_exon</span><span class="o">=</span><span class="n">use_exon</span><span class="p">,</span>
</span><span id="L-1027"><a href="#L-1027"><span class="linenos">1027</span></a>                <span class="n">use_protein</span><span class="o">=</span><span class="n">use_protein</span><span class="p">,</span>
</span><span id="L-1028"><a href="#L-1028"><span class="linenos">1028</span></a>                <span class="n">full_format</span><span class="o">=</span><span class="n">full_format</span><span class="p">,</span>
</span><span id="L-1029"><a href="#L-1029"><span class="linenos">1029</span></a>                <span class="n">use_version</span><span class="o">=</span><span class="n">use_version</span><span class="p">,</span>
</span><span id="L-1030"><a href="#L-1030"><span class="linenos">1030</span></a>            <span class="p">)</span>
</span><span id="L-1031"><a href="#L-1031"><span class="linenos">1031</span></a>            <span class="k">if</span> <span class="n">use_prefix</span>
</span><span id="L-1032"><a href="#L-1032"><span class="linenos">1032</span></a>            <span class="k">else</span> <span class="s2">&quot;&quot;</span>
</span><span id="L-1033"><a href="#L-1033"><span class="linenos">1033</span></a>        <span class="p">)</span>
</span><span id="L-1034"><a href="#L-1034"><span class="linenos">1034</span></a>
</span><span id="L-1035"><a href="#L-1035"><span class="linenos">1035</span></a>        <span class="k">if</span> <span class="n">prefix</span><span class="p">:</span>
</span><span id="L-1036"><a href="#L-1036"><span class="linenos">1036</span></a>            <span class="k">return</span> <span class="n">prefix</span> <span class="o">+</span> <span class="s2">&quot;:&quot;</span> <span class="o">+</span> <span class="n">allele</span>
</span><span id="L-1037"><a href="#L-1037"><span class="linenos">1037</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="L-1038"><a href="#L-1038"><span class="linenos">1038</span></a>            <span class="k">return</span> <span class="n">allele</span>
</span><span id="L-1039"><a href="#L-1039"><span class="linenos">1039</span></a>
</span><span id="L-1040"><a href="#L-1040"><span class="linenos">1040</span></a>    <span class="k">def</span> <span class="nf">format_prefix</span><span class="p">(</span>
</span><span id="L-1041"><a href="#L-1041"><span class="linenos">1041</span></a>        <span class="bp">self</span><span class="p">,</span>
</span><span id="L-1042"><a href="#L-1042"><span class="linenos">1042</span></a>        <span class="n">use_gene</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span>
</span><span id="L-1043"><a href="#L-1043"><span class="linenos">1043</span></a>        <span class="n">use_exon</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
</span><span id="L-1044"><a href="#L-1044"><span class="linenos">1044</span></a>        <span class="n">use_protein</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
</span><span id="L-1045"><a href="#L-1045"><span class="linenos">1045</span></a>        <span class="n">full_format</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
</span><span id="L-1046"><a href="#L-1046"><span class="linenos">1046</span></a>        <span class="n">use_version</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
</span><span id="L-1047"><a href="#L-1047"><span class="linenos">1047</span></a>    <span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
</span><span id="L-1048"><a href="#L-1048"><span class="linenos">1048</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-1049"><a href="#L-1049"><span class="linenos">1049</span></a><span class="sd">        The `format_prefix` function generates an HGVS transcript/gene prefix based on various</span>
</span><span id="L-1050"><a href="#L-1050"><span class="linenos">1050</span></a><span class="sd">        parameters.</span>
</span><span id="L-1051"><a href="#L-1051"><span class="linenos">1051</span></a>
</span><span id="L-1052"><a href="#L-1052"><span class="linenos">1052</span></a><span class="sd">        :param use_gene: A boolean parameter that determines whether to include the gene name in the</span>
</span><span id="L-1053"><a href="#L-1053"><span class="linenos">1053</span></a><span class="sd">        prefix. If set to True, the gene name will be included in the prefix. If set to False, the gene</span>
</span><span id="L-1054"><a href="#L-1054"><span class="linenos">1054</span></a><span class="sd">        name will not be included in the prefix. The default value is True, defaults to True</span>
</span><span id="L-1055"><a href="#L-1055"><span class="linenos">1055</span></a><span class="sd">        :type use_gene: bool (optional)</span>
</span><span id="L-1056"><a href="#L-1056"><span class="linenos">1056</span></a><span class="sd">        :param use_exon: A boolean parameter that determines whether to include the exon information in</span>
</span><span id="L-1057"><a href="#L-1057"><span class="linenos">1057</span></a><span class="sd">        the prefix. If set to True, the exon information will be included in the prefix. If set to</span>
</span><span id="L-1058"><a href="#L-1058"><span class="linenos">1058</span></a><span class="sd">        False, the exon information will not be included, defaults to False</span>
</span><span id="L-1059"><a href="#L-1059"><span class="linenos">1059</span></a><span class="sd">        :type use_exon: bool (optional)</span>
</span><span id="L-1060"><a href="#L-1060"><span class="linenos">1060</span></a><span class="sd">        :param use_protein: A boolean indicating whether to use the protein transcript instead of the</span>
</span><span id="L-1061"><a href="#L-1061"><span class="linenos">1061</span></a><span class="sd">        nucleotide transcript if available. If set to True, the protein transcript will be used. If set</span>
</span><span id="L-1062"><a href="#L-1062"><span class="linenos">1062</span></a><span class="sd">        to False, the nucleotide transcript will be used. The default value is False, defaults to False</span>
</span><span id="L-1063"><a href="#L-1063"><span class="linenos">1063</span></a><span class="sd">        :type use_protein: bool (optional)</span>
</span><span id="L-1064"><a href="#L-1064"><span class="linenos">1064</span></a><span class="sd">        :param full_format: A boolean parameter that determines whether to generate the full HGVS name</span>
</span><span id="L-1065"><a href="#L-1065"><span class="linenos">1065</span></a><span class="sd">        with transcript/gene prefix or not. If set to True, the full format will be generated. If set to</span>
</span><span id="L-1066"><a href="#L-1066"><span class="linenos">1066</span></a><span class="sd">        False, only the transcript/gene prefix will be generated, defaults to False</span>
</span><span id="L-1067"><a href="#L-1067"><span class="linenos">1067</span></a><span class="sd">        :type full_format: bool (optional)</span>
</span><span id="L-1068"><a href="#L-1068"><span class="linenos">1068</span></a><span class="sd">        :param use_version: A boolean parameter that determines whether to include the version number in</span>
</span><span id="L-1069"><a href="#L-1069"><span class="linenos">1069</span></a><span class="sd">        the transcript prefix. If set to True, the version number will be included in the prefix (e.g.,</span>
</span><span id="L-1070"><a href="#L-1070"><span class="linenos">1070</span></a><span class="sd">        NM_007294.3). If set to False, only the transcript ID without the version number will be</span>
</span><span id="L-1071"><a href="#L-1071"><span class="linenos">1071</span></a><span class="sd">        included in the prefix, defaults to False</span>
</span><span id="L-1072"><a href="#L-1072"><span class="linenos">1072</span></a><span class="sd">        :type use_version: bool (optional)</span>
</span><span id="L-1073"><a href="#L-1073"><span class="linenos">1073</span></a><span class="sd">        :return: The function `format_prefix` returns a formatted HGVS transcript/gene prefix as a</span>
</span><span id="L-1074"><a href="#L-1074"><span class="linenos">1074</span></a><span class="sd">        string.</span>
</span><span id="L-1075"><a href="#L-1075"><span class="linenos">1075</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="L-1076"><a href="#L-1076"><span class="linenos">1076</span></a>
</span><span id="L-1077"><a href="#L-1077"><span class="linenos">1077</span></a>        <span class="k">if</span> <span class="n">full_format</span><span class="p">:</span>
</span><span id="L-1078"><a href="#L-1078"><span class="linenos">1078</span></a>
</span><span id="L-1079"><a href="#L-1079"><span class="linenos">1079</span></a>            <span class="n">prefix</span> <span class="o">=</span> <span class="p">[]</span>
</span><span id="L-1080"><a href="#L-1080"><span class="linenos">1080</span></a>            <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">gene</span><span class="p">:</span>
</span><span id="L-1081"><a href="#L-1081"><span class="linenos">1081</span></a>                <span class="n">prefix</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">gene</span><span class="p">)</span>
</span><span id="L-1082"><a href="#L-1082"><span class="linenos">1082</span></a>            <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span><span class="p">:</span>
</span><span id="L-1083"><a href="#L-1083"><span class="linenos">1083</span></a>                <span class="n">prefix</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">transcript</span><span class="p">)</span>
</span><span id="L-1084"><a href="#L-1084"><span class="linenos">1084</span></a>            <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">transcript_protein</span><span class="p">:</span>
</span><span id="L-1085"><a href="#L-1085"><span class="linenos">1085</span></a>                <span class="n">prefix</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">transcript_protein</span><span class="p">)</span>
</span><span id="L-1086"><a href="#L-1086"><span class="linenos">1086</span></a>            <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">exon</span><span class="p">:</span>  <span class="c1"># TODO</span>
</span><span id="L-1087"><a href="#L-1087"><span class="linenos">1087</span></a>                <span class="n">prefix</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;exon&quot;</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">exon</span><span class="p">))</span>
</span><span id="L-1088"><a href="#L-1088"><span class="linenos">1088</span></a>            <span class="k">return</span> <span class="s2">&quot;:&quot;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">prefix</span><span class="p">)</span>
</span><span id="L-1089"><a href="#L-1089"><span class="linenos">1089</span></a>
</span><span id="L-1090"><a href="#L-1090"><span class="linenos">1090</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="L-1091"><a href="#L-1091"><span class="linenos">1091</span></a>
</span><span id="L-1092"><a href="#L-1092"><span class="linenos">1092</span></a>            <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">kind</span> <span class="ow">in</span> <span class="p">(</span><span class="s2">&quot;g&quot;</span><span class="p">,</span> <span class="s2">&quot;m&quot;</span><span class="p">):</span>
</span><span id="L-1093"><a href="#L-1093"><span class="linenos">1093</span></a>                <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">chrom</span><span class="p">:</span>
</span><span id="L-1094"><a href="#L-1094"><span class="linenos">1094</span></a>                    <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">chrom</span>
</span><span id="L-1095"><a href="#L-1095"><span class="linenos">1095</span></a>
</span><span id="L-1096"><a href="#L-1096"><span class="linenos">1096</span></a>            <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span><span class="p">:</span>
</span><span id="L-1097"><a href="#L-1097"><span class="linenos">1097</span></a>                <span class="k">if</span> <span class="n">use_protein</span> <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">transcript_protein</span><span class="p">:</span>
</span><span id="L-1098"><a href="#L-1098"><span class="linenos">1098</span></a>                    <span class="n">transcript</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">transcript_protein</span>
</span><span id="L-1099"><a href="#L-1099"><span class="linenos">1099</span></a>                <span class="k">else</span><span class="p">:</span>
</span><span id="L-1100"><a href="#L-1100"><span class="linenos">1100</span></a>                    <span class="n">transcript</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span>
</span><span id="L-1101"><a href="#L-1101"><span class="linenos">1101</span></a>                <span class="k">if</span> <span class="ow">not</span> <span class="n">use_version</span> <span class="ow">and</span> <span class="n">transcript</span><span class="p">:</span>
</span><span id="L-1102"><a href="#L-1102"><span class="linenos">1102</span></a>                    <span class="n">transcript</span> <span class="o">=</span> <span class="n">transcript</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot;.&quot;</span><span class="p">)[</span><span class="mi">0</span><span class="p">]</span>
</span><span id="L-1103"><a href="#L-1103"><span class="linenos">1103</span></a>                <span class="k">if</span> <span class="n">use_gene</span> <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">gene</span><span class="p">:</span>
</span><span id="L-1104"><a href="#L-1104"><span class="linenos">1104</span></a>                    <span class="k">return</span> <span class="s2">&quot;</span><span class="si">%s</span><span class="s2">(</span><span class="si">%s</span><span class="s2">)&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">transcript</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">gene</span><span class="p">)</span>
</span><span id="L-1105"><a href="#L-1105"><span class="linenos">1105</span></a>                <span class="k">elif</span> <span class="n">use_exon</span> <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">exon</span><span class="p">:</span>
</span><span id="L-1106"><a href="#L-1106"><span class="linenos">1106</span></a>                    <span class="k">return</span> <span class="s2">&quot;</span><span class="si">%s</span><span class="s2">(exon</span><span class="si">%s</span><span class="s2">)&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">transcript</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">exon</span><span class="p">)</span>
</span><span id="L-1107"><a href="#L-1107"><span class="linenos">1107</span></a>                <span class="k">else</span><span class="p">:</span>
</span><span id="L-1108"><a href="#L-1108"><span class="linenos">1108</span></a>                    <span class="k">return</span> <span class="n">transcript</span>
</span><span id="L-1109"><a href="#L-1109"><span class="linenos">1109</span></a>            <span class="k">else</span><span class="p">:</span>
</span><span id="L-1110"><a href="#L-1110"><span class="linenos">1110</span></a>                <span class="k">if</span> <span class="n">use_gene</span><span class="p">:</span>
</span><span id="L-1111"><a href="#L-1111"><span class="linenos">1111</span></a>                    <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">gene</span>
</span><span id="L-1112"><a href="#L-1112"><span class="linenos">1112</span></a>                <span class="k">else</span><span class="p">:</span>
</span><span id="L-1113"><a href="#L-1113"><span class="linenos">1113</span></a>                    <span class="k">return</span> <span class="s2">&quot;&quot;</span>
</span><span id="L-1114"><a href="#L-1114"><span class="linenos">1114</span></a>
</span><span id="L-1115"><a href="#L-1115"><span class="linenos">1115</span></a>    <span class="k">def</span> <span class="nf">format_cdna_coords</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
</span><span id="L-1116"><a href="#L-1116"><span class="linenos">1116</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-1117"><a href="#L-1117"><span class="linenos">1117</span></a><span class="sd">        The function `format_cdna_coords` generates a string representing HGVS cDNA coordinates,</span>
</span><span id="L-1118"><a href="#L-1118"><span class="linenos">1118</span></a><span class="sd">        returning either the start coordinate or a string in the format &quot;start_end&quot; depending on whether</span>
</span><span id="L-1119"><a href="#L-1119"><span class="linenos">1119</span></a><span class="sd">        the start and end coordinates are the same or not.</span>
</span><span id="L-1120"><a href="#L-1120"><span class="linenos">1120</span></a><span class="sd">        :return: a string representing the cDNA coordinates. If the start and end coordinates are the</span>
</span><span id="L-1121"><a href="#L-1121"><span class="linenos">1121</span></a><span class="sd">        same, it returns just the start coordinate. Otherwise, it returns a string in the format</span>
</span><span id="L-1122"><a href="#L-1122"><span class="linenos">1122</span></a><span class="sd">        &quot;start_end&quot;.</span>
</span><span id="L-1123"><a href="#L-1123"><span class="linenos">1123</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="L-1124"><a href="#L-1124"><span class="linenos">1124</span></a>
</span><span id="L-1125"><a href="#L-1125"><span class="linenos">1125</span></a>        <span class="c1"># Format coordinates.</span>
</span><span id="L-1126"><a href="#L-1126"><span class="linenos">1126</span></a>        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">cdna_start</span> <span class="o">==</span> <span class="bp">self</span><span class="o">.</span><span class="n">cdna_end</span><span class="p">:</span>
</span><span id="L-1127"><a href="#L-1127"><span class="linenos">1127</span></a>            <span class="k">return</span> <span class="nb">str</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">cdna_start</span><span class="p">)</span>
</span><span id="L-1128"><a href="#L-1128"><span class="linenos">1128</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="L-1129"><a href="#L-1129"><span class="linenos">1129</span></a>            <span class="k">return</span> <span class="s2">&quot;</span><span class="si">%s</span><span class="s2">_</span><span class="si">%s</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">cdna_start</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">cdna_end</span><span class="p">)</span>
</span><span id="L-1130"><a href="#L-1130"><span class="linenos">1130</span></a>
</span><span id="L-1131"><a href="#L-1131"><span class="linenos">1131</span></a>    <span class="k">def</span> <span class="nf">format_dna_allele</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
</span><span id="L-1132"><a href="#L-1132"><span class="linenos">1132</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-1133"><a href="#L-1133"><span class="linenos">1133</span></a><span class="sd">        The function `format_dna_allele` generates an HGVS DNA allele based on the mutation type and</span>
</span><span id="L-1134"><a href="#L-1134"><span class="linenos">1134</span></a><span class="sd">        alleles provided.</span>
</span><span id="L-1135"><a href="#L-1135"><span class="linenos">1135</span></a><span class="sd">        :return: The function `format_dna_allele` returns a string representing the HGVS DNA allele. The</span>
</span><span id="L-1136"><a href="#L-1136"><span class="linenos">1136</span></a><span class="sd">        specific format of the returned string depends on the value of the `mutation_type` attribute of</span>
</span><span id="L-1137"><a href="#L-1137"><span class="linenos">1137</span></a><span class="sd">        the object. The possible return values are:</span>
</span><span id="L-1138"><a href="#L-1138"><span class="linenos">1138</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="L-1139"><a href="#L-1139"><span class="linenos">1139</span></a>
</span><span id="L-1140"><a href="#L-1140"><span class="linenos">1140</span></a>        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">==</span> <span class="s2">&quot;=&quot;</span><span class="p">:</span>
</span><span id="L-1141"><a href="#L-1141"><span class="linenos">1141</span></a>            <span class="c1"># No change.</span>
</span><span id="L-1142"><a href="#L-1142"><span class="linenos">1142</span></a>            <span class="c1"># example: 101A=</span>
</span><span id="L-1143"><a href="#L-1143"><span class="linenos">1143</span></a>            <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">+</span> <span class="s2">&quot;=&quot;</span>
</span><span id="L-1144"><a href="#L-1144"><span class="linenos">1144</span></a>
</span><span id="L-1145"><a href="#L-1145"><span class="linenos">1145</span></a>        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">==</span> <span class="s2">&quot;&gt;&quot;</span><span class="p">:</span>
</span><span id="L-1146"><a href="#L-1146"><span class="linenos">1146</span></a>            <span class="c1"># SNP.</span>
</span><span id="L-1147"><a href="#L-1147"><span class="linenos">1147</span></a>            <span class="c1"># example: 101A&gt;C</span>
</span><span id="L-1148"><a href="#L-1148"><span class="linenos">1148</span></a>            <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">+</span> <span class="s2">&quot;&gt;&quot;</span> <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span>
</span><span id="L-1149"><a href="#L-1149"><span class="linenos">1149</span></a>
</span><span id="L-1150"><a href="#L-1150"><span class="linenos">1150</span></a>        <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">==</span> <span class="s2">&quot;delins&quot;</span><span class="p">:</span>
</span><span id="L-1151"><a href="#L-1151"><span class="linenos">1151</span></a>            <span class="c1"># Indel.</span>
</span><span id="L-1152"><a href="#L-1152"><span class="linenos">1152</span></a>            <span class="c1"># example: 112_117delAGGTCAinsTG, 112_117delinsTG</span>
</span><span id="L-1153"><a href="#L-1153"><span class="linenos">1153</span></a>            <span class="k">return</span> <span class="s2">&quot;del&quot;</span> <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">+</span> <span class="s2">&quot;ins&quot;</span> <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span>
</span><span id="L-1154"><a href="#L-1154"><span class="linenos">1154</span></a>
</span><span id="L-1155"><a href="#L-1155"><span class="linenos">1155</span></a>        <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="ow">in</span> <span class="p">(</span><span class="s2">&quot;del&quot;</span><span class="p">,</span> <span class="s2">&quot;dup&quot;</span><span class="p">):</span>
</span><span id="L-1156"><a href="#L-1156"><span class="linenos">1156</span></a>            <span class="c1"># Delete, duplication.</span>
</span><span id="L-1157"><a href="#L-1157"><span class="linenos">1157</span></a>            <span class="c1"># example: 1000_1003delATG, 1000_1003dupATG</span>
</span><span id="L-1158"><a href="#L-1158"><span class="linenos">1158</span></a>            <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span>
</span><span id="L-1159"><a href="#L-1159"><span class="linenos">1159</span></a>
</span><span id="L-1160"><a href="#L-1160"><span class="linenos">1160</span></a>        <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">==</span> <span class="s2">&quot;ins&quot;</span><span class="p">:</span>
</span><span id="L-1161"><a href="#L-1161"><span class="linenos">1161</span></a>            <span class="c1"># Insert.</span>
</span><span id="L-1162"><a href="#L-1162"><span class="linenos">1162</span></a>            <span class="c1"># example: 1000_1001insATG</span>
</span><span id="L-1163"><a href="#L-1163"><span class="linenos">1163</span></a>            <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span>
</span><span id="L-1164"><a href="#L-1164"><span class="linenos">1164</span></a>
</span><span id="L-1165"><a href="#L-1165"><span class="linenos">1165</span></a>        <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">==</span> <span class="s2">&quot;inv&quot;</span><span class="p">:</span>
</span><span id="L-1166"><a href="#L-1166"><span class="linenos">1166</span></a>            <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span>
</span><span id="L-1167"><a href="#L-1167"><span class="linenos">1167</span></a>
</span><span id="L-1168"><a href="#L-1168"><span class="linenos">1168</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="L-1169"><a href="#L-1169"><span class="linenos">1169</span></a>            <span class="k">raise</span> <span class="ne">AssertionError</span><span class="p">(</span><span class="s2">&quot;unknown mutation type: &#39;</span><span class="si">%s</span><span class="s2">&#39;&quot;</span> <span class="o">%</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span><span class="p">)</span>
</span><span id="L-1170"><a href="#L-1170"><span class="linenos">1170</span></a>
</span><span id="L-1171"><a href="#L-1171"><span class="linenos">1171</span></a>    <span class="k">def</span> <span class="nf">format_cdna</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
</span><span id="L-1172"><a href="#L-1172"><span class="linenos">1172</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-1173"><a href="#L-1173"><span class="linenos">1173</span></a><span class="sd">        The function &quot;format_cdna&quot; generates an HGVS cDNA allele by combining the cDNA coordinates and</span>
</span><span id="L-1174"><a href="#L-1174"><span class="linenos">1174</span></a><span class="sd">        the DNA allele.</span>
</span><span id="L-1175"><a href="#L-1175"><span class="linenos">1175</span></a>
</span><span id="L-1176"><a href="#L-1176"><span class="linenos">1176</span></a><span class="sd">        Some examples include:</span>
</span><span id="L-1177"><a href="#L-1177"><span class="linenos">1177</span></a><span class="sd">          Substitution: 101A&gt;C,</span>
</span><span id="L-1178"><a href="#L-1178"><span class="linenos">1178</span></a><span class="sd">          Indel: 3428delCinsTA, 1000_1003delATG, 1000_1001insATG</span>
</span><span id="L-1179"><a href="#L-1179"><span class="linenos">1179</span></a>
</span><span id="L-1180"><a href="#L-1180"><span class="linenos">1180</span></a><span class="sd">        :return: a string that represents the HGVS cDNA allele.</span>
</span><span id="L-1181"><a href="#L-1181"><span class="linenos">1181</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="L-1182"><a href="#L-1182"><span class="linenos">1182</span></a>
</span><span id="L-1183"><a href="#L-1183"><span class="linenos">1183</span></a>        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">format_cdna_coords</span><span class="p">()</span> <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">format_dna_allele</span><span class="p">()</span>
</span><span id="L-1184"><a href="#L-1184"><span class="linenos">1184</span></a>
</span><span id="L-1185"><a href="#L-1185"><span class="linenos">1185</span></a>    <span class="k">def</span> <span class="nf">format_protein</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
</span><span id="L-1186"><a href="#L-1186"><span class="linenos">1186</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-1187"><a href="#L-1187"><span class="linenos">1187</span></a><span class="sd">        The `format_protein` function generates an HGVS protein name based on different scenarios such</span>
</span><span id="L-1188"><a href="#L-1188"><span class="linenos">1188</span></a><span class="sd">        as no change, change, frameshift, and range change.</span>
</span><span id="L-1189"><a href="#L-1189"><span class="linenos">1189</span></a>
</span><span id="L-1190"><a href="#L-1190"><span class="linenos">1190</span></a><span class="sd">        Some examples include:</span>
</span><span id="L-1191"><a href="#L-1191"><span class="linenos">1191</span></a><span class="sd">          No change: Glu1161=</span>
</span><span id="L-1192"><a href="#L-1192"><span class="linenos">1192</span></a><span class="sd">          Change: Glu1161Ser</span>
</span><span id="L-1193"><a href="#L-1193"><span class="linenos">1193</span></a><span class="sd">          Frameshift: Glu1161_Ser1164?fs</span>
</span><span id="L-1194"><a href="#L-1194"><span class="linenos">1194</span></a>
</span><span id="L-1195"><a href="#L-1195"><span class="linenos">1195</span></a><span class="sd">        :return: The method `format_protein` returns a string representing the HGVS protein name.</span>
</span><span id="L-1196"><a href="#L-1196"><span class="linenos">1196</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="L-1197"><a href="#L-1197"><span class="linenos">1197</span></a>
</span><span id="L-1198"><a href="#L-1198"><span class="linenos">1198</span></a>        <span class="k">if</span> <span class="p">(</span>
</span><span id="L-1199"><a href="#L-1199"><span class="linenos">1199</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">start</span> <span class="o">==</span> <span class="bp">self</span><span class="o">.</span><span class="n">end</span>
</span><span id="L-1200"><a href="#L-1200"><span class="linenos">1200</span></a>            <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">==</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref2_allele</span> <span class="o">==</span> <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span>
</span><span id="L-1201"><a href="#L-1201"><span class="linenos">1201</span></a>        <span class="p">):</span>
</span><span id="L-1202"><a href="#L-1202"><span class="linenos">1202</span></a>            <span class="c1"># Match.</span>
</span><span id="L-1203"><a href="#L-1203"><span class="linenos">1203</span></a>            <span class="c1"># Example: Glu1161=</span>
</span><span id="L-1204"><a href="#L-1204"><span class="linenos">1204</span></a>            <span class="n">pep_extra</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">pep_extra</span> <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">pep_extra</span> <span class="k">else</span> <span class="s2">&quot;=&quot;</span>
</span><span id="L-1205"><a href="#L-1205"><span class="linenos">1205</span></a>            <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">start</span><span class="p">)</span> <span class="o">+</span> <span class="n">pep_extra</span>
</span><span id="L-1206"><a href="#L-1206"><span class="linenos">1206</span></a>
</span><span id="L-1207"><a href="#L-1207"><span class="linenos">1207</span></a>        <span class="k">elif</span> <span class="p">(</span>
</span><span id="L-1208"><a href="#L-1208"><span class="linenos">1208</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">start</span> <span class="o">==</span> <span class="bp">self</span><span class="o">.</span><span class="n">end</span>
</span><span id="L-1209"><a href="#L-1209"><span class="linenos">1209</span></a>            <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">==</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref2_allele</span>
</span><span id="L-1210"><a href="#L-1210"><span class="linenos">1210</span></a>            <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">!=</span> <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span>
</span><span id="L-1211"><a href="#L-1211"><span class="linenos">1211</span></a>        <span class="p">):</span>
</span><span id="L-1212"><a href="#L-1212"><span class="linenos">1212</span></a>            <span class="c1"># Change.</span>
</span><span id="L-1213"><a href="#L-1213"><span class="linenos">1213</span></a>            <span class="c1"># Example: Glu1161Ser</span>
</span><span id="L-1214"><a href="#L-1214"><span class="linenos">1214</span></a>            <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">start</span><span class="p">)</span> <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span> <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">pep_extra</span>
</span><span id="L-1215"><a href="#L-1215"><span class="linenos">1215</span></a>
</span><span id="L-1216"><a href="#L-1216"><span class="linenos">1216</span></a>        <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">start</span> <span class="o">!=</span> <span class="bp">self</span><span class="o">.</span><span class="n">end</span><span class="p">:</span>
</span><span id="L-1217"><a href="#L-1217"><span class="linenos">1217</span></a>            <span class="c1"># Range change.</span>
</span><span id="L-1218"><a href="#L-1218"><span class="linenos">1218</span></a>            <span class="c1"># Example: Glu1161_Ser1164?fs</span>
</span><span id="L-1219"><a href="#L-1219"><span class="linenos">1219</span></a>            <span class="k">return</span> <span class="p">(</span>
</span><span id="L-1220"><a href="#L-1220"><span class="linenos">1220</span></a>                <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span>
</span><span id="L-1221"><a href="#L-1221"><span class="linenos">1221</span></a>                <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">start</span><span class="p">)</span>
</span><span id="L-1222"><a href="#L-1222"><span class="linenos">1222</span></a>                <span class="o">+</span> <span class="s2">&quot;_&quot;</span>
</span><span id="L-1223"><a href="#L-1223"><span class="linenos">1223</span></a>                <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref2_allele</span>
</span><span id="L-1224"><a href="#L-1224"><span class="linenos">1224</span></a>                <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">end</span><span class="p">)</span>
</span><span id="L-1225"><a href="#L-1225"><span class="linenos">1225</span></a>                <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">pep_extra</span>
</span><span id="L-1226"><a href="#L-1226"><span class="linenos">1226</span></a>            <span class="p">)</span>
</span><span id="L-1227"><a href="#L-1227"><span class="linenos">1227</span></a>        <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">pep_extra</span><span class="p">:</span>
</span><span id="L-1228"><a href="#L-1228"><span class="linenos">1228</span></a>            <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">pep_extra</span>
</span><span id="L-1229"><a href="#L-1229"><span class="linenos">1229</span></a>
</span><span id="L-1230"><a href="#L-1230"><span class="linenos">1230</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="L-1231"><a href="#L-1231"><span class="linenos">1231</span></a>            <span class="k">return</span> <span class="kc">None</span>
</span><span id="L-1232"><a href="#L-1232"><span class="linenos">1232</span></a>
</span><span id="L-1233"><a href="#L-1233"><span class="linenos">1233</span></a>    <span class="k">def</span> <span class="nf">format_coords</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
</span><span id="L-1234"><a href="#L-1234"><span class="linenos">1234</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-1235"><a href="#L-1235"><span class="linenos">1235</span></a><span class="sd">        The function `format_coords` generates a string representation of HGVS cDNA coordinates.</span>
</span><span id="L-1236"><a href="#L-1236"><span class="linenos">1236</span></a><span class="sd">        :return: a string that represents the HGVS cDNA coordinates. If the start and end coordinates</span>
</span><span id="L-1237"><a href="#L-1237"><span class="linenos">1237</span></a><span class="sd">        are the same, it returns just the start coordinate. Otherwise, it returns a string in the format</span>
</span><span id="L-1238"><a href="#L-1238"><span class="linenos">1238</span></a><span class="sd">        &quot;start_end&quot;.</span>
</span><span id="L-1239"><a href="#L-1239"><span class="linenos">1239</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="L-1240"><a href="#L-1240"><span class="linenos">1240</span></a>
</span><span id="L-1241"><a href="#L-1241"><span class="linenos">1241</span></a>        <span class="c1"># Format coordinates.</span>
</span><span id="L-1242"><a href="#L-1242"><span class="linenos">1242</span></a>        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">start</span> <span class="o">==</span> <span class="bp">self</span><span class="o">.</span><span class="n">end</span><span class="p">:</span>
</span><span id="L-1243"><a href="#L-1243"><span class="linenos">1243</span></a>            <span class="k">return</span> <span class="nb">str</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">start</span><span class="p">)</span>
</span><span id="L-1244"><a href="#L-1244"><span class="linenos">1244</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="L-1245"><a href="#L-1245"><span class="linenos">1245</span></a>            <span class="k">return</span> <span class="s2">&quot;</span><span class="si">%s</span><span class="s2">_</span><span class="si">%s</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">start</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">end</span><span class="p">)</span>
</span><span id="L-1246"><a href="#L-1246"><span class="linenos">1246</span></a>
</span><span id="L-1247"><a href="#L-1247"><span class="linenos">1247</span></a>    <span class="k">def</span> <span class="nf">format_genome</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
</span><span id="L-1248"><a href="#L-1248"><span class="linenos">1248</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-1249"><a href="#L-1249"><span class="linenos">1249</span></a><span class="sd">        The function &quot;format_genome&quot; generates an HGVS genomic allele by combining the formatted</span>
</span><span id="L-1250"><a href="#L-1250"><span class="linenos">1250</span></a><span class="sd">        coordinates and DNA allele.</span>
</span><span id="L-1251"><a href="#L-1251"><span class="linenos">1251</span></a>
</span><span id="L-1252"><a href="#L-1252"><span class="linenos">1252</span></a><span class="sd">        Some examples include:</span>
</span><span id="L-1253"><a href="#L-1253"><span class="linenos">1253</span></a><span class="sd">          Substitution: 1000100A&gt;T</span>
</span><span id="L-1254"><a href="#L-1254"><span class="linenos">1254</span></a><span class="sd">          Indel: 1000100_1000102delATG</span>
</span><span id="L-1255"><a href="#L-1255"><span class="linenos">1255</span></a>
</span><span id="L-1256"><a href="#L-1256"><span class="linenos">1256</span></a><span class="sd">        :return: a string that represents the HGVS genomic allele.</span>
</span><span id="L-1257"><a href="#L-1257"><span class="linenos">1257</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="L-1258"><a href="#L-1258"><span class="linenos">1258</span></a>
</span><span id="L-1259"><a href="#L-1259"><span class="linenos">1259</span></a>        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">format_coords</span><span class="p">()</span> <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">format_dna_allele</span><span class="p">()</span>
</span><span id="L-1260"><a href="#L-1260"><span class="linenos">1260</span></a>
</span><span id="L-1261"><a href="#L-1261"><span class="linenos">1261</span></a>    <span class="k">def</span> <span class="nf">get_raw_coords</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">transcript</span><span class="p">:</span> <span class="nb">object</span> <span class="o">=</span> <span class="kc">None</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">tuple</span><span class="p">:</span>
</span><span id="L-1262"><a href="#L-1262"><span class="linenos">1262</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-1263"><a href="#L-1263"><span class="linenos">1263</span></a><span class="sd">        The function `get_raw_coords` returns the genomic coordinates based on the given transcript or</span>
</span><span id="L-1264"><a href="#L-1264"><span class="linenos">1264</span></a><span class="sd">        the provided chromosomal coordinates.</span>
</span><span id="L-1265"><a href="#L-1265"><span class="linenos">1265</span></a>
</span><span id="L-1266"><a href="#L-1266"><span class="linenos">1266</span></a><span class="sd">        :param transcript: The `transcript` parameter is an object that represents a transcript. It is</span>
</span><span id="L-1267"><a href="#L-1267"><span class="linenos">1267</span></a><span class="sd">        used to retrieve genomic coordinates based on the type of HGVS name (`self.kind`). The</span>
</span><span id="L-1268"><a href="#L-1268"><span class="linenos">1268</span></a><span class="sd">        `transcript` object should have the following attributes and methods:</span>
</span><span id="L-1269"><a href="#L-1269"><span class="linenos">1269</span></a><span class="sd">        :type transcript: object</span>
</span><span id="L-1270"><a href="#L-1270"><span class="linenos">1270</span></a><span class="sd">        :return: a tuple containing the genomic coordinates. The tuple consists of three elements: the</span>
</span><span id="L-1271"><a href="#L-1271"><span class="linenos">1271</span></a><span class="sd">        chromosome, the start position, and the end position.</span>
</span><span id="L-1272"><a href="#L-1272"><span class="linenos">1272</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="L-1273"><a href="#L-1273"><span class="linenos">1273</span></a>
</span><span id="L-1274"><a href="#L-1274"><span class="linenos">1274</span></a>        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">kind</span> <span class="ow">in</span> <span class="p">(</span><span class="s2">&quot;c&quot;</span><span class="p">,</span> <span class="s2">&quot;n&quot;</span><span class="p">):</span>
</span><span id="L-1275"><a href="#L-1275"><span class="linenos">1275</span></a>            <span class="n">chrom</span> <span class="o">=</span> <span class="n">transcript</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom</span>
</span><span id="L-1276"><a href="#L-1276"><span class="linenos">1276</span></a>            <span class="n">start</span> <span class="o">=</span> <span class="n">transcript</span><span class="o">.</span><span class="n">cdna_to_genomic_coord</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">cdna_start</span><span class="p">)</span>
</span><span id="L-1277"><a href="#L-1277"><span class="linenos">1277</span></a>            <span class="n">end</span> <span class="o">=</span> <span class="n">transcript</span><span class="o">.</span><span class="n">cdna_to_genomic_coord</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">cdna_end</span><span class="p">)</span>
</span><span id="L-1278"><a href="#L-1278"><span class="linenos">1278</span></a>
</span><span id="L-1279"><a href="#L-1279"><span class="linenos">1279</span></a>            <span class="k">if</span> <span class="ow">not</span> <span class="n">transcript</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">is_forward_strand</span><span class="p">:</span>
</span><span id="L-1280"><a href="#L-1280"><span class="linenos">1280</span></a>                <span class="n">start</span><span class="p">,</span> <span class="n">end</span> <span class="o">=</span> <span class="n">end</span><span class="p">,</span> <span class="n">start</span>
</span><span id="L-1281"><a href="#L-1281"><span class="linenos">1281</span></a>
</span><span id="L-1282"><a href="#L-1282"><span class="linenos">1282</span></a>            <span class="k">if</span> <span class="n">start</span> <span class="o">&gt;</span> <span class="n">end</span><span class="p">:</span>
</span><span id="L-1283"><a href="#L-1283"><span class="linenos">1283</span></a>                <span class="k">raise</span> <span class="ne">AssertionError</span><span class="p">(</span><span class="s2">&quot;cdna_start cannot be greater than cdna_end&quot;</span><span class="p">)</span>
</span><span id="L-1284"><a href="#L-1284"><span class="linenos">1284</span></a>        <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">kind</span> <span class="ow">in</span> <span class="p">(</span><span class="s2">&quot;g&quot;</span><span class="p">,</span> <span class="s2">&quot;m&quot;</span><span class="p">):</span>
</span><span id="L-1285"><a href="#L-1285"><span class="linenos">1285</span></a>            <span class="n">chrom</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">chrom</span>
</span><span id="L-1286"><a href="#L-1286"><span class="linenos">1286</span></a>            <span class="n">start</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">start</span>
</span><span id="L-1287"><a href="#L-1287"><span class="linenos">1287</span></a>            <span class="n">end</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">end</span>
</span><span id="L-1288"><a href="#L-1288"><span class="linenos">1288</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="L-1289"><a href="#L-1289"><span class="linenos">1289</span></a>            <span class="k">raise</span> <span class="ne">NotImplementedError</span><span class="p">(</span>
</span><span id="L-1290"><a href="#L-1290"><span class="linenos">1290</span></a>                <span class="s1">&#39;Coordinates are not available for this kind of HGVS name &quot;</span><span class="si">%s</span><span class="s1">&quot;&#39;</span>
</span><span id="L-1291"><a href="#L-1291"><span class="linenos">1291</span></a>                <span class="o">%</span> <span class="bp">self</span><span class="o">.</span><span class="n">kind</span>
</span><span id="L-1292"><a href="#L-1292"><span class="linenos">1292</span></a>            <span class="p">)</span>
</span><span id="L-1293"><a href="#L-1293"><span class="linenos">1293</span></a>
</span><span id="L-1294"><a href="#L-1294"><span class="linenos">1294</span></a>        <span class="c1"># Check coordinate span is equal to reference bases</span>
</span><span id="L-1295"><a href="#L-1295"><span class="linenos">1295</span></a>        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span><span class="p">:</span>
</span><span id="L-1296"><a href="#L-1296"><span class="linenos">1296</span></a>            <span class="n">coordinate_span</span> <span class="o">=</span> <span class="n">end</span> <span class="o">-</span> <span class="n">start</span> <span class="o">+</span> <span class="mi">1</span>  <span class="c1"># Ref will always be &gt;=1 base</span>
</span><span id="L-1297"><a href="#L-1297"><span class="linenos">1297</span></a>            <span class="n">ref_length</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span><span class="p">)</span>
</span><span id="L-1298"><a href="#L-1298"><span class="linenos">1298</span></a>            <span class="k">if</span> <span class="n">coordinate_span</span> <span class="o">!=</span> <span class="n">ref_length</span><span class="p">:</span>
</span><span id="L-1299"><a href="#L-1299"><span class="linenos">1299</span></a>                <span class="k">raise</span> <span class="n">InvalidHGVSName</span><span class="p">(</span>
</span><span id="L-1300"><a href="#L-1300"><span class="linenos">1300</span></a>                    <span class="s2">&quot;Coordinate span (</span><span class="si">%d</span><span class="s2">) not equal to ref length </span><span class="si">%d</span><span class="s2">&quot;</span>
</span><span id="L-1301"><a href="#L-1301"><span class="linenos">1301</span></a>                    <span class="o">%</span> <span class="p">(</span><span class="n">coordinate_span</span><span class="p">,</span> <span class="n">ref_length</span><span class="p">)</span>
</span><span id="L-1302"><a href="#L-1302"><span class="linenos">1302</span></a>                <span class="p">)</span>
</span><span id="L-1303"><a href="#L-1303"><span class="linenos">1303</span></a>
</span><span id="L-1304"><a href="#L-1304"><span class="linenos">1304</span></a>        <span class="k">return</span> <span class="n">chrom</span><span class="p">,</span> <span class="n">start</span><span class="p">,</span> <span class="n">end</span>
</span><span id="L-1305"><a href="#L-1305"><span class="linenos">1305</span></a>
</span><span id="L-1306"><a href="#L-1306"><span class="linenos">1306</span></a>    <span class="k">def</span> <span class="nf">get_ref_coords</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">transcript</span><span class="p">:</span> <span class="nb">object</span> <span class="o">=</span> <span class="kc">None</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">tuple</span><span class="p">:</span>
</span><span id="L-1307"><a href="#L-1307"><span class="linenos">1307</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-1308"><a href="#L-1308"><span class="linenos">1308</span></a><span class="sd">        The function &quot;get_ref_coords&quot; returns the genomic coordinates of the reference allele, taking</span>
</span><span id="L-1309"><a href="#L-1309"><span class="linenos">1309</span></a><span class="sd">        into account different mutation types.</span>
</span><span id="L-1310"><a href="#L-1310"><span class="linenos">1310</span></a>
</span><span id="L-1311"><a href="#L-1311"><span class="linenos">1311</span></a><span class="sd">        :param transcript: The `transcript` parameter is an optional object that represents a transcript</span>
</span><span id="L-1312"><a href="#L-1312"><span class="linenos">1312</span></a><span class="sd">        or gene. It is used to retrieve the genomic coordinates of the reference allele</span>
</span><span id="L-1313"><a href="#L-1313"><span class="linenos">1313</span></a><span class="sd">        :type transcript: object</span>
</span><span id="L-1314"><a href="#L-1314"><span class="linenos">1314</span></a><span class="sd">        :return: a tuple containing the genomic coordinates of the reference allele. The tuple consists</span>
</span><span id="L-1315"><a href="#L-1315"><span class="linenos">1315</span></a><span class="sd">        of three elements: the chromosome, the start position, and the end position.</span>
</span><span id="L-1316"><a href="#L-1316"><span class="linenos">1316</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="L-1317"><a href="#L-1317"><span class="linenos">1317</span></a>
</span><span id="L-1318"><a href="#L-1318"><span class="linenos">1318</span></a>        <span class="n">chrom</span><span class="p">,</span> <span class="n">start</span><span class="p">,</span> <span class="n">end</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">get_raw_coords</span><span class="p">(</span><span class="n">transcript</span><span class="p">)</span>
</span><span id="L-1319"><a href="#L-1319"><span class="linenos">1319</span></a>
</span><span id="L-1320"><a href="#L-1320"><span class="linenos">1320</span></a>        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">==</span> <span class="s2">&quot;ins&quot;</span><span class="p">:</span>
</span><span id="L-1321"><a href="#L-1321"><span class="linenos">1321</span></a>            <span class="c1"># Inserts have empty interval.</span>
</span><span id="L-1322"><a href="#L-1322"><span class="linenos">1322</span></a>            <span class="k">if</span> <span class="n">start</span> <span class="o">&lt;</span> <span class="n">end</span><span class="p">:</span>
</span><span id="L-1323"><a href="#L-1323"><span class="linenos">1323</span></a>                <span class="n">start</span> <span class="o">+=</span> <span class="mi">1</span>
</span><span id="L-1324"><a href="#L-1324"><span class="linenos">1324</span></a>                <span class="n">end</span> <span class="o">-=</span> <span class="mi">1</span>
</span><span id="L-1325"><a href="#L-1325"><span class="linenos">1325</span></a>            <span class="k">else</span><span class="p">:</span>
</span><span id="L-1326"><a href="#L-1326"><span class="linenos">1326</span></a>                <span class="n">end</span> <span class="o">=</span> <span class="n">start</span> <span class="o">-</span> <span class="mi">1</span>
</span><span id="L-1327"><a href="#L-1327"><span class="linenos">1327</span></a>
</span><span id="L-1328"><a href="#L-1328"><span class="linenos">1328</span></a>        <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">==</span> <span class="s2">&quot;dup&quot;</span><span class="p">:</span>
</span><span id="L-1329"><a href="#L-1329"><span class="linenos">1329</span></a>            <span class="n">end</span> <span class="o">=</span> <span class="n">start</span> <span class="o">-</span> <span class="mi">1</span>
</span><span id="L-1330"><a href="#L-1330"><span class="linenos">1330</span></a>        <span class="k">return</span> <span class="n">chrom</span><span class="p">,</span> <span class="n">start</span><span class="p">,</span> <span class="n">end</span>
</span><span id="L-1331"><a href="#L-1331"><span class="linenos">1331</span></a>
</span><span id="L-1332"><a href="#L-1332"><span class="linenos">1332</span></a>    <span class="k">def</span> <span class="nf">get_vcf_coords</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">transcript</span><span class="p">:</span> <span class="nb">object</span> <span class="o">=</span> <span class="kc">None</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">tuple</span><span class="p">:</span>
</span><span id="L-1333"><a href="#L-1333"><span class="linenos">1333</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-1334"><a href="#L-1334"><span class="linenos">1334</span></a><span class="sd">        The function &quot;get_vcf_coords&quot; returns the genomic coordinates of the reference allele in</span>
</span><span id="L-1335"><a href="#L-1335"><span class="linenos">1335</span></a><span class="sd">        VCF-style, with left-padding for indels.</span>
</span><span id="L-1336"><a href="#L-1336"><span class="linenos">1336</span></a>
</span><span id="L-1337"><a href="#L-1337"><span class="linenos">1337</span></a><span class="sd">        :param transcript: The `transcript` parameter is an object that represents a transcript or gene.</span>
</span><span id="L-1338"><a href="#L-1338"><span class="linenos">1338</span></a><span class="sd">        It is used to retrieve the genomic coordinates of the reference allele</span>
</span><span id="L-1339"><a href="#L-1339"><span class="linenos">1339</span></a><span class="sd">        :type transcript: object</span>
</span><span id="L-1340"><a href="#L-1340"><span class="linenos">1340</span></a><span class="sd">        :return: a tuple containing the genomic coordinates of the reference allele in VCF-style. The</span>
</span><span id="L-1341"><a href="#L-1341"><span class="linenos">1341</span></a><span class="sd">        tuple consists of three elements: the chromosome, the start position, and the end position.</span>
</span><span id="L-1342"><a href="#L-1342"><span class="linenos">1342</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="L-1343"><a href="#L-1343"><span class="linenos">1343</span></a>
</span><span id="L-1344"><a href="#L-1344"><span class="linenos">1344</span></a>        <span class="n">chrom</span><span class="p">,</span> <span class="n">start</span><span class="p">,</span> <span class="n">end</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">get_ref_coords</span><span class="p">(</span><span class="n">transcript</span><span class="p">)</span>
</span><span id="L-1345"><a href="#L-1345"><span class="linenos">1345</span></a>
</span><span id="L-1346"><a href="#L-1346"><span class="linenos">1346</span></a>        <span class="c1"># Inserts and deletes require left-padding by 1 base</span>
</span><span id="L-1347"><a href="#L-1347"><span class="linenos">1347</span></a>        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="ow">in</span> <span class="p">(</span><span class="s2">&quot;=&quot;</span><span class="p">,</span> <span class="s2">&quot;&gt;&quot;</span><span class="p">):</span>
</span><span id="L-1348"><a href="#L-1348"><span class="linenos">1348</span></a>            <span class="k">pass</span>
</span><span id="L-1349"><a href="#L-1349"><span class="linenos">1349</span></a>        <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="ow">in</span> <span class="p">(</span><span class="s2">&quot;del&quot;</span><span class="p">,</span> <span class="s2">&quot;ins&quot;</span><span class="p">,</span> <span class="s2">&quot;dup&quot;</span><span class="p">,</span> <span class="s2">&quot;delins&quot;</span><span class="p">):</span>
</span><span id="L-1350"><a href="#L-1350"><span class="linenos">1350</span></a>            <span class="c1"># Indels have left-padding.</span>
</span><span id="L-1351"><a href="#L-1351"><span class="linenos">1351</span></a>            <span class="n">start</span> <span class="o">-=</span> <span class="mi">1</span>
</span><span id="L-1352"><a href="#L-1352"><span class="linenos">1352</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="L-1353"><a href="#L-1353"><span class="linenos">1353</span></a>            <span class="k">raise</span> <span class="ne">NotImplementedError</span><span class="p">(</span><span class="s2">&quot;Unknown mutation_type &#39;</span><span class="si">%s</span><span class="s2">&#39;&quot;</span> <span class="o">%</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span><span class="p">)</span>
</span><span id="L-1354"><a href="#L-1354"><span class="linenos">1354</span></a>        <span class="k">return</span> <span class="n">chrom</span><span class="p">,</span> <span class="n">start</span><span class="p">,</span> <span class="n">end</span>
</span><span id="L-1355"><a href="#L-1355"><span class="linenos">1355</span></a>
</span><span id="L-1356"><a href="#L-1356"><span class="linenos">1356</span></a>    <span class="k">def</span> <span class="nf">get_ref_alt</span><span class="p">(</span>
</span><span id="L-1357"><a href="#L-1357"><span class="linenos">1357</span></a>        <span class="bp">self</span><span class="p">,</span> <span class="n">is_forward_strand</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span> <span class="n">raw_dup_alleles</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span>
</span><span id="L-1358"><a href="#L-1358"><span class="linenos">1358</span></a>    <span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">tuple</span><span class="p">:</span>
</span><span id="L-1359"><a href="#L-1359"><span class="linenos">1359</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-1360"><a href="#L-1360"><span class="linenos">1360</span></a><span class="sd">        The function `get_ref_alt` returns the reference and alternate alleles, with an option to modify</span>
</span><span id="L-1361"><a href="#L-1361"><span class="linenos">1361</span></a><span class="sd">        duplications to look like inserts.</span>
</span><span id="L-1362"><a href="#L-1362"><span class="linenos">1362</span></a>
</span><span id="L-1363"><a href="#L-1363"><span class="linenos">1363</span></a><span class="sd">        :param is_forward_strand: The parameter `is_forward_strand` is a boolean flag that indicates</span>
</span><span id="L-1364"><a href="#L-1364"><span class="linenos">1364</span></a><span class="sd">        whether the alleles should be returned for the forward strand or the reverse complement strand.</span>
</span><span id="L-1365"><a href="#L-1365"><span class="linenos">1365</span></a><span class="sd">        If `is_forward_strand` is `True`, the alleles will be returned as is. If `is_forward_strand` is</span>
</span><span id="L-1366"><a href="#L-1366"><span class="linenos">1366</span></a><span class="sd">        `False`,, defaults to True</span>
</span><span id="L-1367"><a href="#L-1367"><span class="linenos">1367</span></a><span class="sd">        :type is_forward_strand: bool (optional)</span>
</span><span id="L-1368"><a href="#L-1368"><span class="linenos">1368</span></a><span class="sd">        :param raw_dup_alleles: The `raw_dup_alleles` parameter is a boolean flag that determines</span>
</span><span id="L-1369"><a href="#L-1369"><span class="linenos">1369</span></a><span class="sd">        whether the raw values of duplicated alleles should be returned. By default, it is set to</span>
</span><span id="L-1370"><a href="#L-1370"><span class="linenos">1370</span></a><span class="sd">        `False`, which means that if the mutation type is a duplication (`dup`), the reference allele</span>
</span><span id="L-1371"><a href="#L-1371"><span class="linenos">1371</span></a><span class="sd">        will be represented as an empty string, defaults to False</span>
</span><span id="L-1372"><a href="#L-1372"><span class="linenos">1372</span></a><span class="sd">        :type raw_dup_alleles: bool (optional)</span>
</span><span id="L-1373"><a href="#L-1373"><span class="linenos">1373</span></a><span class="sd">        :return: The function `get_ref_alt` returns a tuple containing the reference and alternate</span>
</span><span id="L-1374"><a href="#L-1374"><span class="linenos">1374</span></a><span class="sd">        alleles.</span>
</span><span id="L-1375"><a href="#L-1375"><span class="linenos">1375</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="L-1376"><a href="#L-1376"><span class="linenos">1376</span></a>
</span><span id="L-1377"><a href="#L-1377"><span class="linenos">1377</span></a>        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">kind</span> <span class="o">==</span> <span class="s2">&quot;p&quot;</span><span class="p">:</span>
</span><span id="L-1378"><a href="#L-1378"><span class="linenos">1378</span></a>            <span class="k">raise</span> <span class="ne">NotImplementedError</span><span class="p">(</span>
</span><span id="L-1379"><a href="#L-1379"><span class="linenos">1379</span></a>                <span class="s2">&quot;get_ref_alt is not implemented for protein HGVS names&quot;</span>
</span><span id="L-1380"><a href="#L-1380"><span class="linenos">1380</span></a>            <span class="p">)</span>
</span><span id="L-1381"><a href="#L-1381"><span class="linenos">1381</span></a>        <span class="n">alleles</span> <span class="o">=</span> <span class="p">[</span><span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span><span class="p">]</span>
</span><span id="L-1382"><a href="#L-1382"><span class="linenos">1382</span></a>
</span><span id="L-1383"><a href="#L-1383"><span class="linenos">1383</span></a>        <span class="c1"># Represent duplications are inserts.</span>
</span><span id="L-1384"><a href="#L-1384"><span class="linenos">1384</span></a>        <span class="k">if</span> <span class="ow">not</span> <span class="n">raw_dup_alleles</span> <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">==</span> <span class="s2">&quot;dup&quot;</span><span class="p">:</span>
</span><span id="L-1385"><a href="#L-1385"><span class="linenos">1385</span></a>            <span class="n">alleles</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span>
</span><span id="L-1386"><a href="#L-1386"><span class="linenos">1386</span></a>            <span class="n">alleles</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="o">=</span> <span class="n">alleles</span><span class="p">[</span><span class="mi">1</span><span class="p">][:</span> <span class="nb">len</span><span class="p">(</span><span class="n">alleles</span><span class="p">[</span><span class="mi">1</span><span class="p">])</span> <span class="o">//</span> <span class="mi">2</span><span class="p">]</span>
</span><span id="L-1387"><a href="#L-1387"><span class="linenos">1387</span></a>
</span><span id="L-1388"><a href="#L-1388"><span class="linenos">1388</span></a>        <span class="k">if</span> <span class="n">is_forward_strand</span><span class="p">:</span>
</span><span id="L-1389"><a href="#L-1389"><span class="linenos">1389</span></a>            <span class="k">return</span> <span class="n">alleles</span>
</span><span id="L-1390"><a href="#L-1390"><span class="linenos">1390</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="L-1391"><a href="#L-1391"><span class="linenos">1391</span></a>            <span class="k">return</span> <span class="nb">tuple</span><span class="p">(</span><span class="nb">map</span><span class="p">(</span><span class="n">revcomp</span><span class="p">,</span> <span class="n">alleles</span><span class="p">))</span>
</span></pre></div>


            </section>
                <section id="CHROM_PREFIX">
                    <div class="attr variable">
            <span class="name">CHROM_PREFIX</span>        =
<span class="default_value">&#39;chr&#39;</span>

        
    </div>
    <a class="headerlink" href="#CHROM_PREFIX"></a>
    
    

                </section>
                <section id="CODON_1">
                    <div class="attr variable">
            <span class="name">CODON_1</span>        =
<input id="CODON_1-view-value" class="view-value-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
            <label class="view-value-button pdoc-button" for="CODON_1-view-value"></label><span class="default_value">{&#39;TTT&#39;: &#39;F&#39;, &#39;TTC&#39;: &#39;F&#39;, &#39;TCT&#39;: &#39;S&#39;, &#39;TCC&#39;: &#39;S&#39;, &#39;TAT&#39;: &#39;Y&#39;, &#39;TAC&#39;: &#39;Y&#39;, &#39;TGT&#39;: &#39;C&#39;, &#39;TGC&#39;: &#39;C&#39;, &#39;TTA&#39;: &#39;L&#39;, &#39;TCA&#39;: &#39;S&#39;, &#39;TAA&#39;: &#39;*&#39;, &#39;TGA&#39;: &#39;*&#39;, &#39;TTG&#39;: &#39;L&#39;, &#39;TCG&#39;: &#39;S&#39;, &#39;TAG&#39;: &#39;*&#39;, &#39;TGG&#39;: &#39;W&#39;, &#39;CTT&#39;: &#39;L&#39;, &#39;CTC&#39;: &#39;L&#39;, &#39;CCT&#39;: &#39;P&#39;, &#39;CCC&#39;: &#39;P&#39;, &#39;CAT&#39;: &#39;H&#39;, &#39;CAC&#39;: &#39;H&#39;, &#39;CGT&#39;: &#39;R&#39;, &#39;CGC&#39;: &#39;R&#39;, &#39;CTA&#39;: &#39;L&#39;, &#39;CTG&#39;: &#39;L&#39;, &#39;CCA&#39;: &#39;P&#39;, &#39;CCG&#39;: &#39;P&#39;, &#39;CAA&#39;: &#39;Q&#39;, &#39;CAG&#39;: &#39;Q&#39;, &#39;CGA&#39;: &#39;R&#39;, &#39;CGG&#39;: &#39;R&#39;, &#39;ATT&#39;: &#39;I&#39;, &#39;ATC&#39;: &#39;I&#39;, &#39;ACT&#39;: &#39;T&#39;, &#39;ACC&#39;: &#39;T&#39;, &#39;AAT&#39;: &#39;N&#39;, &#39;AAC&#39;: &#39;N&#39;, &#39;AGT&#39;: &#39;S&#39;, &#39;AGC&#39;: &#39;S&#39;, &#39;ATA&#39;: &#39;I&#39;, &#39;ACA&#39;: &#39;T&#39;, &#39;AAA&#39;: &#39;K&#39;, &#39;AGA&#39;: &#39;R&#39;, &#39;ATG&#39;: &#39;M&#39;, &#39;ACG&#39;: &#39;T&#39;, &#39;AAG&#39;: &#39;K&#39;, &#39;AGG&#39;: &#39;R&#39;, &#39;GTT&#39;: &#39;V&#39;, &#39;GTC&#39;: &#39;V&#39;, &#39;GCT&#39;: &#39;A&#39;, &#39;GCC&#39;: &#39;A&#39;, &#39;GAT&#39;: &#39;D&#39;, &#39;GAC&#39;: &#39;D&#39;, &#39;GGT&#39;: &#39;G&#39;, &#39;GGC&#39;: &#39;G&#39;, &#39;GTA&#39;: &#39;V&#39;, &#39;GTG&#39;: &#39;V&#39;, &#39;GCA&#39;: &#39;A&#39;, &#39;GCG&#39;: &#39;A&#39;, &#39;GAA&#39;: &#39;E&#39;, &#39;GAG&#39;: &#39;E&#39;, &#39;GGA&#39;: &#39;G&#39;, &#39;GGG&#39;: &#39;G&#39;}</span>

        
    </div>
    <a class="headerlink" href="#CODON_1"></a>
    
    

                </section>
                <section id="CODON_3">
                    <div class="attr variable">
            <span class="name">CODON_3</span>        =
<input id="CODON_3-view-value" class="view-value-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
            <label class="view-value-button pdoc-button" for="CODON_3-view-value"></label><span class="default_value">{&#39;TTT&#39;: &#39;Phe&#39;, &#39;TTC&#39;: &#39;Phe&#39;, &#39;TCT&#39;: &#39;Ser&#39;, &#39;TCC&#39;: &#39;Ser&#39;, &#39;TAT&#39;: &#39;Tyr&#39;, &#39;TAC&#39;: &#39;Tyr&#39;, &#39;TGT&#39;: &#39;Cys&#39;, &#39;TGC&#39;: &#39;Cys&#39;, &#39;TTA&#39;: &#39;Leu&#39;, &#39;TCA&#39;: &#39;Ser&#39;, &#39;TAA&#39;: &#39;*&#39;, &#39;TGA&#39;: &#39;*&#39;, &#39;TTG&#39;: &#39;Leu&#39;, &#39;TCG&#39;: &#39;Ser&#39;, &#39;TAG&#39;: &#39;*&#39;, &#39;TGG&#39;: &#39;Trp&#39;, &#39;CTT&#39;: &#39;Leu&#39;, &#39;CTC&#39;: &#39;Leu&#39;, &#39;CCT&#39;: &#39;Pro&#39;, &#39;CCC&#39;: &#39;Pro&#39;, &#39;CAT&#39;: &#39;His&#39;, &#39;CAC&#39;: &#39;His&#39;, &#39;CGT&#39;: &#39;Arg&#39;, &#39;CGC&#39;: &#39;Arg&#39;, &#39;CTA&#39;: &#39;Leu&#39;, &#39;CTG&#39;: &#39;Leu&#39;, &#39;CCA&#39;: &#39;Pro&#39;, &#39;CCG&#39;: &#39;Pro&#39;, &#39;CAA&#39;: &#39;Gln&#39;, &#39;CAG&#39;: &#39;Gln&#39;, &#39;CGA&#39;: &#39;Arg&#39;, &#39;CGG&#39;: &#39;Arg&#39;, &#39;ATT&#39;: &#39;Ile&#39;, &#39;ATC&#39;: &#39;Ile&#39;, &#39;ACT&#39;: &#39;Thr&#39;, &#39;ACC&#39;: &#39;Thr&#39;, &#39;AAT&#39;: &#39;Asn&#39;, &#39;AAC&#39;: &#39;Asn&#39;, &#39;AGT&#39;: &#39;Ser&#39;, &#39;AGC&#39;: &#39;Ser&#39;, &#39;ATA&#39;: &#39;Ile&#39;, &#39;ACA&#39;: &#39;Thr&#39;, &#39;AAA&#39;: &#39;Lys&#39;, &#39;AGA&#39;: &#39;Arg&#39;, &#39;ATG&#39;: &#39;Met&#39;, &#39;ACG&#39;: &#39;Thr&#39;, &#39;AAG&#39;: &#39;Lys&#39;, &#39;AGG&#39;: &#39;Arg&#39;, &#39;GTT&#39;: &#39;Val&#39;, &#39;GTC&#39;: &#39;Val&#39;, &#39;GCT&#39;: &#39;Ala&#39;, &#39;GCC&#39;: &#39;Ala&#39;, &#39;GAT&#39;: &#39;Asp&#39;, &#39;GAC&#39;: &#39;Asp&#39;, &#39;GGT&#39;: &#39;Gly&#39;, &#39;GGC&#39;: &#39;Gly&#39;, &#39;GTA&#39;: &#39;Val&#39;, &#39;GTG&#39;: &#39;Val&#39;, &#39;GCA&#39;: &#39;Ala&#39;, &#39;GCG&#39;: &#39;Ala&#39;, &#39;GAA&#39;: &#39;Glu&#39;, &#39;GAG&#39;: &#39;Glu&#39;, &#39;GGA&#39;: &#39;Gly&#39;, &#39;GGG&#39;: &#39;Gly&#39;}</span>

        
    </div>
    <a class="headerlink" href="#CODON_3"></a>
    
    

                </section>
                <section id="CODON_FULL">
                    <div class="attr variable">
            <span class="name">CODON_FULL</span>        =
<input id="CODON_FULL-view-value" class="view-value-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
            <label class="view-value-button pdoc-button" for="CODON_FULL-view-value"></label><span class="default_value">{&#39;TTT&#39;: &#39;Phenylalanine&#39;, &#39;TTC&#39;: &#39;Phenylalanine&#39;, &#39;TCT&#39;: &#39;Serine&#39;, &#39;TCC&#39;: &#39;Serine&#39;, &#39;TAT&#39;: &#39;Tyrosine&#39;, &#39;TAC&#39;: &#39;Tyrosine&#39;, &#39;TGT&#39;: &#39;Cysteine&#39;, &#39;TGC&#39;: &#39;Cysteine&#39;, &#39;TTA&#39;: &#39;Leucine&#39;, &#39;TCA&#39;: &#39;Serine&#39;, &#39;TAA&#39;: &#39;Stop&#39;, &#39;TGA&#39;: &#39;Stop&#39;, &#39;TTG&#39;: &#39;Leucine&#39;, &#39;TCG&#39;: &#39;Serine&#39;, &#39;TAG&#39;: &#39;Stop&#39;, &#39;TGG&#39;: &#39;Tryptophan&#39;, &#39;CTT&#39;: &#39;Leucine&#39;, &#39;CTC&#39;: &#39;Leucine&#39;, &#39;CCT&#39;: &#39;Proline&#39;, &#39;CCC&#39;: &#39;Proline&#39;, &#39;CAT&#39;: &#39;Histidine&#39;, &#39;CAC&#39;: &#39;Histidine&#39;, &#39;CGT&#39;: &#39;Arginine&#39;, &#39;CGC&#39;: &#39;Arginine&#39;, &#39;CTA&#39;: &#39;Leucine&#39;, &#39;CTG&#39;: &#39;Leucine&#39;, &#39;CCA&#39;: &#39;Proline&#39;, &#39;CCG&#39;: &#39;Proline&#39;, &#39;CAA&#39;: &#39;Glutamine&#39;, &#39;CAG&#39;: &#39;Glutamine&#39;, &#39;CGA&#39;: &#39;Arginine&#39;, &#39;CGG&#39;: &#39;Arginine&#39;, &#39;ATT&#39;: &#39;Isoleucine&#39;, &#39;ATC&#39;: &#39;Isoleucine&#39;, &#39;ACT&#39;: &#39;Threonine&#39;, &#39;ACC&#39;: &#39;Threonine&#39;, &#39;AAT&#39;: &#39;Asparagine&#39;, &#39;AAC&#39;: &#39;Asparagine&#39;, &#39;AGT&#39;: &#39;Serine&#39;, &#39;AGC&#39;: &#39;Serine&#39;, &#39;ATA&#39;: &#39;Isoleucine&#39;, &#39;ACA&#39;: &#39;Threonine&#39;, &#39;AAA&#39;: &#39;Lysine&#39;, &#39;AGA&#39;: &#39;Arginine&#39;, &#39;ATG&#39;: &#39;Methionine&#39;, &#39;ACG&#39;: &#39;Threonine&#39;, &#39;AAG&#39;: &#39;Lysine&#39;, &#39;AGG&#39;: &#39;Arginine&#39;, &#39;GTT&#39;: &#39;Valine&#39;, &#39;GTC&#39;: &#39;Valine&#39;, &#39;GCT&#39;: &#39;Alanine&#39;, &#39;GCC&#39;: &#39;Alanine&#39;, &#39;GAT&#39;: &#39;Aspartic acid&#39;, &#39;GAC&#39;: &#39;Aspartic acid&#39;, &#39;GGT&#39;: &#39;Glycine&#39;, &#39;GGC&#39;: &#39;Glycine&#39;, &#39;GTA&#39;: &#39;Valine&#39;, &#39;GTG&#39;: &#39;Valine&#39;, &#39;GCA&#39;: &#39;Alanine&#39;, &#39;GCG&#39;: &#39;Alanine&#39;, &#39;GAA&#39;: &#39;Glutamic acid&#39;, &#39;GAG&#39;: &#39;Glutamic acid&#39;, &#39;GGA&#39;: &#39;Glycine&#39;, &#39;GGG&#39;: &#39;Glycine&#39;}</span>

        
    </div>
    <a class="headerlink" href="#CODON_FULL"></a>
    
    

                </section>
                <section id="NUCLEOTIDE_TRANSLATE">
                    <div class="attr variable">
            <span class="name">NUCLEOTIDE_TRANSLATE</span>        =
<span class="default_value">{&#39;T&#39;: &#39;A&#39;, &#39;A&#39;: &#39;T&#39;, &#39;G&#39;: &#39;C&#39;, &#39;C&#39;: &#39;G&#39;}</span>

        
    </div>
    <a class="headerlink" href="#NUCLEOTIDE_TRANSLATE"></a>
    
    

                </section>
                <section id="HGVSRegex">
                            <input id="HGVSRegex-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr class">
            
    <span class="def">class</span>
    <span class="name">HGVSRegex</span>:

                <label class="view-source-button" for="HGVSRegex-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#HGVSRegex"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="HGVSRegex-305"><a href="#HGVSRegex-305"><span class="linenos">305</span></a><span class="k">class</span> <span class="nc">HGVSRegex</span><span class="p">(</span><span class="nb">object</span><span class="p">):</span>
</span><span id="HGVSRegex-306"><a href="#HGVSRegex-306"><span class="linenos">306</span></a><span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="HGVSRegex-307"><a href="#HGVSRegex-307"><span class="linenos">307</span></a><span class="sd">    All regular expression for HGVS names.</span>
</span><span id="HGVSRegex-308"><a href="#HGVSRegex-308"><span class="linenos">308</span></a><span class="sd">    &quot;&quot;&quot;</span>
</span><span id="HGVSRegex-309"><a href="#HGVSRegex-309"><span class="linenos">309</span></a>
</span><span id="HGVSRegex-310"><a href="#HGVSRegex-310"><span class="linenos">310</span></a>    <span class="c1"># DNA syntax</span>
</span><span id="HGVSRegex-311"><a href="#HGVSRegex-311"><span class="linenos">311</span></a>    <span class="c1"># http://www.hgvs.org/mutnomen/standards.html#nucleotide</span>
</span><span id="HGVSRegex-312"><a href="#HGVSRegex-312"><span class="linenos">312</span></a>    <span class="n">BASE</span> <span class="o">=</span> <span class="sa">r</span><span class="s2">&quot;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]|\d+&quot;</span>
</span><span id="HGVSRegex-313"><a href="#HGVSRegex-313"><span class="linenos">313</span></a>    <span class="n">BASES</span> <span class="o">=</span> <span class="sa">r</span><span class="s2">&quot;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\d+&quot;</span>
</span><span id="HGVSRegex-314"><a href="#HGVSRegex-314"><span class="linenos">314</span></a>    <span class="n">DNA_REF</span> <span class="o">=</span> <span class="s2">&quot;(?P&lt;ref&gt;&quot;</span> <span class="o">+</span> <span class="n">BASES</span> <span class="o">+</span> <span class="s2">&quot;)&quot;</span>
</span><span id="HGVSRegex-315"><a href="#HGVSRegex-315"><span class="linenos">315</span></a>    <span class="n">DNA_ALT</span> <span class="o">=</span> <span class="s2">&quot;(?P&lt;alt&gt;&quot;</span> <span class="o">+</span> <span class="n">BASES</span> <span class="o">+</span> <span class="s2">&quot;)&quot;</span>
</span><span id="HGVSRegex-316"><a href="#HGVSRegex-316"><span class="linenos">316</span></a>
</span><span id="HGVSRegex-317"><a href="#HGVSRegex-317"><span class="linenos">317</span></a>    <span class="c1"># Mutation types</span>
</span><span id="HGVSRegex-318"><a href="#HGVSRegex-318"><span class="linenos">318</span></a>    <span class="n">EQUAL</span> <span class="o">=</span> <span class="s2">&quot;(?P&lt;mutation_type&gt;=)&quot;</span>
</span><span id="HGVSRegex-319"><a href="#HGVSRegex-319"><span class="linenos">319</span></a>    <span class="n">SUB</span> <span class="o">=</span> <span class="s2">&quot;(?P&lt;mutation_type&gt;&gt;)&quot;</span>
</span><span id="HGVSRegex-320"><a href="#HGVSRegex-320"><span class="linenos">320</span></a>    <span class="n">INS</span> <span class="o">=</span> <span class="s2">&quot;(?P&lt;mutation_type&gt;ins)&quot;</span>
</span><span id="HGVSRegex-321"><a href="#HGVSRegex-321"><span class="linenos">321</span></a>    <span class="n">DEL</span> <span class="o">=</span> <span class="s2">&quot;(?P&lt;mutation_type&gt;del)&quot;</span>
</span><span id="HGVSRegex-322"><a href="#HGVSRegex-322"><span class="linenos">322</span></a>    <span class="n">DUP</span> <span class="o">=</span> <span class="s2">&quot;(?P&lt;mutation_type&gt;dup)&quot;</span>
</span><span id="HGVSRegex-323"><a href="#HGVSRegex-323"><span class="linenos">323</span></a>
</span><span id="HGVSRegex-324"><a href="#HGVSRegex-324"><span class="linenos">324</span></a>    <span class="c1"># Simple coordinate syntax</span>
</span><span id="HGVSRegex-325"><a href="#HGVSRegex-325"><span class="linenos">325</span></a>    <span class="n">COORD_START</span> <span class="o">=</span> <span class="sa">r</span><span class="s2">&quot;(?P&lt;start&gt;\d+)&quot;</span>
</span><span id="HGVSRegex-326"><a href="#HGVSRegex-326"><span class="linenos">326</span></a>    <span class="n">COORD_END</span> <span class="o">=</span> <span class="sa">r</span><span class="s2">&quot;(?P&lt;end&gt;\d+)&quot;</span>
</span><span id="HGVSRegex-327"><a href="#HGVSRegex-327"><span class="linenos">327</span></a>    <span class="n">COORD_RANGE</span> <span class="o">=</span> <span class="n">COORD_START</span> <span class="o">+</span> <span class="s2">&quot;_&quot;</span> <span class="o">+</span> <span class="n">COORD_END</span>
</span><span id="HGVSRegex-328"><a href="#HGVSRegex-328"><span class="linenos">328</span></a>
</span><span id="HGVSRegex-329"><a href="#HGVSRegex-329"><span class="linenos">329</span></a>    <span class="c1"># cDNA coordinate syntax</span>
</span><span id="HGVSRegex-330"><a href="#HGVSRegex-330"><span class="linenos">330</span></a>    <span class="n">CDNA_COORD</span> <span class="o">=</span> <span class="p">(</span>
</span><span id="HGVSRegex-331"><a href="#HGVSRegex-331"><span class="linenos">331</span></a>        <span class="sa">r</span><span class="s2">&quot;(?P&lt;coord_prefix&gt;|-|\*)(?P&lt;coord&gt;\d+)&quot;</span>
</span><span id="HGVSRegex-332"><a href="#HGVSRegex-332"><span class="linenos">332</span></a>        <span class="sa">r</span><span class="s2">&quot;((?P&lt;offset_prefix&gt;-|\+)(?P&lt;offset&gt;\d+))?&quot;</span>
</span><span id="HGVSRegex-333"><a href="#HGVSRegex-333"><span class="linenos">333</span></a>    <span class="p">)</span>
</span><span id="HGVSRegex-334"><a href="#HGVSRegex-334"><span class="linenos">334</span></a>    <span class="n">CDNA_START</span> <span class="o">=</span> <span class="p">(</span>
</span><span id="HGVSRegex-335"><a href="#HGVSRegex-335"><span class="linenos">335</span></a>        <span class="sa">r</span><span class="s2">&quot;(?P&lt;start&gt;(?P&lt;start_coord_prefix&gt;|-|\*)(?P&lt;start_coord&gt;\d+)&quot;</span>
</span><span id="HGVSRegex-336"><a href="#HGVSRegex-336"><span class="linenos">336</span></a>        <span class="sa">r</span><span class="s2">&quot;((?P&lt;start_offset_prefix&gt;-|\+)(?P&lt;start_offset&gt;\d+))?)&quot;</span>
</span><span id="HGVSRegex-337"><a href="#HGVSRegex-337"><span class="linenos">337</span></a>    <span class="p">)</span>
</span><span id="HGVSRegex-338"><a href="#HGVSRegex-338"><span class="linenos">338</span></a>    <span class="n">CDNA_END</span> <span class="o">=</span> <span class="p">(</span>
</span><span id="HGVSRegex-339"><a href="#HGVSRegex-339"><span class="linenos">339</span></a>        <span class="sa">r</span><span class="s2">&quot;(?P&lt;end&gt;(?P&lt;end_coord_prefix&gt;|-|\*)(?P&lt;end_coord&gt;\d+)&quot;</span>
</span><span id="HGVSRegex-340"><a href="#HGVSRegex-340"><span class="linenos">340</span></a>        <span class="sa">r</span><span class="s2">&quot;((?P&lt;end_offset_prefix&gt;-|\+)(?P&lt;end_offset&gt;\d+))?)&quot;</span>
</span><span id="HGVSRegex-341"><a href="#HGVSRegex-341"><span class="linenos">341</span></a>    <span class="p">)</span>
</span><span id="HGVSRegex-342"><a href="#HGVSRegex-342"><span class="linenos">342</span></a>    <span class="n">CDNA_RANGE</span> <span class="o">=</span> <span class="n">CDNA_START</span> <span class="o">+</span> <span class="s2">&quot;_&quot;</span> <span class="o">+</span> <span class="n">CDNA_END</span>
</span><span id="HGVSRegex-343"><a href="#HGVSRegex-343"><span class="linenos">343</span></a>
</span><span id="HGVSRegex-344"><a href="#HGVSRegex-344"><span class="linenos">344</span></a>    <span class="c1"># cDNA allele syntax</span>
</span><span id="HGVSRegex-345"><a href="#HGVSRegex-345"><span class="linenos">345</span></a>    <span class="n">CDNA_ALLELE</span> <span class="o">=</span> <span class="p">[</span>
</span><span id="HGVSRegex-346"><a href="#HGVSRegex-346"><span class="linenos">346</span></a>        <span class="c1"># No change</span>
</span><span id="HGVSRegex-347"><a href="#HGVSRegex-347"><span class="linenos">347</span></a>        <span class="n">CDNA_START</span> <span class="o">+</span> <span class="n">EQUAL</span><span class="p">,</span>
</span><span id="HGVSRegex-348"><a href="#HGVSRegex-348"><span class="linenos">348</span></a>        <span class="n">CDNA_START</span> <span class="o">+</span> <span class="n">DNA_REF</span> <span class="o">+</span> <span class="n">EQUAL</span><span class="p">,</span>
</span><span id="HGVSRegex-349"><a href="#HGVSRegex-349"><span class="linenos">349</span></a>        <span class="c1"># Substitution</span>
</span><span id="HGVSRegex-350"><a href="#HGVSRegex-350"><span class="linenos">350</span></a>        <span class="n">CDNA_START</span> <span class="o">+</span> <span class="n">DNA_REF</span> <span class="o">+</span> <span class="n">SUB</span> <span class="o">+</span> <span class="n">DNA_ALT</span><span class="p">,</span>
</span><span id="HGVSRegex-351"><a href="#HGVSRegex-351"><span class="linenos">351</span></a>        <span class="c1"># 1bp insertion, deletion, duplication</span>
</span><span id="HGVSRegex-352"><a href="#HGVSRegex-352"><span class="linenos">352</span></a>        <span class="n">CDNA_START</span> <span class="o">+</span> <span class="n">INS</span> <span class="o">+</span> <span class="n">DNA_ALT</span><span class="p">,</span>
</span><span id="HGVSRegex-353"><a href="#HGVSRegex-353"><span class="linenos">353</span></a>        <span class="n">CDNA_START</span> <span class="o">+</span> <span class="n">DEL</span> <span class="o">+</span> <span class="n">DNA_REF</span><span class="p">,</span>
</span><span id="HGVSRegex-354"><a href="#HGVSRegex-354"><span class="linenos">354</span></a>        <span class="n">CDNA_START</span> <span class="o">+</span> <span class="n">DUP</span> <span class="o">+</span> <span class="n">DNA_REF</span><span class="p">,</span>
</span><span id="HGVSRegex-355"><a href="#HGVSRegex-355"><span class="linenos">355</span></a>        <span class="n">CDNA_START</span> <span class="o">+</span> <span class="n">DEL</span><span class="p">,</span>
</span><span id="HGVSRegex-356"><a href="#HGVSRegex-356"><span class="linenos">356</span></a>        <span class="n">CDNA_START</span> <span class="o">+</span> <span class="n">DUP</span><span class="p">,</span>
</span><span id="HGVSRegex-357"><a href="#HGVSRegex-357"><span class="linenos">357</span></a>        <span class="c1"># Insertion, deletion, duplication</span>
</span><span id="HGVSRegex-358"><a href="#HGVSRegex-358"><span class="linenos">358</span></a>        <span class="n">COORD_RANGE</span> <span class="o">+</span> <span class="n">EQUAL</span><span class="p">,</span>
</span><span id="HGVSRegex-359"><a href="#HGVSRegex-359"><span class="linenos">359</span></a>        <span class="n">CDNA_RANGE</span> <span class="o">+</span> <span class="n">INS</span> <span class="o">+</span> <span class="n">DNA_ALT</span><span class="p">,</span>
</span><span id="HGVSRegex-360"><a href="#HGVSRegex-360"><span class="linenos">360</span></a>        <span class="n">CDNA_RANGE</span> <span class="o">+</span> <span class="n">DEL</span> <span class="o">+</span> <span class="n">DNA_REF</span><span class="p">,</span>
</span><span id="HGVSRegex-361"><a href="#HGVSRegex-361"><span class="linenos">361</span></a>        <span class="n">CDNA_RANGE</span> <span class="o">+</span> <span class="n">DUP</span> <span class="o">+</span> <span class="n">DNA_REF</span><span class="p">,</span>
</span><span id="HGVSRegex-362"><a href="#HGVSRegex-362"><span class="linenos">362</span></a>        <span class="n">CDNA_RANGE</span> <span class="o">+</span> <span class="n">DEL</span><span class="p">,</span>
</span><span id="HGVSRegex-363"><a href="#HGVSRegex-363"><span class="linenos">363</span></a>        <span class="n">CDNA_RANGE</span> <span class="o">+</span> <span class="n">DUP</span><span class="p">,</span>
</span><span id="HGVSRegex-364"><a href="#HGVSRegex-364"><span class="linenos">364</span></a>        <span class="c1"># Indels</span>
</span><span id="HGVSRegex-365"><a href="#HGVSRegex-365"><span class="linenos">365</span></a>        <span class="s2">&quot;(?P&lt;delins&gt;&quot;</span> <span class="o">+</span> <span class="n">CDNA_START</span> <span class="o">+</span> <span class="s2">&quot;del&quot;</span> <span class="o">+</span> <span class="n">DNA_REF</span> <span class="o">+</span> <span class="s2">&quot;ins&quot;</span> <span class="o">+</span> <span class="n">DNA_ALT</span> <span class="o">+</span> <span class="s2">&quot;)&quot;</span><span class="p">,</span>
</span><span id="HGVSRegex-366"><a href="#HGVSRegex-366"><span class="linenos">366</span></a>        <span class="s2">&quot;(?P&lt;delins&gt;&quot;</span> <span class="o">+</span> <span class="n">CDNA_RANGE</span> <span class="o">+</span> <span class="s2">&quot;del&quot;</span> <span class="o">+</span> <span class="n">DNA_REF</span> <span class="o">+</span> <span class="s2">&quot;ins&quot;</span> <span class="o">+</span> <span class="n">DNA_ALT</span> <span class="o">+</span> <span class="s2">&quot;)&quot;</span><span class="p">,</span>
</span><span id="HGVSRegex-367"><a href="#HGVSRegex-367"><span class="linenos">367</span></a>        <span class="s2">&quot;(?P&lt;delins&gt;&quot;</span> <span class="o">+</span> <span class="n">CDNA_START</span> <span class="o">+</span> <span class="s2">&quot;delins&quot;</span> <span class="o">+</span> <span class="n">DNA_ALT</span> <span class="o">+</span> <span class="s2">&quot;)&quot;</span><span class="p">,</span>
</span><span id="HGVSRegex-368"><a href="#HGVSRegex-368"><span class="linenos">368</span></a>        <span class="s2">&quot;(?P&lt;delins&gt;&quot;</span> <span class="o">+</span> <span class="n">CDNA_RANGE</span> <span class="o">+</span> <span class="s2">&quot;delins&quot;</span> <span class="o">+</span> <span class="n">DNA_ALT</span> <span class="o">+</span> <span class="s2">&quot;)&quot;</span><span class="p">,</span>
</span><span id="HGVSRegex-369"><a href="#HGVSRegex-369"><span class="linenos">369</span></a>    <span class="p">]</span>
</span><span id="HGVSRegex-370"><a href="#HGVSRegex-370"><span class="linenos">370</span></a>
</span><span id="HGVSRegex-371"><a href="#HGVSRegex-371"><span class="linenos">371</span></a>    <span class="n">CDNA_ALLELE_REGEXES</span> <span class="o">=</span> <span class="p">[</span><span class="n">re</span><span class="o">.</span><span class="n">compile</span><span class="p">(</span><span class="s2">&quot;^&quot;</span> <span class="o">+</span> <span class="n">regex</span> <span class="o">+</span> <span class="s2">&quot;$&quot;</span><span class="p">)</span> <span class="k">for</span> <span class="n">regex</span> <span class="ow">in</span> <span class="n">CDNA_ALLELE</span><span class="p">]</span>
</span><span id="HGVSRegex-372"><a href="#HGVSRegex-372"><span class="linenos">372</span></a>
</span><span id="HGVSRegex-373"><a href="#HGVSRegex-373"><span class="linenos">373</span></a>    <span class="c1"># Peptide syntax</span>
</span><span id="HGVSRegex-374"><a href="#HGVSRegex-374"><span class="linenos">374</span></a>    <span class="n">PEP</span> <span class="o">=</span> <span class="s2">&quot;([A-Z]([a-z]</span><span class="si">{2}</span><span class="s2">))+&quot;</span>
</span><span id="HGVSRegex-375"><a href="#HGVSRegex-375"><span class="linenos">375</span></a>    <span class="n">PEP_REF</span> <span class="o">=</span> <span class="s2">&quot;(?P&lt;ref&gt;&quot;</span> <span class="o">+</span> <span class="n">PEP</span> <span class="o">+</span> <span class="s2">&quot;)&quot;</span>
</span><span id="HGVSRegex-376"><a href="#HGVSRegex-376"><span class="linenos">376</span></a>    <span class="n">PEP_REF2</span> <span class="o">=</span> <span class="s2">&quot;(?P&lt;ref2&gt;&quot;</span> <span class="o">+</span> <span class="n">PEP</span> <span class="o">+</span> <span class="s2">&quot;)&quot;</span>
</span><span id="HGVSRegex-377"><a href="#HGVSRegex-377"><span class="linenos">377</span></a>    <span class="n">PEP_ALT</span> <span class="o">=</span> <span class="s2">&quot;(?P&lt;alt&gt;&quot;</span> <span class="o">+</span> <span class="n">PEP</span> <span class="o">+</span> <span class="s2">&quot;)&quot;</span>
</span><span id="HGVSRegex-378"><a href="#HGVSRegex-378"><span class="linenos">378</span></a>
</span><span id="HGVSRegex-379"><a href="#HGVSRegex-379"><span class="linenos">379</span></a>    <span class="n">PEP_EXTRA</span> <span class="o">=</span> <span class="sa">r</span><span class="s2">&quot;(?P&lt;extra&gt;(|=|\?)(|fs))&quot;</span>
</span><span id="HGVSRegex-380"><a href="#HGVSRegex-380"><span class="linenos">380</span></a>
</span><span id="HGVSRegex-381"><a href="#HGVSRegex-381"><span class="linenos">381</span></a>    <span class="c1"># Peptide allele syntax</span>
</span><span id="HGVSRegex-382"><a href="#HGVSRegex-382"><span class="linenos">382</span></a>    <span class="n">PEP_ALLELE</span> <span class="o">=</span> <span class="p">[</span>
</span><span id="HGVSRegex-383"><a href="#HGVSRegex-383"><span class="linenos">383</span></a>        <span class="c1"># No peptide change</span>
</span><span id="HGVSRegex-384"><a href="#HGVSRegex-384"><span class="linenos">384</span></a>        <span class="c1"># Example: Glu1161=</span>
</span><span id="HGVSRegex-385"><a href="#HGVSRegex-385"><span class="linenos">385</span></a>        <span class="n">PEP_REF</span> <span class="o">+</span> <span class="n">COORD_START</span> <span class="o">+</span> <span class="n">PEP_EXTRA</span><span class="p">,</span>
</span><span id="HGVSRegex-386"><a href="#HGVSRegex-386"><span class="linenos">386</span></a>        <span class="c1"># Peptide change</span>
</span><span id="HGVSRegex-387"><a href="#HGVSRegex-387"><span class="linenos">387</span></a>        <span class="c1"># Example: Glu1161Ser</span>
</span><span id="HGVSRegex-388"><a href="#HGVSRegex-388"><span class="linenos">388</span></a>        <span class="n">PEP_REF</span> <span class="o">+</span> <span class="n">COORD_START</span> <span class="o">+</span> <span class="n">PEP_ALT</span> <span class="o">+</span> <span class="n">PEP_EXTRA</span><span class="p">,</span>
</span><span id="HGVSRegex-389"><a href="#HGVSRegex-389"><span class="linenos">389</span></a>        <span class="c1"># Peptide indel</span>
</span><span id="HGVSRegex-390"><a href="#HGVSRegex-390"><span class="linenos">390</span></a>        <span class="c1"># Example: Glu1161_Ser1164?fs</span>
</span><span id="HGVSRegex-391"><a href="#HGVSRegex-391"><span class="linenos">391</span></a>        <span class="s2">&quot;(?P&lt;delins&gt;&quot;</span>
</span><span id="HGVSRegex-392"><a href="#HGVSRegex-392"><span class="linenos">392</span></a>        <span class="o">+</span> <span class="n">PEP_REF</span>
</span><span id="HGVSRegex-393"><a href="#HGVSRegex-393"><span class="linenos">393</span></a>        <span class="o">+</span> <span class="n">COORD_START</span>
</span><span id="HGVSRegex-394"><a href="#HGVSRegex-394"><span class="linenos">394</span></a>        <span class="o">+</span> <span class="s2">&quot;_&quot;</span>
</span><span id="HGVSRegex-395"><a href="#HGVSRegex-395"><span class="linenos">395</span></a>        <span class="o">+</span> <span class="n">PEP_REF2</span>
</span><span id="HGVSRegex-396"><a href="#HGVSRegex-396"><span class="linenos">396</span></a>        <span class="o">+</span> <span class="n">COORD_END</span>
</span><span id="HGVSRegex-397"><a href="#HGVSRegex-397"><span class="linenos">397</span></a>        <span class="o">+</span> <span class="n">PEP_EXTRA</span>
</span><span id="HGVSRegex-398"><a href="#HGVSRegex-398"><span class="linenos">398</span></a>        <span class="o">+</span> <span class="s2">&quot;)&quot;</span><span class="p">,</span>
</span><span id="HGVSRegex-399"><a href="#HGVSRegex-399"><span class="linenos">399</span></a>        <span class="s2">&quot;(?P&lt;delins&gt;&quot;</span>
</span><span id="HGVSRegex-400"><a href="#HGVSRegex-400"><span class="linenos">400</span></a>        <span class="o">+</span> <span class="n">PEP_REF</span>
</span><span id="HGVSRegex-401"><a href="#HGVSRegex-401"><span class="linenos">401</span></a>        <span class="o">+</span> <span class="n">COORD_START</span>
</span><span id="HGVSRegex-402"><a href="#HGVSRegex-402"><span class="linenos">402</span></a>        <span class="o">+</span> <span class="s2">&quot;_&quot;</span>
</span><span id="HGVSRegex-403"><a href="#HGVSRegex-403"><span class="linenos">403</span></a>        <span class="o">+</span> <span class="n">PEP_REF2</span>
</span><span id="HGVSRegex-404"><a href="#HGVSRegex-404"><span class="linenos">404</span></a>        <span class="o">+</span> <span class="n">COORD_END</span>
</span><span id="HGVSRegex-405"><a href="#HGVSRegex-405"><span class="linenos">405</span></a>        <span class="o">+</span> <span class="n">PEP_ALT</span>
</span><span id="HGVSRegex-406"><a href="#HGVSRegex-406"><span class="linenos">406</span></a>        <span class="o">+</span> <span class="n">PEP_EXTRA</span>
</span><span id="HGVSRegex-407"><a href="#HGVSRegex-407"><span class="linenos">407</span></a>        <span class="o">+</span> <span class="s2">&quot;)&quot;</span><span class="p">,</span>
</span><span id="HGVSRegex-408"><a href="#HGVSRegex-408"><span class="linenos">408</span></a>    <span class="p">]</span>
</span><span id="HGVSRegex-409"><a href="#HGVSRegex-409"><span class="linenos">409</span></a>
</span><span id="HGVSRegex-410"><a href="#HGVSRegex-410"><span class="linenos">410</span></a>    <span class="n">PEP_ALLELE_REGEXES</span> <span class="o">=</span> <span class="p">[</span><span class="n">re</span><span class="o">.</span><span class="n">compile</span><span class="p">(</span><span class="s2">&quot;^&quot;</span> <span class="o">+</span> <span class="n">regex</span> <span class="o">+</span> <span class="s2">&quot;$&quot;</span><span class="p">)</span> <span class="k">for</span> <span class="n">regex</span> <span class="ow">in</span> <span class="n">PEP_ALLELE</span><span class="p">]</span>
</span><span id="HGVSRegex-411"><a href="#HGVSRegex-411"><span class="linenos">411</span></a>
</span><span id="HGVSRegex-412"><a href="#HGVSRegex-412"><span class="linenos">412</span></a>    <span class="c1"># Genomic allele syntax</span>
</span><span id="HGVSRegex-413"><a href="#HGVSRegex-413"><span class="linenos">413</span></a>    <span class="n">GENOMIC_ALLELE</span> <span class="o">=</span> <span class="p">[</span>
</span><span id="HGVSRegex-414"><a href="#HGVSRegex-414"><span class="linenos">414</span></a>        <span class="c1"># No change</span>
</span><span id="HGVSRegex-415"><a href="#HGVSRegex-415"><span class="linenos">415</span></a>        <span class="n">COORD_START</span> <span class="o">+</span> <span class="n">EQUAL</span><span class="p">,</span>
</span><span id="HGVSRegex-416"><a href="#HGVSRegex-416"><span class="linenos">416</span></a>        <span class="n">COORD_START</span> <span class="o">+</span> <span class="n">DNA_REF</span> <span class="o">+</span> <span class="n">EQUAL</span><span class="p">,</span>
</span><span id="HGVSRegex-417"><a href="#HGVSRegex-417"><span class="linenos">417</span></a>        <span class="c1"># Substitution</span>
</span><span id="HGVSRegex-418"><a href="#HGVSRegex-418"><span class="linenos">418</span></a>        <span class="n">COORD_START</span> <span class="o">+</span> <span class="n">DNA_REF</span> <span class="o">+</span> <span class="n">SUB</span> <span class="o">+</span> <span class="n">DNA_ALT</span><span class="p">,</span>
</span><span id="HGVSRegex-419"><a href="#HGVSRegex-419"><span class="linenos">419</span></a>        <span class="c1"># 1bp insertion, deletion, duplication</span>
</span><span id="HGVSRegex-420"><a href="#HGVSRegex-420"><span class="linenos">420</span></a>        <span class="n">COORD_START</span> <span class="o">+</span> <span class="n">INS</span> <span class="o">+</span> <span class="n">DNA_ALT</span><span class="p">,</span>
</span><span id="HGVSRegex-421"><a href="#HGVSRegex-421"><span class="linenos">421</span></a>        <span class="n">COORD_START</span> <span class="o">+</span> <span class="n">DEL</span> <span class="o">+</span> <span class="n">DNA_REF</span><span class="p">,</span>
</span><span id="HGVSRegex-422"><a href="#HGVSRegex-422"><span class="linenos">422</span></a>        <span class="n">COORD_START</span> <span class="o">+</span> <span class="n">DUP</span> <span class="o">+</span> <span class="n">DNA_REF</span><span class="p">,</span>
</span><span id="HGVSRegex-423"><a href="#HGVSRegex-423"><span class="linenos">423</span></a>        <span class="n">COORD_START</span> <span class="o">+</span> <span class="n">DEL</span><span class="p">,</span>
</span><span id="HGVSRegex-424"><a href="#HGVSRegex-424"><span class="linenos">424</span></a>        <span class="n">COORD_START</span> <span class="o">+</span> <span class="n">DUP</span><span class="p">,</span>
</span><span id="HGVSRegex-425"><a href="#HGVSRegex-425"><span class="linenos">425</span></a>        <span class="c1"># Insertion, deletion, duplication</span>
</span><span id="HGVSRegex-426"><a href="#HGVSRegex-426"><span class="linenos">426</span></a>        <span class="n">COORD_RANGE</span> <span class="o">+</span> <span class="n">EQUAL</span><span class="p">,</span>
</span><span id="HGVSRegex-427"><a href="#HGVSRegex-427"><span class="linenos">427</span></a>        <span class="n">COORD_RANGE</span> <span class="o">+</span> <span class="n">INS</span> <span class="o">+</span> <span class="n">DNA_ALT</span><span class="p">,</span>
</span><span id="HGVSRegex-428"><a href="#HGVSRegex-428"><span class="linenos">428</span></a>        <span class="n">COORD_RANGE</span> <span class="o">+</span> <span class="n">DEL</span> <span class="o">+</span> <span class="n">DNA_REF</span><span class="p">,</span>
</span><span id="HGVSRegex-429"><a href="#HGVSRegex-429"><span class="linenos">429</span></a>        <span class="n">COORD_RANGE</span> <span class="o">+</span> <span class="n">DUP</span> <span class="o">+</span> <span class="n">DNA_REF</span><span class="p">,</span>
</span><span id="HGVSRegex-430"><a href="#HGVSRegex-430"><span class="linenos">430</span></a>        <span class="n">COORD_RANGE</span> <span class="o">+</span> <span class="n">DEL</span><span class="p">,</span>
</span><span id="HGVSRegex-431"><a href="#HGVSRegex-431"><span class="linenos">431</span></a>        <span class="n">COORD_RANGE</span> <span class="o">+</span> <span class="n">DUP</span><span class="p">,</span>
</span><span id="HGVSRegex-432"><a href="#HGVSRegex-432"><span class="linenos">432</span></a>        <span class="c1"># Indels</span>
</span><span id="HGVSRegex-433"><a href="#HGVSRegex-433"><span class="linenos">433</span></a>        <span class="s2">&quot;(?P&lt;delins&gt;&quot;</span> <span class="o">+</span> <span class="n">COORD_START</span> <span class="o">+</span> <span class="s2">&quot;del&quot;</span> <span class="o">+</span> <span class="n">DNA_REF</span> <span class="o">+</span> <span class="s2">&quot;ins&quot;</span> <span class="o">+</span> <span class="n">DNA_ALT</span> <span class="o">+</span> <span class="s2">&quot;)&quot;</span><span class="p">,</span>
</span><span id="HGVSRegex-434"><a href="#HGVSRegex-434"><span class="linenos">434</span></a>        <span class="s2">&quot;(?P&lt;delins&gt;&quot;</span> <span class="o">+</span> <span class="n">COORD_RANGE</span> <span class="o">+</span> <span class="s2">&quot;del&quot;</span> <span class="o">+</span> <span class="n">DNA_REF</span> <span class="o">+</span> <span class="s2">&quot;ins&quot;</span> <span class="o">+</span> <span class="n">DNA_ALT</span> <span class="o">+</span> <span class="s2">&quot;)&quot;</span><span class="p">,</span>
</span><span id="HGVSRegex-435"><a href="#HGVSRegex-435"><span class="linenos">435</span></a>        <span class="s2">&quot;(?P&lt;delins&gt;&quot;</span> <span class="o">+</span> <span class="n">COORD_START</span> <span class="o">+</span> <span class="s2">&quot;delins&quot;</span> <span class="o">+</span> <span class="n">DNA_ALT</span> <span class="o">+</span> <span class="s2">&quot;)&quot;</span><span class="p">,</span>
</span><span id="HGVSRegex-436"><a href="#HGVSRegex-436"><span class="linenos">436</span></a>        <span class="s2">&quot;(?P&lt;delins&gt;&quot;</span> <span class="o">+</span> <span class="n">COORD_RANGE</span> <span class="o">+</span> <span class="s2">&quot;delins&quot;</span> <span class="o">+</span> <span class="n">DNA_ALT</span> <span class="o">+</span> <span class="s2">&quot;)&quot;</span><span class="p">,</span>
</span><span id="HGVSRegex-437"><a href="#HGVSRegex-437"><span class="linenos">437</span></a>    <span class="p">]</span>
</span><span id="HGVSRegex-438"><a href="#HGVSRegex-438"><span class="linenos">438</span></a>
</span><span id="HGVSRegex-439"><a href="#HGVSRegex-439"><span class="linenos">439</span></a>    <span class="n">GENOMIC_ALLELE_REGEXES</span> <span class="o">=</span> <span class="p">[</span><span class="n">re</span><span class="o">.</span><span class="n">compile</span><span class="p">(</span><span class="s2">&quot;^&quot;</span> <span class="o">+</span> <span class="n">regex</span> <span class="o">+</span> <span class="s2">&quot;$&quot;</span><span class="p">)</span> <span class="k">for</span> <span class="n">regex</span> <span class="ow">in</span> <span class="n">GENOMIC_ALLELE</span><span class="p">]</span>
</span></pre></div>


            <div class="docstring"><p>All regular expression for HGVS names.</p>
</div>


                            <div id="HGVSRegex.BASE" class="classattr">
                                <div class="attr variable">
            <span class="name">BASE</span>        =
<span class="default_value">&#39;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]|\\d+&#39;</span>

        
    </div>
    <a class="headerlink" href="#HGVSRegex.BASE"></a>
    
    

                            </div>
                            <div id="HGVSRegex.BASES" class="classattr">
                                <div class="attr variable">
            <span class="name">BASES</span>        =
<span class="default_value">&#39;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+&#39;</span>

        
    </div>
    <a class="headerlink" href="#HGVSRegex.BASES"></a>
    
    

                            </div>
                            <div id="HGVSRegex.DNA_REF" class="classattr">
                                <div class="attr variable">
            <span class="name">DNA_REF</span>        =
<span class="default_value">&#39;(?P&lt;ref&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+)&#39;</span>

        
    </div>
    <a class="headerlink" href="#HGVSRegex.DNA_REF"></a>
    
    

                            </div>
                            <div id="HGVSRegex.DNA_ALT" class="classattr">
                                <div class="attr variable">
            <span class="name">DNA_ALT</span>        =
<span class="default_value">&#39;(?P&lt;alt&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+)&#39;</span>

        
    </div>
    <a class="headerlink" href="#HGVSRegex.DNA_ALT"></a>
    
    

                            </div>
                            <div id="HGVSRegex.EQUAL" class="classattr">
                                <div class="attr variable">
            <span class="name">EQUAL</span>        =
<span class="default_value">&#39;(?P&lt;mutation_type&gt;=)&#39;</span>

        
    </div>
    <a class="headerlink" href="#HGVSRegex.EQUAL"></a>
    
    

                            </div>
                            <div id="HGVSRegex.SUB" class="classattr">
                                <div class="attr variable">
            <span class="name">SUB</span>        =
<span class="default_value">&#39;(?P&lt;mutation_type&gt;&gt;)&#39;</span>

        
    </div>
    <a class="headerlink" href="#HGVSRegex.SUB"></a>
    
    

                            </div>
                            <div id="HGVSRegex.INS" class="classattr">
                                <div class="attr variable">
            <span class="name">INS</span>        =
<span class="default_value">&#39;(?P&lt;mutation_type&gt;ins)&#39;</span>

        
    </div>
    <a class="headerlink" href="#HGVSRegex.INS"></a>
    
    

                            </div>
                            <div id="HGVSRegex.DEL" class="classattr">
                                <div class="attr variable">
            <span class="name">DEL</span>        =
<span class="default_value">&#39;(?P&lt;mutation_type&gt;del)&#39;</span>

        
    </div>
    <a class="headerlink" href="#HGVSRegex.DEL"></a>
    
    

                            </div>
                            <div id="HGVSRegex.DUP" class="classattr">
                                <div class="attr variable">
            <span class="name">DUP</span>        =
<span class="default_value">&#39;(?P&lt;mutation_type&gt;dup)&#39;</span>

        
    </div>
    <a class="headerlink" href="#HGVSRegex.DUP"></a>
    
    

                            </div>
                            <div id="HGVSRegex.COORD_START" class="classattr">
                                <div class="attr variable">
            <span class="name">COORD_START</span>        =
<span class="default_value">&#39;(?P&lt;start&gt;\\d+)&#39;</span>

        
    </div>
    <a class="headerlink" href="#HGVSRegex.COORD_START"></a>
    
    

                            </div>
                            <div id="HGVSRegex.COORD_END" class="classattr">
                                <div class="attr variable">
            <span class="name">COORD_END</span>        =
<span class="default_value">&#39;(?P&lt;end&gt;\\d+)&#39;</span>

        
    </div>
    <a class="headerlink" href="#HGVSRegex.COORD_END"></a>
    
    

                            </div>
                            <div id="HGVSRegex.COORD_RANGE" class="classattr">
                                <div class="attr variable">
            <span class="name">COORD_RANGE</span>        =
<span class="default_value">&#39;(?P&lt;start&gt;\\d+)_(?P&lt;end&gt;\\d+)&#39;</span>

        
    </div>
    <a class="headerlink" href="#HGVSRegex.COORD_RANGE"></a>
    
    

                            </div>
                            <div id="HGVSRegex.CDNA_COORD" class="classattr">
                                <div class="attr variable">
            <span class="name">CDNA_COORD</span>        =
<span class="default_value">&#39;(?P&lt;coord_prefix&gt;|-|\\*)(?P&lt;coord&gt;\\d+)((?P&lt;offset_prefix&gt;-|\\+)(?P&lt;offset&gt;\\d+))?&#39;</span>

        
    </div>
    <a class="headerlink" href="#HGVSRegex.CDNA_COORD"></a>
    
    

                            </div>
                            <div id="HGVSRegex.CDNA_START" class="classattr">
                                <div class="attr variable">
            <span class="name">CDNA_START</span>        =
<input id="HGVSRegex.CDNA_START-view-value" class="view-value-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
            <label class="view-value-button pdoc-button" for="HGVSRegex.CDNA_START-view-value"></label><span class="default_value">&#39;(?P&lt;start&gt;(?P&lt;start_coord_prefix&gt;|-|\\*)(?P&lt;start_coord&gt;\\d+)((?P&lt;start_offset_prefix&gt;-|\\+)(?P&lt;start_offset&gt;\\d+))?)&#39;</span>

        
    </div>
    <a class="headerlink" href="#HGVSRegex.CDNA_START"></a>
    
    

                            </div>
                            <div id="HGVSRegex.CDNA_END" class="classattr">
                                <div class="attr variable">
            <span class="name">CDNA_END</span>        =
<input id="HGVSRegex.CDNA_END-view-value" class="view-value-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
            <label class="view-value-button pdoc-button" for="HGVSRegex.CDNA_END-view-value"></label><span class="default_value">&#39;(?P&lt;end&gt;(?P&lt;end_coord_prefix&gt;|-|\\*)(?P&lt;end_coord&gt;\\d+)((?P&lt;end_offset_prefix&gt;-|\\+)(?P&lt;end_offset&gt;\\d+))?)&#39;</span>

        
    </div>
    <a class="headerlink" href="#HGVSRegex.CDNA_END"></a>
    
    

                            </div>
                            <div id="HGVSRegex.CDNA_RANGE" class="classattr">
                                <div class="attr variable">
            <span class="name">CDNA_RANGE</span>        =
<input id="HGVSRegex.CDNA_RANGE-view-value" class="view-value-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
            <label class="view-value-button pdoc-button" for="HGVSRegex.CDNA_RANGE-view-value"></label><span class="default_value">&#39;(?P&lt;start&gt;(?P&lt;start_coord_prefix&gt;|-|\\*)(?P&lt;start_coord&gt;\\d+)((?P&lt;start_offset_prefix&gt;-|\\+)(?P&lt;start_offset&gt;\\d+))?)_(?P&lt;end&gt;(?P&lt;end_coord_prefix&gt;|-|\\*)(?P&lt;end_coord&gt;\\d+)((?P&lt;end_offset_prefix&gt;-|\\+)(?P&lt;end_offset&gt;\\d+))?)&#39;</span>

        
    </div>
    <a class="headerlink" href="#HGVSRegex.CDNA_RANGE"></a>
    
    

                            </div>
                            <div id="HGVSRegex.CDNA_ALLELE" class="classattr">
                                <div class="attr variable">
            <span class="name">CDNA_ALLELE</span>        =
<input id="HGVSRegex.CDNA_ALLELE-view-value" class="view-value-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
            <label class="view-value-button pdoc-button" for="HGVSRegex.CDNA_ALLELE-view-value"></label><span class="default_value">[&#39;(?P&lt;start&gt;(?P&lt;start_coord_prefix&gt;|-|\\*)(?P&lt;start_coord&gt;\\d+)((?P&lt;start_offset_prefix&gt;-|\\+)(?P&lt;start_offset&gt;\\d+))?)(?P&lt;mutation_type&gt;=)&#39;, &#39;(?P&lt;start&gt;(?P&lt;start_coord_prefix&gt;|-|\\*)(?P&lt;start_coord&gt;\\d+)((?P&lt;start_offset_prefix&gt;-|\\+)(?P&lt;start_offset&gt;\\d+))?)(?P&lt;ref&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+)(?P&lt;mutation_type&gt;=)&#39;, &#39;(?P&lt;start&gt;(?P&lt;start_coord_prefix&gt;|-|\\*)(?P&lt;start_coord&gt;\\d+)((?P&lt;start_offset_prefix&gt;-|\\+)(?P&lt;start_offset&gt;\\d+))?)(?P&lt;ref&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+)(?P&lt;mutation_type&gt;&gt;)(?P&lt;alt&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+)&#39;, &#39;(?P&lt;start&gt;(?P&lt;start_coord_prefix&gt;|-|\\*)(?P&lt;start_coord&gt;\\d+)((?P&lt;start_offset_prefix&gt;-|\\+)(?P&lt;start_offset&gt;\\d+))?)(?P&lt;mutation_type&gt;ins)(?P&lt;alt&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+)&#39;, &#39;(?P&lt;start&gt;(?P&lt;start_coord_prefix&gt;|-|\\*)(?P&lt;start_coord&gt;\\d+)((?P&lt;start_offset_prefix&gt;-|\\+)(?P&lt;start_offset&gt;\\d+))?)(?P&lt;mutation_type&gt;del)(?P&lt;ref&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+)&#39;, &#39;(?P&lt;start&gt;(?P&lt;start_coord_prefix&gt;|-|\\*)(?P&lt;start_coord&gt;\\d+)((?P&lt;start_offset_prefix&gt;-|\\+)(?P&lt;start_offset&gt;\\d+))?)(?P&lt;mutation_type&gt;dup)(?P&lt;ref&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+)&#39;, &#39;(?P&lt;start&gt;(?P&lt;start_coord_prefix&gt;|-|\\*)(?P&lt;start_coord&gt;\\d+)((?P&lt;start_offset_prefix&gt;-|\\+)(?P&lt;start_offset&gt;\\d+))?)(?P&lt;mutation_type&gt;del)&#39;, &#39;(?P&lt;start&gt;(?P&lt;start_coord_prefix&gt;|-|\\*)(?P&lt;start_coord&gt;\\d+)((?P&lt;start_offset_prefix&gt;-|\\+)(?P&lt;start_offset&gt;\\d+))?)(?P&lt;mutation_type&gt;dup)&#39;, &#39;(?P&lt;start&gt;\\d+)_(?P&lt;end&gt;\\d+)(?P&lt;mutation_type&gt;=)&#39;, &#39;(?P&lt;start&gt;(?P&lt;start_coord_prefix&gt;|-|\\*)(?P&lt;start_coord&gt;\\d+)((?P&lt;start_offset_prefix&gt;-|\\+)(?P&lt;start_offset&gt;\\d+))?)_(?P&lt;end&gt;(?P&lt;end_coord_prefix&gt;|-|\\*)(?P&lt;end_coord&gt;\\d+)((?P&lt;end_offset_prefix&gt;-|\\+)(?P&lt;end_offset&gt;\\d+))?)(?P&lt;mutation_type&gt;ins)(?P&lt;alt&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+)&#39;, &#39;(?P&lt;start&gt;(?P&lt;start_coord_prefix&gt;|-|\\*)(?P&lt;start_coord&gt;\\d+)((?P&lt;start_offset_prefix&gt;-|\\+)(?P&lt;start_offset&gt;\\d+))?)_(?P&lt;end&gt;(?P&lt;end_coord_prefix&gt;|-|\\*)(?P&lt;end_coord&gt;\\d+)((?P&lt;end_offset_prefix&gt;-|\\+)(?P&lt;end_offset&gt;\\d+))?)(?P&lt;mutation_type&gt;del)(?P&lt;ref&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+)&#39;, &#39;(?P&lt;start&gt;(?P&lt;start_coord_prefix&gt;|-|\\*)(?P&lt;start_coord&gt;\\d+)((?P&lt;start_offset_prefix&gt;-|\\+)(?P&lt;start_offset&gt;\\d+))?)_(?P&lt;end&gt;(?P&lt;end_coord_prefix&gt;|-|\\*)(?P&lt;end_coord&gt;\\d+)((?P&lt;end_offset_prefix&gt;-|\\+)(?P&lt;end_offset&gt;\\d+))?)(?P&lt;mutation_type&gt;dup)(?P&lt;ref&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+)&#39;, &#39;(?P&lt;start&gt;(?P&lt;start_coord_prefix&gt;|-|\\*)(?P&lt;start_coord&gt;\\d+)((?P&lt;start_offset_prefix&gt;-|\\+)(?P&lt;start_offset&gt;\\d+))?)_(?P&lt;end&gt;(?P&lt;end_coord_prefix&gt;|-|\\*)(?P&lt;end_coord&gt;\\d+)((?P&lt;end_offset_prefix&gt;-|\\+)(?P&lt;end_offset&gt;\\d+))?)(?P&lt;mutation_type&gt;del)&#39;, &#39;(?P&lt;start&gt;(?P&lt;start_coord_prefix&gt;|-|\\*)(?P&lt;start_coord&gt;\\d+)((?P&lt;start_offset_prefix&gt;-|\\+)(?P&lt;start_offset&gt;\\d+))?)_(?P&lt;end&gt;(?P&lt;end_coord_prefix&gt;|-|\\*)(?P&lt;end_coord&gt;\\d+)((?P&lt;end_offset_prefix&gt;-|\\+)(?P&lt;end_offset&gt;\\d+))?)(?P&lt;mutation_type&gt;dup)&#39;, &#39;(?P&lt;delins&gt;(?P&lt;start&gt;(?P&lt;start_coord_prefix&gt;|-|\\*)(?P&lt;start_coord&gt;\\d+)((?P&lt;start_offset_prefix&gt;-|\\+)(?P&lt;start_offset&gt;\\d+))?)del(?P&lt;ref&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+)ins(?P&lt;alt&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+))&#39;, &#39;(?P&lt;delins&gt;(?P&lt;start&gt;(?P&lt;start_coord_prefix&gt;|-|\\*)(?P&lt;start_coord&gt;\\d+)((?P&lt;start_offset_prefix&gt;-|\\+)(?P&lt;start_offset&gt;\\d+))?)_(?P&lt;end&gt;(?P&lt;end_coord_prefix&gt;|-|\\*)(?P&lt;end_coord&gt;\\d+)((?P&lt;end_offset_prefix&gt;-|\\+)(?P&lt;end_offset&gt;\\d+))?)del(?P&lt;ref&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+)ins(?P&lt;alt&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+))&#39;, &#39;(?P&lt;delins&gt;(?P&lt;start&gt;(?P&lt;start_coord_prefix&gt;|-|\\*)(?P&lt;start_coord&gt;\\d+)((?P&lt;start_offset_prefix&gt;-|\\+)(?P&lt;start_offset&gt;\\d+))?)delins(?P&lt;alt&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+))&#39;, &#39;(?P&lt;delins&gt;(?P&lt;start&gt;(?P&lt;start_coord_prefix&gt;|-|\\*)(?P&lt;start_coord&gt;\\d+)((?P&lt;start_offset_prefix&gt;-|\\+)(?P&lt;start_offset&gt;\\d+))?)_(?P&lt;end&gt;(?P&lt;end_coord_prefix&gt;|-|\\*)(?P&lt;end_coord&gt;\\d+)((?P&lt;end_offset_prefix&gt;-|\\+)(?P&lt;end_offset&gt;\\d+))?)delins(?P&lt;alt&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+))&#39;]</span>

        
    </div>
    <a class="headerlink" href="#HGVSRegex.CDNA_ALLELE"></a>
    
    

                            </div>
                            <div id="HGVSRegex.CDNA_ALLELE_REGEXES" class="classattr">
                                <div class="attr variable">
            <span class="name">CDNA_ALLELE_REGEXES</span>        =
<input id="HGVSRegex.CDNA_ALLELE_REGEXES-view-value" class="view-value-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
            <label class="view-value-button pdoc-button" for="HGVSRegex.CDNA_ALLELE_REGEXES-view-value"></label><span class="default_value">[re.compile(&#39;^(?P&lt;start&gt;(?P&lt;start_coord_prefix&gt;|-|\\*)(?P&lt;start_coord&gt;\\d+)((?P&lt;start_offset_prefix&gt;-|\\+)(?P&lt;start_offset&gt;\\d+))?)(?P&lt;mutation_type&gt;=)$&#39;), re.compile(&#39;^(?P&lt;start&gt;(?P&lt;start_coord_prefix&gt;|-|\\*)(?P&lt;start_coord&gt;\\d+)((?P&lt;start_offset_prefix&gt;-|\\+)(?P&lt;start_offset&gt;\\d+))?)(?P&lt;ref&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+)(?P&lt;mutation_type&gt;=)$&#39;), re.compile(&#39;^(?P&lt;start&gt;(?P&lt;start_coord_prefix&gt;|-|\\*)(?P&lt;start_coord&gt;\\d+)((?P&lt;start_offset_prefix&gt;-|\\+)(?P&lt;start_offset&gt;\\d+))?)(?P&lt;ref&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+)(?P&lt;mutation_type&gt;&gt;)(?P&lt;alt&gt;[acgtb), re.compile(&#39;^(?P&lt;start&gt;(?P&lt;start_coord_prefix&gt;|-|\\*)(?P&lt;start_coord&gt;\\d+)((?P&lt;start_offset_prefix&gt;-|\\+)(?P&lt;start_offset&gt;\\d+))?)(?P&lt;mutation_type&gt;ins)(?P&lt;alt&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+)$&#39;), re.compile(&#39;^(?P&lt;start&gt;(?P&lt;start_coord_prefix&gt;|-|\\*)(?P&lt;start_coord&gt;\\d+)((?P&lt;start_offset_prefix&gt;-|\\+)(?P&lt;start_offset&gt;\\d+))?)(?P&lt;mutation_type&gt;del)(?P&lt;ref&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+)$&#39;), re.compile(&#39;^(?P&lt;start&gt;(?P&lt;start_coord_prefix&gt;|-|\\*)(?P&lt;start_coord&gt;\\d+)((?P&lt;start_offset_prefix&gt;-|\\+)(?P&lt;start_offset&gt;\\d+))?)(?P&lt;mutation_type&gt;dup)(?P&lt;ref&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+)$&#39;), re.compile(&#39;^(?P&lt;start&gt;(?P&lt;start_coord_prefix&gt;|-|\\*)(?P&lt;start_coord&gt;\\d+)((?P&lt;start_offset_prefix&gt;-|\\+)(?P&lt;start_offset&gt;\\d+))?)(?P&lt;mutation_type&gt;del)$&#39;), re.compile(&#39;^(?P&lt;start&gt;(?P&lt;start_coord_prefix&gt;|-|\\*)(?P&lt;start_coord&gt;\\d+)((?P&lt;start_offset_prefix&gt;-|\\+)(?P&lt;start_offset&gt;\\d+))?)(?P&lt;mutation_type&gt;dup)$&#39;), re.compile(&#39;^(?P&lt;start&gt;\\d+)_(?P&lt;end&gt;\\d+)(?P&lt;mutation_type&gt;=)$&#39;), re.compile(&#39;^(?P&lt;start&gt;(?P&lt;start_coord_prefix&gt;|-|\\*)(?P&lt;start_coord&gt;\\d+)((?P&lt;start_offset_prefix&gt;-|\\+)(?P&lt;start_offset&gt;\\d+))?)_(?P&lt;end&gt;(?P&lt;end_coord_prefix&gt;|-|\\*)(?P&lt;end_coord&gt;\\d+)((?P&lt;end_offset_prefix&gt;-|), re.compile(&#39;^(?P&lt;start&gt;(?P&lt;start_coord_prefix&gt;|-|\\*)(?P&lt;start_coord&gt;\\d+)((?P&lt;start_offset_prefix&gt;-|\\+)(?P&lt;start_offset&gt;\\d+))?)_(?P&lt;end&gt;(?P&lt;end_coord_prefix&gt;|-|\\*)(?P&lt;end_coord&gt;\\d+)((?P&lt;end_offset_prefix&gt;-|), re.compile(&#39;^(?P&lt;start&gt;(?P&lt;start_coord_prefix&gt;|-|\\*)(?P&lt;start_coord&gt;\\d+)((?P&lt;start_offset_prefix&gt;-|\\+)(?P&lt;start_offset&gt;\\d+))?)_(?P&lt;end&gt;(?P&lt;end_coord_prefix&gt;|-|\\*)(?P&lt;end_coord&gt;\\d+)((?P&lt;end_offset_prefix&gt;-|), re.compile(&#39;^(?P&lt;start&gt;(?P&lt;start_coord_prefix&gt;|-|\\*)(?P&lt;start_coord&gt;\\d+)((?P&lt;start_offset_prefix&gt;-|\\+)(?P&lt;start_offset&gt;\\d+))?)_(?P&lt;end&gt;(?P&lt;end_coord_prefix&gt;|-|\\*)(?P&lt;end_coord&gt;\\d+)((?P&lt;end_offset_prefix&gt;-|), re.compile(&#39;^(?P&lt;start&gt;(?P&lt;start_coord_prefix&gt;|-|\\*)(?P&lt;start_coord&gt;\\d+)((?P&lt;start_offset_prefix&gt;-|\\+)(?P&lt;start_offset&gt;\\d+))?)_(?P&lt;end&gt;(?P&lt;end_coord_prefix&gt;|-|\\*)(?P&lt;end_coord&gt;\\d+)((?P&lt;end_offset_prefix&gt;-|), re.compile(&#39;^(?P&lt;delins&gt;(?P&lt;start&gt;(?P&lt;start_coord_prefix&gt;|-|\\*)(?P&lt;start_coord&gt;\\d+)((?P&lt;start_offset_prefix&gt;-|\\+)(?P&lt;start_offset&gt;\\d+))?)del(?P&lt;ref&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+)ins(?P&lt;alt&gt;[acgtbdhk), re.compile(&#39;^(?P&lt;delins&gt;(?P&lt;start&gt;(?P&lt;start_coord_prefix&gt;|-|\\*)(?P&lt;start_coord&gt;\\d+)((?P&lt;start_offset_prefix&gt;-|\\+)(?P&lt;start_offset&gt;\\d+))?)_(?P&lt;end&gt;(?P&lt;end_coord_prefix&gt;|-|\\*)(?P&lt;end_coord&gt;\\d+)((?P&lt;end_offse), re.compile(&#39;^(?P&lt;delins&gt;(?P&lt;start&gt;(?P&lt;start_coord_prefix&gt;|-|\\*)(?P&lt;start_coord&gt;\\d+)((?P&lt;start_offset_prefix&gt;-|\\+)(?P&lt;start_offset&gt;\\d+))?)delins(?P&lt;alt&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+))$&#39;), re.compile(&#39;^(?P&lt;delins&gt;(?P&lt;start&gt;(?P&lt;start_coord_prefix&gt;|-|\\*)(?P&lt;start_coord&gt;\\d+)((?P&lt;start_offset_prefix&gt;-|\\+)(?P&lt;start_offset&gt;\\d+))?)_(?P&lt;end&gt;(?P&lt;end_coord_prefix&gt;|-|\\*)(?P&lt;end_coord&gt;\\d+)((?P&lt;end_offse)]</span>

        
    </div>
    <a class="headerlink" href="#HGVSRegex.CDNA_ALLELE_REGEXES"></a>
    
    

                            </div>
                            <div id="HGVSRegex.PEP" class="classattr">
                                <div class="attr variable">
            <span class="name">PEP</span>        =
<span class="default_value">&#39;([A-Z]([a-z]{2}))+&#39;</span>

        
    </div>
    <a class="headerlink" href="#HGVSRegex.PEP"></a>
    
    

                            </div>
                            <div id="HGVSRegex.PEP_REF" class="classattr">
                                <div class="attr variable">
            <span class="name">PEP_REF</span>        =
<span class="default_value">&#39;(?P&lt;ref&gt;([A-Z]([a-z]{2}))+)&#39;</span>

        
    </div>
    <a class="headerlink" href="#HGVSRegex.PEP_REF"></a>
    
    

                            </div>
                            <div id="HGVSRegex.PEP_REF2" class="classattr">
                                <div class="attr variable">
            <span class="name">PEP_REF2</span>        =
<span class="default_value">&#39;(?P&lt;ref2&gt;([A-Z]([a-z]{2}))+)&#39;</span>

        
    </div>
    <a class="headerlink" href="#HGVSRegex.PEP_REF2"></a>
    
    

                            </div>
                            <div id="HGVSRegex.PEP_ALT" class="classattr">
                                <div class="attr variable">
            <span class="name">PEP_ALT</span>        =
<span class="default_value">&#39;(?P&lt;alt&gt;([A-Z]([a-z]{2}))+)&#39;</span>

        
    </div>
    <a class="headerlink" href="#HGVSRegex.PEP_ALT"></a>
    
    

                            </div>
                            <div id="HGVSRegex.PEP_EXTRA" class="classattr">
                                <div class="attr variable">
            <span class="name">PEP_EXTRA</span>        =
<span class="default_value">&#39;(?P&lt;extra&gt;(|=|\\?)(|fs))&#39;</span>

        
    </div>
    <a class="headerlink" href="#HGVSRegex.PEP_EXTRA"></a>
    
    

                            </div>
                            <div id="HGVSRegex.PEP_ALLELE" class="classattr">
                                <div class="attr variable">
            <span class="name">PEP_ALLELE</span>        =
<input id="HGVSRegex.PEP_ALLELE-view-value" class="view-value-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
            <label class="view-value-button pdoc-button" for="HGVSRegex.PEP_ALLELE-view-value"></label><span class="default_value">[&#39;(?P&lt;ref&gt;([A-Z]([a-z]{2}))+)(?P&lt;start&gt;\\d+)(?P&lt;extra&gt;(|=|\\?)(|fs))&#39;, &#39;(?P&lt;ref&gt;([A-Z]([a-z]{2}))+)(?P&lt;start&gt;\\d+)(?P&lt;alt&gt;([A-Z]([a-z]{2}))+)(?P&lt;extra&gt;(|=|\\?)(|fs))&#39;, &#39;(?P&lt;delins&gt;(?P&lt;ref&gt;([A-Z]([a-z]{2}))+)(?P&lt;start&gt;\\d+)_(?P&lt;ref2&gt;([A-Z]([a-z]{2}))+)(?P&lt;end&gt;\\d+)(?P&lt;extra&gt;(|=|\\?)(|fs)))&#39;, &#39;(?P&lt;delins&gt;(?P&lt;ref&gt;([A-Z]([a-z]{2}))+)(?P&lt;start&gt;\\d+)_(?P&lt;ref2&gt;([A-Z]([a-z]{2}))+)(?P&lt;end&gt;\\d+)(?P&lt;alt&gt;([A-Z]([a-z]{2}))+)(?P&lt;extra&gt;(|=|\\?)(|fs)))&#39;]</span>

        
    </div>
    <a class="headerlink" href="#HGVSRegex.PEP_ALLELE"></a>
    
    

                            </div>
                            <div id="HGVSRegex.PEP_ALLELE_REGEXES" class="classattr">
                                <div class="attr variable">
            <span class="name">PEP_ALLELE_REGEXES</span>        =
<input id="HGVSRegex.PEP_ALLELE_REGEXES-view-value" class="view-value-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
            <label class="view-value-button pdoc-button" for="HGVSRegex.PEP_ALLELE_REGEXES-view-value"></label><span class="default_value">[re.compile(&#39;^(?P&lt;ref&gt;([A-Z]([a-z]{2}))+)(?P&lt;start&gt;\\d+)(?P&lt;extra&gt;(|=|\\?)(|fs))$&#39;), re.compile(&#39;^(?P&lt;ref&gt;([A-Z]([a-z]{2}))+)(?P&lt;start&gt;\\d+)(?P&lt;alt&gt;([A-Z]([a-z]{2}))+)(?P&lt;extra&gt;(|=|\\?)(|fs))$&#39;), re.compile(&#39;^(?P&lt;delins&gt;(?P&lt;ref&gt;([A-Z]([a-z]{2}))+)(?P&lt;start&gt;\\d+)_(?P&lt;ref2&gt;([A-Z]([a-z]{2}))+)(?P&lt;end&gt;\\d+)(?P&lt;extra&gt;(|=|\\?)(|fs)))$&#39;), re.compile(&#39;^(?P&lt;delins&gt;(?P&lt;ref&gt;([A-Z]([a-z]{2}))+)(?P&lt;start&gt;\\d+)_(?P&lt;ref2&gt;([A-Z]([a-z]{2}))+)(?P&lt;end&gt;\\d+)(?P&lt;alt&gt;([A-Z]([a-z]{2}))+)(?P&lt;extra&gt;(|=|\\?)(|fs)))$&#39;)]</span>

        
    </div>
    <a class="headerlink" href="#HGVSRegex.PEP_ALLELE_REGEXES"></a>
    
    

                            </div>
                            <div id="HGVSRegex.GENOMIC_ALLELE" class="classattr">
                                <div class="attr variable">
            <span class="name">GENOMIC_ALLELE</span>        =
<input id="HGVSRegex.GENOMIC_ALLELE-view-value" class="view-value-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
            <label class="view-value-button pdoc-button" for="HGVSRegex.GENOMIC_ALLELE-view-value"></label><span class="default_value">[&#39;(?P&lt;start&gt;\\d+)(?P&lt;mutation_type&gt;=)&#39;, &#39;(?P&lt;start&gt;\\d+)(?P&lt;ref&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+)(?P&lt;mutation_type&gt;=)&#39;, &#39;(?P&lt;start&gt;\\d+)(?P&lt;ref&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+)(?P&lt;mutation_type&gt;&gt;)(?P&lt;alt&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+)&#39;, &#39;(?P&lt;start&gt;\\d+)(?P&lt;mutation_type&gt;ins)(?P&lt;alt&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+)&#39;, &#39;(?P&lt;start&gt;\\d+)(?P&lt;mutation_type&gt;del)(?P&lt;ref&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+)&#39;, &#39;(?P&lt;start&gt;\\d+)(?P&lt;mutation_type&gt;dup)(?P&lt;ref&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+)&#39;, &#39;(?P&lt;start&gt;\\d+)(?P&lt;mutation_type&gt;del)&#39;, &#39;(?P&lt;start&gt;\\d+)(?P&lt;mutation_type&gt;dup)&#39;, &#39;(?P&lt;start&gt;\\d+)_(?P&lt;end&gt;\\d+)(?P&lt;mutation_type&gt;=)&#39;, &#39;(?P&lt;start&gt;\\d+)_(?P&lt;end&gt;\\d+)(?P&lt;mutation_type&gt;ins)(?P&lt;alt&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+)&#39;, &#39;(?P&lt;start&gt;\\d+)_(?P&lt;end&gt;\\d+)(?P&lt;mutation_type&gt;del)(?P&lt;ref&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+)&#39;, &#39;(?P&lt;start&gt;\\d+)_(?P&lt;end&gt;\\d+)(?P&lt;mutation_type&gt;dup)(?P&lt;ref&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+)&#39;, &#39;(?P&lt;start&gt;\\d+)_(?P&lt;end&gt;\\d+)(?P&lt;mutation_type&gt;del)&#39;, &#39;(?P&lt;start&gt;\\d+)_(?P&lt;end&gt;\\d+)(?P&lt;mutation_type&gt;dup)&#39;, &#39;(?P&lt;delins&gt;(?P&lt;start&gt;\\d+)del(?P&lt;ref&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+)ins(?P&lt;alt&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+))&#39;, &#39;(?P&lt;delins&gt;(?P&lt;start&gt;\\d+)_(?P&lt;end&gt;\\d+)del(?P&lt;ref&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+)ins(?P&lt;alt&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+))&#39;, &#39;(?P&lt;delins&gt;(?P&lt;start&gt;\\d+)delins(?P&lt;alt&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+))&#39;, &#39;(?P&lt;delins&gt;(?P&lt;start&gt;\\d+)_(?P&lt;end&gt;\\d+)delins(?P&lt;alt&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+))&#39;]</span>

        
    </div>
    <a class="headerlink" href="#HGVSRegex.GENOMIC_ALLELE"></a>
    
    

                            </div>
                            <div id="HGVSRegex.GENOMIC_ALLELE_REGEXES" class="classattr">
                                <div class="attr variable">
            <span class="name">GENOMIC_ALLELE_REGEXES</span>        =
<input id="HGVSRegex.GENOMIC_ALLELE_REGEXES-view-value" class="view-value-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
            <label class="view-value-button pdoc-button" for="HGVSRegex.GENOMIC_ALLELE_REGEXES-view-value"></label><span class="default_value">[re.compile(&#39;^(?P&lt;start&gt;\\d+)(?P&lt;mutation_type&gt;=)$&#39;), re.compile(&#39;^(?P&lt;start&gt;\\d+)(?P&lt;ref&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+)(?P&lt;mutation_type&gt;=)$&#39;), re.compile(&#39;^(?P&lt;start&gt;\\d+)(?P&lt;ref&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+)(?P&lt;mutation_type&gt;&gt;)(?P&lt;alt&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+)$&#39;), re.compile(&#39;^(?P&lt;start&gt;\\d+)(?P&lt;mutation_type&gt;ins)(?P&lt;alt&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+)$&#39;), re.compile(&#39;^(?P&lt;start&gt;\\d+)(?P&lt;mutation_type&gt;del)(?P&lt;ref&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+)$&#39;), re.compile(&#39;^(?P&lt;start&gt;\\d+)(?P&lt;mutation_type&gt;dup)(?P&lt;ref&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+)$&#39;), re.compile(&#39;^(?P&lt;start&gt;\\d+)(?P&lt;mutation_type&gt;del)$&#39;), re.compile(&#39;^(?P&lt;start&gt;\\d+)(?P&lt;mutation_type&gt;dup)$&#39;), re.compile(&#39;^(?P&lt;start&gt;\\d+)_(?P&lt;end&gt;\\d+)(?P&lt;mutation_type&gt;=)$&#39;), re.compile(&#39;^(?P&lt;start&gt;\\d+)_(?P&lt;end&gt;\\d+)(?P&lt;mutation_type&gt;ins)(?P&lt;alt&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+)$&#39;), re.compile(&#39;^(?P&lt;start&gt;\\d+)_(?P&lt;end&gt;\\d+)(?P&lt;mutation_type&gt;del)(?P&lt;ref&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+)$&#39;), re.compile(&#39;^(?P&lt;start&gt;\\d+)_(?P&lt;end&gt;\\d+)(?P&lt;mutation_type&gt;dup)(?P&lt;ref&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+)$&#39;), re.compile(&#39;^(?P&lt;start&gt;\\d+)_(?P&lt;end&gt;\\d+)(?P&lt;mutation_type&gt;del)$&#39;), re.compile(&#39;^(?P&lt;start&gt;\\d+)_(?P&lt;end&gt;\\d+)(?P&lt;mutation_type&gt;dup)$&#39;), re.compile(&#39;^(?P&lt;delins&gt;(?P&lt;start&gt;\\d+)del(?P&lt;ref&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+)ins(?P&lt;alt&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+))$&#39;), re.compile(&#39;^(?P&lt;delins&gt;(?P&lt;start&gt;\\d+)_(?P&lt;end&gt;\\d+)del(?P&lt;ref&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+)ins(?P&lt;alt&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+))$&#39;), re.compile(&#39;^(?P&lt;delins&gt;(?P&lt;start&gt;\\d+)delins(?P&lt;alt&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+))$&#39;), re.compile(&#39;^(?P&lt;delins&gt;(?P&lt;start&gt;\\d+)_(?P&lt;end&gt;\\d+)delins(?P&lt;alt&gt;[acgtbdhkmnrsvwyACGTBDHKMNRSVWY]+|\\d+))$&#39;)]</span>

        
    </div>
    <a class="headerlink" href="#HGVSRegex.GENOMIC_ALLELE_REGEXES"></a>
    
    

                            </div>
                </section>
                <section id="REFSEQ_PREFIXES">
                    <div class="attr variable">
            <span class="name">REFSEQ_PREFIXES</span>        =
<input id="REFSEQ_PREFIXES-view-value" class="view-value-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
            <label class="view-value-button pdoc-button" for="REFSEQ_PREFIXES-view-value"></label><span class="default_value">[(&#39;AC_&#39;, &#39;genomic&#39;, &#39;Complete genomic molecule, usually alternate assembly&#39;), (&#39;NC_&#39;, &#39;genomic&#39;, &#39;Complete genomic molecule, usually reference assembly&#39;), (&#39;NG_&#39;, &#39;genomic&#39;, &#39;Incomplete genomic region&#39;), (&#39;NT_&#39;, &#39;genomic&#39;, &#39;Contig or scaffold, clone-based or WGS&#39;), (&#39;NW_&#39;, &#39;genomic&#39;, &#39;Contig or scaffold, primarily WGS&#39;), (&#39;NS_&#39;, &#39;genomic&#39;, &#39;Environmental sequence&#39;), (&#39;NZ_&#39;, &#39;genomic&#39;, &#39;Unfinished WGS&#39;), (&#39;NM_&#39;, &#39;mRNA&#39;, &#39;&#39;), (&#39;NR_&#39;, &#39;RNA&#39;, &#39;&#39;), (&#39;XM_&#39;, &#39;mRNA&#39;, &#39;Predicted model&#39;), (&#39;XR_&#39;, &#39;RNA&#39;, &#39;Predicted model&#39;), (&#39;AP_&#39;, &#39;Protein&#39;, &#39;Annotated on AC_ alternate assembly&#39;), (&#39;NP_&#39;, &#39;Protein&#39;, &#39;Associated with an NM_ or NC_ accession&#39;), (&#39;YP_&#39;, &#39;Protein&#39;, &#39;&#39;), (&#39;XP_&#39;, &#39;Protein&#39;, &#39;Predicted model, associated with an XM_ accession&#39;), (&#39;ZP_&#39;, &#39;Protein&#39;, &#39;Predicted model, annotated on NZ_ genomic records&#39;)]</span>

        
    </div>
    <a class="headerlink" href="#REFSEQ_PREFIXES"></a>
    
    

                </section>
                <section id="REFSEQ_PREFIX_LOOKUP">
                    <div class="attr variable">
            <span class="name">REFSEQ_PREFIX_LOOKUP</span>        =
<input id="REFSEQ_PREFIX_LOOKUP-view-value" class="view-value-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
            <label class="view-value-button pdoc-button" for="REFSEQ_PREFIX_LOOKUP-view-value"></label><span class="default_value">{&#39;AC_&#39;: (&#39;genomic&#39;, &#39;Complete genomic molecule, usually alternate assembly&#39;), &#39;NC_&#39;: (&#39;genomic&#39;, &#39;Complete genomic molecule, usually reference assembly&#39;), &#39;NG_&#39;: (&#39;genomic&#39;, &#39;Incomplete genomic region&#39;), &#39;NT_&#39;: (&#39;genomic&#39;, &#39;Contig or scaffold, clone-based or WGS&#39;), &#39;NW_&#39;: (&#39;genomic&#39;, &#39;Contig or scaffold, primarily WGS&#39;), &#39;NS_&#39;: (&#39;genomic&#39;, &#39;Environmental sequence&#39;), &#39;NZ_&#39;: (&#39;genomic&#39;, &#39;Unfinished WGS&#39;), &#39;NM_&#39;: (&#39;mRNA&#39;, &#39;&#39;), &#39;NR_&#39;: (&#39;RNA&#39;, &#39;&#39;), &#39;XM_&#39;: (&#39;mRNA&#39;, &#39;Predicted model&#39;), &#39;XR_&#39;: (&#39;RNA&#39;, &#39;Predicted model&#39;), &#39;AP_&#39;: (&#39;Protein&#39;, &#39;Annotated on AC_ alternate assembly&#39;), &#39;NP_&#39;: (&#39;Protein&#39;, &#39;Associated with an NM_ or NC_ accession&#39;), &#39;YP_&#39;: (&#39;Protein&#39;, &#39;&#39;), &#39;XP_&#39;: (&#39;Protein&#39;, &#39;Predicted model, associated with an XM_ accession&#39;), &#39;ZP_&#39;: (&#39;Protein&#39;, &#39;Predicted model, annotated on NZ_ genomic records&#39;)}</span>

        
    </div>
    <a class="headerlink" href="#REFSEQ_PREFIX_LOOKUP"></a>
    
    

                </section>
                <section id="get_refseq_type">
                            <input id="get_refseq_type-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr function">
            
        <span class="def">def</span>
        <span class="name">get_refseq_type</span><span class="signature pdoc-code condensed">(<span class="param"><span class="n">name</span><span class="p">:</span> <span class="nb">str</span></span><span class="return-annotation">) -> <span class="nb">str</span>:</span></span>

                <label class="view-source-button" for="get_refseq_type-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#get_refseq_type"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="get_refseq_type-469"><a href="#get_refseq_type-469"><span class="linenos">469</span></a><span class="k">def</span> <span class="nf">get_refseq_type</span><span class="p">(</span><span class="n">name</span><span class="p">:</span> <span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
</span><span id="get_refseq_type-470"><a href="#get_refseq_type-470"><span class="linenos">470</span></a><span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="get_refseq_type-471"><a href="#get_refseq_type-471"><span class="linenos">471</span></a><span class="sd">    The `get_refseq_type` function returns the RefSeq type for a given RefSeq name.</span>
</span><span id="get_refseq_type-472"><a href="#get_refseq_type-472"><span class="linenos">472</span></a>
</span><span id="get_refseq_type-473"><a href="#get_refseq_type-473"><span class="linenos">473</span></a><span class="sd">    :param name: The `name` parameter is a string representing a RefSeq name</span>
</span><span id="get_refseq_type-474"><a href="#get_refseq_type-474"><span class="linenos">474</span></a><span class="sd">    :type name: str</span>
</span><span id="get_refseq_type-475"><a href="#get_refseq_type-475"><span class="linenos">475</span></a><span class="sd">    :return: The function `get_refseq_type` returns the RefSeq type for a given RefSeq name.</span>
</span><span id="get_refseq_type-476"><a href="#get_refseq_type-476"><span class="linenos">476</span></a><span class="sd">    &quot;&quot;&quot;</span>
</span><span id="get_refseq_type-477"><a href="#get_refseq_type-477"><span class="linenos">477</span></a>
</span><span id="get_refseq_type-478"><a href="#get_refseq_type-478"><span class="linenos">478</span></a>    <span class="n">prefix</span> <span class="o">=</span> <span class="n">name</span><span class="p">[:</span><span class="mi">3</span><span class="p">]</span>
</span><span id="get_refseq_type-479"><a href="#get_refseq_type-479"><span class="linenos">479</span></a>    <span class="k">return</span> <span class="n">REFSEQ_PREFIX_LOOKUP</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="n">prefix</span><span class="p">,</span> <span class="p">(</span><span class="kc">None</span><span class="p">,</span> <span class="s2">&quot;&quot;</span><span class="p">))[</span><span class="mi">0</span><span class="p">]</span>
</span></pre></div>


            <div class="docstring"><p>The <code><a href="#get_refseq_type">get_refseq_type</a></code> function returns the RefSeq type for a given RefSeq name.</p>

<h6 id="parameters">Parameters</h6>

<ul>
<li><strong>name</strong>:  The <code>name</code> parameter is a string representing a RefSeq name</li>
</ul>

<h6 id="returns">Returns</h6>

<blockquote>
  <p>The function <code><a href="#get_refseq_type">get_refseq_type</a></code> returns the RefSeq type for a given RefSeq name.</p>
</blockquote>
</div>


                </section>
                <section id="InvalidHGVSName">
                            <input id="InvalidHGVSName-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr class">
            
    <span class="def">class</span>
    <span class="name">InvalidHGVSName</span><wbr>(<span class="base">builtins.ValueError</span>):

                <label class="view-source-button" for="InvalidHGVSName-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#InvalidHGVSName"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="InvalidHGVSName-484"><a href="#InvalidHGVSName-484"><span class="linenos">484</span></a><span class="k">class</span> <span class="nc">InvalidHGVSName</span><span class="p">(</span><span class="ne">ValueError</span><span class="p">):</span>
</span><span id="InvalidHGVSName-485"><a href="#InvalidHGVSName-485"><span class="linenos">485</span></a>
</span><span id="InvalidHGVSName-486"><a href="#InvalidHGVSName-486"><span class="linenos">486</span></a>    <span class="k">def</span> <span class="fm">__init__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">name</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span><span class="p">,</span> <span class="n">part</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;name&quot;</span><span class="p">,</span> <span class="n">reason</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
</span><span id="InvalidHGVSName-487"><a href="#InvalidHGVSName-487"><span class="linenos">487</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="InvalidHGVSName-488"><a href="#InvalidHGVSName-488"><span class="linenos">488</span></a><span class="sd">        The function initializes an InvalidHGVSName object with a message, name, part, and reason.</span>
</span><span id="InvalidHGVSName-489"><a href="#InvalidHGVSName-489"><span class="linenos">489</span></a>
</span><span id="InvalidHGVSName-490"><a href="#InvalidHGVSName-490"><span class="linenos">490</span></a><span class="sd">        :param name: The name parameter is a string that represents the invalid HGVS name. It is the</span>
</span><span id="InvalidHGVSName-491"><a href="#InvalidHGVSName-491"><span class="linenos">491</span></a><span class="sd">        name that is considered invalid and does not meet the required criteria</span>
</span><span id="InvalidHGVSName-492"><a href="#InvalidHGVSName-492"><span class="linenos">492</span></a><span class="sd">        :type name: str</span>
</span><span id="InvalidHGVSName-493"><a href="#InvalidHGVSName-493"><span class="linenos">493</span></a><span class="sd">        :param part: The &quot;part&quot; parameter represents the part of the HGVS (Human Genome Variation</span>
</span><span id="InvalidHGVSName-494"><a href="#InvalidHGVSName-494"><span class="linenos">494</span></a><span class="sd">        Society) name that is invalid. It is used to provide more specific information about the error</span>
</span><span id="InvalidHGVSName-495"><a href="#InvalidHGVSName-495"><span class="linenos">495</span></a><span class="sd">        that occurred, defaults to name</span>
</span><span id="InvalidHGVSName-496"><a href="#InvalidHGVSName-496"><span class="linenos">496</span></a><span class="sd">        :type part: str (optional)</span>
</span><span id="InvalidHGVSName-497"><a href="#InvalidHGVSName-497"><span class="linenos">497</span></a><span class="sd">        :param reason: The &quot;reason&quot; parameter is an optional argument that provides additional</span>
</span><span id="InvalidHGVSName-498"><a href="#InvalidHGVSName-498"><span class="linenos">498</span></a><span class="sd">        information or context for why the HGVS name is considered invalid. It can be used to provide</span>
</span><span id="InvalidHGVSName-499"><a href="#InvalidHGVSName-499"><span class="linenos">499</span></a><span class="sd">        specific details about the error or to explain why the name does not meet the required criteria</span>
</span><span id="InvalidHGVSName-500"><a href="#InvalidHGVSName-500"><span class="linenos">500</span></a><span class="sd">        :type reason: str</span>
</span><span id="InvalidHGVSName-501"><a href="#InvalidHGVSName-501"><span class="linenos">501</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="InvalidHGVSName-502"><a href="#InvalidHGVSName-502"><span class="linenos">502</span></a>
</span><span id="InvalidHGVSName-503"><a href="#InvalidHGVSName-503"><span class="linenos">503</span></a>        <span class="k">if</span> <span class="n">name</span><span class="p">:</span>
</span><span id="InvalidHGVSName-504"><a href="#InvalidHGVSName-504"><span class="linenos">504</span></a>            <span class="n">message</span> <span class="o">=</span> <span class="s1">&#39;Invalid HGVS </span><span class="si">%s</span><span class="s1"> &quot;</span><span class="si">%s</span><span class="s1">&quot;&#39;</span> <span class="o">%</span> <span class="p">(</span><span class="n">part</span><span class="p">,</span> <span class="n">name</span><span class="p">)</span>
</span><span id="InvalidHGVSName-505"><a href="#InvalidHGVSName-505"><span class="linenos">505</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="InvalidHGVSName-506"><a href="#InvalidHGVSName-506"><span class="linenos">506</span></a>            <span class="n">message</span> <span class="o">=</span> <span class="s2">&quot;Invalid HGVS </span><span class="si">%s</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="n">part</span>
</span><span id="InvalidHGVSName-507"><a href="#InvalidHGVSName-507"><span class="linenos">507</span></a>        <span class="k">if</span> <span class="n">reason</span><span class="p">:</span>
</span><span id="InvalidHGVSName-508"><a href="#InvalidHGVSName-508"><span class="linenos">508</span></a>            <span class="n">message</span> <span class="o">+=</span> <span class="s2">&quot;: &quot;</span> <span class="o">+</span> <span class="n">reason</span>
</span><span id="InvalidHGVSName-509"><a href="#InvalidHGVSName-509"><span class="linenos">509</span></a>        <span class="nb">super</span><span class="p">(</span><span class="n">InvalidHGVSName</span><span class="p">,</span> <span class="bp">self</span><span class="p">)</span><span class="o">.</span><span class="fm">__init__</span><span class="p">(</span><span class="n">message</span><span class="p">)</span>
</span><span id="InvalidHGVSName-510"><a href="#InvalidHGVSName-510"><span class="linenos">510</span></a>
</span><span id="InvalidHGVSName-511"><a href="#InvalidHGVSName-511"><span class="linenos">511</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">name</span> <span class="o">=</span> <span class="n">name</span>
</span><span id="InvalidHGVSName-512"><a href="#InvalidHGVSName-512"><span class="linenos">512</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">part</span> <span class="o">=</span> <span class="n">part</span>
</span><span id="InvalidHGVSName-513"><a href="#InvalidHGVSName-513"><span class="linenos">513</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">reason</span> <span class="o">=</span> <span class="n">reason</span>
</span></pre></div>


            <div class="docstring"><p>Inappropriate argument value (of correct type).</p>
</div>


                            <div id="InvalidHGVSName.__init__" class="classattr">
                                        <input id="InvalidHGVSName.__init__-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr function">
            
        <span class="name">InvalidHGVSName</span><span class="signature pdoc-code condensed">(<span class="param"><span class="n">name</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s1">&#39;&#39;</span>, </span><span class="param"><span class="n">part</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s1">&#39;name&#39;</span>, </span><span class="param"><span class="n">reason</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s1">&#39;&#39;</span></span>)</span>

                <label class="view-source-button" for="InvalidHGVSName.__init__-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#InvalidHGVSName.__init__"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="InvalidHGVSName.__init__-486"><a href="#InvalidHGVSName.__init__-486"><span class="linenos">486</span></a>    <span class="k">def</span> <span class="fm">__init__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">name</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span><span class="p">,</span> <span class="n">part</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;name&quot;</span><span class="p">,</span> <span class="n">reason</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
</span><span id="InvalidHGVSName.__init__-487"><a href="#InvalidHGVSName.__init__-487"><span class="linenos">487</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="InvalidHGVSName.__init__-488"><a href="#InvalidHGVSName.__init__-488"><span class="linenos">488</span></a><span class="sd">        The function initializes an InvalidHGVSName object with a message, name, part, and reason.</span>
</span><span id="InvalidHGVSName.__init__-489"><a href="#InvalidHGVSName.__init__-489"><span class="linenos">489</span></a>
</span><span id="InvalidHGVSName.__init__-490"><a href="#InvalidHGVSName.__init__-490"><span class="linenos">490</span></a><span class="sd">        :param name: The name parameter is a string that represents the invalid HGVS name. It is the</span>
</span><span id="InvalidHGVSName.__init__-491"><a href="#InvalidHGVSName.__init__-491"><span class="linenos">491</span></a><span class="sd">        name that is considered invalid and does not meet the required criteria</span>
</span><span id="InvalidHGVSName.__init__-492"><a href="#InvalidHGVSName.__init__-492"><span class="linenos">492</span></a><span class="sd">        :type name: str</span>
</span><span id="InvalidHGVSName.__init__-493"><a href="#InvalidHGVSName.__init__-493"><span class="linenos">493</span></a><span class="sd">        :param part: The &quot;part&quot; parameter represents the part of the HGVS (Human Genome Variation</span>
</span><span id="InvalidHGVSName.__init__-494"><a href="#InvalidHGVSName.__init__-494"><span class="linenos">494</span></a><span class="sd">        Society) name that is invalid. It is used to provide more specific information about the error</span>
</span><span id="InvalidHGVSName.__init__-495"><a href="#InvalidHGVSName.__init__-495"><span class="linenos">495</span></a><span class="sd">        that occurred, defaults to name</span>
</span><span id="InvalidHGVSName.__init__-496"><a href="#InvalidHGVSName.__init__-496"><span class="linenos">496</span></a><span class="sd">        :type part: str (optional)</span>
</span><span id="InvalidHGVSName.__init__-497"><a href="#InvalidHGVSName.__init__-497"><span class="linenos">497</span></a><span class="sd">        :param reason: The &quot;reason&quot; parameter is an optional argument that provides additional</span>
</span><span id="InvalidHGVSName.__init__-498"><a href="#InvalidHGVSName.__init__-498"><span class="linenos">498</span></a><span class="sd">        information or context for why the HGVS name is considered invalid. It can be used to provide</span>
</span><span id="InvalidHGVSName.__init__-499"><a href="#InvalidHGVSName.__init__-499"><span class="linenos">499</span></a><span class="sd">        specific details about the error or to explain why the name does not meet the required criteria</span>
</span><span id="InvalidHGVSName.__init__-500"><a href="#InvalidHGVSName.__init__-500"><span class="linenos">500</span></a><span class="sd">        :type reason: str</span>
</span><span id="InvalidHGVSName.__init__-501"><a href="#InvalidHGVSName.__init__-501"><span class="linenos">501</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="InvalidHGVSName.__init__-502"><a href="#InvalidHGVSName.__init__-502"><span class="linenos">502</span></a>
</span><span id="InvalidHGVSName.__init__-503"><a href="#InvalidHGVSName.__init__-503"><span class="linenos">503</span></a>        <span class="k">if</span> <span class="n">name</span><span class="p">:</span>
</span><span id="InvalidHGVSName.__init__-504"><a href="#InvalidHGVSName.__init__-504"><span class="linenos">504</span></a>            <span class="n">message</span> <span class="o">=</span> <span class="s1">&#39;Invalid HGVS </span><span class="si">%s</span><span class="s1"> &quot;</span><span class="si">%s</span><span class="s1">&quot;&#39;</span> <span class="o">%</span> <span class="p">(</span><span class="n">part</span><span class="p">,</span> <span class="n">name</span><span class="p">)</span>
</span><span id="InvalidHGVSName.__init__-505"><a href="#InvalidHGVSName.__init__-505"><span class="linenos">505</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="InvalidHGVSName.__init__-506"><a href="#InvalidHGVSName.__init__-506"><span class="linenos">506</span></a>            <span class="n">message</span> <span class="o">=</span> <span class="s2">&quot;Invalid HGVS </span><span class="si">%s</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="n">part</span>
</span><span id="InvalidHGVSName.__init__-507"><a href="#InvalidHGVSName.__init__-507"><span class="linenos">507</span></a>        <span class="k">if</span> <span class="n">reason</span><span class="p">:</span>
</span><span id="InvalidHGVSName.__init__-508"><a href="#InvalidHGVSName.__init__-508"><span class="linenos">508</span></a>            <span class="n">message</span> <span class="o">+=</span> <span class="s2">&quot;: &quot;</span> <span class="o">+</span> <span class="n">reason</span>
</span><span id="InvalidHGVSName.__init__-509"><a href="#InvalidHGVSName.__init__-509"><span class="linenos">509</span></a>        <span class="nb">super</span><span class="p">(</span><span class="n">InvalidHGVSName</span><span class="p">,</span> <span class="bp">self</span><span class="p">)</span><span class="o">.</span><span class="fm">__init__</span><span class="p">(</span><span class="n">message</span><span class="p">)</span>
</span><span id="InvalidHGVSName.__init__-510"><a href="#InvalidHGVSName.__init__-510"><span class="linenos">510</span></a>
</span><span id="InvalidHGVSName.__init__-511"><a href="#InvalidHGVSName.__init__-511"><span class="linenos">511</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">name</span> <span class="o">=</span> <span class="n">name</span>
</span><span id="InvalidHGVSName.__init__-512"><a href="#InvalidHGVSName.__init__-512"><span class="linenos">512</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">part</span> <span class="o">=</span> <span class="n">part</span>
</span><span id="InvalidHGVSName.__init__-513"><a href="#InvalidHGVSName.__init__-513"><span class="linenos">513</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">reason</span> <span class="o">=</span> <span class="n">reason</span>
</span></pre></div>


            <div class="docstring"><p>The function initializes an InvalidHGVSName object with a message, name, part, and reason.</p>

<h6 id="parameters">Parameters</h6>

<ul>
<li><strong>name</strong>:  The name parameter is a string that represents the invalid HGVS name. It is the
name that is considered invalid and does not meet the required criteria</li>
<li><strong>part</strong>:  The "part" parameter represents the part of the HGVS (Human Genome Variation
Society) name that is invalid. It is used to provide more specific information about the error
that occurred, defaults to name</li>
<li><strong>reason</strong>:  The "reason" parameter is an optional argument that provides additional
information or context for why the HGVS name is considered invalid. It can be used to provide
specific details about the error or to explain why the name does not meet the required criteria</li>
</ul>
</div>


                            </div>
                            <div id="InvalidHGVSName.name" class="classattr">
                                <div class="attr variable">
            <span class="name">name</span>

        
    </div>
    <a class="headerlink" href="#InvalidHGVSName.name"></a>
    
    

                            </div>
                            <div id="InvalidHGVSName.part" class="classattr">
                                <div class="attr variable">
            <span class="name">part</span>

        
    </div>
    <a class="headerlink" href="#InvalidHGVSName.part"></a>
    
    

                            </div>
                            <div id="InvalidHGVSName.reason" class="classattr">
                                <div class="attr variable">
            <span class="name">reason</span>

        
    </div>
    <a class="headerlink" href="#InvalidHGVSName.reason"></a>
    
    

                            </div>
                            <div class="inherited">
                                <h5>Inherited Members</h5>
                                <dl>
                                    <div><dt>builtins.BaseException</dt>
                                <dd id="InvalidHGVSName.with_traceback" class="function">with_traceback</dd>
                <dd id="InvalidHGVSName.args" class="variable">args</dd>

            </div>
                                </dl>
                            </div>
                </section>
                <section id="HGVSName">
                            <input id="HGVSName-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr class">
            
    <span class="def">class</span>
    <span class="name">HGVSName</span>:

                <label class="view-source-button" for="HGVSName-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#HGVSName"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="HGVSName-517"><a href="#HGVSName-517"><span class="linenos"> 517</span></a><span class="k">class</span> <span class="nc">HGVSName</span><span class="p">(</span><span class="nb">object</span><span class="p">):</span>
</span><span id="HGVSName-518"><a href="#HGVSName-518"><span class="linenos"> 518</span></a><span class="w">    </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="HGVSName-519"><a href="#HGVSName-519"><span class="linenos"> 519</span></a><span class="sd">    Represents a HGVS variant name.</span>
</span><span id="HGVSName-520"><a href="#HGVSName-520"><span class="linenos"> 520</span></a><span class="sd">    &quot;&quot;&quot;</span>
</span><span id="HGVSName-521"><a href="#HGVSName-521"><span class="linenos"> 521</span></a>
</span><span id="HGVSName-522"><a href="#HGVSName-522"><span class="linenos"> 522</span></a>    <span class="k">def</span> <span class="fm">__init__</span><span class="p">(</span>
</span><span id="HGVSName-523"><a href="#HGVSName-523"><span class="linenos"> 523</span></a>        <span class="bp">self</span><span class="p">,</span>
</span><span id="HGVSName-524"><a href="#HGVSName-524"><span class="linenos"> 524</span></a>        <span class="n">name</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span><span class="p">,</span>
</span><span id="HGVSName-525"><a href="#HGVSName-525"><span class="linenos"> 525</span></a>        <span class="n">prefix</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span><span class="p">,</span>
</span><span id="HGVSName-526"><a href="#HGVSName-526"><span class="linenos"> 526</span></a>        <span class="n">chrom</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span><span class="p">,</span>
</span><span id="HGVSName-527"><a href="#HGVSName-527"><span class="linenos"> 527</span></a>        <span class="n">transcript</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span><span class="p">,</span>
</span><span id="HGVSName-528"><a href="#HGVSName-528"><span class="linenos"> 528</span></a>        <span class="n">transcript_protein</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
</span><span id="HGVSName-529"><a href="#HGVSName-529"><span class="linenos"> 529</span></a>        <span class="n">gene</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span><span class="p">,</span>
</span><span id="HGVSName-530"><a href="#HGVSName-530"><span class="linenos"> 530</span></a>        <span class="n">exon</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
</span><span id="HGVSName-531"><a href="#HGVSName-531"><span class="linenos"> 531</span></a>        <span class="n">kind</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span><span class="p">,</span>
</span><span id="HGVSName-532"><a href="#HGVSName-532"><span class="linenos"> 532</span></a>        <span class="n">mutation_type</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
</span><span id="HGVSName-533"><a href="#HGVSName-533"><span class="linenos"> 533</span></a>        <span class="n">start</span><span class="p">:</span> <span class="nb">int</span> <span class="o">=</span> <span class="mi">0</span><span class="p">,</span>
</span><span id="HGVSName-534"><a href="#HGVSName-534"><span class="linenos"> 534</span></a>        <span class="n">end</span><span class="p">:</span> <span class="nb">int</span> <span class="o">=</span> <span class="mi">0</span><span class="p">,</span>
</span><span id="HGVSName-535"><a href="#HGVSName-535"><span class="linenos"> 535</span></a>        <span class="n">ref_allele</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span><span class="p">,</span>
</span><span id="HGVSName-536"><a href="#HGVSName-536"><span class="linenos"> 536</span></a>        <span class="n">ref2_allele</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span><span class="p">,</span>
</span><span id="HGVSName-537"><a href="#HGVSName-537"><span class="linenos"> 537</span></a>        <span class="n">alt_allele</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span><span class="p">,</span>
</span><span id="HGVSName-538"><a href="#HGVSName-538"><span class="linenos"> 538</span></a>        <span class="n">cdna_start</span><span class="p">:</span> <span class="nb">int</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
</span><span id="HGVSName-539"><a href="#HGVSName-539"><span class="linenos"> 539</span></a>        <span class="n">cdna_end</span><span class="p">:</span> <span class="nb">int</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
</span><span id="HGVSName-540"><a href="#HGVSName-540"><span class="linenos"> 540</span></a>        <span class="n">pep_extra</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span><span class="p">,</span>
</span><span id="HGVSName-541"><a href="#HGVSName-541"><span class="linenos"> 541</span></a>    <span class="p">):</span>
</span><span id="HGVSName-542"><a href="#HGVSName-542"><span class="linenos"> 542</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="HGVSName-543"><a href="#HGVSName-543"><span class="linenos"> 543</span></a><span class="sd">        The function is a constructor that initializes various attributes of an object and parses a</span>
</span><span id="HGVSName-544"><a href="#HGVSName-544"><span class="linenos"> 544</span></a><span class="sd">        given name to populate those attributes.</span>
</span><span id="HGVSName-545"><a href="#HGVSName-545"><span class="linenos"> 545</span></a>
</span><span id="HGVSName-546"><a href="#HGVSName-546"><span class="linenos"> 546</span></a><span class="sd">        :param name: The full HGVS name of the variant</span>
</span><span id="HGVSName-547"><a href="#HGVSName-547"><span class="linenos"> 547</span></a><span class="sd">        :type name: str</span>
</span><span id="HGVSName-548"><a href="#HGVSName-548"><span class="linenos"> 548</span></a><span class="sd">        :param prefix: The `prefix` parameter is a string that is used as a prefix for the HGVS name. It</span>
</span><span id="HGVSName-549"><a href="#HGVSName-549"><span class="linenos"> 549</span></a><span class="sd">        can be used to indicate additional information or context about the variant</span>
</span><span id="HGVSName-550"><a href="#HGVSName-550"><span class="linenos"> 550</span></a><span class="sd">        :type prefix: str</span>
</span><span id="HGVSName-551"><a href="#HGVSName-551"><span class="linenos"> 551</span></a><span class="sd">        :param chrom: The `chrom` parameter represents the chromosome where the mutation occurs. It is a</span>
</span><span id="HGVSName-552"><a href="#HGVSName-552"><span class="linenos"> 552</span></a><span class="sd">        string that specifies the chromosome number or identifier</span>
</span><span id="HGVSName-553"><a href="#HGVSName-553"><span class="linenos"> 553</span></a><span class="sd">        :type chrom: str</span>
</span><span id="HGVSName-554"><a href="#HGVSName-554"><span class="linenos"> 554</span></a><span class="sd">        :param transcript: The `transcript` parameter represents the transcript ID or name associated</span>
</span><span id="HGVSName-555"><a href="#HGVSName-555"><span class="linenos"> 555</span></a><span class="sd">        with the mutation. It is used to specify the specific transcript in which the mutation occurs</span>
</span><span id="HGVSName-556"><a href="#HGVSName-556"><span class="linenos"> 556</span></a><span class="sd">        :type transcript: str</span>
</span><span id="HGVSName-557"><a href="#HGVSName-557"><span class="linenos"> 557</span></a><span class="sd">        :param transcript_protein: The `transcript_protein` parameter is used to store information about</span>
</span><span id="HGVSName-558"><a href="#HGVSName-558"><span class="linenos"> 558</span></a><span class="sd">        the protein associated with the transcript. It can be used to specify the protein variant or</span>
</span><span id="HGVSName-559"><a href="#HGVSName-559"><span class="linenos"> 559</span></a><span class="sd">        isoform that is affected by the mutation</span>
</span><span id="HGVSName-560"><a href="#HGVSName-560"><span class="linenos"> 560</span></a><span class="sd">        :type transcript_protein: str</span>
</span><span id="HGVSName-561"><a href="#HGVSName-561"><span class="linenos"> 561</span></a><span class="sd">        :param gene: The &quot;gene&quot; parameter represents the gene associated with the variant. It is a</span>
</span><span id="HGVSName-562"><a href="#HGVSName-562"><span class="linenos"> 562</span></a><span class="sd">        string that specifies the gene name or identifier</span>
</span><span id="HGVSName-563"><a href="#HGVSName-563"><span class="linenos"> 563</span></a><span class="sd">        :type gene: str</span>
</span><span id="HGVSName-564"><a href="#HGVSName-564"><span class="linenos"> 564</span></a><span class="sd">        :param exon: The `exon` parameter represents the exon number or range in which the mutation</span>
</span><span id="HGVSName-565"><a href="#HGVSName-565"><span class="linenos"> 565</span></a><span class="sd">        occurs. It is used to specify the location of the mutation within the transcript</span>
</span><span id="HGVSName-566"><a href="#HGVSName-566"><span class="linenos"> 566</span></a><span class="sd">        :type exon: str</span>
</span><span id="HGVSName-567"><a href="#HGVSName-567"><span class="linenos"> 567</span></a><span class="sd">        :param kind: The &quot;kind&quot; parameter is used to specify the type of variant or mutation. It can be</span>
</span><span id="HGVSName-568"><a href="#HGVSName-568"><span class="linenos"> 568</span></a><span class="sd">        a string that represents the kind of mutation, such as &quot;substitution&quot;, &quot;deletion&quot;, &quot;insertion&quot;,</span>
</span><span id="HGVSName-569"><a href="#HGVSName-569"><span class="linenos"> 569</span></a><span class="sd">        etc. This parameter helps to categorize and describe the type of mutation being represented by</span>
</span><span id="HGVSName-570"><a href="#HGVSName-570"><span class="linenos"> 570</span></a><span class="sd">        the</span>
</span><span id="HGVSName-571"><a href="#HGVSName-571"><span class="linenos"> 571</span></a><span class="sd">        :type kind: str</span>
</span><span id="HGVSName-572"><a href="#HGVSName-572"><span class="linenos"> 572</span></a><span class="sd">        :param mutation_type: The `mutation_type` parameter is used to specify the type of mutation. It</span>
</span><span id="HGVSName-573"><a href="#HGVSName-573"><span class="linenos"> 573</span></a><span class="sd">        can be a string that represents the type of mutation, such as &quot;SNP&quot; (single nucleotide</span>
</span><span id="HGVSName-574"><a href="#HGVSName-574"><span class="linenos"> 574</span></a><span class="sd">        polymorphism), &quot;DEL&quot; (deletion), &quot;INS&quot; (insertion), etc</span>
</span><span id="HGVSName-575"><a href="#HGVSName-575"><span class="linenos"> 575</span></a><span class="sd">        :type mutation_type: str</span>
</span><span id="HGVSName-576"><a href="#HGVSName-576"><span class="linenos"> 576</span></a><span class="sd">        :param start: The `start` parameter represents the starting position of the mutation or variant</span>
</span><span id="HGVSName-577"><a href="#HGVSName-577"><span class="linenos"> 577</span></a><span class="sd">        in the genomic sequence. It is an integer value that indicates the position of the mutation or</span>
</span><span id="HGVSName-578"><a href="#HGVSName-578"><span class="linenos"> 578</span></a><span class="sd">        variant on the genomic sequence. If not provided, it defaults to 0, defaults to 0</span>
</span><span id="HGVSName-579"><a href="#HGVSName-579"><span class="linenos"> 579</span></a><span class="sd">        :type start: int (optional)</span>
</span><span id="HGVSName-580"><a href="#HGVSName-580"><span class="linenos"> 580</span></a><span class="sd">        :param end: The &quot;end&quot; parameter represents the end position of the mutation or variant. It is an</span>
</span><span id="HGVSName-581"><a href="#HGVSName-581"><span class="linenos"> 581</span></a><span class="sd">        integer value that indicates the position of the mutation or variant on the genomic sequence,</span>
</span><span id="HGVSName-582"><a href="#HGVSName-582"><span class="linenos"> 582</span></a><span class="sd">        defaults to 0</span>
</span><span id="HGVSName-583"><a href="#HGVSName-583"><span class="linenos"> 583</span></a><span class="sd">        :type end: int (optional)</span>
</span><span id="HGVSName-584"><a href="#HGVSName-584"><span class="linenos"> 584</span></a><span class="sd">        :param ref_allele: The `ref_allele` parameter represents the reference allele in a genetic</span>
</span><span id="HGVSName-585"><a href="#HGVSName-585"><span class="linenos"> 585</span></a><span class="sd">        mutation. It is the allele that is present in the reference genome at a specific position</span>
</span><span id="HGVSName-586"><a href="#HGVSName-586"><span class="linenos"> 586</span></a><span class="sd">        :type ref_allele: str</span>
</span><span id="HGVSName-587"><a href="#HGVSName-587"><span class="linenos"> 587</span></a><span class="sd">        :param ref2_allele: The `ref2_allele` parameter represents the reference allele at the end of a</span>
</span><span id="HGVSName-588"><a href="#HGVSName-588"><span class="linenos"> 588</span></a><span class="sd">        peptide indel. In the context of genetic mutations, an indel refers to the insertion or deletion</span>
</span><span id="HGVSName-589"><a href="#HGVSName-589"><span class="linenos"> 589</span></a><span class="sd">        of nucleotides in a DNA sequence. The `ref2_allele` specifically represents the reference allele</span>
</span><span id="HGVSName-590"><a href="#HGVSName-590"><span class="linenos"> 590</span></a><span class="sd">        that is</span>
</span><span id="HGVSName-591"><a href="#HGVSName-591"><span class="linenos"> 591</span></a><span class="sd">        :type ref2_allele: str</span>
</span><span id="HGVSName-592"><a href="#HGVSName-592"><span class="linenos"> 592</span></a><span class="sd">        :param alt_allele: The `alt_allele` parameter represents the alternate allele in a genetic</span>
</span><span id="HGVSName-593"><a href="#HGVSName-593"><span class="linenos"> 593</span></a><span class="sd">        mutation. In genetics, an allele is one of the possible forms of a gene. In the context of this</span>
</span><span id="HGVSName-594"><a href="#HGVSName-594"><span class="linenos"> 594</span></a><span class="sd">        code, `alt_allele` is used to store the alternate allele that is present in a mutation</span>
</span><span id="HGVSName-595"><a href="#HGVSName-595"><span class="linenos"> 595</span></a><span class="sd">        :type alt_allele: str</span>
</span><span id="HGVSName-596"><a href="#HGVSName-596"><span class="linenos"> 596</span></a><span class="sd">        :param cdna_start: The `cdna_start` parameter is used to specify the start position of the</span>
</span><span id="HGVSName-597"><a href="#HGVSName-597"><span class="linenos"> 597</span></a><span class="sd">        mutation in the cDNA sequence. It is an optional parameter and if not provided, it will be set</span>
</span><span id="HGVSName-598"><a href="#HGVSName-598"><span class="linenos"> 598</span></a><span class="sd">        to a default value of `CDNACoord()`</span>
</span><span id="HGVSName-599"><a href="#HGVSName-599"><span class="linenos"> 599</span></a><span class="sd">        :type cdna_start: int</span>
</span><span id="HGVSName-600"><a href="#HGVSName-600"><span class="linenos"> 600</span></a><span class="sd">        :param cdna_end: The `cdna_end` parameter is used to store the end coordinate of the cDNA</span>
</span><span id="HGVSName-601"><a href="#HGVSName-601"><span class="linenos"> 601</span></a><span class="sd">        (complementary DNA) sequence. It is an optional parameter and if not provided, it will be</span>
</span><span id="HGVSName-602"><a href="#HGVSName-602"><span class="linenos"> 602</span></a><span class="sd">        initialized as a `CDNACoord` object. The `CDNACoord` object is likely a</span>
</span><span id="HGVSName-603"><a href="#HGVSName-603"><span class="linenos"> 603</span></a><span class="sd">        :type cdna_end: int</span>
</span><span id="HGVSName-604"><a href="#HGVSName-604"><span class="linenos"> 604</span></a><span class="sd">        :param pep_extra: The `pep_extra` parameter is a string that represents any additional</span>
</span><span id="HGVSName-605"><a href="#HGVSName-605"><span class="linenos"> 605</span></a><span class="sd">        information related to the protein. It is used in the context of protein-specific fields</span>
</span><span id="HGVSName-606"><a href="#HGVSName-606"><span class="linenos"> 606</span></a><span class="sd">        :type pep_extra: str</span>
</span><span id="HGVSName-607"><a href="#HGVSName-607"><span class="linenos"> 607</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="HGVSName-608"><a href="#HGVSName-608"><span class="linenos"> 608</span></a>
</span><span id="HGVSName-609"><a href="#HGVSName-609"><span class="linenos"> 609</span></a>        <span class="c1"># Full HGVS name.</span>
</span><span id="HGVSName-610"><a href="#HGVSName-610"><span class="linenos"> 610</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">name</span> <span class="o">=</span> <span class="n">name</span>
</span><span id="HGVSName-611"><a href="#HGVSName-611"><span class="linenos"> 611</span></a>
</span><span id="HGVSName-612"><a href="#HGVSName-612"><span class="linenos"> 612</span></a>        <span class="c1"># Name parts.</span>
</span><span id="HGVSName-613"><a href="#HGVSName-613"><span class="linenos"> 613</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">prefix</span> <span class="o">=</span> <span class="n">prefix</span>
</span><span id="HGVSName-614"><a href="#HGVSName-614"><span class="linenos"> 614</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">chrom</span> <span class="o">=</span> <span class="n">chrom</span>
</span><span id="HGVSName-615"><a href="#HGVSName-615"><span class="linenos"> 615</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span> <span class="o">=</span> <span class="n">transcript</span>
</span><span id="HGVSName-616"><a href="#HGVSName-616"><span class="linenos"> 616</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">transcript_protein</span> <span class="o">=</span> <span class="n">transcript_protein</span>
</span><span id="HGVSName-617"><a href="#HGVSName-617"><span class="linenos"> 617</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">gene</span> <span class="o">=</span> <span class="n">gene</span>
</span><span id="HGVSName-618"><a href="#HGVSName-618"><span class="linenos"> 618</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">exon</span> <span class="o">=</span> <span class="n">exon</span>
</span><span id="HGVSName-619"><a href="#HGVSName-619"><span class="linenos"> 619</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">kind</span> <span class="o">=</span> <span class="n">kind</span>
</span><span id="HGVSName-620"><a href="#HGVSName-620"><span class="linenos"> 620</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">=</span> <span class="n">mutation_type</span>
</span><span id="HGVSName-621"><a href="#HGVSName-621"><span class="linenos"> 621</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">start</span> <span class="o">=</span> <span class="n">start</span>
</span><span id="HGVSName-622"><a href="#HGVSName-622"><span class="linenos"> 622</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">end</span> <span class="o">=</span> <span class="n">end</span>
</span><span id="HGVSName-623"><a href="#HGVSName-623"><span class="linenos"> 623</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">=</span> <span class="n">ref_allele</span>  <span class="c1"># reference allele</span>
</span><span id="HGVSName-624"><a href="#HGVSName-624"><span class="linenos"> 624</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">ref2_allele</span> <span class="o">=</span> <span class="n">ref2_allele</span>  <span class="c1"># reference allele at end of pep indel</span>
</span><span id="HGVSName-625"><a href="#HGVSName-625"><span class="linenos"> 625</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span> <span class="o">=</span> <span class="n">alt_allele</span>  <span class="c1"># alternate allele</span>
</span><span id="HGVSName-626"><a href="#HGVSName-626"><span class="linenos"> 626</span></a>
</span><span id="HGVSName-627"><a href="#HGVSName-627"><span class="linenos"> 627</span></a>        <span class="c1"># cDNA-specific fields</span>
</span><span id="HGVSName-628"><a href="#HGVSName-628"><span class="linenos"> 628</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">cdna_start</span> <span class="o">=</span> <span class="n">cdna_start</span> <span class="k">if</span> <span class="n">cdna_start</span> <span class="k">else</span> <span class="n">CDNACoord</span><span class="p">()</span>
</span><span id="HGVSName-629"><a href="#HGVSName-629"><span class="linenos"> 629</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">cdna_end</span> <span class="o">=</span> <span class="n">cdna_end</span> <span class="k">if</span> <span class="n">cdna_end</span> <span class="k">else</span> <span class="n">CDNACoord</span><span class="p">()</span>
</span><span id="HGVSName-630"><a href="#HGVSName-630"><span class="linenos"> 630</span></a>
</span><span id="HGVSName-631"><a href="#HGVSName-631"><span class="linenos"> 631</span></a>        <span class="c1"># Protein-specific fields</span>
</span><span id="HGVSName-632"><a href="#HGVSName-632"><span class="linenos"> 632</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">pep_extra</span> <span class="o">=</span> <span class="n">pep_extra</span>
</span><span id="HGVSName-633"><a href="#HGVSName-633"><span class="linenos"> 633</span></a>
</span><span id="HGVSName-634"><a href="#HGVSName-634"><span class="linenos"> 634</span></a>        <span class="k">if</span> <span class="n">name</span><span class="p">:</span>
</span><span id="HGVSName-635"><a href="#HGVSName-635"><span class="linenos"> 635</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">parse</span><span class="p">(</span><span class="n">name</span><span class="p">)</span>
</span><span id="HGVSName-636"><a href="#HGVSName-636"><span class="linenos"> 636</span></a>
</span><span id="HGVSName-637"><a href="#HGVSName-637"><span class="linenos"> 637</span></a>    <span class="k">def</span> <span class="nf">parse</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">name</span><span class="p">:</span> <span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
</span><span id="HGVSName-638"><a href="#HGVSName-638"><span class="linenos"> 638</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="HGVSName-639"><a href="#HGVSName-639"><span class="linenos"> 639</span></a><span class="sd">        The `parse` function is used to split an HGVS name into a prefix and allele, and then validate</span>
</span><span id="HGVSName-640"><a href="#HGVSName-640"><span class="linenos"> 640</span></a><span class="sd">        the parsed components.</span>
</span><span id="HGVSName-641"><a href="#HGVSName-641"><span class="linenos"> 641</span></a>
</span><span id="HGVSName-642"><a href="#HGVSName-642"><span class="linenos"> 642</span></a><span class="sd">        :param name: The `name` parameter is a string that represents an HGVS name. It is the input to</span>
</span><span id="HGVSName-643"><a href="#HGVSName-643"><span class="linenos"> 643</span></a><span class="sd">        the `parse` function and is used to parse the HGVS name by splitting it into a prefix and allele</span>
</span><span id="HGVSName-644"><a href="#HGVSName-644"><span class="linenos"> 644</span></a><span class="sd">        :type name: str</span>
</span><span id="HGVSName-645"><a href="#HGVSName-645"><span class="linenos"> 645</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="HGVSName-646"><a href="#HGVSName-646"><span class="linenos"> 646</span></a>
</span><span id="HGVSName-647"><a href="#HGVSName-647"><span class="linenos"> 647</span></a>        <span class="c1"># Does HGVS name have transcript/gene prefix?</span>
</span><span id="HGVSName-648"><a href="#HGVSName-648"><span class="linenos"> 648</span></a>        <span class="k">if</span> <span class="s2">&quot;:&quot;</span> <span class="ow">in</span> <span class="n">name</span><span class="p">:</span>
</span><span id="HGVSName-649"><a href="#HGVSName-649"><span class="linenos"> 649</span></a>            <span class="n">prefix</span><span class="p">,</span> <span class="n">allele</span> <span class="o">=</span> <span class="n">name</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot;:&quot;</span><span class="p">,</span> <span class="mi">1</span><span class="p">)</span>
</span><span id="HGVSName-650"><a href="#HGVSName-650"><span class="linenos"> 650</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName-651"><a href="#HGVSName-651"><span class="linenos"> 651</span></a>            <span class="n">prefix</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span>
</span><span id="HGVSName-652"><a href="#HGVSName-652"><span class="linenos"> 652</span></a>            <span class="n">allele</span> <span class="o">=</span> <span class="n">name</span>
</span><span id="HGVSName-653"><a href="#HGVSName-653"><span class="linenos"> 653</span></a>
</span><span id="HGVSName-654"><a href="#HGVSName-654"><span class="linenos"> 654</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">name</span> <span class="o">=</span> <span class="n">name</span>
</span><span id="HGVSName-655"><a href="#HGVSName-655"><span class="linenos"> 655</span></a>
</span><span id="HGVSName-656"><a href="#HGVSName-656"><span class="linenos"> 656</span></a>        <span class="c1"># Parse prefix and allele.</span>
</span><span id="HGVSName-657"><a href="#HGVSName-657"><span class="linenos"> 657</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">parse_allele</span><span class="p">(</span><span class="n">allele</span><span class="p">)</span>
</span><span id="HGVSName-658"><a href="#HGVSName-658"><span class="linenos"> 658</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">parse_prefix</span><span class="p">(</span><span class="n">prefix</span><span class="p">)</span>
</span><span id="HGVSName-659"><a href="#HGVSName-659"><span class="linenos"> 659</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">_validate</span><span class="p">()</span>
</span><span id="HGVSName-660"><a href="#HGVSName-660"><span class="linenos"> 660</span></a>
</span><span id="HGVSName-661"><a href="#HGVSName-661"><span class="linenos"> 661</span></a>    <span class="k">def</span> <span class="nf">parse_prefix</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">prefix</span><span class="p">:</span> <span class="nb">str</span><span class="p">):</span>
</span><span id="HGVSName-662"><a href="#HGVSName-662"><span class="linenos"> 662</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="HGVSName-663"><a href="#HGVSName-663"><span class="linenos"> 663</span></a><span class="sd">        The `parse_prefix` function is used to parse a HGVS prefix (gene/transcript/chromosome) and</span>
</span><span id="HGVSName-664"><a href="#HGVSName-664"><span class="linenos"> 664</span></a><span class="sd">        assign the parsed values to the corresponding attributes of the object.</span>
</span><span id="HGVSName-665"><a href="#HGVSName-665"><span class="linenos"> 665</span></a>
</span><span id="HGVSName-666"><a href="#HGVSName-666"><span class="linenos"> 666</span></a><span class="sd">        :param prefix: The `prefix` parameter is a string that represents a HGVS prefix, which can be a</span>
</span><span id="HGVSName-667"><a href="#HGVSName-667"><span class="linenos"> 667</span></a><span class="sd">        gene, transcript, or chromosome identifier. It is used to determine the type of prefix and</span>
</span><span id="HGVSName-668"><a href="#HGVSName-668"><span class="linenos"> 668</span></a><span class="sd">        assign the parsed values to the corresponding attributes of the object</span>
</span><span id="HGVSName-669"><a href="#HGVSName-669"><span class="linenos"> 669</span></a><span class="sd">        :type prefix: str</span>
</span><span id="HGVSName-670"><a href="#HGVSName-670"><span class="linenos"> 670</span></a><span class="sd">        :return: The function `parse_prefix` returns the parsed values for the transcript and gene</span>
</span><span id="HGVSName-671"><a href="#HGVSName-671"><span class="linenos"> 671</span></a><span class="sd">        attributes, or sets the chrom or gene attributes based on the given prefix.</span>
</span><span id="HGVSName-672"><a href="#HGVSName-672"><span class="linenos"> 672</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="HGVSName-673"><a href="#HGVSName-673"><span class="linenos"> 673</span></a>
</span><span id="HGVSName-674"><a href="#HGVSName-674"><span class="linenos"> 674</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">prefix</span> <span class="o">=</span> <span class="n">prefix</span>
</span><span id="HGVSName-675"><a href="#HGVSName-675"><span class="linenos"> 675</span></a>
</span><span id="HGVSName-676"><a href="#HGVSName-676"><span class="linenos"> 676</span></a>        <span class="c1"># No prefix.</span>
</span><span id="HGVSName-677"><a href="#HGVSName-677"><span class="linenos"> 677</span></a>        <span class="k">if</span> <span class="n">prefix</span> <span class="o">==</span> <span class="s2">&quot;&quot;</span><span class="p">:</span>
</span><span id="HGVSName-678"><a href="#HGVSName-678"><span class="linenos"> 678</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">chrom</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span>
</span><span id="HGVSName-679"><a href="#HGVSName-679"><span class="linenos"> 679</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span>
</span><span id="HGVSName-680"><a href="#HGVSName-680"><span class="linenos"> 680</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">gene</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span>
</span><span id="HGVSName-681"><a href="#HGVSName-681"><span class="linenos"> 681</span></a>            <span class="k">return</span>
</span><span id="HGVSName-682"><a href="#HGVSName-682"><span class="linenos"> 682</span></a>
</span><span id="HGVSName-683"><a href="#HGVSName-683"><span class="linenos"> 683</span></a>        <span class="c1"># Transcript and gene given with parens.</span>
</span><span id="HGVSName-684"><a href="#HGVSName-684"><span class="linenos"> 684</span></a>        <span class="c1"># example: NM_007294.3(BRCA1):c.2207A&gt;C</span>
</span><span id="HGVSName-685"><a href="#HGVSName-685"><span class="linenos"> 685</span></a>        <span class="n">match</span> <span class="o">=</span> <span class="n">re</span><span class="o">.</span><span class="n">match</span><span class="p">(</span><span class="sa">r</span><span class="s2">&quot;^(?P&lt;transcript&gt;[^(]+)\((?P&lt;gene&gt;[^)]+)\)$&quot;</span><span class="p">,</span> <span class="n">prefix</span><span class="p">)</span>
</span><span id="HGVSName-686"><a href="#HGVSName-686"><span class="linenos"> 686</span></a>        <span class="k">if</span> <span class="n">match</span><span class="p">:</span>
</span><span id="HGVSName-687"><a href="#HGVSName-687"><span class="linenos"> 687</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span> <span class="o">=</span> <span class="n">match</span><span class="o">.</span><span class="n">group</span><span class="p">(</span><span class="s2">&quot;transcript&quot;</span><span class="p">)</span>
</span><span id="HGVSName-688"><a href="#HGVSName-688"><span class="linenos"> 688</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">gene</span> <span class="o">=</span> <span class="n">match</span><span class="o">.</span><span class="n">group</span><span class="p">(</span><span class="s2">&quot;gene&quot;</span><span class="p">)</span>
</span><span id="HGVSName-689"><a href="#HGVSName-689"><span class="linenos"> 689</span></a>            <span class="k">return</span>
</span><span id="HGVSName-690"><a href="#HGVSName-690"><span class="linenos"> 690</span></a>
</span><span id="HGVSName-691"><a href="#HGVSName-691"><span class="linenos"> 691</span></a>        <span class="c1"># Transcript and gene given with braces.</span>
</span><span id="HGVSName-692"><a href="#HGVSName-692"><span class="linenos"> 692</span></a>        <span class="c1"># example: BRCA1{NM_007294.3}:c.2207A&gt;C</span>
</span><span id="HGVSName-693"><a href="#HGVSName-693"><span class="linenos"> 693</span></a>        <span class="n">match</span> <span class="o">=</span> <span class="n">re</span><span class="o">.</span><span class="n">match</span><span class="p">(</span><span class="sa">r</span><span class="s2">&quot;^(?P&lt;gene&gt;[^{]+){(?P&lt;transcript&gt;[^}]+)}$&quot;</span><span class="p">,</span> <span class="n">prefix</span><span class="p">)</span>
</span><span id="HGVSName-694"><a href="#HGVSName-694"><span class="linenos"> 694</span></a>        <span class="k">if</span> <span class="n">match</span><span class="p">:</span>
</span><span id="HGVSName-695"><a href="#HGVSName-695"><span class="linenos"> 695</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span> <span class="o">=</span> <span class="n">match</span><span class="o">.</span><span class="n">group</span><span class="p">(</span><span class="s2">&quot;transcript&quot;</span><span class="p">)</span>
</span><span id="HGVSName-696"><a href="#HGVSName-696"><span class="linenos"> 696</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">gene</span> <span class="o">=</span> <span class="n">match</span><span class="o">.</span><span class="n">group</span><span class="p">(</span><span class="s2">&quot;gene&quot;</span><span class="p">)</span>
</span><span id="HGVSName-697"><a href="#HGVSName-697"><span class="linenos"> 697</span></a>            <span class="k">return</span>
</span><span id="HGVSName-698"><a href="#HGVSName-698"><span class="linenos"> 698</span></a>
</span><span id="HGVSName-699"><a href="#HGVSName-699"><span class="linenos"> 699</span></a>        <span class="c1"># Determine using Ensembl type.</span>
</span><span id="HGVSName-700"><a href="#HGVSName-700"><span class="linenos"> 700</span></a>        <span class="k">if</span> <span class="n">prefix</span><span class="o">.</span><span class="n">startswith</span><span class="p">(</span><span class="s2">&quot;ENST&quot;</span><span class="p">):</span>
</span><span id="HGVSName-701"><a href="#HGVSName-701"><span class="linenos"> 701</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span> <span class="o">=</span> <span class="n">prefix</span>
</span><span id="HGVSName-702"><a href="#HGVSName-702"><span class="linenos"> 702</span></a>            <span class="k">return</span>
</span><span id="HGVSName-703"><a href="#HGVSName-703"><span class="linenos"> 703</span></a>
</span><span id="HGVSName-704"><a href="#HGVSName-704"><span class="linenos"> 704</span></a>        <span class="c1"># Determine using LRG type.</span>
</span><span id="HGVSName-705"><a href="#HGVSName-705"><span class="linenos"> 705</span></a>        <span class="k">if</span> <span class="n">prefix</span><span class="o">.</span><span class="n">startswith</span><span class="p">(</span><span class="s2">&quot;LRG_&quot;</span><span class="p">):</span>
</span><span id="HGVSName-706"><a href="#HGVSName-706"><span class="linenos"> 706</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span> <span class="o">=</span> <span class="n">prefix</span>
</span><span id="HGVSName-707"><a href="#HGVSName-707"><span class="linenos"> 707</span></a>            <span class="k">return</span>
</span><span id="HGVSName-708"><a href="#HGVSName-708"><span class="linenos"> 708</span></a>
</span><span id="HGVSName-709"><a href="#HGVSName-709"><span class="linenos"> 709</span></a>        <span class="c1"># Determine using refseq type.</span>
</span><span id="HGVSName-710"><a href="#HGVSName-710"><span class="linenos"> 710</span></a>        <span class="n">refseq_type</span> <span class="o">=</span> <span class="n">get_refseq_type</span><span class="p">(</span><span class="n">prefix</span><span class="p">)</span>
</span><span id="HGVSName-711"><a href="#HGVSName-711"><span class="linenos"> 711</span></a>        <span class="k">if</span> <span class="n">refseq_type</span> <span class="ow">in</span> <span class="p">(</span><span class="s2">&quot;mRNA&quot;</span><span class="p">,</span> <span class="s2">&quot;RNA&quot;</span><span class="p">):</span>
</span><span id="HGVSName-712"><a href="#HGVSName-712"><span class="linenos"> 712</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span> <span class="o">=</span> <span class="n">prefix</span>
</span><span id="HGVSName-713"><a href="#HGVSName-713"><span class="linenos"> 713</span></a>            <span class="k">return</span>
</span><span id="HGVSName-714"><a href="#HGVSName-714"><span class="linenos"> 714</span></a>
</span><span id="HGVSName-715"><a href="#HGVSName-715"><span class="linenos"> 715</span></a>        <span class="c1"># Determine using refseq type.</span>
</span><span id="HGVSName-716"><a href="#HGVSName-716"><span class="linenos"> 716</span></a>        <span class="k">if</span> <span class="n">prefix</span><span class="o">.</span><span class="n">startswith</span><span class="p">(</span><span class="n">CHROM_PREFIX</span><span class="p">)</span> <span class="ow">or</span> <span class="n">refseq_type</span> <span class="o">==</span> <span class="s2">&quot;genomic&quot;</span><span class="p">:</span>
</span><span id="HGVSName-717"><a href="#HGVSName-717"><span class="linenos"> 717</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">chrom</span> <span class="o">=</span> <span class="n">prefix</span>
</span><span id="HGVSName-718"><a href="#HGVSName-718"><span class="linenos"> 718</span></a>            <span class="k">return</span>
</span><span id="HGVSName-719"><a href="#HGVSName-719"><span class="linenos"> 719</span></a>
</span><span id="HGVSName-720"><a href="#HGVSName-720"><span class="linenos"> 720</span></a>        <span class="c1"># Assume gene name.</span>
</span><span id="HGVSName-721"><a href="#HGVSName-721"><span class="linenos"> 721</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">gene</span> <span class="o">=</span> <span class="n">prefix</span>
</span><span id="HGVSName-722"><a href="#HGVSName-722"><span class="linenos"> 722</span></a>
</span><span id="HGVSName-723"><a href="#HGVSName-723"><span class="linenos"> 723</span></a>    <span class="k">def</span> <span class="nf">parse_allele</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">allele</span><span class="p">:</span> <span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
</span><span id="HGVSName-724"><a href="#HGVSName-724"><span class="linenos"> 724</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="HGVSName-725"><a href="#HGVSName-725"><span class="linenos"> 725</span></a><span class="sd">        The function `parse_allele` parses a HGVS allele description and determines the kind of HGVS</span>
</span><span id="HGVSName-726"><a href="#HGVSName-726"><span class="linenos"> 726</span></a><span class="sd">        name (c., p., g., etc.) and the mutation type.</span>
</span><span id="HGVSName-727"><a href="#HGVSName-727"><span class="linenos"> 727</span></a>
</span><span id="HGVSName-728"><a href="#HGVSName-728"><span class="linenos"> 728</span></a><span class="sd">        Some examples include:</span>
</span><span id="HGVSName-729"><a href="#HGVSName-729"><span class="linenos"> 729</span></a><span class="sd">          cDNA substitution: c.101A&gt;C,</span>
</span><span id="HGVSName-730"><a href="#HGVSName-730"><span class="linenos"> 730</span></a><span class="sd">          cDNA indel: c.3428delCinsTA, c.1000_1003delATG, c.1000_1001insATG</span>
</span><span id="HGVSName-731"><a href="#HGVSName-731"><span class="linenos"> 731</span></a><span class="sd">          No protein change: p.Glu1161=</span>
</span><span id="HGVSName-732"><a href="#HGVSName-732"><span class="linenos"> 732</span></a><span class="sd">          Protein change: p.Glu1161Ser</span>
</span><span id="HGVSName-733"><a href="#HGVSName-733"><span class="linenos"> 733</span></a><span class="sd">          Protein frameshift: p.Glu1161_Ser1164?fs</span>
</span><span id="HGVSName-734"><a href="#HGVSName-734"><span class="linenos"> 734</span></a><span class="sd">          Genomic substitution: g.1000100A&gt;T</span>
</span><span id="HGVSName-735"><a href="#HGVSName-735"><span class="linenos"> 735</span></a><span class="sd">          Genomic indel: g.1000100_1000102delATG</span>
</span><span id="HGVSName-736"><a href="#HGVSName-736"><span class="linenos"> 736</span></a>
</span><span id="HGVSName-737"><a href="#HGVSName-737"><span class="linenos"> 737</span></a><span class="sd">        :param allele: The `allele` parameter is a string that represents a HGVS allele description. It</span>
</span><span id="HGVSName-738"><a href="#HGVSName-738"><span class="linenos"> 738</span></a><span class="sd">        can contain various types of mutations, such as cDNA substitutions, cDNA indels, protein</span>
</span><span id="HGVSName-739"><a href="#HGVSName-739"><span class="linenos"> 739</span></a><span class="sd">        changes, protein frameshifts, genomic substitutions, and genomic indels. The purpose of the</span>
</span><span id="HGVSName-740"><a href="#HGVSName-740"><span class="linenos"> 740</span></a><span class="sd">        `parse_allele`</span>
</span><span id="HGVSName-741"><a href="#HGVSName-741"><span class="linenos"> 741</span></a><span class="sd">        :type allele: str</span>
</span><span id="HGVSName-742"><a href="#HGVSName-742"><span class="linenos"> 742</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="HGVSName-743"><a href="#HGVSName-743"><span class="linenos"> 743</span></a>
</span><span id="HGVSName-744"><a href="#HGVSName-744"><span class="linenos"> 744</span></a>        <span class="k">if</span> <span class="s2">&quot;.&quot;</span> <span class="ow">not</span> <span class="ow">in</span> <span class="n">allele</span><span class="p">:</span>
</span><span id="HGVSName-745"><a href="#HGVSName-745"><span class="linenos"> 745</span></a>            <span class="k">raise</span> <span class="n">InvalidHGVSName</span><span class="p">(</span>
</span><span id="HGVSName-746"><a href="#HGVSName-746"><span class="linenos"> 746</span></a>                <span class="n">allele</span><span class="p">,</span> <span class="s2">&quot;allele&quot;</span><span class="p">,</span> <span class="s1">&#39;expected kind &quot;c.&quot;, &quot;p.&quot;, &quot;g.&quot;, etc&#39;</span>
</span><span id="HGVSName-747"><a href="#HGVSName-747"><span class="linenos"> 747</span></a>            <span class="p">)</span>
</span><span id="HGVSName-748"><a href="#HGVSName-748"><span class="linenos"> 748</span></a>
</span><span id="HGVSName-749"><a href="#HGVSName-749"><span class="linenos"> 749</span></a>        <span class="c1"># Determine HGVS name kind.</span>
</span><span id="HGVSName-750"><a href="#HGVSName-750"><span class="linenos"> 750</span></a>        <span class="n">kind</span><span class="p">,</span> <span class="n">details</span> <span class="o">=</span> <span class="n">allele</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot;.&quot;</span><span class="p">,</span> <span class="mi">1</span><span class="p">)</span>
</span><span id="HGVSName-751"><a href="#HGVSName-751"><span class="linenos"> 751</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">kind</span> <span class="o">=</span> <span class="n">kind</span>
</span><span id="HGVSName-752"><a href="#HGVSName-752"><span class="linenos"> 752</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">=</span> <span class="kc">None</span>
</span><span id="HGVSName-753"><a href="#HGVSName-753"><span class="linenos"> 753</span></a>
</span><span id="HGVSName-754"><a href="#HGVSName-754"><span class="linenos"> 754</span></a>        <span class="k">if</span> <span class="n">kind</span> <span class="ow">in</span> <span class="p">(</span><span class="s2">&quot;c&quot;</span><span class="p">,</span> <span class="s2">&quot;n&quot;</span><span class="p">):</span>
</span><span id="HGVSName-755"><a href="#HGVSName-755"><span class="linenos"> 755</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">parse_cdna</span><span class="p">(</span><span class="n">details</span><span class="p">)</span>
</span><span id="HGVSName-756"><a href="#HGVSName-756"><span class="linenos"> 756</span></a>            <span class="k">if</span> <span class="n">kind</span> <span class="o">==</span> <span class="s2">&quot;n&quot;</span><span class="p">:</span>  <span class="c1"># Ensure no 3&#39;UTR or 5&#39;UTR coords in non-coding</span>
</span><span id="HGVSName-757"><a href="#HGVSName-757"><span class="linenos"> 757</span></a>                <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">cdna_start</span><span class="o">.</span><span class="n">coord</span> <span class="o">&lt;</span> <span class="mi">0</span><span class="p">:</span>
</span><span id="HGVSName-758"><a href="#HGVSName-758"><span class="linenos"> 758</span></a>                    <span class="k">raise</span> <span class="n">InvalidHGVSName</span><span class="p">(</span>
</span><span id="HGVSName-759"><a href="#HGVSName-759"><span class="linenos"> 759</span></a>                        <span class="n">allele</span><span class="p">,</span>
</span><span id="HGVSName-760"><a href="#HGVSName-760"><span class="linenos"> 760</span></a>                        <span class="s2">&quot;allele&quot;</span><span class="p">,</span>
</span><span id="HGVSName-761"><a href="#HGVSName-761"><span class="linenos"> 761</span></a>                        <span class="s2">&quot;Non-coding transcript cannot contain negative (5&#39;UTR) coordinates&quot;</span><span class="p">,</span>
</span><span id="HGVSName-762"><a href="#HGVSName-762"><span class="linenos"> 762</span></a>                    <span class="p">)</span>
</span><span id="HGVSName-763"><a href="#HGVSName-763"><span class="linenos"> 763</span></a>                <span class="k">if</span> <span class="p">(</span>
</span><span id="HGVSName-764"><a href="#HGVSName-764"><span class="linenos"> 764</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">cdna_start</span><span class="o">.</span><span class="n">landmark</span> <span class="o">==</span> <span class="s2">&quot;cdna_stop&quot;</span>
</span><span id="HGVSName-765"><a href="#HGVSName-765"><span class="linenos"> 765</span></a>                    <span class="ow">or</span> <span class="bp">self</span><span class="o">.</span><span class="n">cdna_end</span>
</span><span id="HGVSName-766"><a href="#HGVSName-766"><span class="linenos"> 766</span></a>                    <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">cdna_end</span><span class="o">.</span><span class="n">landmark</span> <span class="o">==</span> <span class="s2">&quot;cdna_stop&quot;</span>
</span><span id="HGVSName-767"><a href="#HGVSName-767"><span class="linenos"> 767</span></a>                <span class="p">):</span>
</span><span id="HGVSName-768"><a href="#HGVSName-768"><span class="linenos"> 768</span></a>                    <span class="k">raise</span> <span class="n">InvalidHGVSName</span><span class="p">(</span>
</span><span id="HGVSName-769"><a href="#HGVSName-769"><span class="linenos"> 769</span></a>                        <span class="n">allele</span><span class="p">,</span>
</span><span id="HGVSName-770"><a href="#HGVSName-770"><span class="linenos"> 770</span></a>                        <span class="s2">&quot;allele&quot;</span><span class="p">,</span>
</span><span id="HGVSName-771"><a href="#HGVSName-771"><span class="linenos"> 771</span></a>                        <span class="s2">&quot;Non-coding transcript cannot contain &#39;*&#39; (3&#39;UTR) coordinates&quot;</span><span class="p">,</span>
</span><span id="HGVSName-772"><a href="#HGVSName-772"><span class="linenos"> 772</span></a>                    <span class="p">)</span>
</span><span id="HGVSName-773"><a href="#HGVSName-773"><span class="linenos"> 773</span></a>        <span class="k">elif</span> <span class="n">kind</span> <span class="o">==</span> <span class="s2">&quot;p&quot;</span><span class="p">:</span>
</span><span id="HGVSName-774"><a href="#HGVSName-774"><span class="linenos"> 774</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">parse_protein</span><span class="p">(</span><span class="n">details</span><span class="p">)</span>
</span><span id="HGVSName-775"><a href="#HGVSName-775"><span class="linenos"> 775</span></a>        <span class="k">elif</span> <span class="n">kind</span> <span class="ow">in</span> <span class="p">(</span><span class="s2">&quot;g&quot;</span><span class="p">,</span> <span class="s2">&quot;m&quot;</span><span class="p">):</span>
</span><span id="HGVSName-776"><a href="#HGVSName-776"><span class="linenos"> 776</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">parse_genome</span><span class="p">(</span><span class="n">details</span><span class="p">)</span>
</span><span id="HGVSName-777"><a href="#HGVSName-777"><span class="linenos"> 777</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName-778"><a href="#HGVSName-778"><span class="linenos"> 778</span></a>            <span class="k">raise</span> <span class="ne">NotImplementedError</span><span class="p">(</span><span class="s2">&quot;unknown kind: </span><span class="si">%s</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="n">allele</span><span class="p">)</span>
</span><span id="HGVSName-779"><a href="#HGVSName-779"><span class="linenos"> 779</span></a>
</span><span id="HGVSName-780"><a href="#HGVSName-780"><span class="linenos"> 780</span></a>    <span class="k">def</span> <span class="nf">parse_cdna</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">details</span><span class="p">:</span> <span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
</span><span id="HGVSName-781"><a href="#HGVSName-781"><span class="linenos"> 781</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="HGVSName-782"><a href="#HGVSName-782"><span class="linenos"> 782</span></a><span class="sd">        The function `parse_cdna` is used to parse a HGVS cDNA name and extract information such as</span>
</span><span id="HGVSName-783"><a href="#HGVSName-783"><span class="linenos"> 783</span></a><span class="sd">        mutation type, coordinates, and alleles.</span>
</span><span id="HGVSName-784"><a href="#HGVSName-784"><span class="linenos"> 784</span></a>
</span><span id="HGVSName-785"><a href="#HGVSName-785"><span class="linenos"> 785</span></a><span class="sd">        Some examples include:</span>
</span><span id="HGVSName-786"><a href="#HGVSName-786"><span class="linenos"> 786</span></a><span class="sd">          Substitution: 101A&gt;C,</span>
</span><span id="HGVSName-787"><a href="#HGVSName-787"><span class="linenos"> 787</span></a><span class="sd">          Indel: 3428delCinsTA, 1000_1003delATG, 1000_1001insATG</span>
</span><span id="HGVSName-788"><a href="#HGVSName-788"><span class="linenos"> 788</span></a>
</span><span id="HGVSName-789"><a href="#HGVSName-789"><span class="linenos"> 789</span></a><span class="sd">        :param details: The `details` parameter is a string that represents a HGVS cDNA name. It</span>
</span><span id="HGVSName-790"><a href="#HGVSName-790"><span class="linenos"> 790</span></a><span class="sd">        contains information about a genetic mutation, such as a substitution or an indel, along with</span>
</span><span id="HGVSName-791"><a href="#HGVSName-791"><span class="linenos"> 791</span></a><span class="sd">        the specific coordinates and alleles involved in the mutation</span>
</span><span id="HGVSName-792"><a href="#HGVSName-792"><span class="linenos"> 792</span></a><span class="sd">        :type details: str</span>
</span><span id="HGVSName-793"><a href="#HGVSName-793"><span class="linenos"> 793</span></a><span class="sd">        :return: None.</span>
</span><span id="HGVSName-794"><a href="#HGVSName-794"><span class="linenos"> 794</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="HGVSName-795"><a href="#HGVSName-795"><span class="linenos"> 795</span></a>
</span><span id="HGVSName-796"><a href="#HGVSName-796"><span class="linenos"> 796</span></a>        <span class="k">for</span> <span class="n">regex</span> <span class="ow">in</span> <span class="n">HGVSRegex</span><span class="o">.</span><span class="n">CDNA_ALLELE_REGEXES</span><span class="p">:</span>
</span><span id="HGVSName-797"><a href="#HGVSName-797"><span class="linenos"> 797</span></a>            <span class="n">match</span> <span class="o">=</span> <span class="n">re</span><span class="o">.</span><span class="n">match</span><span class="p">(</span><span class="n">regex</span><span class="p">,</span> <span class="n">details</span><span class="p">)</span>
</span><span id="HGVSName-798"><a href="#HGVSName-798"><span class="linenos"> 798</span></a>            <span class="k">if</span> <span class="n">match</span><span class="p">:</span>
</span><span id="HGVSName-799"><a href="#HGVSName-799"><span class="linenos"> 799</span></a>                <span class="n">groups</span> <span class="o">=</span> <span class="n">match</span><span class="o">.</span><span class="n">groupdict</span><span class="p">()</span>
</span><span id="HGVSName-800"><a href="#HGVSName-800"><span class="linenos"> 800</span></a>
</span><span id="HGVSName-801"><a href="#HGVSName-801"><span class="linenos"> 801</span></a>                <span class="c1"># Parse mutation type.</span>
</span><span id="HGVSName-802"><a href="#HGVSName-802"><span class="linenos"> 802</span></a>                <span class="k">if</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;delins&quot;</span><span class="p">):</span>
</span><span id="HGVSName-803"><a href="#HGVSName-803"><span class="linenos"> 803</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">=</span> <span class="s2">&quot;delins&quot;</span>
</span><span id="HGVSName-804"><a href="#HGVSName-804"><span class="linenos"> 804</span></a>                <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName-805"><a href="#HGVSName-805"><span class="linenos"> 805</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">=</span> <span class="n">groups</span><span class="p">[</span><span class="s2">&quot;mutation_type&quot;</span><span class="p">]</span>
</span><span id="HGVSName-806"><a href="#HGVSName-806"><span class="linenos"> 806</span></a>
</span><span id="HGVSName-807"><a href="#HGVSName-807"><span class="linenos"> 807</span></a>                <span class="c1"># Parse coordinates.</span>
</span><span id="HGVSName-808"><a href="#HGVSName-808"><span class="linenos"> 808</span></a>                <span class="bp">self</span><span class="o">.</span><span class="n">cdna_start</span> <span class="o">=</span> <span class="n">CDNACoord</span><span class="p">(</span><span class="n">string</span><span class="o">=</span><span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;start&quot;</span><span class="p">))</span>
</span><span id="HGVSName-809"><a href="#HGVSName-809"><span class="linenos"> 809</span></a>                <span class="k">if</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;end&quot;</span><span class="p">):</span>
</span><span id="HGVSName-810"><a href="#HGVSName-810"><span class="linenos"> 810</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">cdna_end</span> <span class="o">=</span> <span class="n">CDNACoord</span><span class="p">(</span><span class="n">string</span><span class="o">=</span><span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;end&quot;</span><span class="p">))</span>
</span><span id="HGVSName-811"><a href="#HGVSName-811"><span class="linenos"> 811</span></a>                <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName-812"><a href="#HGVSName-812"><span class="linenos"> 812</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">cdna_end</span> <span class="o">=</span> <span class="n">CDNACoord</span><span class="p">(</span><span class="n">string</span><span class="o">=</span><span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;start&quot;</span><span class="p">))</span>
</span><span id="HGVSName-813"><a href="#HGVSName-813"><span class="linenos"> 813</span></a>
</span><span id="HGVSName-814"><a href="#HGVSName-814"><span class="linenos"> 814</span></a>                <span class="c1"># Parse alleles.</span>
</span><span id="HGVSName-815"><a href="#HGVSName-815"><span class="linenos"> 815</span></a>                <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">=</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;ref&quot;</span><span class="p">,</span> <span class="s2">&quot;&quot;</span><span class="p">)</span>
</span><span id="HGVSName-816"><a href="#HGVSName-816"><span class="linenos"> 816</span></a>                <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span> <span class="o">=</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;alt&quot;</span><span class="p">,</span> <span class="s2">&quot;&quot;</span><span class="p">)</span>
</span><span id="HGVSName-817"><a href="#HGVSName-817"><span class="linenos"> 817</span></a>
</span><span id="HGVSName-818"><a href="#HGVSName-818"><span class="linenos"> 818</span></a>                <span class="c1"># Convert numerical allelles.</span>
</span><span id="HGVSName-819"><a href="#HGVSName-819"><span class="linenos"> 819</span></a>                <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span><span class="o">.</span><span class="n">isdigit</span><span class="p">():</span>
</span><span id="HGVSName-820"><a href="#HGVSName-820"><span class="linenos"> 820</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">=</span> <span class="s2">&quot;N&quot;</span> <span class="o">*</span> <span class="nb">int</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span><span class="p">)</span>
</span><span id="HGVSName-821"><a href="#HGVSName-821"><span class="linenos"> 821</span></a>                <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span><span class="o">.</span><span class="n">isdigit</span><span class="p">():</span>
</span><span id="HGVSName-822"><a href="#HGVSName-822"><span class="linenos"> 822</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span> <span class="o">=</span> <span class="s2">&quot;N&quot;</span> <span class="o">*</span> <span class="nb">int</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span><span class="p">)</span>
</span><span id="HGVSName-823"><a href="#HGVSName-823"><span class="linenos"> 823</span></a>
</span><span id="HGVSName-824"><a href="#HGVSName-824"><span class="linenos"> 824</span></a>                <span class="c1"># Convert duplication alleles.</span>
</span><span id="HGVSName-825"><a href="#HGVSName-825"><span class="linenos"> 825</span></a>                <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">==</span> <span class="s2">&quot;dup&quot;</span><span class="p">:</span>
</span><span id="HGVSName-826"><a href="#HGVSName-826"><span class="linenos"> 826</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">*</span> <span class="mi">2</span>
</span><span id="HGVSName-827"><a href="#HGVSName-827"><span class="linenos"> 827</span></a>
</span><span id="HGVSName-828"><a href="#HGVSName-828"><span class="linenos"> 828</span></a>                <span class="c1"># Convert no match alleles.</span>
</span><span id="HGVSName-829"><a href="#HGVSName-829"><span class="linenos"> 829</span></a>                <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">==</span> <span class="s2">&quot;=&quot;</span><span class="p">:</span>
</span><span id="HGVSName-830"><a href="#HGVSName-830"><span class="linenos"> 830</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span>
</span><span id="HGVSName-831"><a href="#HGVSName-831"><span class="linenos"> 831</span></a>                <span class="k">return</span>
</span><span id="HGVSName-832"><a href="#HGVSName-832"><span class="linenos"> 832</span></a>
</span><span id="HGVSName-833"><a href="#HGVSName-833"><span class="linenos"> 833</span></a>        <span class="k">raise</span> <span class="n">InvalidHGVSName</span><span class="p">(</span><span class="n">details</span><span class="p">,</span> <span class="s2">&quot;cDNA allele&quot;</span><span class="p">)</span>
</span><span id="HGVSName-834"><a href="#HGVSName-834"><span class="linenos"> 834</span></a>
</span><span id="HGVSName-835"><a href="#HGVSName-835"><span class="linenos"> 835</span></a>    <span class="k">def</span> <span class="nf">parse_protein</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">details</span><span class="p">:</span> <span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
</span><span id="HGVSName-836"><a href="#HGVSName-836"><span class="linenos"> 836</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="HGVSName-837"><a href="#HGVSName-837"><span class="linenos"> 837</span></a><span class="sd">        The function `parse_protein` is used to parse a HGVS protein name and extract information such</span>
</span><span id="HGVSName-838"><a href="#HGVSName-838"><span class="linenos"> 838</span></a><span class="sd">        as mutation type, coordinates, alleles, and additional details.</span>
</span><span id="HGVSName-839"><a href="#HGVSName-839"><span class="linenos"> 839</span></a>
</span><span id="HGVSName-840"><a href="#HGVSName-840"><span class="linenos"> 840</span></a><span class="sd">        Some examples include:</span>
</span><span id="HGVSName-841"><a href="#HGVSName-841"><span class="linenos"> 841</span></a><span class="sd">          No change: Glu1161=</span>
</span><span id="HGVSName-842"><a href="#HGVSName-842"><span class="linenos"> 842</span></a><span class="sd">          Change: Glu1161Ser</span>
</span><span id="HGVSName-843"><a href="#HGVSName-843"><span class="linenos"> 843</span></a><span class="sd">          Frameshift: Glu1161_Ser1164?fs</span>
</span><span id="HGVSName-844"><a href="#HGVSName-844"><span class="linenos"> 844</span></a>
</span><span id="HGVSName-845"><a href="#HGVSName-845"><span class="linenos"> 845</span></a><span class="sd">        :param details: The `details` parameter is a string that represents a HGVS protein name. It</span>
</span><span id="HGVSName-846"><a href="#HGVSName-846"><span class="linenos"> 846</span></a><span class="sd">        contains information about a protein mutation, such as the amino acid change and the position of</span>
</span><span id="HGVSName-847"><a href="#HGVSName-847"><span class="linenos"> 847</span></a><span class="sd">        the mutation</span>
</span><span id="HGVSName-848"><a href="#HGVSName-848"><span class="linenos"> 848</span></a><span class="sd">        :type details: str</span>
</span><span id="HGVSName-849"><a href="#HGVSName-849"><span class="linenos"> 849</span></a><span class="sd">        :return: The method `parse_protein` does not return anything. It updates the instance variables</span>
</span><span id="HGVSName-850"><a href="#HGVSName-850"><span class="linenos"> 850</span></a><span class="sd">        of the object it is called on.</span>
</span><span id="HGVSName-851"><a href="#HGVSName-851"><span class="linenos"> 851</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="HGVSName-852"><a href="#HGVSName-852"><span class="linenos"> 852</span></a>
</span><span id="HGVSName-853"><a href="#HGVSName-853"><span class="linenos"> 853</span></a>        <span class="k">for</span> <span class="n">regex</span> <span class="ow">in</span> <span class="n">HGVSRegex</span><span class="o">.</span><span class="n">PEP_ALLELE_REGEXES</span><span class="p">:</span>
</span><span id="HGVSName-854"><a href="#HGVSName-854"><span class="linenos"> 854</span></a>            <span class="n">match</span> <span class="o">=</span> <span class="n">re</span><span class="o">.</span><span class="n">match</span><span class="p">(</span><span class="n">regex</span><span class="p">,</span> <span class="n">details</span><span class="p">)</span>
</span><span id="HGVSName-855"><a href="#HGVSName-855"><span class="linenos"> 855</span></a>            <span class="k">if</span> <span class="n">match</span><span class="p">:</span>
</span><span id="HGVSName-856"><a href="#HGVSName-856"><span class="linenos"> 856</span></a>                <span class="n">groups</span> <span class="o">=</span> <span class="n">match</span><span class="o">.</span><span class="n">groupdict</span><span class="p">()</span>
</span><span id="HGVSName-857"><a href="#HGVSName-857"><span class="linenos"> 857</span></a>
</span><span id="HGVSName-858"><a href="#HGVSName-858"><span class="linenos"> 858</span></a>                <span class="c1"># Parse mutation type.</span>
</span><span id="HGVSName-859"><a href="#HGVSName-859"><span class="linenos"> 859</span></a>                <span class="k">if</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;delins&quot;</span><span class="p">):</span>
</span><span id="HGVSName-860"><a href="#HGVSName-860"><span class="linenos"> 860</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">=</span> <span class="s2">&quot;delins&quot;</span>
</span><span id="HGVSName-861"><a href="#HGVSName-861"><span class="linenos"> 861</span></a>                <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName-862"><a href="#HGVSName-862"><span class="linenos"> 862</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">=</span> <span class="s2">&quot;&gt;&quot;</span>
</span><span id="HGVSName-863"><a href="#HGVSName-863"><span class="linenos"> 863</span></a>
</span><span id="HGVSName-864"><a href="#HGVSName-864"><span class="linenos"> 864</span></a>                <span class="c1"># Parse coordinates.</span>
</span><span id="HGVSName-865"><a href="#HGVSName-865"><span class="linenos"> 865</span></a>                <span class="bp">self</span><span class="o">.</span><span class="n">start</span> <span class="o">=</span> <span class="nb">int</span><span class="p">(</span><span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;start&quot;</span><span class="p">))</span>
</span><span id="HGVSName-866"><a href="#HGVSName-866"><span class="linenos"> 866</span></a>                <span class="k">if</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;end&quot;</span><span class="p">):</span>
</span><span id="HGVSName-867"><a href="#HGVSName-867"><span class="linenos"> 867</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">end</span> <span class="o">=</span> <span class="nb">int</span><span class="p">(</span><span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;end&quot;</span><span class="p">))</span>
</span><span id="HGVSName-868"><a href="#HGVSName-868"><span class="linenos"> 868</span></a>                <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName-869"><a href="#HGVSName-869"><span class="linenos"> 869</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">end</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">start</span>
</span><span id="HGVSName-870"><a href="#HGVSName-870"><span class="linenos"> 870</span></a>
</span><span id="HGVSName-871"><a href="#HGVSName-871"><span class="linenos"> 871</span></a>                <span class="c1"># Parse alleles.</span>
</span><span id="HGVSName-872"><a href="#HGVSName-872"><span class="linenos"> 872</span></a>                <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">=</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;ref&quot;</span><span class="p">,</span> <span class="s2">&quot;&quot;</span><span class="p">)</span>
</span><span id="HGVSName-873"><a href="#HGVSName-873"><span class="linenos"> 873</span></a>                <span class="k">if</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;ref2&quot;</span><span class="p">):</span>
</span><span id="HGVSName-874"><a href="#HGVSName-874"><span class="linenos"> 874</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">ref2_allele</span> <span class="o">=</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;ref2&quot;</span><span class="p">)</span>
</span><span id="HGVSName-875"><a href="#HGVSName-875"><span class="linenos"> 875</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span> <span class="o">=</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;alt&quot;</span><span class="p">,</span> <span class="s2">&quot;&quot;</span><span class="p">)</span>
</span><span id="HGVSName-876"><a href="#HGVSName-876"><span class="linenos"> 876</span></a>                <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName-877"><a href="#HGVSName-877"><span class="linenos"> 877</span></a>                    <span class="c1"># If alt is not given, assume matching with ref</span>
</span><span id="HGVSName-878"><a href="#HGVSName-878"><span class="linenos"> 878</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">ref2_allele</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span>
</span><span id="HGVSName-879"><a href="#HGVSName-879"><span class="linenos"> 879</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span> <span class="o">=</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;alt&quot;</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span><span class="p">)</span>
</span><span id="HGVSName-880"><a href="#HGVSName-880"><span class="linenos"> 880</span></a>
</span><span id="HGVSName-881"><a href="#HGVSName-881"><span class="linenos"> 881</span></a>                <span class="bp">self</span><span class="o">.</span><span class="n">pep_extra</span> <span class="o">=</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;extra&quot;</span><span class="p">)</span>
</span><span id="HGVSName-882"><a href="#HGVSName-882"><span class="linenos"> 882</span></a>                <span class="k">return</span>
</span><span id="HGVSName-883"><a href="#HGVSName-883"><span class="linenos"> 883</span></a>
</span><span id="HGVSName-884"><a href="#HGVSName-884"><span class="linenos"> 884</span></a>        <span class="k">raise</span> <span class="n">InvalidHGVSName</span><span class="p">(</span><span class="n">details</span><span class="p">,</span> <span class="s2">&quot;protein allele&quot;</span><span class="p">)</span>
</span><span id="HGVSName-885"><a href="#HGVSName-885"><span class="linenos"> 885</span></a>
</span><span id="HGVSName-886"><a href="#HGVSName-886"><span class="linenos"> 886</span></a>    <span class="k">def</span> <span class="nf">parse_genome</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">details</span><span class="p">:</span> <span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
</span><span id="HGVSName-887"><a href="#HGVSName-887"><span class="linenos"> 887</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="HGVSName-888"><a href="#HGVSName-888"><span class="linenos"> 888</span></a><span class="sd">        The function `parse_genome` is used to parse a HGVS genomic name and extract information such as</span>
</span><span id="HGVSName-889"><a href="#HGVSName-889"><span class="linenos"> 889</span></a><span class="sd">        mutation type, coordinates, and alleles.</span>
</span><span id="HGVSName-890"><a href="#HGVSName-890"><span class="linenos"> 890</span></a>
</span><span id="HGVSName-891"><a href="#HGVSName-891"><span class="linenos"> 891</span></a><span class="sd">        Some examples include:</span>
</span><span id="HGVSName-892"><a href="#HGVSName-892"><span class="linenos"> 892</span></a><span class="sd">          Substitution: 1000100A&gt;T</span>
</span><span id="HGVSName-893"><a href="#HGVSName-893"><span class="linenos"> 893</span></a><span class="sd">          Indel: 1000100_1000102delATG</span>
</span><span id="HGVSName-894"><a href="#HGVSName-894"><span class="linenos"> 894</span></a>
</span><span id="HGVSName-895"><a href="#HGVSName-895"><span class="linenos"> 895</span></a><span class="sd">        :param details: The `details` parameter is a string that represents a HGVS genomic name. It</span>
</span><span id="HGVSName-896"><a href="#HGVSName-896"><span class="linenos"> 896</span></a><span class="sd">        contains information about a genomic mutation, such as a substitution or an indel</span>
</span><span id="HGVSName-897"><a href="#HGVSName-897"><span class="linenos"> 897</span></a><span class="sd">        :type details: str</span>
</span><span id="HGVSName-898"><a href="#HGVSName-898"><span class="linenos"> 898</span></a><span class="sd">        :return: None.</span>
</span><span id="HGVSName-899"><a href="#HGVSName-899"><span class="linenos"> 899</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="HGVSName-900"><a href="#HGVSName-900"><span class="linenos"> 900</span></a>
</span><span id="HGVSName-901"><a href="#HGVSName-901"><span class="linenos"> 901</span></a>        <span class="k">for</span> <span class="n">regex</span> <span class="ow">in</span> <span class="n">HGVSRegex</span><span class="o">.</span><span class="n">GENOMIC_ALLELE_REGEXES</span><span class="p">:</span>
</span><span id="HGVSName-902"><a href="#HGVSName-902"><span class="linenos"> 902</span></a>            <span class="n">match</span> <span class="o">=</span> <span class="n">re</span><span class="o">.</span><span class="n">match</span><span class="p">(</span><span class="n">regex</span><span class="p">,</span> <span class="n">details</span><span class="p">)</span>
</span><span id="HGVSName-903"><a href="#HGVSName-903"><span class="linenos"> 903</span></a>            <span class="k">if</span> <span class="n">match</span><span class="p">:</span>
</span><span id="HGVSName-904"><a href="#HGVSName-904"><span class="linenos"> 904</span></a>                <span class="n">groups</span> <span class="o">=</span> <span class="n">match</span><span class="o">.</span><span class="n">groupdict</span><span class="p">()</span>
</span><span id="HGVSName-905"><a href="#HGVSName-905"><span class="linenos"> 905</span></a>
</span><span id="HGVSName-906"><a href="#HGVSName-906"><span class="linenos"> 906</span></a>                <span class="c1"># Parse mutation type.</span>
</span><span id="HGVSName-907"><a href="#HGVSName-907"><span class="linenos"> 907</span></a>                <span class="k">if</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;delins&quot;</span><span class="p">):</span>
</span><span id="HGVSName-908"><a href="#HGVSName-908"><span class="linenos"> 908</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">=</span> <span class="s2">&quot;delins&quot;</span>
</span><span id="HGVSName-909"><a href="#HGVSName-909"><span class="linenos"> 909</span></a>                <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName-910"><a href="#HGVSName-910"><span class="linenos"> 910</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">=</span> <span class="n">groups</span><span class="p">[</span><span class="s2">&quot;mutation_type&quot;</span><span class="p">]</span>
</span><span id="HGVSName-911"><a href="#HGVSName-911"><span class="linenos"> 911</span></a>
</span><span id="HGVSName-912"><a href="#HGVSName-912"><span class="linenos"> 912</span></a>                <span class="c1"># Parse coordinates.</span>
</span><span id="HGVSName-913"><a href="#HGVSName-913"><span class="linenos"> 913</span></a>                <span class="bp">self</span><span class="o">.</span><span class="n">start</span> <span class="o">=</span> <span class="nb">int</span><span class="p">(</span><span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;start&quot;</span><span class="p">))</span>
</span><span id="HGVSName-914"><a href="#HGVSName-914"><span class="linenos"> 914</span></a>                <span class="k">if</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;end&quot;</span><span class="p">):</span>
</span><span id="HGVSName-915"><a href="#HGVSName-915"><span class="linenos"> 915</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">end</span> <span class="o">=</span> <span class="nb">int</span><span class="p">(</span><span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;end&quot;</span><span class="p">))</span>
</span><span id="HGVSName-916"><a href="#HGVSName-916"><span class="linenos"> 916</span></a>                <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName-917"><a href="#HGVSName-917"><span class="linenos"> 917</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">end</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">start</span>
</span><span id="HGVSName-918"><a href="#HGVSName-918"><span class="linenos"> 918</span></a>
</span><span id="HGVSName-919"><a href="#HGVSName-919"><span class="linenos"> 919</span></a>                <span class="c1"># Parse alleles.</span>
</span><span id="HGVSName-920"><a href="#HGVSName-920"><span class="linenos"> 920</span></a>                <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">=</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;ref&quot;</span><span class="p">,</span> <span class="s2">&quot;&quot;</span><span class="p">)</span>
</span><span id="HGVSName-921"><a href="#HGVSName-921"><span class="linenos"> 921</span></a>                <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span> <span class="o">=</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;alt&quot;</span><span class="p">,</span> <span class="s2">&quot;&quot;</span><span class="p">)</span>
</span><span id="HGVSName-922"><a href="#HGVSName-922"><span class="linenos"> 922</span></a>
</span><span id="HGVSName-923"><a href="#HGVSName-923"><span class="linenos"> 923</span></a>                <span class="c1"># Convert numerical alleles.</span>
</span><span id="HGVSName-924"><a href="#HGVSName-924"><span class="linenos"> 924</span></a>                <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span><span class="o">.</span><span class="n">isdigit</span><span class="p">():</span>
</span><span id="HGVSName-925"><a href="#HGVSName-925"><span class="linenos"> 925</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">=</span> <span class="s2">&quot;N&quot;</span> <span class="o">*</span> <span class="nb">int</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span><span class="p">)</span>
</span><span id="HGVSName-926"><a href="#HGVSName-926"><span class="linenos"> 926</span></a>                <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span><span class="o">.</span><span class="n">isdigit</span><span class="p">():</span>
</span><span id="HGVSName-927"><a href="#HGVSName-927"><span class="linenos"> 927</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span> <span class="o">=</span> <span class="s2">&quot;N&quot;</span> <span class="o">*</span> <span class="nb">int</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span><span class="p">)</span>
</span><span id="HGVSName-928"><a href="#HGVSName-928"><span class="linenos"> 928</span></a>
</span><span id="HGVSName-929"><a href="#HGVSName-929"><span class="linenos"> 929</span></a>                <span class="c1"># Convert duplication alleles.</span>
</span><span id="HGVSName-930"><a href="#HGVSName-930"><span class="linenos"> 930</span></a>                <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">==</span> <span class="s2">&quot;dup&quot;</span><span class="p">:</span>
</span><span id="HGVSName-931"><a href="#HGVSName-931"><span class="linenos"> 931</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">*</span> <span class="mi">2</span>
</span><span id="HGVSName-932"><a href="#HGVSName-932"><span class="linenos"> 932</span></a>
</span><span id="HGVSName-933"><a href="#HGVSName-933"><span class="linenos"> 933</span></a>                <span class="c1"># Convert no match alleles.</span>
</span><span id="HGVSName-934"><a href="#HGVSName-934"><span class="linenos"> 934</span></a>                <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">==</span> <span class="s2">&quot;=&quot;</span><span class="p">:</span>
</span><span id="HGVSName-935"><a href="#HGVSName-935"><span class="linenos"> 935</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span>
</span><span id="HGVSName-936"><a href="#HGVSName-936"><span class="linenos"> 936</span></a>                <span class="k">return</span>
</span><span id="HGVSName-937"><a href="#HGVSName-937"><span class="linenos"> 937</span></a>
</span><span id="HGVSName-938"><a href="#HGVSName-938"><span class="linenos"> 938</span></a>        <span class="k">raise</span> <span class="n">InvalidHGVSName</span><span class="p">(</span><span class="n">details</span><span class="p">,</span> <span class="s2">&quot;genomic allele&quot;</span><span class="p">)</span>
</span><span id="HGVSName-939"><a href="#HGVSName-939"><span class="linenos"> 939</span></a>
</span><span id="HGVSName-940"><a href="#HGVSName-940"><span class="linenos"> 940</span></a>    <span class="k">def</span> <span class="nf">_validate</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
</span><span id="HGVSName-941"><a href="#HGVSName-941"><span class="linenos"> 941</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="HGVSName-942"><a href="#HGVSName-942"><span class="linenos"> 942</span></a><span class="sd">        The function checks for internal inconsistencies in the representation of coordinates.</span>
</span><span id="HGVSName-943"><a href="#HGVSName-943"><span class="linenos"> 943</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="HGVSName-944"><a href="#HGVSName-944"><span class="linenos"> 944</span></a>
</span><span id="HGVSName-945"><a href="#HGVSName-945"><span class="linenos"> 945</span></a>        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">start</span> <span class="o">&gt;</span> <span class="bp">self</span><span class="o">.</span><span class="n">end</span><span class="p">:</span>
</span><span id="HGVSName-946"><a href="#HGVSName-946"><span class="linenos"> 946</span></a>            <span class="k">raise</span> <span class="n">InvalidHGVSName</span><span class="p">(</span><span class="n">reason</span><span class="o">=</span><span class="s2">&quot;Coordinates are nonincreasing&quot;</span><span class="p">)</span>
</span><span id="HGVSName-947"><a href="#HGVSName-947"><span class="linenos"> 947</span></a>
</span><span id="HGVSName-948"><a href="#HGVSName-948"><span class="linenos"> 948</span></a>    <span class="k">def</span> <span class="fm">__repr__</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
</span><span id="HGVSName-949"><a href="#HGVSName-949"><span class="linenos"> 949</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="HGVSName-950"><a href="#HGVSName-950"><span class="linenos"> 950</span></a><span class="sd">        The function returns a string representation of an HGVSName object.</span>
</span><span id="HGVSName-951"><a href="#HGVSName-951"><span class="linenos"> 951</span></a><span class="sd">        :return: The `__repr__` method is returning a string representation of the object. If the</span>
</span><span id="HGVSName-952"><a href="#HGVSName-952"><span class="linenos"> 952</span></a><span class="sd">        `format` method is implemented, it will return a string in the format</span>
</span><span id="HGVSName-953"><a href="#HGVSName-953"><span class="linenos"> 953</span></a><span class="sd">        &quot;HGVSName(&#39;formatted_string&#39;)&quot;. If the `format` method is not implemented, it will return a</span>
</span><span id="HGVSName-954"><a href="#HGVSName-954"><span class="linenos"> 954</span></a><span class="sd">        string in the format &quot;HGVSName(&#39;name&#39;)&quot;.</span>
</span><span id="HGVSName-955"><a href="#HGVSName-955"><span class="linenos"> 955</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="HGVSName-956"><a href="#HGVSName-956"><span class="linenos"> 956</span></a>
</span><span id="HGVSName-957"><a href="#HGVSName-957"><span class="linenos"> 957</span></a>        <span class="k">try</span><span class="p">:</span>
</span><span id="HGVSName-958"><a href="#HGVSName-958"><span class="linenos"> 958</span></a>            <span class="k">return</span> <span class="s2">&quot;HGVSName(&#39;</span><span class="si">%s</span><span class="s2">&#39;)&quot;</span> <span class="o">%</span> <span class="bp">self</span><span class="o">.</span><span class="n">format</span><span class="p">()</span>
</span><span id="HGVSName-959"><a href="#HGVSName-959"><span class="linenos"> 959</span></a>        <span class="k">except</span> <span class="ne">NotImplementedError</span><span class="p">:</span>
</span><span id="HGVSName-960"><a href="#HGVSName-960"><span class="linenos"> 960</span></a>            <span class="k">return</span> <span class="s2">&quot;HGVSName(&#39;</span><span class="si">%s</span><span class="s2">&#39;)&quot;</span> <span class="o">%</span> <span class="bp">self</span><span class="o">.</span><span class="n">name</span>
</span><span id="HGVSName-961"><a href="#HGVSName-961"><span class="linenos"> 961</span></a>
</span><span id="HGVSName-962"><a href="#HGVSName-962"><span class="linenos"> 962</span></a>    <span class="k">def</span> <span class="nf">__unicode__</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
</span><span id="HGVSName-963"><a href="#HGVSName-963"><span class="linenos"> 963</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="HGVSName-964"><a href="#HGVSName-964"><span class="linenos"> 964</span></a><span class="sd">        The function returns a formatted string representation of the object.</span>
</span><span id="HGVSName-965"><a href="#HGVSName-965"><span class="linenos"> 965</span></a><span class="sd">        :return: The `__unicode__` method is returning the result of the `format()` method.</span>
</span><span id="HGVSName-966"><a href="#HGVSName-966"><span class="linenos"> 966</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="HGVSName-967"><a href="#HGVSName-967"><span class="linenos"> 967</span></a>
</span><span id="HGVSName-968"><a href="#HGVSName-968"><span class="linenos"> 968</span></a>        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">format</span><span class="p">()</span>
</span><span id="HGVSName-969"><a href="#HGVSName-969"><span class="linenos"> 969</span></a>
</span><span id="HGVSName-970"><a href="#HGVSName-970"><span class="linenos"> 970</span></a>    <span class="k">def</span> <span class="nf">format</span><span class="p">(</span>
</span><span id="HGVSName-971"><a href="#HGVSName-971"><span class="linenos"> 971</span></a>        <span class="bp">self</span><span class="p">,</span>
</span><span id="HGVSName-972"><a href="#HGVSName-972"><span class="linenos"> 972</span></a>        <span class="n">use_prefix</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span>
</span><span id="HGVSName-973"><a href="#HGVSName-973"><span class="linenos"> 973</span></a>        <span class="n">use_gene</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span>
</span><span id="HGVSName-974"><a href="#HGVSName-974"><span class="linenos"> 974</span></a>        <span class="n">use_exon</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
</span><span id="HGVSName-975"><a href="#HGVSName-975"><span class="linenos"> 975</span></a>        <span class="n">use_protein</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
</span><span id="HGVSName-976"><a href="#HGVSName-976"><span class="linenos"> 976</span></a>        <span class="n">full_format</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
</span><span id="HGVSName-977"><a href="#HGVSName-977"><span class="linenos"> 977</span></a>        <span class="n">use_version</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
</span><span id="HGVSName-978"><a href="#HGVSName-978"><span class="linenos"> 978</span></a>    <span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
</span><span id="HGVSName-979"><a href="#HGVSName-979"><span class="linenos"> 979</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="HGVSName-980"><a href="#HGVSName-980"><span class="linenos"> 980</span></a><span class="sd">        The `format` function generates a HGVS name as a string based on various formatting options.</span>
</span><span id="HGVSName-981"><a href="#HGVSName-981"><span class="linenos"> 981</span></a>
</span><span id="HGVSName-982"><a href="#HGVSName-982"><span class="linenos"> 982</span></a><span class="sd">        :param use_prefix: A boolean indicating whether to include the prefix in the HGVS name. If set</span>
</span><span id="HGVSName-983"><a href="#HGVSName-983"><span class="linenos"> 983</span></a><span class="sd">        to True, the prefix will be included in the HGVS name. If set to False, the prefix will be</span>
</span><span id="HGVSName-984"><a href="#HGVSName-984"><span class="linenos"> 984</span></a><span class="sd">        excluded. The default value is True, defaults to True</span>
</span><span id="HGVSName-985"><a href="#HGVSName-985"><span class="linenos"> 985</span></a><span class="sd">        :type use_prefix: bool (optional)</span>
</span><span id="HGVSName-986"><a href="#HGVSName-986"><span class="linenos"> 986</span></a><span class="sd">        :param use_gene: A boolean indicating whether to include the gene name in the HGVS name. If set</span>
</span><span id="HGVSName-987"><a href="#HGVSName-987"><span class="linenos"> 987</span></a><span class="sd">        to True, the gene name will be included in the HGVS name. If set to False, the gene name will</span>
</span><span id="HGVSName-988"><a href="#HGVSName-988"><span class="linenos"> 988</span></a><span class="sd">        not be included. The default value is True, defaults to True</span>
</span><span id="HGVSName-989"><a href="#HGVSName-989"><span class="linenos"> 989</span></a><span class="sd">        :type use_gene: bool (optional)</span>
</span><span id="HGVSName-990"><a href="#HGVSName-990"><span class="linenos"> 990</span></a><span class="sd">        :param use_exon: A boolean indicating whether to include exon information in the HGVS name. If</span>
</span><span id="HGVSName-991"><a href="#HGVSName-991"><span class="linenos"> 991</span></a><span class="sd">        set to True, exon information will be included in the HGVS name. If set to False, exon</span>
</span><span id="HGVSName-992"><a href="#HGVSName-992"><span class="linenos"> 992</span></a><span class="sd">        information will not be included, defaults to False</span>
</span><span id="HGVSName-993"><a href="#HGVSName-993"><span class="linenos"> 993</span></a><span class="sd">        :type use_exon: bool (optional)</span>
</span><span id="HGVSName-994"><a href="#HGVSName-994"><span class="linenos"> 994</span></a><span class="sd">        :param use_protein: A boolean indicating whether to include the protein change in the HGVS name.</span>
</span><span id="HGVSName-995"><a href="#HGVSName-995"><span class="linenos"> 995</span></a><span class="sd">        If set to True, the protein change will be included in the HGVS name. If set to False, the</span>
</span><span id="HGVSName-996"><a href="#HGVSName-996"><span class="linenos"> 996</span></a><span class="sd">        protein change will not be included, defaults to False</span>
</span><span id="HGVSName-997"><a href="#HGVSName-997"><span class="linenos"> 997</span></a><span class="sd">        :type use_protein: bool (optional)</span>
</span><span id="HGVSName-998"><a href="#HGVSName-998"><span class="linenos"> 998</span></a><span class="sd">        :param full_format: A boolean parameter that determines whether the full format of the allele</span>
</span><span id="HGVSName-999"><a href="#HGVSName-999"><span class="linenos"> 999</span></a><span class="sd">        should be included in the output. If set to True, and if the allele is not a protein variant,</span>
</span><span id="HGVSName-1000"><a href="#HGVSName-1000"><span class="linenos">1000</span></a><span class="sd">        the allele will be appended with &#39;:p.&#39; followed by the formatted protein variant, defaults to</span>
</span><span id="HGVSName-1001"><a href="#HGVSName-1001"><span class="linenos">1001</span></a><span class="sd">        False (optional)</span>
</span><span id="HGVSName-1002"><a href="#HGVSName-1002"><span class="linenos">1002</span></a><span class="sd">        :param use_version: A boolean parameter that determines whether to include the version number in</span>
</span><span id="HGVSName-1003"><a href="#HGVSName-1003"><span class="linenos">1003</span></a><span class="sd">        the formatted HGVS name. If set to True, the version number will be included in the output. If</span>
</span><span id="HGVSName-1004"><a href="#HGVSName-1004"><span class="linenos">1004</span></a><span class="sd">        set to False, the version number will not be included, defaults to False</span>
</span><span id="HGVSName-1005"><a href="#HGVSName-1005"><span class="linenos">1005</span></a><span class="sd">        :type use_version: bool (optional)</span>
</span><span id="HGVSName-1006"><a href="#HGVSName-1006"><span class="linenos">1006</span></a><span class="sd">        :return: a HGVS name as a string.</span>
</span><span id="HGVSName-1007"><a href="#HGVSName-1007"><span class="linenos">1007</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="HGVSName-1008"><a href="#HGVSName-1008"><span class="linenos">1008</span></a>
</span><span id="HGVSName-1009"><a href="#HGVSName-1009"><span class="linenos">1009</span></a>        <span class="k">if</span> <span class="n">use_protein</span> <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">format_protein</span><span class="p">():</span>
</span><span id="HGVSName-1010"><a href="#HGVSName-1010"><span class="linenos">1010</span></a>            <span class="n">allele</span> <span class="o">=</span> <span class="s2">&quot;p.&quot;</span> <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">format_protein</span><span class="p">()</span>
</span><span id="HGVSName-1011"><a href="#HGVSName-1011"><span class="linenos">1011</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName-1012"><a href="#HGVSName-1012"><span class="linenos">1012</span></a>            <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">kind</span> <span class="ow">in</span> <span class="p">(</span><span class="s2">&quot;c&quot;</span><span class="p">,</span> <span class="s2">&quot;n&quot;</span><span class="p">):</span>
</span><span id="HGVSName-1013"><a href="#HGVSName-1013"><span class="linenos">1013</span></a>                <span class="n">allele</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">kind</span> <span class="o">+</span> <span class="s2">&quot;.&quot;</span> <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">format_cdna</span><span class="p">()</span>
</span><span id="HGVSName-1014"><a href="#HGVSName-1014"><span class="linenos">1014</span></a>            <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">kind</span> <span class="o">==</span> <span class="s2">&quot;p&quot;</span><span class="p">:</span>
</span><span id="HGVSName-1015"><a href="#HGVSName-1015"><span class="linenos">1015</span></a>                <span class="n">allele</span> <span class="o">=</span> <span class="s2">&quot;p.&quot;</span> <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">format_protein</span><span class="p">()</span>
</span><span id="HGVSName-1016"><a href="#HGVSName-1016"><span class="linenos">1016</span></a>            <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">kind</span> <span class="ow">in</span> <span class="p">(</span><span class="s2">&quot;g&quot;</span><span class="p">,</span> <span class="s2">&quot;m&quot;</span><span class="p">):</span>
</span><span id="HGVSName-1017"><a href="#HGVSName-1017"><span class="linenos">1017</span></a>                <span class="n">allele</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">kind</span> <span class="o">+</span> <span class="s2">&quot;.&quot;</span> <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">format_genome</span><span class="p">()</span>
</span><span id="HGVSName-1018"><a href="#HGVSName-1018"><span class="linenos">1018</span></a>            <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName-1019"><a href="#HGVSName-1019"><span class="linenos">1019</span></a>                <span class="k">raise</span> <span class="ne">NotImplementedError</span><span class="p">(</span><span class="s2">&quot;not implemented: &#39;</span><span class="si">%s</span><span class="s2">&#39;&quot;</span> <span class="o">%</span> <span class="bp">self</span><span class="o">.</span><span class="n">kind</span><span class="p">)</span>
</span><span id="HGVSName-1020"><a href="#HGVSName-1020"><span class="linenos">1020</span></a>
</span><span id="HGVSName-1021"><a href="#HGVSName-1021"><span class="linenos">1021</span></a>        <span class="k">if</span> <span class="n">full_format</span> <span class="ow">and</span> <span class="ow">not</span> <span class="n">use_protein</span> <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">format_protein</span><span class="p">():</span>
</span><span id="HGVSName-1022"><a href="#HGVSName-1022"><span class="linenos">1022</span></a>            <span class="n">allele</span> <span class="o">+=</span> <span class="s2">&quot;:p.&quot;</span> <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">format_protein</span><span class="p">()</span>
</span><span id="HGVSName-1023"><a href="#HGVSName-1023"><span class="linenos">1023</span></a>
</span><span id="HGVSName-1024"><a href="#HGVSName-1024"><span class="linenos">1024</span></a>        <span class="n">prefix</span> <span class="o">=</span> <span class="p">(</span>
</span><span id="HGVSName-1025"><a href="#HGVSName-1025"><span class="linenos">1025</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">format_prefix</span><span class="p">(</span>
</span><span id="HGVSName-1026"><a href="#HGVSName-1026"><span class="linenos">1026</span></a>                <span class="n">use_gene</span><span class="o">=</span><span class="n">use_gene</span><span class="p">,</span>
</span><span id="HGVSName-1027"><a href="#HGVSName-1027"><span class="linenos">1027</span></a>                <span class="n">use_exon</span><span class="o">=</span><span class="n">use_exon</span><span class="p">,</span>
</span><span id="HGVSName-1028"><a href="#HGVSName-1028"><span class="linenos">1028</span></a>                <span class="n">use_protein</span><span class="o">=</span><span class="n">use_protein</span><span class="p">,</span>
</span><span id="HGVSName-1029"><a href="#HGVSName-1029"><span class="linenos">1029</span></a>                <span class="n">full_format</span><span class="o">=</span><span class="n">full_format</span><span class="p">,</span>
</span><span id="HGVSName-1030"><a href="#HGVSName-1030"><span class="linenos">1030</span></a>                <span class="n">use_version</span><span class="o">=</span><span class="n">use_version</span><span class="p">,</span>
</span><span id="HGVSName-1031"><a href="#HGVSName-1031"><span class="linenos">1031</span></a>            <span class="p">)</span>
</span><span id="HGVSName-1032"><a href="#HGVSName-1032"><span class="linenos">1032</span></a>            <span class="k">if</span> <span class="n">use_prefix</span>
</span><span id="HGVSName-1033"><a href="#HGVSName-1033"><span class="linenos">1033</span></a>            <span class="k">else</span> <span class="s2">&quot;&quot;</span>
</span><span id="HGVSName-1034"><a href="#HGVSName-1034"><span class="linenos">1034</span></a>        <span class="p">)</span>
</span><span id="HGVSName-1035"><a href="#HGVSName-1035"><span class="linenos">1035</span></a>
</span><span id="HGVSName-1036"><a href="#HGVSName-1036"><span class="linenos">1036</span></a>        <span class="k">if</span> <span class="n">prefix</span><span class="p">:</span>
</span><span id="HGVSName-1037"><a href="#HGVSName-1037"><span class="linenos">1037</span></a>            <span class="k">return</span> <span class="n">prefix</span> <span class="o">+</span> <span class="s2">&quot;:&quot;</span> <span class="o">+</span> <span class="n">allele</span>
</span><span id="HGVSName-1038"><a href="#HGVSName-1038"><span class="linenos">1038</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName-1039"><a href="#HGVSName-1039"><span class="linenos">1039</span></a>            <span class="k">return</span> <span class="n">allele</span>
</span><span id="HGVSName-1040"><a href="#HGVSName-1040"><span class="linenos">1040</span></a>
</span><span id="HGVSName-1041"><a href="#HGVSName-1041"><span class="linenos">1041</span></a>    <span class="k">def</span> <span class="nf">format_prefix</span><span class="p">(</span>
</span><span id="HGVSName-1042"><a href="#HGVSName-1042"><span class="linenos">1042</span></a>        <span class="bp">self</span><span class="p">,</span>
</span><span id="HGVSName-1043"><a href="#HGVSName-1043"><span class="linenos">1043</span></a>        <span class="n">use_gene</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span>
</span><span id="HGVSName-1044"><a href="#HGVSName-1044"><span class="linenos">1044</span></a>        <span class="n">use_exon</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
</span><span id="HGVSName-1045"><a href="#HGVSName-1045"><span class="linenos">1045</span></a>        <span class="n">use_protein</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
</span><span id="HGVSName-1046"><a href="#HGVSName-1046"><span class="linenos">1046</span></a>        <span class="n">full_format</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
</span><span id="HGVSName-1047"><a href="#HGVSName-1047"><span class="linenos">1047</span></a>        <span class="n">use_version</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
</span><span id="HGVSName-1048"><a href="#HGVSName-1048"><span class="linenos">1048</span></a>    <span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
</span><span id="HGVSName-1049"><a href="#HGVSName-1049"><span class="linenos">1049</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="HGVSName-1050"><a href="#HGVSName-1050"><span class="linenos">1050</span></a><span class="sd">        The `format_prefix` function generates an HGVS transcript/gene prefix based on various</span>
</span><span id="HGVSName-1051"><a href="#HGVSName-1051"><span class="linenos">1051</span></a><span class="sd">        parameters.</span>
</span><span id="HGVSName-1052"><a href="#HGVSName-1052"><span class="linenos">1052</span></a>
</span><span id="HGVSName-1053"><a href="#HGVSName-1053"><span class="linenos">1053</span></a><span class="sd">        :param use_gene: A boolean parameter that determines whether to include the gene name in the</span>
</span><span id="HGVSName-1054"><a href="#HGVSName-1054"><span class="linenos">1054</span></a><span class="sd">        prefix. If set to True, the gene name will be included in the prefix. If set to False, the gene</span>
</span><span id="HGVSName-1055"><a href="#HGVSName-1055"><span class="linenos">1055</span></a><span class="sd">        name will not be included in the prefix. The default value is True, defaults to True</span>
</span><span id="HGVSName-1056"><a href="#HGVSName-1056"><span class="linenos">1056</span></a><span class="sd">        :type use_gene: bool (optional)</span>
</span><span id="HGVSName-1057"><a href="#HGVSName-1057"><span class="linenos">1057</span></a><span class="sd">        :param use_exon: A boolean parameter that determines whether to include the exon information in</span>
</span><span id="HGVSName-1058"><a href="#HGVSName-1058"><span class="linenos">1058</span></a><span class="sd">        the prefix. If set to True, the exon information will be included in the prefix. If set to</span>
</span><span id="HGVSName-1059"><a href="#HGVSName-1059"><span class="linenos">1059</span></a><span class="sd">        False, the exon information will not be included, defaults to False</span>
</span><span id="HGVSName-1060"><a href="#HGVSName-1060"><span class="linenos">1060</span></a><span class="sd">        :type use_exon: bool (optional)</span>
</span><span id="HGVSName-1061"><a href="#HGVSName-1061"><span class="linenos">1061</span></a><span class="sd">        :param use_protein: A boolean indicating whether to use the protein transcript instead of the</span>
</span><span id="HGVSName-1062"><a href="#HGVSName-1062"><span class="linenos">1062</span></a><span class="sd">        nucleotide transcript if available. If set to True, the protein transcript will be used. If set</span>
</span><span id="HGVSName-1063"><a href="#HGVSName-1063"><span class="linenos">1063</span></a><span class="sd">        to False, the nucleotide transcript will be used. The default value is False, defaults to False</span>
</span><span id="HGVSName-1064"><a href="#HGVSName-1064"><span class="linenos">1064</span></a><span class="sd">        :type use_protein: bool (optional)</span>
</span><span id="HGVSName-1065"><a href="#HGVSName-1065"><span class="linenos">1065</span></a><span class="sd">        :param full_format: A boolean parameter that determines whether to generate the full HGVS name</span>
</span><span id="HGVSName-1066"><a href="#HGVSName-1066"><span class="linenos">1066</span></a><span class="sd">        with transcript/gene prefix or not. If set to True, the full format will be generated. If set to</span>
</span><span id="HGVSName-1067"><a href="#HGVSName-1067"><span class="linenos">1067</span></a><span class="sd">        False, only the transcript/gene prefix will be generated, defaults to False</span>
</span><span id="HGVSName-1068"><a href="#HGVSName-1068"><span class="linenos">1068</span></a><span class="sd">        :type full_format: bool (optional)</span>
</span><span id="HGVSName-1069"><a href="#HGVSName-1069"><span class="linenos">1069</span></a><span class="sd">        :param use_version: A boolean parameter that determines whether to include the version number in</span>
</span><span id="HGVSName-1070"><a href="#HGVSName-1070"><span class="linenos">1070</span></a><span class="sd">        the transcript prefix. If set to True, the version number will be included in the prefix (e.g.,</span>
</span><span id="HGVSName-1071"><a href="#HGVSName-1071"><span class="linenos">1071</span></a><span class="sd">        NM_007294.3). If set to False, only the transcript ID without the version number will be</span>
</span><span id="HGVSName-1072"><a href="#HGVSName-1072"><span class="linenos">1072</span></a><span class="sd">        included in the prefix, defaults to False</span>
</span><span id="HGVSName-1073"><a href="#HGVSName-1073"><span class="linenos">1073</span></a><span class="sd">        :type use_version: bool (optional)</span>
</span><span id="HGVSName-1074"><a href="#HGVSName-1074"><span class="linenos">1074</span></a><span class="sd">        :return: The function `format_prefix` returns a formatted HGVS transcript/gene prefix as a</span>
</span><span id="HGVSName-1075"><a href="#HGVSName-1075"><span class="linenos">1075</span></a><span class="sd">        string.</span>
</span><span id="HGVSName-1076"><a href="#HGVSName-1076"><span class="linenos">1076</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="HGVSName-1077"><a href="#HGVSName-1077"><span class="linenos">1077</span></a>
</span><span id="HGVSName-1078"><a href="#HGVSName-1078"><span class="linenos">1078</span></a>        <span class="k">if</span> <span class="n">full_format</span><span class="p">:</span>
</span><span id="HGVSName-1079"><a href="#HGVSName-1079"><span class="linenos">1079</span></a>
</span><span id="HGVSName-1080"><a href="#HGVSName-1080"><span class="linenos">1080</span></a>            <span class="n">prefix</span> <span class="o">=</span> <span class="p">[]</span>
</span><span id="HGVSName-1081"><a href="#HGVSName-1081"><span class="linenos">1081</span></a>            <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">gene</span><span class="p">:</span>
</span><span id="HGVSName-1082"><a href="#HGVSName-1082"><span class="linenos">1082</span></a>                <span class="n">prefix</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">gene</span><span class="p">)</span>
</span><span id="HGVSName-1083"><a href="#HGVSName-1083"><span class="linenos">1083</span></a>            <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span><span class="p">:</span>
</span><span id="HGVSName-1084"><a href="#HGVSName-1084"><span class="linenos">1084</span></a>                <span class="n">prefix</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">transcript</span><span class="p">)</span>
</span><span id="HGVSName-1085"><a href="#HGVSName-1085"><span class="linenos">1085</span></a>            <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">transcript_protein</span><span class="p">:</span>
</span><span id="HGVSName-1086"><a href="#HGVSName-1086"><span class="linenos">1086</span></a>                <span class="n">prefix</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">transcript_protein</span><span class="p">)</span>
</span><span id="HGVSName-1087"><a href="#HGVSName-1087"><span class="linenos">1087</span></a>            <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">exon</span><span class="p">:</span>  <span class="c1"># TODO</span>
</span><span id="HGVSName-1088"><a href="#HGVSName-1088"><span class="linenos">1088</span></a>                <span class="n">prefix</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;exon&quot;</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">exon</span><span class="p">))</span>
</span><span id="HGVSName-1089"><a href="#HGVSName-1089"><span class="linenos">1089</span></a>            <span class="k">return</span> <span class="s2">&quot;:&quot;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">prefix</span><span class="p">)</span>
</span><span id="HGVSName-1090"><a href="#HGVSName-1090"><span class="linenos">1090</span></a>
</span><span id="HGVSName-1091"><a href="#HGVSName-1091"><span class="linenos">1091</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName-1092"><a href="#HGVSName-1092"><span class="linenos">1092</span></a>
</span><span id="HGVSName-1093"><a href="#HGVSName-1093"><span class="linenos">1093</span></a>            <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">kind</span> <span class="ow">in</span> <span class="p">(</span><span class="s2">&quot;g&quot;</span><span class="p">,</span> <span class="s2">&quot;m&quot;</span><span class="p">):</span>
</span><span id="HGVSName-1094"><a href="#HGVSName-1094"><span class="linenos">1094</span></a>                <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">chrom</span><span class="p">:</span>
</span><span id="HGVSName-1095"><a href="#HGVSName-1095"><span class="linenos">1095</span></a>                    <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">chrom</span>
</span><span id="HGVSName-1096"><a href="#HGVSName-1096"><span class="linenos">1096</span></a>
</span><span id="HGVSName-1097"><a href="#HGVSName-1097"><span class="linenos">1097</span></a>            <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span><span class="p">:</span>
</span><span id="HGVSName-1098"><a href="#HGVSName-1098"><span class="linenos">1098</span></a>                <span class="k">if</span> <span class="n">use_protein</span> <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">transcript_protein</span><span class="p">:</span>
</span><span id="HGVSName-1099"><a href="#HGVSName-1099"><span class="linenos">1099</span></a>                    <span class="n">transcript</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">transcript_protein</span>
</span><span id="HGVSName-1100"><a href="#HGVSName-1100"><span class="linenos">1100</span></a>                <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName-1101"><a href="#HGVSName-1101"><span class="linenos">1101</span></a>                    <span class="n">transcript</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span>
</span><span id="HGVSName-1102"><a href="#HGVSName-1102"><span class="linenos">1102</span></a>                <span class="k">if</span> <span class="ow">not</span> <span class="n">use_version</span> <span class="ow">and</span> <span class="n">transcript</span><span class="p">:</span>
</span><span id="HGVSName-1103"><a href="#HGVSName-1103"><span class="linenos">1103</span></a>                    <span class="n">transcript</span> <span class="o">=</span> <span class="n">transcript</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot;.&quot;</span><span class="p">)[</span><span class="mi">0</span><span class="p">]</span>
</span><span id="HGVSName-1104"><a href="#HGVSName-1104"><span class="linenos">1104</span></a>                <span class="k">if</span> <span class="n">use_gene</span> <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">gene</span><span class="p">:</span>
</span><span id="HGVSName-1105"><a href="#HGVSName-1105"><span class="linenos">1105</span></a>                    <span class="k">return</span> <span class="s2">&quot;</span><span class="si">%s</span><span class="s2">(</span><span class="si">%s</span><span class="s2">)&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">transcript</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">gene</span><span class="p">)</span>
</span><span id="HGVSName-1106"><a href="#HGVSName-1106"><span class="linenos">1106</span></a>                <span class="k">elif</span> <span class="n">use_exon</span> <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">exon</span><span class="p">:</span>
</span><span id="HGVSName-1107"><a href="#HGVSName-1107"><span class="linenos">1107</span></a>                    <span class="k">return</span> <span class="s2">&quot;</span><span class="si">%s</span><span class="s2">(exon</span><span class="si">%s</span><span class="s2">)&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">transcript</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">exon</span><span class="p">)</span>
</span><span id="HGVSName-1108"><a href="#HGVSName-1108"><span class="linenos">1108</span></a>                <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName-1109"><a href="#HGVSName-1109"><span class="linenos">1109</span></a>                    <span class="k">return</span> <span class="n">transcript</span>
</span><span id="HGVSName-1110"><a href="#HGVSName-1110"><span class="linenos">1110</span></a>            <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName-1111"><a href="#HGVSName-1111"><span class="linenos">1111</span></a>                <span class="k">if</span> <span class="n">use_gene</span><span class="p">:</span>
</span><span id="HGVSName-1112"><a href="#HGVSName-1112"><span class="linenos">1112</span></a>                    <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">gene</span>
</span><span id="HGVSName-1113"><a href="#HGVSName-1113"><span class="linenos">1113</span></a>                <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName-1114"><a href="#HGVSName-1114"><span class="linenos">1114</span></a>                    <span class="k">return</span> <span class="s2">&quot;&quot;</span>
</span><span id="HGVSName-1115"><a href="#HGVSName-1115"><span class="linenos">1115</span></a>
</span><span id="HGVSName-1116"><a href="#HGVSName-1116"><span class="linenos">1116</span></a>    <span class="k">def</span> <span class="nf">format_cdna_coords</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
</span><span id="HGVSName-1117"><a href="#HGVSName-1117"><span class="linenos">1117</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="HGVSName-1118"><a href="#HGVSName-1118"><span class="linenos">1118</span></a><span class="sd">        The function `format_cdna_coords` generates a string representing HGVS cDNA coordinates,</span>
</span><span id="HGVSName-1119"><a href="#HGVSName-1119"><span class="linenos">1119</span></a><span class="sd">        returning either the start coordinate or a string in the format &quot;start_end&quot; depending on whether</span>
</span><span id="HGVSName-1120"><a href="#HGVSName-1120"><span class="linenos">1120</span></a><span class="sd">        the start and end coordinates are the same or not.</span>
</span><span id="HGVSName-1121"><a href="#HGVSName-1121"><span class="linenos">1121</span></a><span class="sd">        :return: a string representing the cDNA coordinates. If the start and end coordinates are the</span>
</span><span id="HGVSName-1122"><a href="#HGVSName-1122"><span class="linenos">1122</span></a><span class="sd">        same, it returns just the start coordinate. Otherwise, it returns a string in the format</span>
</span><span id="HGVSName-1123"><a href="#HGVSName-1123"><span class="linenos">1123</span></a><span class="sd">        &quot;start_end&quot;.</span>
</span><span id="HGVSName-1124"><a href="#HGVSName-1124"><span class="linenos">1124</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="HGVSName-1125"><a href="#HGVSName-1125"><span class="linenos">1125</span></a>
</span><span id="HGVSName-1126"><a href="#HGVSName-1126"><span class="linenos">1126</span></a>        <span class="c1"># Format coordinates.</span>
</span><span id="HGVSName-1127"><a href="#HGVSName-1127"><span class="linenos">1127</span></a>        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">cdna_start</span> <span class="o">==</span> <span class="bp">self</span><span class="o">.</span><span class="n">cdna_end</span><span class="p">:</span>
</span><span id="HGVSName-1128"><a href="#HGVSName-1128"><span class="linenos">1128</span></a>            <span class="k">return</span> <span class="nb">str</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">cdna_start</span><span class="p">)</span>
</span><span id="HGVSName-1129"><a href="#HGVSName-1129"><span class="linenos">1129</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName-1130"><a href="#HGVSName-1130"><span class="linenos">1130</span></a>            <span class="k">return</span> <span class="s2">&quot;</span><span class="si">%s</span><span class="s2">_</span><span class="si">%s</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">cdna_start</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">cdna_end</span><span class="p">)</span>
</span><span id="HGVSName-1131"><a href="#HGVSName-1131"><span class="linenos">1131</span></a>
</span><span id="HGVSName-1132"><a href="#HGVSName-1132"><span class="linenos">1132</span></a>    <span class="k">def</span> <span class="nf">format_dna_allele</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
</span><span id="HGVSName-1133"><a href="#HGVSName-1133"><span class="linenos">1133</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="HGVSName-1134"><a href="#HGVSName-1134"><span class="linenos">1134</span></a><span class="sd">        The function `format_dna_allele` generates an HGVS DNA allele based on the mutation type and</span>
</span><span id="HGVSName-1135"><a href="#HGVSName-1135"><span class="linenos">1135</span></a><span class="sd">        alleles provided.</span>
</span><span id="HGVSName-1136"><a href="#HGVSName-1136"><span class="linenos">1136</span></a><span class="sd">        :return: The function `format_dna_allele` returns a string representing the HGVS DNA allele. The</span>
</span><span id="HGVSName-1137"><a href="#HGVSName-1137"><span class="linenos">1137</span></a><span class="sd">        specific format of the returned string depends on the value of the `mutation_type` attribute of</span>
</span><span id="HGVSName-1138"><a href="#HGVSName-1138"><span class="linenos">1138</span></a><span class="sd">        the object. The possible return values are:</span>
</span><span id="HGVSName-1139"><a href="#HGVSName-1139"><span class="linenos">1139</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="HGVSName-1140"><a href="#HGVSName-1140"><span class="linenos">1140</span></a>
</span><span id="HGVSName-1141"><a href="#HGVSName-1141"><span class="linenos">1141</span></a>        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">==</span> <span class="s2">&quot;=&quot;</span><span class="p">:</span>
</span><span id="HGVSName-1142"><a href="#HGVSName-1142"><span class="linenos">1142</span></a>            <span class="c1"># No change.</span>
</span><span id="HGVSName-1143"><a href="#HGVSName-1143"><span class="linenos">1143</span></a>            <span class="c1"># example: 101A=</span>
</span><span id="HGVSName-1144"><a href="#HGVSName-1144"><span class="linenos">1144</span></a>            <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">+</span> <span class="s2">&quot;=&quot;</span>
</span><span id="HGVSName-1145"><a href="#HGVSName-1145"><span class="linenos">1145</span></a>
</span><span id="HGVSName-1146"><a href="#HGVSName-1146"><span class="linenos">1146</span></a>        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">==</span> <span class="s2">&quot;&gt;&quot;</span><span class="p">:</span>
</span><span id="HGVSName-1147"><a href="#HGVSName-1147"><span class="linenos">1147</span></a>            <span class="c1"># SNP.</span>
</span><span id="HGVSName-1148"><a href="#HGVSName-1148"><span class="linenos">1148</span></a>            <span class="c1"># example: 101A&gt;C</span>
</span><span id="HGVSName-1149"><a href="#HGVSName-1149"><span class="linenos">1149</span></a>            <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">+</span> <span class="s2">&quot;&gt;&quot;</span> <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span>
</span><span id="HGVSName-1150"><a href="#HGVSName-1150"><span class="linenos">1150</span></a>
</span><span id="HGVSName-1151"><a href="#HGVSName-1151"><span class="linenos">1151</span></a>        <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">==</span> <span class="s2">&quot;delins&quot;</span><span class="p">:</span>
</span><span id="HGVSName-1152"><a href="#HGVSName-1152"><span class="linenos">1152</span></a>            <span class="c1"># Indel.</span>
</span><span id="HGVSName-1153"><a href="#HGVSName-1153"><span class="linenos">1153</span></a>            <span class="c1"># example: 112_117delAGGTCAinsTG, 112_117delinsTG</span>
</span><span id="HGVSName-1154"><a href="#HGVSName-1154"><span class="linenos">1154</span></a>            <span class="k">return</span> <span class="s2">&quot;del&quot;</span> <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">+</span> <span class="s2">&quot;ins&quot;</span> <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span>
</span><span id="HGVSName-1155"><a href="#HGVSName-1155"><span class="linenos">1155</span></a>
</span><span id="HGVSName-1156"><a href="#HGVSName-1156"><span class="linenos">1156</span></a>        <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="ow">in</span> <span class="p">(</span><span class="s2">&quot;del&quot;</span><span class="p">,</span> <span class="s2">&quot;dup&quot;</span><span class="p">):</span>
</span><span id="HGVSName-1157"><a href="#HGVSName-1157"><span class="linenos">1157</span></a>            <span class="c1"># Delete, duplication.</span>
</span><span id="HGVSName-1158"><a href="#HGVSName-1158"><span class="linenos">1158</span></a>            <span class="c1"># example: 1000_1003delATG, 1000_1003dupATG</span>
</span><span id="HGVSName-1159"><a href="#HGVSName-1159"><span class="linenos">1159</span></a>            <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span>
</span><span id="HGVSName-1160"><a href="#HGVSName-1160"><span class="linenos">1160</span></a>
</span><span id="HGVSName-1161"><a href="#HGVSName-1161"><span class="linenos">1161</span></a>        <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">==</span> <span class="s2">&quot;ins&quot;</span><span class="p">:</span>
</span><span id="HGVSName-1162"><a href="#HGVSName-1162"><span class="linenos">1162</span></a>            <span class="c1"># Insert.</span>
</span><span id="HGVSName-1163"><a href="#HGVSName-1163"><span class="linenos">1163</span></a>            <span class="c1"># example: 1000_1001insATG</span>
</span><span id="HGVSName-1164"><a href="#HGVSName-1164"><span class="linenos">1164</span></a>            <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span>
</span><span id="HGVSName-1165"><a href="#HGVSName-1165"><span class="linenos">1165</span></a>
</span><span id="HGVSName-1166"><a href="#HGVSName-1166"><span class="linenos">1166</span></a>        <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">==</span> <span class="s2">&quot;inv&quot;</span><span class="p">:</span>
</span><span id="HGVSName-1167"><a href="#HGVSName-1167"><span class="linenos">1167</span></a>            <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span>
</span><span id="HGVSName-1168"><a href="#HGVSName-1168"><span class="linenos">1168</span></a>
</span><span id="HGVSName-1169"><a href="#HGVSName-1169"><span class="linenos">1169</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName-1170"><a href="#HGVSName-1170"><span class="linenos">1170</span></a>            <span class="k">raise</span> <span class="ne">AssertionError</span><span class="p">(</span><span class="s2">&quot;unknown mutation type: &#39;</span><span class="si">%s</span><span class="s2">&#39;&quot;</span> <span class="o">%</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span><span class="p">)</span>
</span><span id="HGVSName-1171"><a href="#HGVSName-1171"><span class="linenos">1171</span></a>
</span><span id="HGVSName-1172"><a href="#HGVSName-1172"><span class="linenos">1172</span></a>    <span class="k">def</span> <span class="nf">format_cdna</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
</span><span id="HGVSName-1173"><a href="#HGVSName-1173"><span class="linenos">1173</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="HGVSName-1174"><a href="#HGVSName-1174"><span class="linenos">1174</span></a><span class="sd">        The function &quot;format_cdna&quot; generates an HGVS cDNA allele by combining the cDNA coordinates and</span>
</span><span id="HGVSName-1175"><a href="#HGVSName-1175"><span class="linenos">1175</span></a><span class="sd">        the DNA allele.</span>
</span><span id="HGVSName-1176"><a href="#HGVSName-1176"><span class="linenos">1176</span></a>
</span><span id="HGVSName-1177"><a href="#HGVSName-1177"><span class="linenos">1177</span></a><span class="sd">        Some examples include:</span>
</span><span id="HGVSName-1178"><a href="#HGVSName-1178"><span class="linenos">1178</span></a><span class="sd">          Substitution: 101A&gt;C,</span>
</span><span id="HGVSName-1179"><a href="#HGVSName-1179"><span class="linenos">1179</span></a><span class="sd">          Indel: 3428delCinsTA, 1000_1003delATG, 1000_1001insATG</span>
</span><span id="HGVSName-1180"><a href="#HGVSName-1180"><span class="linenos">1180</span></a>
</span><span id="HGVSName-1181"><a href="#HGVSName-1181"><span class="linenos">1181</span></a><span class="sd">        :return: a string that represents the HGVS cDNA allele.</span>
</span><span id="HGVSName-1182"><a href="#HGVSName-1182"><span class="linenos">1182</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="HGVSName-1183"><a href="#HGVSName-1183"><span class="linenos">1183</span></a>
</span><span id="HGVSName-1184"><a href="#HGVSName-1184"><span class="linenos">1184</span></a>        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">format_cdna_coords</span><span class="p">()</span> <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">format_dna_allele</span><span class="p">()</span>
</span><span id="HGVSName-1185"><a href="#HGVSName-1185"><span class="linenos">1185</span></a>
</span><span id="HGVSName-1186"><a href="#HGVSName-1186"><span class="linenos">1186</span></a>    <span class="k">def</span> <span class="nf">format_protein</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
</span><span id="HGVSName-1187"><a href="#HGVSName-1187"><span class="linenos">1187</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="HGVSName-1188"><a href="#HGVSName-1188"><span class="linenos">1188</span></a><span class="sd">        The `format_protein` function generates an HGVS protein name based on different scenarios such</span>
</span><span id="HGVSName-1189"><a href="#HGVSName-1189"><span class="linenos">1189</span></a><span class="sd">        as no change, change, frameshift, and range change.</span>
</span><span id="HGVSName-1190"><a href="#HGVSName-1190"><span class="linenos">1190</span></a>
</span><span id="HGVSName-1191"><a href="#HGVSName-1191"><span class="linenos">1191</span></a><span class="sd">        Some examples include:</span>
</span><span id="HGVSName-1192"><a href="#HGVSName-1192"><span class="linenos">1192</span></a><span class="sd">          No change: Glu1161=</span>
</span><span id="HGVSName-1193"><a href="#HGVSName-1193"><span class="linenos">1193</span></a><span class="sd">          Change: Glu1161Ser</span>
</span><span id="HGVSName-1194"><a href="#HGVSName-1194"><span class="linenos">1194</span></a><span class="sd">          Frameshift: Glu1161_Ser1164?fs</span>
</span><span id="HGVSName-1195"><a href="#HGVSName-1195"><span class="linenos">1195</span></a>
</span><span id="HGVSName-1196"><a href="#HGVSName-1196"><span class="linenos">1196</span></a><span class="sd">        :return: The method `format_protein` returns a string representing the HGVS protein name.</span>
</span><span id="HGVSName-1197"><a href="#HGVSName-1197"><span class="linenos">1197</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="HGVSName-1198"><a href="#HGVSName-1198"><span class="linenos">1198</span></a>
</span><span id="HGVSName-1199"><a href="#HGVSName-1199"><span class="linenos">1199</span></a>        <span class="k">if</span> <span class="p">(</span>
</span><span id="HGVSName-1200"><a href="#HGVSName-1200"><span class="linenos">1200</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">start</span> <span class="o">==</span> <span class="bp">self</span><span class="o">.</span><span class="n">end</span>
</span><span id="HGVSName-1201"><a href="#HGVSName-1201"><span class="linenos">1201</span></a>            <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">==</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref2_allele</span> <span class="o">==</span> <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span>
</span><span id="HGVSName-1202"><a href="#HGVSName-1202"><span class="linenos">1202</span></a>        <span class="p">):</span>
</span><span id="HGVSName-1203"><a href="#HGVSName-1203"><span class="linenos">1203</span></a>            <span class="c1"># Match.</span>
</span><span id="HGVSName-1204"><a href="#HGVSName-1204"><span class="linenos">1204</span></a>            <span class="c1"># Example: Glu1161=</span>
</span><span id="HGVSName-1205"><a href="#HGVSName-1205"><span class="linenos">1205</span></a>            <span class="n">pep_extra</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">pep_extra</span> <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">pep_extra</span> <span class="k">else</span> <span class="s2">&quot;=&quot;</span>
</span><span id="HGVSName-1206"><a href="#HGVSName-1206"><span class="linenos">1206</span></a>            <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">start</span><span class="p">)</span> <span class="o">+</span> <span class="n">pep_extra</span>
</span><span id="HGVSName-1207"><a href="#HGVSName-1207"><span class="linenos">1207</span></a>
</span><span id="HGVSName-1208"><a href="#HGVSName-1208"><span class="linenos">1208</span></a>        <span class="k">elif</span> <span class="p">(</span>
</span><span id="HGVSName-1209"><a href="#HGVSName-1209"><span class="linenos">1209</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">start</span> <span class="o">==</span> <span class="bp">self</span><span class="o">.</span><span class="n">end</span>
</span><span id="HGVSName-1210"><a href="#HGVSName-1210"><span class="linenos">1210</span></a>            <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">==</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref2_allele</span>
</span><span id="HGVSName-1211"><a href="#HGVSName-1211"><span class="linenos">1211</span></a>            <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">!=</span> <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span>
</span><span id="HGVSName-1212"><a href="#HGVSName-1212"><span class="linenos">1212</span></a>        <span class="p">):</span>
</span><span id="HGVSName-1213"><a href="#HGVSName-1213"><span class="linenos">1213</span></a>            <span class="c1"># Change.</span>
</span><span id="HGVSName-1214"><a href="#HGVSName-1214"><span class="linenos">1214</span></a>            <span class="c1"># Example: Glu1161Ser</span>
</span><span id="HGVSName-1215"><a href="#HGVSName-1215"><span class="linenos">1215</span></a>            <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">start</span><span class="p">)</span> <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span> <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">pep_extra</span>
</span><span id="HGVSName-1216"><a href="#HGVSName-1216"><span class="linenos">1216</span></a>
</span><span id="HGVSName-1217"><a href="#HGVSName-1217"><span class="linenos">1217</span></a>        <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">start</span> <span class="o">!=</span> <span class="bp">self</span><span class="o">.</span><span class="n">end</span><span class="p">:</span>
</span><span id="HGVSName-1218"><a href="#HGVSName-1218"><span class="linenos">1218</span></a>            <span class="c1"># Range change.</span>
</span><span id="HGVSName-1219"><a href="#HGVSName-1219"><span class="linenos">1219</span></a>            <span class="c1"># Example: Glu1161_Ser1164?fs</span>
</span><span id="HGVSName-1220"><a href="#HGVSName-1220"><span class="linenos">1220</span></a>            <span class="k">return</span> <span class="p">(</span>
</span><span id="HGVSName-1221"><a href="#HGVSName-1221"><span class="linenos">1221</span></a>                <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span>
</span><span id="HGVSName-1222"><a href="#HGVSName-1222"><span class="linenos">1222</span></a>                <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">start</span><span class="p">)</span>
</span><span id="HGVSName-1223"><a href="#HGVSName-1223"><span class="linenos">1223</span></a>                <span class="o">+</span> <span class="s2">&quot;_&quot;</span>
</span><span id="HGVSName-1224"><a href="#HGVSName-1224"><span class="linenos">1224</span></a>                <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref2_allele</span>
</span><span id="HGVSName-1225"><a href="#HGVSName-1225"><span class="linenos">1225</span></a>                <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">end</span><span class="p">)</span>
</span><span id="HGVSName-1226"><a href="#HGVSName-1226"><span class="linenos">1226</span></a>                <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">pep_extra</span>
</span><span id="HGVSName-1227"><a href="#HGVSName-1227"><span class="linenos">1227</span></a>            <span class="p">)</span>
</span><span id="HGVSName-1228"><a href="#HGVSName-1228"><span class="linenos">1228</span></a>        <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">pep_extra</span><span class="p">:</span>
</span><span id="HGVSName-1229"><a href="#HGVSName-1229"><span class="linenos">1229</span></a>            <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">pep_extra</span>
</span><span id="HGVSName-1230"><a href="#HGVSName-1230"><span class="linenos">1230</span></a>
</span><span id="HGVSName-1231"><a href="#HGVSName-1231"><span class="linenos">1231</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName-1232"><a href="#HGVSName-1232"><span class="linenos">1232</span></a>            <span class="k">return</span> <span class="kc">None</span>
</span><span id="HGVSName-1233"><a href="#HGVSName-1233"><span class="linenos">1233</span></a>
</span><span id="HGVSName-1234"><a href="#HGVSName-1234"><span class="linenos">1234</span></a>    <span class="k">def</span> <span class="nf">format_coords</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
</span><span id="HGVSName-1235"><a href="#HGVSName-1235"><span class="linenos">1235</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="HGVSName-1236"><a href="#HGVSName-1236"><span class="linenos">1236</span></a><span class="sd">        The function `format_coords` generates a string representation of HGVS cDNA coordinates.</span>
</span><span id="HGVSName-1237"><a href="#HGVSName-1237"><span class="linenos">1237</span></a><span class="sd">        :return: a string that represents the HGVS cDNA coordinates. If the start and end coordinates</span>
</span><span id="HGVSName-1238"><a href="#HGVSName-1238"><span class="linenos">1238</span></a><span class="sd">        are the same, it returns just the start coordinate. Otherwise, it returns a string in the format</span>
</span><span id="HGVSName-1239"><a href="#HGVSName-1239"><span class="linenos">1239</span></a><span class="sd">        &quot;start_end&quot;.</span>
</span><span id="HGVSName-1240"><a href="#HGVSName-1240"><span class="linenos">1240</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="HGVSName-1241"><a href="#HGVSName-1241"><span class="linenos">1241</span></a>
</span><span id="HGVSName-1242"><a href="#HGVSName-1242"><span class="linenos">1242</span></a>        <span class="c1"># Format coordinates.</span>
</span><span id="HGVSName-1243"><a href="#HGVSName-1243"><span class="linenos">1243</span></a>        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">start</span> <span class="o">==</span> <span class="bp">self</span><span class="o">.</span><span class="n">end</span><span class="p">:</span>
</span><span id="HGVSName-1244"><a href="#HGVSName-1244"><span class="linenos">1244</span></a>            <span class="k">return</span> <span class="nb">str</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">start</span><span class="p">)</span>
</span><span id="HGVSName-1245"><a href="#HGVSName-1245"><span class="linenos">1245</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName-1246"><a href="#HGVSName-1246"><span class="linenos">1246</span></a>            <span class="k">return</span> <span class="s2">&quot;</span><span class="si">%s</span><span class="s2">_</span><span class="si">%s</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">start</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">end</span><span class="p">)</span>
</span><span id="HGVSName-1247"><a href="#HGVSName-1247"><span class="linenos">1247</span></a>
</span><span id="HGVSName-1248"><a href="#HGVSName-1248"><span class="linenos">1248</span></a>    <span class="k">def</span> <span class="nf">format_genome</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
</span><span id="HGVSName-1249"><a href="#HGVSName-1249"><span class="linenos">1249</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="HGVSName-1250"><a href="#HGVSName-1250"><span class="linenos">1250</span></a><span class="sd">        The function &quot;format_genome&quot; generates an HGVS genomic allele by combining the formatted</span>
</span><span id="HGVSName-1251"><a href="#HGVSName-1251"><span class="linenos">1251</span></a><span class="sd">        coordinates and DNA allele.</span>
</span><span id="HGVSName-1252"><a href="#HGVSName-1252"><span class="linenos">1252</span></a>
</span><span id="HGVSName-1253"><a href="#HGVSName-1253"><span class="linenos">1253</span></a><span class="sd">        Some examples include:</span>
</span><span id="HGVSName-1254"><a href="#HGVSName-1254"><span class="linenos">1254</span></a><span class="sd">          Substitution: 1000100A&gt;T</span>
</span><span id="HGVSName-1255"><a href="#HGVSName-1255"><span class="linenos">1255</span></a><span class="sd">          Indel: 1000100_1000102delATG</span>
</span><span id="HGVSName-1256"><a href="#HGVSName-1256"><span class="linenos">1256</span></a>
</span><span id="HGVSName-1257"><a href="#HGVSName-1257"><span class="linenos">1257</span></a><span class="sd">        :return: a string that represents the HGVS genomic allele.</span>
</span><span id="HGVSName-1258"><a href="#HGVSName-1258"><span class="linenos">1258</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="HGVSName-1259"><a href="#HGVSName-1259"><span class="linenos">1259</span></a>
</span><span id="HGVSName-1260"><a href="#HGVSName-1260"><span class="linenos">1260</span></a>        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">format_coords</span><span class="p">()</span> <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">format_dna_allele</span><span class="p">()</span>
</span><span id="HGVSName-1261"><a href="#HGVSName-1261"><span class="linenos">1261</span></a>
</span><span id="HGVSName-1262"><a href="#HGVSName-1262"><span class="linenos">1262</span></a>    <span class="k">def</span> <span class="nf">get_raw_coords</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">transcript</span><span class="p">:</span> <span class="nb">object</span> <span class="o">=</span> <span class="kc">None</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">tuple</span><span class="p">:</span>
</span><span id="HGVSName-1263"><a href="#HGVSName-1263"><span class="linenos">1263</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="HGVSName-1264"><a href="#HGVSName-1264"><span class="linenos">1264</span></a><span class="sd">        The function `get_raw_coords` returns the genomic coordinates based on the given transcript or</span>
</span><span id="HGVSName-1265"><a href="#HGVSName-1265"><span class="linenos">1265</span></a><span class="sd">        the provided chromosomal coordinates.</span>
</span><span id="HGVSName-1266"><a href="#HGVSName-1266"><span class="linenos">1266</span></a>
</span><span id="HGVSName-1267"><a href="#HGVSName-1267"><span class="linenos">1267</span></a><span class="sd">        :param transcript: The `transcript` parameter is an object that represents a transcript. It is</span>
</span><span id="HGVSName-1268"><a href="#HGVSName-1268"><span class="linenos">1268</span></a><span class="sd">        used to retrieve genomic coordinates based on the type of HGVS name (`self.kind`). The</span>
</span><span id="HGVSName-1269"><a href="#HGVSName-1269"><span class="linenos">1269</span></a><span class="sd">        `transcript` object should have the following attributes and methods:</span>
</span><span id="HGVSName-1270"><a href="#HGVSName-1270"><span class="linenos">1270</span></a><span class="sd">        :type transcript: object</span>
</span><span id="HGVSName-1271"><a href="#HGVSName-1271"><span class="linenos">1271</span></a><span class="sd">        :return: a tuple containing the genomic coordinates. The tuple consists of three elements: the</span>
</span><span id="HGVSName-1272"><a href="#HGVSName-1272"><span class="linenos">1272</span></a><span class="sd">        chromosome, the start position, and the end position.</span>
</span><span id="HGVSName-1273"><a href="#HGVSName-1273"><span class="linenos">1273</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="HGVSName-1274"><a href="#HGVSName-1274"><span class="linenos">1274</span></a>
</span><span id="HGVSName-1275"><a href="#HGVSName-1275"><span class="linenos">1275</span></a>        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">kind</span> <span class="ow">in</span> <span class="p">(</span><span class="s2">&quot;c&quot;</span><span class="p">,</span> <span class="s2">&quot;n&quot;</span><span class="p">):</span>
</span><span id="HGVSName-1276"><a href="#HGVSName-1276"><span class="linenos">1276</span></a>            <span class="n">chrom</span> <span class="o">=</span> <span class="n">transcript</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom</span>
</span><span id="HGVSName-1277"><a href="#HGVSName-1277"><span class="linenos">1277</span></a>            <span class="n">start</span> <span class="o">=</span> <span class="n">transcript</span><span class="o">.</span><span class="n">cdna_to_genomic_coord</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">cdna_start</span><span class="p">)</span>
</span><span id="HGVSName-1278"><a href="#HGVSName-1278"><span class="linenos">1278</span></a>            <span class="n">end</span> <span class="o">=</span> <span class="n">transcript</span><span class="o">.</span><span class="n">cdna_to_genomic_coord</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">cdna_end</span><span class="p">)</span>
</span><span id="HGVSName-1279"><a href="#HGVSName-1279"><span class="linenos">1279</span></a>
</span><span id="HGVSName-1280"><a href="#HGVSName-1280"><span class="linenos">1280</span></a>            <span class="k">if</span> <span class="ow">not</span> <span class="n">transcript</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">is_forward_strand</span><span class="p">:</span>
</span><span id="HGVSName-1281"><a href="#HGVSName-1281"><span class="linenos">1281</span></a>                <span class="n">start</span><span class="p">,</span> <span class="n">end</span> <span class="o">=</span> <span class="n">end</span><span class="p">,</span> <span class="n">start</span>
</span><span id="HGVSName-1282"><a href="#HGVSName-1282"><span class="linenos">1282</span></a>
</span><span id="HGVSName-1283"><a href="#HGVSName-1283"><span class="linenos">1283</span></a>            <span class="k">if</span> <span class="n">start</span> <span class="o">&gt;</span> <span class="n">end</span><span class="p">:</span>
</span><span id="HGVSName-1284"><a href="#HGVSName-1284"><span class="linenos">1284</span></a>                <span class="k">raise</span> <span class="ne">AssertionError</span><span class="p">(</span><span class="s2">&quot;cdna_start cannot be greater than cdna_end&quot;</span><span class="p">)</span>
</span><span id="HGVSName-1285"><a href="#HGVSName-1285"><span class="linenos">1285</span></a>        <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">kind</span> <span class="ow">in</span> <span class="p">(</span><span class="s2">&quot;g&quot;</span><span class="p">,</span> <span class="s2">&quot;m&quot;</span><span class="p">):</span>
</span><span id="HGVSName-1286"><a href="#HGVSName-1286"><span class="linenos">1286</span></a>            <span class="n">chrom</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">chrom</span>
</span><span id="HGVSName-1287"><a href="#HGVSName-1287"><span class="linenos">1287</span></a>            <span class="n">start</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">start</span>
</span><span id="HGVSName-1288"><a href="#HGVSName-1288"><span class="linenos">1288</span></a>            <span class="n">end</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">end</span>
</span><span id="HGVSName-1289"><a href="#HGVSName-1289"><span class="linenos">1289</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName-1290"><a href="#HGVSName-1290"><span class="linenos">1290</span></a>            <span class="k">raise</span> <span class="ne">NotImplementedError</span><span class="p">(</span>
</span><span id="HGVSName-1291"><a href="#HGVSName-1291"><span class="linenos">1291</span></a>                <span class="s1">&#39;Coordinates are not available for this kind of HGVS name &quot;</span><span class="si">%s</span><span class="s1">&quot;&#39;</span>
</span><span id="HGVSName-1292"><a href="#HGVSName-1292"><span class="linenos">1292</span></a>                <span class="o">%</span> <span class="bp">self</span><span class="o">.</span><span class="n">kind</span>
</span><span id="HGVSName-1293"><a href="#HGVSName-1293"><span class="linenos">1293</span></a>            <span class="p">)</span>
</span><span id="HGVSName-1294"><a href="#HGVSName-1294"><span class="linenos">1294</span></a>
</span><span id="HGVSName-1295"><a href="#HGVSName-1295"><span class="linenos">1295</span></a>        <span class="c1"># Check coordinate span is equal to reference bases</span>
</span><span id="HGVSName-1296"><a href="#HGVSName-1296"><span class="linenos">1296</span></a>        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span><span class="p">:</span>
</span><span id="HGVSName-1297"><a href="#HGVSName-1297"><span class="linenos">1297</span></a>            <span class="n">coordinate_span</span> <span class="o">=</span> <span class="n">end</span> <span class="o">-</span> <span class="n">start</span> <span class="o">+</span> <span class="mi">1</span>  <span class="c1"># Ref will always be &gt;=1 base</span>
</span><span id="HGVSName-1298"><a href="#HGVSName-1298"><span class="linenos">1298</span></a>            <span class="n">ref_length</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span><span class="p">)</span>
</span><span id="HGVSName-1299"><a href="#HGVSName-1299"><span class="linenos">1299</span></a>            <span class="k">if</span> <span class="n">coordinate_span</span> <span class="o">!=</span> <span class="n">ref_length</span><span class="p">:</span>
</span><span id="HGVSName-1300"><a href="#HGVSName-1300"><span class="linenos">1300</span></a>                <span class="k">raise</span> <span class="n">InvalidHGVSName</span><span class="p">(</span>
</span><span id="HGVSName-1301"><a href="#HGVSName-1301"><span class="linenos">1301</span></a>                    <span class="s2">&quot;Coordinate span (</span><span class="si">%d</span><span class="s2">) not equal to ref length </span><span class="si">%d</span><span class="s2">&quot;</span>
</span><span id="HGVSName-1302"><a href="#HGVSName-1302"><span class="linenos">1302</span></a>                    <span class="o">%</span> <span class="p">(</span><span class="n">coordinate_span</span><span class="p">,</span> <span class="n">ref_length</span><span class="p">)</span>
</span><span id="HGVSName-1303"><a href="#HGVSName-1303"><span class="linenos">1303</span></a>                <span class="p">)</span>
</span><span id="HGVSName-1304"><a href="#HGVSName-1304"><span class="linenos">1304</span></a>
</span><span id="HGVSName-1305"><a href="#HGVSName-1305"><span class="linenos">1305</span></a>        <span class="k">return</span> <span class="n">chrom</span><span class="p">,</span> <span class="n">start</span><span class="p">,</span> <span class="n">end</span>
</span><span id="HGVSName-1306"><a href="#HGVSName-1306"><span class="linenos">1306</span></a>
</span><span id="HGVSName-1307"><a href="#HGVSName-1307"><span class="linenos">1307</span></a>    <span class="k">def</span> <span class="nf">get_ref_coords</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">transcript</span><span class="p">:</span> <span class="nb">object</span> <span class="o">=</span> <span class="kc">None</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">tuple</span><span class="p">:</span>
</span><span id="HGVSName-1308"><a href="#HGVSName-1308"><span class="linenos">1308</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="HGVSName-1309"><a href="#HGVSName-1309"><span class="linenos">1309</span></a><span class="sd">        The function &quot;get_ref_coords&quot; returns the genomic coordinates of the reference allele, taking</span>
</span><span id="HGVSName-1310"><a href="#HGVSName-1310"><span class="linenos">1310</span></a><span class="sd">        into account different mutation types.</span>
</span><span id="HGVSName-1311"><a href="#HGVSName-1311"><span class="linenos">1311</span></a>
</span><span id="HGVSName-1312"><a href="#HGVSName-1312"><span class="linenos">1312</span></a><span class="sd">        :param transcript: The `transcript` parameter is an optional object that represents a transcript</span>
</span><span id="HGVSName-1313"><a href="#HGVSName-1313"><span class="linenos">1313</span></a><span class="sd">        or gene. It is used to retrieve the genomic coordinates of the reference allele</span>
</span><span id="HGVSName-1314"><a href="#HGVSName-1314"><span class="linenos">1314</span></a><span class="sd">        :type transcript: object</span>
</span><span id="HGVSName-1315"><a href="#HGVSName-1315"><span class="linenos">1315</span></a><span class="sd">        :return: a tuple containing the genomic coordinates of the reference allele. The tuple consists</span>
</span><span id="HGVSName-1316"><a href="#HGVSName-1316"><span class="linenos">1316</span></a><span class="sd">        of three elements: the chromosome, the start position, and the end position.</span>
</span><span id="HGVSName-1317"><a href="#HGVSName-1317"><span class="linenos">1317</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="HGVSName-1318"><a href="#HGVSName-1318"><span class="linenos">1318</span></a>
</span><span id="HGVSName-1319"><a href="#HGVSName-1319"><span class="linenos">1319</span></a>        <span class="n">chrom</span><span class="p">,</span> <span class="n">start</span><span class="p">,</span> <span class="n">end</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">get_raw_coords</span><span class="p">(</span><span class="n">transcript</span><span class="p">)</span>
</span><span id="HGVSName-1320"><a href="#HGVSName-1320"><span class="linenos">1320</span></a>
</span><span id="HGVSName-1321"><a href="#HGVSName-1321"><span class="linenos">1321</span></a>        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">==</span> <span class="s2">&quot;ins&quot;</span><span class="p">:</span>
</span><span id="HGVSName-1322"><a href="#HGVSName-1322"><span class="linenos">1322</span></a>            <span class="c1"># Inserts have empty interval.</span>
</span><span id="HGVSName-1323"><a href="#HGVSName-1323"><span class="linenos">1323</span></a>            <span class="k">if</span> <span class="n">start</span> <span class="o">&lt;</span> <span class="n">end</span><span class="p">:</span>
</span><span id="HGVSName-1324"><a href="#HGVSName-1324"><span class="linenos">1324</span></a>                <span class="n">start</span> <span class="o">+=</span> <span class="mi">1</span>
</span><span id="HGVSName-1325"><a href="#HGVSName-1325"><span class="linenos">1325</span></a>                <span class="n">end</span> <span class="o">-=</span> <span class="mi">1</span>
</span><span id="HGVSName-1326"><a href="#HGVSName-1326"><span class="linenos">1326</span></a>            <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName-1327"><a href="#HGVSName-1327"><span class="linenos">1327</span></a>                <span class="n">end</span> <span class="o">=</span> <span class="n">start</span> <span class="o">-</span> <span class="mi">1</span>
</span><span id="HGVSName-1328"><a href="#HGVSName-1328"><span class="linenos">1328</span></a>
</span><span id="HGVSName-1329"><a href="#HGVSName-1329"><span class="linenos">1329</span></a>        <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">==</span> <span class="s2">&quot;dup&quot;</span><span class="p">:</span>
</span><span id="HGVSName-1330"><a href="#HGVSName-1330"><span class="linenos">1330</span></a>            <span class="n">end</span> <span class="o">=</span> <span class="n">start</span> <span class="o">-</span> <span class="mi">1</span>
</span><span id="HGVSName-1331"><a href="#HGVSName-1331"><span class="linenos">1331</span></a>        <span class="k">return</span> <span class="n">chrom</span><span class="p">,</span> <span class="n">start</span><span class="p">,</span> <span class="n">end</span>
</span><span id="HGVSName-1332"><a href="#HGVSName-1332"><span class="linenos">1332</span></a>
</span><span id="HGVSName-1333"><a href="#HGVSName-1333"><span class="linenos">1333</span></a>    <span class="k">def</span> <span class="nf">get_vcf_coords</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">transcript</span><span class="p">:</span> <span class="nb">object</span> <span class="o">=</span> <span class="kc">None</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">tuple</span><span class="p">:</span>
</span><span id="HGVSName-1334"><a href="#HGVSName-1334"><span class="linenos">1334</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="HGVSName-1335"><a href="#HGVSName-1335"><span class="linenos">1335</span></a><span class="sd">        The function &quot;get_vcf_coords&quot; returns the genomic coordinates of the reference allele in</span>
</span><span id="HGVSName-1336"><a href="#HGVSName-1336"><span class="linenos">1336</span></a><span class="sd">        VCF-style, with left-padding for indels.</span>
</span><span id="HGVSName-1337"><a href="#HGVSName-1337"><span class="linenos">1337</span></a>
</span><span id="HGVSName-1338"><a href="#HGVSName-1338"><span class="linenos">1338</span></a><span class="sd">        :param transcript: The `transcript` parameter is an object that represents a transcript or gene.</span>
</span><span id="HGVSName-1339"><a href="#HGVSName-1339"><span class="linenos">1339</span></a><span class="sd">        It is used to retrieve the genomic coordinates of the reference allele</span>
</span><span id="HGVSName-1340"><a href="#HGVSName-1340"><span class="linenos">1340</span></a><span class="sd">        :type transcript: object</span>
</span><span id="HGVSName-1341"><a href="#HGVSName-1341"><span class="linenos">1341</span></a><span class="sd">        :return: a tuple containing the genomic coordinates of the reference allele in VCF-style. The</span>
</span><span id="HGVSName-1342"><a href="#HGVSName-1342"><span class="linenos">1342</span></a><span class="sd">        tuple consists of three elements: the chromosome, the start position, and the end position.</span>
</span><span id="HGVSName-1343"><a href="#HGVSName-1343"><span class="linenos">1343</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="HGVSName-1344"><a href="#HGVSName-1344"><span class="linenos">1344</span></a>
</span><span id="HGVSName-1345"><a href="#HGVSName-1345"><span class="linenos">1345</span></a>        <span class="n">chrom</span><span class="p">,</span> <span class="n">start</span><span class="p">,</span> <span class="n">end</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">get_ref_coords</span><span class="p">(</span><span class="n">transcript</span><span class="p">)</span>
</span><span id="HGVSName-1346"><a href="#HGVSName-1346"><span class="linenos">1346</span></a>
</span><span id="HGVSName-1347"><a href="#HGVSName-1347"><span class="linenos">1347</span></a>        <span class="c1"># Inserts and deletes require left-padding by 1 base</span>
</span><span id="HGVSName-1348"><a href="#HGVSName-1348"><span class="linenos">1348</span></a>        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="ow">in</span> <span class="p">(</span><span class="s2">&quot;=&quot;</span><span class="p">,</span> <span class="s2">&quot;&gt;&quot;</span><span class="p">):</span>
</span><span id="HGVSName-1349"><a href="#HGVSName-1349"><span class="linenos">1349</span></a>            <span class="k">pass</span>
</span><span id="HGVSName-1350"><a href="#HGVSName-1350"><span class="linenos">1350</span></a>        <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="ow">in</span> <span class="p">(</span><span class="s2">&quot;del&quot;</span><span class="p">,</span> <span class="s2">&quot;ins&quot;</span><span class="p">,</span> <span class="s2">&quot;dup&quot;</span><span class="p">,</span> <span class="s2">&quot;delins&quot;</span><span class="p">):</span>
</span><span id="HGVSName-1351"><a href="#HGVSName-1351"><span class="linenos">1351</span></a>            <span class="c1"># Indels have left-padding.</span>
</span><span id="HGVSName-1352"><a href="#HGVSName-1352"><span class="linenos">1352</span></a>            <span class="n">start</span> <span class="o">-=</span> <span class="mi">1</span>
</span><span id="HGVSName-1353"><a href="#HGVSName-1353"><span class="linenos">1353</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName-1354"><a href="#HGVSName-1354"><span class="linenos">1354</span></a>            <span class="k">raise</span> <span class="ne">NotImplementedError</span><span class="p">(</span><span class="s2">&quot;Unknown mutation_type &#39;</span><span class="si">%s</span><span class="s2">&#39;&quot;</span> <span class="o">%</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span><span class="p">)</span>
</span><span id="HGVSName-1355"><a href="#HGVSName-1355"><span class="linenos">1355</span></a>        <span class="k">return</span> <span class="n">chrom</span><span class="p">,</span> <span class="n">start</span><span class="p">,</span> <span class="n">end</span>
</span><span id="HGVSName-1356"><a href="#HGVSName-1356"><span class="linenos">1356</span></a>
</span><span id="HGVSName-1357"><a href="#HGVSName-1357"><span class="linenos">1357</span></a>    <span class="k">def</span> <span class="nf">get_ref_alt</span><span class="p">(</span>
</span><span id="HGVSName-1358"><a href="#HGVSName-1358"><span class="linenos">1358</span></a>        <span class="bp">self</span><span class="p">,</span> <span class="n">is_forward_strand</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span> <span class="n">raw_dup_alleles</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span>
</span><span id="HGVSName-1359"><a href="#HGVSName-1359"><span class="linenos">1359</span></a>    <span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">tuple</span><span class="p">:</span>
</span><span id="HGVSName-1360"><a href="#HGVSName-1360"><span class="linenos">1360</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="HGVSName-1361"><a href="#HGVSName-1361"><span class="linenos">1361</span></a><span class="sd">        The function `get_ref_alt` returns the reference and alternate alleles, with an option to modify</span>
</span><span id="HGVSName-1362"><a href="#HGVSName-1362"><span class="linenos">1362</span></a><span class="sd">        duplications to look like inserts.</span>
</span><span id="HGVSName-1363"><a href="#HGVSName-1363"><span class="linenos">1363</span></a>
</span><span id="HGVSName-1364"><a href="#HGVSName-1364"><span class="linenos">1364</span></a><span class="sd">        :param is_forward_strand: The parameter `is_forward_strand` is a boolean flag that indicates</span>
</span><span id="HGVSName-1365"><a href="#HGVSName-1365"><span class="linenos">1365</span></a><span class="sd">        whether the alleles should be returned for the forward strand or the reverse complement strand.</span>
</span><span id="HGVSName-1366"><a href="#HGVSName-1366"><span class="linenos">1366</span></a><span class="sd">        If `is_forward_strand` is `True`, the alleles will be returned as is. If `is_forward_strand` is</span>
</span><span id="HGVSName-1367"><a href="#HGVSName-1367"><span class="linenos">1367</span></a><span class="sd">        `False`,, defaults to True</span>
</span><span id="HGVSName-1368"><a href="#HGVSName-1368"><span class="linenos">1368</span></a><span class="sd">        :type is_forward_strand: bool (optional)</span>
</span><span id="HGVSName-1369"><a href="#HGVSName-1369"><span class="linenos">1369</span></a><span class="sd">        :param raw_dup_alleles: The `raw_dup_alleles` parameter is a boolean flag that determines</span>
</span><span id="HGVSName-1370"><a href="#HGVSName-1370"><span class="linenos">1370</span></a><span class="sd">        whether the raw values of duplicated alleles should be returned. By default, it is set to</span>
</span><span id="HGVSName-1371"><a href="#HGVSName-1371"><span class="linenos">1371</span></a><span class="sd">        `False`, which means that if the mutation type is a duplication (`dup`), the reference allele</span>
</span><span id="HGVSName-1372"><a href="#HGVSName-1372"><span class="linenos">1372</span></a><span class="sd">        will be represented as an empty string, defaults to False</span>
</span><span id="HGVSName-1373"><a href="#HGVSName-1373"><span class="linenos">1373</span></a><span class="sd">        :type raw_dup_alleles: bool (optional)</span>
</span><span id="HGVSName-1374"><a href="#HGVSName-1374"><span class="linenos">1374</span></a><span class="sd">        :return: The function `get_ref_alt` returns a tuple containing the reference and alternate</span>
</span><span id="HGVSName-1375"><a href="#HGVSName-1375"><span class="linenos">1375</span></a><span class="sd">        alleles.</span>
</span><span id="HGVSName-1376"><a href="#HGVSName-1376"><span class="linenos">1376</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="HGVSName-1377"><a href="#HGVSName-1377"><span class="linenos">1377</span></a>
</span><span id="HGVSName-1378"><a href="#HGVSName-1378"><span class="linenos">1378</span></a>        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">kind</span> <span class="o">==</span> <span class="s2">&quot;p&quot;</span><span class="p">:</span>
</span><span id="HGVSName-1379"><a href="#HGVSName-1379"><span class="linenos">1379</span></a>            <span class="k">raise</span> <span class="ne">NotImplementedError</span><span class="p">(</span>
</span><span id="HGVSName-1380"><a href="#HGVSName-1380"><span class="linenos">1380</span></a>                <span class="s2">&quot;get_ref_alt is not implemented for protein HGVS names&quot;</span>
</span><span id="HGVSName-1381"><a href="#HGVSName-1381"><span class="linenos">1381</span></a>            <span class="p">)</span>
</span><span id="HGVSName-1382"><a href="#HGVSName-1382"><span class="linenos">1382</span></a>        <span class="n">alleles</span> <span class="o">=</span> <span class="p">[</span><span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span><span class="p">]</span>
</span><span id="HGVSName-1383"><a href="#HGVSName-1383"><span class="linenos">1383</span></a>
</span><span id="HGVSName-1384"><a href="#HGVSName-1384"><span class="linenos">1384</span></a>        <span class="c1"># Represent duplications are inserts.</span>
</span><span id="HGVSName-1385"><a href="#HGVSName-1385"><span class="linenos">1385</span></a>        <span class="k">if</span> <span class="ow">not</span> <span class="n">raw_dup_alleles</span> <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">==</span> <span class="s2">&quot;dup&quot;</span><span class="p">:</span>
</span><span id="HGVSName-1386"><a href="#HGVSName-1386"><span class="linenos">1386</span></a>            <span class="n">alleles</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span>
</span><span id="HGVSName-1387"><a href="#HGVSName-1387"><span class="linenos">1387</span></a>            <span class="n">alleles</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="o">=</span> <span class="n">alleles</span><span class="p">[</span><span class="mi">1</span><span class="p">][:</span> <span class="nb">len</span><span class="p">(</span><span class="n">alleles</span><span class="p">[</span><span class="mi">1</span><span class="p">])</span> <span class="o">//</span> <span class="mi">2</span><span class="p">]</span>
</span><span id="HGVSName-1388"><a href="#HGVSName-1388"><span class="linenos">1388</span></a>
</span><span id="HGVSName-1389"><a href="#HGVSName-1389"><span class="linenos">1389</span></a>        <span class="k">if</span> <span class="n">is_forward_strand</span><span class="p">:</span>
</span><span id="HGVSName-1390"><a href="#HGVSName-1390"><span class="linenos">1390</span></a>            <span class="k">return</span> <span class="n">alleles</span>
</span><span id="HGVSName-1391"><a href="#HGVSName-1391"><span class="linenos">1391</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName-1392"><a href="#HGVSName-1392"><span class="linenos">1392</span></a>            <span class="k">return</span> <span class="nb">tuple</span><span class="p">(</span><span class="nb">map</span><span class="p">(</span><span class="n">revcomp</span><span class="p">,</span> <span class="n">alleles</span><span class="p">))</span>
</span></pre></div>


            <div class="docstring"><p>Represents a HGVS variant name.</p>
</div>


                            <div id="HGVSName.__init__" class="classattr">
                                        <input id="HGVSName.__init__-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr function">
            
        <span class="name">HGVSName</span><span class="signature pdoc-code multiline">(<span class="param">	<span class="n">name</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s1">&#39;&#39;</span>,</span><span class="param">	<span class="n">prefix</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s1">&#39;&#39;</span>,</span><span class="param">	<span class="n">chrom</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s1">&#39;&#39;</span>,</span><span class="param">	<span class="n">transcript</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s1">&#39;&#39;</span>,</span><span class="param">	<span class="n">transcript_protein</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="kc">None</span>,</span><span class="param">	<span class="n">gene</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s1">&#39;&#39;</span>,</span><span class="param">	<span class="n">exon</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="kc">None</span>,</span><span class="param">	<span class="n">kind</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s1">&#39;&#39;</span>,</span><span class="param">	<span class="n">mutation_type</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="kc">None</span>,</span><span class="param">	<span class="n">start</span><span class="p">:</span> <span class="nb">int</span> <span class="o">=</span> <span class="mi">0</span>,</span><span class="param">	<span class="n">end</span><span class="p">:</span> <span class="nb">int</span> <span class="o">=</span> <span class="mi">0</span>,</span><span class="param">	<span class="n">ref_allele</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s1">&#39;&#39;</span>,</span><span class="param">	<span class="n">ref2_allele</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s1">&#39;&#39;</span>,</span><span class="param">	<span class="n">alt_allele</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s1">&#39;&#39;</span>,</span><span class="param">	<span class="n">cdna_start</span><span class="p">:</span> <span class="nb">int</span> <span class="o">=</span> <span class="kc">None</span>,</span><span class="param">	<span class="n">cdna_end</span><span class="p">:</span> <span class="nb">int</span> <span class="o">=</span> <span class="kc">None</span>,</span><span class="param">	<span class="n">pep_extra</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s1">&#39;&#39;</span></span>)</span>

                <label class="view-source-button" for="HGVSName.__init__-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#HGVSName.__init__"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="HGVSName.__init__-522"><a href="#HGVSName.__init__-522"><span class="linenos">522</span></a>    <span class="k">def</span> <span class="fm">__init__</span><span class="p">(</span>
</span><span id="HGVSName.__init__-523"><a href="#HGVSName.__init__-523"><span class="linenos">523</span></a>        <span class="bp">self</span><span class="p">,</span>
</span><span id="HGVSName.__init__-524"><a href="#HGVSName.__init__-524"><span class="linenos">524</span></a>        <span class="n">name</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span><span class="p">,</span>
</span><span id="HGVSName.__init__-525"><a href="#HGVSName.__init__-525"><span class="linenos">525</span></a>        <span class="n">prefix</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span><span class="p">,</span>
</span><span id="HGVSName.__init__-526"><a href="#HGVSName.__init__-526"><span class="linenos">526</span></a>        <span class="n">chrom</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span><span class="p">,</span>
</span><span id="HGVSName.__init__-527"><a href="#HGVSName.__init__-527"><span class="linenos">527</span></a>        <span class="n">transcript</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span><span class="p">,</span>
</span><span id="HGVSName.__init__-528"><a href="#HGVSName.__init__-528"><span class="linenos">528</span></a>        <span class="n">transcript_protein</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
</span><span id="HGVSName.__init__-529"><a href="#HGVSName.__init__-529"><span class="linenos">529</span></a>        <span class="n">gene</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span><span class="p">,</span>
</span><span id="HGVSName.__init__-530"><a href="#HGVSName.__init__-530"><span class="linenos">530</span></a>        <span class="n">exon</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
</span><span id="HGVSName.__init__-531"><a href="#HGVSName.__init__-531"><span class="linenos">531</span></a>        <span class="n">kind</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span><span class="p">,</span>
</span><span id="HGVSName.__init__-532"><a href="#HGVSName.__init__-532"><span class="linenos">532</span></a>        <span class="n">mutation_type</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
</span><span id="HGVSName.__init__-533"><a href="#HGVSName.__init__-533"><span class="linenos">533</span></a>        <span class="n">start</span><span class="p">:</span> <span class="nb">int</span> <span class="o">=</span> <span class="mi">0</span><span class="p">,</span>
</span><span id="HGVSName.__init__-534"><a href="#HGVSName.__init__-534"><span class="linenos">534</span></a>        <span class="n">end</span><span class="p">:</span> <span class="nb">int</span> <span class="o">=</span> <span class="mi">0</span><span class="p">,</span>
</span><span id="HGVSName.__init__-535"><a href="#HGVSName.__init__-535"><span class="linenos">535</span></a>        <span class="n">ref_allele</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span><span class="p">,</span>
</span><span id="HGVSName.__init__-536"><a href="#HGVSName.__init__-536"><span class="linenos">536</span></a>        <span class="n">ref2_allele</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span><span class="p">,</span>
</span><span id="HGVSName.__init__-537"><a href="#HGVSName.__init__-537"><span class="linenos">537</span></a>        <span class="n">alt_allele</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span><span class="p">,</span>
</span><span id="HGVSName.__init__-538"><a href="#HGVSName.__init__-538"><span class="linenos">538</span></a>        <span class="n">cdna_start</span><span class="p">:</span> <span class="nb">int</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
</span><span id="HGVSName.__init__-539"><a href="#HGVSName.__init__-539"><span class="linenos">539</span></a>        <span class="n">cdna_end</span><span class="p">:</span> <span class="nb">int</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
</span><span id="HGVSName.__init__-540"><a href="#HGVSName.__init__-540"><span class="linenos">540</span></a>        <span class="n">pep_extra</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span><span class="p">,</span>
</span><span id="HGVSName.__init__-541"><a href="#HGVSName.__init__-541"><span class="linenos">541</span></a>    <span class="p">):</span>
</span><span id="HGVSName.__init__-542"><a href="#HGVSName.__init__-542"><span class="linenos">542</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="HGVSName.__init__-543"><a href="#HGVSName.__init__-543"><span class="linenos">543</span></a><span class="sd">        The function is a constructor that initializes various attributes of an object and parses a</span>
</span><span id="HGVSName.__init__-544"><a href="#HGVSName.__init__-544"><span class="linenos">544</span></a><span class="sd">        given name to populate those attributes.</span>
</span><span id="HGVSName.__init__-545"><a href="#HGVSName.__init__-545"><span class="linenos">545</span></a>
</span><span id="HGVSName.__init__-546"><a href="#HGVSName.__init__-546"><span class="linenos">546</span></a><span class="sd">        :param name: The full HGVS name of the variant</span>
</span><span id="HGVSName.__init__-547"><a href="#HGVSName.__init__-547"><span class="linenos">547</span></a><span class="sd">        :type name: str</span>
</span><span id="HGVSName.__init__-548"><a href="#HGVSName.__init__-548"><span class="linenos">548</span></a><span class="sd">        :param prefix: The `prefix` parameter is a string that is used as a prefix for the HGVS name. It</span>
</span><span id="HGVSName.__init__-549"><a href="#HGVSName.__init__-549"><span class="linenos">549</span></a><span class="sd">        can be used to indicate additional information or context about the variant</span>
</span><span id="HGVSName.__init__-550"><a href="#HGVSName.__init__-550"><span class="linenos">550</span></a><span class="sd">        :type prefix: str</span>
</span><span id="HGVSName.__init__-551"><a href="#HGVSName.__init__-551"><span class="linenos">551</span></a><span class="sd">        :param chrom: The `chrom` parameter represents the chromosome where the mutation occurs. It is a</span>
</span><span id="HGVSName.__init__-552"><a href="#HGVSName.__init__-552"><span class="linenos">552</span></a><span class="sd">        string that specifies the chromosome number or identifier</span>
</span><span id="HGVSName.__init__-553"><a href="#HGVSName.__init__-553"><span class="linenos">553</span></a><span class="sd">        :type chrom: str</span>
</span><span id="HGVSName.__init__-554"><a href="#HGVSName.__init__-554"><span class="linenos">554</span></a><span class="sd">        :param transcript: The `transcript` parameter represents the transcript ID or name associated</span>
</span><span id="HGVSName.__init__-555"><a href="#HGVSName.__init__-555"><span class="linenos">555</span></a><span class="sd">        with the mutation. It is used to specify the specific transcript in which the mutation occurs</span>
</span><span id="HGVSName.__init__-556"><a href="#HGVSName.__init__-556"><span class="linenos">556</span></a><span class="sd">        :type transcript: str</span>
</span><span id="HGVSName.__init__-557"><a href="#HGVSName.__init__-557"><span class="linenos">557</span></a><span class="sd">        :param transcript_protein: The `transcript_protein` parameter is used to store information about</span>
</span><span id="HGVSName.__init__-558"><a href="#HGVSName.__init__-558"><span class="linenos">558</span></a><span class="sd">        the protein associated with the transcript. It can be used to specify the protein variant or</span>
</span><span id="HGVSName.__init__-559"><a href="#HGVSName.__init__-559"><span class="linenos">559</span></a><span class="sd">        isoform that is affected by the mutation</span>
</span><span id="HGVSName.__init__-560"><a href="#HGVSName.__init__-560"><span class="linenos">560</span></a><span class="sd">        :type transcript_protein: str</span>
</span><span id="HGVSName.__init__-561"><a href="#HGVSName.__init__-561"><span class="linenos">561</span></a><span class="sd">        :param gene: The &quot;gene&quot; parameter represents the gene associated with the variant. It is a</span>
</span><span id="HGVSName.__init__-562"><a href="#HGVSName.__init__-562"><span class="linenos">562</span></a><span class="sd">        string that specifies the gene name or identifier</span>
</span><span id="HGVSName.__init__-563"><a href="#HGVSName.__init__-563"><span class="linenos">563</span></a><span class="sd">        :type gene: str</span>
</span><span id="HGVSName.__init__-564"><a href="#HGVSName.__init__-564"><span class="linenos">564</span></a><span class="sd">        :param exon: The `exon` parameter represents the exon number or range in which the mutation</span>
</span><span id="HGVSName.__init__-565"><a href="#HGVSName.__init__-565"><span class="linenos">565</span></a><span class="sd">        occurs. It is used to specify the location of the mutation within the transcript</span>
</span><span id="HGVSName.__init__-566"><a href="#HGVSName.__init__-566"><span class="linenos">566</span></a><span class="sd">        :type exon: str</span>
</span><span id="HGVSName.__init__-567"><a href="#HGVSName.__init__-567"><span class="linenos">567</span></a><span class="sd">        :param kind: The &quot;kind&quot; parameter is used to specify the type of variant or mutation. It can be</span>
</span><span id="HGVSName.__init__-568"><a href="#HGVSName.__init__-568"><span class="linenos">568</span></a><span class="sd">        a string that represents the kind of mutation, such as &quot;substitution&quot;, &quot;deletion&quot;, &quot;insertion&quot;,</span>
</span><span id="HGVSName.__init__-569"><a href="#HGVSName.__init__-569"><span class="linenos">569</span></a><span class="sd">        etc. This parameter helps to categorize and describe the type of mutation being represented by</span>
</span><span id="HGVSName.__init__-570"><a href="#HGVSName.__init__-570"><span class="linenos">570</span></a><span class="sd">        the</span>
</span><span id="HGVSName.__init__-571"><a href="#HGVSName.__init__-571"><span class="linenos">571</span></a><span class="sd">        :type kind: str</span>
</span><span id="HGVSName.__init__-572"><a href="#HGVSName.__init__-572"><span class="linenos">572</span></a><span class="sd">        :param mutation_type: The `mutation_type` parameter is used to specify the type of mutation. It</span>
</span><span id="HGVSName.__init__-573"><a href="#HGVSName.__init__-573"><span class="linenos">573</span></a><span class="sd">        can be a string that represents the type of mutation, such as &quot;SNP&quot; (single nucleotide</span>
</span><span id="HGVSName.__init__-574"><a href="#HGVSName.__init__-574"><span class="linenos">574</span></a><span class="sd">        polymorphism), &quot;DEL&quot; (deletion), &quot;INS&quot; (insertion), etc</span>
</span><span id="HGVSName.__init__-575"><a href="#HGVSName.__init__-575"><span class="linenos">575</span></a><span class="sd">        :type mutation_type: str</span>
</span><span id="HGVSName.__init__-576"><a href="#HGVSName.__init__-576"><span class="linenos">576</span></a><span class="sd">        :param start: The `start` parameter represents the starting position of the mutation or variant</span>
</span><span id="HGVSName.__init__-577"><a href="#HGVSName.__init__-577"><span class="linenos">577</span></a><span class="sd">        in the genomic sequence. It is an integer value that indicates the position of the mutation or</span>
</span><span id="HGVSName.__init__-578"><a href="#HGVSName.__init__-578"><span class="linenos">578</span></a><span class="sd">        variant on the genomic sequence. If not provided, it defaults to 0, defaults to 0</span>
</span><span id="HGVSName.__init__-579"><a href="#HGVSName.__init__-579"><span class="linenos">579</span></a><span class="sd">        :type start: int (optional)</span>
</span><span id="HGVSName.__init__-580"><a href="#HGVSName.__init__-580"><span class="linenos">580</span></a><span class="sd">        :param end: The &quot;end&quot; parameter represents the end position of the mutation or variant. It is an</span>
</span><span id="HGVSName.__init__-581"><a href="#HGVSName.__init__-581"><span class="linenos">581</span></a><span class="sd">        integer value that indicates the position of the mutation or variant on the genomic sequence,</span>
</span><span id="HGVSName.__init__-582"><a href="#HGVSName.__init__-582"><span class="linenos">582</span></a><span class="sd">        defaults to 0</span>
</span><span id="HGVSName.__init__-583"><a href="#HGVSName.__init__-583"><span class="linenos">583</span></a><span class="sd">        :type end: int (optional)</span>
</span><span id="HGVSName.__init__-584"><a href="#HGVSName.__init__-584"><span class="linenos">584</span></a><span class="sd">        :param ref_allele: The `ref_allele` parameter represents the reference allele in a genetic</span>
</span><span id="HGVSName.__init__-585"><a href="#HGVSName.__init__-585"><span class="linenos">585</span></a><span class="sd">        mutation. It is the allele that is present in the reference genome at a specific position</span>
</span><span id="HGVSName.__init__-586"><a href="#HGVSName.__init__-586"><span class="linenos">586</span></a><span class="sd">        :type ref_allele: str</span>
</span><span id="HGVSName.__init__-587"><a href="#HGVSName.__init__-587"><span class="linenos">587</span></a><span class="sd">        :param ref2_allele: The `ref2_allele` parameter represents the reference allele at the end of a</span>
</span><span id="HGVSName.__init__-588"><a href="#HGVSName.__init__-588"><span class="linenos">588</span></a><span class="sd">        peptide indel. In the context of genetic mutations, an indel refers to the insertion or deletion</span>
</span><span id="HGVSName.__init__-589"><a href="#HGVSName.__init__-589"><span class="linenos">589</span></a><span class="sd">        of nucleotides in a DNA sequence. The `ref2_allele` specifically represents the reference allele</span>
</span><span id="HGVSName.__init__-590"><a href="#HGVSName.__init__-590"><span class="linenos">590</span></a><span class="sd">        that is</span>
</span><span id="HGVSName.__init__-591"><a href="#HGVSName.__init__-591"><span class="linenos">591</span></a><span class="sd">        :type ref2_allele: str</span>
</span><span id="HGVSName.__init__-592"><a href="#HGVSName.__init__-592"><span class="linenos">592</span></a><span class="sd">        :param alt_allele: The `alt_allele` parameter represents the alternate allele in a genetic</span>
</span><span id="HGVSName.__init__-593"><a href="#HGVSName.__init__-593"><span class="linenos">593</span></a><span class="sd">        mutation. In genetics, an allele is one of the possible forms of a gene. In the context of this</span>
</span><span id="HGVSName.__init__-594"><a href="#HGVSName.__init__-594"><span class="linenos">594</span></a><span class="sd">        code, `alt_allele` is used to store the alternate allele that is present in a mutation</span>
</span><span id="HGVSName.__init__-595"><a href="#HGVSName.__init__-595"><span class="linenos">595</span></a><span class="sd">        :type alt_allele: str</span>
</span><span id="HGVSName.__init__-596"><a href="#HGVSName.__init__-596"><span class="linenos">596</span></a><span class="sd">        :param cdna_start: The `cdna_start` parameter is used to specify the start position of the</span>
</span><span id="HGVSName.__init__-597"><a href="#HGVSName.__init__-597"><span class="linenos">597</span></a><span class="sd">        mutation in the cDNA sequence. It is an optional parameter and if not provided, it will be set</span>
</span><span id="HGVSName.__init__-598"><a href="#HGVSName.__init__-598"><span class="linenos">598</span></a><span class="sd">        to a default value of `CDNACoord()`</span>
</span><span id="HGVSName.__init__-599"><a href="#HGVSName.__init__-599"><span class="linenos">599</span></a><span class="sd">        :type cdna_start: int</span>
</span><span id="HGVSName.__init__-600"><a href="#HGVSName.__init__-600"><span class="linenos">600</span></a><span class="sd">        :param cdna_end: The `cdna_end` parameter is used to store the end coordinate of the cDNA</span>
</span><span id="HGVSName.__init__-601"><a href="#HGVSName.__init__-601"><span class="linenos">601</span></a><span class="sd">        (complementary DNA) sequence. It is an optional parameter and if not provided, it will be</span>
</span><span id="HGVSName.__init__-602"><a href="#HGVSName.__init__-602"><span class="linenos">602</span></a><span class="sd">        initialized as a `CDNACoord` object. The `CDNACoord` object is likely a</span>
</span><span id="HGVSName.__init__-603"><a href="#HGVSName.__init__-603"><span class="linenos">603</span></a><span class="sd">        :type cdna_end: int</span>
</span><span id="HGVSName.__init__-604"><a href="#HGVSName.__init__-604"><span class="linenos">604</span></a><span class="sd">        :param pep_extra: The `pep_extra` parameter is a string that represents any additional</span>
</span><span id="HGVSName.__init__-605"><a href="#HGVSName.__init__-605"><span class="linenos">605</span></a><span class="sd">        information related to the protein. It is used in the context of protein-specific fields</span>
</span><span id="HGVSName.__init__-606"><a href="#HGVSName.__init__-606"><span class="linenos">606</span></a><span class="sd">        :type pep_extra: str</span>
</span><span id="HGVSName.__init__-607"><a href="#HGVSName.__init__-607"><span class="linenos">607</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="HGVSName.__init__-608"><a href="#HGVSName.__init__-608"><span class="linenos">608</span></a>
</span><span id="HGVSName.__init__-609"><a href="#HGVSName.__init__-609"><span class="linenos">609</span></a>        <span class="c1"># Full HGVS name.</span>
</span><span id="HGVSName.__init__-610"><a href="#HGVSName.__init__-610"><span class="linenos">610</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">name</span> <span class="o">=</span> <span class="n">name</span>
</span><span id="HGVSName.__init__-611"><a href="#HGVSName.__init__-611"><span class="linenos">611</span></a>
</span><span id="HGVSName.__init__-612"><a href="#HGVSName.__init__-612"><span class="linenos">612</span></a>        <span class="c1"># Name parts.</span>
</span><span id="HGVSName.__init__-613"><a href="#HGVSName.__init__-613"><span class="linenos">613</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">prefix</span> <span class="o">=</span> <span class="n">prefix</span>
</span><span id="HGVSName.__init__-614"><a href="#HGVSName.__init__-614"><span class="linenos">614</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">chrom</span> <span class="o">=</span> <span class="n">chrom</span>
</span><span id="HGVSName.__init__-615"><a href="#HGVSName.__init__-615"><span class="linenos">615</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span> <span class="o">=</span> <span class="n">transcript</span>
</span><span id="HGVSName.__init__-616"><a href="#HGVSName.__init__-616"><span class="linenos">616</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">transcript_protein</span> <span class="o">=</span> <span class="n">transcript_protein</span>
</span><span id="HGVSName.__init__-617"><a href="#HGVSName.__init__-617"><span class="linenos">617</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">gene</span> <span class="o">=</span> <span class="n">gene</span>
</span><span id="HGVSName.__init__-618"><a href="#HGVSName.__init__-618"><span class="linenos">618</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">exon</span> <span class="o">=</span> <span class="n">exon</span>
</span><span id="HGVSName.__init__-619"><a href="#HGVSName.__init__-619"><span class="linenos">619</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">kind</span> <span class="o">=</span> <span class="n">kind</span>
</span><span id="HGVSName.__init__-620"><a href="#HGVSName.__init__-620"><span class="linenos">620</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">=</span> <span class="n">mutation_type</span>
</span><span id="HGVSName.__init__-621"><a href="#HGVSName.__init__-621"><span class="linenos">621</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">start</span> <span class="o">=</span> <span class="n">start</span>
</span><span id="HGVSName.__init__-622"><a href="#HGVSName.__init__-622"><span class="linenos">622</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">end</span> <span class="o">=</span> <span class="n">end</span>
</span><span id="HGVSName.__init__-623"><a href="#HGVSName.__init__-623"><span class="linenos">623</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">=</span> <span class="n">ref_allele</span>  <span class="c1"># reference allele</span>
</span><span id="HGVSName.__init__-624"><a href="#HGVSName.__init__-624"><span class="linenos">624</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">ref2_allele</span> <span class="o">=</span> <span class="n">ref2_allele</span>  <span class="c1"># reference allele at end of pep indel</span>
</span><span id="HGVSName.__init__-625"><a href="#HGVSName.__init__-625"><span class="linenos">625</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span> <span class="o">=</span> <span class="n">alt_allele</span>  <span class="c1"># alternate allele</span>
</span><span id="HGVSName.__init__-626"><a href="#HGVSName.__init__-626"><span class="linenos">626</span></a>
</span><span id="HGVSName.__init__-627"><a href="#HGVSName.__init__-627"><span class="linenos">627</span></a>        <span class="c1"># cDNA-specific fields</span>
</span><span id="HGVSName.__init__-628"><a href="#HGVSName.__init__-628"><span class="linenos">628</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">cdna_start</span> <span class="o">=</span> <span class="n">cdna_start</span> <span class="k">if</span> <span class="n">cdna_start</span> <span class="k">else</span> <span class="n">CDNACoord</span><span class="p">()</span>
</span><span id="HGVSName.__init__-629"><a href="#HGVSName.__init__-629"><span class="linenos">629</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">cdna_end</span> <span class="o">=</span> <span class="n">cdna_end</span> <span class="k">if</span> <span class="n">cdna_end</span> <span class="k">else</span> <span class="n">CDNACoord</span><span class="p">()</span>
</span><span id="HGVSName.__init__-630"><a href="#HGVSName.__init__-630"><span class="linenos">630</span></a>
</span><span id="HGVSName.__init__-631"><a href="#HGVSName.__init__-631"><span class="linenos">631</span></a>        <span class="c1"># Protein-specific fields</span>
</span><span id="HGVSName.__init__-632"><a href="#HGVSName.__init__-632"><span class="linenos">632</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">pep_extra</span> <span class="o">=</span> <span class="n">pep_extra</span>
</span><span id="HGVSName.__init__-633"><a href="#HGVSName.__init__-633"><span class="linenos">633</span></a>
</span><span id="HGVSName.__init__-634"><a href="#HGVSName.__init__-634"><span class="linenos">634</span></a>        <span class="k">if</span> <span class="n">name</span><span class="p">:</span>
</span><span id="HGVSName.__init__-635"><a href="#HGVSName.__init__-635"><span class="linenos">635</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">parse</span><span class="p">(</span><span class="n">name</span><span class="p">)</span>
</span></pre></div>


            <div class="docstring"><p>The function is a constructor that initializes various attributes of an object and parses a
given name to populate those attributes.</p>

<h6 id="parameters">Parameters</h6>

<ul>
<li><strong>name</strong>:  The full HGVS name of the variant</li>
<li><strong>prefix</strong>:  The <code><a href="#HGVSName.prefix">prefix</a></code> parameter is a string that is used as a prefix for the HGVS name. It
can be used to indicate additional information or context about the variant</li>
<li><strong>chrom</strong>:  The <code><a href="#HGVSName.chrom">chrom</a></code> parameter represents the chromosome where the mutation occurs. It is a
string that specifies the chromosome number or identifier</li>
<li><strong>transcript</strong>:  The <code><a href="#HGVSName.transcript">transcript</a></code> parameter represents the transcript ID or name associated
with the mutation. It is used to specify the specific transcript in which the mutation occurs</li>
<li><strong>transcript_protein</strong>:  The <code><a href="#HGVSName.transcript_protein">transcript_protein</a></code> parameter is used to store information about
the protein associated with the transcript. It can be used to specify the protein variant or
isoform that is affected by the mutation</li>
<li><strong>gene</strong>:  The "gene" parameter represents the gene associated with the variant. It is a
string that specifies the gene name or identifier</li>
<li><strong>exon</strong>:  The <code><a href="#HGVSName.exon">exon</a></code> parameter represents the exon number or range in which the mutation
occurs. It is used to specify the location of the mutation within the transcript</li>
<li><strong>kind</strong>:  The "kind" parameter is used to specify the type of variant or mutation. It can be
a string that represents the kind of mutation, such as "substitution", "deletion", "insertion",
etc. This parameter helps to categorize and describe the type of mutation being represented by
the</li>
<li><strong>mutation_type</strong>:  The <code><a href="#HGVSName.mutation_type">mutation_type</a></code> parameter is used to specify the type of mutation. It
can be a string that represents the type of mutation, such as "SNP" (single nucleotide
polymorphism), "DEL" (deletion), "INS" (insertion), etc</li>
<li><strong>start</strong>:  The <code><a href="#HGVSName.start">start</a></code> parameter represents the starting position of the mutation or variant
in the genomic sequence. It is an integer value that indicates the position of the mutation or
variant on the genomic sequence. If not provided, it defaults to 0, defaults to 0</li>
<li><strong>end</strong>:  The "end" parameter represents the end position of the mutation or variant. It is an
integer value that indicates the position of the mutation or variant on the genomic sequence,
defaults to 0</li>
<li><strong>ref_allele</strong>:  The <code><a href="#HGVSName.ref_allele">ref_allele</a></code> parameter represents the reference allele in a genetic
mutation. It is the allele that is present in the reference genome at a specific position</li>
<li><strong>ref2_allele</strong>:  The <code><a href="#HGVSName.ref2_allele">ref2_allele</a></code> parameter represents the reference allele at the end of a
peptide indel. In the context of genetic mutations, an indel refers to the insertion or deletion
of nucleotides in a DNA sequence. The <code><a href="#HGVSName.ref2_allele">ref2_allele</a></code> specifically represents the reference allele
that is</li>
<li><strong>alt_allele</strong>:  The <code><a href="#HGVSName.alt_allele">alt_allele</a></code> parameter represents the alternate allele in a genetic
mutation. In genetics, an allele is one of the possible forms of a gene. In the context of this
code, <code><a href="#HGVSName.alt_allele">alt_allele</a></code> is used to store the alternate allele that is present in a mutation</li>
<li><strong>cdna_start</strong>:  The <code><a href="#HGVSName.cdna_start">cdna_start</a></code> parameter is used to specify the start position of the
mutation in the cDNA sequence. It is an optional parameter and if not provided, it will be set
to a default value of <code>CDNACoord()</code></li>
<li><strong>cdna_end</strong>:  The <code><a href="#HGVSName.cdna_end">cdna_end</a></code> parameter is used to store the end coordinate of the cDNA
(complementary DNA) sequence. It is an optional parameter and if not provided, it will be
initialized as a <code>CDNACoord</code> object. The <code>CDNACoord</code> object is likely a</li>
<li><strong>pep_extra</strong>:  The <code><a href="#HGVSName.pep_extra">pep_extra</a></code> parameter is a string that represents any additional
information related to the protein. It is used in the context of protein-specific fields</li>
</ul>
</div>


                            </div>
                            <div id="HGVSName.name" class="classattr">
                                <div class="attr variable">
            <span class="name">name</span>

        
    </div>
    <a class="headerlink" href="#HGVSName.name"></a>
    
    

                            </div>
                            <div id="HGVSName.prefix" class="classattr">
                                <div class="attr variable">
            <span class="name">prefix</span>

        
    </div>
    <a class="headerlink" href="#HGVSName.prefix"></a>
    
    

                            </div>
                            <div id="HGVSName.chrom" class="classattr">
                                <div class="attr variable">
            <span class="name">chrom</span>

        
    </div>
    <a class="headerlink" href="#HGVSName.chrom"></a>
    
    

                            </div>
                            <div id="HGVSName.transcript" class="classattr">
                                <div class="attr variable">
            <span class="name">transcript</span>

        
    </div>
    <a class="headerlink" href="#HGVSName.transcript"></a>
    
    

                            </div>
                            <div id="HGVSName.transcript_protein" class="classattr">
                                <div class="attr variable">
            <span class="name">transcript_protein</span>

        
    </div>
    <a class="headerlink" href="#HGVSName.transcript_protein"></a>
    
    

                            </div>
                            <div id="HGVSName.gene" class="classattr">
                                <div class="attr variable">
            <span class="name">gene</span>

        
    </div>
    <a class="headerlink" href="#HGVSName.gene"></a>
    
    

                            </div>
                            <div id="HGVSName.exon" class="classattr">
                                <div class="attr variable">
            <span class="name">exon</span>

        
    </div>
    <a class="headerlink" href="#HGVSName.exon"></a>
    
    

                            </div>
                            <div id="HGVSName.kind" class="classattr">
                                <div class="attr variable">
            <span class="name">kind</span>

        
    </div>
    <a class="headerlink" href="#HGVSName.kind"></a>
    
    

                            </div>
                            <div id="HGVSName.mutation_type" class="classattr">
                                <div class="attr variable">
            <span class="name">mutation_type</span>

        
    </div>
    <a class="headerlink" href="#HGVSName.mutation_type"></a>
    
    

                            </div>
                            <div id="HGVSName.start" class="classattr">
                                <div class="attr variable">
            <span class="name">start</span>

        
    </div>
    <a class="headerlink" href="#HGVSName.start"></a>
    
    

                            </div>
                            <div id="HGVSName.end" class="classattr">
                                <div class="attr variable">
            <span class="name">end</span>

        
    </div>
    <a class="headerlink" href="#HGVSName.end"></a>
    
    

                            </div>
                            <div id="HGVSName.ref_allele" class="classattr">
                                <div class="attr variable">
            <span class="name">ref_allele</span>

        
    </div>
    <a class="headerlink" href="#HGVSName.ref_allele"></a>
    
    

                            </div>
                            <div id="HGVSName.ref2_allele" class="classattr">
                                <div class="attr variable">
            <span class="name">ref2_allele</span>

        
    </div>
    <a class="headerlink" href="#HGVSName.ref2_allele"></a>
    
    

                            </div>
                            <div id="HGVSName.alt_allele" class="classattr">
                                <div class="attr variable">
            <span class="name">alt_allele</span>

        
    </div>
    <a class="headerlink" href="#HGVSName.alt_allele"></a>
    
    

                            </div>
                            <div id="HGVSName.cdna_start" class="classattr">
                                <div class="attr variable">
            <span class="name">cdna_start</span>

        
    </div>
    <a class="headerlink" href="#HGVSName.cdna_start"></a>
    
    

                            </div>
                            <div id="HGVSName.cdna_end" class="classattr">
                                <div class="attr variable">
            <span class="name">cdna_end</span>

        
    </div>
    <a class="headerlink" href="#HGVSName.cdna_end"></a>
    
    

                            </div>
                            <div id="HGVSName.pep_extra" class="classattr">
                                <div class="attr variable">
            <span class="name">pep_extra</span>

        
    </div>
    <a class="headerlink" href="#HGVSName.pep_extra"></a>
    
    

                            </div>
                            <div id="HGVSName.parse" class="classattr">
                                        <input id="HGVSName.parse-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr function">
            
        <span class="def">def</span>
        <span class="name">parse</span><span class="signature pdoc-code condensed">(<span class="param"><span class="bp">self</span>, </span><span class="param"><span class="n">name</span><span class="p">:</span> <span class="nb">str</span></span><span class="return-annotation">) -> <span class="kc">None</span>:</span></span>

                <label class="view-source-button" for="HGVSName.parse-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#HGVSName.parse"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="HGVSName.parse-637"><a href="#HGVSName.parse-637"><span class="linenos">637</span></a>    <span class="k">def</span> <span class="nf">parse</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">name</span><span class="p">:</span> <span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
</span><span id="HGVSName.parse-638"><a href="#HGVSName.parse-638"><span class="linenos">638</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="HGVSName.parse-639"><a href="#HGVSName.parse-639"><span class="linenos">639</span></a><span class="sd">        The `parse` function is used to split an HGVS name into a prefix and allele, and then validate</span>
</span><span id="HGVSName.parse-640"><a href="#HGVSName.parse-640"><span class="linenos">640</span></a><span class="sd">        the parsed components.</span>
</span><span id="HGVSName.parse-641"><a href="#HGVSName.parse-641"><span class="linenos">641</span></a>
</span><span id="HGVSName.parse-642"><a href="#HGVSName.parse-642"><span class="linenos">642</span></a><span class="sd">        :param name: The `name` parameter is a string that represents an HGVS name. It is the input to</span>
</span><span id="HGVSName.parse-643"><a href="#HGVSName.parse-643"><span class="linenos">643</span></a><span class="sd">        the `parse` function and is used to parse the HGVS name by splitting it into a prefix and allele</span>
</span><span id="HGVSName.parse-644"><a href="#HGVSName.parse-644"><span class="linenos">644</span></a><span class="sd">        :type name: str</span>
</span><span id="HGVSName.parse-645"><a href="#HGVSName.parse-645"><span class="linenos">645</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="HGVSName.parse-646"><a href="#HGVSName.parse-646"><span class="linenos">646</span></a>
</span><span id="HGVSName.parse-647"><a href="#HGVSName.parse-647"><span class="linenos">647</span></a>        <span class="c1"># Does HGVS name have transcript/gene prefix?</span>
</span><span id="HGVSName.parse-648"><a href="#HGVSName.parse-648"><span class="linenos">648</span></a>        <span class="k">if</span> <span class="s2">&quot;:&quot;</span> <span class="ow">in</span> <span class="n">name</span><span class="p">:</span>
</span><span id="HGVSName.parse-649"><a href="#HGVSName.parse-649"><span class="linenos">649</span></a>            <span class="n">prefix</span><span class="p">,</span> <span class="n">allele</span> <span class="o">=</span> <span class="n">name</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot;:&quot;</span><span class="p">,</span> <span class="mi">1</span><span class="p">)</span>
</span><span id="HGVSName.parse-650"><a href="#HGVSName.parse-650"><span class="linenos">650</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName.parse-651"><a href="#HGVSName.parse-651"><span class="linenos">651</span></a>            <span class="n">prefix</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span>
</span><span id="HGVSName.parse-652"><a href="#HGVSName.parse-652"><span class="linenos">652</span></a>            <span class="n">allele</span> <span class="o">=</span> <span class="n">name</span>
</span><span id="HGVSName.parse-653"><a href="#HGVSName.parse-653"><span class="linenos">653</span></a>
</span><span id="HGVSName.parse-654"><a href="#HGVSName.parse-654"><span class="linenos">654</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">name</span> <span class="o">=</span> <span class="n">name</span>
</span><span id="HGVSName.parse-655"><a href="#HGVSName.parse-655"><span class="linenos">655</span></a>
</span><span id="HGVSName.parse-656"><a href="#HGVSName.parse-656"><span class="linenos">656</span></a>        <span class="c1"># Parse prefix and allele.</span>
</span><span id="HGVSName.parse-657"><a href="#HGVSName.parse-657"><span class="linenos">657</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">parse_allele</span><span class="p">(</span><span class="n">allele</span><span class="p">)</span>
</span><span id="HGVSName.parse-658"><a href="#HGVSName.parse-658"><span class="linenos">658</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">parse_prefix</span><span class="p">(</span><span class="n">prefix</span><span class="p">)</span>
</span><span id="HGVSName.parse-659"><a href="#HGVSName.parse-659"><span class="linenos">659</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">_validate</span><span class="p">()</span>
</span></pre></div>


            <div class="docstring"><p>The <code><a href="#HGVSName.parse">parse</a></code> function is used to split an HGVS name into a prefix and allele, and then validate
the parsed components.</p>

<h6 id="parameters">Parameters</h6>

<ul>
<li><strong>name</strong>:  The <code><a href="#HGVSName.name">name</a></code> parameter is a string that represents an HGVS name. It is the input to
the <code><a href="#HGVSName.parse">parse</a></code> function and is used to parse the HGVS name by splitting it into a prefix and allele</li>
</ul>
</div>


                            </div>
                            <div id="HGVSName.parse_prefix" class="classattr">
                                        <input id="HGVSName.parse_prefix-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr function">
            
        <span class="def">def</span>
        <span class="name">parse_prefix</span><span class="signature pdoc-code condensed">(<span class="param"><span class="bp">self</span>, </span><span class="param"><span class="n">prefix</span><span class="p">:</span> <span class="nb">str</span></span><span class="return-annotation">):</span></span>

                <label class="view-source-button" for="HGVSName.parse_prefix-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#HGVSName.parse_prefix"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="HGVSName.parse_prefix-661"><a href="#HGVSName.parse_prefix-661"><span class="linenos">661</span></a>    <span class="k">def</span> <span class="nf">parse_prefix</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">prefix</span><span class="p">:</span> <span class="nb">str</span><span class="p">):</span>
</span><span id="HGVSName.parse_prefix-662"><a href="#HGVSName.parse_prefix-662"><span class="linenos">662</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="HGVSName.parse_prefix-663"><a href="#HGVSName.parse_prefix-663"><span class="linenos">663</span></a><span class="sd">        The `parse_prefix` function is used to parse a HGVS prefix (gene/transcript/chromosome) and</span>
</span><span id="HGVSName.parse_prefix-664"><a href="#HGVSName.parse_prefix-664"><span class="linenos">664</span></a><span class="sd">        assign the parsed values to the corresponding attributes of the object.</span>
</span><span id="HGVSName.parse_prefix-665"><a href="#HGVSName.parse_prefix-665"><span class="linenos">665</span></a>
</span><span id="HGVSName.parse_prefix-666"><a href="#HGVSName.parse_prefix-666"><span class="linenos">666</span></a><span class="sd">        :param prefix: The `prefix` parameter is a string that represents a HGVS prefix, which can be a</span>
</span><span id="HGVSName.parse_prefix-667"><a href="#HGVSName.parse_prefix-667"><span class="linenos">667</span></a><span class="sd">        gene, transcript, or chromosome identifier. It is used to determine the type of prefix and</span>
</span><span id="HGVSName.parse_prefix-668"><a href="#HGVSName.parse_prefix-668"><span class="linenos">668</span></a><span class="sd">        assign the parsed values to the corresponding attributes of the object</span>
</span><span id="HGVSName.parse_prefix-669"><a href="#HGVSName.parse_prefix-669"><span class="linenos">669</span></a><span class="sd">        :type prefix: str</span>
</span><span id="HGVSName.parse_prefix-670"><a href="#HGVSName.parse_prefix-670"><span class="linenos">670</span></a><span class="sd">        :return: The function `parse_prefix` returns the parsed values for the transcript and gene</span>
</span><span id="HGVSName.parse_prefix-671"><a href="#HGVSName.parse_prefix-671"><span class="linenos">671</span></a><span class="sd">        attributes, or sets the chrom or gene attributes based on the given prefix.</span>
</span><span id="HGVSName.parse_prefix-672"><a href="#HGVSName.parse_prefix-672"><span class="linenos">672</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="HGVSName.parse_prefix-673"><a href="#HGVSName.parse_prefix-673"><span class="linenos">673</span></a>
</span><span id="HGVSName.parse_prefix-674"><a href="#HGVSName.parse_prefix-674"><span class="linenos">674</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">prefix</span> <span class="o">=</span> <span class="n">prefix</span>
</span><span id="HGVSName.parse_prefix-675"><a href="#HGVSName.parse_prefix-675"><span class="linenos">675</span></a>
</span><span id="HGVSName.parse_prefix-676"><a href="#HGVSName.parse_prefix-676"><span class="linenos">676</span></a>        <span class="c1"># No prefix.</span>
</span><span id="HGVSName.parse_prefix-677"><a href="#HGVSName.parse_prefix-677"><span class="linenos">677</span></a>        <span class="k">if</span> <span class="n">prefix</span> <span class="o">==</span> <span class="s2">&quot;&quot;</span><span class="p">:</span>
</span><span id="HGVSName.parse_prefix-678"><a href="#HGVSName.parse_prefix-678"><span class="linenos">678</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">chrom</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span>
</span><span id="HGVSName.parse_prefix-679"><a href="#HGVSName.parse_prefix-679"><span class="linenos">679</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span>
</span><span id="HGVSName.parse_prefix-680"><a href="#HGVSName.parse_prefix-680"><span class="linenos">680</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">gene</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span>
</span><span id="HGVSName.parse_prefix-681"><a href="#HGVSName.parse_prefix-681"><span class="linenos">681</span></a>            <span class="k">return</span>
</span><span id="HGVSName.parse_prefix-682"><a href="#HGVSName.parse_prefix-682"><span class="linenos">682</span></a>
</span><span id="HGVSName.parse_prefix-683"><a href="#HGVSName.parse_prefix-683"><span class="linenos">683</span></a>        <span class="c1"># Transcript and gene given with parens.</span>
</span><span id="HGVSName.parse_prefix-684"><a href="#HGVSName.parse_prefix-684"><span class="linenos">684</span></a>        <span class="c1"># example: NM_007294.3(BRCA1):c.2207A&gt;C</span>
</span><span id="HGVSName.parse_prefix-685"><a href="#HGVSName.parse_prefix-685"><span class="linenos">685</span></a>        <span class="n">match</span> <span class="o">=</span> <span class="n">re</span><span class="o">.</span><span class="n">match</span><span class="p">(</span><span class="sa">r</span><span class="s2">&quot;^(?P&lt;transcript&gt;[^(]+)\((?P&lt;gene&gt;[^)]+)\)$&quot;</span><span class="p">,</span> <span class="n">prefix</span><span class="p">)</span>
</span><span id="HGVSName.parse_prefix-686"><a href="#HGVSName.parse_prefix-686"><span class="linenos">686</span></a>        <span class="k">if</span> <span class="n">match</span><span class="p">:</span>
</span><span id="HGVSName.parse_prefix-687"><a href="#HGVSName.parse_prefix-687"><span class="linenos">687</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span> <span class="o">=</span> <span class="n">match</span><span class="o">.</span><span class="n">group</span><span class="p">(</span><span class="s2">&quot;transcript&quot;</span><span class="p">)</span>
</span><span id="HGVSName.parse_prefix-688"><a href="#HGVSName.parse_prefix-688"><span class="linenos">688</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">gene</span> <span class="o">=</span> <span class="n">match</span><span class="o">.</span><span class="n">group</span><span class="p">(</span><span class="s2">&quot;gene&quot;</span><span class="p">)</span>
</span><span id="HGVSName.parse_prefix-689"><a href="#HGVSName.parse_prefix-689"><span class="linenos">689</span></a>            <span class="k">return</span>
</span><span id="HGVSName.parse_prefix-690"><a href="#HGVSName.parse_prefix-690"><span class="linenos">690</span></a>
</span><span id="HGVSName.parse_prefix-691"><a href="#HGVSName.parse_prefix-691"><span class="linenos">691</span></a>        <span class="c1"># Transcript and gene given with braces.</span>
</span><span id="HGVSName.parse_prefix-692"><a href="#HGVSName.parse_prefix-692"><span class="linenos">692</span></a>        <span class="c1"># example: BRCA1{NM_007294.3}:c.2207A&gt;C</span>
</span><span id="HGVSName.parse_prefix-693"><a href="#HGVSName.parse_prefix-693"><span class="linenos">693</span></a>        <span class="n">match</span> <span class="o">=</span> <span class="n">re</span><span class="o">.</span><span class="n">match</span><span class="p">(</span><span class="sa">r</span><span class="s2">&quot;^(?P&lt;gene&gt;[^{]+){(?P&lt;transcript&gt;[^}]+)}$&quot;</span><span class="p">,</span> <span class="n">prefix</span><span class="p">)</span>
</span><span id="HGVSName.parse_prefix-694"><a href="#HGVSName.parse_prefix-694"><span class="linenos">694</span></a>        <span class="k">if</span> <span class="n">match</span><span class="p">:</span>
</span><span id="HGVSName.parse_prefix-695"><a href="#HGVSName.parse_prefix-695"><span class="linenos">695</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span> <span class="o">=</span> <span class="n">match</span><span class="o">.</span><span class="n">group</span><span class="p">(</span><span class="s2">&quot;transcript&quot;</span><span class="p">)</span>
</span><span id="HGVSName.parse_prefix-696"><a href="#HGVSName.parse_prefix-696"><span class="linenos">696</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">gene</span> <span class="o">=</span> <span class="n">match</span><span class="o">.</span><span class="n">group</span><span class="p">(</span><span class="s2">&quot;gene&quot;</span><span class="p">)</span>
</span><span id="HGVSName.parse_prefix-697"><a href="#HGVSName.parse_prefix-697"><span class="linenos">697</span></a>            <span class="k">return</span>
</span><span id="HGVSName.parse_prefix-698"><a href="#HGVSName.parse_prefix-698"><span class="linenos">698</span></a>
</span><span id="HGVSName.parse_prefix-699"><a href="#HGVSName.parse_prefix-699"><span class="linenos">699</span></a>        <span class="c1"># Determine using Ensembl type.</span>
</span><span id="HGVSName.parse_prefix-700"><a href="#HGVSName.parse_prefix-700"><span class="linenos">700</span></a>        <span class="k">if</span> <span class="n">prefix</span><span class="o">.</span><span class="n">startswith</span><span class="p">(</span><span class="s2">&quot;ENST&quot;</span><span class="p">):</span>
</span><span id="HGVSName.parse_prefix-701"><a href="#HGVSName.parse_prefix-701"><span class="linenos">701</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span> <span class="o">=</span> <span class="n">prefix</span>
</span><span id="HGVSName.parse_prefix-702"><a href="#HGVSName.parse_prefix-702"><span class="linenos">702</span></a>            <span class="k">return</span>
</span><span id="HGVSName.parse_prefix-703"><a href="#HGVSName.parse_prefix-703"><span class="linenos">703</span></a>
</span><span id="HGVSName.parse_prefix-704"><a href="#HGVSName.parse_prefix-704"><span class="linenos">704</span></a>        <span class="c1"># Determine using LRG type.</span>
</span><span id="HGVSName.parse_prefix-705"><a href="#HGVSName.parse_prefix-705"><span class="linenos">705</span></a>        <span class="k">if</span> <span class="n">prefix</span><span class="o">.</span><span class="n">startswith</span><span class="p">(</span><span class="s2">&quot;LRG_&quot;</span><span class="p">):</span>
</span><span id="HGVSName.parse_prefix-706"><a href="#HGVSName.parse_prefix-706"><span class="linenos">706</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span> <span class="o">=</span> <span class="n">prefix</span>
</span><span id="HGVSName.parse_prefix-707"><a href="#HGVSName.parse_prefix-707"><span class="linenos">707</span></a>            <span class="k">return</span>
</span><span id="HGVSName.parse_prefix-708"><a href="#HGVSName.parse_prefix-708"><span class="linenos">708</span></a>
</span><span id="HGVSName.parse_prefix-709"><a href="#HGVSName.parse_prefix-709"><span class="linenos">709</span></a>        <span class="c1"># Determine using refseq type.</span>
</span><span id="HGVSName.parse_prefix-710"><a href="#HGVSName.parse_prefix-710"><span class="linenos">710</span></a>        <span class="n">refseq_type</span> <span class="o">=</span> <span class="n">get_refseq_type</span><span class="p">(</span><span class="n">prefix</span><span class="p">)</span>
</span><span id="HGVSName.parse_prefix-711"><a href="#HGVSName.parse_prefix-711"><span class="linenos">711</span></a>        <span class="k">if</span> <span class="n">refseq_type</span> <span class="ow">in</span> <span class="p">(</span><span class="s2">&quot;mRNA&quot;</span><span class="p">,</span> <span class="s2">&quot;RNA&quot;</span><span class="p">):</span>
</span><span id="HGVSName.parse_prefix-712"><a href="#HGVSName.parse_prefix-712"><span class="linenos">712</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span> <span class="o">=</span> <span class="n">prefix</span>
</span><span id="HGVSName.parse_prefix-713"><a href="#HGVSName.parse_prefix-713"><span class="linenos">713</span></a>            <span class="k">return</span>
</span><span id="HGVSName.parse_prefix-714"><a href="#HGVSName.parse_prefix-714"><span class="linenos">714</span></a>
</span><span id="HGVSName.parse_prefix-715"><a href="#HGVSName.parse_prefix-715"><span class="linenos">715</span></a>        <span class="c1"># Determine using refseq type.</span>
</span><span id="HGVSName.parse_prefix-716"><a href="#HGVSName.parse_prefix-716"><span class="linenos">716</span></a>        <span class="k">if</span> <span class="n">prefix</span><span class="o">.</span><span class="n">startswith</span><span class="p">(</span><span class="n">CHROM_PREFIX</span><span class="p">)</span> <span class="ow">or</span> <span class="n">refseq_type</span> <span class="o">==</span> <span class="s2">&quot;genomic&quot;</span><span class="p">:</span>
</span><span id="HGVSName.parse_prefix-717"><a href="#HGVSName.parse_prefix-717"><span class="linenos">717</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">chrom</span> <span class="o">=</span> <span class="n">prefix</span>
</span><span id="HGVSName.parse_prefix-718"><a href="#HGVSName.parse_prefix-718"><span class="linenos">718</span></a>            <span class="k">return</span>
</span><span id="HGVSName.parse_prefix-719"><a href="#HGVSName.parse_prefix-719"><span class="linenos">719</span></a>
</span><span id="HGVSName.parse_prefix-720"><a href="#HGVSName.parse_prefix-720"><span class="linenos">720</span></a>        <span class="c1"># Assume gene name.</span>
</span><span id="HGVSName.parse_prefix-721"><a href="#HGVSName.parse_prefix-721"><span class="linenos">721</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">gene</span> <span class="o">=</span> <span class="n">prefix</span>
</span></pre></div>


            <div class="docstring"><p>The <code><a href="#HGVSName.parse_prefix">parse_prefix</a></code> function is used to parse a HGVS prefix (gene/transcript/chromosome) and
assign the parsed values to the corresponding attributes of the object.</p>

<h6 id="parameters">Parameters</h6>

<ul>
<li><strong>prefix</strong>:  The <code><a href="#HGVSName.prefix">prefix</a></code> parameter is a string that represents a HGVS prefix, which can be a
gene, transcript, or chromosome identifier. It is used to determine the type of prefix and
assign the parsed values to the corresponding attributes of the object</li>
</ul>

<h6 id="returns">Returns</h6>

<blockquote>
  <p>The function <code><a href="#HGVSName.parse_prefix">parse_prefix</a></code> returns the parsed values for the transcript and gene
  attributes, or sets the chrom or gene attributes based on the given prefix.</p>
</blockquote>
</div>


                            </div>
                            <div id="HGVSName.parse_allele" class="classattr">
                                        <input id="HGVSName.parse_allele-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr function">
            
        <span class="def">def</span>
        <span class="name">parse_allele</span><span class="signature pdoc-code condensed">(<span class="param"><span class="bp">self</span>, </span><span class="param"><span class="n">allele</span><span class="p">:</span> <span class="nb">str</span></span><span class="return-annotation">) -> <span class="kc">None</span>:</span></span>

                <label class="view-source-button" for="HGVSName.parse_allele-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#HGVSName.parse_allele"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="HGVSName.parse_allele-723"><a href="#HGVSName.parse_allele-723"><span class="linenos">723</span></a>    <span class="k">def</span> <span class="nf">parse_allele</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">allele</span><span class="p">:</span> <span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
</span><span id="HGVSName.parse_allele-724"><a href="#HGVSName.parse_allele-724"><span class="linenos">724</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="HGVSName.parse_allele-725"><a href="#HGVSName.parse_allele-725"><span class="linenos">725</span></a><span class="sd">        The function `parse_allele` parses a HGVS allele description and determines the kind of HGVS</span>
</span><span id="HGVSName.parse_allele-726"><a href="#HGVSName.parse_allele-726"><span class="linenos">726</span></a><span class="sd">        name (c., p., g., etc.) and the mutation type.</span>
</span><span id="HGVSName.parse_allele-727"><a href="#HGVSName.parse_allele-727"><span class="linenos">727</span></a>
</span><span id="HGVSName.parse_allele-728"><a href="#HGVSName.parse_allele-728"><span class="linenos">728</span></a><span class="sd">        Some examples include:</span>
</span><span id="HGVSName.parse_allele-729"><a href="#HGVSName.parse_allele-729"><span class="linenos">729</span></a><span class="sd">          cDNA substitution: c.101A&gt;C,</span>
</span><span id="HGVSName.parse_allele-730"><a href="#HGVSName.parse_allele-730"><span class="linenos">730</span></a><span class="sd">          cDNA indel: c.3428delCinsTA, c.1000_1003delATG, c.1000_1001insATG</span>
</span><span id="HGVSName.parse_allele-731"><a href="#HGVSName.parse_allele-731"><span class="linenos">731</span></a><span class="sd">          No protein change: p.Glu1161=</span>
</span><span id="HGVSName.parse_allele-732"><a href="#HGVSName.parse_allele-732"><span class="linenos">732</span></a><span class="sd">          Protein change: p.Glu1161Ser</span>
</span><span id="HGVSName.parse_allele-733"><a href="#HGVSName.parse_allele-733"><span class="linenos">733</span></a><span class="sd">          Protein frameshift: p.Glu1161_Ser1164?fs</span>
</span><span id="HGVSName.parse_allele-734"><a href="#HGVSName.parse_allele-734"><span class="linenos">734</span></a><span class="sd">          Genomic substitution: g.1000100A&gt;T</span>
</span><span id="HGVSName.parse_allele-735"><a href="#HGVSName.parse_allele-735"><span class="linenos">735</span></a><span class="sd">          Genomic indel: g.1000100_1000102delATG</span>
</span><span id="HGVSName.parse_allele-736"><a href="#HGVSName.parse_allele-736"><span class="linenos">736</span></a>
</span><span id="HGVSName.parse_allele-737"><a href="#HGVSName.parse_allele-737"><span class="linenos">737</span></a><span class="sd">        :param allele: The `allele` parameter is a string that represents a HGVS allele description. It</span>
</span><span id="HGVSName.parse_allele-738"><a href="#HGVSName.parse_allele-738"><span class="linenos">738</span></a><span class="sd">        can contain various types of mutations, such as cDNA substitutions, cDNA indels, protein</span>
</span><span id="HGVSName.parse_allele-739"><a href="#HGVSName.parse_allele-739"><span class="linenos">739</span></a><span class="sd">        changes, protein frameshifts, genomic substitutions, and genomic indels. The purpose of the</span>
</span><span id="HGVSName.parse_allele-740"><a href="#HGVSName.parse_allele-740"><span class="linenos">740</span></a><span class="sd">        `parse_allele`</span>
</span><span id="HGVSName.parse_allele-741"><a href="#HGVSName.parse_allele-741"><span class="linenos">741</span></a><span class="sd">        :type allele: str</span>
</span><span id="HGVSName.parse_allele-742"><a href="#HGVSName.parse_allele-742"><span class="linenos">742</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="HGVSName.parse_allele-743"><a href="#HGVSName.parse_allele-743"><span class="linenos">743</span></a>
</span><span id="HGVSName.parse_allele-744"><a href="#HGVSName.parse_allele-744"><span class="linenos">744</span></a>        <span class="k">if</span> <span class="s2">&quot;.&quot;</span> <span class="ow">not</span> <span class="ow">in</span> <span class="n">allele</span><span class="p">:</span>
</span><span id="HGVSName.parse_allele-745"><a href="#HGVSName.parse_allele-745"><span class="linenos">745</span></a>            <span class="k">raise</span> <span class="n">InvalidHGVSName</span><span class="p">(</span>
</span><span id="HGVSName.parse_allele-746"><a href="#HGVSName.parse_allele-746"><span class="linenos">746</span></a>                <span class="n">allele</span><span class="p">,</span> <span class="s2">&quot;allele&quot;</span><span class="p">,</span> <span class="s1">&#39;expected kind &quot;c.&quot;, &quot;p.&quot;, &quot;g.&quot;, etc&#39;</span>
</span><span id="HGVSName.parse_allele-747"><a href="#HGVSName.parse_allele-747"><span class="linenos">747</span></a>            <span class="p">)</span>
</span><span id="HGVSName.parse_allele-748"><a href="#HGVSName.parse_allele-748"><span class="linenos">748</span></a>
</span><span id="HGVSName.parse_allele-749"><a href="#HGVSName.parse_allele-749"><span class="linenos">749</span></a>        <span class="c1"># Determine HGVS name kind.</span>
</span><span id="HGVSName.parse_allele-750"><a href="#HGVSName.parse_allele-750"><span class="linenos">750</span></a>        <span class="n">kind</span><span class="p">,</span> <span class="n">details</span> <span class="o">=</span> <span class="n">allele</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot;.&quot;</span><span class="p">,</span> <span class="mi">1</span><span class="p">)</span>
</span><span id="HGVSName.parse_allele-751"><a href="#HGVSName.parse_allele-751"><span class="linenos">751</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">kind</span> <span class="o">=</span> <span class="n">kind</span>
</span><span id="HGVSName.parse_allele-752"><a href="#HGVSName.parse_allele-752"><span class="linenos">752</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">=</span> <span class="kc">None</span>
</span><span id="HGVSName.parse_allele-753"><a href="#HGVSName.parse_allele-753"><span class="linenos">753</span></a>
</span><span id="HGVSName.parse_allele-754"><a href="#HGVSName.parse_allele-754"><span class="linenos">754</span></a>        <span class="k">if</span> <span class="n">kind</span> <span class="ow">in</span> <span class="p">(</span><span class="s2">&quot;c&quot;</span><span class="p">,</span> <span class="s2">&quot;n&quot;</span><span class="p">):</span>
</span><span id="HGVSName.parse_allele-755"><a href="#HGVSName.parse_allele-755"><span class="linenos">755</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">parse_cdna</span><span class="p">(</span><span class="n">details</span><span class="p">)</span>
</span><span id="HGVSName.parse_allele-756"><a href="#HGVSName.parse_allele-756"><span class="linenos">756</span></a>            <span class="k">if</span> <span class="n">kind</span> <span class="o">==</span> <span class="s2">&quot;n&quot;</span><span class="p">:</span>  <span class="c1"># Ensure no 3&#39;UTR or 5&#39;UTR coords in non-coding</span>
</span><span id="HGVSName.parse_allele-757"><a href="#HGVSName.parse_allele-757"><span class="linenos">757</span></a>                <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">cdna_start</span><span class="o">.</span><span class="n">coord</span> <span class="o">&lt;</span> <span class="mi">0</span><span class="p">:</span>
</span><span id="HGVSName.parse_allele-758"><a href="#HGVSName.parse_allele-758"><span class="linenos">758</span></a>                    <span class="k">raise</span> <span class="n">InvalidHGVSName</span><span class="p">(</span>
</span><span id="HGVSName.parse_allele-759"><a href="#HGVSName.parse_allele-759"><span class="linenos">759</span></a>                        <span class="n">allele</span><span class="p">,</span>
</span><span id="HGVSName.parse_allele-760"><a href="#HGVSName.parse_allele-760"><span class="linenos">760</span></a>                        <span class="s2">&quot;allele&quot;</span><span class="p">,</span>
</span><span id="HGVSName.parse_allele-761"><a href="#HGVSName.parse_allele-761"><span class="linenos">761</span></a>                        <span class="s2">&quot;Non-coding transcript cannot contain negative (5&#39;UTR) coordinates&quot;</span><span class="p">,</span>
</span><span id="HGVSName.parse_allele-762"><a href="#HGVSName.parse_allele-762"><span class="linenos">762</span></a>                    <span class="p">)</span>
</span><span id="HGVSName.parse_allele-763"><a href="#HGVSName.parse_allele-763"><span class="linenos">763</span></a>                <span class="k">if</span> <span class="p">(</span>
</span><span id="HGVSName.parse_allele-764"><a href="#HGVSName.parse_allele-764"><span class="linenos">764</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">cdna_start</span><span class="o">.</span><span class="n">landmark</span> <span class="o">==</span> <span class="s2">&quot;cdna_stop&quot;</span>
</span><span id="HGVSName.parse_allele-765"><a href="#HGVSName.parse_allele-765"><span class="linenos">765</span></a>                    <span class="ow">or</span> <span class="bp">self</span><span class="o">.</span><span class="n">cdna_end</span>
</span><span id="HGVSName.parse_allele-766"><a href="#HGVSName.parse_allele-766"><span class="linenos">766</span></a>                    <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">cdna_end</span><span class="o">.</span><span class="n">landmark</span> <span class="o">==</span> <span class="s2">&quot;cdna_stop&quot;</span>
</span><span id="HGVSName.parse_allele-767"><a href="#HGVSName.parse_allele-767"><span class="linenos">767</span></a>                <span class="p">):</span>
</span><span id="HGVSName.parse_allele-768"><a href="#HGVSName.parse_allele-768"><span class="linenos">768</span></a>                    <span class="k">raise</span> <span class="n">InvalidHGVSName</span><span class="p">(</span>
</span><span id="HGVSName.parse_allele-769"><a href="#HGVSName.parse_allele-769"><span class="linenos">769</span></a>                        <span class="n">allele</span><span class="p">,</span>
</span><span id="HGVSName.parse_allele-770"><a href="#HGVSName.parse_allele-770"><span class="linenos">770</span></a>                        <span class="s2">&quot;allele&quot;</span><span class="p">,</span>
</span><span id="HGVSName.parse_allele-771"><a href="#HGVSName.parse_allele-771"><span class="linenos">771</span></a>                        <span class="s2">&quot;Non-coding transcript cannot contain &#39;*&#39; (3&#39;UTR) coordinates&quot;</span><span class="p">,</span>
</span><span id="HGVSName.parse_allele-772"><a href="#HGVSName.parse_allele-772"><span class="linenos">772</span></a>                    <span class="p">)</span>
</span><span id="HGVSName.parse_allele-773"><a href="#HGVSName.parse_allele-773"><span class="linenos">773</span></a>        <span class="k">elif</span> <span class="n">kind</span> <span class="o">==</span> <span class="s2">&quot;p&quot;</span><span class="p">:</span>
</span><span id="HGVSName.parse_allele-774"><a href="#HGVSName.parse_allele-774"><span class="linenos">774</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">parse_protein</span><span class="p">(</span><span class="n">details</span><span class="p">)</span>
</span><span id="HGVSName.parse_allele-775"><a href="#HGVSName.parse_allele-775"><span class="linenos">775</span></a>        <span class="k">elif</span> <span class="n">kind</span> <span class="ow">in</span> <span class="p">(</span><span class="s2">&quot;g&quot;</span><span class="p">,</span> <span class="s2">&quot;m&quot;</span><span class="p">):</span>
</span><span id="HGVSName.parse_allele-776"><a href="#HGVSName.parse_allele-776"><span class="linenos">776</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">parse_genome</span><span class="p">(</span><span class="n">details</span><span class="p">)</span>
</span><span id="HGVSName.parse_allele-777"><a href="#HGVSName.parse_allele-777"><span class="linenos">777</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName.parse_allele-778"><a href="#HGVSName.parse_allele-778"><span class="linenos">778</span></a>            <span class="k">raise</span> <span class="ne">NotImplementedError</span><span class="p">(</span><span class="s2">&quot;unknown kind: </span><span class="si">%s</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="n">allele</span><span class="p">)</span>
</span></pre></div>


            <div class="docstring"><p>The function <code><a href="#HGVSName.parse_allele">parse_allele</a></code> parses a HGVS allele description and determines the kind of HGVS
name (c., p., g., etc.) and the mutation type.</p>

<p>Some examples include:
  cDNA substitution: c.101A>C,
  cDNA indel: c.3428delCinsTA, c.1000_1003delATG, c.1000_1001insATG
  No protein change: p.Glu1161=
  Protein change: p.Glu1161Ser
  Protein frameshift: p.Glu1161_Ser1164?fs
  Genomic substitution: g.1000100A>T
  Genomic indel: g.1000100_1000102delATG</p>

<h6 id="parameters">Parameters</h6>

<ul>
<li><strong>allele</strong>:  The <code>allele</code> parameter is a string that represents a HGVS allele description. It
can contain various types of mutations, such as cDNA substitutions, cDNA indels, protein
changes, protein frameshifts, genomic substitutions, and genomic indels. The purpose of the
<code><a href="#HGVSName.parse_allele">parse_allele</a></code></li>
</ul>
</div>


                            </div>
                            <div id="HGVSName.parse_cdna" class="classattr">
                                        <input id="HGVSName.parse_cdna-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr function">
            
        <span class="def">def</span>
        <span class="name">parse_cdna</span><span class="signature pdoc-code condensed">(<span class="param"><span class="bp">self</span>, </span><span class="param"><span class="n">details</span><span class="p">:</span> <span class="nb">str</span></span><span class="return-annotation">) -> <span class="kc">None</span>:</span></span>

                <label class="view-source-button" for="HGVSName.parse_cdna-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#HGVSName.parse_cdna"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="HGVSName.parse_cdna-780"><a href="#HGVSName.parse_cdna-780"><span class="linenos">780</span></a>    <span class="k">def</span> <span class="nf">parse_cdna</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">details</span><span class="p">:</span> <span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
</span><span id="HGVSName.parse_cdna-781"><a href="#HGVSName.parse_cdna-781"><span class="linenos">781</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="HGVSName.parse_cdna-782"><a href="#HGVSName.parse_cdna-782"><span class="linenos">782</span></a><span class="sd">        The function `parse_cdna` is used to parse a HGVS cDNA name and extract information such as</span>
</span><span id="HGVSName.parse_cdna-783"><a href="#HGVSName.parse_cdna-783"><span class="linenos">783</span></a><span class="sd">        mutation type, coordinates, and alleles.</span>
</span><span id="HGVSName.parse_cdna-784"><a href="#HGVSName.parse_cdna-784"><span class="linenos">784</span></a>
</span><span id="HGVSName.parse_cdna-785"><a href="#HGVSName.parse_cdna-785"><span class="linenos">785</span></a><span class="sd">        Some examples include:</span>
</span><span id="HGVSName.parse_cdna-786"><a href="#HGVSName.parse_cdna-786"><span class="linenos">786</span></a><span class="sd">          Substitution: 101A&gt;C,</span>
</span><span id="HGVSName.parse_cdna-787"><a href="#HGVSName.parse_cdna-787"><span class="linenos">787</span></a><span class="sd">          Indel: 3428delCinsTA, 1000_1003delATG, 1000_1001insATG</span>
</span><span id="HGVSName.parse_cdna-788"><a href="#HGVSName.parse_cdna-788"><span class="linenos">788</span></a>
</span><span id="HGVSName.parse_cdna-789"><a href="#HGVSName.parse_cdna-789"><span class="linenos">789</span></a><span class="sd">        :param details: The `details` parameter is a string that represents a HGVS cDNA name. It</span>
</span><span id="HGVSName.parse_cdna-790"><a href="#HGVSName.parse_cdna-790"><span class="linenos">790</span></a><span class="sd">        contains information about a genetic mutation, such as a substitution or an indel, along with</span>
</span><span id="HGVSName.parse_cdna-791"><a href="#HGVSName.parse_cdna-791"><span class="linenos">791</span></a><span class="sd">        the specific coordinates and alleles involved in the mutation</span>
</span><span id="HGVSName.parse_cdna-792"><a href="#HGVSName.parse_cdna-792"><span class="linenos">792</span></a><span class="sd">        :type details: str</span>
</span><span id="HGVSName.parse_cdna-793"><a href="#HGVSName.parse_cdna-793"><span class="linenos">793</span></a><span class="sd">        :return: None.</span>
</span><span id="HGVSName.parse_cdna-794"><a href="#HGVSName.parse_cdna-794"><span class="linenos">794</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="HGVSName.parse_cdna-795"><a href="#HGVSName.parse_cdna-795"><span class="linenos">795</span></a>
</span><span id="HGVSName.parse_cdna-796"><a href="#HGVSName.parse_cdna-796"><span class="linenos">796</span></a>        <span class="k">for</span> <span class="n">regex</span> <span class="ow">in</span> <span class="n">HGVSRegex</span><span class="o">.</span><span class="n">CDNA_ALLELE_REGEXES</span><span class="p">:</span>
</span><span id="HGVSName.parse_cdna-797"><a href="#HGVSName.parse_cdna-797"><span class="linenos">797</span></a>            <span class="n">match</span> <span class="o">=</span> <span class="n">re</span><span class="o">.</span><span class="n">match</span><span class="p">(</span><span class="n">regex</span><span class="p">,</span> <span class="n">details</span><span class="p">)</span>
</span><span id="HGVSName.parse_cdna-798"><a href="#HGVSName.parse_cdna-798"><span class="linenos">798</span></a>            <span class="k">if</span> <span class="n">match</span><span class="p">:</span>
</span><span id="HGVSName.parse_cdna-799"><a href="#HGVSName.parse_cdna-799"><span class="linenos">799</span></a>                <span class="n">groups</span> <span class="o">=</span> <span class="n">match</span><span class="o">.</span><span class="n">groupdict</span><span class="p">()</span>
</span><span id="HGVSName.parse_cdna-800"><a href="#HGVSName.parse_cdna-800"><span class="linenos">800</span></a>
</span><span id="HGVSName.parse_cdna-801"><a href="#HGVSName.parse_cdna-801"><span class="linenos">801</span></a>                <span class="c1"># Parse mutation type.</span>
</span><span id="HGVSName.parse_cdna-802"><a href="#HGVSName.parse_cdna-802"><span class="linenos">802</span></a>                <span class="k">if</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;delins&quot;</span><span class="p">):</span>
</span><span id="HGVSName.parse_cdna-803"><a href="#HGVSName.parse_cdna-803"><span class="linenos">803</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">=</span> <span class="s2">&quot;delins&quot;</span>
</span><span id="HGVSName.parse_cdna-804"><a href="#HGVSName.parse_cdna-804"><span class="linenos">804</span></a>                <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName.parse_cdna-805"><a href="#HGVSName.parse_cdna-805"><span class="linenos">805</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">=</span> <span class="n">groups</span><span class="p">[</span><span class="s2">&quot;mutation_type&quot;</span><span class="p">]</span>
</span><span id="HGVSName.parse_cdna-806"><a href="#HGVSName.parse_cdna-806"><span class="linenos">806</span></a>
</span><span id="HGVSName.parse_cdna-807"><a href="#HGVSName.parse_cdna-807"><span class="linenos">807</span></a>                <span class="c1"># Parse coordinates.</span>
</span><span id="HGVSName.parse_cdna-808"><a href="#HGVSName.parse_cdna-808"><span class="linenos">808</span></a>                <span class="bp">self</span><span class="o">.</span><span class="n">cdna_start</span> <span class="o">=</span> <span class="n">CDNACoord</span><span class="p">(</span><span class="n">string</span><span class="o">=</span><span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;start&quot;</span><span class="p">))</span>
</span><span id="HGVSName.parse_cdna-809"><a href="#HGVSName.parse_cdna-809"><span class="linenos">809</span></a>                <span class="k">if</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;end&quot;</span><span class="p">):</span>
</span><span id="HGVSName.parse_cdna-810"><a href="#HGVSName.parse_cdna-810"><span class="linenos">810</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">cdna_end</span> <span class="o">=</span> <span class="n">CDNACoord</span><span class="p">(</span><span class="n">string</span><span class="o">=</span><span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;end&quot;</span><span class="p">))</span>
</span><span id="HGVSName.parse_cdna-811"><a href="#HGVSName.parse_cdna-811"><span class="linenos">811</span></a>                <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName.parse_cdna-812"><a href="#HGVSName.parse_cdna-812"><span class="linenos">812</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">cdna_end</span> <span class="o">=</span> <span class="n">CDNACoord</span><span class="p">(</span><span class="n">string</span><span class="o">=</span><span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;start&quot;</span><span class="p">))</span>
</span><span id="HGVSName.parse_cdna-813"><a href="#HGVSName.parse_cdna-813"><span class="linenos">813</span></a>
</span><span id="HGVSName.parse_cdna-814"><a href="#HGVSName.parse_cdna-814"><span class="linenos">814</span></a>                <span class="c1"># Parse alleles.</span>
</span><span id="HGVSName.parse_cdna-815"><a href="#HGVSName.parse_cdna-815"><span class="linenos">815</span></a>                <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">=</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;ref&quot;</span><span class="p">,</span> <span class="s2">&quot;&quot;</span><span class="p">)</span>
</span><span id="HGVSName.parse_cdna-816"><a href="#HGVSName.parse_cdna-816"><span class="linenos">816</span></a>                <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span> <span class="o">=</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;alt&quot;</span><span class="p">,</span> <span class="s2">&quot;&quot;</span><span class="p">)</span>
</span><span id="HGVSName.parse_cdna-817"><a href="#HGVSName.parse_cdna-817"><span class="linenos">817</span></a>
</span><span id="HGVSName.parse_cdna-818"><a href="#HGVSName.parse_cdna-818"><span class="linenos">818</span></a>                <span class="c1"># Convert numerical allelles.</span>
</span><span id="HGVSName.parse_cdna-819"><a href="#HGVSName.parse_cdna-819"><span class="linenos">819</span></a>                <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span><span class="o">.</span><span class="n">isdigit</span><span class="p">():</span>
</span><span id="HGVSName.parse_cdna-820"><a href="#HGVSName.parse_cdna-820"><span class="linenos">820</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">=</span> <span class="s2">&quot;N&quot;</span> <span class="o">*</span> <span class="nb">int</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span><span class="p">)</span>
</span><span id="HGVSName.parse_cdna-821"><a href="#HGVSName.parse_cdna-821"><span class="linenos">821</span></a>                <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span><span class="o">.</span><span class="n">isdigit</span><span class="p">():</span>
</span><span id="HGVSName.parse_cdna-822"><a href="#HGVSName.parse_cdna-822"><span class="linenos">822</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span> <span class="o">=</span> <span class="s2">&quot;N&quot;</span> <span class="o">*</span> <span class="nb">int</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span><span class="p">)</span>
</span><span id="HGVSName.parse_cdna-823"><a href="#HGVSName.parse_cdna-823"><span class="linenos">823</span></a>
</span><span id="HGVSName.parse_cdna-824"><a href="#HGVSName.parse_cdna-824"><span class="linenos">824</span></a>                <span class="c1"># Convert duplication alleles.</span>
</span><span id="HGVSName.parse_cdna-825"><a href="#HGVSName.parse_cdna-825"><span class="linenos">825</span></a>                <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">==</span> <span class="s2">&quot;dup&quot;</span><span class="p">:</span>
</span><span id="HGVSName.parse_cdna-826"><a href="#HGVSName.parse_cdna-826"><span class="linenos">826</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">*</span> <span class="mi">2</span>
</span><span id="HGVSName.parse_cdna-827"><a href="#HGVSName.parse_cdna-827"><span class="linenos">827</span></a>
</span><span id="HGVSName.parse_cdna-828"><a href="#HGVSName.parse_cdna-828"><span class="linenos">828</span></a>                <span class="c1"># Convert no match alleles.</span>
</span><span id="HGVSName.parse_cdna-829"><a href="#HGVSName.parse_cdna-829"><span class="linenos">829</span></a>                <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">==</span> <span class="s2">&quot;=&quot;</span><span class="p">:</span>
</span><span id="HGVSName.parse_cdna-830"><a href="#HGVSName.parse_cdna-830"><span class="linenos">830</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span>
</span><span id="HGVSName.parse_cdna-831"><a href="#HGVSName.parse_cdna-831"><span class="linenos">831</span></a>                <span class="k">return</span>
</span><span id="HGVSName.parse_cdna-832"><a href="#HGVSName.parse_cdna-832"><span class="linenos">832</span></a>
</span><span id="HGVSName.parse_cdna-833"><a href="#HGVSName.parse_cdna-833"><span class="linenos">833</span></a>        <span class="k">raise</span> <span class="n">InvalidHGVSName</span><span class="p">(</span><span class="n">details</span><span class="p">,</span> <span class="s2">&quot;cDNA allele&quot;</span><span class="p">)</span>
</span></pre></div>


            <div class="docstring"><p>The function <code><a href="#HGVSName.parse_cdna">parse_cdna</a></code> is used to parse a HGVS cDNA name and extract information such as
mutation type, coordinates, and alleles.</p>

<p>Some examples include:
  Substitution: 101A>C,
  Indel: 3428delCinsTA, 1000_1003delATG, 1000_1001insATG</p>

<h6 id="parameters">Parameters</h6>

<ul>
<li><strong>details</strong>:  The <code>details</code> parameter is a string that represents a HGVS cDNA name. It
contains information about a genetic mutation, such as a substitution or an indel, along with
the specific coordinates and alleles involved in the mutation</li>
</ul>

<h6 id="returns">Returns</h6>

<blockquote>
  <p>None.</p>
</blockquote>
</div>


                            </div>
                            <div id="HGVSName.parse_protein" class="classattr">
                                        <input id="HGVSName.parse_protein-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr function">
            
        <span class="def">def</span>
        <span class="name">parse_protein</span><span class="signature pdoc-code condensed">(<span class="param"><span class="bp">self</span>, </span><span class="param"><span class="n">details</span><span class="p">:</span> <span class="nb">str</span></span><span class="return-annotation">) -> <span class="kc">None</span>:</span></span>

                <label class="view-source-button" for="HGVSName.parse_protein-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#HGVSName.parse_protein"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="HGVSName.parse_protein-835"><a href="#HGVSName.parse_protein-835"><span class="linenos">835</span></a>    <span class="k">def</span> <span class="nf">parse_protein</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">details</span><span class="p">:</span> <span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
</span><span id="HGVSName.parse_protein-836"><a href="#HGVSName.parse_protein-836"><span class="linenos">836</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="HGVSName.parse_protein-837"><a href="#HGVSName.parse_protein-837"><span class="linenos">837</span></a><span class="sd">        The function `parse_protein` is used to parse a HGVS protein name and extract information such</span>
</span><span id="HGVSName.parse_protein-838"><a href="#HGVSName.parse_protein-838"><span class="linenos">838</span></a><span class="sd">        as mutation type, coordinates, alleles, and additional details.</span>
</span><span id="HGVSName.parse_protein-839"><a href="#HGVSName.parse_protein-839"><span class="linenos">839</span></a>
</span><span id="HGVSName.parse_protein-840"><a href="#HGVSName.parse_protein-840"><span class="linenos">840</span></a><span class="sd">        Some examples include:</span>
</span><span id="HGVSName.parse_protein-841"><a href="#HGVSName.parse_protein-841"><span class="linenos">841</span></a><span class="sd">          No change: Glu1161=</span>
</span><span id="HGVSName.parse_protein-842"><a href="#HGVSName.parse_protein-842"><span class="linenos">842</span></a><span class="sd">          Change: Glu1161Ser</span>
</span><span id="HGVSName.parse_protein-843"><a href="#HGVSName.parse_protein-843"><span class="linenos">843</span></a><span class="sd">          Frameshift: Glu1161_Ser1164?fs</span>
</span><span id="HGVSName.parse_protein-844"><a href="#HGVSName.parse_protein-844"><span class="linenos">844</span></a>
</span><span id="HGVSName.parse_protein-845"><a href="#HGVSName.parse_protein-845"><span class="linenos">845</span></a><span class="sd">        :param details: The `details` parameter is a string that represents a HGVS protein name. It</span>
</span><span id="HGVSName.parse_protein-846"><a href="#HGVSName.parse_protein-846"><span class="linenos">846</span></a><span class="sd">        contains information about a protein mutation, such as the amino acid change and the position of</span>
</span><span id="HGVSName.parse_protein-847"><a href="#HGVSName.parse_protein-847"><span class="linenos">847</span></a><span class="sd">        the mutation</span>
</span><span id="HGVSName.parse_protein-848"><a href="#HGVSName.parse_protein-848"><span class="linenos">848</span></a><span class="sd">        :type details: str</span>
</span><span id="HGVSName.parse_protein-849"><a href="#HGVSName.parse_protein-849"><span class="linenos">849</span></a><span class="sd">        :return: The method `parse_protein` does not return anything. It updates the instance variables</span>
</span><span id="HGVSName.parse_protein-850"><a href="#HGVSName.parse_protein-850"><span class="linenos">850</span></a><span class="sd">        of the object it is called on.</span>
</span><span id="HGVSName.parse_protein-851"><a href="#HGVSName.parse_protein-851"><span class="linenos">851</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="HGVSName.parse_protein-852"><a href="#HGVSName.parse_protein-852"><span class="linenos">852</span></a>
</span><span id="HGVSName.parse_protein-853"><a href="#HGVSName.parse_protein-853"><span class="linenos">853</span></a>        <span class="k">for</span> <span class="n">regex</span> <span class="ow">in</span> <span class="n">HGVSRegex</span><span class="o">.</span><span class="n">PEP_ALLELE_REGEXES</span><span class="p">:</span>
</span><span id="HGVSName.parse_protein-854"><a href="#HGVSName.parse_protein-854"><span class="linenos">854</span></a>            <span class="n">match</span> <span class="o">=</span> <span class="n">re</span><span class="o">.</span><span class="n">match</span><span class="p">(</span><span class="n">regex</span><span class="p">,</span> <span class="n">details</span><span class="p">)</span>
</span><span id="HGVSName.parse_protein-855"><a href="#HGVSName.parse_protein-855"><span class="linenos">855</span></a>            <span class="k">if</span> <span class="n">match</span><span class="p">:</span>
</span><span id="HGVSName.parse_protein-856"><a href="#HGVSName.parse_protein-856"><span class="linenos">856</span></a>                <span class="n">groups</span> <span class="o">=</span> <span class="n">match</span><span class="o">.</span><span class="n">groupdict</span><span class="p">()</span>
</span><span id="HGVSName.parse_protein-857"><a href="#HGVSName.parse_protein-857"><span class="linenos">857</span></a>
</span><span id="HGVSName.parse_protein-858"><a href="#HGVSName.parse_protein-858"><span class="linenos">858</span></a>                <span class="c1"># Parse mutation type.</span>
</span><span id="HGVSName.parse_protein-859"><a href="#HGVSName.parse_protein-859"><span class="linenos">859</span></a>                <span class="k">if</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;delins&quot;</span><span class="p">):</span>
</span><span id="HGVSName.parse_protein-860"><a href="#HGVSName.parse_protein-860"><span class="linenos">860</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">=</span> <span class="s2">&quot;delins&quot;</span>
</span><span id="HGVSName.parse_protein-861"><a href="#HGVSName.parse_protein-861"><span class="linenos">861</span></a>                <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName.parse_protein-862"><a href="#HGVSName.parse_protein-862"><span class="linenos">862</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">=</span> <span class="s2">&quot;&gt;&quot;</span>
</span><span id="HGVSName.parse_protein-863"><a href="#HGVSName.parse_protein-863"><span class="linenos">863</span></a>
</span><span id="HGVSName.parse_protein-864"><a href="#HGVSName.parse_protein-864"><span class="linenos">864</span></a>                <span class="c1"># Parse coordinates.</span>
</span><span id="HGVSName.parse_protein-865"><a href="#HGVSName.parse_protein-865"><span class="linenos">865</span></a>                <span class="bp">self</span><span class="o">.</span><span class="n">start</span> <span class="o">=</span> <span class="nb">int</span><span class="p">(</span><span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;start&quot;</span><span class="p">))</span>
</span><span id="HGVSName.parse_protein-866"><a href="#HGVSName.parse_protein-866"><span class="linenos">866</span></a>                <span class="k">if</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;end&quot;</span><span class="p">):</span>
</span><span id="HGVSName.parse_protein-867"><a href="#HGVSName.parse_protein-867"><span class="linenos">867</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">end</span> <span class="o">=</span> <span class="nb">int</span><span class="p">(</span><span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;end&quot;</span><span class="p">))</span>
</span><span id="HGVSName.parse_protein-868"><a href="#HGVSName.parse_protein-868"><span class="linenos">868</span></a>                <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName.parse_protein-869"><a href="#HGVSName.parse_protein-869"><span class="linenos">869</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">end</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">start</span>
</span><span id="HGVSName.parse_protein-870"><a href="#HGVSName.parse_protein-870"><span class="linenos">870</span></a>
</span><span id="HGVSName.parse_protein-871"><a href="#HGVSName.parse_protein-871"><span class="linenos">871</span></a>                <span class="c1"># Parse alleles.</span>
</span><span id="HGVSName.parse_protein-872"><a href="#HGVSName.parse_protein-872"><span class="linenos">872</span></a>                <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">=</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;ref&quot;</span><span class="p">,</span> <span class="s2">&quot;&quot;</span><span class="p">)</span>
</span><span id="HGVSName.parse_protein-873"><a href="#HGVSName.parse_protein-873"><span class="linenos">873</span></a>                <span class="k">if</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;ref2&quot;</span><span class="p">):</span>
</span><span id="HGVSName.parse_protein-874"><a href="#HGVSName.parse_protein-874"><span class="linenos">874</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">ref2_allele</span> <span class="o">=</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;ref2&quot;</span><span class="p">)</span>
</span><span id="HGVSName.parse_protein-875"><a href="#HGVSName.parse_protein-875"><span class="linenos">875</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span> <span class="o">=</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;alt&quot;</span><span class="p">,</span> <span class="s2">&quot;&quot;</span><span class="p">)</span>
</span><span id="HGVSName.parse_protein-876"><a href="#HGVSName.parse_protein-876"><span class="linenos">876</span></a>                <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName.parse_protein-877"><a href="#HGVSName.parse_protein-877"><span class="linenos">877</span></a>                    <span class="c1"># If alt is not given, assume matching with ref</span>
</span><span id="HGVSName.parse_protein-878"><a href="#HGVSName.parse_protein-878"><span class="linenos">878</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">ref2_allele</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span>
</span><span id="HGVSName.parse_protein-879"><a href="#HGVSName.parse_protein-879"><span class="linenos">879</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span> <span class="o">=</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;alt&quot;</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span><span class="p">)</span>
</span><span id="HGVSName.parse_protein-880"><a href="#HGVSName.parse_protein-880"><span class="linenos">880</span></a>
</span><span id="HGVSName.parse_protein-881"><a href="#HGVSName.parse_protein-881"><span class="linenos">881</span></a>                <span class="bp">self</span><span class="o">.</span><span class="n">pep_extra</span> <span class="o">=</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;extra&quot;</span><span class="p">)</span>
</span><span id="HGVSName.parse_protein-882"><a href="#HGVSName.parse_protein-882"><span class="linenos">882</span></a>                <span class="k">return</span>
</span><span id="HGVSName.parse_protein-883"><a href="#HGVSName.parse_protein-883"><span class="linenos">883</span></a>
</span><span id="HGVSName.parse_protein-884"><a href="#HGVSName.parse_protein-884"><span class="linenos">884</span></a>        <span class="k">raise</span> <span class="n">InvalidHGVSName</span><span class="p">(</span><span class="n">details</span><span class="p">,</span> <span class="s2">&quot;protein allele&quot;</span><span class="p">)</span>
</span></pre></div>


            <div class="docstring"><p>The function <code><a href="#HGVSName.parse_protein">parse_protein</a></code> is used to parse a HGVS protein name and extract information such
as mutation type, coordinates, alleles, and additional details.</p>

<p>Some examples include:
  No change: Glu1161=
  Change: Glu1161Ser
  Frameshift: Glu1161_Ser1164?fs</p>

<h6 id="parameters">Parameters</h6>

<ul>
<li><strong>details</strong>:  The <code>details</code> parameter is a string that represents a HGVS protein name. It
contains information about a protein mutation, such as the amino acid change and the position of
the mutation</li>
</ul>

<h6 id="returns">Returns</h6>

<blockquote>
  <p>The method <code><a href="#HGVSName.parse_protein">parse_protein</a></code> does not return anything. It updates the instance variables
  of the object it is called on.</p>
</blockquote>
</div>


                            </div>
                            <div id="HGVSName.parse_genome" class="classattr">
                                        <input id="HGVSName.parse_genome-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr function">
            
        <span class="def">def</span>
        <span class="name">parse_genome</span><span class="signature pdoc-code condensed">(<span class="param"><span class="bp">self</span>, </span><span class="param"><span class="n">details</span><span class="p">:</span> <span class="nb">str</span></span><span class="return-annotation">) -> <span class="kc">None</span>:</span></span>

                <label class="view-source-button" for="HGVSName.parse_genome-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#HGVSName.parse_genome"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="HGVSName.parse_genome-886"><a href="#HGVSName.parse_genome-886"><span class="linenos">886</span></a>    <span class="k">def</span> <span class="nf">parse_genome</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">details</span><span class="p">:</span> <span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
</span><span id="HGVSName.parse_genome-887"><a href="#HGVSName.parse_genome-887"><span class="linenos">887</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="HGVSName.parse_genome-888"><a href="#HGVSName.parse_genome-888"><span class="linenos">888</span></a><span class="sd">        The function `parse_genome` is used to parse a HGVS genomic name and extract information such as</span>
</span><span id="HGVSName.parse_genome-889"><a href="#HGVSName.parse_genome-889"><span class="linenos">889</span></a><span class="sd">        mutation type, coordinates, and alleles.</span>
</span><span id="HGVSName.parse_genome-890"><a href="#HGVSName.parse_genome-890"><span class="linenos">890</span></a>
</span><span id="HGVSName.parse_genome-891"><a href="#HGVSName.parse_genome-891"><span class="linenos">891</span></a><span class="sd">        Some examples include:</span>
</span><span id="HGVSName.parse_genome-892"><a href="#HGVSName.parse_genome-892"><span class="linenos">892</span></a><span class="sd">          Substitution: 1000100A&gt;T</span>
</span><span id="HGVSName.parse_genome-893"><a href="#HGVSName.parse_genome-893"><span class="linenos">893</span></a><span class="sd">          Indel: 1000100_1000102delATG</span>
</span><span id="HGVSName.parse_genome-894"><a href="#HGVSName.parse_genome-894"><span class="linenos">894</span></a>
</span><span id="HGVSName.parse_genome-895"><a href="#HGVSName.parse_genome-895"><span class="linenos">895</span></a><span class="sd">        :param details: The `details` parameter is a string that represents a HGVS genomic name. It</span>
</span><span id="HGVSName.parse_genome-896"><a href="#HGVSName.parse_genome-896"><span class="linenos">896</span></a><span class="sd">        contains information about a genomic mutation, such as a substitution or an indel</span>
</span><span id="HGVSName.parse_genome-897"><a href="#HGVSName.parse_genome-897"><span class="linenos">897</span></a><span class="sd">        :type details: str</span>
</span><span id="HGVSName.parse_genome-898"><a href="#HGVSName.parse_genome-898"><span class="linenos">898</span></a><span class="sd">        :return: None.</span>
</span><span id="HGVSName.parse_genome-899"><a href="#HGVSName.parse_genome-899"><span class="linenos">899</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="HGVSName.parse_genome-900"><a href="#HGVSName.parse_genome-900"><span class="linenos">900</span></a>
</span><span id="HGVSName.parse_genome-901"><a href="#HGVSName.parse_genome-901"><span class="linenos">901</span></a>        <span class="k">for</span> <span class="n">regex</span> <span class="ow">in</span> <span class="n">HGVSRegex</span><span class="o">.</span><span class="n">GENOMIC_ALLELE_REGEXES</span><span class="p">:</span>
</span><span id="HGVSName.parse_genome-902"><a href="#HGVSName.parse_genome-902"><span class="linenos">902</span></a>            <span class="n">match</span> <span class="o">=</span> <span class="n">re</span><span class="o">.</span><span class="n">match</span><span class="p">(</span><span class="n">regex</span><span class="p">,</span> <span class="n">details</span><span class="p">)</span>
</span><span id="HGVSName.parse_genome-903"><a href="#HGVSName.parse_genome-903"><span class="linenos">903</span></a>            <span class="k">if</span> <span class="n">match</span><span class="p">:</span>
</span><span id="HGVSName.parse_genome-904"><a href="#HGVSName.parse_genome-904"><span class="linenos">904</span></a>                <span class="n">groups</span> <span class="o">=</span> <span class="n">match</span><span class="o">.</span><span class="n">groupdict</span><span class="p">()</span>
</span><span id="HGVSName.parse_genome-905"><a href="#HGVSName.parse_genome-905"><span class="linenos">905</span></a>
</span><span id="HGVSName.parse_genome-906"><a href="#HGVSName.parse_genome-906"><span class="linenos">906</span></a>                <span class="c1"># Parse mutation type.</span>
</span><span id="HGVSName.parse_genome-907"><a href="#HGVSName.parse_genome-907"><span class="linenos">907</span></a>                <span class="k">if</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;delins&quot;</span><span class="p">):</span>
</span><span id="HGVSName.parse_genome-908"><a href="#HGVSName.parse_genome-908"><span class="linenos">908</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">=</span> <span class="s2">&quot;delins&quot;</span>
</span><span id="HGVSName.parse_genome-909"><a href="#HGVSName.parse_genome-909"><span class="linenos">909</span></a>                <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName.parse_genome-910"><a href="#HGVSName.parse_genome-910"><span class="linenos">910</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">=</span> <span class="n">groups</span><span class="p">[</span><span class="s2">&quot;mutation_type&quot;</span><span class="p">]</span>
</span><span id="HGVSName.parse_genome-911"><a href="#HGVSName.parse_genome-911"><span class="linenos">911</span></a>
</span><span id="HGVSName.parse_genome-912"><a href="#HGVSName.parse_genome-912"><span class="linenos">912</span></a>                <span class="c1"># Parse coordinates.</span>
</span><span id="HGVSName.parse_genome-913"><a href="#HGVSName.parse_genome-913"><span class="linenos">913</span></a>                <span class="bp">self</span><span class="o">.</span><span class="n">start</span> <span class="o">=</span> <span class="nb">int</span><span class="p">(</span><span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;start&quot;</span><span class="p">))</span>
</span><span id="HGVSName.parse_genome-914"><a href="#HGVSName.parse_genome-914"><span class="linenos">914</span></a>                <span class="k">if</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;end&quot;</span><span class="p">):</span>
</span><span id="HGVSName.parse_genome-915"><a href="#HGVSName.parse_genome-915"><span class="linenos">915</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">end</span> <span class="o">=</span> <span class="nb">int</span><span class="p">(</span><span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;end&quot;</span><span class="p">))</span>
</span><span id="HGVSName.parse_genome-916"><a href="#HGVSName.parse_genome-916"><span class="linenos">916</span></a>                <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName.parse_genome-917"><a href="#HGVSName.parse_genome-917"><span class="linenos">917</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">end</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">start</span>
</span><span id="HGVSName.parse_genome-918"><a href="#HGVSName.parse_genome-918"><span class="linenos">918</span></a>
</span><span id="HGVSName.parse_genome-919"><a href="#HGVSName.parse_genome-919"><span class="linenos">919</span></a>                <span class="c1"># Parse alleles.</span>
</span><span id="HGVSName.parse_genome-920"><a href="#HGVSName.parse_genome-920"><span class="linenos">920</span></a>                <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">=</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;ref&quot;</span><span class="p">,</span> <span class="s2">&quot;&quot;</span><span class="p">)</span>
</span><span id="HGVSName.parse_genome-921"><a href="#HGVSName.parse_genome-921"><span class="linenos">921</span></a>                <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span> <span class="o">=</span> <span class="n">groups</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;alt&quot;</span><span class="p">,</span> <span class="s2">&quot;&quot;</span><span class="p">)</span>
</span><span id="HGVSName.parse_genome-922"><a href="#HGVSName.parse_genome-922"><span class="linenos">922</span></a>
</span><span id="HGVSName.parse_genome-923"><a href="#HGVSName.parse_genome-923"><span class="linenos">923</span></a>                <span class="c1"># Convert numerical alleles.</span>
</span><span id="HGVSName.parse_genome-924"><a href="#HGVSName.parse_genome-924"><span class="linenos">924</span></a>                <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span><span class="o">.</span><span class="n">isdigit</span><span class="p">():</span>
</span><span id="HGVSName.parse_genome-925"><a href="#HGVSName.parse_genome-925"><span class="linenos">925</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">=</span> <span class="s2">&quot;N&quot;</span> <span class="o">*</span> <span class="nb">int</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span><span class="p">)</span>
</span><span id="HGVSName.parse_genome-926"><a href="#HGVSName.parse_genome-926"><span class="linenos">926</span></a>                <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span><span class="o">.</span><span class="n">isdigit</span><span class="p">():</span>
</span><span id="HGVSName.parse_genome-927"><a href="#HGVSName.parse_genome-927"><span class="linenos">927</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span> <span class="o">=</span> <span class="s2">&quot;N&quot;</span> <span class="o">*</span> <span class="nb">int</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span><span class="p">)</span>
</span><span id="HGVSName.parse_genome-928"><a href="#HGVSName.parse_genome-928"><span class="linenos">928</span></a>
</span><span id="HGVSName.parse_genome-929"><a href="#HGVSName.parse_genome-929"><span class="linenos">929</span></a>                <span class="c1"># Convert duplication alleles.</span>
</span><span id="HGVSName.parse_genome-930"><a href="#HGVSName.parse_genome-930"><span class="linenos">930</span></a>                <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">==</span> <span class="s2">&quot;dup&quot;</span><span class="p">:</span>
</span><span id="HGVSName.parse_genome-931"><a href="#HGVSName.parse_genome-931"><span class="linenos">931</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">*</span> <span class="mi">2</span>
</span><span id="HGVSName.parse_genome-932"><a href="#HGVSName.parse_genome-932"><span class="linenos">932</span></a>
</span><span id="HGVSName.parse_genome-933"><a href="#HGVSName.parse_genome-933"><span class="linenos">933</span></a>                <span class="c1"># Convert no match alleles.</span>
</span><span id="HGVSName.parse_genome-934"><a href="#HGVSName.parse_genome-934"><span class="linenos">934</span></a>                <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">==</span> <span class="s2">&quot;=&quot;</span><span class="p">:</span>
</span><span id="HGVSName.parse_genome-935"><a href="#HGVSName.parse_genome-935"><span class="linenos">935</span></a>                    <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span>
</span><span id="HGVSName.parse_genome-936"><a href="#HGVSName.parse_genome-936"><span class="linenos">936</span></a>                <span class="k">return</span>
</span><span id="HGVSName.parse_genome-937"><a href="#HGVSName.parse_genome-937"><span class="linenos">937</span></a>
</span><span id="HGVSName.parse_genome-938"><a href="#HGVSName.parse_genome-938"><span class="linenos">938</span></a>        <span class="k">raise</span> <span class="n">InvalidHGVSName</span><span class="p">(</span><span class="n">details</span><span class="p">,</span> <span class="s2">&quot;genomic allele&quot;</span><span class="p">)</span>
</span></pre></div>


            <div class="docstring"><p>The function <code><a href="#HGVSName.parse_genome">parse_genome</a></code> is used to parse a HGVS genomic name and extract information such as
mutation type, coordinates, and alleles.</p>

<p>Some examples include:
  Substitution: 1000100A>T
  Indel: 1000100_1000102delATG</p>

<h6 id="parameters">Parameters</h6>

<ul>
<li><strong>details</strong>:  The <code>details</code> parameter is a string that represents a HGVS genomic name. It
contains information about a genomic mutation, such as a substitution or an indel</li>
</ul>

<h6 id="returns">Returns</h6>

<blockquote>
  <p>None.</p>
</blockquote>
</div>


                            </div>
                            <div id="HGVSName.format" class="classattr">
                                        <input id="HGVSName.format-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr function">
            
        <span class="def">def</span>
        <span class="name">format</span><span class="signature pdoc-code multiline">(<span class="param">	<span class="bp">self</span>,</span><span class="param">	<span class="n">use_prefix</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">True</span>,</span><span class="param">	<span class="n">use_gene</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">True</span>,</span><span class="param">	<span class="n">use_exon</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span>,</span><span class="param">	<span class="n">use_protein</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span>,</span><span class="param">	<span class="n">full_format</span><span class="o">=</span><span class="kc">False</span>,</span><span class="param">	<span class="n">use_version</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span></span><span class="return-annotation">) -> <span class="nb">str</span>:</span></span>

                <label class="view-source-button" for="HGVSName.format-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#HGVSName.format"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="HGVSName.format-970"><a href="#HGVSName.format-970"><span class="linenos"> 970</span></a>    <span class="k">def</span> <span class="nf">format</span><span class="p">(</span>
</span><span id="HGVSName.format-971"><a href="#HGVSName.format-971"><span class="linenos"> 971</span></a>        <span class="bp">self</span><span class="p">,</span>
</span><span id="HGVSName.format-972"><a href="#HGVSName.format-972"><span class="linenos"> 972</span></a>        <span class="n">use_prefix</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span>
</span><span id="HGVSName.format-973"><a href="#HGVSName.format-973"><span class="linenos"> 973</span></a>        <span class="n">use_gene</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span>
</span><span id="HGVSName.format-974"><a href="#HGVSName.format-974"><span class="linenos"> 974</span></a>        <span class="n">use_exon</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
</span><span id="HGVSName.format-975"><a href="#HGVSName.format-975"><span class="linenos"> 975</span></a>        <span class="n">use_protein</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
</span><span id="HGVSName.format-976"><a href="#HGVSName.format-976"><span class="linenos"> 976</span></a>        <span class="n">full_format</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span>
</span><span id="HGVSName.format-977"><a href="#HGVSName.format-977"><span class="linenos"> 977</span></a>        <span class="n">use_version</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
</span><span id="HGVSName.format-978"><a href="#HGVSName.format-978"><span class="linenos"> 978</span></a>    <span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
</span><span id="HGVSName.format-979"><a href="#HGVSName.format-979"><span class="linenos"> 979</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="HGVSName.format-980"><a href="#HGVSName.format-980"><span class="linenos"> 980</span></a><span class="sd">        The `format` function generates a HGVS name as a string based on various formatting options.</span>
</span><span id="HGVSName.format-981"><a href="#HGVSName.format-981"><span class="linenos"> 981</span></a>
</span><span id="HGVSName.format-982"><a href="#HGVSName.format-982"><span class="linenos"> 982</span></a><span class="sd">        :param use_prefix: A boolean indicating whether to include the prefix in the HGVS name. If set</span>
</span><span id="HGVSName.format-983"><a href="#HGVSName.format-983"><span class="linenos"> 983</span></a><span class="sd">        to True, the prefix will be included in the HGVS name. If set to False, the prefix will be</span>
</span><span id="HGVSName.format-984"><a href="#HGVSName.format-984"><span class="linenos"> 984</span></a><span class="sd">        excluded. The default value is True, defaults to True</span>
</span><span id="HGVSName.format-985"><a href="#HGVSName.format-985"><span class="linenos"> 985</span></a><span class="sd">        :type use_prefix: bool (optional)</span>
</span><span id="HGVSName.format-986"><a href="#HGVSName.format-986"><span class="linenos"> 986</span></a><span class="sd">        :param use_gene: A boolean indicating whether to include the gene name in the HGVS name. If set</span>
</span><span id="HGVSName.format-987"><a href="#HGVSName.format-987"><span class="linenos"> 987</span></a><span class="sd">        to True, the gene name will be included in the HGVS name. If set to False, the gene name will</span>
</span><span id="HGVSName.format-988"><a href="#HGVSName.format-988"><span class="linenos"> 988</span></a><span class="sd">        not be included. The default value is True, defaults to True</span>
</span><span id="HGVSName.format-989"><a href="#HGVSName.format-989"><span class="linenos"> 989</span></a><span class="sd">        :type use_gene: bool (optional)</span>
</span><span id="HGVSName.format-990"><a href="#HGVSName.format-990"><span class="linenos"> 990</span></a><span class="sd">        :param use_exon: A boolean indicating whether to include exon information in the HGVS name. If</span>
</span><span id="HGVSName.format-991"><a href="#HGVSName.format-991"><span class="linenos"> 991</span></a><span class="sd">        set to True, exon information will be included in the HGVS name. If set to False, exon</span>
</span><span id="HGVSName.format-992"><a href="#HGVSName.format-992"><span class="linenos"> 992</span></a><span class="sd">        information will not be included, defaults to False</span>
</span><span id="HGVSName.format-993"><a href="#HGVSName.format-993"><span class="linenos"> 993</span></a><span class="sd">        :type use_exon: bool (optional)</span>
</span><span id="HGVSName.format-994"><a href="#HGVSName.format-994"><span class="linenos"> 994</span></a><span class="sd">        :param use_protein: A boolean indicating whether to include the protein change in the HGVS name.</span>
</span><span id="HGVSName.format-995"><a href="#HGVSName.format-995"><span class="linenos"> 995</span></a><span class="sd">        If set to True, the protein change will be included in the HGVS name. If set to False, the</span>
</span><span id="HGVSName.format-996"><a href="#HGVSName.format-996"><span class="linenos"> 996</span></a><span class="sd">        protein change will not be included, defaults to False</span>
</span><span id="HGVSName.format-997"><a href="#HGVSName.format-997"><span class="linenos"> 997</span></a><span class="sd">        :type use_protein: bool (optional)</span>
</span><span id="HGVSName.format-998"><a href="#HGVSName.format-998"><span class="linenos"> 998</span></a><span class="sd">        :param full_format: A boolean parameter that determines whether the full format of the allele</span>
</span><span id="HGVSName.format-999"><a href="#HGVSName.format-999"><span class="linenos"> 999</span></a><span class="sd">        should be included in the output. If set to True, and if the allele is not a protein variant,</span>
</span><span id="HGVSName.format-1000"><a href="#HGVSName.format-1000"><span class="linenos">1000</span></a><span class="sd">        the allele will be appended with &#39;:p.&#39; followed by the formatted protein variant, defaults to</span>
</span><span id="HGVSName.format-1001"><a href="#HGVSName.format-1001"><span class="linenos">1001</span></a><span class="sd">        False (optional)</span>
</span><span id="HGVSName.format-1002"><a href="#HGVSName.format-1002"><span class="linenos">1002</span></a><span class="sd">        :param use_version: A boolean parameter that determines whether to include the version number in</span>
</span><span id="HGVSName.format-1003"><a href="#HGVSName.format-1003"><span class="linenos">1003</span></a><span class="sd">        the formatted HGVS name. If set to True, the version number will be included in the output. If</span>
</span><span id="HGVSName.format-1004"><a href="#HGVSName.format-1004"><span class="linenos">1004</span></a><span class="sd">        set to False, the version number will not be included, defaults to False</span>
</span><span id="HGVSName.format-1005"><a href="#HGVSName.format-1005"><span class="linenos">1005</span></a><span class="sd">        :type use_version: bool (optional)</span>
</span><span id="HGVSName.format-1006"><a href="#HGVSName.format-1006"><span class="linenos">1006</span></a><span class="sd">        :return: a HGVS name as a string.</span>
</span><span id="HGVSName.format-1007"><a href="#HGVSName.format-1007"><span class="linenos">1007</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="HGVSName.format-1008"><a href="#HGVSName.format-1008"><span class="linenos">1008</span></a>
</span><span id="HGVSName.format-1009"><a href="#HGVSName.format-1009"><span class="linenos">1009</span></a>        <span class="k">if</span> <span class="n">use_protein</span> <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">format_protein</span><span class="p">():</span>
</span><span id="HGVSName.format-1010"><a href="#HGVSName.format-1010"><span class="linenos">1010</span></a>            <span class="n">allele</span> <span class="o">=</span> <span class="s2">&quot;p.&quot;</span> <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">format_protein</span><span class="p">()</span>
</span><span id="HGVSName.format-1011"><a href="#HGVSName.format-1011"><span class="linenos">1011</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName.format-1012"><a href="#HGVSName.format-1012"><span class="linenos">1012</span></a>            <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">kind</span> <span class="ow">in</span> <span class="p">(</span><span class="s2">&quot;c&quot;</span><span class="p">,</span> <span class="s2">&quot;n&quot;</span><span class="p">):</span>
</span><span id="HGVSName.format-1013"><a href="#HGVSName.format-1013"><span class="linenos">1013</span></a>                <span class="n">allele</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">kind</span> <span class="o">+</span> <span class="s2">&quot;.&quot;</span> <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">format_cdna</span><span class="p">()</span>
</span><span id="HGVSName.format-1014"><a href="#HGVSName.format-1014"><span class="linenos">1014</span></a>            <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">kind</span> <span class="o">==</span> <span class="s2">&quot;p&quot;</span><span class="p">:</span>
</span><span id="HGVSName.format-1015"><a href="#HGVSName.format-1015"><span class="linenos">1015</span></a>                <span class="n">allele</span> <span class="o">=</span> <span class="s2">&quot;p.&quot;</span> <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">format_protein</span><span class="p">()</span>
</span><span id="HGVSName.format-1016"><a href="#HGVSName.format-1016"><span class="linenos">1016</span></a>            <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">kind</span> <span class="ow">in</span> <span class="p">(</span><span class="s2">&quot;g&quot;</span><span class="p">,</span> <span class="s2">&quot;m&quot;</span><span class="p">):</span>
</span><span id="HGVSName.format-1017"><a href="#HGVSName.format-1017"><span class="linenos">1017</span></a>                <span class="n">allele</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">kind</span> <span class="o">+</span> <span class="s2">&quot;.&quot;</span> <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">format_genome</span><span class="p">()</span>
</span><span id="HGVSName.format-1018"><a href="#HGVSName.format-1018"><span class="linenos">1018</span></a>            <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName.format-1019"><a href="#HGVSName.format-1019"><span class="linenos">1019</span></a>                <span class="k">raise</span> <span class="ne">NotImplementedError</span><span class="p">(</span><span class="s2">&quot;not implemented: &#39;</span><span class="si">%s</span><span class="s2">&#39;&quot;</span> <span class="o">%</span> <span class="bp">self</span><span class="o">.</span><span class="n">kind</span><span class="p">)</span>
</span><span id="HGVSName.format-1020"><a href="#HGVSName.format-1020"><span class="linenos">1020</span></a>
</span><span id="HGVSName.format-1021"><a href="#HGVSName.format-1021"><span class="linenos">1021</span></a>        <span class="k">if</span> <span class="n">full_format</span> <span class="ow">and</span> <span class="ow">not</span> <span class="n">use_protein</span> <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">format_protein</span><span class="p">():</span>
</span><span id="HGVSName.format-1022"><a href="#HGVSName.format-1022"><span class="linenos">1022</span></a>            <span class="n">allele</span> <span class="o">+=</span> <span class="s2">&quot;:p.&quot;</span> <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">format_protein</span><span class="p">()</span>
</span><span id="HGVSName.format-1023"><a href="#HGVSName.format-1023"><span class="linenos">1023</span></a>
</span><span id="HGVSName.format-1024"><a href="#HGVSName.format-1024"><span class="linenos">1024</span></a>        <span class="n">prefix</span> <span class="o">=</span> <span class="p">(</span>
</span><span id="HGVSName.format-1025"><a href="#HGVSName.format-1025"><span class="linenos">1025</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">format_prefix</span><span class="p">(</span>
</span><span id="HGVSName.format-1026"><a href="#HGVSName.format-1026"><span class="linenos">1026</span></a>                <span class="n">use_gene</span><span class="o">=</span><span class="n">use_gene</span><span class="p">,</span>
</span><span id="HGVSName.format-1027"><a href="#HGVSName.format-1027"><span class="linenos">1027</span></a>                <span class="n">use_exon</span><span class="o">=</span><span class="n">use_exon</span><span class="p">,</span>
</span><span id="HGVSName.format-1028"><a href="#HGVSName.format-1028"><span class="linenos">1028</span></a>                <span class="n">use_protein</span><span class="o">=</span><span class="n">use_protein</span><span class="p">,</span>
</span><span id="HGVSName.format-1029"><a href="#HGVSName.format-1029"><span class="linenos">1029</span></a>                <span class="n">full_format</span><span class="o">=</span><span class="n">full_format</span><span class="p">,</span>
</span><span id="HGVSName.format-1030"><a href="#HGVSName.format-1030"><span class="linenos">1030</span></a>                <span class="n">use_version</span><span class="o">=</span><span class="n">use_version</span><span class="p">,</span>
</span><span id="HGVSName.format-1031"><a href="#HGVSName.format-1031"><span class="linenos">1031</span></a>            <span class="p">)</span>
</span><span id="HGVSName.format-1032"><a href="#HGVSName.format-1032"><span class="linenos">1032</span></a>            <span class="k">if</span> <span class="n">use_prefix</span>
</span><span id="HGVSName.format-1033"><a href="#HGVSName.format-1033"><span class="linenos">1033</span></a>            <span class="k">else</span> <span class="s2">&quot;&quot;</span>
</span><span id="HGVSName.format-1034"><a href="#HGVSName.format-1034"><span class="linenos">1034</span></a>        <span class="p">)</span>
</span><span id="HGVSName.format-1035"><a href="#HGVSName.format-1035"><span class="linenos">1035</span></a>
</span><span id="HGVSName.format-1036"><a href="#HGVSName.format-1036"><span class="linenos">1036</span></a>        <span class="k">if</span> <span class="n">prefix</span><span class="p">:</span>
</span><span id="HGVSName.format-1037"><a href="#HGVSName.format-1037"><span class="linenos">1037</span></a>            <span class="k">return</span> <span class="n">prefix</span> <span class="o">+</span> <span class="s2">&quot;:&quot;</span> <span class="o">+</span> <span class="n">allele</span>
</span><span id="HGVSName.format-1038"><a href="#HGVSName.format-1038"><span class="linenos">1038</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName.format-1039"><a href="#HGVSName.format-1039"><span class="linenos">1039</span></a>            <span class="k">return</span> <span class="n">allele</span>
</span></pre></div>


            <div class="docstring"><p>The <code><a href="#HGVSName.format">format</a></code> function generates a HGVS name as a string based on various formatting options.</p>

<h6 id="parameters">Parameters</h6>

<ul>
<li><strong>use_prefix</strong>:  A boolean indicating whether to include the prefix in the HGVS name. If set
to True, the prefix will be included in the HGVS name. If set to False, the prefix will be
excluded. The default value is True, defaults to True</li>
<li><strong>use_gene</strong>:  A boolean indicating whether to include the gene name in the HGVS name. If set
to True, the gene name will be included in the HGVS name. If set to False, the gene name will
not be included. The default value is True, defaults to True</li>
<li><strong>use_exon</strong>:  A boolean indicating whether to include exon information in the HGVS name. If
set to True, exon information will be included in the HGVS name. If set to False, exon
information will not be included, defaults to False</li>
<li><strong>use_protein</strong>:  A boolean indicating whether to include the protein change in the HGVS name.
If set to True, the protein change will be included in the HGVS name. If set to False, the
protein change will not be included, defaults to False</li>
<li><strong>full_format</strong>:  A boolean parameter that determines whether the full format of the allele
should be included in the output. If set to True, and if the allele is not a protein variant,
the allele will be appended with ':p.' followed by the formatted protein variant, defaults to
False (optional)</li>
<li><strong>use_version</strong>:  A boolean parameter that determines whether to include the version number in
the formatted HGVS name. If set to True, the version number will be included in the output. If
set to False, the version number will not be included, defaults to False</li>
</ul>

<h6 id="returns">Returns</h6>

<blockquote>
  <p>a HGVS name as a string.</p>
</blockquote>
</div>


                            </div>
                            <div id="HGVSName.format_prefix" class="classattr">
                                        <input id="HGVSName.format_prefix-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr function">
            
        <span class="def">def</span>
        <span class="name">format_prefix</span><span class="signature pdoc-code multiline">(<span class="param">	<span class="bp">self</span>,</span><span class="param">	<span class="n">use_gene</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">True</span>,</span><span class="param">	<span class="n">use_exon</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span>,</span><span class="param">	<span class="n">use_protein</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span>,</span><span class="param">	<span class="n">full_format</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span>,</span><span class="param">	<span class="n">use_version</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span></span><span class="return-annotation">) -> <span class="nb">str</span>:</span></span>

                <label class="view-source-button" for="HGVSName.format_prefix-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#HGVSName.format_prefix"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="HGVSName.format_prefix-1041"><a href="#HGVSName.format_prefix-1041"><span class="linenos">1041</span></a>    <span class="k">def</span> <span class="nf">format_prefix</span><span class="p">(</span>
</span><span id="HGVSName.format_prefix-1042"><a href="#HGVSName.format_prefix-1042"><span class="linenos">1042</span></a>        <span class="bp">self</span><span class="p">,</span>
</span><span id="HGVSName.format_prefix-1043"><a href="#HGVSName.format_prefix-1043"><span class="linenos">1043</span></a>        <span class="n">use_gene</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span>
</span><span id="HGVSName.format_prefix-1044"><a href="#HGVSName.format_prefix-1044"><span class="linenos">1044</span></a>        <span class="n">use_exon</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
</span><span id="HGVSName.format_prefix-1045"><a href="#HGVSName.format_prefix-1045"><span class="linenos">1045</span></a>        <span class="n">use_protein</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
</span><span id="HGVSName.format_prefix-1046"><a href="#HGVSName.format_prefix-1046"><span class="linenos">1046</span></a>        <span class="n">full_format</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
</span><span id="HGVSName.format_prefix-1047"><a href="#HGVSName.format_prefix-1047"><span class="linenos">1047</span></a>        <span class="n">use_version</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
</span><span id="HGVSName.format_prefix-1048"><a href="#HGVSName.format_prefix-1048"><span class="linenos">1048</span></a>    <span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
</span><span id="HGVSName.format_prefix-1049"><a href="#HGVSName.format_prefix-1049"><span class="linenos">1049</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="HGVSName.format_prefix-1050"><a href="#HGVSName.format_prefix-1050"><span class="linenos">1050</span></a><span class="sd">        The `format_prefix` function generates an HGVS transcript/gene prefix based on various</span>
</span><span id="HGVSName.format_prefix-1051"><a href="#HGVSName.format_prefix-1051"><span class="linenos">1051</span></a><span class="sd">        parameters.</span>
</span><span id="HGVSName.format_prefix-1052"><a href="#HGVSName.format_prefix-1052"><span class="linenos">1052</span></a>
</span><span id="HGVSName.format_prefix-1053"><a href="#HGVSName.format_prefix-1053"><span class="linenos">1053</span></a><span class="sd">        :param use_gene: A boolean parameter that determines whether to include the gene name in the</span>
</span><span id="HGVSName.format_prefix-1054"><a href="#HGVSName.format_prefix-1054"><span class="linenos">1054</span></a><span class="sd">        prefix. If set to True, the gene name will be included in the prefix. If set to False, the gene</span>
</span><span id="HGVSName.format_prefix-1055"><a href="#HGVSName.format_prefix-1055"><span class="linenos">1055</span></a><span class="sd">        name will not be included in the prefix. The default value is True, defaults to True</span>
</span><span id="HGVSName.format_prefix-1056"><a href="#HGVSName.format_prefix-1056"><span class="linenos">1056</span></a><span class="sd">        :type use_gene: bool (optional)</span>
</span><span id="HGVSName.format_prefix-1057"><a href="#HGVSName.format_prefix-1057"><span class="linenos">1057</span></a><span class="sd">        :param use_exon: A boolean parameter that determines whether to include the exon information in</span>
</span><span id="HGVSName.format_prefix-1058"><a href="#HGVSName.format_prefix-1058"><span class="linenos">1058</span></a><span class="sd">        the prefix. If set to True, the exon information will be included in the prefix. If set to</span>
</span><span id="HGVSName.format_prefix-1059"><a href="#HGVSName.format_prefix-1059"><span class="linenos">1059</span></a><span class="sd">        False, the exon information will not be included, defaults to False</span>
</span><span id="HGVSName.format_prefix-1060"><a href="#HGVSName.format_prefix-1060"><span class="linenos">1060</span></a><span class="sd">        :type use_exon: bool (optional)</span>
</span><span id="HGVSName.format_prefix-1061"><a href="#HGVSName.format_prefix-1061"><span class="linenos">1061</span></a><span class="sd">        :param use_protein: A boolean indicating whether to use the protein transcript instead of the</span>
</span><span id="HGVSName.format_prefix-1062"><a href="#HGVSName.format_prefix-1062"><span class="linenos">1062</span></a><span class="sd">        nucleotide transcript if available. If set to True, the protein transcript will be used. If set</span>
</span><span id="HGVSName.format_prefix-1063"><a href="#HGVSName.format_prefix-1063"><span class="linenos">1063</span></a><span class="sd">        to False, the nucleotide transcript will be used. The default value is False, defaults to False</span>
</span><span id="HGVSName.format_prefix-1064"><a href="#HGVSName.format_prefix-1064"><span class="linenos">1064</span></a><span class="sd">        :type use_protein: bool (optional)</span>
</span><span id="HGVSName.format_prefix-1065"><a href="#HGVSName.format_prefix-1065"><span class="linenos">1065</span></a><span class="sd">        :param full_format: A boolean parameter that determines whether to generate the full HGVS name</span>
</span><span id="HGVSName.format_prefix-1066"><a href="#HGVSName.format_prefix-1066"><span class="linenos">1066</span></a><span class="sd">        with transcript/gene prefix or not. If set to True, the full format will be generated. If set to</span>
</span><span id="HGVSName.format_prefix-1067"><a href="#HGVSName.format_prefix-1067"><span class="linenos">1067</span></a><span class="sd">        False, only the transcript/gene prefix will be generated, defaults to False</span>
</span><span id="HGVSName.format_prefix-1068"><a href="#HGVSName.format_prefix-1068"><span class="linenos">1068</span></a><span class="sd">        :type full_format: bool (optional)</span>
</span><span id="HGVSName.format_prefix-1069"><a href="#HGVSName.format_prefix-1069"><span class="linenos">1069</span></a><span class="sd">        :param use_version: A boolean parameter that determines whether to include the version number in</span>
</span><span id="HGVSName.format_prefix-1070"><a href="#HGVSName.format_prefix-1070"><span class="linenos">1070</span></a><span class="sd">        the transcript prefix. If set to True, the version number will be included in the prefix (e.g.,</span>
</span><span id="HGVSName.format_prefix-1071"><a href="#HGVSName.format_prefix-1071"><span class="linenos">1071</span></a><span class="sd">        NM_007294.3). If set to False, only the transcript ID without the version number will be</span>
</span><span id="HGVSName.format_prefix-1072"><a href="#HGVSName.format_prefix-1072"><span class="linenos">1072</span></a><span class="sd">        included in the prefix, defaults to False</span>
</span><span id="HGVSName.format_prefix-1073"><a href="#HGVSName.format_prefix-1073"><span class="linenos">1073</span></a><span class="sd">        :type use_version: bool (optional)</span>
</span><span id="HGVSName.format_prefix-1074"><a href="#HGVSName.format_prefix-1074"><span class="linenos">1074</span></a><span class="sd">        :return: The function `format_prefix` returns a formatted HGVS transcript/gene prefix as a</span>
</span><span id="HGVSName.format_prefix-1075"><a href="#HGVSName.format_prefix-1075"><span class="linenos">1075</span></a><span class="sd">        string.</span>
</span><span id="HGVSName.format_prefix-1076"><a href="#HGVSName.format_prefix-1076"><span class="linenos">1076</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="HGVSName.format_prefix-1077"><a href="#HGVSName.format_prefix-1077"><span class="linenos">1077</span></a>
</span><span id="HGVSName.format_prefix-1078"><a href="#HGVSName.format_prefix-1078"><span class="linenos">1078</span></a>        <span class="k">if</span> <span class="n">full_format</span><span class="p">:</span>
</span><span id="HGVSName.format_prefix-1079"><a href="#HGVSName.format_prefix-1079"><span class="linenos">1079</span></a>
</span><span id="HGVSName.format_prefix-1080"><a href="#HGVSName.format_prefix-1080"><span class="linenos">1080</span></a>            <span class="n">prefix</span> <span class="o">=</span> <span class="p">[]</span>
</span><span id="HGVSName.format_prefix-1081"><a href="#HGVSName.format_prefix-1081"><span class="linenos">1081</span></a>            <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">gene</span><span class="p">:</span>
</span><span id="HGVSName.format_prefix-1082"><a href="#HGVSName.format_prefix-1082"><span class="linenos">1082</span></a>                <span class="n">prefix</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">gene</span><span class="p">)</span>
</span><span id="HGVSName.format_prefix-1083"><a href="#HGVSName.format_prefix-1083"><span class="linenos">1083</span></a>            <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span><span class="p">:</span>
</span><span id="HGVSName.format_prefix-1084"><a href="#HGVSName.format_prefix-1084"><span class="linenos">1084</span></a>                <span class="n">prefix</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">transcript</span><span class="p">)</span>
</span><span id="HGVSName.format_prefix-1085"><a href="#HGVSName.format_prefix-1085"><span class="linenos">1085</span></a>            <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">transcript_protein</span><span class="p">:</span>
</span><span id="HGVSName.format_prefix-1086"><a href="#HGVSName.format_prefix-1086"><span class="linenos">1086</span></a>                <span class="n">prefix</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">transcript_protein</span><span class="p">)</span>
</span><span id="HGVSName.format_prefix-1087"><a href="#HGVSName.format_prefix-1087"><span class="linenos">1087</span></a>            <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">exon</span><span class="p">:</span>  <span class="c1"># TODO</span>
</span><span id="HGVSName.format_prefix-1088"><a href="#HGVSName.format_prefix-1088"><span class="linenos">1088</span></a>                <span class="n">prefix</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s2">&quot;exon&quot;</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">exon</span><span class="p">))</span>
</span><span id="HGVSName.format_prefix-1089"><a href="#HGVSName.format_prefix-1089"><span class="linenos">1089</span></a>            <span class="k">return</span> <span class="s2">&quot;:&quot;</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">prefix</span><span class="p">)</span>
</span><span id="HGVSName.format_prefix-1090"><a href="#HGVSName.format_prefix-1090"><span class="linenos">1090</span></a>
</span><span id="HGVSName.format_prefix-1091"><a href="#HGVSName.format_prefix-1091"><span class="linenos">1091</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName.format_prefix-1092"><a href="#HGVSName.format_prefix-1092"><span class="linenos">1092</span></a>
</span><span id="HGVSName.format_prefix-1093"><a href="#HGVSName.format_prefix-1093"><span class="linenos">1093</span></a>            <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">kind</span> <span class="ow">in</span> <span class="p">(</span><span class="s2">&quot;g&quot;</span><span class="p">,</span> <span class="s2">&quot;m&quot;</span><span class="p">):</span>
</span><span id="HGVSName.format_prefix-1094"><a href="#HGVSName.format_prefix-1094"><span class="linenos">1094</span></a>                <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">chrom</span><span class="p">:</span>
</span><span id="HGVSName.format_prefix-1095"><a href="#HGVSName.format_prefix-1095"><span class="linenos">1095</span></a>                    <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">chrom</span>
</span><span id="HGVSName.format_prefix-1096"><a href="#HGVSName.format_prefix-1096"><span class="linenos">1096</span></a>
</span><span id="HGVSName.format_prefix-1097"><a href="#HGVSName.format_prefix-1097"><span class="linenos">1097</span></a>            <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span><span class="p">:</span>
</span><span id="HGVSName.format_prefix-1098"><a href="#HGVSName.format_prefix-1098"><span class="linenos">1098</span></a>                <span class="k">if</span> <span class="n">use_protein</span> <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">transcript_protein</span><span class="p">:</span>
</span><span id="HGVSName.format_prefix-1099"><a href="#HGVSName.format_prefix-1099"><span class="linenos">1099</span></a>                    <span class="n">transcript</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">transcript_protein</span>
</span><span id="HGVSName.format_prefix-1100"><a href="#HGVSName.format_prefix-1100"><span class="linenos">1100</span></a>                <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName.format_prefix-1101"><a href="#HGVSName.format_prefix-1101"><span class="linenos">1101</span></a>                    <span class="n">transcript</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span>
</span><span id="HGVSName.format_prefix-1102"><a href="#HGVSName.format_prefix-1102"><span class="linenos">1102</span></a>                <span class="k">if</span> <span class="ow">not</span> <span class="n">use_version</span> <span class="ow">and</span> <span class="n">transcript</span><span class="p">:</span>
</span><span id="HGVSName.format_prefix-1103"><a href="#HGVSName.format_prefix-1103"><span class="linenos">1103</span></a>                    <span class="n">transcript</span> <span class="o">=</span> <span class="n">transcript</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot;.&quot;</span><span class="p">)[</span><span class="mi">0</span><span class="p">]</span>
</span><span id="HGVSName.format_prefix-1104"><a href="#HGVSName.format_prefix-1104"><span class="linenos">1104</span></a>                <span class="k">if</span> <span class="n">use_gene</span> <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">gene</span><span class="p">:</span>
</span><span id="HGVSName.format_prefix-1105"><a href="#HGVSName.format_prefix-1105"><span class="linenos">1105</span></a>                    <span class="k">return</span> <span class="s2">&quot;</span><span class="si">%s</span><span class="s2">(</span><span class="si">%s</span><span class="s2">)&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">transcript</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">gene</span><span class="p">)</span>
</span><span id="HGVSName.format_prefix-1106"><a href="#HGVSName.format_prefix-1106"><span class="linenos">1106</span></a>                <span class="k">elif</span> <span class="n">use_exon</span> <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">exon</span><span class="p">:</span>
</span><span id="HGVSName.format_prefix-1107"><a href="#HGVSName.format_prefix-1107"><span class="linenos">1107</span></a>                    <span class="k">return</span> <span class="s2">&quot;</span><span class="si">%s</span><span class="s2">(exon</span><span class="si">%s</span><span class="s2">)&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">transcript</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">exon</span><span class="p">)</span>
</span><span id="HGVSName.format_prefix-1108"><a href="#HGVSName.format_prefix-1108"><span class="linenos">1108</span></a>                <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName.format_prefix-1109"><a href="#HGVSName.format_prefix-1109"><span class="linenos">1109</span></a>                    <span class="k">return</span> <span class="n">transcript</span>
</span><span id="HGVSName.format_prefix-1110"><a href="#HGVSName.format_prefix-1110"><span class="linenos">1110</span></a>            <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName.format_prefix-1111"><a href="#HGVSName.format_prefix-1111"><span class="linenos">1111</span></a>                <span class="k">if</span> <span class="n">use_gene</span><span class="p">:</span>
</span><span id="HGVSName.format_prefix-1112"><a href="#HGVSName.format_prefix-1112"><span class="linenos">1112</span></a>                    <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">gene</span>
</span><span id="HGVSName.format_prefix-1113"><a href="#HGVSName.format_prefix-1113"><span class="linenos">1113</span></a>                <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName.format_prefix-1114"><a href="#HGVSName.format_prefix-1114"><span class="linenos">1114</span></a>                    <span class="k">return</span> <span class="s2">&quot;&quot;</span>
</span></pre></div>


            <div class="docstring"><p>The <code><a href="#HGVSName.format_prefix">format_prefix</a></code> function generates an HGVS transcript/gene prefix based on various
parameters.</p>

<h6 id="parameters">Parameters</h6>

<ul>
<li><strong>use_gene</strong>:  A boolean parameter that determines whether to include the gene name in the
prefix. If set to True, the gene name will be included in the prefix. If set to False, the gene
name will not be included in the prefix. The default value is True, defaults to True</li>
<li><strong>use_exon</strong>:  A boolean parameter that determines whether to include the exon information in
the prefix. If set to True, the exon information will be included in the prefix. If set to
False, the exon information will not be included, defaults to False</li>
<li><strong>use_protein</strong>:  A boolean indicating whether to use the protein transcript instead of the
nucleotide transcript if available. If set to True, the protein transcript will be used. If set
to False, the nucleotide transcript will be used. The default value is False, defaults to False</li>
<li><strong>full_format</strong>:  A boolean parameter that determines whether to generate the full HGVS name
with transcript/gene prefix or not. If set to True, the full format will be generated. If set to
False, only the transcript/gene prefix will be generated, defaults to False</li>
<li><strong>use_version</strong>:  A boolean parameter that determines whether to include the version number in
the transcript prefix. If set to True, the version number will be included in the prefix (e.g.,
NM_007294.3). If set to False, only the transcript ID without the version number will be
included in the prefix, defaults to False</li>
</ul>

<h6 id="returns">Returns</h6>

<blockquote>
  <p>The function <code><a href="#HGVSName.format_prefix">format_prefix</a></code> returns a formatted HGVS transcript/gene prefix as a
  string.</p>
</blockquote>
</div>


                            </div>
                            <div id="HGVSName.format_cdna_coords" class="classattr">
                                        <input id="HGVSName.format_cdna_coords-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr function">
            
        <span class="def">def</span>
        <span class="name">format_cdna_coords</span><span class="signature pdoc-code condensed">(<span class="param"><span class="bp">self</span></span><span class="return-annotation">) -> <span class="nb">str</span>:</span></span>

                <label class="view-source-button" for="HGVSName.format_cdna_coords-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#HGVSName.format_cdna_coords"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="HGVSName.format_cdna_coords-1116"><a href="#HGVSName.format_cdna_coords-1116"><span class="linenos">1116</span></a>    <span class="k">def</span> <span class="nf">format_cdna_coords</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
</span><span id="HGVSName.format_cdna_coords-1117"><a href="#HGVSName.format_cdna_coords-1117"><span class="linenos">1117</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="HGVSName.format_cdna_coords-1118"><a href="#HGVSName.format_cdna_coords-1118"><span class="linenos">1118</span></a><span class="sd">        The function `format_cdna_coords` generates a string representing HGVS cDNA coordinates,</span>
</span><span id="HGVSName.format_cdna_coords-1119"><a href="#HGVSName.format_cdna_coords-1119"><span class="linenos">1119</span></a><span class="sd">        returning either the start coordinate or a string in the format &quot;start_end&quot; depending on whether</span>
</span><span id="HGVSName.format_cdna_coords-1120"><a href="#HGVSName.format_cdna_coords-1120"><span class="linenos">1120</span></a><span class="sd">        the start and end coordinates are the same or not.</span>
</span><span id="HGVSName.format_cdna_coords-1121"><a href="#HGVSName.format_cdna_coords-1121"><span class="linenos">1121</span></a><span class="sd">        :return: a string representing the cDNA coordinates. If the start and end coordinates are the</span>
</span><span id="HGVSName.format_cdna_coords-1122"><a href="#HGVSName.format_cdna_coords-1122"><span class="linenos">1122</span></a><span class="sd">        same, it returns just the start coordinate. Otherwise, it returns a string in the format</span>
</span><span id="HGVSName.format_cdna_coords-1123"><a href="#HGVSName.format_cdna_coords-1123"><span class="linenos">1123</span></a><span class="sd">        &quot;start_end&quot;.</span>
</span><span id="HGVSName.format_cdna_coords-1124"><a href="#HGVSName.format_cdna_coords-1124"><span class="linenos">1124</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="HGVSName.format_cdna_coords-1125"><a href="#HGVSName.format_cdna_coords-1125"><span class="linenos">1125</span></a>
</span><span id="HGVSName.format_cdna_coords-1126"><a href="#HGVSName.format_cdna_coords-1126"><span class="linenos">1126</span></a>        <span class="c1"># Format coordinates.</span>
</span><span id="HGVSName.format_cdna_coords-1127"><a href="#HGVSName.format_cdna_coords-1127"><span class="linenos">1127</span></a>        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">cdna_start</span> <span class="o">==</span> <span class="bp">self</span><span class="o">.</span><span class="n">cdna_end</span><span class="p">:</span>
</span><span id="HGVSName.format_cdna_coords-1128"><a href="#HGVSName.format_cdna_coords-1128"><span class="linenos">1128</span></a>            <span class="k">return</span> <span class="nb">str</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">cdna_start</span><span class="p">)</span>
</span><span id="HGVSName.format_cdna_coords-1129"><a href="#HGVSName.format_cdna_coords-1129"><span class="linenos">1129</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName.format_cdna_coords-1130"><a href="#HGVSName.format_cdna_coords-1130"><span class="linenos">1130</span></a>            <span class="k">return</span> <span class="s2">&quot;</span><span class="si">%s</span><span class="s2">_</span><span class="si">%s</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">cdna_start</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">cdna_end</span><span class="p">)</span>
</span></pre></div>


            <div class="docstring"><p>The function <code><a href="#HGVSName.format_cdna_coords">format_cdna_coords</a></code> generates a string representing HGVS cDNA coordinates,
returning either the start coordinate or a string in the format "start_end" depending on whether
the start and end coordinates are the same or not.</p>

<h6 id="returns">Returns</h6>

<blockquote>
  <p>a string representing the cDNA coordinates. If the start and end coordinates are the
  same, it returns just the start coordinate. Otherwise, it returns a string in the format
  "start_end".</p>
</blockquote>
</div>


                            </div>
                            <div id="HGVSName.format_dna_allele" class="classattr">
                                        <input id="HGVSName.format_dna_allele-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr function">
            
        <span class="def">def</span>
        <span class="name">format_dna_allele</span><span class="signature pdoc-code condensed">(<span class="param"><span class="bp">self</span></span><span class="return-annotation">) -> <span class="nb">str</span>:</span></span>

                <label class="view-source-button" for="HGVSName.format_dna_allele-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#HGVSName.format_dna_allele"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="HGVSName.format_dna_allele-1132"><a href="#HGVSName.format_dna_allele-1132"><span class="linenos">1132</span></a>    <span class="k">def</span> <span class="nf">format_dna_allele</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
</span><span id="HGVSName.format_dna_allele-1133"><a href="#HGVSName.format_dna_allele-1133"><span class="linenos">1133</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="HGVSName.format_dna_allele-1134"><a href="#HGVSName.format_dna_allele-1134"><span class="linenos">1134</span></a><span class="sd">        The function `format_dna_allele` generates an HGVS DNA allele based on the mutation type and</span>
</span><span id="HGVSName.format_dna_allele-1135"><a href="#HGVSName.format_dna_allele-1135"><span class="linenos">1135</span></a><span class="sd">        alleles provided.</span>
</span><span id="HGVSName.format_dna_allele-1136"><a href="#HGVSName.format_dna_allele-1136"><span class="linenos">1136</span></a><span class="sd">        :return: The function `format_dna_allele` returns a string representing the HGVS DNA allele. The</span>
</span><span id="HGVSName.format_dna_allele-1137"><a href="#HGVSName.format_dna_allele-1137"><span class="linenos">1137</span></a><span class="sd">        specific format of the returned string depends on the value of the `mutation_type` attribute of</span>
</span><span id="HGVSName.format_dna_allele-1138"><a href="#HGVSName.format_dna_allele-1138"><span class="linenos">1138</span></a><span class="sd">        the object. The possible return values are:</span>
</span><span id="HGVSName.format_dna_allele-1139"><a href="#HGVSName.format_dna_allele-1139"><span class="linenos">1139</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="HGVSName.format_dna_allele-1140"><a href="#HGVSName.format_dna_allele-1140"><span class="linenos">1140</span></a>
</span><span id="HGVSName.format_dna_allele-1141"><a href="#HGVSName.format_dna_allele-1141"><span class="linenos">1141</span></a>        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">==</span> <span class="s2">&quot;=&quot;</span><span class="p">:</span>
</span><span id="HGVSName.format_dna_allele-1142"><a href="#HGVSName.format_dna_allele-1142"><span class="linenos">1142</span></a>            <span class="c1"># No change.</span>
</span><span id="HGVSName.format_dna_allele-1143"><a href="#HGVSName.format_dna_allele-1143"><span class="linenos">1143</span></a>            <span class="c1"># example: 101A=</span>
</span><span id="HGVSName.format_dna_allele-1144"><a href="#HGVSName.format_dna_allele-1144"><span class="linenos">1144</span></a>            <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">+</span> <span class="s2">&quot;=&quot;</span>
</span><span id="HGVSName.format_dna_allele-1145"><a href="#HGVSName.format_dna_allele-1145"><span class="linenos">1145</span></a>
</span><span id="HGVSName.format_dna_allele-1146"><a href="#HGVSName.format_dna_allele-1146"><span class="linenos">1146</span></a>        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">==</span> <span class="s2">&quot;&gt;&quot;</span><span class="p">:</span>
</span><span id="HGVSName.format_dna_allele-1147"><a href="#HGVSName.format_dna_allele-1147"><span class="linenos">1147</span></a>            <span class="c1"># SNP.</span>
</span><span id="HGVSName.format_dna_allele-1148"><a href="#HGVSName.format_dna_allele-1148"><span class="linenos">1148</span></a>            <span class="c1"># example: 101A&gt;C</span>
</span><span id="HGVSName.format_dna_allele-1149"><a href="#HGVSName.format_dna_allele-1149"><span class="linenos">1149</span></a>            <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">+</span> <span class="s2">&quot;&gt;&quot;</span> <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span>
</span><span id="HGVSName.format_dna_allele-1150"><a href="#HGVSName.format_dna_allele-1150"><span class="linenos">1150</span></a>
</span><span id="HGVSName.format_dna_allele-1151"><a href="#HGVSName.format_dna_allele-1151"><span class="linenos">1151</span></a>        <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">==</span> <span class="s2">&quot;delins&quot;</span><span class="p">:</span>
</span><span id="HGVSName.format_dna_allele-1152"><a href="#HGVSName.format_dna_allele-1152"><span class="linenos">1152</span></a>            <span class="c1"># Indel.</span>
</span><span id="HGVSName.format_dna_allele-1153"><a href="#HGVSName.format_dna_allele-1153"><span class="linenos">1153</span></a>            <span class="c1"># example: 112_117delAGGTCAinsTG, 112_117delinsTG</span>
</span><span id="HGVSName.format_dna_allele-1154"><a href="#HGVSName.format_dna_allele-1154"><span class="linenos">1154</span></a>            <span class="k">return</span> <span class="s2">&quot;del&quot;</span> <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">+</span> <span class="s2">&quot;ins&quot;</span> <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span>
</span><span id="HGVSName.format_dna_allele-1155"><a href="#HGVSName.format_dna_allele-1155"><span class="linenos">1155</span></a>
</span><span id="HGVSName.format_dna_allele-1156"><a href="#HGVSName.format_dna_allele-1156"><span class="linenos">1156</span></a>        <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="ow">in</span> <span class="p">(</span><span class="s2">&quot;del&quot;</span><span class="p">,</span> <span class="s2">&quot;dup&quot;</span><span class="p">):</span>
</span><span id="HGVSName.format_dna_allele-1157"><a href="#HGVSName.format_dna_allele-1157"><span class="linenos">1157</span></a>            <span class="c1"># Delete, duplication.</span>
</span><span id="HGVSName.format_dna_allele-1158"><a href="#HGVSName.format_dna_allele-1158"><span class="linenos">1158</span></a>            <span class="c1"># example: 1000_1003delATG, 1000_1003dupATG</span>
</span><span id="HGVSName.format_dna_allele-1159"><a href="#HGVSName.format_dna_allele-1159"><span class="linenos">1159</span></a>            <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span>
</span><span id="HGVSName.format_dna_allele-1160"><a href="#HGVSName.format_dna_allele-1160"><span class="linenos">1160</span></a>
</span><span id="HGVSName.format_dna_allele-1161"><a href="#HGVSName.format_dna_allele-1161"><span class="linenos">1161</span></a>        <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">==</span> <span class="s2">&quot;ins&quot;</span><span class="p">:</span>
</span><span id="HGVSName.format_dna_allele-1162"><a href="#HGVSName.format_dna_allele-1162"><span class="linenos">1162</span></a>            <span class="c1"># Insert.</span>
</span><span id="HGVSName.format_dna_allele-1163"><a href="#HGVSName.format_dna_allele-1163"><span class="linenos">1163</span></a>            <span class="c1"># example: 1000_1001insATG</span>
</span><span id="HGVSName.format_dna_allele-1164"><a href="#HGVSName.format_dna_allele-1164"><span class="linenos">1164</span></a>            <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span>
</span><span id="HGVSName.format_dna_allele-1165"><a href="#HGVSName.format_dna_allele-1165"><span class="linenos">1165</span></a>
</span><span id="HGVSName.format_dna_allele-1166"><a href="#HGVSName.format_dna_allele-1166"><span class="linenos">1166</span></a>        <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">==</span> <span class="s2">&quot;inv&quot;</span><span class="p">:</span>
</span><span id="HGVSName.format_dna_allele-1167"><a href="#HGVSName.format_dna_allele-1167"><span class="linenos">1167</span></a>            <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span>
</span><span id="HGVSName.format_dna_allele-1168"><a href="#HGVSName.format_dna_allele-1168"><span class="linenos">1168</span></a>
</span><span id="HGVSName.format_dna_allele-1169"><a href="#HGVSName.format_dna_allele-1169"><span class="linenos">1169</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName.format_dna_allele-1170"><a href="#HGVSName.format_dna_allele-1170"><span class="linenos">1170</span></a>            <span class="k">raise</span> <span class="ne">AssertionError</span><span class="p">(</span><span class="s2">&quot;unknown mutation type: &#39;</span><span class="si">%s</span><span class="s2">&#39;&quot;</span> <span class="o">%</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span><span class="p">)</span>
</span></pre></div>


            <div class="docstring"><p>The function <code><a href="#HGVSName.format_dna_allele">format_dna_allele</a></code> generates an HGVS DNA allele based on the mutation type and
alleles provided.</p>

<h6 id="returns">Returns</h6>

<blockquote>
  <p>The function <code><a href="#HGVSName.format_dna_allele">format_dna_allele</a></code> returns a string representing the HGVS DNA allele. The
  specific format of the returned string depends on the value of the <code><a href="#HGVSName.mutation_type">mutation_type</a></code> attribute of
  the object. The possible return values are:</p>
</blockquote>
</div>


                            </div>
                            <div id="HGVSName.format_cdna" class="classattr">
                                        <input id="HGVSName.format_cdna-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr function">
            
        <span class="def">def</span>
        <span class="name">format_cdna</span><span class="signature pdoc-code condensed">(<span class="param"><span class="bp">self</span></span><span class="return-annotation">) -> <span class="nb">str</span>:</span></span>

                <label class="view-source-button" for="HGVSName.format_cdna-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#HGVSName.format_cdna"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="HGVSName.format_cdna-1172"><a href="#HGVSName.format_cdna-1172"><span class="linenos">1172</span></a>    <span class="k">def</span> <span class="nf">format_cdna</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
</span><span id="HGVSName.format_cdna-1173"><a href="#HGVSName.format_cdna-1173"><span class="linenos">1173</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="HGVSName.format_cdna-1174"><a href="#HGVSName.format_cdna-1174"><span class="linenos">1174</span></a><span class="sd">        The function &quot;format_cdna&quot; generates an HGVS cDNA allele by combining the cDNA coordinates and</span>
</span><span id="HGVSName.format_cdna-1175"><a href="#HGVSName.format_cdna-1175"><span class="linenos">1175</span></a><span class="sd">        the DNA allele.</span>
</span><span id="HGVSName.format_cdna-1176"><a href="#HGVSName.format_cdna-1176"><span class="linenos">1176</span></a>
</span><span id="HGVSName.format_cdna-1177"><a href="#HGVSName.format_cdna-1177"><span class="linenos">1177</span></a><span class="sd">        Some examples include:</span>
</span><span id="HGVSName.format_cdna-1178"><a href="#HGVSName.format_cdna-1178"><span class="linenos">1178</span></a><span class="sd">          Substitution: 101A&gt;C,</span>
</span><span id="HGVSName.format_cdna-1179"><a href="#HGVSName.format_cdna-1179"><span class="linenos">1179</span></a><span class="sd">          Indel: 3428delCinsTA, 1000_1003delATG, 1000_1001insATG</span>
</span><span id="HGVSName.format_cdna-1180"><a href="#HGVSName.format_cdna-1180"><span class="linenos">1180</span></a>
</span><span id="HGVSName.format_cdna-1181"><a href="#HGVSName.format_cdna-1181"><span class="linenos">1181</span></a><span class="sd">        :return: a string that represents the HGVS cDNA allele.</span>
</span><span id="HGVSName.format_cdna-1182"><a href="#HGVSName.format_cdna-1182"><span class="linenos">1182</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="HGVSName.format_cdna-1183"><a href="#HGVSName.format_cdna-1183"><span class="linenos">1183</span></a>
</span><span id="HGVSName.format_cdna-1184"><a href="#HGVSName.format_cdna-1184"><span class="linenos">1184</span></a>        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">format_cdna_coords</span><span class="p">()</span> <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">format_dna_allele</span><span class="p">()</span>
</span></pre></div>


            <div class="docstring"><p>The function "format_cdna" generates an HGVS cDNA allele by combining the cDNA coordinates and
the DNA allele.</p>

<p>Some examples include:
  Substitution: 101A>C,
  Indel: 3428delCinsTA, 1000_1003delATG, 1000_1001insATG</p>

<h6 id="returns">Returns</h6>

<blockquote>
  <p>a string that represents the HGVS cDNA allele.</p>
</blockquote>
</div>


                            </div>
                            <div id="HGVSName.format_protein" class="classattr">
                                        <input id="HGVSName.format_protein-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr function">
            
        <span class="def">def</span>
        <span class="name">format_protein</span><span class="signature pdoc-code condensed">(<span class="param"><span class="bp">self</span></span><span class="return-annotation">) -> <span class="nb">str</span>:</span></span>

                <label class="view-source-button" for="HGVSName.format_protein-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#HGVSName.format_protein"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="HGVSName.format_protein-1186"><a href="#HGVSName.format_protein-1186"><span class="linenos">1186</span></a>    <span class="k">def</span> <span class="nf">format_protein</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
</span><span id="HGVSName.format_protein-1187"><a href="#HGVSName.format_protein-1187"><span class="linenos">1187</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="HGVSName.format_protein-1188"><a href="#HGVSName.format_protein-1188"><span class="linenos">1188</span></a><span class="sd">        The `format_protein` function generates an HGVS protein name based on different scenarios such</span>
</span><span id="HGVSName.format_protein-1189"><a href="#HGVSName.format_protein-1189"><span class="linenos">1189</span></a><span class="sd">        as no change, change, frameshift, and range change.</span>
</span><span id="HGVSName.format_protein-1190"><a href="#HGVSName.format_protein-1190"><span class="linenos">1190</span></a>
</span><span id="HGVSName.format_protein-1191"><a href="#HGVSName.format_protein-1191"><span class="linenos">1191</span></a><span class="sd">        Some examples include:</span>
</span><span id="HGVSName.format_protein-1192"><a href="#HGVSName.format_protein-1192"><span class="linenos">1192</span></a><span class="sd">          No change: Glu1161=</span>
</span><span id="HGVSName.format_protein-1193"><a href="#HGVSName.format_protein-1193"><span class="linenos">1193</span></a><span class="sd">          Change: Glu1161Ser</span>
</span><span id="HGVSName.format_protein-1194"><a href="#HGVSName.format_protein-1194"><span class="linenos">1194</span></a><span class="sd">          Frameshift: Glu1161_Ser1164?fs</span>
</span><span id="HGVSName.format_protein-1195"><a href="#HGVSName.format_protein-1195"><span class="linenos">1195</span></a>
</span><span id="HGVSName.format_protein-1196"><a href="#HGVSName.format_protein-1196"><span class="linenos">1196</span></a><span class="sd">        :return: The method `format_protein` returns a string representing the HGVS protein name.</span>
</span><span id="HGVSName.format_protein-1197"><a href="#HGVSName.format_protein-1197"><span class="linenos">1197</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="HGVSName.format_protein-1198"><a href="#HGVSName.format_protein-1198"><span class="linenos">1198</span></a>
</span><span id="HGVSName.format_protein-1199"><a href="#HGVSName.format_protein-1199"><span class="linenos">1199</span></a>        <span class="k">if</span> <span class="p">(</span>
</span><span id="HGVSName.format_protein-1200"><a href="#HGVSName.format_protein-1200"><span class="linenos">1200</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">start</span> <span class="o">==</span> <span class="bp">self</span><span class="o">.</span><span class="n">end</span>
</span><span id="HGVSName.format_protein-1201"><a href="#HGVSName.format_protein-1201"><span class="linenos">1201</span></a>            <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">==</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref2_allele</span> <span class="o">==</span> <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span>
</span><span id="HGVSName.format_protein-1202"><a href="#HGVSName.format_protein-1202"><span class="linenos">1202</span></a>        <span class="p">):</span>
</span><span id="HGVSName.format_protein-1203"><a href="#HGVSName.format_protein-1203"><span class="linenos">1203</span></a>            <span class="c1"># Match.</span>
</span><span id="HGVSName.format_protein-1204"><a href="#HGVSName.format_protein-1204"><span class="linenos">1204</span></a>            <span class="c1"># Example: Glu1161=</span>
</span><span id="HGVSName.format_protein-1205"><a href="#HGVSName.format_protein-1205"><span class="linenos">1205</span></a>            <span class="n">pep_extra</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">pep_extra</span> <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">pep_extra</span> <span class="k">else</span> <span class="s2">&quot;=&quot;</span>
</span><span id="HGVSName.format_protein-1206"><a href="#HGVSName.format_protein-1206"><span class="linenos">1206</span></a>            <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">start</span><span class="p">)</span> <span class="o">+</span> <span class="n">pep_extra</span>
</span><span id="HGVSName.format_protein-1207"><a href="#HGVSName.format_protein-1207"><span class="linenos">1207</span></a>
</span><span id="HGVSName.format_protein-1208"><a href="#HGVSName.format_protein-1208"><span class="linenos">1208</span></a>        <span class="k">elif</span> <span class="p">(</span>
</span><span id="HGVSName.format_protein-1209"><a href="#HGVSName.format_protein-1209"><span class="linenos">1209</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">start</span> <span class="o">==</span> <span class="bp">self</span><span class="o">.</span><span class="n">end</span>
</span><span id="HGVSName.format_protein-1210"><a href="#HGVSName.format_protein-1210"><span class="linenos">1210</span></a>            <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">==</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref2_allele</span>
</span><span id="HGVSName.format_protein-1211"><a href="#HGVSName.format_protein-1211"><span class="linenos">1211</span></a>            <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">!=</span> <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span>
</span><span id="HGVSName.format_protein-1212"><a href="#HGVSName.format_protein-1212"><span class="linenos">1212</span></a>        <span class="p">):</span>
</span><span id="HGVSName.format_protein-1213"><a href="#HGVSName.format_protein-1213"><span class="linenos">1213</span></a>            <span class="c1"># Change.</span>
</span><span id="HGVSName.format_protein-1214"><a href="#HGVSName.format_protein-1214"><span class="linenos">1214</span></a>            <span class="c1"># Example: Glu1161Ser</span>
</span><span id="HGVSName.format_protein-1215"><a href="#HGVSName.format_protein-1215"><span class="linenos">1215</span></a>            <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span> <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">start</span><span class="p">)</span> <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span> <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">pep_extra</span>
</span><span id="HGVSName.format_protein-1216"><a href="#HGVSName.format_protein-1216"><span class="linenos">1216</span></a>
</span><span id="HGVSName.format_protein-1217"><a href="#HGVSName.format_protein-1217"><span class="linenos">1217</span></a>        <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">start</span> <span class="o">!=</span> <span class="bp">self</span><span class="o">.</span><span class="n">end</span><span class="p">:</span>
</span><span id="HGVSName.format_protein-1218"><a href="#HGVSName.format_protein-1218"><span class="linenos">1218</span></a>            <span class="c1"># Range change.</span>
</span><span id="HGVSName.format_protein-1219"><a href="#HGVSName.format_protein-1219"><span class="linenos">1219</span></a>            <span class="c1"># Example: Glu1161_Ser1164?fs</span>
</span><span id="HGVSName.format_protein-1220"><a href="#HGVSName.format_protein-1220"><span class="linenos">1220</span></a>            <span class="k">return</span> <span class="p">(</span>
</span><span id="HGVSName.format_protein-1221"><a href="#HGVSName.format_protein-1221"><span class="linenos">1221</span></a>                <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span>
</span><span id="HGVSName.format_protein-1222"><a href="#HGVSName.format_protein-1222"><span class="linenos">1222</span></a>                <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">start</span><span class="p">)</span>
</span><span id="HGVSName.format_protein-1223"><a href="#HGVSName.format_protein-1223"><span class="linenos">1223</span></a>                <span class="o">+</span> <span class="s2">&quot;_&quot;</span>
</span><span id="HGVSName.format_protein-1224"><a href="#HGVSName.format_protein-1224"><span class="linenos">1224</span></a>                <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref2_allele</span>
</span><span id="HGVSName.format_protein-1225"><a href="#HGVSName.format_protein-1225"><span class="linenos">1225</span></a>                <span class="o">+</span> <span class="nb">str</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">end</span><span class="p">)</span>
</span><span id="HGVSName.format_protein-1226"><a href="#HGVSName.format_protein-1226"><span class="linenos">1226</span></a>                <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">pep_extra</span>
</span><span id="HGVSName.format_protein-1227"><a href="#HGVSName.format_protein-1227"><span class="linenos">1227</span></a>            <span class="p">)</span>
</span><span id="HGVSName.format_protein-1228"><a href="#HGVSName.format_protein-1228"><span class="linenos">1228</span></a>        <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">pep_extra</span><span class="p">:</span>
</span><span id="HGVSName.format_protein-1229"><a href="#HGVSName.format_protein-1229"><span class="linenos">1229</span></a>            <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">pep_extra</span>
</span><span id="HGVSName.format_protein-1230"><a href="#HGVSName.format_protein-1230"><span class="linenos">1230</span></a>
</span><span id="HGVSName.format_protein-1231"><a href="#HGVSName.format_protein-1231"><span class="linenos">1231</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName.format_protein-1232"><a href="#HGVSName.format_protein-1232"><span class="linenos">1232</span></a>            <span class="k">return</span> <span class="kc">None</span>
</span></pre></div>


            <div class="docstring"><p>The <code><a href="#HGVSName.format_protein">format_protein</a></code> function generates an HGVS protein name based on different scenarios such
as no change, change, frameshift, and range change.</p>

<p>Some examples include:
  No change: Glu1161=
  Change: Glu1161Ser
  Frameshift: Glu1161_Ser1164?fs</p>

<h6 id="returns">Returns</h6>

<blockquote>
  <p>The method <code><a href="#HGVSName.format_protein">format_protein</a></code> returns a string representing the HGVS protein name.</p>
</blockquote>
</div>


                            </div>
                            <div id="HGVSName.format_coords" class="classattr">
                                        <input id="HGVSName.format_coords-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr function">
            
        <span class="def">def</span>
        <span class="name">format_coords</span><span class="signature pdoc-code condensed">(<span class="param"><span class="bp">self</span></span><span class="return-annotation">) -> <span class="nb">str</span>:</span></span>

                <label class="view-source-button" for="HGVSName.format_coords-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#HGVSName.format_coords"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="HGVSName.format_coords-1234"><a href="#HGVSName.format_coords-1234"><span class="linenos">1234</span></a>    <span class="k">def</span> <span class="nf">format_coords</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
</span><span id="HGVSName.format_coords-1235"><a href="#HGVSName.format_coords-1235"><span class="linenos">1235</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="HGVSName.format_coords-1236"><a href="#HGVSName.format_coords-1236"><span class="linenos">1236</span></a><span class="sd">        The function `format_coords` generates a string representation of HGVS cDNA coordinates.</span>
</span><span id="HGVSName.format_coords-1237"><a href="#HGVSName.format_coords-1237"><span class="linenos">1237</span></a><span class="sd">        :return: a string that represents the HGVS cDNA coordinates. If the start and end coordinates</span>
</span><span id="HGVSName.format_coords-1238"><a href="#HGVSName.format_coords-1238"><span class="linenos">1238</span></a><span class="sd">        are the same, it returns just the start coordinate. Otherwise, it returns a string in the format</span>
</span><span id="HGVSName.format_coords-1239"><a href="#HGVSName.format_coords-1239"><span class="linenos">1239</span></a><span class="sd">        &quot;start_end&quot;.</span>
</span><span id="HGVSName.format_coords-1240"><a href="#HGVSName.format_coords-1240"><span class="linenos">1240</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="HGVSName.format_coords-1241"><a href="#HGVSName.format_coords-1241"><span class="linenos">1241</span></a>
</span><span id="HGVSName.format_coords-1242"><a href="#HGVSName.format_coords-1242"><span class="linenos">1242</span></a>        <span class="c1"># Format coordinates.</span>
</span><span id="HGVSName.format_coords-1243"><a href="#HGVSName.format_coords-1243"><span class="linenos">1243</span></a>        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">start</span> <span class="o">==</span> <span class="bp">self</span><span class="o">.</span><span class="n">end</span><span class="p">:</span>
</span><span id="HGVSName.format_coords-1244"><a href="#HGVSName.format_coords-1244"><span class="linenos">1244</span></a>            <span class="k">return</span> <span class="nb">str</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">start</span><span class="p">)</span>
</span><span id="HGVSName.format_coords-1245"><a href="#HGVSName.format_coords-1245"><span class="linenos">1245</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName.format_coords-1246"><a href="#HGVSName.format_coords-1246"><span class="linenos">1246</span></a>            <span class="k">return</span> <span class="s2">&quot;</span><span class="si">%s</span><span class="s2">_</span><span class="si">%s</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">start</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">end</span><span class="p">)</span>
</span></pre></div>


            <div class="docstring"><p>The function <code><a href="#HGVSName.format_coords">format_coords</a></code> generates a string representation of HGVS cDNA coordinates.</p>

<h6 id="returns">Returns</h6>

<blockquote>
  <p>a string that represents the HGVS cDNA coordinates. If the start and end coordinates
  are the same, it returns just the start coordinate. Otherwise, it returns a string in the format
  "start_end".</p>
</blockquote>
</div>


                            </div>
                            <div id="HGVSName.format_genome" class="classattr">
                                        <input id="HGVSName.format_genome-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr function">
            
        <span class="def">def</span>
        <span class="name">format_genome</span><span class="signature pdoc-code condensed">(<span class="param"><span class="bp">self</span></span><span class="return-annotation">) -> <span class="nb">str</span>:</span></span>

                <label class="view-source-button" for="HGVSName.format_genome-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#HGVSName.format_genome"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="HGVSName.format_genome-1248"><a href="#HGVSName.format_genome-1248"><span class="linenos">1248</span></a>    <span class="k">def</span> <span class="nf">format_genome</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
</span><span id="HGVSName.format_genome-1249"><a href="#HGVSName.format_genome-1249"><span class="linenos">1249</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="HGVSName.format_genome-1250"><a href="#HGVSName.format_genome-1250"><span class="linenos">1250</span></a><span class="sd">        The function &quot;format_genome&quot; generates an HGVS genomic allele by combining the formatted</span>
</span><span id="HGVSName.format_genome-1251"><a href="#HGVSName.format_genome-1251"><span class="linenos">1251</span></a><span class="sd">        coordinates and DNA allele.</span>
</span><span id="HGVSName.format_genome-1252"><a href="#HGVSName.format_genome-1252"><span class="linenos">1252</span></a>
</span><span id="HGVSName.format_genome-1253"><a href="#HGVSName.format_genome-1253"><span class="linenos">1253</span></a><span class="sd">        Some examples include:</span>
</span><span id="HGVSName.format_genome-1254"><a href="#HGVSName.format_genome-1254"><span class="linenos">1254</span></a><span class="sd">          Substitution: 1000100A&gt;T</span>
</span><span id="HGVSName.format_genome-1255"><a href="#HGVSName.format_genome-1255"><span class="linenos">1255</span></a><span class="sd">          Indel: 1000100_1000102delATG</span>
</span><span id="HGVSName.format_genome-1256"><a href="#HGVSName.format_genome-1256"><span class="linenos">1256</span></a>
</span><span id="HGVSName.format_genome-1257"><a href="#HGVSName.format_genome-1257"><span class="linenos">1257</span></a><span class="sd">        :return: a string that represents the HGVS genomic allele.</span>
</span><span id="HGVSName.format_genome-1258"><a href="#HGVSName.format_genome-1258"><span class="linenos">1258</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="HGVSName.format_genome-1259"><a href="#HGVSName.format_genome-1259"><span class="linenos">1259</span></a>
</span><span id="HGVSName.format_genome-1260"><a href="#HGVSName.format_genome-1260"><span class="linenos">1260</span></a>        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">format_coords</span><span class="p">()</span> <span class="o">+</span> <span class="bp">self</span><span class="o">.</span><span class="n">format_dna_allele</span><span class="p">()</span>
</span></pre></div>


            <div class="docstring"><p>The function "format_genome" generates an HGVS genomic allele by combining the formatted
coordinates and DNA allele.</p>

<p>Some examples include:
  Substitution: 1000100A>T
  Indel: 1000100_1000102delATG</p>

<h6 id="returns">Returns</h6>

<blockquote>
  <p>a string that represents the HGVS genomic allele.</p>
</blockquote>
</div>


                            </div>
                            <div id="HGVSName.get_raw_coords" class="classattr">
                                        <input id="HGVSName.get_raw_coords-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr function">
            
        <span class="def">def</span>
        <span class="name">get_raw_coords</span><span class="signature pdoc-code condensed">(<span class="param"><span class="bp">self</span>, </span><span class="param"><span class="n">transcript</span><span class="p">:</span> <span class="nb">object</span> <span class="o">=</span> <span class="kc">None</span></span><span class="return-annotation">) -> <span class="nb">tuple</span>:</span></span>

                <label class="view-source-button" for="HGVSName.get_raw_coords-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#HGVSName.get_raw_coords"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="HGVSName.get_raw_coords-1262"><a href="#HGVSName.get_raw_coords-1262"><span class="linenos">1262</span></a>    <span class="k">def</span> <span class="nf">get_raw_coords</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">transcript</span><span class="p">:</span> <span class="nb">object</span> <span class="o">=</span> <span class="kc">None</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">tuple</span><span class="p">:</span>
</span><span id="HGVSName.get_raw_coords-1263"><a href="#HGVSName.get_raw_coords-1263"><span class="linenos">1263</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="HGVSName.get_raw_coords-1264"><a href="#HGVSName.get_raw_coords-1264"><span class="linenos">1264</span></a><span class="sd">        The function `get_raw_coords` returns the genomic coordinates based on the given transcript or</span>
</span><span id="HGVSName.get_raw_coords-1265"><a href="#HGVSName.get_raw_coords-1265"><span class="linenos">1265</span></a><span class="sd">        the provided chromosomal coordinates.</span>
</span><span id="HGVSName.get_raw_coords-1266"><a href="#HGVSName.get_raw_coords-1266"><span class="linenos">1266</span></a>
</span><span id="HGVSName.get_raw_coords-1267"><a href="#HGVSName.get_raw_coords-1267"><span class="linenos">1267</span></a><span class="sd">        :param transcript: The `transcript` parameter is an object that represents a transcript. It is</span>
</span><span id="HGVSName.get_raw_coords-1268"><a href="#HGVSName.get_raw_coords-1268"><span class="linenos">1268</span></a><span class="sd">        used to retrieve genomic coordinates based on the type of HGVS name (`self.kind`). The</span>
</span><span id="HGVSName.get_raw_coords-1269"><a href="#HGVSName.get_raw_coords-1269"><span class="linenos">1269</span></a><span class="sd">        `transcript` object should have the following attributes and methods:</span>
</span><span id="HGVSName.get_raw_coords-1270"><a href="#HGVSName.get_raw_coords-1270"><span class="linenos">1270</span></a><span class="sd">        :type transcript: object</span>
</span><span id="HGVSName.get_raw_coords-1271"><a href="#HGVSName.get_raw_coords-1271"><span class="linenos">1271</span></a><span class="sd">        :return: a tuple containing the genomic coordinates. The tuple consists of three elements: the</span>
</span><span id="HGVSName.get_raw_coords-1272"><a href="#HGVSName.get_raw_coords-1272"><span class="linenos">1272</span></a><span class="sd">        chromosome, the start position, and the end position.</span>
</span><span id="HGVSName.get_raw_coords-1273"><a href="#HGVSName.get_raw_coords-1273"><span class="linenos">1273</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="HGVSName.get_raw_coords-1274"><a href="#HGVSName.get_raw_coords-1274"><span class="linenos">1274</span></a>
</span><span id="HGVSName.get_raw_coords-1275"><a href="#HGVSName.get_raw_coords-1275"><span class="linenos">1275</span></a>        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">kind</span> <span class="ow">in</span> <span class="p">(</span><span class="s2">&quot;c&quot;</span><span class="p">,</span> <span class="s2">&quot;n&quot;</span><span class="p">):</span>
</span><span id="HGVSName.get_raw_coords-1276"><a href="#HGVSName.get_raw_coords-1276"><span class="linenos">1276</span></a>            <span class="n">chrom</span> <span class="o">=</span> <span class="n">transcript</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom</span>
</span><span id="HGVSName.get_raw_coords-1277"><a href="#HGVSName.get_raw_coords-1277"><span class="linenos">1277</span></a>            <span class="n">start</span> <span class="o">=</span> <span class="n">transcript</span><span class="o">.</span><span class="n">cdna_to_genomic_coord</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">cdna_start</span><span class="p">)</span>
</span><span id="HGVSName.get_raw_coords-1278"><a href="#HGVSName.get_raw_coords-1278"><span class="linenos">1278</span></a>            <span class="n">end</span> <span class="o">=</span> <span class="n">transcript</span><span class="o">.</span><span class="n">cdna_to_genomic_coord</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">cdna_end</span><span class="p">)</span>
</span><span id="HGVSName.get_raw_coords-1279"><a href="#HGVSName.get_raw_coords-1279"><span class="linenos">1279</span></a>
</span><span id="HGVSName.get_raw_coords-1280"><a href="#HGVSName.get_raw_coords-1280"><span class="linenos">1280</span></a>            <span class="k">if</span> <span class="ow">not</span> <span class="n">transcript</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">is_forward_strand</span><span class="p">:</span>
</span><span id="HGVSName.get_raw_coords-1281"><a href="#HGVSName.get_raw_coords-1281"><span class="linenos">1281</span></a>                <span class="n">start</span><span class="p">,</span> <span class="n">end</span> <span class="o">=</span> <span class="n">end</span><span class="p">,</span> <span class="n">start</span>
</span><span id="HGVSName.get_raw_coords-1282"><a href="#HGVSName.get_raw_coords-1282"><span class="linenos">1282</span></a>
</span><span id="HGVSName.get_raw_coords-1283"><a href="#HGVSName.get_raw_coords-1283"><span class="linenos">1283</span></a>            <span class="k">if</span> <span class="n">start</span> <span class="o">&gt;</span> <span class="n">end</span><span class="p">:</span>
</span><span id="HGVSName.get_raw_coords-1284"><a href="#HGVSName.get_raw_coords-1284"><span class="linenos">1284</span></a>                <span class="k">raise</span> <span class="ne">AssertionError</span><span class="p">(</span><span class="s2">&quot;cdna_start cannot be greater than cdna_end&quot;</span><span class="p">)</span>
</span><span id="HGVSName.get_raw_coords-1285"><a href="#HGVSName.get_raw_coords-1285"><span class="linenos">1285</span></a>        <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">kind</span> <span class="ow">in</span> <span class="p">(</span><span class="s2">&quot;g&quot;</span><span class="p">,</span> <span class="s2">&quot;m&quot;</span><span class="p">):</span>
</span><span id="HGVSName.get_raw_coords-1286"><a href="#HGVSName.get_raw_coords-1286"><span class="linenos">1286</span></a>            <span class="n">chrom</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">chrom</span>
</span><span id="HGVSName.get_raw_coords-1287"><a href="#HGVSName.get_raw_coords-1287"><span class="linenos">1287</span></a>            <span class="n">start</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">start</span>
</span><span id="HGVSName.get_raw_coords-1288"><a href="#HGVSName.get_raw_coords-1288"><span class="linenos">1288</span></a>            <span class="n">end</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">end</span>
</span><span id="HGVSName.get_raw_coords-1289"><a href="#HGVSName.get_raw_coords-1289"><span class="linenos">1289</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName.get_raw_coords-1290"><a href="#HGVSName.get_raw_coords-1290"><span class="linenos">1290</span></a>            <span class="k">raise</span> <span class="ne">NotImplementedError</span><span class="p">(</span>
</span><span id="HGVSName.get_raw_coords-1291"><a href="#HGVSName.get_raw_coords-1291"><span class="linenos">1291</span></a>                <span class="s1">&#39;Coordinates are not available for this kind of HGVS name &quot;</span><span class="si">%s</span><span class="s1">&quot;&#39;</span>
</span><span id="HGVSName.get_raw_coords-1292"><a href="#HGVSName.get_raw_coords-1292"><span class="linenos">1292</span></a>                <span class="o">%</span> <span class="bp">self</span><span class="o">.</span><span class="n">kind</span>
</span><span id="HGVSName.get_raw_coords-1293"><a href="#HGVSName.get_raw_coords-1293"><span class="linenos">1293</span></a>            <span class="p">)</span>
</span><span id="HGVSName.get_raw_coords-1294"><a href="#HGVSName.get_raw_coords-1294"><span class="linenos">1294</span></a>
</span><span id="HGVSName.get_raw_coords-1295"><a href="#HGVSName.get_raw_coords-1295"><span class="linenos">1295</span></a>        <span class="c1"># Check coordinate span is equal to reference bases</span>
</span><span id="HGVSName.get_raw_coords-1296"><a href="#HGVSName.get_raw_coords-1296"><span class="linenos">1296</span></a>        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span><span class="p">:</span>
</span><span id="HGVSName.get_raw_coords-1297"><a href="#HGVSName.get_raw_coords-1297"><span class="linenos">1297</span></a>            <span class="n">coordinate_span</span> <span class="o">=</span> <span class="n">end</span> <span class="o">-</span> <span class="n">start</span> <span class="o">+</span> <span class="mi">1</span>  <span class="c1"># Ref will always be &gt;=1 base</span>
</span><span id="HGVSName.get_raw_coords-1298"><a href="#HGVSName.get_raw_coords-1298"><span class="linenos">1298</span></a>            <span class="n">ref_length</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span><span class="p">)</span>
</span><span id="HGVSName.get_raw_coords-1299"><a href="#HGVSName.get_raw_coords-1299"><span class="linenos">1299</span></a>            <span class="k">if</span> <span class="n">coordinate_span</span> <span class="o">!=</span> <span class="n">ref_length</span><span class="p">:</span>
</span><span id="HGVSName.get_raw_coords-1300"><a href="#HGVSName.get_raw_coords-1300"><span class="linenos">1300</span></a>                <span class="k">raise</span> <span class="n">InvalidHGVSName</span><span class="p">(</span>
</span><span id="HGVSName.get_raw_coords-1301"><a href="#HGVSName.get_raw_coords-1301"><span class="linenos">1301</span></a>                    <span class="s2">&quot;Coordinate span (</span><span class="si">%d</span><span class="s2">) not equal to ref length </span><span class="si">%d</span><span class="s2">&quot;</span>
</span><span id="HGVSName.get_raw_coords-1302"><a href="#HGVSName.get_raw_coords-1302"><span class="linenos">1302</span></a>                    <span class="o">%</span> <span class="p">(</span><span class="n">coordinate_span</span><span class="p">,</span> <span class="n">ref_length</span><span class="p">)</span>
</span><span id="HGVSName.get_raw_coords-1303"><a href="#HGVSName.get_raw_coords-1303"><span class="linenos">1303</span></a>                <span class="p">)</span>
</span><span id="HGVSName.get_raw_coords-1304"><a href="#HGVSName.get_raw_coords-1304"><span class="linenos">1304</span></a>
</span><span id="HGVSName.get_raw_coords-1305"><a href="#HGVSName.get_raw_coords-1305"><span class="linenos">1305</span></a>        <span class="k">return</span> <span class="n">chrom</span><span class="p">,</span> <span class="n">start</span><span class="p">,</span> <span class="n">end</span>
</span></pre></div>


            <div class="docstring"><p>The function <code><a href="#HGVSName.get_raw_coords">get_raw_coords</a></code> returns the genomic coordinates based on the given transcript or
the provided chromosomal coordinates.</p>

<h6 id="parameters">Parameters</h6>

<ul>
<li><strong>transcript</strong>:  The <code><a href="#HGVSName.transcript">transcript</a></code> parameter is an object that represents a transcript. It is
used to retrieve genomic coordinates based on the type of HGVS name (<code>self.kind</code>). The
<code><a href="#HGVSName.transcript">transcript</a></code> object should have the following attributes and methods:</li>
</ul>

<h6 id="returns">Returns</h6>

<blockquote>
  <p>a tuple containing the genomic coordinates. The tuple consists of three elements: the
  chromosome, the start position, and the end position.</p>
</blockquote>
</div>


                            </div>
                            <div id="HGVSName.get_ref_coords" class="classattr">
                                        <input id="HGVSName.get_ref_coords-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr function">
            
        <span class="def">def</span>
        <span class="name">get_ref_coords</span><span class="signature pdoc-code condensed">(<span class="param"><span class="bp">self</span>, </span><span class="param"><span class="n">transcript</span><span class="p">:</span> <span class="nb">object</span> <span class="o">=</span> <span class="kc">None</span></span><span class="return-annotation">) -> <span class="nb">tuple</span>:</span></span>

                <label class="view-source-button" for="HGVSName.get_ref_coords-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#HGVSName.get_ref_coords"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="HGVSName.get_ref_coords-1307"><a href="#HGVSName.get_ref_coords-1307"><span class="linenos">1307</span></a>    <span class="k">def</span> <span class="nf">get_ref_coords</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">transcript</span><span class="p">:</span> <span class="nb">object</span> <span class="o">=</span> <span class="kc">None</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">tuple</span><span class="p">:</span>
</span><span id="HGVSName.get_ref_coords-1308"><a href="#HGVSName.get_ref_coords-1308"><span class="linenos">1308</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="HGVSName.get_ref_coords-1309"><a href="#HGVSName.get_ref_coords-1309"><span class="linenos">1309</span></a><span class="sd">        The function &quot;get_ref_coords&quot; returns the genomic coordinates of the reference allele, taking</span>
</span><span id="HGVSName.get_ref_coords-1310"><a href="#HGVSName.get_ref_coords-1310"><span class="linenos">1310</span></a><span class="sd">        into account different mutation types.</span>
</span><span id="HGVSName.get_ref_coords-1311"><a href="#HGVSName.get_ref_coords-1311"><span class="linenos">1311</span></a>
</span><span id="HGVSName.get_ref_coords-1312"><a href="#HGVSName.get_ref_coords-1312"><span class="linenos">1312</span></a><span class="sd">        :param transcript: The `transcript` parameter is an optional object that represents a transcript</span>
</span><span id="HGVSName.get_ref_coords-1313"><a href="#HGVSName.get_ref_coords-1313"><span class="linenos">1313</span></a><span class="sd">        or gene. It is used to retrieve the genomic coordinates of the reference allele</span>
</span><span id="HGVSName.get_ref_coords-1314"><a href="#HGVSName.get_ref_coords-1314"><span class="linenos">1314</span></a><span class="sd">        :type transcript: object</span>
</span><span id="HGVSName.get_ref_coords-1315"><a href="#HGVSName.get_ref_coords-1315"><span class="linenos">1315</span></a><span class="sd">        :return: a tuple containing the genomic coordinates of the reference allele. The tuple consists</span>
</span><span id="HGVSName.get_ref_coords-1316"><a href="#HGVSName.get_ref_coords-1316"><span class="linenos">1316</span></a><span class="sd">        of three elements: the chromosome, the start position, and the end position.</span>
</span><span id="HGVSName.get_ref_coords-1317"><a href="#HGVSName.get_ref_coords-1317"><span class="linenos">1317</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="HGVSName.get_ref_coords-1318"><a href="#HGVSName.get_ref_coords-1318"><span class="linenos">1318</span></a>
</span><span id="HGVSName.get_ref_coords-1319"><a href="#HGVSName.get_ref_coords-1319"><span class="linenos">1319</span></a>        <span class="n">chrom</span><span class="p">,</span> <span class="n">start</span><span class="p">,</span> <span class="n">end</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">get_raw_coords</span><span class="p">(</span><span class="n">transcript</span><span class="p">)</span>
</span><span id="HGVSName.get_ref_coords-1320"><a href="#HGVSName.get_ref_coords-1320"><span class="linenos">1320</span></a>
</span><span id="HGVSName.get_ref_coords-1321"><a href="#HGVSName.get_ref_coords-1321"><span class="linenos">1321</span></a>        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">==</span> <span class="s2">&quot;ins&quot;</span><span class="p">:</span>
</span><span id="HGVSName.get_ref_coords-1322"><a href="#HGVSName.get_ref_coords-1322"><span class="linenos">1322</span></a>            <span class="c1"># Inserts have empty interval.</span>
</span><span id="HGVSName.get_ref_coords-1323"><a href="#HGVSName.get_ref_coords-1323"><span class="linenos">1323</span></a>            <span class="k">if</span> <span class="n">start</span> <span class="o">&lt;</span> <span class="n">end</span><span class="p">:</span>
</span><span id="HGVSName.get_ref_coords-1324"><a href="#HGVSName.get_ref_coords-1324"><span class="linenos">1324</span></a>                <span class="n">start</span> <span class="o">+=</span> <span class="mi">1</span>
</span><span id="HGVSName.get_ref_coords-1325"><a href="#HGVSName.get_ref_coords-1325"><span class="linenos">1325</span></a>                <span class="n">end</span> <span class="o">-=</span> <span class="mi">1</span>
</span><span id="HGVSName.get_ref_coords-1326"><a href="#HGVSName.get_ref_coords-1326"><span class="linenos">1326</span></a>            <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName.get_ref_coords-1327"><a href="#HGVSName.get_ref_coords-1327"><span class="linenos">1327</span></a>                <span class="n">end</span> <span class="o">=</span> <span class="n">start</span> <span class="o">-</span> <span class="mi">1</span>
</span><span id="HGVSName.get_ref_coords-1328"><a href="#HGVSName.get_ref_coords-1328"><span class="linenos">1328</span></a>
</span><span id="HGVSName.get_ref_coords-1329"><a href="#HGVSName.get_ref_coords-1329"><span class="linenos">1329</span></a>        <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">==</span> <span class="s2">&quot;dup&quot;</span><span class="p">:</span>
</span><span id="HGVSName.get_ref_coords-1330"><a href="#HGVSName.get_ref_coords-1330"><span class="linenos">1330</span></a>            <span class="n">end</span> <span class="o">=</span> <span class="n">start</span> <span class="o">-</span> <span class="mi">1</span>
</span><span id="HGVSName.get_ref_coords-1331"><a href="#HGVSName.get_ref_coords-1331"><span class="linenos">1331</span></a>        <span class="k">return</span> <span class="n">chrom</span><span class="p">,</span> <span class="n">start</span><span class="p">,</span> <span class="n">end</span>
</span></pre></div>


            <div class="docstring"><p>The function "get_ref_coords" returns the genomic coordinates of the reference allele, taking
into account different mutation types.</p>

<h6 id="parameters">Parameters</h6>

<ul>
<li><strong>transcript</strong>:  The <code><a href="#HGVSName.transcript">transcript</a></code> parameter is an optional object that represents a transcript
or gene. It is used to retrieve the genomic coordinates of the reference allele</li>
</ul>

<h6 id="returns">Returns</h6>

<blockquote>
  <p>a tuple containing the genomic coordinates of the reference allele. The tuple consists
  of three elements: the chromosome, the start position, and the end position.</p>
</blockquote>
</div>


                            </div>
                            <div id="HGVSName.get_vcf_coords" class="classattr">
                                        <input id="HGVSName.get_vcf_coords-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr function">
            
        <span class="def">def</span>
        <span class="name">get_vcf_coords</span><span class="signature pdoc-code condensed">(<span class="param"><span class="bp">self</span>, </span><span class="param"><span class="n">transcript</span><span class="p">:</span> <span class="nb">object</span> <span class="o">=</span> <span class="kc">None</span></span><span class="return-annotation">) -> <span class="nb">tuple</span>:</span></span>

                <label class="view-source-button" for="HGVSName.get_vcf_coords-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#HGVSName.get_vcf_coords"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="HGVSName.get_vcf_coords-1333"><a href="#HGVSName.get_vcf_coords-1333"><span class="linenos">1333</span></a>    <span class="k">def</span> <span class="nf">get_vcf_coords</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">transcript</span><span class="p">:</span> <span class="nb">object</span> <span class="o">=</span> <span class="kc">None</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">tuple</span><span class="p">:</span>
</span><span id="HGVSName.get_vcf_coords-1334"><a href="#HGVSName.get_vcf_coords-1334"><span class="linenos">1334</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="HGVSName.get_vcf_coords-1335"><a href="#HGVSName.get_vcf_coords-1335"><span class="linenos">1335</span></a><span class="sd">        The function &quot;get_vcf_coords&quot; returns the genomic coordinates of the reference allele in</span>
</span><span id="HGVSName.get_vcf_coords-1336"><a href="#HGVSName.get_vcf_coords-1336"><span class="linenos">1336</span></a><span class="sd">        VCF-style, with left-padding for indels.</span>
</span><span id="HGVSName.get_vcf_coords-1337"><a href="#HGVSName.get_vcf_coords-1337"><span class="linenos">1337</span></a>
</span><span id="HGVSName.get_vcf_coords-1338"><a href="#HGVSName.get_vcf_coords-1338"><span class="linenos">1338</span></a><span class="sd">        :param transcript: The `transcript` parameter is an object that represents a transcript or gene.</span>
</span><span id="HGVSName.get_vcf_coords-1339"><a href="#HGVSName.get_vcf_coords-1339"><span class="linenos">1339</span></a><span class="sd">        It is used to retrieve the genomic coordinates of the reference allele</span>
</span><span id="HGVSName.get_vcf_coords-1340"><a href="#HGVSName.get_vcf_coords-1340"><span class="linenos">1340</span></a><span class="sd">        :type transcript: object</span>
</span><span id="HGVSName.get_vcf_coords-1341"><a href="#HGVSName.get_vcf_coords-1341"><span class="linenos">1341</span></a><span class="sd">        :return: a tuple containing the genomic coordinates of the reference allele in VCF-style. The</span>
</span><span id="HGVSName.get_vcf_coords-1342"><a href="#HGVSName.get_vcf_coords-1342"><span class="linenos">1342</span></a><span class="sd">        tuple consists of three elements: the chromosome, the start position, and the end position.</span>
</span><span id="HGVSName.get_vcf_coords-1343"><a href="#HGVSName.get_vcf_coords-1343"><span class="linenos">1343</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="HGVSName.get_vcf_coords-1344"><a href="#HGVSName.get_vcf_coords-1344"><span class="linenos">1344</span></a>
</span><span id="HGVSName.get_vcf_coords-1345"><a href="#HGVSName.get_vcf_coords-1345"><span class="linenos">1345</span></a>        <span class="n">chrom</span><span class="p">,</span> <span class="n">start</span><span class="p">,</span> <span class="n">end</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">get_ref_coords</span><span class="p">(</span><span class="n">transcript</span><span class="p">)</span>
</span><span id="HGVSName.get_vcf_coords-1346"><a href="#HGVSName.get_vcf_coords-1346"><span class="linenos">1346</span></a>
</span><span id="HGVSName.get_vcf_coords-1347"><a href="#HGVSName.get_vcf_coords-1347"><span class="linenos">1347</span></a>        <span class="c1"># Inserts and deletes require left-padding by 1 base</span>
</span><span id="HGVSName.get_vcf_coords-1348"><a href="#HGVSName.get_vcf_coords-1348"><span class="linenos">1348</span></a>        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="ow">in</span> <span class="p">(</span><span class="s2">&quot;=&quot;</span><span class="p">,</span> <span class="s2">&quot;&gt;&quot;</span><span class="p">):</span>
</span><span id="HGVSName.get_vcf_coords-1349"><a href="#HGVSName.get_vcf_coords-1349"><span class="linenos">1349</span></a>            <span class="k">pass</span>
</span><span id="HGVSName.get_vcf_coords-1350"><a href="#HGVSName.get_vcf_coords-1350"><span class="linenos">1350</span></a>        <span class="k">elif</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="ow">in</span> <span class="p">(</span><span class="s2">&quot;del&quot;</span><span class="p">,</span> <span class="s2">&quot;ins&quot;</span><span class="p">,</span> <span class="s2">&quot;dup&quot;</span><span class="p">,</span> <span class="s2">&quot;delins&quot;</span><span class="p">):</span>
</span><span id="HGVSName.get_vcf_coords-1351"><a href="#HGVSName.get_vcf_coords-1351"><span class="linenos">1351</span></a>            <span class="c1"># Indels have left-padding.</span>
</span><span id="HGVSName.get_vcf_coords-1352"><a href="#HGVSName.get_vcf_coords-1352"><span class="linenos">1352</span></a>            <span class="n">start</span> <span class="o">-=</span> <span class="mi">1</span>
</span><span id="HGVSName.get_vcf_coords-1353"><a href="#HGVSName.get_vcf_coords-1353"><span class="linenos">1353</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName.get_vcf_coords-1354"><a href="#HGVSName.get_vcf_coords-1354"><span class="linenos">1354</span></a>            <span class="k">raise</span> <span class="ne">NotImplementedError</span><span class="p">(</span><span class="s2">&quot;Unknown mutation_type &#39;</span><span class="si">%s</span><span class="s2">&#39;&quot;</span> <span class="o">%</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span><span class="p">)</span>
</span><span id="HGVSName.get_vcf_coords-1355"><a href="#HGVSName.get_vcf_coords-1355"><span class="linenos">1355</span></a>        <span class="k">return</span> <span class="n">chrom</span><span class="p">,</span> <span class="n">start</span><span class="p">,</span> <span class="n">end</span>
</span></pre></div>


            <div class="docstring"><p>The function "get_vcf_coords" returns the genomic coordinates of the reference allele in
VCF-style, with left-padding for indels.</p>

<h6 id="parameters">Parameters</h6>

<ul>
<li><strong>transcript</strong>:  The <code><a href="#HGVSName.transcript">transcript</a></code> parameter is an object that represents a transcript or gene.
It is used to retrieve the genomic coordinates of the reference allele</li>
</ul>

<h6 id="returns">Returns</h6>

<blockquote>
  <p>a tuple containing the genomic coordinates of the reference allele in VCF-style. The
  tuple consists of three elements: the chromosome, the start position, and the end position.</p>
</blockquote>
</div>


                            </div>
                            <div id="HGVSName.get_ref_alt" class="classattr">
                                        <input id="HGVSName.get_ref_alt-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr function">
            
        <span class="def">def</span>
        <span class="name">get_ref_alt</span><span class="signature pdoc-code multiline">(<span class="param">	<span class="bp">self</span>,</span><span class="param">	<span class="n">is_forward_strand</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">True</span>,</span><span class="param">	<span class="n">raw_dup_alleles</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span></span><span class="return-annotation">) -> <span class="nb">tuple</span>:</span></span>

                <label class="view-source-button" for="HGVSName.get_ref_alt-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#HGVSName.get_ref_alt"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="HGVSName.get_ref_alt-1357"><a href="#HGVSName.get_ref_alt-1357"><span class="linenos">1357</span></a>    <span class="k">def</span> <span class="nf">get_ref_alt</span><span class="p">(</span>
</span><span id="HGVSName.get_ref_alt-1358"><a href="#HGVSName.get_ref_alt-1358"><span class="linenos">1358</span></a>        <span class="bp">self</span><span class="p">,</span> <span class="n">is_forward_strand</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span> <span class="n">raw_dup_alleles</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span>
</span><span id="HGVSName.get_ref_alt-1359"><a href="#HGVSName.get_ref_alt-1359"><span class="linenos">1359</span></a>    <span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">tuple</span><span class="p">:</span>
</span><span id="HGVSName.get_ref_alt-1360"><a href="#HGVSName.get_ref_alt-1360"><span class="linenos">1360</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="HGVSName.get_ref_alt-1361"><a href="#HGVSName.get_ref_alt-1361"><span class="linenos">1361</span></a><span class="sd">        The function `get_ref_alt` returns the reference and alternate alleles, with an option to modify</span>
</span><span id="HGVSName.get_ref_alt-1362"><a href="#HGVSName.get_ref_alt-1362"><span class="linenos">1362</span></a><span class="sd">        duplications to look like inserts.</span>
</span><span id="HGVSName.get_ref_alt-1363"><a href="#HGVSName.get_ref_alt-1363"><span class="linenos">1363</span></a>
</span><span id="HGVSName.get_ref_alt-1364"><a href="#HGVSName.get_ref_alt-1364"><span class="linenos">1364</span></a><span class="sd">        :param is_forward_strand: The parameter `is_forward_strand` is a boolean flag that indicates</span>
</span><span id="HGVSName.get_ref_alt-1365"><a href="#HGVSName.get_ref_alt-1365"><span class="linenos">1365</span></a><span class="sd">        whether the alleles should be returned for the forward strand or the reverse complement strand.</span>
</span><span id="HGVSName.get_ref_alt-1366"><a href="#HGVSName.get_ref_alt-1366"><span class="linenos">1366</span></a><span class="sd">        If `is_forward_strand` is `True`, the alleles will be returned as is. If `is_forward_strand` is</span>
</span><span id="HGVSName.get_ref_alt-1367"><a href="#HGVSName.get_ref_alt-1367"><span class="linenos">1367</span></a><span class="sd">        `False`,, defaults to True</span>
</span><span id="HGVSName.get_ref_alt-1368"><a href="#HGVSName.get_ref_alt-1368"><span class="linenos">1368</span></a><span class="sd">        :type is_forward_strand: bool (optional)</span>
</span><span id="HGVSName.get_ref_alt-1369"><a href="#HGVSName.get_ref_alt-1369"><span class="linenos">1369</span></a><span class="sd">        :param raw_dup_alleles: The `raw_dup_alleles` parameter is a boolean flag that determines</span>
</span><span id="HGVSName.get_ref_alt-1370"><a href="#HGVSName.get_ref_alt-1370"><span class="linenos">1370</span></a><span class="sd">        whether the raw values of duplicated alleles should be returned. By default, it is set to</span>
</span><span id="HGVSName.get_ref_alt-1371"><a href="#HGVSName.get_ref_alt-1371"><span class="linenos">1371</span></a><span class="sd">        `False`, which means that if the mutation type is a duplication (`dup`), the reference allele</span>
</span><span id="HGVSName.get_ref_alt-1372"><a href="#HGVSName.get_ref_alt-1372"><span class="linenos">1372</span></a><span class="sd">        will be represented as an empty string, defaults to False</span>
</span><span id="HGVSName.get_ref_alt-1373"><a href="#HGVSName.get_ref_alt-1373"><span class="linenos">1373</span></a><span class="sd">        :type raw_dup_alleles: bool (optional)</span>
</span><span id="HGVSName.get_ref_alt-1374"><a href="#HGVSName.get_ref_alt-1374"><span class="linenos">1374</span></a><span class="sd">        :return: The function `get_ref_alt` returns a tuple containing the reference and alternate</span>
</span><span id="HGVSName.get_ref_alt-1375"><a href="#HGVSName.get_ref_alt-1375"><span class="linenos">1375</span></a><span class="sd">        alleles.</span>
</span><span id="HGVSName.get_ref_alt-1376"><a href="#HGVSName.get_ref_alt-1376"><span class="linenos">1376</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="HGVSName.get_ref_alt-1377"><a href="#HGVSName.get_ref_alt-1377"><span class="linenos">1377</span></a>
</span><span id="HGVSName.get_ref_alt-1378"><a href="#HGVSName.get_ref_alt-1378"><span class="linenos">1378</span></a>        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">kind</span> <span class="o">==</span> <span class="s2">&quot;p&quot;</span><span class="p">:</span>
</span><span id="HGVSName.get_ref_alt-1379"><a href="#HGVSName.get_ref_alt-1379"><span class="linenos">1379</span></a>            <span class="k">raise</span> <span class="ne">NotImplementedError</span><span class="p">(</span>
</span><span id="HGVSName.get_ref_alt-1380"><a href="#HGVSName.get_ref_alt-1380"><span class="linenos">1380</span></a>                <span class="s2">&quot;get_ref_alt is not implemented for protein HGVS names&quot;</span>
</span><span id="HGVSName.get_ref_alt-1381"><a href="#HGVSName.get_ref_alt-1381"><span class="linenos">1381</span></a>            <span class="p">)</span>
</span><span id="HGVSName.get_ref_alt-1382"><a href="#HGVSName.get_ref_alt-1382"><span class="linenos">1382</span></a>        <span class="n">alleles</span> <span class="o">=</span> <span class="p">[</span><span class="bp">self</span><span class="o">.</span><span class="n">ref_allele</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">alt_allele</span><span class="p">]</span>
</span><span id="HGVSName.get_ref_alt-1383"><a href="#HGVSName.get_ref_alt-1383"><span class="linenos">1383</span></a>
</span><span id="HGVSName.get_ref_alt-1384"><a href="#HGVSName.get_ref_alt-1384"><span class="linenos">1384</span></a>        <span class="c1"># Represent duplications are inserts.</span>
</span><span id="HGVSName.get_ref_alt-1385"><a href="#HGVSName.get_ref_alt-1385"><span class="linenos">1385</span></a>        <span class="k">if</span> <span class="ow">not</span> <span class="n">raw_dup_alleles</span> <span class="ow">and</span> <span class="bp">self</span><span class="o">.</span><span class="n">mutation_type</span> <span class="o">==</span> <span class="s2">&quot;dup&quot;</span><span class="p">:</span>
</span><span id="HGVSName.get_ref_alt-1386"><a href="#HGVSName.get_ref_alt-1386"><span class="linenos">1386</span></a>            <span class="n">alleles</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span>
</span><span id="HGVSName.get_ref_alt-1387"><a href="#HGVSName.get_ref_alt-1387"><span class="linenos">1387</span></a>            <span class="n">alleles</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="o">=</span> <span class="n">alleles</span><span class="p">[</span><span class="mi">1</span><span class="p">][:</span> <span class="nb">len</span><span class="p">(</span><span class="n">alleles</span><span class="p">[</span><span class="mi">1</span><span class="p">])</span> <span class="o">//</span> <span class="mi">2</span><span class="p">]</span>
</span><span id="HGVSName.get_ref_alt-1388"><a href="#HGVSName.get_ref_alt-1388"><span class="linenos">1388</span></a>
</span><span id="HGVSName.get_ref_alt-1389"><a href="#HGVSName.get_ref_alt-1389"><span class="linenos">1389</span></a>        <span class="k">if</span> <span class="n">is_forward_strand</span><span class="p">:</span>
</span><span id="HGVSName.get_ref_alt-1390"><a href="#HGVSName.get_ref_alt-1390"><span class="linenos">1390</span></a>            <span class="k">return</span> <span class="n">alleles</span>
</span><span id="HGVSName.get_ref_alt-1391"><a href="#HGVSName.get_ref_alt-1391"><span class="linenos">1391</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="HGVSName.get_ref_alt-1392"><a href="#HGVSName.get_ref_alt-1392"><span class="linenos">1392</span></a>            <span class="k">return</span> <span class="nb">tuple</span><span class="p">(</span><span class="nb">map</span><span class="p">(</span><span class="n">revcomp</span><span class="p">,</span> <span class="n">alleles</span><span class="p">))</span>
</span></pre></div>


            <div class="docstring"><p>The function <code><a href="#HGVSName.get_ref_alt">get_ref_alt</a></code> returns the reference and alternate alleles, with an option to modify
duplications to look like inserts.</p>

<h6 id="parameters">Parameters</h6>

<ul>
<li><strong>is_forward_strand</strong>:  The parameter <code>is_forward_strand</code> is a boolean flag that indicates
whether the alleles should be returned for the forward strand or the reverse complement strand.
If <code>is_forward_strand</code> is <code>True</code>, the alleles will be returned as is. If <code>is_forward_strand</code> is
<code>False</code>,, defaults to True</li>
<li><strong>raw_dup_alleles</strong>:  The <code>raw_dup_alleles</code> parameter is a boolean flag that determines
whether the raw values of duplicated alleles should be returned. By default, it is set to
<code>False</code>, which means that if the mutation type is a duplication (<code>dup</code>), the reference allele
will be represented as an empty string, defaults to False</li>
</ul>

<h6 id="returns">Returns</h6>

<blockquote>
  <p>The function <code><a href="#HGVSName.get_ref_alt">get_ref_alt</a></code> returns a tuple containing the reference and alternate
  alleles.</p>
</blockquote>
</div>


                            </div>
                </section>
    </main>
<script>
    function escapeHTML(html) {
        return document.createElement('div').appendChild(document.createTextNode(html)).parentNode.innerHTML;
    }

    const originalContent = document.querySelector("main.pdoc");
    let currentContent = originalContent;

    function setContent(innerHTML) {
        let elem;
        if (innerHTML) {
            elem = document.createElement("main");
            elem.classList.add("pdoc");
            elem.innerHTML = innerHTML;
        } else {
            elem = originalContent;
        }
        if (currentContent !== elem) {
            currentContent.replaceWith(elem);
            currentContent = elem;
        }
    }

    function getSearchTerm() {
        return (new URL(window.location)).searchParams.get("search");
    }

    const searchBox = document.querySelector(".pdoc input[type=search]");
    searchBox.addEventListener("input", function () {
        let url = new URL(window.location);
        if (searchBox.value.trim()) {
            url.hash = "";
            url.searchParams.set("search", searchBox.value);
        } else {
            url.searchParams.delete("search");
        }
        history.replaceState("", "", url.toString());
        onInput();
    });
    window.addEventListener("popstate", onInput);


    let search, searchErr;

    async function initialize() {
        try {
            search = await new Promise((resolve, reject) => {
                const script = document.createElement("script");
                script.type = "text/javascript";
                script.async = true;
                script.onload = () => resolve(window.pdocSearch);
                script.onerror = (e) => reject(e);
                script.src = "../../search.js";
                document.getElementsByTagName("head")[0].appendChild(script);
            });
        } catch (e) {
            console.error("Cannot fetch pdoc search index");
            searchErr = "Cannot fetch search index.";
        }
        onInput();

        document.querySelector("nav.pdoc").addEventListener("click", e => {
            if (e.target.hash) {
                searchBox.value = "";
                searchBox.dispatchEvent(new Event("input"));
            }
        });
    }

    function onInput() {
        setContent((() => {
            const term = getSearchTerm();
            if (!term) {
                return null
            }
            if (searchErr) {
                return `<h3>Error: ${searchErr}</h3>`
            }
            if (!search) {
                return "<h3>Searching...</h3>"
            }

            window.scrollTo({top: 0, left: 0, behavior: 'auto'});

            const results = search(term);

            let html;
            if (results.length === 0) {
                html = `No search results for '${escapeHTML(term)}'.`
            } else {
                html = `<h4>${results.length} search result${results.length > 1 ? "s" : ""} for '${escapeHTML(term)}'.</h4>`;
            }
            for (let result of results.slice(0, 10)) {
                let doc = result.doc;
                let url = `../../${doc.modulename.replaceAll(".", "/")}.html`;
                if (doc.qualname) {
                    url += `#${doc.qualname}`;
                }

                let heading;
                switch (result.doc.kind) {
                    case "function":
                        if (doc.fullname.endsWith(".__init__")) {
                            heading = `<span class="name">${doc.fullname.replace(/\.__init__$/, "")}</span>${doc.signature}`;
                        } else {
                            heading = `<span class="def">${doc.funcdef}</span> <span class="name">${doc.fullname}</span>${doc.signature}`;
                        }
                        break;
                    case "class":
                        heading = `<span class="def">class</span> <span class="name">${doc.fullname}</span>`;
                        if (doc.bases)
                            heading += `<wbr>(<span class="base">${doc.bases}</span>)`;
                        heading += `:`;
                        break;
                    case "variable":
                        heading = `<span class="name">${doc.fullname}</span>`;
                        if (doc.annotation)
                            heading += `<span class="annotation">${doc.annotation}</span>`;
                        if (doc.default_value)
                            heading += `<span class="default_value"> = ${doc.default_value}</span>`;
                        break;
                    default:
                        heading = `<span class="name">${doc.fullname}</span>`;
                        break;
                }
                html += `
                        <section class="search-result">
                        <a href="${url}" class="attr ${doc.kind}">${heading}</a>
                        <div class="docstring">${doc.doc}</div>
                        </section>
                    `;

            }
            return html;
        })());
    }

    if (getSearchTerm()) {
        initialize();
        searchBox.value = getSearchTerm();
        onInput();
    } else {
        searchBox.addEventListener("focus", initialize, {once: true});
    }

    searchBox.addEventListener("keydown", e => {
        if (["ArrowDown", "ArrowUp", "Enter"].includes(e.key)) {
            let focused = currentContent.querySelector(".search-result.focused");
            if (!focused) {
                currentContent.querySelector(".search-result").classList.add("focused");
            } else if (
                e.key === "ArrowDown"
                && focused.nextElementSibling
                && focused.nextElementSibling.classList.contains("search-result")
            ) {
                focused.classList.remove("focused");
                focused.nextElementSibling.classList.add("focused");
                focused.nextElementSibling.scrollIntoView({
                    behavior: "smooth",
                    block: "nearest",
                    inline: "nearest"
                });
            } else if (
                e.key === "ArrowUp"
                && focused.previousElementSibling
                && focused.previousElementSibling.classList.contains("search-result")
            ) {
                focused.classList.remove("focused");
                focused.previousElementSibling.classList.add("focused");
                focused.previousElementSibling.scrollIntoView({
                    behavior: "smooth",
                    block: "nearest",
                    inline: "nearest"
                });
            } else if (
                e.key === "Enter"
            ) {
                focused.querySelector("a").click();
            }
        }
    });
</script></body>
</html>