<!doctype html>
<html lang="en">
<head>
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, minimum-scale=1" />
<meta name="generator" content="pdoc 0.10.0" />
<title>howard.objects.transcript API documentation</title>
<meta name="description" content="Models for representing genomic elements." />
<link rel="preload stylesheet" as="style" href="https://cdnjs.cloudflare.com/ajax/libs/10up-sanitize.css/11.0.1/sanitize.min.css" integrity="sha256-PK9q560IAAa6WVRRh76LtCaI8pjTJ2z11v0miyNNjrs=" crossorigin>
<link rel="preload stylesheet" as="style" href="https://cdnjs.cloudflare.com/ajax/libs/10up-sanitize.css/11.0.1/typography.min.css" integrity="sha256-7l/o7C8jubJiy74VsKTidCy1yBkRtiUGbVkYBylBqUg=" crossorigin>
<link rel="stylesheet preload" as="style" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.1.1/styles/github.min.css" crossorigin>
<style>:root{--highlight-color:#fe9}.flex{display:flex !important}body{line-height:1.5em}#content{padding:20px}#sidebar{padding:30px;overflow:hidden}#sidebar > *:last-child{margin-bottom:2cm}.http-server-breadcrumbs{font-size:130%;margin:0 0 15px 0}#footer{font-size:.75em;padding:5px 30px;border-top:1px solid #ddd;text-align:right}#footer p{margin:0 0 0 1em;display:inline-block}#footer p:last-child{margin-right:30px}h1,h2,h3,h4,h5{font-weight:300}h1{font-size:2.5em;line-height:1.1em}h2{font-size:1.75em;margin:1em 0 .50em 0}h3{font-size:1.4em;margin:25px 0 10px 0}h4{margin:0;font-size:105%}h1:target,h2:target,h3:target,h4:target,h5:target,h6:target{background:var(--highlight-color);padding:.2em 0}a{color:#058;text-decoration:none;transition:color .3s ease-in-out}a:hover{color:#e82}.title code{font-weight:bold}h2[id^="header-"]{margin-top:2em}.ident{color:#900}pre code{background:#f8f8f8;font-size:.8em;line-height:1.4em}code{background:#f2f2f1;padding:1px 4px;overflow-wrap:break-word}h1 code{background:transparent}pre{background:#f8f8f8;border:0;border-top:1px solid #ccc;border-bottom:1px solid #ccc;margin:1em 0;padding:1ex}#http-server-module-list{display:flex;flex-flow:column}#http-server-module-list div{display:flex}#http-server-module-list dt{min-width:10%}#http-server-module-list p{margin-top:0}.toc ul,#index{list-style-type:none;margin:0;padding:0}#index code{background:transparent}#index h3{border-bottom:1px solid #ddd}#index ul{padding:0}#index h4{margin-top:.6em;font-weight:bold}@media (min-width:200ex){#index .two-column{column-count:2}}@media (min-width:300ex){#index .two-column{column-count:3}}dl{margin-bottom:2em}dl dl:last-child{margin-bottom:4em}dd{margin:0 0 1em 3em}#header-classes + dl > dd{margin-bottom:3em}dd dd{margin-left:2em}dd p{margin:10px 0}.name{background:#eee;font-weight:bold;font-size:.85em;padding:5px 10px;display:inline-block;min-width:40%}.name:hover{background:#e0e0e0}dt:target .name{background:var(--highlight-color)}.name > span:first-child{white-space:nowrap}.name.class > span:nth-child(2){margin-left:.4em}.inherited{color:#999;border-left:5px solid #eee;padding-left:1em}.inheritance em{font-style:normal;font-weight:bold}.desc h2{font-weight:400;font-size:1.25em}.desc h3{font-size:1em}.desc dt code{background:inherit}.source summary,.git-link-div{color:#666;text-align:right;font-weight:400;font-size:.8em;text-transform:uppercase}.source summary > *{white-space:nowrap;cursor:pointer}.git-link{color:inherit;margin-left:1em}.source pre{max-height:500px;overflow:auto;margin:0}.source pre code{font-size:12px;overflow:visible}.hlist{list-style:none}.hlist li{display:inline}.hlist li:after{content:',\2002'}.hlist li:last-child:after{content:none}.hlist .hlist{display:inline;padding-left:1em}img{max-width:100%}td{padding:0 .5em}.admonition{padding:.1em .5em;margin-bottom:1em}.admonition-title{font-weight:bold}.admonition.note,.admonition.info,.admonition.important{background:#aef}.admonition.todo,.admonition.versionadded,.admonition.tip,.admonition.hint{background:#dfd}.admonition.warning,.admonition.versionchanged,.admonition.deprecated{background:#fd4}.admonition.error,.admonition.danger,.admonition.caution{background:lightpink}</style>
<style media="screen and (min-width: 700px)">@media screen and (min-width:700px){#sidebar{width:30%;height:100vh;overflow:auto;position:sticky;top:0}#content{width:70%;max-width:100ch;padding:3em 4em;border-left:1px solid #ddd}pre code{font-size:1em}.item .name{font-size:1em}main{display:flex;flex-direction:row-reverse;justify-content:flex-end}.toc ul ul,#index ul{padding-left:1.5em}.toc > ul > li{margin-top:.5em}}</style>
<style media="print">@media print{#sidebar h1{page-break-before:always}.source{display:none}}@media print{*{background:transparent !important;color:#000 !important;box-shadow:none !important;text-shadow:none !important}a[href]:after{content:" (" attr(href) ")";font-size:90%}a[href][title]:after{content:none}abbr[title]:after{content:" (" attr(title) ")"}.ir a:after,a[href^="javascript:"]:after,a[href^="#"]:after{content:""}pre,blockquote{border:1px solid #999;page-break-inside:avoid}thead{display:table-header-group}tr,img{page-break-inside:avoid}img{max-width:100% !important}@page{margin:0.5cm}p,h2,h3{orphans:3;widows:3}h1,h2,h3,h4,h5,h6{page-break-after:avoid}}</style>
<script defer src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.1.1/highlight.min.js" integrity="sha256-Uv3H6lx7dJmRfRvH8TH6kJD1TSK1aFcwgx+mdg3epi8=" crossorigin></script>
<script>window.addEventListener('DOMContentLoaded', () => hljs.initHighlighting())</script>
</head>
<body>
<main>
<article id="content">
<header>
<h1 class="title">Module <code>howard.objects.transcript</code></h1>
</header>
<section id="section-intro">
<p>Models for representing genomic elements.</p>
</section>
<section>
</section>
<section>
</section>
<section>
</section>
<section>
<h2 class="section-title" id="header-classes">Classes</h2>
<dl>
<dt id="howard.objects.transcript.BED6Interval"><code class="flex name class">
<span>class <span class="ident">BED6Interval</span></span>
<span>(</span><span>chrom, chrom_start, chrom_end, name, score, strand)</span>
</code></dt>
<dd>
<div class="desc"><p>BED6Interval_base(chrom, chrom_start, chrom_end, name, score, strand)</p></div>
<h3>Ancestors</h3>
<ul class="hlist">
<li><a title="howard.objects.transcript.BED6Interval_base" href="#howard.objects.transcript.BED6Interval_base">BED6Interval_base</a></li>
<li>builtins.tuple</li>
</ul>
<h3>Methods</h3>
<dl>
<dt id="howard.objects.transcript.BED6Interval.distance"><code class="name flex">
<span>def <span class="ident">distance</span></span>(<span>self, offset: int) ‑> int</span>
</code></dt>
<dd>
<div class="desc"><p>The <code>distance</code> function calculates the distance between an offset and an interval, returning
zero if the offset is inside the interval, a positive value if the interval comes after the
offset, and a negative value if the interval comes before the offset.
if offset is inside the exon, distance is zero.
otherwise, distance is the distance to the nearest edge.
distance is positive if the exon comes after the offset.
distance is negative if the exon comes before the offset.</p>
<p>:param offset: The offset parameter represents a position or point in the genome. It is an
integer value that indicates the position within the genome sequence
:type offset: int
:return: an integer value, which represents the distance to the interval.</p></div>
</dd>
</dl>
<h3>Inherited members</h3>
<ul class="hlist">
<li><code><b><a title="howard.objects.transcript.BED6Interval_base" href="#howard.objects.transcript.BED6Interval_base">BED6Interval_base</a></b></code>:
<ul class="hlist">
<li><code><a title="howard.objects.transcript.BED6Interval_base.chrom" href="#howard.objects.transcript.BED6Interval_base.chrom">chrom</a></code></li>
<li><code><a title="howard.objects.transcript.BED6Interval_base.chrom_end" href="#howard.objects.transcript.BED6Interval_base.chrom_end">chrom_end</a></code></li>
<li><code><a title="howard.objects.transcript.BED6Interval_base.chrom_start" href="#howard.objects.transcript.BED6Interval_base.chrom_start">chrom_start</a></code></li>
<li><code><a title="howard.objects.transcript.BED6Interval_base.name" href="#howard.objects.transcript.BED6Interval_base.name">name</a></code></li>
<li><code><a title="howard.objects.transcript.BED6Interval_base.score" href="#howard.objects.transcript.BED6Interval_base.score">score</a></code></li>
<li><code><a title="howard.objects.transcript.BED6Interval_base.strand" href="#howard.objects.transcript.BED6Interval_base.strand">strand</a></code></li>
</ul>
</li>
</ul>
</dd>
<dt id="howard.objects.transcript.BED6Interval_base"><code class="flex name class">
<span>class <span class="ident">BED6Interval_base</span></span>
<span>(</span><span>chrom, chrom_start, chrom_end, name, score, strand)</span>
</code></dt>
<dd>
<div class="desc"><p>BED6Interval_base(chrom, chrom_start, chrom_end, name, score, strand)</p></div>
<h3>Ancestors</h3>
<ul class="hlist">
<li>builtins.tuple</li>
</ul>
<h3>Subclasses</h3>
<ul class="hlist">
<li><a title="howard.objects.transcript.BED6Interval" href="#howard.objects.transcript.BED6Interval">BED6Interval</a></li>
</ul>
<h3>Instance variables</h3>
<dl>
<dt id="howard.objects.transcript.BED6Interval_base.chrom"><code class="name">var <span class="ident">chrom</span></code></dt>
<dd>
<div class="desc"><p>Alias for field number 0</p></div>
</dd>
<dt id="howard.objects.transcript.BED6Interval_base.chrom_end"><code class="name">var <span class="ident">chrom_end</span></code></dt>
<dd>
<div class="desc"><p>Alias for field number 2</p></div>
</dd>
<dt id="howard.objects.transcript.BED6Interval_base.chrom_start"><code class="name">var <span class="ident">chrom_start</span></code></dt>
<dd>
<div class="desc"><p>Alias for field number 1</p></div>
</dd>
<dt id="howard.objects.transcript.BED6Interval_base.name"><code class="name">var <span class="ident">name</span></code></dt>
<dd>
<div class="desc"><p>Alias for field number 3</p></div>
</dd>
<dt id="howard.objects.transcript.BED6Interval_base.score"><code class="name">var <span class="ident">score</span></code></dt>
<dd>
<div class="desc"><p>Alias for field number 4</p></div>
</dd>
<dt id="howard.objects.transcript.BED6Interval_base.strand"><code class="name">var <span class="ident">strand</span></code></dt>
<dd>
<div class="desc"><p>Alias for field number 5</p></div>
</dd>
</dl>
</dd>
<dt id="howard.objects.transcript.CDNA_Match"><code class="flex name class">
<span>class <span class="ident">CDNA_Match</span></span>
<span>(</span><span>transcript: <a title="howard.objects.transcript.Transcript" href="#howard.objects.transcript.Transcript">Transcript</a>, tx_position: int, cdna_start: int, cdna_end: int, gap: int, number: int)</span>
</code></dt>
<dd>
<div class="desc"><p>The function initializes a CDNA_Match object with specified attributes.</p>
<p>:param transcript: The <code>transcript</code> parameter is an instance of the <code><a title="howard.objects.transcript.Transcript" href="#howard.objects.transcript.Transcript">Transcript</a></code> class. It
represents the transcript that the CDNA match belongs to
:type transcript: Transcript
:param tx_position: The <code>tx_position</code> parameter represents the position of the transcript in the
genome. It is an integer value
:type tx_position: int
:param cdna_start: The <code>cdna_start</code> parameter represents the starting position of the cDNA
match. It is an integer value
:type cdna_start: int
:param cdna_end: The <code>cdna_end</code> parameter represents the end position of the cDNA match. It is
an integer value that indicates the position of the last nucleotide in the cDNA sequence that
matches the transcript
:type cdna_end: int
:param gap: The "gap" parameter represents the number of nucleotides that are missing or
inserted in the cDNA sequence compared to the reference transcript sequence. It indicates the
presence of gaps or insertions in the alignment between the cDNA and the reference transcript
:type gap: int
:param number: The <code>number</code> parameter represents the number of the CDNA match. It is used to
uniquely identify each CDNA match object
:type number: int</p></div>
<h3>Ancestors</h3>
<ul class="hlist">
<li><a title="howard.objects.transcript.Exon" href="#howard.objects.transcript.Exon">Exon</a></li>
</ul>
<h3>Instance variables</h3>
<dl>
<dt id="howard.objects.transcript.CDNA_Match.length"><code class="name">var <span class="ident">length</span> : int</code></dt>
<dd>
<div class="desc"><p>The function calculates the length of a sequence by subtracting the start position from the end
position and adding 1.
:return: The length of the sequence, calculated by subtracting the cdna_start from the cdna_end
and adding 1.</p></div>
</dd>
</dl>
<h3>Methods</h3>
<dl>
<dt id="howard.objects.transcript.CDNA_Match.get_offset"><code class="name flex">
<span>def <span class="ident">get_offset</span></span>(<span>self, position: int, validate: bool = True) ‑> int</span>
</code></dt>
<dd>
<div class="desc"><p>The <code>get_offset</code> function calculates the offset for a given position in a cDNA sequence based on
the GAP attribute.
cdna_match GAP attribute looks like: 'M185 I3 M250' which is code/length
@see <a href="https://github.com/The-Sequence-Ontology/Specifications/blob/master/gff3.md#the-gap-attribute">https://github.com/The-Sequence-Ontology/Specifications/blob/master/gff3.md#the-gap-attribute</a>
codes operation
M
match
I
insert a gap into the reference sequence
D
insert a gap into the target (delete from reference)
If you want the whole exon, then pass the end</p>
<p>:param position: The <code>position</code> parameter is an integer that represents the position in the
sequence. It is used to calculate the offset based on the GAP attribute of the cDNA match
:type position: int
:param validate: The <code>validate</code> parameter is a boolean flag that determines whether to perform
validation checks during the calculation of the offset. If <code>validate</code> is set to <code>True</code>, the
function will raise a <code>ValueError</code> if the given position falls within an insertion or deletion
gap. If <code>validate</code> is set, defaults to True
:type validate: bool (optional)
:return: an integer value representing the offset for a given position in the cDNA sequence.</p></div>
</dd>
</dl>
<h3>Inherited members</h3>
<ul class="hlist">
<li><code><b><a title="howard.objects.transcript.Exon" href="#howard.objects.transcript.Exon">Exon</a></b></code>:
<ul class="hlist">
<li><code><a title="howard.objects.transcript.Exon.get_as_interval" href="#howard.objects.transcript.Exon.get_as_interval">get_as_interval</a></code></li>
<li><code><a title="howard.objects.transcript.Exon.name" href="#howard.objects.transcript.Exon.name">name</a></code></li>
</ul>
</li>
</ul>
</dd>
<dt id="howard.objects.transcript.Exon"><code class="flex name class">
<span>class <span class="ident">Exon</span></span>
<span>(</span><span>transcript: <a title="howard.objects.transcript.Transcript" href="#howard.objects.transcript.Transcript">Transcript</a>, tx_position: int, number: int)</span>
</code></dt>
<dd>
<div class="desc"><p>The function initializes an object with a transcript, a position in the transcript, and a
number.</p>
<p>:param transcript: The <code>transcript</code> parameter is of type <code><a title="howard.objects.transcript.Transcript" href="#howard.objects.transcript.Transcript">Transcript</a></code>. It represents a
transcript object that contains information about a conversation or dialogue
:type transcript: Transcript
:param tx_position: The <code>tx_position</code> parameter represents the position of the transcript in a
list or array. It is an integer value that indicates the index of the transcript in the list or
array
:type tx_position: int
:param number: The "number" parameter is an integer that represents a specific number. It is
used as a parameter in the constructor of a class
:type number: int</p></div>
<h3>Subclasses</h3>
<ul class="hlist">
<li><a title="howard.objects.transcript.CDNA_Match" href="#howard.objects.transcript.CDNA_Match">CDNA_Match</a></li>
</ul>
<h3>Instance variables</h3>
<dl>
<dt id="howard.objects.transcript.Exon.name"><code class="name">var <span class="ident">name</span> : str</code></dt>
<dd>
<div class="desc"><p>The function returns a string that combines the name of the transcript and a number.
:return: a string that combines the name of the transcript with the number. The format of the
string is "{transcript name}.{number}".</p></div>
</dd>
<dt id="howard.objects.transcript.Exon.strand"><code class="name">var <span class="ident">strand</span></code></dt>
<dd>
<div class="desc"></div>
</dd>
</dl>
<h3>Methods</h3>
<dl>
<dt id="howard.objects.transcript.Exon.get_as_interval"><code class="name flex">
<span>def <span class="ident">get_as_interval</span></span>(<span>self, coding_only: bool = False) ‑> object</span>
</code></dt>
<dd>
<div class="desc"><p>The function <code>get_as_interval</code> returns the coding region for an exon as a BED6Interval object.
This function returns a BED6Interval objects containing
position
information for this exon. This may be used as input for
pybedtools.create_interval_from_list() after casting chrom_start
and chrom_end as strings.</p>
<p>:param coding_only: The <code>coding_only</code> parameter is a boolean flag that determines whether to
include only exons in the coding region. If <code>coding_only</code> is set to <code>True</code>, the function will
check if the exon is completely outside the coding region defined by the transcript's CDS
(coding sequence) position, defaults to False
:type coding_only: bool (optional)
:return: a BED6Interval object.</p></div>
</dd>
</dl>
</dd>
<dt id="howard.objects.transcript.Gene"><code class="flex name class">
<span>class <span class="ident">Gene</span></span>
<span>(</span><span>name)</span>
</code></dt>
<dd>
<div class="desc"></div>
</dd>
<dt id="howard.objects.transcript.Transcript"><code class="flex name class">
<span>class <span class="ident">Transcript</span></span>
<span>(</span><span>name: str, version: str, gene: str, tx_position: int, cds_position: int, is_default: bool = False, cdna_match: list = None, start_codon_transcript_pos: int = None, stop_codon_transcript_pos: int = None)</span>
</code></dt>
<dd>
<div class="desc"><p>The function initializes an object with various attributes related to a gene and its transcript.</p>
<p>:param name: A string representing the name of the coding sequence
:type name: str
:param version: The <code>version</code> parameter is a string that represents the version of the object.
It is used to track changes or updates to the object over time
:type version: str
:param gene: The <code>gene</code> parameter is a string that represents the gene associated with the
coding sequence
:type gene: str
:param tx_position: The <code>tx_position</code> parameter represents the position of the transcript. It is
an integer value that indicates the position of the transcript in the genome
:type tx_position: int
:param cds_position: The <code>cds_position</code> parameter represents the position of the coding sequence
(CDS) within the transcript. It is an integer value that indicates the starting position of the
CDS within the transcript sequence
:type cds_position: int
:param is_default: The <code>is_default</code> parameter is a boolean flag that indicates whether the
instance of the class is the default version of the gene. It is set to <code>False</code> by default, but
can be set to <code>True</code> if the instance is the default version, defaults to False
:type is_default: bool (optional)
:param cdna_match: The <code>cdna_match</code> parameter is a list that contains the positions of the
matching cDNA sequences. It is an optional parameter and if not provided, it defaults to an
empty list
:type cdna_match: list
:param start_codon_transcript_pos: The parameter "start_codon_transcript_pos" is an optional
parameter that represents the transcript position of the start codon. It is used to store the
pre-calculated transcript position of the start codon for a specific gene
:type start_codon_transcript_pos: int
:param stop_codon_transcript_pos: The parameter <code>stop_codon_transcript_pos</code> is an optional
integer that represents the transcript position of the stop codon. It is used to store the
pre-calculated transcript coordinate of the stop codon. If not provided, it will be set to
<code>None</code>
:type stop_codon_transcript_pos: int</p></div>
<h3>Instance variables</h3>
<dl>
<dt id="howard.objects.transcript.Transcript.full_name"><code class="name">var <span class="ident">full_name</span> : str</code></dt>
<dd>
<div class="desc"><p>The function <code>full_name</code> returns the full name of an object, including its version if it exists.
:return: a string. If the <code>version</code> attribute of the object is not <code>None</code>, it returns a string
in the format <code>name.version</code>. Otherwise, it returns just the <code>name</code> attribute.</p></div>
</dd>
<dt id="howard.objects.transcript.Transcript.is_coding"><code class="name">var <span class="ident">is_coding</span> : bool</code></dt>
<dd>
<div class="desc"><p>The function checks if a coding transcript has a non-zero length coding sequence.
:return: a boolean value indicating whether the coding transcript has a coding sequence (CDS)
with a non-zero length.</p></div>
</dd>
<dt id="howard.objects.transcript.Transcript.strand"><code class="name">var <span class="ident">strand</span> : str</code></dt>
<dd>
<div class="desc"><p>The function returns a string '+' if the tx_position is on the forward strand, and '-' if it is
on the reverse strand.
:return: a string that represents the strand of the given <code>self.tx_position</code>. If
<code>self.tx_position.is_forward_strand</code> is <code>True</code>, then the string returned is '+'. Otherwise, the
string returned is '-'.</p></div>
</dd>
</dl>
<h3>Methods</h3>
<dl>
<dt id="howard.objects.transcript.Transcript.cdna_to_genomic_coord"><code class="name flex">
<span>def <span class="ident">cdna_to_genomic_coord</span></span>(<span>self, coord: object) ‑> int</span>
</code></dt>
<dd>
<div class="desc"><p>The function <code>cdna_to_genomic_coord</code> converts a HGVS cDNA coordinate to a genomic coordinate.</p>
<p>:param coord: The parameter <code>coord</code> is an object that represents a cDNA coordinate. It is used
to specify a position along a cDNA sequence
:type coord: object
:return: an integer value, which represents the genomic coordinate corresponding to the given
cDNA coordinate.</p></div>
</dd>
<dt id="howard.objects.transcript.Transcript.find_exon_number"><code class="name flex">
<span>def <span class="ident">find_exon_number</span></span>(<span>self, offset: int) ‑> int</span>
</code></dt>
<dd>
<div class="desc"><p>The function <code>find_exon_number</code> returns the exon number for a given position.</p>
<p>:param offset: The offset parameter represents a position in the genome. It is an integer value
that indicates the position of interest within the genome
:type offset: int
:return: an integer value, which represents the exon number for a given position.</p></div>
</dd>
<dt id="howard.objects.transcript.Transcript.genomic_to_cdna_coord"><code class="name flex">
<span>def <span class="ident">genomic_to_cdna_coord</span></span>(<span>self, genomic_coord: int) ‑> object</span>
</code></dt>
<dd>
<div class="desc"><p>The function <code>genomic_to_cdna_coord</code> converts a genomic coordinate to a cDNA coordinate and
offset, taking into account exons, strand, and coding transcript information.</p>
<p>:param genomic_coord: The <code>genomic_coord</code> parameter is an integer representing a genomic
coordinate
:type genomic_coord: int
:return: an object of type <code>CDNACoord</code>.</p></div>
</dd>
<dt id="howard.objects.transcript.Transcript.get_cds_start_stop"><code class="name flex">
<span>def <span class="ident">get_cds_start_stop</span></span>(<span>self) ‑> tuple</span>
</code></dt>
<dd>
<div class="desc"><p>The function "get_cds_start_stop" returns the start and stop positions of a coding sequence,
taking into account the direction of the strand.
:return: a tuple containing the start and stop positions of the coding sequence (CDS).</p></div>
</dd>
<dt id="howard.objects.transcript.Transcript.ordered_cdna_match"><code class="name flex">
<span>def <span class="ident">ordered_cdna_match</span></span>(<span>self) ‑> str</span>
</code></dt>
<dd>
<div class="desc"><p>The function "ordered_cdna_match" sorts a list of cdna_match objects based on their
tx_position.chrom_start attribute and returns the sorted list.
:return: a sorted list of cdna_match objects.</p></div>
</dd>
<dt id="howard.objects.transcript.Transcript.start_codon"><code class="name flex">
<span>def <span class="ident">start_codon</span></span>(<span>self) ‑> int</span>
</code></dt>
<dd>
<div class="desc"><p>The function returns the transcript position of the start codon.
:return: the transcript position of the start codon.</p></div>
</dd>
<dt id="howard.objects.transcript.Transcript.stop_codon"><code class="name flex">
<span>def <span class="ident">stop_codon</span></span>(<span>self) ‑> int</span>
</code></dt>
<dd>
<div class="desc"><p>The function returns the transcript position of the stop codon.
:return: The method <code>stop_codon</code> returns an integer, which represents the transcript position of
the stop codon.</p></div>
</dd>
</dl>
</dd>
</dl>
</section>
</article>
<nav id="sidebar">
<h1>Index</h1>
<div class="toc">
<ul></ul>
</div>
<ul id="index">
<li><h3><a href="#header-classes">Classes</a></h3>
<ul>
<li>
<h4><code><a title="howard.objects.transcript.BED6Interval" href="#howard.objects.transcript.BED6Interval">BED6Interval</a></code></h4>
<ul class="">
<li><code><a title="howard.objects.transcript.BED6Interval.distance" href="#howard.objects.transcript.BED6Interval.distance">distance</a></code></li>
</ul>
</li>
<li>
<h4><code><a title="howard.objects.transcript.BED6Interval_base" href="#howard.objects.transcript.BED6Interval_base">BED6Interval_base</a></code></h4>
<ul class="two-column">
<li><code><a title="howard.objects.transcript.BED6Interval_base.chrom" href="#howard.objects.transcript.BED6Interval_base.chrom">chrom</a></code></li>
<li><code><a title="howard.objects.transcript.BED6Interval_base.chrom_end" href="#howard.objects.transcript.BED6Interval_base.chrom_end">chrom_end</a></code></li>
<li><code><a title="howard.objects.transcript.BED6Interval_base.chrom_start" href="#howard.objects.transcript.BED6Interval_base.chrom_start">chrom_start</a></code></li>
<li><code><a title="howard.objects.transcript.BED6Interval_base.name" href="#howard.objects.transcript.BED6Interval_base.name">name</a></code></li>
<li><code><a title="howard.objects.transcript.BED6Interval_base.score" href="#howard.objects.transcript.BED6Interval_base.score">score</a></code></li>
<li><code><a title="howard.objects.transcript.BED6Interval_base.strand" href="#howard.objects.transcript.BED6Interval_base.strand">strand</a></code></li>
</ul>
</li>
<li>
<h4><code><a title="howard.objects.transcript.CDNA_Match" href="#howard.objects.transcript.CDNA_Match">CDNA_Match</a></code></h4>
<ul class="">
<li><code><a title="howard.objects.transcript.CDNA_Match.get_offset" href="#howard.objects.transcript.CDNA_Match.get_offset">get_offset</a></code></li>
<li><code><a title="howard.objects.transcript.CDNA_Match.length" href="#howard.objects.transcript.CDNA_Match.length">length</a></code></li>
</ul>
</li>
<li>
<h4><code><a title="howard.objects.transcript.Exon" href="#howard.objects.transcript.Exon">Exon</a></code></h4>
<ul class="">
<li><code><a title="howard.objects.transcript.Exon.get_as_interval" href="#howard.objects.transcript.Exon.get_as_interval">get_as_interval</a></code></li>
<li><code><a title="howard.objects.transcript.Exon.name" href="#howard.objects.transcript.Exon.name">name</a></code></li>
<li><code><a title="howard.objects.transcript.Exon.strand" href="#howard.objects.transcript.Exon.strand">strand</a></code></li>
</ul>
</li>
<li>
<h4><code><a title="howard.objects.transcript.Gene" href="#howard.objects.transcript.Gene">Gene</a></code></h4>
</li>
<li>
<h4><code><a title="howard.objects.transcript.Transcript" href="#howard.objects.transcript.Transcript">Transcript</a></code></h4>
<ul class="">
<li><code><a title="howard.objects.transcript.Transcript.cdna_to_genomic_coord" href="#howard.objects.transcript.Transcript.cdna_to_genomic_coord">cdna_to_genomic_coord</a></code></li>
<li><code><a title="howard.objects.transcript.Transcript.find_exon_number" href="#howard.objects.transcript.Transcript.find_exon_number">find_exon_number</a></code></li>
<li><code><a title="howard.objects.transcript.Transcript.full_name" href="#howard.objects.transcript.Transcript.full_name">full_name</a></code></li>
<li><code><a title="howard.objects.transcript.Transcript.genomic_to_cdna_coord" href="#howard.objects.transcript.Transcript.genomic_to_cdna_coord">genomic_to_cdna_coord</a></code></li>
<li><code><a title="howard.objects.transcript.Transcript.get_cds_start_stop" href="#howard.objects.transcript.Transcript.get_cds_start_stop">get_cds_start_stop</a></code></li>
<li><code><a title="howard.objects.transcript.Transcript.is_coding" href="#howard.objects.transcript.Transcript.is_coding">is_coding</a></code></li>
<li><code><a title="howard.objects.transcript.Transcript.ordered_cdna_match" href="#howard.objects.transcript.Transcript.ordered_cdna_match">ordered_cdna_match</a></code></li>
<li><code><a title="howard.objects.transcript.Transcript.start_codon" href="#howard.objects.transcript.Transcript.start_codon">start_codon</a></code></li>
<li><code><a title="howard.objects.transcript.Transcript.stop_codon" href="#howard.objects.transcript.Transcript.stop_codon">stop_codon</a></code></li>
<li><code><a title="howard.objects.transcript.Transcript.strand" href="#howard.objects.transcript.Transcript.strand">strand</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</nav>
</main>
<footer id="footer">
<p>Generated by <a href="https://pdoc3.github.io/pdoc" title="pdoc: Python API documentation generator"><cite>pdoc</cite> 0.10.0</a>.</p>
</footer>
</body>
</html>