<!doctype html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="generator" content="pdoc 14.5.1"/>
    <title>howard.objects.transcript API documentation</title>

    <style>/*! * Bootstrap Reboot v5.0.0 (https://getbootstrap.com/) * Copyright 2011-2021 The Bootstrap Authors * Copyright 2011-2021 Twitter, Inc. * Licensed under MIT (https://github.com/twbs/bootstrap/blob/main/LICENSE) * Forked from Normalize.css, licensed MIT (https://github.com/necolas/normalize.css/blob/master/LICENSE.md) */*,::after,::before{box-sizing:border-box}@media (prefers-reduced-motion:no-preference){:root{scroll-behavior:smooth}}body{margin:0;font-family:system-ui,-apple-system,"Segoe UI",Roboto,"Helvetica Neue",Arial,"Noto Sans","Liberation Sans",sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol","Noto Color Emoji";font-size:1rem;font-weight:400;line-height:1.5;color:#212529;background-color:#fff;-webkit-text-size-adjust:100%;-webkit-tap-highlight-color:transparent}hr{margin:1rem 0;color:inherit;background-color:currentColor;border:0;opacity:.25}hr:not([size]){height:1px}h1,h2,h3,h4,h5,h6{margin-top:0;margin-bottom:.5rem;font-weight:500;line-height:1.2}h1{font-size:calc(1.375rem + 1.5vw)}@media (min-width:1200px){h1{font-size:2.5rem}}h2{font-size:calc(1.325rem + .9vw)}@media (min-width:1200px){h2{font-size:2rem}}h3{font-size:calc(1.3rem + .6vw)}@media (min-width:1200px){h3{font-size:1.75rem}}h4{font-size:calc(1.275rem + .3vw)}@media (min-width:1200px){h4{font-size:1.5rem}}h5{font-size:1.25rem}h6{font-size:1rem}p{margin-top:0;margin-bottom:1rem}abbr[data-bs-original-title],abbr[title]{-webkit-text-decoration:underline dotted;text-decoration:underline dotted;cursor:help;-webkit-text-decoration-skip-ink:none;text-decoration-skip-ink:none}address{margin-bottom:1rem;font-style:normal;line-height:inherit}ol,ul{padding-left:2rem}dl,ol,ul{margin-top:0;margin-bottom:1rem}ol ol,ol ul,ul ol,ul ul{margin-bottom:0}dt{font-weight:700}dd{margin-bottom:.5rem;margin-left:0}blockquote{margin:0 0 1rem}b,strong{font-weight:bolder}small{font-size:.875em}mark{padding:.2em;background-color:#fcf8e3}sub,sup{position:relative;font-size:.75em;line-height:0;vertical-align:baseline}sub{bottom:-.25em}sup{top:-.5em}a{color:#0d6efd;text-decoration:underline}a:hover{color:#0a58ca}a:not([href]):not([class]),a:not([href]):not([class]):hover{color:inherit;text-decoration:none}code,kbd,pre,samp{font-family:SFMono-Regular,Menlo,Monaco,Consolas,"Liberation Mono","Courier New",monospace;font-size:1em;direction:ltr;unicode-bidi:bidi-override}pre{display:block;margin-top:0;margin-bottom:1rem;overflow:auto;font-size:.875em}pre code{font-size:inherit;color:inherit;word-break:normal}code{font-size:.875em;color:#d63384;word-wrap:break-word}a>code{color:inherit}kbd{padding:.2rem .4rem;font-size:.875em;color:#fff;background-color:#212529;border-radius:.2rem}kbd kbd{padding:0;font-size:1em;font-weight:700}figure{margin:0 0 1rem}img,svg{vertical-align:middle}table{caption-side:bottom;border-collapse:collapse}caption{padding-top:.5rem;padding-bottom:.5rem;color:#6c757d;text-align:left}th{text-align:inherit;text-align:-webkit-match-parent}tbody,td,tfoot,th,thead,tr{border-color:inherit;border-style:solid;border-width:0}label{display:inline-block}button{border-radius:0}button:focus:not(:focus-visible){outline:0}button,input,optgroup,select,textarea{margin:0;font-family:inherit;font-size:inherit;line-height:inherit}button,select{text-transform:none}[role=button]{cursor:pointer}select{word-wrap:normal}select:disabled{opacity:1}[list]::-webkit-calendar-picker-indicator{display:none}[type=button],[type=reset],[type=submit],button{-webkit-appearance:button}[type=button]:not(:disabled),[type=reset]:not(:disabled),[type=submit]:not(:disabled),button:not(:disabled){cursor:pointer}::-moz-focus-inner{padding:0;border-style:none}textarea{resize:vertical}fieldset{min-width:0;padding:0;margin:0;border:0}legend{float:left;width:100%;padding:0;margin-bottom:.5rem;font-size:calc(1.275rem + .3vw);line-height:inherit}@media (min-width:1200px){legend{font-size:1.5rem}}legend+*{clear:left}::-webkit-datetime-edit-day-field,::-webkit-datetime-edit-fields-wrapper,::-webkit-datetime-edit-hour-field,::-webkit-datetime-edit-minute,::-webkit-datetime-edit-month-field,::-webkit-datetime-edit-text,::-webkit-datetime-edit-year-field{padding:0}::-webkit-inner-spin-button{height:auto}[type=search]{outline-offset:-2px;-webkit-appearance:textfield}::-webkit-search-decoration{-webkit-appearance:none}::-webkit-color-swatch-wrapper{padding:0}::file-selector-button{font:inherit}::-webkit-file-upload-button{font:inherit;-webkit-appearance:button}output{display:inline-block}iframe{border:0}summary{display:list-item;cursor:pointer}progress{vertical-align:baseline}[hidden]{display:none!important}</style>
    <style>/*! syntax-highlighting.css */pre{line-height:125%;}span.linenos{color:inherit; background-color:transparent; padding-left:5px; padding-right:20px;}.pdoc-code .hll{background-color:#ffffcc}.pdoc-code{background:#f8f8f8;}.pdoc-code .c{color:#3D7B7B; font-style:italic}.pdoc-code .err{border:1px solid #FF0000}.pdoc-code .k{color:#008000; font-weight:bold}.pdoc-code .o{color:#666666}.pdoc-code .ch{color:#3D7B7B; font-style:italic}.pdoc-code .cm{color:#3D7B7B; font-style:italic}.pdoc-code .cp{color:#9C6500}.pdoc-code .cpf{color:#3D7B7B; font-style:italic}.pdoc-code .c1{color:#3D7B7B; font-style:italic}.pdoc-code .cs{color:#3D7B7B; font-style:italic}.pdoc-code .gd{color:#A00000}.pdoc-code .ge{font-style:italic}.pdoc-code .gr{color:#E40000}.pdoc-code .gh{color:#000080; font-weight:bold}.pdoc-code .gi{color:#008400}.pdoc-code .go{color:#717171}.pdoc-code .gp{color:#000080; font-weight:bold}.pdoc-code .gs{font-weight:bold}.pdoc-code .gu{color:#800080; font-weight:bold}.pdoc-code .gt{color:#0044DD}.pdoc-code .kc{color:#008000; font-weight:bold}.pdoc-code .kd{color:#008000; font-weight:bold}.pdoc-code .kn{color:#008000; font-weight:bold}.pdoc-code .kp{color:#008000}.pdoc-code .kr{color:#008000; font-weight:bold}.pdoc-code .kt{color:#B00040}.pdoc-code .m{color:#666666}.pdoc-code .s{color:#BA2121}.pdoc-code .na{color:#687822}.pdoc-code .nb{color:#008000}.pdoc-code .nc{color:#0000FF; font-weight:bold}.pdoc-code .no{color:#880000}.pdoc-code .nd{color:#AA22FF}.pdoc-code .ni{color:#717171; font-weight:bold}.pdoc-code .ne{color:#CB3F38; font-weight:bold}.pdoc-code .nf{color:#0000FF}.pdoc-code .nl{color:#767600}.pdoc-code .nn{color:#0000FF; font-weight:bold}.pdoc-code .nt{color:#008000; font-weight:bold}.pdoc-code .nv{color:#19177C}.pdoc-code .ow{color:#AA22FF; font-weight:bold}.pdoc-code .w{color:#bbbbbb}.pdoc-code .mb{color:#666666}.pdoc-code .mf{color:#666666}.pdoc-code .mh{color:#666666}.pdoc-code .mi{color:#666666}.pdoc-code .mo{color:#666666}.pdoc-code .sa{color:#BA2121}.pdoc-code .sb{color:#BA2121}.pdoc-code .sc{color:#BA2121}.pdoc-code .dl{color:#BA2121}.pdoc-code .sd{color:#BA2121; font-style:italic}.pdoc-code .s2{color:#BA2121}.pdoc-code .se{color:#AA5D1F; font-weight:bold}.pdoc-code .sh{color:#BA2121}.pdoc-code .si{color:#A45A77; font-weight:bold}.pdoc-code .sx{color:#008000}.pdoc-code .sr{color:#A45A77}.pdoc-code .s1{color:#BA2121}.pdoc-code .ss{color:#19177C}.pdoc-code .bp{color:#008000}.pdoc-code .fm{color:#0000FF}.pdoc-code .vc{color:#19177C}.pdoc-code .vg{color:#19177C}.pdoc-code .vi{color:#19177C}.pdoc-code .vm{color:#19177C}.pdoc-code .il{color:#666666}</style>
    <style>/*! theme.css */:root{--pdoc-background:#fff;}.pdoc{--text:#212529;--muted:#6c757d;--link:#3660a5;--link-hover:#1659c5;--code:#f8f8f8;--active:#fff598;--accent:#eee;--accent2:#c1c1c1;--nav-hover:rgba(255, 255, 255, 0.5);--name:#0066BB;--def:#008800;--annotation:#007020;}</style>
    <style>/*! layout.css */html, body{width:100%;height:100%;}html, main{scroll-behavior:smooth;}body{background-color:var(--pdoc-background);}@media (max-width:769px){#navtoggle{cursor:pointer;position:absolute;width:50px;height:40px;top:1rem;right:1rem;border-color:var(--text);color:var(--text);display:flex;opacity:0.8;z-index:999;}#navtoggle:hover{opacity:1;}#togglestate + div{display:none;}#togglestate:checked + div{display:inherit;}main, header{padding:2rem 3vw;}header + main{margin-top:-3rem;}.git-button{display:none !important;}nav input[type="search"]{max-width:77%;}nav input[type="search"]:first-child{margin-top:-6px;}nav input[type="search"]:valid ~ *{display:none !important;}}@media (min-width:770px){:root{--sidebar-width:clamp(12.5rem, 28vw, 22rem);}nav{position:fixed;overflow:auto;height:100vh;width:var(--sidebar-width);}main, header{padding:3rem 2rem 3rem calc(var(--sidebar-width) + 3rem);width:calc(54rem + var(--sidebar-width));max-width:100%;}header + main{margin-top:-4rem;}#navtoggle{display:none;}}#togglestate{position:absolute;height:0;opacity:0;}nav.pdoc{--pad:clamp(0.5rem, 2vw, 1.75rem);--indent:1.5rem;background-color:var(--accent);border-right:1px solid var(--accent2);box-shadow:0 0 20px rgba(50, 50, 50, .2) inset;padding:0 0 0 var(--pad);overflow-wrap:anywhere;scrollbar-width:thin; scrollbar-color:var(--accent2) transparent; z-index:1}nav.pdoc::-webkit-scrollbar{width:.4rem; }nav.pdoc::-webkit-scrollbar-thumb{background-color:var(--accent2); }nav.pdoc > div{padding:var(--pad) 0;}nav.pdoc .module-list-button{display:inline-flex;align-items:center;color:var(--text);border-color:var(--muted);margin-bottom:1rem;}nav.pdoc .module-list-button:hover{border-color:var(--text);}nav.pdoc input[type=search]{display:block;outline-offset:0;width:calc(100% - var(--pad));}nav.pdoc .logo{max-width:calc(100% - var(--pad));max-height:35vh;display:block;margin:0 auto 1rem;transform:translate(calc(-.5 * var(--pad)), 0);}nav.pdoc ul{list-style:none;padding-left:0;}nav.pdoc > div > ul{margin-left:calc(0px - var(--pad));}nav.pdoc li a{padding:.2rem 0 .2rem calc(var(--pad) + var(--indent));}nav.pdoc > div > ul > li > a{padding-left:var(--pad);}nav.pdoc li{transition:all 100ms;}nav.pdoc li:hover{background-color:var(--nav-hover);}nav.pdoc a, nav.pdoc a:hover{color:var(--text);}nav.pdoc a{display:block;}nav.pdoc > h2:first-of-type{margin-top:1.5rem;}nav.pdoc .class:before{content:"class ";color:var(--muted);}nav.pdoc .function:after{content:"()";color:var(--muted);}nav.pdoc footer:before{content:"";display:block;width:calc(100% - var(--pad));border-top:solid var(--accent2) 1px;margin-top:1.5rem;padding-top:.5rem;}nav.pdoc footer{font-size:small;}</style>
    <style>/*! content.css */.pdoc{color:var(--text);box-sizing:border-box;line-height:1.5;background:none;}.pdoc .pdoc-button{cursor:pointer;display:inline-block;border:solid black 1px;border-radius:2px;font-size:.75rem;padding:calc(0.5em - 1px) 1em;transition:100ms all;}.pdoc .pdoc-alert{padding:1rem 1rem 1rem calc(1.5rem + 24px);border:1px solid transparent;border-radius:.25rem;background-repeat:no-repeat;background-position:1rem center;margin-bottom:1rem;}.pdoc .pdoc-alert > *:last-child{margin-bottom:0;}.pdoc .pdoc-alert-note {color:#084298;background-color:#cfe2ff;border-color:#b6d4fe;background-image:url("data:image/svg+xml,%3Csvg%20xmlns%3D%22http%3A//www.w3.org/2000/svg%22%20width%3D%2224%22%20height%3D%2224%22%20fill%3D%22%23084298%22%20viewBox%3D%220%200%2016%2016%22%3E%3Cpath%20d%3D%22M8%2016A8%208%200%201%200%208%200a8%208%200%200%200%200%2016zm.93-9.412-1%204.705c-.07.34.029.533.304.533.194%200%20.487-.07.686-.246l-.088.416c-.287.346-.92.598-1.465.598-.703%200-1.002-.422-.808-1.319l.738-3.468c.064-.293.006-.399-.287-.47l-.451-.081.082-.381%202.29-.287zM8%205.5a1%201%200%201%201%200-2%201%201%200%200%201%200%202z%22/%3E%3C/svg%3E");}.pdoc .pdoc-alert-warning{color:#664d03;background-color:#fff3cd;border-color:#ffecb5;background-image:url("data:image/svg+xml,%3Csvg%20xmlns%3D%22http%3A//www.w3.org/2000/svg%22%20width%3D%2224%22%20height%3D%2224%22%20fill%3D%22%23664d03%22%20viewBox%3D%220%200%2016%2016%22%3E%3Cpath%20d%3D%22M8.982%201.566a1.13%201.13%200%200%200-1.96%200L.165%2013.233c-.457.778.091%201.767.98%201.767h13.713c.889%200%201.438-.99.98-1.767L8.982%201.566zM8%205c.535%200%20.954.462.9.995l-.35%203.507a.552.552%200%200%201-1.1%200L7.1%205.995A.905.905%200%200%201%208%205zm.002%206a1%201%200%201%201%200%202%201%201%200%200%201%200-2z%22/%3E%3C/svg%3E");}.pdoc .pdoc-alert-danger{color:#842029;background-color:#f8d7da;border-color:#f5c2c7;background-image:url("data:image/svg+xml,%3Csvg%20xmlns%3D%22http%3A//www.w3.org/2000/svg%22%20width%3D%2224%22%20height%3D%2224%22%20fill%3D%22%23842029%22%20viewBox%3D%220%200%2016%2016%22%3E%3Cpath%20d%3D%22M5.52.359A.5.5%200%200%201%206%200h4a.5.5%200%200%201%20.474.658L8.694%206H12.5a.5.5%200%200%201%20.395.807l-7%209a.5.5%200%200%201-.873-.454L6.823%209.5H3.5a.5.5%200%200%201-.48-.641l2.5-8.5z%22/%3E%3C/svg%3E");}.pdoc .visually-hidden{position:absolute !important;width:1px !important;height:1px !important;padding:0 !important;margin:-1px !important;overflow:hidden !important;clip:rect(0, 0, 0, 0) !important;white-space:nowrap !important;border:0 !important;}.pdoc h1, .pdoc h2, .pdoc h3{font-weight:300;margin:.3em 0;padding:.2em 0;}.pdoc > section:not(.module-info) h1{font-size:1.5rem;font-weight:500;}.pdoc > section:not(.module-info) h2{font-size:1.4rem;font-weight:500;}.pdoc > section:not(.module-info) h3{font-size:1.3rem;font-weight:500;}.pdoc > section:not(.module-info) h4{font-size:1.2rem;}.pdoc > section:not(.module-info) h5{font-size:1.1rem;}.pdoc a{text-decoration:none;color:var(--link);}.pdoc a:hover{color:var(--link-hover);}.pdoc blockquote{margin-left:2rem;}.pdoc pre{border-top:1px solid var(--accent2);border-bottom:1px solid var(--accent2);margin-top:0;margin-bottom:1em;padding:.5rem 0 .5rem .5rem;overflow-x:auto;background-color:var(--code);}.pdoc code{color:var(--text);padding:.2em .4em;margin:0;font-size:85%;background-color:var(--accent);border-radius:6px;}.pdoc a > code{color:inherit;}.pdoc pre > code{display:inline-block;font-size:inherit;background:none;border:none;padding:0;}.pdoc > section:not(.module-info){margin-bottom:1.5rem;}.pdoc .modulename{margin-top:0;font-weight:bold;}.pdoc .modulename a{color:var(--link);transition:100ms all;}.pdoc .git-button{float:right;border:solid var(--link) 1px;}.pdoc .git-button:hover{background-color:var(--link);color:var(--pdoc-background);}.view-source-toggle-state,.view-source-toggle-state ~ .pdoc-code{display:none;}.view-source-toggle-state:checked ~ .pdoc-code{display:block;}.view-source-button{display:inline-block;float:right;font-size:.75rem;line-height:1.5rem;color:var(--muted);padding:0 .4rem 0 1.3rem;cursor:pointer;text-indent:-2px;}.view-source-button > span{visibility:hidden;}.module-info .view-source-button{float:none;display:flex;justify-content:flex-end;margin:-1.2rem .4rem -.2rem 0;}.view-source-button::before{position:absolute;content:"View Source";display:list-item;list-style-type:disclosure-closed;}.view-source-toggle-state:checked ~ .attr .view-source-button::before,.view-source-toggle-state:checked ~ .view-source-button::before{list-style-type:disclosure-open;}.pdoc .docstring{margin-bottom:1.5rem;}.pdoc section:not(.module-info) .docstring{margin-left:clamp(0rem, 5vw - 2rem, 1rem);}.pdoc .docstring .pdoc-code{margin-left:1em;margin-right:1em;}.pdoc h1:target,.pdoc h2:target,.pdoc h3:target,.pdoc h4:target,.pdoc h5:target,.pdoc h6:target,.pdoc .pdoc-code > pre > span:target{background-color:var(--active);box-shadow:-1rem 0 0 0 var(--active);}.pdoc .pdoc-code > pre > span:target{display:block;}.pdoc div:target > .attr,.pdoc section:target > .attr,.pdoc dd:target > a{background-color:var(--active);}.pdoc *{scroll-margin:2rem;}.pdoc .pdoc-code .linenos{user-select:none;}.pdoc .attr:hover{filter:contrast(0.95);}.pdoc section, .pdoc .classattr{position:relative;}.pdoc .headerlink{--width:clamp(1rem, 3vw, 2rem);position:absolute;top:0;left:calc(0rem - var(--width));transition:all 100ms ease-in-out;opacity:0;}.pdoc .headerlink::before{content:"#";display:block;text-align:center;width:var(--width);height:2.3rem;line-height:2.3rem;font-size:1.5rem;}.pdoc .attr:hover ~ .headerlink,.pdoc *:target > .headerlink,.pdoc .headerlink:hover{opacity:1;}.pdoc .attr{display:block;margin:.5rem 0 .5rem;padding:.4rem .4rem .4rem 1rem;background-color:var(--accent);overflow-x:auto;}.pdoc .classattr{margin-left:2rem;}.pdoc .name{color:var(--name);font-weight:bold;}.pdoc .def{color:var(--def);font-weight:bold;}.pdoc .signature{background-color:transparent;}.pdoc .param, .pdoc .return-annotation{white-space:pre;}.pdoc .signature.multiline .param{display:block;}.pdoc .signature.condensed .param{display:inline-block;}.pdoc .annotation{color:var(--annotation);}.pdoc .view-value-toggle-state,.pdoc .view-value-toggle-state ~ .default_value{display:none;}.pdoc .view-value-toggle-state:checked ~ .default_value{display:inherit;}.pdoc .view-value-button{font-size:.5rem;vertical-align:middle;border-style:dashed;margin-top:-0.1rem;}.pdoc .view-value-button:hover{background:white;}.pdoc .view-value-button::before{content:"show";text-align:center;width:2.2em;display:inline-block;}.pdoc .view-value-toggle-state:checked ~ .view-value-button::before{content:"hide";}.pdoc .inherited{margin-left:2rem;}.pdoc .inherited dt{font-weight:700;}.pdoc .inherited dt, .pdoc .inherited dd{display:inline;margin-left:0;margin-bottom:.5rem;}.pdoc .inherited dd:not(:last-child):after{content:", ";}.pdoc .inherited .class:before{content:"class ";}.pdoc .inherited .function a:after{content:"()";}.pdoc .search-result .docstring{overflow:auto;max-height:25vh;}.pdoc .search-result.focused > .attr{background-color:var(--active);}.pdoc .attribution{margin-top:2rem;display:block;opacity:0.5;transition:all 200ms;filter:grayscale(100%);}.pdoc .attribution:hover{opacity:1;filter:grayscale(0%);}.pdoc .attribution img{margin-left:5px;height:35px;vertical-align:middle;width:70px;transition:all 200ms;}.pdoc table{display:block;width:max-content;max-width:100%;overflow:auto;margin-bottom:1rem;}.pdoc table th{font-weight:600;}.pdoc table th, .pdoc table td{padding:6px 13px;border:1px solid var(--accent2);}</style>
    <style>/*! custom.css */</style></head>
<body>
    <nav class="pdoc">
        <label id="navtoggle" for="togglestate" class="pdoc-button"><svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 30 30'><path stroke-linecap='round' stroke="currentColor" stroke-miterlimit='10' stroke-width='2' d='M4 7h22M4 15h22M4 23h22'/></svg></label>
        <input id="togglestate" type="checkbox" aria-hidden="true" tabindex="-1">
        <div>            <a class="pdoc-button module-list-button" href="../objects.html">
<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-box-arrow-in-left" viewBox="0 0 16 16">
  <path fill-rule="evenodd" d="M10 3.5a.5.5 0 0 0-.5-.5h-8a.5.5 0 0 0-.5.5v9a.5.5 0 0 0 .5.5h8a.5.5 0 0 0 .5-.5v-2a.5.5 0 0 1 1 0v2A1.5 1.5 0 0 1 9.5 14h-8A1.5 1.5 0 0 1 0 12.5v-9A1.5 1.5 0 0 1 1.5 2h8A1.5 1.5 0 0 1 11 3.5v2a.5.5 0 0 1-1 0v-2z"/>
  <path fill-rule="evenodd" d="M4.146 8.354a.5.5 0 0 1 0-.708l3-3a.5.5 0 1 1 .708.708L5.707 7.5H14.5a.5.5 0 0 1 0 1H5.707l2.147 2.146a.5.5 0 0 1-.708.708l-3-3z"/>
</svg>                &nbsp;howard.objects</a>


            <input type="search" placeholder="Search..." role="searchbox" aria-label="search"
                   pattern=".+" required>



            <h2>API Documentation</h2>
                <ul class="memberlist">
            <li>
                    <a class="class" href="#Gene">Gene</a>
                            <ul class="memberlist">
                        <li>
                                <a class="function" href="#Gene.__init__">Gene</a>
                        </li>
                        <li>
                                <a class="variable" href="#Gene.name">name</a>
                        </li>
                </ul>

            </li>
            <li>
                    <a class="class" href="#Transcript">Transcript</a>
                            <ul class="memberlist">
                        <li>
                                <a class="function" href="#Transcript.__init__">Transcript</a>
                        </li>
                        <li>
                                <a class="variable" href="#Transcript.name">name</a>
                        </li>
                        <li>
                                <a class="variable" href="#Transcript.version">version</a>
                        </li>
                        <li>
                                <a class="variable" href="#Transcript.gene">gene</a>
                        </li>
                        <li>
                                <a class="variable" href="#Transcript.tx_position">tx_position</a>
                        </li>
                        <li>
                                <a class="variable" href="#Transcript.cds_position">cds_position</a>
                        </li>
                        <li>
                                <a class="variable" href="#Transcript.is_default">is_default</a>
                        </li>
                        <li>
                                <a class="variable" href="#Transcript.cdna_match">cdna_match</a>
                        </li>
                        <li>
                                <a class="variable" href="#Transcript.full_name">full_name</a>
                        </li>
                        <li>
                                <a class="variable" href="#Transcript.is_coding">is_coding</a>
                        </li>
                        <li>
                                <a class="variable" href="#Transcript.strand">strand</a>
                        </li>
                        <li>
                                <a class="function" href="#Transcript.ordered_cdna_match">ordered_cdna_match</a>
                        </li>
                        <li>
                                <a class="function" href="#Transcript.get_cds_start_stop">get_cds_start_stop</a>
                        </li>
                        <li>
                                <a class="function" href="#Transcript.start_codon">start_codon</a>
                        </li>
                        <li>
                                <a class="function" href="#Transcript.stop_codon">stop_codon</a>
                        </li>
                        <li>
                                <a class="function" href="#Transcript.cdna_to_genomic_coord">cdna_to_genomic_coord</a>
                        </li>
                        <li>
                                <a class="function" href="#Transcript.genomic_to_cdna_coord">genomic_to_cdna_coord</a>
                        </li>
                        <li>
                                <a class="function" href="#Transcript.find_exon_number">find_exon_number</a>
                        </li>
                </ul>

            </li>
            <li>
                    <a class="class" href="#BED6Interval_base">BED6Interval_base</a>
                            <ul class="memberlist">
                        <li>
                                <a class="function" href="#BED6Interval_base.__init__">BED6Interval_base</a>
                        </li>
                        <li>
                                <a class="variable" href="#BED6Interval_base.chrom">chrom</a>
                        </li>
                        <li>
                                <a class="variable" href="#BED6Interval_base.chrom_start">chrom_start</a>
                        </li>
                        <li>
                                <a class="variable" href="#BED6Interval_base.chrom_end">chrom_end</a>
                        </li>
                        <li>
                                <a class="variable" href="#BED6Interval_base.name">name</a>
                        </li>
                        <li>
                                <a class="variable" href="#BED6Interval_base.score">score</a>
                        </li>
                        <li>
                                <a class="variable" href="#BED6Interval_base.strand">strand</a>
                        </li>
                </ul>

            </li>
            <li>
                    <a class="class" href="#BED6Interval">BED6Interval</a>
                            <ul class="memberlist">
                        <li>
                                <a class="function" href="#BED6Interval.__init__">BED6Interval</a>
                        </li>
                        <li>
                                <a class="function" href="#BED6Interval.distance">distance</a>
                        </li>
                </ul>

            </li>
            <li>
                    <a class="class" href="#Exon">Exon</a>
                            <ul class="memberlist">
                        <li>
                                <a class="function" href="#Exon.__init__">Exon</a>
                        </li>
                        <li>
                                <a class="variable" href="#Exon.transcript">transcript</a>
                        </li>
                        <li>
                                <a class="variable" href="#Exon.tx_position">tx_position</a>
                        </li>
                        <li>
                                <a class="variable" href="#Exon.number">number</a>
                        </li>
                        <li>
                                <a class="variable" href="#Exon.name">name</a>
                        </li>
                        <li>
                                <a class="function" href="#Exon.get_as_interval">get_as_interval</a>
                        </li>
                        <li>
                                <a class="variable" href="#Exon.strand">strand</a>
                        </li>
                </ul>

            </li>
            <li>
                    <a class="class" href="#CDNA_Match">CDNA_Match</a>
                            <ul class="memberlist">
                        <li>
                                <a class="function" href="#CDNA_Match.__init__">CDNA_Match</a>
                        </li>
                        <li>
                                <a class="variable" href="#CDNA_Match.cdna_start">cdna_start</a>
                        </li>
                        <li>
                                <a class="variable" href="#CDNA_Match.cdna_end">cdna_end</a>
                        </li>
                        <li>
                                <a class="variable" href="#CDNA_Match.gap">gap</a>
                        </li>
                        <li>
                                <a class="variable" href="#CDNA_Match.length">length</a>
                        </li>
                        <li>
                                <a class="function" href="#CDNA_Match.get_offset">get_offset</a>
                        </li>
                </ul>

            </li>
    </ul>



        <a class="attribution" title="pdoc: Python API documentation generator" href="https://pdoc.dev" target="_blank">
            built with <span class="visually-hidden">pdoc</span><img
                alt="pdoc logo"
                src="data:image/svg+xml,%3Csvg%20xmlns%3D%22http%3A//www.w3.org/2000/svg%22%20role%3D%22img%22%20aria-label%3D%22pdoc%20logo%22%20width%3D%22300%22%20height%3D%22150%22%20viewBox%3D%22-1%200%2060%2030%22%3E%3Ctitle%3Epdoc%3C/title%3E%3Cpath%20d%3D%22M29.621%2021.293c-.011-.273-.214-.475-.511-.481a.5.5%200%200%200-.489.503l-.044%201.393c-.097.551-.695%201.215-1.566%201.704-.577.428-1.306.486-2.193.182-1.426-.617-2.467-1.654-3.304-2.487l-.173-.172a3.43%203.43%200%200%200-.365-.306.49.49%200%200%200-.286-.196c-1.718-1.06-4.931-1.47-7.353.191l-.219.15c-1.707%201.187-3.413%202.131-4.328%201.03-.02-.027-.49-.685-.141-1.763.233-.721.546-2.408.772-4.076.042-.09.067-.187.046-.288.166-1.347.277-2.625.241-3.351%201.378-1.008%202.271-2.586%202.271-4.362%200-.976-.272-1.935-.788-2.774-.057-.094-.122-.18-.184-.268.033-.167.052-.339.052-.516%200-1.477-1.202-2.679-2.679-2.679-.791%200-1.496.352-1.987.9a6.3%206.3%200%200%200-1.001.029c-.492-.564-1.207-.929-2.012-.929-1.477%200-2.679%201.202-2.679%202.679A2.65%202.65%200%200%200%20.97%206.554c-.383.747-.595%201.572-.595%202.41%200%202.311%201.507%204.29%203.635%205.107-.037.699-.147%202.27-.423%203.294l-.137.461c-.622%202.042-2.515%208.257%201.727%2010.643%201.614.908%203.06%201.248%204.317%201.248%202.665%200%204.492-1.524%205.322-2.401%201.476-1.559%202.886-1.854%206.491.82%201.877%201.393%203.514%201.753%204.861%201.068%202.223-1.713%202.811-3.867%203.399-6.374.077-.846.056-1.469.054-1.537zm-4.835%204.313c-.054.305-.156.586-.242.629-.034-.007-.131-.022-.307-.157-.145-.111-.314-.478-.456-.908.221.121.432.25.675.355.115.039.219.051.33.081zm-2.251-1.238c-.05.33-.158.648-.252.694-.022.001-.125-.018-.307-.157-.217-.166-.488-.906-.639-1.573.358.344.754.693%201.198%201.036zm-3.887-2.337c-.006-.116-.018-.231-.041-.342.635.145%201.189.368%201.599.625.097.231.166.481.174.642-.03.049-.055.101-.067.158-.046.013-.128.026-.298.004-.278-.037-.901-.57-1.367-1.087zm-1.127-.497c.116.306.176.625.12.71-.019.014-.117.045-.345.016-.206-.027-.604-.332-.986-.695.41-.051.816-.056%201.211-.031zm-4.535%201.535c.209.22.379.47.358.598-.006.041-.088.138-.351.234-.144.055-.539-.063-.979-.259a11.66%2011.66%200%200%200%20.972-.573zm.983-.664c.359-.237.738-.418%201.126-.554.25.237.479.548.457.694-.006.042-.087.138-.351.235-.174.064-.694-.105-1.232-.375zm-3.381%201.794c-.022.145-.061.29-.149.401-.133.166-.358.248-.69.251h-.002c-.133%200-.306-.26-.45-.621.417.091.854.07%201.291-.031zm-2.066-8.077a4.78%204.78%200%200%201-.775-.584c.172-.115.505-.254.88-.378l-.105.962zm-.331%202.302a10.32%2010.32%200%200%201-.828-.502c.202-.143.576-.328.984-.49l-.156.992zm-.45%202.157l-.701-.403c.214-.115.536-.249.891-.376a11.57%2011.57%200%200%201-.19.779zm-.181%201.716c.064.398.194.702.298.893-.194-.051-.435-.162-.736-.398.061-.119.224-.3.438-.495zM8.87%204.141c0%20.152-.123.276-.276.276s-.275-.124-.275-.276.123-.276.276-.276.275.124.275.276zm-.735-.389a1.15%201.15%200%200%200-.314.783%201.16%201.16%200%200%200%201.162%201.162c.457%200%20.842-.27%201.032-.653.026.117.042.238.042.362a1.68%201.68%200%200%201-1.679%201.679%201.68%201.68%200%200%201-1.679-1.679c0-.843.626-1.535%201.436-1.654zM5.059%205.406A1.68%201.68%200%200%201%203.38%207.085a1.68%201.68%200%200%201-1.679-1.679c0-.037.009-.072.011-.109.21.3.541.508.935.508a1.16%201.16%200%200%200%201.162-1.162%201.14%201.14%200%200%200-.474-.912c.015%200%20.03-.005.045-.005.926.001%201.679.754%201.679%201.68zM3.198%204.141c0%20.152-.123.276-.276.276s-.275-.124-.275-.276.123-.276.276-.276.275.124.275.276zM1.375%208.964c0-.52.103-1.035.288-1.52.466.394%201.06.64%201.717.64%201.144%200%202.116-.725%202.499-1.738.383%201.012%201.355%201.738%202.499%201.738.867%200%201.631-.421%202.121-1.062.307.605.478%201.267.478%201.942%200%202.486-2.153%204.51-4.801%204.51s-4.801-2.023-4.801-4.51zm24.342%2019.349c-.985.498-2.267.168-3.813-.979-3.073-2.281-5.453-3.199-7.813-.705-1.315%201.391-4.163%203.365-8.423.97-3.174-1.786-2.239-6.266-1.261-9.479l.146-.492c.276-1.02.395-2.457.444-3.268a6.11%206.11%200%200%200%201.18.115%206.01%206.01%200%200%200%202.536-.562l-.006.175c-.802.215-1.848.612-2.021%201.25-.079.295.021.601.274.837.219.203.415.364.598.501-.667.304-1.243.698-1.311%201.179-.02.144-.022.507.393.787.213.144.395.26.564.365-1.285.521-1.361.96-1.381%201.126-.018.142-.011.496.427.746l.854.489c-.473.389-.971.914-.999%201.429-.018.278.095.532.316.713.675.556%201.231.721%201.653.721.059%200%20.104-.014.158-.02.207.707.641%201.64%201.513%201.64h.013c.8-.008%201.236-.345%201.462-.626.173-.216.268-.457.325-.692.424.195.93.374%201.372.374.151%200%20.294-.021.423-.068.732-.27.944-.704.993-1.021.009-.061.003-.119.002-.179.266.086.538.147.789.147.15%200%20.294-.021.423-.069.542-.2.797-.489.914-.754.237.147.478.258.704.288.106.014.205.021.296.021.356%200%20.595-.101.767-.229.438.435%201.094.992%201.656%201.067.106.014.205.021.296.021a1.56%201.56%200%200%200%20.323-.035c.17.575.453%201.289.866%201.605.358.273.665.362.914.362a.99.99%200%200%200%20.421-.093%201.03%201.03%200%200%200%20.245-.164c.168.428.39.846.68%201.068.358.273.665.362.913.362a.99.99%200%200%200%20.421-.093c.317-.148.512-.448.639-.762.251.157.495.257.726.257.127%200%20.25-.024.37-.071.427-.17.706-.617.841-1.314.022-.015.047-.022.068-.038.067-.051.133-.104.196-.159-.443%201.486-1.107%202.761-2.086%203.257zM8.66%209.925a.5.5%200%201%200-1%200c0%20.653-.818%201.205-1.787%201.205s-1.787-.552-1.787-1.205a.5.5%200%201%200-1%200c0%201.216%201.25%202.205%202.787%202.205s2.787-.989%202.787-2.205zm4.4%2015.965l-.208.097c-2.661%201.258-4.708%201.436-6.086.527-1.542-1.017-1.88-3.19-1.844-4.198a.4.4%200%200%200-.385-.414c-.242-.029-.406.164-.414.385-.046%201.249.367%203.686%202.202%204.896.708.467%201.547.7%202.51.7%201.248%200%202.706-.392%204.362-1.174l.185-.086a.4.4%200%200%200%20.205-.527c-.089-.204-.326-.291-.527-.206zM9.547%202.292c.093.077.205.114.317.114a.5.5%200%200%200%20.318-.886L8.817.397a.5.5%200%200%200-.703.068.5.5%200%200%200%20.069.703l1.364%201.124zm-7.661-.065c.086%200%20.173-.022.253-.068l1.523-.893a.5.5%200%200%200-.506-.863l-1.523.892a.5.5%200%200%200-.179.685c.094.158.261.247.432.247z%22%20transform%3D%22matrix%28-1%200%200%201%2058%200%29%22%20fill%3D%22%233bb300%22/%3E%3Cpath%20d%3D%22M.3%2021.86V10.18q0-.46.02-.68.04-.22.18-.5.28-.54%201.34-.54%201.06%200%201.42.28.38.26.44.78.76-1.04%202.38-1.04%201.64%200%203.1%201.54%201.46%201.54%201.46%203.58%200%202.04-1.46%203.58-1.44%201.54-3.08%201.54-1.64%200-2.38-.92v4.04q0%20.46-.04.68-.02.22-.18.5-.14.3-.5.42-.36.12-.98.12-.62%200-1-.12-.36-.12-.52-.4-.14-.28-.18-.5-.02-.22-.02-.68zm3.96-9.42q-.46.54-.46%201.18%200%20.64.46%201.18.48.52%201.2.52.74%200%201.24-.52.52-.52.52-1.18%200-.66-.48-1.18-.48-.54-1.26-.54-.76%200-1.22.54zm14.741-8.36q.16-.3.54-.42.38-.12%201-.12.64%200%201.02.12.38.12.52.42.16.3.18.54.04.22.04.68v11.94q0%20.46-.04.7-.02.22-.18.5-.3.54-1.7.54-1.38%200-1.54-.98-.84.96-2.34.96-1.8%200-3.28-1.56-1.48-1.58-1.48-3.66%200-2.1%201.48-3.68%201.5-1.58%203.28-1.58%201.48%200%202.3%201v-4.2q0-.46.02-.68.04-.24.18-.52zm-3.24%2010.86q.52.54%201.26.54.74%200%201.22-.54.5-.54.5-1.18%200-.66-.48-1.22-.46-.56-1.26-.56-.8%200-1.28.56-.48.54-.48%201.2%200%20.66.52%201.2zm7.833-1.2q0-2.4%201.68-3.96%201.68-1.56%203.84-1.56%202.16%200%203.82%201.56%201.66%201.54%201.66%203.94%200%201.66-.86%202.96-.86%201.28-2.1%201.9-1.22.6-2.54.6-1.32%200-2.56-.64-1.24-.66-2.1-1.92-.84-1.28-.84-2.88zm4.18%201.44q.64.48%201.3.48.66%200%201.32-.5.66-.5.66-1.48%200-.98-.62-1.46-.62-.48-1.34-.48-.72%200-1.34.5-.62.5-.62%201.48%200%20.96.64%201.46zm11.412-1.44q0%20.84.56%201.32.56.46%201.18.46.64%200%201.18-.36.56-.38.9-.38.6%200%201.46%201.06.46.58.46%201.04%200%20.76-1.1%201.42-1.14.8-2.8.8-1.86%200-3.58-1.34-.82-.64-1.34-1.7-.52-1.08-.52-2.36%200-1.3.52-2.34.52-1.06%201.34-1.7%201.66-1.32%203.54-1.32.76%200%201.48.22.72.2%201.06.4l.32.2q.36.24.56.38.52.4.52.92%200%20.5-.42%201.14-.72%201.1-1.38%201.1-.38%200-1.08-.44-.36-.34-1.04-.34-.66%200-1.24.48-.58.48-.58%201.34z%22%20fill%3D%22green%22/%3E%3C/svg%3E"/>
        </a>
</div>
    </nav>
    <main class="pdoc">
            <section class="module-info">
                    <h1 class="modulename">
<a href="./../../howard.html">howard</a><wbr>.<a href="./../objects.html">objects</a><wbr>.transcript    </h1>

                        <div class="docstring"><p>Models for representing genomic elements.</p>
</div>

                        <input id="mod-transcript-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">

                        <label class="view-source-button" for="mod-transcript-view-source"><span>View Source</span></label>

                        <div class="pdoc-code codehilite"><pre><span></span><span id="L-1"><a href="#L-1"><span class="linenos">  1</span></a><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-2"><a href="#L-2"><span class="linenos">  2</span></a><span class="sd">Models for representing genomic elements.</span>
</span><span id="L-3"><a href="#L-3"><span class="linenos">  3</span></a><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-4"><a href="#L-4"><span class="linenos">  4</span></a>
</span><span id="L-5"><a href="#L-5"><span class="linenos">  5</span></a><span class="kn">from</span> <span class="nn">__future__</span> <span class="kn">import</span> <span class="n">unicode_literals</span>
</span><span id="L-6"><a href="#L-6"><span class="linenos">  6</span></a>
</span><span id="L-7"><a href="#L-7"><span class="linenos">  7</span></a><span class="kn">from</span> <span class="nn">collections</span> <span class="kn">import</span> <span class="n">namedtuple</span>
</span><span id="L-8"><a href="#L-8"><span class="linenos">  8</span></a>
</span><span id="L-9"><a href="#L-9"><span class="linenos">  9</span></a><span class="kn">from</span> <span class="nn">lazy</span> <span class="kn">import</span> <span class="n">lazy</span>
</span><span id="L-10"><a href="#L-10"><span class="linenos"> 10</span></a>
</span><span id="L-11"><a href="#L-11"><span class="linenos"> 11</span></a><span class="kn">from</span> <span class="nn">howard.objects.cdna</span> <span class="kn">import</span> <span class="n">CDNA_START_CODON</span><span class="p">,</span> <span class="n">CDNA_STOP_CODON</span><span class="p">,</span> <span class="n">CDNACoord</span>
</span><span id="L-12"><a href="#L-12"><span class="linenos"> 12</span></a>
</span><span id="L-13"><a href="#L-13"><span class="linenos"> 13</span></a>
</span><span id="L-14"><a href="#L-14"><span class="linenos"> 14</span></a><span class="c1"># The Gene class is a basic template for representing genes in Python.</span>
</span><span id="L-15"><a href="#L-15"><span class="linenos"> 15</span></a><span class="k">class</span> <span class="nc">Gene</span><span class="p">(</span><span class="nb">object</span><span class="p">):</span>
</span><span id="L-16"><a href="#L-16"><span class="linenos"> 16</span></a>    <span class="k">def</span> <span class="fm">__init__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">name</span><span class="p">):</span>
</span><span id="L-17"><a href="#L-17"><span class="linenos"> 17</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">name</span> <span class="o">=</span> <span class="n">name</span>
</span><span id="L-18"><a href="#L-18"><span class="linenos"> 18</span></a>
</span><span id="L-19"><a href="#L-19"><span class="linenos"> 19</span></a>
</span><span id="L-20"><a href="#L-20"><span class="linenos"> 20</span></a><span class="c1"># The Transcript class is a blueprint for creating objects that represent a transcript.</span>
</span><span id="L-21"><a href="#L-21"><span class="linenos"> 21</span></a><span class="c1"># A gene may have multiple transcripts with different combinations of exons.</span>
</span><span id="L-22"><a href="#L-22"><span class="linenos"> 22</span></a><span class="c1"># We need both exons and cdna_match as need to know exact exon boundaries to work out flanking</span>
</span><span id="L-23"><a href="#L-23"><span class="linenos"> 23</span></a><span class="k">class</span> <span class="nc">Transcript</span><span class="p">(</span><span class="nb">object</span><span class="p">):</span>
</span><span id="L-24"><a href="#L-24"><span class="linenos"> 24</span></a>
</span><span id="L-25"><a href="#L-25"><span class="linenos"> 25</span></a>    <span class="k">def</span> <span class="fm">__init__</span><span class="p">(</span>
</span><span id="L-26"><a href="#L-26"><span class="linenos"> 26</span></a>        <span class="bp">self</span><span class="p">,</span>
</span><span id="L-27"><a href="#L-27"><span class="linenos"> 27</span></a>        <span class="n">name</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
</span><span id="L-28"><a href="#L-28"><span class="linenos"> 28</span></a>        <span class="n">version</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
</span><span id="L-29"><a href="#L-29"><span class="linenos"> 29</span></a>        <span class="n">gene</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
</span><span id="L-30"><a href="#L-30"><span class="linenos"> 30</span></a>        <span class="n">tx_position</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span>
</span><span id="L-31"><a href="#L-31"><span class="linenos"> 31</span></a>        <span class="n">cds_position</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span>
</span><span id="L-32"><a href="#L-32"><span class="linenos"> 32</span></a>        <span class="n">is_default</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
</span><span id="L-33"><a href="#L-33"><span class="linenos"> 33</span></a>        <span class="n">cdna_match</span><span class="p">:</span> <span class="nb">list</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
</span><span id="L-34"><a href="#L-34"><span class="linenos"> 34</span></a>        <span class="n">start_codon_transcript_pos</span><span class="p">:</span> <span class="nb">int</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
</span><span id="L-35"><a href="#L-35"><span class="linenos"> 35</span></a>        <span class="n">stop_codon_transcript_pos</span><span class="p">:</span> <span class="nb">int</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
</span><span id="L-36"><a href="#L-36"><span class="linenos"> 36</span></a>    <span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
</span><span id="L-37"><a href="#L-37"><span class="linenos"> 37</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-38"><a href="#L-38"><span class="linenos"> 38</span></a><span class="sd">        The function initializes an object with various attributes related to a gene and its transcript.</span>
</span><span id="L-39"><a href="#L-39"><span class="linenos"> 39</span></a>
</span><span id="L-40"><a href="#L-40"><span class="linenos"> 40</span></a><span class="sd">        :param name: A string representing the name of the coding sequence</span>
</span><span id="L-41"><a href="#L-41"><span class="linenos"> 41</span></a><span class="sd">        :type name: str</span>
</span><span id="L-42"><a href="#L-42"><span class="linenos"> 42</span></a><span class="sd">        :param version: The `version` parameter is a string that represents the version of the object.</span>
</span><span id="L-43"><a href="#L-43"><span class="linenos"> 43</span></a><span class="sd">        It is used to track changes or updates to the object over time</span>
</span><span id="L-44"><a href="#L-44"><span class="linenos"> 44</span></a><span class="sd">        :type version: str</span>
</span><span id="L-45"><a href="#L-45"><span class="linenos"> 45</span></a><span class="sd">        :param gene: The `gene` parameter is a string that represents the gene associated with the</span>
</span><span id="L-46"><a href="#L-46"><span class="linenos"> 46</span></a><span class="sd">        coding sequence</span>
</span><span id="L-47"><a href="#L-47"><span class="linenos"> 47</span></a><span class="sd">        :type gene: str</span>
</span><span id="L-48"><a href="#L-48"><span class="linenos"> 48</span></a><span class="sd">        :param tx_position: The `tx_position` parameter represents the position of the transcript. It is</span>
</span><span id="L-49"><a href="#L-49"><span class="linenos"> 49</span></a><span class="sd">        an integer value that indicates the position of the transcript in the genome</span>
</span><span id="L-50"><a href="#L-50"><span class="linenos"> 50</span></a><span class="sd">        :type tx_position: int</span>
</span><span id="L-51"><a href="#L-51"><span class="linenos"> 51</span></a><span class="sd">        :param cds_position: The `cds_position` parameter represents the position of the coding sequence</span>
</span><span id="L-52"><a href="#L-52"><span class="linenos"> 52</span></a><span class="sd">        (CDS) within the transcript. It is an integer value that indicates the starting position of the</span>
</span><span id="L-53"><a href="#L-53"><span class="linenos"> 53</span></a><span class="sd">        CDS within the transcript sequence</span>
</span><span id="L-54"><a href="#L-54"><span class="linenos"> 54</span></a><span class="sd">        :type cds_position: int</span>
</span><span id="L-55"><a href="#L-55"><span class="linenos"> 55</span></a><span class="sd">        :param is_default: The `is_default` parameter is a boolean flag that indicates whether the</span>
</span><span id="L-56"><a href="#L-56"><span class="linenos"> 56</span></a><span class="sd">        instance of the class is the default version of the gene. It is set to `False` by default, but</span>
</span><span id="L-57"><a href="#L-57"><span class="linenos"> 57</span></a><span class="sd">        can be set to `True` if the instance is the default version, defaults to False</span>
</span><span id="L-58"><a href="#L-58"><span class="linenos"> 58</span></a><span class="sd">        :type is_default: bool (optional)</span>
</span><span id="L-59"><a href="#L-59"><span class="linenos"> 59</span></a><span class="sd">        :param cdna_match: The `cdna_match` parameter is a list that contains the positions of the</span>
</span><span id="L-60"><a href="#L-60"><span class="linenos"> 60</span></a><span class="sd">        matching cDNA sequences. It is an optional parameter and if not provided, it defaults to an</span>
</span><span id="L-61"><a href="#L-61"><span class="linenos"> 61</span></a><span class="sd">        empty list</span>
</span><span id="L-62"><a href="#L-62"><span class="linenos"> 62</span></a><span class="sd">        :type cdna_match: list</span>
</span><span id="L-63"><a href="#L-63"><span class="linenos"> 63</span></a><span class="sd">        :param start_codon_transcript_pos: The parameter &quot;start_codon_transcript_pos&quot; is an optional</span>
</span><span id="L-64"><a href="#L-64"><span class="linenos"> 64</span></a><span class="sd">        parameter that represents the transcript position of the start codon. It is used to store the</span>
</span><span id="L-65"><a href="#L-65"><span class="linenos"> 65</span></a><span class="sd">        pre-calculated transcript position of the start codon for a specific gene</span>
</span><span id="L-66"><a href="#L-66"><span class="linenos"> 66</span></a><span class="sd">        :type start_codon_transcript_pos: int</span>
</span><span id="L-67"><a href="#L-67"><span class="linenos"> 67</span></a><span class="sd">        :param stop_codon_transcript_pos: The parameter `stop_codon_transcript_pos` is an optional</span>
</span><span id="L-68"><a href="#L-68"><span class="linenos"> 68</span></a><span class="sd">        integer that represents the transcript position of the stop codon. It is used to store the</span>
</span><span id="L-69"><a href="#L-69"><span class="linenos"> 69</span></a><span class="sd">        pre-calculated transcript coordinate of the stop codon. If not provided, it will be set to</span>
</span><span id="L-70"><a href="#L-70"><span class="linenos"> 70</span></a><span class="sd">        `None`</span>
</span><span id="L-71"><a href="#L-71"><span class="linenos"> 71</span></a><span class="sd">        :type stop_codon_transcript_pos: int</span>
</span><span id="L-72"><a href="#L-72"><span class="linenos"> 72</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="L-73"><a href="#L-73"><span class="linenos"> 73</span></a>
</span><span id="L-74"><a href="#L-74"><span class="linenos"> 74</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">name</span> <span class="o">=</span> <span class="n">name</span>
</span><span id="L-75"><a href="#L-75"><span class="linenos"> 75</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">version</span> <span class="o">=</span> <span class="n">version</span>
</span><span id="L-76"><a href="#L-76"><span class="linenos"> 76</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">gene</span> <span class="o">=</span> <span class="n">Gene</span><span class="p">(</span><span class="n">gene</span><span class="p">)</span>
</span><span id="L-77"><a href="#L-77"><span class="linenos"> 77</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span> <span class="o">=</span> <span class="n">tx_position</span>
</span><span id="L-78"><a href="#L-78"><span class="linenos"> 78</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">cds_position</span> <span class="o">=</span> <span class="n">cds_position</span>
</span><span id="L-79"><a href="#L-79"><span class="linenos"> 79</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">is_default</span> <span class="o">=</span> <span class="n">is_default</span>
</span><span id="L-80"><a href="#L-80"><span class="linenos"> 80</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">cdna_match</span> <span class="o">=</span> <span class="n">cdna_match</span> <span class="ow">or</span> <span class="p">[]</span>
</span><span id="L-81"><a href="#L-81"><span class="linenos"> 81</span></a>        <span class="c1"># Optional pre-calculated transcript coordinates</span>
</span><span id="L-82"><a href="#L-82"><span class="linenos"> 82</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">_start_codon_transcript_pos</span> <span class="o">=</span> <span class="n">start_codon_transcript_pos</span>
</span><span id="L-83"><a href="#L-83"><span class="linenos"> 83</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">_stop_codon_transcript_pos</span> <span class="o">=</span> <span class="n">stop_codon_transcript_pos</span>
</span><span id="L-84"><a href="#L-84"><span class="linenos"> 84</span></a>
</span><span id="L-85"><a href="#L-85"><span class="linenos"> 85</span></a>    <span class="nd">@property</span>
</span><span id="L-86"><a href="#L-86"><span class="linenos"> 86</span></a>    <span class="k">def</span> <span class="nf">full_name</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
</span><span id="L-87"><a href="#L-87"><span class="linenos"> 87</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-88"><a href="#L-88"><span class="linenos"> 88</span></a><span class="sd">        The function `full_name` returns the full name of an object, including its version if it exists.</span>
</span><span id="L-89"><a href="#L-89"><span class="linenos"> 89</span></a><span class="sd">        :return: a string. If the `version` attribute of the object is not `None`, it returns a string</span>
</span><span id="L-90"><a href="#L-90"><span class="linenos"> 90</span></a><span class="sd">        in the format `name.version`. Otherwise, it returns just the `name` attribute.</span>
</span><span id="L-91"><a href="#L-91"><span class="linenos"> 91</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="L-92"><a href="#L-92"><span class="linenos"> 92</span></a>
</span><span id="L-93"><a href="#L-93"><span class="linenos"> 93</span></a>        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">version</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
</span><span id="L-94"><a href="#L-94"><span class="linenos"> 94</span></a>            <span class="k">return</span> <span class="s2">&quot;</span><span class="si">%s</span><span class="s2">.</span><span class="si">%d</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">name</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">version</span><span class="p">)</span>
</span><span id="L-95"><a href="#L-95"><span class="linenos"> 95</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="L-96"><a href="#L-96"><span class="linenos"> 96</span></a>            <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">name</span>
</span><span id="L-97"><a href="#L-97"><span class="linenos"> 97</span></a>
</span><span id="L-98"><a href="#L-98"><span class="linenos"> 98</span></a>    <span class="nd">@property</span>
</span><span id="L-99"><a href="#L-99"><span class="linenos"> 99</span></a>    <span class="k">def</span> <span class="nf">is_coding</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">bool</span><span class="p">:</span>
</span><span id="L-100"><a href="#L-100"><span class="linenos">100</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-101"><a href="#L-101"><span class="linenos">101</span></a><span class="sd">        The function checks if a coding transcript has a non-zero length coding sequence.</span>
</span><span id="L-102"><a href="#L-102"><span class="linenos">102</span></a><span class="sd">        :return: a boolean value indicating whether the coding transcript has a coding sequence (CDS)</span>
</span><span id="L-103"><a href="#L-103"><span class="linenos">103</span></a><span class="sd">        with a non-zero length.</span>
</span><span id="L-104"><a href="#L-104"><span class="linenos">104</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="L-105"><a href="#L-105"><span class="linenos">105</span></a>
</span><span id="L-106"><a href="#L-106"><span class="linenos">106</span></a>        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">cds_position</span><span class="o">.</span><span class="n">chrom_stop</span> <span class="o">-</span> <span class="bp">self</span><span class="o">.</span><span class="n">cds_position</span><span class="o">.</span><span class="n">chrom_start</span> <span class="o">&gt;</span> <span class="mi">0</span>
</span><span id="L-107"><a href="#L-107"><span class="linenos">107</span></a>
</span><span id="L-108"><a href="#L-108"><span class="linenos">108</span></a>    <span class="nd">@property</span>
</span><span id="L-109"><a href="#L-109"><span class="linenos">109</span></a>    <span class="k">def</span> <span class="nf">strand</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
</span><span id="L-110"><a href="#L-110"><span class="linenos">110</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-111"><a href="#L-111"><span class="linenos">111</span></a><span class="sd">        The function returns a string &#39;+&#39; if the tx_position is on the forward strand, and &#39;-&#39; if it is</span>
</span><span id="L-112"><a href="#L-112"><span class="linenos">112</span></a><span class="sd">        on the reverse strand.</span>
</span><span id="L-113"><a href="#L-113"><span class="linenos">113</span></a><span class="sd">        :return: a string that represents the strand of the given `self.tx_position`. If</span>
</span><span id="L-114"><a href="#L-114"><span class="linenos">114</span></a><span class="sd">        `self.tx_position.is_forward_strand` is `True`, then the string returned is &#39;+&#39;. Otherwise, the</span>
</span><span id="L-115"><a href="#L-115"><span class="linenos">115</span></a><span class="sd">        string returned is &#39;-&#39;.</span>
</span><span id="L-116"><a href="#L-116"><span class="linenos">116</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="L-117"><a href="#L-117"><span class="linenos">117</span></a>
</span><span id="L-118"><a href="#L-118"><span class="linenos">118</span></a>        <span class="k">return</span> <span class="s2">&quot;+&quot;</span> <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">is_forward_strand</span> <span class="k">else</span> <span class="s2">&quot;-&quot;</span>
</span><span id="L-119"><a href="#L-119"><span class="linenos">119</span></a>
</span><span id="L-120"><a href="#L-120"><span class="linenos">120</span></a>    <span class="nd">@lazy</span>
</span><span id="L-121"><a href="#L-121"><span class="linenos">121</span></a>    <span class="k">def</span> <span class="nf">ordered_cdna_match</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
</span><span id="L-122"><a href="#L-122"><span class="linenos">122</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-123"><a href="#L-123"><span class="linenos">123</span></a><span class="sd">        The function &quot;ordered_cdna_match&quot; sorts a list of cdna_match objects based on their</span>
</span><span id="L-124"><a href="#L-124"><span class="linenos">124</span></a><span class="sd">        tx_position.chrom_start attribute and returns the sorted list.</span>
</span><span id="L-125"><a href="#L-125"><span class="linenos">125</span></a><span class="sd">        :return: a sorted list of cdna_match objects.</span>
</span><span id="L-126"><a href="#L-126"><span class="linenos">126</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="L-127"><a href="#L-127"><span class="linenos">127</span></a>
</span><span id="L-128"><a href="#L-128"><span class="linenos">128</span></a>        <span class="n">transcript_strand</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">is_forward_strand</span>
</span><span id="L-129"><a href="#L-129"><span class="linenos">129</span></a>        <span class="n">cdna_match</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">cdna_match</span><span class="p">)</span>
</span><span id="L-130"><a href="#L-130"><span class="linenos">130</span></a>        <span class="n">cdna_match</span><span class="o">.</span><span class="n">sort</span><span class="p">(</span><span class="n">key</span><span class="o">=</span><span class="k">lambda</span> <span class="n">cm</span><span class="p">:</span> <span class="n">cm</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_start</span><span class="p">)</span>
</span><span id="L-131"><a href="#L-131"><span class="linenos">131</span></a>        <span class="k">if</span> <span class="ow">not</span> <span class="n">transcript_strand</span><span class="p">:</span>
</span><span id="L-132"><a href="#L-132"><span class="linenos">132</span></a>            <span class="n">cdna_match</span><span class="o">.</span><span class="n">reverse</span><span class="p">()</span>
</span><span id="L-133"><a href="#L-133"><span class="linenos">133</span></a>        <span class="k">return</span> <span class="n">cdna_match</span>
</span><span id="L-134"><a href="#L-134"><span class="linenos">134</span></a>
</span><span id="L-135"><a href="#L-135"><span class="linenos">135</span></a>    <span class="k">def</span> <span class="nf">get_cds_start_stop</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">tuple</span><span class="p">:</span>
</span><span id="L-136"><a href="#L-136"><span class="linenos">136</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-137"><a href="#L-137"><span class="linenos">137</span></a><span class="sd">        The function &quot;get_cds_start_stop&quot; returns the start and stop positions of a coding sequence,</span>
</span><span id="L-138"><a href="#L-138"><span class="linenos">138</span></a><span class="sd">        taking into account the direction of the strand.</span>
</span><span id="L-139"><a href="#L-139"><span class="linenos">139</span></a><span class="sd">        :return: a tuple containing the start and stop positions of the coding sequence (CDS).</span>
</span><span id="L-140"><a href="#L-140"><span class="linenos">140</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="L-141"><a href="#L-141"><span class="linenos">141</span></a>
</span><span id="L-142"><a href="#L-142"><span class="linenos">142</span></a>        <span class="p">(</span><span class="n">start</span><span class="p">,</span> <span class="n">stop</span><span class="p">)</span> <span class="o">=</span> <span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">cds_position</span><span class="o">.</span><span class="n">chrom_start</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">cds_position</span><span class="o">.</span><span class="n">chrom_stop</span><span class="p">)</span>
</span><span id="L-143"><a href="#L-143"><span class="linenos">143</span></a>        <span class="k">if</span> <span class="ow">not</span> <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">is_forward_strand</span><span class="p">:</span>
</span><span id="L-144"><a href="#L-144"><span class="linenos">144</span></a>            <span class="p">(</span><span class="n">stop</span><span class="p">,</span> <span class="n">start</span><span class="p">)</span> <span class="o">=</span> <span class="p">(</span><span class="n">start</span><span class="p">,</span> <span class="n">stop</span><span class="p">)</span>
</span><span id="L-145"><a href="#L-145"><span class="linenos">145</span></a>        <span class="k">return</span> <span class="n">start</span><span class="p">,</span> <span class="n">stop</span>
</span><span id="L-146"><a href="#L-146"><span class="linenos">146</span></a>
</span><span id="L-147"><a href="#L-147"><span class="linenos">147</span></a>    <span class="nd">@lazy</span>
</span><span id="L-148"><a href="#L-148"><span class="linenos">148</span></a>    <span class="k">def</span> <span class="nf">start_codon</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">int</span><span class="p">:</span>
</span><span id="L-149"><a href="#L-149"><span class="linenos">149</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-150"><a href="#L-150"><span class="linenos">150</span></a><span class="sd">        The function returns the transcript position of the start codon.</span>
</span><span id="L-151"><a href="#L-151"><span class="linenos">151</span></a><span class="sd">        :return: the transcript position of the start codon.</span>
</span><span id="L-152"><a href="#L-152"><span class="linenos">152</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="L-153"><a href="#L-153"><span class="linenos">153</span></a>
</span><span id="L-154"><a href="#L-154"><span class="linenos">154</span></a>        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">_start_codon_transcript_pos</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
</span><span id="L-155"><a href="#L-155"><span class="linenos">155</span></a>            <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">_start_codon_transcript_pos</span>
</span><span id="L-156"><a href="#L-156"><span class="linenos">156</span></a>
</span><span id="L-157"><a href="#L-157"><span class="linenos">157</span></a>        <span class="n">start_genomic_coordinate</span><span class="p">,</span> <span class="n">_</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">get_cds_start_stop</span><span class="p">()</span>
</span><span id="L-158"><a href="#L-158"><span class="linenos">158</span></a>        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">_get_transcript_position</span><span class="p">(</span><span class="n">start_genomic_coordinate</span><span class="p">)</span>
</span><span id="L-159"><a href="#L-159"><span class="linenos">159</span></a>
</span><span id="L-160"><a href="#L-160"><span class="linenos">160</span></a>    <span class="nd">@lazy</span>
</span><span id="L-161"><a href="#L-161"><span class="linenos">161</span></a>    <span class="k">def</span> <span class="nf">stop_codon</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">int</span><span class="p">:</span>
</span><span id="L-162"><a href="#L-162"><span class="linenos">162</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-163"><a href="#L-163"><span class="linenos">163</span></a><span class="sd">        The function returns the transcript position of the stop codon.</span>
</span><span id="L-164"><a href="#L-164"><span class="linenos">164</span></a><span class="sd">        :return: The method `stop_codon` returns an integer, which represents the transcript position of</span>
</span><span id="L-165"><a href="#L-165"><span class="linenos">165</span></a><span class="sd">        the stop codon.</span>
</span><span id="L-166"><a href="#L-166"><span class="linenos">166</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="L-167"><a href="#L-167"><span class="linenos">167</span></a>
</span><span id="L-168"><a href="#L-168"><span class="linenos">168</span></a>        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">_stop_codon_transcript_pos</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
</span><span id="L-169"><a href="#L-169"><span class="linenos">169</span></a>            <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">_stop_codon_transcript_pos</span>
</span><span id="L-170"><a href="#L-170"><span class="linenos">170</span></a>
</span><span id="L-171"><a href="#L-171"><span class="linenos">171</span></a>        <span class="n">_</span><span class="p">,</span> <span class="n">stop_genomic_coordinate</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">get_cds_start_stop</span><span class="p">()</span>
</span><span id="L-172"><a href="#L-172"><span class="linenos">172</span></a>        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">_get_transcript_position</span><span class="p">(</span><span class="n">stop_genomic_coordinate</span><span class="p">)</span>
</span><span id="L-173"><a href="#L-173"><span class="linenos">173</span></a>
</span><span id="L-174"><a href="#L-174"><span class="linenos">174</span></a>    <span class="k">def</span> <span class="nf">_get_transcript_position</span><span class="p">(</span>
</span><span id="L-175"><a href="#L-175"><span class="linenos">175</span></a>        <span class="bp">self</span><span class="p">,</span> <span class="n">genomic_coordinate</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span> <span class="n">label</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="kc">None</span>
</span><span id="L-176"><a href="#L-176"><span class="linenos">176</span></a>    <span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">int</span><span class="p">:</span>
</span><span id="L-177"><a href="#L-177"><span class="linenos">177</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-178"><a href="#L-178"><span class="linenos">178</span></a><span class="sd">        The function `_get_transcript_position` calculates the position of a genomic coordinate within a</span>
</span><span id="L-179"><a href="#L-179"><span class="linenos">179</span></a><span class="sd">        transcript.</span>
</span><span id="L-180"><a href="#L-180"><span class="linenos">180</span></a>
</span><span id="L-181"><a href="#L-181"><span class="linenos">181</span></a><span class="sd">        :param genomic_coordinate: The `genomic_coordinate` parameter is an integer representing a</span>
</span><span id="L-182"><a href="#L-182"><span class="linenos">182</span></a><span class="sd">        specific position on the genome. It is the coordinate for which we want to determine the</span>
</span><span id="L-183"><a href="#L-183"><span class="linenos">183</span></a><span class="sd">        corresponding position in the transcript</span>
</span><span id="L-184"><a href="#L-184"><span class="linenos">184</span></a><span class="sd">        :type genomic_coordinate: int</span>
</span><span id="L-185"><a href="#L-185"><span class="linenos">185</span></a><span class="sd">        :param label: The `label` parameter is an optional string that represents a label or identifier</span>
</span><span id="L-186"><a href="#L-186"><span class="linenos">186</span></a><span class="sd">        for the genomic coordinate. It is used in the error message if the genomic coordinate is not</span>
</span><span id="L-187"><a href="#L-187"><span class="linenos">187</span></a><span class="sd">        found in any of the exons. If no label is provided, a default label is generated using the value</span>
</span><span id="L-188"><a href="#L-188"><span class="linenos">188</span></a><span class="sd">        of the `genomic</span>
</span><span id="L-189"><a href="#L-189"><span class="linenos">189</span></a><span class="sd">        :type label: str</span>
</span><span id="L-190"><a href="#L-190"><span class="linenos">190</span></a><span class="sd">        :return: an integer representing the position of the genomic coordinate within the transcript.</span>
</span><span id="L-191"><a href="#L-191"><span class="linenos">191</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="L-192"><a href="#L-192"><span class="linenos">192</span></a>
</span><span id="L-193"><a href="#L-193"><span class="linenos">193</span></a>        <span class="n">transcript_strand</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">is_forward_strand</span>
</span><span id="L-194"><a href="#L-194"><span class="linenos">194</span></a>
</span><span id="L-195"><a href="#L-195"><span class="linenos">195</span></a>        <span class="n">cdna_offset</span> <span class="o">=</span> <span class="mi">0</span>
</span><span id="L-196"><a href="#L-196"><span class="linenos">196</span></a>        <span class="k">for</span> <span class="n">cdna_match</span> <span class="ow">in</span> <span class="bp">self</span><span class="o">.</span><span class="n">ordered_cdna_match</span><span class="p">:</span>
</span><span id="L-197"><a href="#L-197"><span class="linenos">197</span></a>            <span class="n">start</span> <span class="o">=</span> <span class="n">cdna_match</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_start</span>
</span><span id="L-198"><a href="#L-198"><span class="linenos">198</span></a>            <span class="n">stop</span> <span class="o">=</span> <span class="n">cdna_match</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_stop</span>
</span><span id="L-199"><a href="#L-199"><span class="linenos">199</span></a>
</span><span id="L-200"><a href="#L-200"><span class="linenos">200</span></a>            <span class="k">if</span> <span class="n">start</span> <span class="o">&lt;=</span> <span class="n">genomic_coordinate</span> <span class="o">&lt;=</span> <span class="n">stop</span><span class="p">:</span>
</span><span id="L-201"><a href="#L-201"><span class="linenos">201</span></a>                <span class="c1"># We&#39;re inside this match</span>
</span><span id="L-202"><a href="#L-202"><span class="linenos">202</span></a>                <span class="k">if</span> <span class="n">transcript_strand</span><span class="p">:</span>
</span><span id="L-203"><a href="#L-203"><span class="linenos">203</span></a>                    <span class="n">position</span> <span class="o">=</span> <span class="n">genomic_coordinate</span> <span class="o">-</span> <span class="n">start</span>
</span><span id="L-204"><a href="#L-204"><span class="linenos">204</span></a>                <span class="k">else</span><span class="p">:</span>
</span><span id="L-205"><a href="#L-205"><span class="linenos">205</span></a>                    <span class="n">position</span> <span class="o">=</span> <span class="n">stop</span> <span class="o">-</span> <span class="n">genomic_coordinate</span>
</span><span id="L-206"><a href="#L-206"><span class="linenos">206</span></a>                <span class="k">return</span> <span class="n">cdna_offset</span> <span class="o">+</span> <span class="n">position</span> <span class="o">+</span> <span class="n">cdna_match</span><span class="o">.</span><span class="n">get_offset</span><span class="p">(</span><span class="n">position</span><span class="p">)</span>
</span><span id="L-207"><a href="#L-207"><span class="linenos">207</span></a>            <span class="k">else</span><span class="p">:</span>
</span><span id="L-208"><a href="#L-208"><span class="linenos">208</span></a>                <span class="n">cdna_offset</span> <span class="o">+=</span> <span class="n">cdna_match</span><span class="o">.</span><span class="n">length</span>
</span><span id="L-209"><a href="#L-209"><span class="linenos">209</span></a>        <span class="k">if</span> <span class="n">label</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">:</span>
</span><span id="L-210"><a href="#L-210"><span class="linenos">210</span></a>            <span class="n">label</span> <span class="o">=</span> <span class="s2">&quot;Genomic coordinate: </span><span class="si">%d</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="n">genomic_coordinate</span>
</span><span id="L-211"><a href="#L-211"><span class="linenos">211</span></a>        <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="s2">&quot;</span><span class="si">%s</span><span class="s2"> is not in any of the exons&quot;</span> <span class="o">%</span> <span class="n">label</span><span class="p">)</span>
</span><span id="L-212"><a href="#L-212"><span class="linenos">212</span></a>
</span><span id="L-213"><a href="#L-213"><span class="linenos">213</span></a>    <span class="k">def</span> <span class="nf">cdna_to_genomic_coord</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">coord</span><span class="p">:</span> <span class="nb">object</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">int</span><span class="p">:</span>
</span><span id="L-214"><a href="#L-214"><span class="linenos">214</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-215"><a href="#L-215"><span class="linenos">215</span></a><span class="sd">        The function `cdna_to_genomic_coord` converts a HGVS cDNA coordinate to a genomic coordinate.</span>
</span><span id="L-216"><a href="#L-216"><span class="linenos">216</span></a>
</span><span id="L-217"><a href="#L-217"><span class="linenos">217</span></a><span class="sd">        :param coord: The parameter `coord` is an object that represents a cDNA coordinate. It is used</span>
</span><span id="L-218"><a href="#L-218"><span class="linenos">218</span></a><span class="sd">        to specify a position along a cDNA sequence</span>
</span><span id="L-219"><a href="#L-219"><span class="linenos">219</span></a><span class="sd">        :type coord: object</span>
</span><span id="L-220"><a href="#L-220"><span class="linenos">220</span></a><span class="sd">        :return: an integer value, which represents the genomic coordinate corresponding to the given</span>
</span><span id="L-221"><a href="#L-221"><span class="linenos">221</span></a><span class="sd">        cDNA coordinate.</span>
</span><span id="L-222"><a href="#L-222"><span class="linenos">222</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="L-223"><a href="#L-223"><span class="linenos">223</span></a>
</span><span id="L-224"><a href="#L-224"><span class="linenos">224</span></a>        <span class="n">transcript_strand</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">is_forward_strand</span>
</span><span id="L-225"><a href="#L-225"><span class="linenos">225</span></a>
</span><span id="L-226"><a href="#L-226"><span class="linenos">226</span></a>        <span class="c1"># compute starting position along spliced transcript.</span>
</span><span id="L-227"><a href="#L-227"><span class="linenos">227</span></a>        <span class="k">if</span> <span class="n">coord</span><span class="o">.</span><span class="n">landmark</span> <span class="o">==</span> <span class="n">CDNA_START_CODON</span><span class="p">:</span>
</span><span id="L-228"><a href="#L-228"><span class="linenos">228</span></a>            <span class="n">utr5p</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">start_codon</span> <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">is_coding</span> <span class="k">else</span> <span class="mi">0</span>
</span><span id="L-229"><a href="#L-229"><span class="linenos">229</span></a>
</span><span id="L-230"><a href="#L-230"><span class="linenos">230</span></a>            <span class="k">if</span> <span class="n">coord</span><span class="o">.</span><span class="n">coord</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">:</span>
</span><span id="L-231"><a href="#L-231"><span class="linenos">231</span></a>                <span class="n">cdna_pos</span> <span class="o">=</span> <span class="n">utr5p</span> <span class="o">+</span> <span class="n">coord</span><span class="o">.</span><span class="n">coord</span>
</span><span id="L-232"><a href="#L-232"><span class="linenos">232</span></a>            <span class="k">else</span><span class="p">:</span>
</span><span id="L-233"><a href="#L-233"><span class="linenos">233</span></a>                <span class="n">cdna_pos</span> <span class="o">=</span> <span class="n">utr5p</span> <span class="o">+</span> <span class="n">coord</span><span class="o">.</span><span class="n">coord</span> <span class="o">+</span> <span class="mi">1</span>
</span><span id="L-234"><a href="#L-234"><span class="linenos">234</span></a>        <span class="k">elif</span> <span class="n">coord</span><span class="o">.</span><span class="n">landmark</span> <span class="o">==</span> <span class="n">CDNA_STOP_CODON</span><span class="p">:</span>
</span><span id="L-235"><a href="#L-235"><span class="linenos">235</span></a>            <span class="k">if</span> <span class="n">coord</span><span class="o">.</span><span class="n">coord</span> <span class="o">&lt;</span> <span class="mi">0</span><span class="p">:</span>
</span><span id="L-236"><a href="#L-236"><span class="linenos">236</span></a>                <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span>
</span><span id="L-237"><a href="#L-237"><span class="linenos">237</span></a>                    <span class="s2">&quot;CDNACoord cannot have a negative coord and &quot;</span>
</span><span id="L-238"><a href="#L-238"><span class="linenos">238</span></a>                    <span class="s2">&quot;landmark CDNA_STOP_CODON&quot;</span>
</span><span id="L-239"><a href="#L-239"><span class="linenos">239</span></a>                <span class="p">)</span>
</span><span id="L-240"><a href="#L-240"><span class="linenos">240</span></a>            <span class="n">cdna_pos</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">stop_codon</span> <span class="o">+</span> <span class="n">coord</span><span class="o">.</span><span class="n">coord</span>
</span><span id="L-241"><a href="#L-241"><span class="linenos">241</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="L-242"><a href="#L-242"><span class="linenos">242</span></a>            <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="s1">&#39;unknown CDNACoord landmark &quot;</span><span class="si">%s</span><span class="s1">&quot;&#39;</span> <span class="o">%</span> <span class="n">coord</span><span class="o">.</span><span class="n">landmark</span><span class="p">)</span>
</span><span id="L-243"><a href="#L-243"><span class="linenos">243</span></a>
</span><span id="L-244"><a href="#L-244"><span class="linenos">244</span></a>        <span class="c1"># 5&#39; flanking sequence (no need to account for gaps)</span>
</span><span id="L-245"><a href="#L-245"><span class="linenos">245</span></a>        <span class="k">if</span> <span class="n">cdna_pos</span> <span class="o">&lt;</span> <span class="mi">1</span><span class="p">:</span>
</span><span id="L-246"><a href="#L-246"><span class="linenos">246</span></a>            <span class="k">if</span> <span class="n">transcript_strand</span><span class="p">:</span>
</span><span id="L-247"><a href="#L-247"><span class="linenos">247</span></a>                <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_start</span> <span class="o">+</span> <span class="n">cdna_pos</span>
</span><span id="L-248"><a href="#L-248"><span class="linenos">248</span></a>            <span class="k">else</span><span class="p">:</span>
</span><span id="L-249"><a href="#L-249"><span class="linenos">249</span></a>                <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_stop</span> <span class="o">-</span> <span class="n">cdna_pos</span> <span class="o">+</span> <span class="mi">1</span>
</span><span id="L-250"><a href="#L-250"><span class="linenos">250</span></a>
</span><span id="L-251"><a href="#L-251"><span class="linenos">251</span></a>        <span class="c1"># Walk along transcript until we find an exon that contains cdna_pos.</span>
</span><span id="L-252"><a href="#L-252"><span class="linenos">252</span></a>        <span class="k">for</span> <span class="n">cdna_match</span> <span class="ow">in</span> <span class="bp">self</span><span class="o">.</span><span class="n">ordered_cdna_match</span><span class="p">:</span>
</span><span id="L-253"><a href="#L-253"><span class="linenos">253</span></a>            <span class="k">if</span> <span class="n">cdna_match</span><span class="o">.</span><span class="n">cdna_start</span> <span class="o">&lt;=</span> <span class="n">cdna_pos</span> <span class="o">&lt;=</span> <span class="n">cdna_match</span><span class="o">.</span><span class="n">cdna_end</span><span class="p">:</span>
</span><span id="L-254"><a href="#L-254"><span class="linenos">254</span></a>                <span class="n">match_pos</span> <span class="o">=</span> <span class="n">cdna_pos</span> <span class="o">-</span> <span class="n">cdna_match</span><span class="o">.</span><span class="n">cdna_start</span>
</span><span id="L-255"><a href="#L-255"><span class="linenos">255</span></a>                <span class="n">match_pos</span> <span class="o">-=</span> <span class="n">cdna_match</span><span class="o">.</span><span class="n">get_offset</span><span class="p">(</span><span class="n">match_pos</span><span class="p">)</span>
</span><span id="L-256"><a href="#L-256"><span class="linenos">256</span></a>                <span class="c1"># Compute genomic coordinate using offset.</span>
</span><span id="L-257"><a href="#L-257"><span class="linenos">257</span></a>                <span class="k">if</span> <span class="n">transcript_strand</span><span class="p">:</span>
</span><span id="L-258"><a href="#L-258"><span class="linenos">258</span></a>                    <span class="c1"># Plus strand.</span>
</span><span id="L-259"><a href="#L-259"><span class="linenos">259</span></a>                    <span class="n">start</span> <span class="o">=</span> <span class="n">cdna_match</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_start</span> <span class="o">+</span> <span class="mi">1</span>
</span><span id="L-260"><a href="#L-260"><span class="linenos">260</span></a>                    <span class="k">return</span> <span class="n">start</span> <span class="o">+</span> <span class="n">match_pos</span> <span class="o">+</span> <span class="n">coord</span><span class="o">.</span><span class="n">offset</span>
</span><span id="L-261"><a href="#L-261"><span class="linenos">261</span></a>                <span class="k">else</span><span class="p">:</span>
</span><span id="L-262"><a href="#L-262"><span class="linenos">262</span></a>                    <span class="c1"># Minus strand.</span>
</span><span id="L-263"><a href="#L-263"><span class="linenos">263</span></a>                    <span class="n">end</span> <span class="o">=</span> <span class="n">cdna_match</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_stop</span>
</span><span id="L-264"><a href="#L-264"><span class="linenos">264</span></a>                    <span class="k">return</span> <span class="n">end</span> <span class="o">-</span> <span class="n">match_pos</span> <span class="o">-</span> <span class="n">coord</span><span class="o">.</span><span class="n">offset</span>
</span><span id="L-265"><a href="#L-265"><span class="linenos">265</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="L-266"><a href="#L-266"><span class="linenos">266</span></a>            <span class="c1"># 3&#39; flanking sequence (no need to account for gaps)</span>
</span><span id="L-267"><a href="#L-267"><span class="linenos">267</span></a>            <span class="k">if</span> <span class="n">transcript_strand</span><span class="p">:</span>
</span><span id="L-268"><a href="#L-268"><span class="linenos">268</span></a>                <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">cds_position</span><span class="o">.</span><span class="n">chrom_stop</span> <span class="o">+</span> <span class="n">coord</span><span class="o">.</span><span class="n">coord</span>
</span><span id="L-269"><a href="#L-269"><span class="linenos">269</span></a>            <span class="k">else</span><span class="p">:</span>
</span><span id="L-270"><a href="#L-270"><span class="linenos">270</span></a>                <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">cds_position</span><span class="o">.</span><span class="n">chrom_start</span> <span class="o">+</span> <span class="mi">1</span> <span class="o">-</span> <span class="n">coord</span><span class="o">.</span><span class="n">coord</span>
</span><span id="L-271"><a href="#L-271"><span class="linenos">271</span></a>
</span><span id="L-272"><a href="#L-272"><span class="linenos">272</span></a>    <span class="k">def</span> <span class="nf">genomic_to_cdna_coord</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">genomic_coord</span><span class="p">:</span> <span class="nb">int</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">object</span><span class="p">:</span>
</span><span id="L-273"><a href="#L-273"><span class="linenos">273</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-274"><a href="#L-274"><span class="linenos">274</span></a><span class="sd">        The function `genomic_to_cdna_coord` converts a genomic coordinate to a cDNA coordinate and</span>
</span><span id="L-275"><a href="#L-275"><span class="linenos">275</span></a><span class="sd">        offset, taking into account exons, strand, and coding transcript information.</span>
</span><span id="L-276"><a href="#L-276"><span class="linenos">276</span></a>
</span><span id="L-277"><a href="#L-277"><span class="linenos">277</span></a><span class="sd">        :param genomic_coord: The `genomic_coord` parameter is an integer representing a genomic</span>
</span><span id="L-278"><a href="#L-278"><span class="linenos">278</span></a><span class="sd">        coordinate</span>
</span><span id="L-279"><a href="#L-279"><span class="linenos">279</span></a><span class="sd">        :type genomic_coord: int</span>
</span><span id="L-280"><a href="#L-280"><span class="linenos">280</span></a><span class="sd">        :return: an object of type `CDNACoord`.</span>
</span><span id="L-281"><a href="#L-281"><span class="linenos">281</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="L-282"><a href="#L-282"><span class="linenos">282</span></a>
</span><span id="L-283"><a href="#L-283"><span class="linenos">283</span></a>        <span class="n">exons</span> <span class="o">=</span> <span class="p">[</span><span class="n">exon</span><span class="o">.</span><span class="n">get_as_interval</span><span class="p">()</span> <span class="k">for</span> <span class="n">exon</span> <span class="ow">in</span> <span class="bp">self</span><span class="o">.</span><span class="n">ordered_cdna_match</span><span class="p">]</span>
</span><span id="L-284"><a href="#L-284"><span class="linenos">284</span></a>
</span><span id="L-285"><a href="#L-285"><span class="linenos">285</span></a>        <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">exons</span><span class="p">)</span> <span class="o">==</span> <span class="mi">0</span><span class="p">:</span>
</span><span id="L-286"><a href="#L-286"><span class="linenos">286</span></a>            <span class="k">return</span> <span class="kc">None</span>
</span><span id="L-287"><a href="#L-287"><span class="linenos">287</span></a>
</span><span id="L-288"><a href="#L-288"><span class="linenos">288</span></a>        <span class="n">strand</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">strand</span>
</span><span id="L-289"><a href="#L-289"><span class="linenos">289</span></a>        <span class="n">distances</span> <span class="o">=</span> <span class="p">[</span><span class="n">exon</span><span class="o">.</span><span class="n">distance</span><span class="p">(</span><span class="n">genomic_coord</span><span class="p">)</span> <span class="k">for</span> <span class="n">exon</span> <span class="ow">in</span> <span class="n">exons</span><span class="p">]</span>
</span><span id="L-290"><a href="#L-290"><span class="linenos">290</span></a>        <span class="n">min_distance_to_exon</span> <span class="o">=</span> <span class="nb">min</span><span class="p">(</span><span class="nb">map</span><span class="p">(</span><span class="nb">abs</span><span class="p">,</span> <span class="n">distances</span><span class="p">))</span>
</span><span id="L-291"><a href="#L-291"><span class="linenos">291</span></a>        <span class="k">if</span> <span class="n">min_distance_to_exon</span><span class="p">:</span>
</span><span id="L-292"><a href="#L-292"><span class="linenos">292</span></a>            <span class="c1"># We&#39;re outside of exon - so need to find closest point</span>
</span><span id="L-293"><a href="#L-293"><span class="linenos">293</span></a>            <span class="k">for</span> <span class="n">exon</span> <span class="ow">in</span> <span class="n">exons</span><span class="p">:</span>
</span><span id="L-294"><a href="#L-294"><span class="linenos">294</span></a>                <span class="n">distance</span> <span class="o">=</span> <span class="n">exon</span><span class="o">.</span><span class="n">distance</span><span class="p">(</span><span class="n">genomic_coord</span><span class="p">)</span>
</span><span id="L-295"><a href="#L-295"><span class="linenos">295</span></a>                <span class="k">if</span> <span class="nb">abs</span><span class="p">(</span><span class="n">distance</span><span class="p">)</span> <span class="o">==</span> <span class="n">min_distance_to_exon</span><span class="p">:</span>
</span><span id="L-296"><a href="#L-296"><span class="linenos">296</span></a>
</span><span id="L-297"><a href="#L-297"><span class="linenos">297</span></a>                    <span class="c1"># Outside the exon.</span>
</span><span id="L-298"><a href="#L-298"><span class="linenos">298</span></a>                    <span class="k">if</span> <span class="n">distance</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">:</span>
</span><span id="L-299"><a href="#L-299"><span class="linenos">299</span></a>                        <span class="n">genomic_nearest_exon</span> <span class="o">=</span> <span class="n">exon</span><span class="o">.</span><span class="n">chrom_start</span> <span class="o">+</span> <span class="mi">1</span>
</span><span id="L-300"><a href="#L-300"><span class="linenos">300</span></a>                    <span class="k">else</span><span class="p">:</span>
</span><span id="L-301"><a href="#L-301"><span class="linenos">301</span></a>                        <span class="n">genomic_nearest_exon</span> <span class="o">=</span> <span class="n">exon</span><span class="o">.</span><span class="n">chrom_end</span>
</span><span id="L-302"><a href="#L-302"><span class="linenos">302</span></a>
</span><span id="L-303"><a href="#L-303"><span class="linenos">303</span></a>                    <span class="k">if</span> <span class="n">strand</span> <span class="o">==</span> <span class="s2">&quot;+&quot;</span><span class="p">:</span>
</span><span id="L-304"><a href="#L-304"><span class="linenos">304</span></a>                        <span class="n">distance</span> <span class="o">*=</span> <span class="o">-</span><span class="mi">1</span>
</span><span id="L-305"><a href="#L-305"><span class="linenos">305</span></a>
</span><span id="L-306"><a href="#L-306"><span class="linenos">306</span></a>                    <span class="n">nearest_exon_coord</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_exon_genomic_to_cdna_coord</span><span class="p">(</span>
</span><span id="L-307"><a href="#L-307"><span class="linenos">307</span></a>                        <span class="n">genomic_nearest_exon</span>
</span><span id="L-308"><a href="#L-308"><span class="linenos">308</span></a>                    <span class="p">)</span>
</span><span id="L-309"><a href="#L-309"><span class="linenos">309</span></a>
</span><span id="L-310"><a href="#L-310"><span class="linenos">310</span></a>                    <span class="c1"># If outside transcript, don&#39;t use offset.</span>
</span><span id="L-311"><a href="#L-311"><span class="linenos">311</span></a>                    <span class="k">if</span> <span class="p">(</span>
</span><span id="L-312"><a href="#L-312"><span class="linenos">312</span></a>                        <span class="n">genomic_coord</span> <span class="o">&lt;</span> <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_start</span> <span class="o">+</span> <span class="mi">1</span>
</span><span id="L-313"><a href="#L-313"><span class="linenos">313</span></a>                        <span class="ow">or</span> <span class="n">genomic_coord</span> <span class="o">&gt;</span> <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_stop</span>
</span><span id="L-314"><a href="#L-314"><span class="linenos">314</span></a>                    <span class="p">):</span>
</span><span id="L-315"><a href="#L-315"><span class="linenos">315</span></a>                        <span class="n">nearest_exon_coord</span> <span class="o">+=</span> <span class="n">distance</span>
</span><span id="L-316"><a href="#L-316"><span class="linenos">316</span></a>                        <span class="n">distance</span> <span class="o">=</span> <span class="mi">0</span>
</span><span id="L-317"><a href="#L-317"><span class="linenos">317</span></a>                    <span class="n">cdna_coord</span> <span class="o">=</span> <span class="n">CDNACoord</span><span class="p">(</span><span class="n">nearest_exon_coord</span><span class="p">,</span> <span class="n">distance</span><span class="p">)</span>
</span><span id="L-318"><a href="#L-318"><span class="linenos">318</span></a>                    <span class="k">break</span>
</span><span id="L-319"><a href="#L-319"><span class="linenos">319</span></a>            <span class="k">else</span><span class="p">:</span>
</span><span id="L-320"><a href="#L-320"><span class="linenos">320</span></a>                <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="s2">&quot;Could not find closest exon!&quot;</span><span class="p">)</span>  <span class="c1"># Should never happen</span>
</span><span id="L-321"><a href="#L-321"><span class="linenos">321</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="L-322"><a href="#L-322"><span class="linenos">322</span></a>            <span class="n">coord</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_exon_genomic_to_cdna_coord</span><span class="p">(</span><span class="n">genomic_coord</span><span class="p">)</span>
</span><span id="L-323"><a href="#L-323"><span class="linenos">323</span></a>            <span class="n">cdna_coord</span> <span class="o">=</span> <span class="n">CDNACoord</span><span class="p">(</span><span class="n">coord</span><span class="p">,</span> <span class="mi">0</span><span class="p">)</span>
</span><span id="L-324"><a href="#L-324"><span class="linenos">324</span></a>
</span><span id="L-325"><a href="#L-325"><span class="linenos">325</span></a>        <span class="c1"># Adjust coordinates for coding transcript.</span>
</span><span id="L-326"><a href="#L-326"><span class="linenos">326</span></a>        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">is_coding</span><span class="p">:</span>
</span><span id="L-327"><a href="#L-327"><span class="linenos">327</span></a>            <span class="c1"># Detect if position before start codon.</span>
</span><span id="L-328"><a href="#L-328"><span class="linenos">328</span></a>            <span class="n">utr5p</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">start_codon</span>
</span><span id="L-329"><a href="#L-329"><span class="linenos">329</span></a>            <span class="n">cdna_coord</span><span class="o">.</span><span class="n">coord</span> <span class="o">-=</span> <span class="n">utr5p</span>
</span><span id="L-330"><a href="#L-330"><span class="linenos">330</span></a>            <span class="k">if</span> <span class="n">cdna_coord</span><span class="o">.</span><span class="n">coord</span> <span class="o">&lt;=</span> <span class="mi">0</span><span class="p">:</span>
</span><span id="L-331"><a href="#L-331"><span class="linenos">331</span></a>                <span class="n">cdna_coord</span><span class="o">.</span><span class="n">coord</span> <span class="o">-=</span> <span class="mi">1</span>
</span><span id="L-332"><a href="#L-332"><span class="linenos">332</span></a>            <span class="k">else</span><span class="p">:</span>
</span><span id="L-333"><a href="#L-333"><span class="linenos">333</span></a>                <span class="c1"># Detect if position is after stop_codon.</span>
</span><span id="L-334"><a href="#L-334"><span class="linenos">334</span></a>                <span class="n">stop_codon</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">stop_codon</span>
</span><span id="L-335"><a href="#L-335"><span class="linenos">335</span></a>                <span class="n">stop_codon</span> <span class="o">-=</span> <span class="n">utr5p</span>
</span><span id="L-336"><a href="#L-336"><span class="linenos">336</span></a>                <span class="k">if</span> <span class="p">(</span>
</span><span id="L-337"><a href="#L-337"><span class="linenos">337</span></a>                    <span class="n">cdna_coord</span><span class="o">.</span><span class="n">coord</span> <span class="o">&gt;</span> <span class="n">stop_codon</span>
</span><span id="L-338"><a href="#L-338"><span class="linenos">338</span></a>                    <span class="ow">or</span> <span class="n">cdna_coord</span><span class="o">.</span><span class="n">coord</span> <span class="o">==</span> <span class="n">stop_codon</span>
</span><span id="L-339"><a href="#L-339"><span class="linenos">339</span></a>                    <span class="ow">and</span> <span class="n">cdna_coord</span><span class="o">.</span><span class="n">offset</span> <span class="o">&gt;</span> <span class="mi">0</span>
</span><span id="L-340"><a href="#L-340"><span class="linenos">340</span></a>                <span class="p">):</span>
</span><span id="L-341"><a href="#L-341"><span class="linenos">341</span></a>                    <span class="n">cdna_coord</span><span class="o">.</span><span class="n">coord</span> <span class="o">-=</span> <span class="n">stop_codon</span>
</span><span id="L-342"><a href="#L-342"><span class="linenos">342</span></a>                    <span class="n">cdna_coord</span><span class="o">.</span><span class="n">landmark</span> <span class="o">=</span> <span class="n">CDNA_STOP_CODON</span>
</span><span id="L-343"><a href="#L-343"><span class="linenos">343</span></a>
</span><span id="L-344"><a href="#L-344"><span class="linenos">344</span></a>        <span class="k">return</span> <span class="n">cdna_coord</span>
</span><span id="L-345"><a href="#L-345"><span class="linenos">345</span></a>
</span><span id="L-346"><a href="#L-346"><span class="linenos">346</span></a>    <span class="k">def</span> <span class="nf">_exon_genomic_to_cdna_coord</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">genomic_coord</span><span class="p">:</span> <span class="nb">int</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">int</span><span class="p">:</span>
</span><span id="L-347"><a href="#L-347"><span class="linenos">347</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-348"><a href="#L-348"><span class="linenos">348</span></a><span class="sd">        The function `_exon_genomic_to_cdna_coord` converts a genomic coordinate to a cDNA coordinate</span>
</span><span id="L-349"><a href="#L-349"><span class="linenos">349</span></a><span class="sd">        within an exon.</span>
</span><span id="L-350"><a href="#L-350"><span class="linenos">350</span></a>
</span><span id="L-351"><a href="#L-351"><span class="linenos">351</span></a><span class="sd">        :param genomic_coord: The `genomic_coord` parameter represents the genomic coordinate that you</span>
</span><span id="L-352"><a href="#L-352"><span class="linenos">352</span></a><span class="sd">        want to convert to cDNA coordinate. It is an integer value that specifies the position on the</span>
</span><span id="L-353"><a href="#L-353"><span class="linenos">353</span></a><span class="sd">        genome</span>
</span><span id="L-354"><a href="#L-354"><span class="linenos">354</span></a><span class="sd">        :type genomic_coord: int</span>
</span><span id="L-355"><a href="#L-355"><span class="linenos">355</span></a><span class="sd">        :return: the cDNA coordinate corresponding to the given genomic coordinate.</span>
</span><span id="L-356"><a href="#L-356"><span class="linenos">356</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="L-357"><a href="#L-357"><span class="linenos">357</span></a>
</span><span id="L-358"><a href="#L-358"><span class="linenos">358</span></a>        <span class="n">cdna_offset</span> <span class="o">=</span> <span class="mi">0</span>
</span><span id="L-359"><a href="#L-359"><span class="linenos">359</span></a>        <span class="k">for</span> <span class="n">cdna_match</span> <span class="ow">in</span> <span class="bp">self</span><span class="o">.</span><span class="n">ordered_cdna_match</span><span class="p">:</span>
</span><span id="L-360"><a href="#L-360"><span class="linenos">360</span></a>            <span class="c1"># Inside the exon.</span>
</span><span id="L-361"><a href="#L-361"><span class="linenos">361</span></a>            <span class="k">if</span> <span class="p">(</span>
</span><span id="L-362"><a href="#L-362"><span class="linenos">362</span></a>                <span class="n">cdna_match</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_start</span>
</span><span id="L-363"><a href="#L-363"><span class="linenos">363</span></a>                <span class="o">&lt;=</span> <span class="n">genomic_coord</span>
</span><span id="L-364"><a href="#L-364"><span class="linenos">364</span></a>                <span class="o">&lt;=</span> <span class="n">cdna_match</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_stop</span>
</span><span id="L-365"><a href="#L-365"><span class="linenos">365</span></a>            <span class="p">):</span>
</span><span id="L-366"><a href="#L-366"><span class="linenos">366</span></a>                <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">strand</span> <span class="o">==</span> <span class="s2">&quot;+&quot;</span><span class="p">:</span>
</span><span id="L-367"><a href="#L-367"><span class="linenos">367</span></a>                    <span class="n">position</span> <span class="o">=</span> <span class="n">genomic_coord</span> <span class="o">-</span> <span class="p">(</span><span class="n">cdna_match</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_start</span> <span class="o">+</span> <span class="mi">1</span><span class="p">)</span>
</span><span id="L-368"><a href="#L-368"><span class="linenos">368</span></a>                <span class="k">else</span><span class="p">:</span>
</span><span id="L-369"><a href="#L-369"><span class="linenos">369</span></a>                    <span class="n">position</span> <span class="o">=</span> <span class="n">cdna_match</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_stop</span> <span class="o">-</span> <span class="n">genomic_coord</span>
</span><span id="L-370"><a href="#L-370"><span class="linenos">370</span></a>                <span class="n">offset</span> <span class="o">=</span> <span class="n">cdna_match</span><span class="o">.</span><span class="n">get_offset</span><span class="p">(</span><span class="n">position</span><span class="p">,</span> <span class="n">validate</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
</span><span id="L-371"><a href="#L-371"><span class="linenos">371</span></a>                <span class="k">return</span> <span class="n">cdna_offset</span> <span class="o">+</span> <span class="n">position</span> <span class="o">+</span> <span class="n">offset</span> <span class="o">+</span> <span class="mi">1</span>
</span><span id="L-372"><a href="#L-372"><span class="linenos">372</span></a>            <span class="n">cdna_offset</span> <span class="o">+=</span> <span class="n">cdna_match</span><span class="o">.</span><span class="n">length</span>
</span><span id="L-373"><a href="#L-373"><span class="linenos">373</span></a>
</span><span id="L-374"><a href="#L-374"><span class="linenos">374</span></a>        <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Couldn&#39;t find </span><span class="si">{</span><span class="n">genomic_coord</span><span class="si">=}</span><span class="s2">!&quot;</span><span class="p">)</span>
</span><span id="L-375"><a href="#L-375"><span class="linenos">375</span></a>
</span><span id="L-376"><a href="#L-376"><span class="linenos">376</span></a>    <span class="k">def</span> <span class="nf">find_exon_number</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">offset</span><span class="p">:</span> <span class="nb">int</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">int</span><span class="p">:</span>
</span><span id="L-377"><a href="#L-377"><span class="linenos">377</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-378"><a href="#L-378"><span class="linenos">378</span></a><span class="sd">        The function `find_exon_number` returns the exon number for a given position.</span>
</span><span id="L-379"><a href="#L-379"><span class="linenos">379</span></a>
</span><span id="L-380"><a href="#L-380"><span class="linenos">380</span></a><span class="sd">        :param offset: The offset parameter represents a position in the genome. It is an integer value</span>
</span><span id="L-381"><a href="#L-381"><span class="linenos">381</span></a><span class="sd">        that indicates the position of interest within the genome</span>
</span><span id="L-382"><a href="#L-382"><span class="linenos">382</span></a><span class="sd">        :type offset: int</span>
</span><span id="L-383"><a href="#L-383"><span class="linenos">383</span></a><span class="sd">        :return: an integer value, which represents the exon number for a given position.</span>
</span><span id="L-384"><a href="#L-384"><span class="linenos">384</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="L-385"><a href="#L-385"><span class="linenos">385</span></a>
</span><span id="L-386"><a href="#L-386"><span class="linenos">386</span></a>        <span class="n">exon_number</span> <span class="o">=</span> <span class="mi">1</span>
</span><span id="L-387"><a href="#L-387"><span class="linenos">387</span></a>        <span class="k">for</span> <span class="n">cdna_match</span> <span class="ow">in</span> <span class="bp">self</span><span class="o">.</span><span class="n">ordered_cdna_match</span><span class="p">:</span>
</span><span id="L-388"><a href="#L-388"><span class="linenos">388</span></a>            <span class="k">if</span> <span class="p">(</span>
</span><span id="L-389"><a href="#L-389"><span class="linenos">389</span></a>                <span class="n">cdna_match</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_start</span>
</span><span id="L-390"><a href="#L-390"><span class="linenos">390</span></a>                <span class="o">&lt;=</span> <span class="n">offset</span>
</span><span id="L-391"><a href="#L-391"><span class="linenos">391</span></a>                <span class="o">&lt;=</span> <span class="n">cdna_match</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_stop</span>
</span><span id="L-392"><a href="#L-392"><span class="linenos">392</span></a>            <span class="p">):</span>
</span><span id="L-393"><a href="#L-393"><span class="linenos">393</span></a>                <span class="k">return</span> <span class="n">exon_number</span>
</span><span id="L-394"><a href="#L-394"><span class="linenos">394</span></a>            <span class="n">exon_number</span> <span class="o">+=</span> <span class="mi">1</span>
</span><span id="L-395"><a href="#L-395"><span class="linenos">395</span></a>        <span class="k">return</span> <span class="kc">None</span>
</span><span id="L-396"><a href="#L-396"><span class="linenos">396</span></a>
</span><span id="L-397"><a href="#L-397"><span class="linenos">397</span></a>
</span><span id="L-398"><a href="#L-398"><span class="linenos">398</span></a><span class="n">BED6Interval_base</span> <span class="o">=</span> <span class="n">namedtuple</span><span class="p">(</span>
</span><span id="L-399"><a href="#L-399"><span class="linenos">399</span></a>    <span class="s2">&quot;BED6Interval_base&quot;</span><span class="p">,</span>
</span><span id="L-400"><a href="#L-400"><span class="linenos">400</span></a>    <span class="p">(</span><span class="s2">&quot;chrom&quot;</span><span class="p">,</span> <span class="s2">&quot;chrom_start&quot;</span><span class="p">,</span> <span class="s2">&quot;chrom_end&quot;</span><span class="p">,</span> <span class="s2">&quot;name&quot;</span><span class="p">,</span> <span class="s2">&quot;score&quot;</span><span class="p">,</span> <span class="s2">&quot;strand&quot;</span><span class="p">),</span>
</span><span id="L-401"><a href="#L-401"><span class="linenos">401</span></a><span class="p">)</span>
</span><span id="L-402"><a href="#L-402"><span class="linenos">402</span></a>
</span><span id="L-403"><a href="#L-403"><span class="linenos">403</span></a>
</span><span id="L-404"><a href="#L-404"><span class="linenos">404</span></a><span class="c1"># The BED6Interval class is a subclass of BED6Interval_base.</span>
</span><span id="L-405"><a href="#L-405"><span class="linenos">405</span></a><span class="k">class</span> <span class="nc">BED6Interval</span><span class="p">(</span><span class="n">BED6Interval_base</span><span class="p">):</span>
</span><span id="L-406"><a href="#L-406"><span class="linenos">406</span></a>
</span><span id="L-407"><a href="#L-407"><span class="linenos">407</span></a>    <span class="k">def</span> <span class="nf">distance</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">offset</span><span class="p">:</span> <span class="nb">int</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">int</span><span class="p">:</span>
</span><span id="L-408"><a href="#L-408"><span class="linenos">408</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-409"><a href="#L-409"><span class="linenos">409</span></a><span class="sd">        The `distance` function calculates the distance between an offset and an interval, returning</span>
</span><span id="L-410"><a href="#L-410"><span class="linenos">410</span></a><span class="sd">        zero if the offset is inside the interval, a positive value if the interval comes after the</span>
</span><span id="L-411"><a href="#L-411"><span class="linenos">411</span></a><span class="sd">        offset, and a negative value if the interval comes before the offset.</span>
</span><span id="L-412"><a href="#L-412"><span class="linenos">412</span></a><span class="sd">        if offset is inside the exon, distance is zero.</span>
</span><span id="L-413"><a href="#L-413"><span class="linenos">413</span></a><span class="sd">        otherwise, distance is the distance to the nearest edge.</span>
</span><span id="L-414"><a href="#L-414"><span class="linenos">414</span></a><span class="sd">        distance is positive if the exon comes after the offset.</span>
</span><span id="L-415"><a href="#L-415"><span class="linenos">415</span></a><span class="sd">        distance is negative if the exon comes before the offset.</span>
</span><span id="L-416"><a href="#L-416"><span class="linenos">416</span></a>
</span><span id="L-417"><a href="#L-417"><span class="linenos">417</span></a><span class="sd">        :param offset: The offset parameter represents a position or point in the genome. It is an</span>
</span><span id="L-418"><a href="#L-418"><span class="linenos">418</span></a><span class="sd">        integer value that indicates the position within the genome sequence</span>
</span><span id="L-419"><a href="#L-419"><span class="linenos">419</span></a><span class="sd">        :type offset: int</span>
</span><span id="L-420"><a href="#L-420"><span class="linenos">420</span></a><span class="sd">        :return: an integer value, which represents the distance to the interval.</span>
</span><span id="L-421"><a href="#L-421"><span class="linenos">421</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="L-422"><a href="#L-422"><span class="linenos">422</span></a>
</span><span id="L-423"><a href="#L-423"><span class="linenos">423</span></a>        <span class="n">start</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">chrom_start</span> <span class="o">+</span> <span class="mi">1</span>
</span><span id="L-424"><a href="#L-424"><span class="linenos">424</span></a>        <span class="n">end</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">chrom_end</span>
</span><span id="L-425"><a href="#L-425"><span class="linenos">425</span></a>
</span><span id="L-426"><a href="#L-426"><span class="linenos">426</span></a>        <span class="k">if</span> <span class="n">start</span> <span class="o">&lt;=</span> <span class="n">offset</span> <span class="o">&lt;=</span> <span class="n">end</span><span class="p">:</span>
</span><span id="L-427"><a href="#L-427"><span class="linenos">427</span></a>            <span class="k">return</span> <span class="mi">0</span>
</span><span id="L-428"><a href="#L-428"><span class="linenos">428</span></a>
</span><span id="L-429"><a href="#L-429"><span class="linenos">429</span></a>        <span class="n">start_distance</span> <span class="o">=</span> <span class="n">start</span> <span class="o">-</span> <span class="n">offset</span>
</span><span id="L-430"><a href="#L-430"><span class="linenos">430</span></a>        <span class="n">end_distance</span> <span class="o">=</span> <span class="n">offset</span> <span class="o">-</span> <span class="n">end</span>
</span><span id="L-431"><a href="#L-431"><span class="linenos">431</span></a>
</span><span id="L-432"><a href="#L-432"><span class="linenos">432</span></a>        <span class="k">if</span> <span class="nb">abs</span><span class="p">(</span><span class="n">start_distance</span><span class="p">)</span> <span class="o">&lt;</span> <span class="nb">abs</span><span class="p">(</span><span class="n">end_distance</span><span class="p">):</span>
</span><span id="L-433"><a href="#L-433"><span class="linenos">433</span></a>            <span class="k">return</span> <span class="n">start_distance</span>
</span><span id="L-434"><a href="#L-434"><span class="linenos">434</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="L-435"><a href="#L-435"><span class="linenos">435</span></a>            <span class="k">return</span> <span class="o">-</span><span class="n">end_distance</span>
</span><span id="L-436"><a href="#L-436"><span class="linenos">436</span></a>
</span><span id="L-437"><a href="#L-437"><span class="linenos">437</span></a>
</span><span id="L-438"><a href="#L-438"><span class="linenos">438</span></a><span class="c1"># The Exon class is a blueprint for creating objects that represent exons in genetic sequences.</span>
</span><span id="L-439"><a href="#L-439"><span class="linenos">439</span></a><span class="c1"># We still need exons to work out the flanking boundaries</span>
</span><span id="L-440"><a href="#L-440"><span class="linenos">440</span></a><span class="k">class</span> <span class="nc">Exon</span><span class="p">(</span><span class="nb">object</span><span class="p">):</span>
</span><span id="L-441"><a href="#L-441"><span class="linenos">441</span></a>
</span><span id="L-442"><a href="#L-442"><span class="linenos">442</span></a>    <span class="k">def</span> <span class="fm">__init__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">transcript</span><span class="p">:</span> <span class="n">Transcript</span><span class="p">,</span> <span class="n">tx_position</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span> <span class="n">number</span><span class="p">:</span> <span class="nb">int</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
</span><span id="L-443"><a href="#L-443"><span class="linenos">443</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-444"><a href="#L-444"><span class="linenos">444</span></a><span class="sd">        The function initializes an object with a transcript, a position in the transcript, and a</span>
</span><span id="L-445"><a href="#L-445"><span class="linenos">445</span></a><span class="sd">        number.</span>
</span><span id="L-446"><a href="#L-446"><span class="linenos">446</span></a>
</span><span id="L-447"><a href="#L-447"><span class="linenos">447</span></a><span class="sd">        :param transcript: The `transcript` parameter is of type `Transcript`. It represents a</span>
</span><span id="L-448"><a href="#L-448"><span class="linenos">448</span></a><span class="sd">        transcript object that contains information about a conversation or dialogue</span>
</span><span id="L-449"><a href="#L-449"><span class="linenos">449</span></a><span class="sd">        :type transcript: Transcript</span>
</span><span id="L-450"><a href="#L-450"><span class="linenos">450</span></a><span class="sd">        :param tx_position: The `tx_position` parameter represents the position of the transcript in a</span>
</span><span id="L-451"><a href="#L-451"><span class="linenos">451</span></a><span class="sd">        list or array. It is an integer value that indicates the index of the transcript in the list or</span>
</span><span id="L-452"><a href="#L-452"><span class="linenos">452</span></a><span class="sd">        array</span>
</span><span id="L-453"><a href="#L-453"><span class="linenos">453</span></a><span class="sd">        :type tx_position: int</span>
</span><span id="L-454"><a href="#L-454"><span class="linenos">454</span></a><span class="sd">        :param number: The &quot;number&quot; parameter is an integer that represents a specific number. It is</span>
</span><span id="L-455"><a href="#L-455"><span class="linenos">455</span></a><span class="sd">        used as a parameter in the constructor of a class</span>
</span><span id="L-456"><a href="#L-456"><span class="linenos">456</span></a><span class="sd">        :type number: int</span>
</span><span id="L-457"><a href="#L-457"><span class="linenos">457</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="L-458"><a href="#L-458"><span class="linenos">458</span></a>
</span><span id="L-459"><a href="#L-459"><span class="linenos">459</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span> <span class="o">=</span> <span class="n">transcript</span>
</span><span id="L-460"><a href="#L-460"><span class="linenos">460</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span> <span class="o">=</span> <span class="n">tx_position</span>
</span><span id="L-461"><a href="#L-461"><span class="linenos">461</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">number</span> <span class="o">=</span> <span class="n">number</span>
</span><span id="L-462"><a href="#L-462"><span class="linenos">462</span></a>
</span><span id="L-463"><a href="#L-463"><span class="linenos">463</span></a>    <span class="nd">@property</span>
</span><span id="L-464"><a href="#L-464"><span class="linenos">464</span></a>    <span class="k">def</span> <span class="nf">name</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
</span><span id="L-465"><a href="#L-465"><span class="linenos">465</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-466"><a href="#L-466"><span class="linenos">466</span></a><span class="sd">        The function returns a string that combines the name of the transcript and a number.</span>
</span><span id="L-467"><a href="#L-467"><span class="linenos">467</span></a><span class="sd">        :return: a string that combines the name of the transcript with the number. The format of the</span>
</span><span id="L-468"><a href="#L-468"><span class="linenos">468</span></a><span class="sd">        string is &quot;{transcript name}.{number}&quot;.</span>
</span><span id="L-469"><a href="#L-469"><span class="linenos">469</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="L-470"><a href="#L-470"><span class="linenos">470</span></a>
</span><span id="L-471"><a href="#L-471"><span class="linenos">471</span></a>        <span class="k">return</span> <span class="s2">&quot;</span><span class="si">%s</span><span class="s2">.</span><span class="si">%d</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">transcript</span><span class="o">.</span><span class="n">name</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">number</span><span class="p">)</span>
</span><span id="L-472"><a href="#L-472"><span class="linenos">472</span></a>
</span><span id="L-473"><a href="#L-473"><span class="linenos">473</span></a>    <span class="k">def</span> <span class="nf">get_as_interval</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">coding_only</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">object</span><span class="p">:</span>
</span><span id="L-474"><a href="#L-474"><span class="linenos">474</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-475"><a href="#L-475"><span class="linenos">475</span></a><span class="sd">        The function `get_as_interval` returns the coding region for an exon as a BED6Interval object.</span>
</span><span id="L-476"><a href="#L-476"><span class="linenos">476</span></a><span class="sd">        This function returns a BED6Interval objects containing  position</span>
</span><span id="L-477"><a href="#L-477"><span class="linenos">477</span></a><span class="sd">        information for this exon. This may be used as input for</span>
</span><span id="L-478"><a href="#L-478"><span class="linenos">478</span></a><span class="sd">        pybedtools.create_interval_from_list() after casting chrom_start</span>
</span><span id="L-479"><a href="#L-479"><span class="linenos">479</span></a><span class="sd">        and chrom_end as strings.</span>
</span><span id="L-480"><a href="#L-480"><span class="linenos">480</span></a>
</span><span id="L-481"><a href="#L-481"><span class="linenos">481</span></a><span class="sd">        :param coding_only: The `coding_only` parameter is a boolean flag that determines whether to</span>
</span><span id="L-482"><a href="#L-482"><span class="linenos">482</span></a><span class="sd">        include only exons in the coding region. If `coding_only` is set to `True`, the function will</span>
</span><span id="L-483"><a href="#L-483"><span class="linenos">483</span></a><span class="sd">        check if the exon is completely outside the coding region defined by the transcript&#39;s CDS</span>
</span><span id="L-484"><a href="#L-484"><span class="linenos">484</span></a><span class="sd">        (coding sequence) position, defaults to False</span>
</span><span id="L-485"><a href="#L-485"><span class="linenos">485</span></a><span class="sd">        :type coding_only: bool (optional)</span>
</span><span id="L-486"><a href="#L-486"><span class="linenos">486</span></a><span class="sd">        :return: a BED6Interval object.</span>
</span><span id="L-487"><a href="#L-487"><span class="linenos">487</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="L-488"><a href="#L-488"><span class="linenos">488</span></a>
</span><span id="L-489"><a href="#L-489"><span class="linenos">489</span></a>        <span class="n">exon_start</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_start</span>
</span><span id="L-490"><a href="#L-490"><span class="linenos">490</span></a>        <span class="n">exon_stop</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_stop</span>
</span><span id="L-491"><a href="#L-491"><span class="linenos">491</span></a>
</span><span id="L-492"><a href="#L-492"><span class="linenos">492</span></a>        <span class="c1"># Get only exon coding region if requested</span>
</span><span id="L-493"><a href="#L-493"><span class="linenos">493</span></a>        <span class="k">if</span> <span class="n">coding_only</span><span class="p">:</span>
</span><span id="L-494"><a href="#L-494"><span class="linenos">494</span></a>            <span class="k">if</span> <span class="p">(</span>
</span><span id="L-495"><a href="#L-495"><span class="linenos">495</span></a>                <span class="n">exon_stop</span> <span class="o">&lt;=</span> <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span><span class="o">.</span><span class="n">cds_position</span><span class="o">.</span><span class="n">chrom_start</span>
</span><span id="L-496"><a href="#L-496"><span class="linenos">496</span></a>                <span class="ow">or</span> <span class="n">exon_start</span> <span class="o">&gt;=</span> <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span><span class="o">.</span><span class="n">cds_position</span><span class="o">.</span><span class="n">chrom_stop</span>
</span><span id="L-497"><a href="#L-497"><span class="linenos">497</span></a>            <span class="p">):</span>
</span><span id="L-498"><a href="#L-498"><span class="linenos">498</span></a>                <span class="k">return</span> <span class="kc">None</span>
</span><span id="L-499"><a href="#L-499"><span class="linenos">499</span></a>            <span class="n">exon_start</span> <span class="o">=</span> <span class="nb">max</span><span class="p">(</span><span class="n">exon_start</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span><span class="o">.</span><span class="n">cds_position</span><span class="o">.</span><span class="n">chrom_start</span><span class="p">)</span>
</span><span id="L-500"><a href="#L-500"><span class="linenos">500</span></a>            <span class="n">exon_stop</span> <span class="o">=</span> <span class="nb">min</span><span class="p">(</span>
</span><span id="L-501"><a href="#L-501"><span class="linenos">501</span></a>                <span class="nb">max</span><span class="p">(</span><span class="n">exon_stop</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span><span class="o">.</span><span class="n">cds_position</span><span class="o">.</span><span class="n">chrom_start</span><span class="p">),</span>
</span><span id="L-502"><a href="#L-502"><span class="linenos">502</span></a>                <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span><span class="o">.</span><span class="n">cds_position</span><span class="o">.</span><span class="n">chrom_stop</span><span class="p">,</span>
</span><span id="L-503"><a href="#L-503"><span class="linenos">503</span></a>            <span class="p">)</span>
</span><span id="L-504"><a href="#L-504"><span class="linenos">504</span></a>
</span><span id="L-505"><a href="#L-505"><span class="linenos">505</span></a>        <span class="k">return</span> <span class="n">BED6Interval</span><span class="p">(</span>
</span><span id="L-506"><a href="#L-506"><span class="linenos">506</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom</span><span class="p">,</span>
</span><span id="L-507"><a href="#L-507"><span class="linenos">507</span></a>            <span class="n">exon_start</span><span class="p">,</span>
</span><span id="L-508"><a href="#L-508"><span class="linenos">508</span></a>            <span class="n">exon_stop</span><span class="p">,</span>
</span><span id="L-509"><a href="#L-509"><span class="linenos">509</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">name</span><span class="p">,</span>
</span><span id="L-510"><a href="#L-510"><span class="linenos">510</span></a>            <span class="s2">&quot;.&quot;</span><span class="p">,</span>
</span><span id="L-511"><a href="#L-511"><span class="linenos">511</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">strand</span><span class="p">,</span>
</span><span id="L-512"><a href="#L-512"><span class="linenos">512</span></a>        <span class="p">)</span>
</span><span id="L-513"><a href="#L-513"><span class="linenos">513</span></a>
</span><span id="L-514"><a href="#L-514"><span class="linenos">514</span></a>    <span class="nd">@property</span>
</span><span id="L-515"><a href="#L-515"><span class="linenos">515</span></a>    <span class="k">def</span> <span class="nf">strand</span><span class="p">(</span><span class="bp">self</span><span class="p">):</span>
</span><span id="L-516"><a href="#L-516"><span class="linenos">516</span></a>        <span class="n">strand</span> <span class="o">=</span> <span class="s2">&quot;+&quot;</span> <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">is_forward_strand</span> <span class="k">else</span> <span class="s2">&quot;-&quot;</span>
</span><span id="L-517"><a href="#L-517"><span class="linenos">517</span></a>        <span class="k">return</span> <span class="n">strand</span>
</span><span id="L-518"><a href="#L-518"><span class="linenos">518</span></a>
</span><span id="L-519"><a href="#L-519"><span class="linenos">519</span></a>
</span><span id="L-520"><a href="#L-520"><span class="linenos">520</span></a><span class="c1"># The CDNA_Match class is a subclass of the Exon class.</span>
</span><span id="L-521"><a href="#L-521"><span class="linenos">521</span></a><span class="k">class</span> <span class="nc">CDNA_Match</span><span class="p">(</span><span class="n">Exon</span><span class="p">):</span>
</span><span id="L-522"><a href="#L-522"><span class="linenos">522</span></a>
</span><span id="L-523"><a href="#L-523"><span class="linenos">523</span></a>    <span class="k">def</span> <span class="fm">__init__</span><span class="p">(</span>
</span><span id="L-524"><a href="#L-524"><span class="linenos">524</span></a>        <span class="bp">self</span><span class="p">,</span>
</span><span id="L-525"><a href="#L-525"><span class="linenos">525</span></a>        <span class="n">transcript</span><span class="p">:</span> <span class="n">Transcript</span><span class="p">,</span>
</span><span id="L-526"><a href="#L-526"><span class="linenos">526</span></a>        <span class="n">tx_position</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span>
</span><span id="L-527"><a href="#L-527"><span class="linenos">527</span></a>        <span class="n">cdna_start</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span>
</span><span id="L-528"><a href="#L-528"><span class="linenos">528</span></a>        <span class="n">cdna_end</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span>
</span><span id="L-529"><a href="#L-529"><span class="linenos">529</span></a>        <span class="n">gap</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span>
</span><span id="L-530"><a href="#L-530"><span class="linenos">530</span></a>        <span class="n">number</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span>
</span><span id="L-531"><a href="#L-531"><span class="linenos">531</span></a>    <span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
</span><span id="L-532"><a href="#L-532"><span class="linenos">532</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-533"><a href="#L-533"><span class="linenos">533</span></a><span class="sd">        The function initializes a CDNA_Match object with specified attributes.</span>
</span><span id="L-534"><a href="#L-534"><span class="linenos">534</span></a>
</span><span id="L-535"><a href="#L-535"><span class="linenos">535</span></a><span class="sd">        :param transcript: The `transcript` parameter is an instance of the `Transcript` class. It</span>
</span><span id="L-536"><a href="#L-536"><span class="linenos">536</span></a><span class="sd">        represents the transcript that the CDNA match belongs to</span>
</span><span id="L-537"><a href="#L-537"><span class="linenos">537</span></a><span class="sd">        :type transcript: Transcript</span>
</span><span id="L-538"><a href="#L-538"><span class="linenos">538</span></a><span class="sd">        :param tx_position: The `tx_position` parameter represents the position of the transcript in the</span>
</span><span id="L-539"><a href="#L-539"><span class="linenos">539</span></a><span class="sd">        genome. It is an integer value</span>
</span><span id="L-540"><a href="#L-540"><span class="linenos">540</span></a><span class="sd">        :type tx_position: int</span>
</span><span id="L-541"><a href="#L-541"><span class="linenos">541</span></a><span class="sd">        :param cdna_start: The `cdna_start` parameter represents the starting position of the cDNA</span>
</span><span id="L-542"><a href="#L-542"><span class="linenos">542</span></a><span class="sd">        match. It is an integer value</span>
</span><span id="L-543"><a href="#L-543"><span class="linenos">543</span></a><span class="sd">        :type cdna_start: int</span>
</span><span id="L-544"><a href="#L-544"><span class="linenos">544</span></a><span class="sd">        :param cdna_end: The `cdna_end` parameter represents the end position of the cDNA match. It is</span>
</span><span id="L-545"><a href="#L-545"><span class="linenos">545</span></a><span class="sd">        an integer value that indicates the position of the last nucleotide in the cDNA sequence that</span>
</span><span id="L-546"><a href="#L-546"><span class="linenos">546</span></a><span class="sd">        matches the transcript</span>
</span><span id="L-547"><a href="#L-547"><span class="linenos">547</span></a><span class="sd">        :type cdna_end: int</span>
</span><span id="L-548"><a href="#L-548"><span class="linenos">548</span></a><span class="sd">        :param gap: The &quot;gap&quot; parameter represents the number of nucleotides that are missing or</span>
</span><span id="L-549"><a href="#L-549"><span class="linenos">549</span></a><span class="sd">        inserted in the cDNA sequence compared to the reference transcript sequence. It indicates the</span>
</span><span id="L-550"><a href="#L-550"><span class="linenos">550</span></a><span class="sd">        presence of gaps or insertions in the alignment between the cDNA and the reference transcript</span>
</span><span id="L-551"><a href="#L-551"><span class="linenos">551</span></a><span class="sd">        :type gap: int</span>
</span><span id="L-552"><a href="#L-552"><span class="linenos">552</span></a><span class="sd">        :param number: The `number` parameter represents the number of the CDNA match. It is used to</span>
</span><span id="L-553"><a href="#L-553"><span class="linenos">553</span></a><span class="sd">        uniquely identify each CDNA match object</span>
</span><span id="L-554"><a href="#L-554"><span class="linenos">554</span></a><span class="sd">        :type number: int</span>
</span><span id="L-555"><a href="#L-555"><span class="linenos">555</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="L-556"><a href="#L-556"><span class="linenos">556</span></a>
</span><span id="L-557"><a href="#L-557"><span class="linenos">557</span></a>        <span class="nb">super</span><span class="p">(</span><span class="n">CDNA_Match</span><span class="p">,</span> <span class="bp">self</span><span class="p">)</span><span class="o">.</span><span class="fm">__init__</span><span class="p">(</span><span class="n">transcript</span><span class="p">,</span> <span class="n">tx_position</span><span class="p">,</span> <span class="n">number</span><span class="p">)</span>
</span><span id="L-558"><a href="#L-558"><span class="linenos">558</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">cdna_start</span> <span class="o">=</span> <span class="n">cdna_start</span>
</span><span id="L-559"><a href="#L-559"><span class="linenos">559</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">cdna_end</span> <span class="o">=</span> <span class="n">cdna_end</span>
</span><span id="L-560"><a href="#L-560"><span class="linenos">560</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">gap</span> <span class="o">=</span> <span class="n">gap</span>
</span><span id="L-561"><a href="#L-561"><span class="linenos">561</span></a>
</span><span id="L-562"><a href="#L-562"><span class="linenos">562</span></a>    <span class="nd">@property</span>
</span><span id="L-563"><a href="#L-563"><span class="linenos">563</span></a>    <span class="k">def</span> <span class="nf">length</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">int</span><span class="p">:</span>
</span><span id="L-564"><a href="#L-564"><span class="linenos">564</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-565"><a href="#L-565"><span class="linenos">565</span></a><span class="sd">        The function calculates the length of a sequence by subtracting the start position from the end</span>
</span><span id="L-566"><a href="#L-566"><span class="linenos">566</span></a><span class="sd">        position and adding 1.</span>
</span><span id="L-567"><a href="#L-567"><span class="linenos">567</span></a><span class="sd">        :return: The length of the sequence, calculated by subtracting the cdna_start from the cdna_end</span>
</span><span id="L-568"><a href="#L-568"><span class="linenos">568</span></a><span class="sd">        and adding 1.</span>
</span><span id="L-569"><a href="#L-569"><span class="linenos">569</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="L-570"><a href="#L-570"><span class="linenos">570</span></a>
</span><span id="L-571"><a href="#L-571"><span class="linenos">571</span></a>        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">cdna_end</span> <span class="o">-</span> <span class="bp">self</span><span class="o">.</span><span class="n">cdna_start</span> <span class="o">+</span> <span class="mi">1</span>
</span><span id="L-572"><a href="#L-572"><span class="linenos">572</span></a>
</span><span id="L-573"><a href="#L-573"><span class="linenos">573</span></a>    <span class="k">def</span> <span class="nf">get_offset</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">position</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span> <span class="n">validate</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">True</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">int</span><span class="p">:</span>
</span><span id="L-574"><a href="#L-574"><span class="linenos">574</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="L-575"><a href="#L-575"><span class="linenos">575</span></a><span class="sd">        The `get_offset` function calculates the offset for a given position in a cDNA sequence based on</span>
</span><span id="L-576"><a href="#L-576"><span class="linenos">576</span></a><span class="sd">        the GAP attribute.</span>
</span><span id="L-577"><a href="#L-577"><span class="linenos">577</span></a><span class="sd">        cdna_match GAP attribute looks like: &#39;M185 I3 M250&#39; which is code/length</span>
</span><span id="L-578"><a href="#L-578"><span class="linenos">578</span></a><span class="sd">        @see https://github.com/The-Sequence-Ontology/Specifications/blob/master/gff3.md#the-gap-attribute</span>
</span><span id="L-579"><a href="#L-579"><span class="linenos">579</span></a><span class="sd">        codes operation</span>
</span><span id="L-580"><a href="#L-580"><span class="linenos">580</span></a><span class="sd">        M 	match</span>
</span><span id="L-581"><a href="#L-581"><span class="linenos">581</span></a><span class="sd">        I 	insert a gap into the reference sequence</span>
</span><span id="L-582"><a href="#L-582"><span class="linenos">582</span></a><span class="sd">        D 	insert a gap into the target (delete from reference)</span>
</span><span id="L-583"><a href="#L-583"><span class="linenos">583</span></a><span class="sd">        If you want the whole exon, then pass the end</span>
</span><span id="L-584"><a href="#L-584"><span class="linenos">584</span></a>
</span><span id="L-585"><a href="#L-585"><span class="linenos">585</span></a><span class="sd">        :param position: The `position` parameter is an integer that represents the position in the</span>
</span><span id="L-586"><a href="#L-586"><span class="linenos">586</span></a><span class="sd">        sequence. It is used to calculate the offset based on the GAP attribute of the cDNA match</span>
</span><span id="L-587"><a href="#L-587"><span class="linenos">587</span></a><span class="sd">        :type position: int</span>
</span><span id="L-588"><a href="#L-588"><span class="linenos">588</span></a><span class="sd">        :param validate: The `validate` parameter is a boolean flag that determines whether to perform</span>
</span><span id="L-589"><a href="#L-589"><span class="linenos">589</span></a><span class="sd">        validation checks during the calculation of the offset. If `validate` is set to `True`, the</span>
</span><span id="L-590"><a href="#L-590"><span class="linenos">590</span></a><span class="sd">        function will raise a `ValueError` if the given position falls within an insertion or deletion</span>
</span><span id="L-591"><a href="#L-591"><span class="linenos">591</span></a><span class="sd">        gap. If `validate` is set, defaults to True</span>
</span><span id="L-592"><a href="#L-592"><span class="linenos">592</span></a><span class="sd">        :type validate: bool (optional)</span>
</span><span id="L-593"><a href="#L-593"><span class="linenos">593</span></a><span class="sd">        :return: an integer value representing the offset for a given position in the cDNA sequence.</span>
</span><span id="L-594"><a href="#L-594"><span class="linenos">594</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="L-595"><a href="#L-595"><span class="linenos">595</span></a>
</span><span id="L-596"><a href="#L-596"><span class="linenos">596</span></a>        <span class="k">if</span> <span class="ow">not</span> <span class="bp">self</span><span class="o">.</span><span class="n">gap</span><span class="p">:</span>
</span><span id="L-597"><a href="#L-597"><span class="linenos">597</span></a>            <span class="k">return</span> <span class="mi">0</span>
</span><span id="L-598"><a href="#L-598"><span class="linenos">598</span></a>
</span><span id="L-599"><a href="#L-599"><span class="linenos">599</span></a>        <span class="n">position_1_based</span> <span class="o">=</span> <span class="n">position</span> <span class="o">+</span> <span class="mi">1</span>
</span><span id="L-600"><a href="#L-600"><span class="linenos">600</span></a>        <span class="n">cdna_match_index</span> <span class="o">=</span> <span class="mi">1</span>
</span><span id="L-601"><a href="#L-601"><span class="linenos">601</span></a>        <span class="n">offset</span> <span class="o">=</span> <span class="mi">0</span>
</span><span id="L-602"><a href="#L-602"><span class="linenos">602</span></a>        <span class="k">for</span> <span class="n">gap_op</span> <span class="ow">in</span> <span class="bp">self</span><span class="o">.</span><span class="n">gap</span><span class="o">.</span><span class="n">split</span><span class="p">():</span>
</span><span id="L-603"><a href="#L-603"><span class="linenos">603</span></a>            <span class="n">code</span> <span class="o">=</span> <span class="n">gap_op</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>
</span><span id="L-604"><a href="#L-604"><span class="linenos">604</span></a>            <span class="n">length</span> <span class="o">=</span> <span class="nb">int</span><span class="p">(</span><span class="n">gap_op</span><span class="p">[</span><span class="mi">1</span><span class="p">:])</span>
</span><span id="L-605"><a href="#L-605"><span class="linenos">605</span></a>            <span class="k">if</span> <span class="n">code</span> <span class="o">==</span> <span class="s2">&quot;M&quot;</span><span class="p">:</span>
</span><span id="L-606"><a href="#L-606"><span class="linenos">606</span></a>                <span class="n">cdna_match_index</span> <span class="o">+=</span> <span class="n">length</span>
</span><span id="L-607"><a href="#L-607"><span class="linenos">607</span></a>            <span class="k">elif</span> <span class="n">code</span> <span class="o">==</span> <span class="s2">&quot;I&quot;</span><span class="p">:</span>
</span><span id="L-608"><a href="#L-608"><span class="linenos">608</span></a>                <span class="k">if</span> <span class="n">validate</span> <span class="ow">and</span> <span class="n">position_1_based</span> <span class="o">&lt;</span> <span class="n">cdna_match_index</span> <span class="o">+</span> <span class="n">length</span><span class="p">:</span>
</span><span id="L-609"><a href="#L-609"><span class="linenos">609</span></a>                    <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span>
</span><span id="L-610"><a href="#L-610"><span class="linenos">610</span></a>                        <span class="s2">&quot;Coordinate (</span><span class="si">%d</span><span class="s2">) inside insertion (</span><span class="si">%s</span><span class="s2">) - no mapping possible!&quot;</span>
</span><span id="L-611"><a href="#L-611"><span class="linenos">611</span></a>                        <span class="o">%</span> <span class="p">(</span><span class="n">position_1_based</span><span class="p">,</span> <span class="n">gap_op</span><span class="p">)</span>
</span><span id="L-612"><a href="#L-612"><span class="linenos">612</span></a>                    <span class="p">)</span>
</span><span id="L-613"><a href="#L-613"><span class="linenos">613</span></a>                <span class="n">offset</span> <span class="o">+=</span> <span class="n">length</span>
</span><span id="L-614"><a href="#L-614"><span class="linenos">614</span></a>            <span class="k">elif</span> <span class="n">code</span> <span class="o">==</span> <span class="s2">&quot;D&quot;</span><span class="p">:</span>
</span><span id="L-615"><a href="#L-615"><span class="linenos">615</span></a>                <span class="k">if</span> <span class="n">validate</span> <span class="ow">and</span> <span class="n">position</span> <span class="o">&lt;</span> <span class="n">cdna_match_index</span> <span class="o">+</span> <span class="n">length</span><span class="p">:</span>
</span><span id="L-616"><a href="#L-616"><span class="linenos">616</span></a>                    <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span>
</span><span id="L-617"><a href="#L-617"><span class="linenos">617</span></a>                        <span class="s2">&quot;Coordinate (</span><span class="si">%d</span><span class="s2">) inside deletion (</span><span class="si">%s</span><span class="s2">) - no mapping possible!&quot;</span>
</span><span id="L-618"><a href="#L-618"><span class="linenos">618</span></a>                        <span class="o">%</span> <span class="p">(</span><span class="n">position_1_based</span><span class="p">,</span> <span class="n">gap_op</span><span class="p">)</span>
</span><span id="L-619"><a href="#L-619"><span class="linenos">619</span></a>                    <span class="p">)</span>
</span><span id="L-620"><a href="#L-620"><span class="linenos">620</span></a>                <span class="n">offset</span> <span class="o">-=</span> <span class="n">length</span>
</span><span id="L-621"><a href="#L-621"><span class="linenos">621</span></a>            <span class="k">else</span><span class="p">:</span>
</span><span id="L-622"><a href="#L-622"><span class="linenos">622</span></a>                <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Unknown code in cDNA GAP: </span><span class="si">{</span><span class="n">gap_op</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
</span><span id="L-623"><a href="#L-623"><span class="linenos">623</span></a>
</span><span id="L-624"><a href="#L-624"><span class="linenos">624</span></a>            <span class="k">if</span> <span class="n">cdna_match_index</span> <span class="o">&gt;</span> <span class="n">position_1_based</span><span class="p">:</span>
</span><span id="L-625"><a href="#L-625"><span class="linenos">625</span></a>                <span class="k">break</span>
</span><span id="L-626"><a href="#L-626"><span class="linenos">626</span></a>
</span><span id="L-627"><a href="#L-627"><span class="linenos">627</span></a>        <span class="k">return</span> <span class="n">offset</span>
</span></pre></div>


            </section>
                <section id="Gene">
                            <input id="Gene-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr class">
            
    <span class="def">class</span>
    <span class="name">Gene</span>:

                <label class="view-source-button" for="Gene-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#Gene"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="Gene-16"><a href="#Gene-16"><span class="linenos">16</span></a><span class="k">class</span> <span class="nc">Gene</span><span class="p">(</span><span class="nb">object</span><span class="p">):</span>
</span><span id="Gene-17"><a href="#Gene-17"><span class="linenos">17</span></a>    <span class="k">def</span> <span class="fm">__init__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">name</span><span class="p">):</span>
</span><span id="Gene-18"><a href="#Gene-18"><span class="linenos">18</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">name</span> <span class="o">=</span> <span class="n">name</span>
</span></pre></div>


    

                            <div id="Gene.__init__" class="classattr">
                                        <input id="Gene.__init__-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr function">
            
        <span class="name">Gene</span><span class="signature pdoc-code condensed">(<span class="param"><span class="n">name</span></span>)</span>

                <label class="view-source-button" for="Gene.__init__-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#Gene.__init__"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="Gene.__init__-17"><a href="#Gene.__init__-17"><span class="linenos">17</span></a>    <span class="k">def</span> <span class="fm">__init__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">name</span><span class="p">):</span>
</span><span id="Gene.__init__-18"><a href="#Gene.__init__-18"><span class="linenos">18</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">name</span> <span class="o">=</span> <span class="n">name</span>
</span></pre></div>


    

                            </div>
                            <div id="Gene.name" class="classattr">
                                <div class="attr variable">
            <span class="name">name</span>

        
    </div>
    <a class="headerlink" href="#Gene.name"></a>
    
    

                            </div>
                </section>
                <section id="Transcript">
                            <input id="Transcript-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr class">
            
    <span class="def">class</span>
    <span class="name">Transcript</span>:

                <label class="view-source-button" for="Transcript-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#Transcript"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="Transcript-24"><a href="#Transcript-24"><span class="linenos"> 24</span></a><span class="k">class</span> <span class="nc">Transcript</span><span class="p">(</span><span class="nb">object</span><span class="p">):</span>
</span><span id="Transcript-25"><a href="#Transcript-25"><span class="linenos"> 25</span></a>
</span><span id="Transcript-26"><a href="#Transcript-26"><span class="linenos"> 26</span></a>    <span class="k">def</span> <span class="fm">__init__</span><span class="p">(</span>
</span><span id="Transcript-27"><a href="#Transcript-27"><span class="linenos"> 27</span></a>        <span class="bp">self</span><span class="p">,</span>
</span><span id="Transcript-28"><a href="#Transcript-28"><span class="linenos"> 28</span></a>        <span class="n">name</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
</span><span id="Transcript-29"><a href="#Transcript-29"><span class="linenos"> 29</span></a>        <span class="n">version</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
</span><span id="Transcript-30"><a href="#Transcript-30"><span class="linenos"> 30</span></a>        <span class="n">gene</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
</span><span id="Transcript-31"><a href="#Transcript-31"><span class="linenos"> 31</span></a>        <span class="n">tx_position</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span>
</span><span id="Transcript-32"><a href="#Transcript-32"><span class="linenos"> 32</span></a>        <span class="n">cds_position</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span>
</span><span id="Transcript-33"><a href="#Transcript-33"><span class="linenos"> 33</span></a>        <span class="n">is_default</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
</span><span id="Transcript-34"><a href="#Transcript-34"><span class="linenos"> 34</span></a>        <span class="n">cdna_match</span><span class="p">:</span> <span class="nb">list</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
</span><span id="Transcript-35"><a href="#Transcript-35"><span class="linenos"> 35</span></a>        <span class="n">start_codon_transcript_pos</span><span class="p">:</span> <span class="nb">int</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
</span><span id="Transcript-36"><a href="#Transcript-36"><span class="linenos"> 36</span></a>        <span class="n">stop_codon_transcript_pos</span><span class="p">:</span> <span class="nb">int</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
</span><span id="Transcript-37"><a href="#Transcript-37"><span class="linenos"> 37</span></a>    <span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
</span><span id="Transcript-38"><a href="#Transcript-38"><span class="linenos"> 38</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="Transcript-39"><a href="#Transcript-39"><span class="linenos"> 39</span></a><span class="sd">        The function initializes an object with various attributes related to a gene and its transcript.</span>
</span><span id="Transcript-40"><a href="#Transcript-40"><span class="linenos"> 40</span></a>
</span><span id="Transcript-41"><a href="#Transcript-41"><span class="linenos"> 41</span></a><span class="sd">        :param name: A string representing the name of the coding sequence</span>
</span><span id="Transcript-42"><a href="#Transcript-42"><span class="linenos"> 42</span></a><span class="sd">        :type name: str</span>
</span><span id="Transcript-43"><a href="#Transcript-43"><span class="linenos"> 43</span></a><span class="sd">        :param version: The `version` parameter is a string that represents the version of the object.</span>
</span><span id="Transcript-44"><a href="#Transcript-44"><span class="linenos"> 44</span></a><span class="sd">        It is used to track changes or updates to the object over time</span>
</span><span id="Transcript-45"><a href="#Transcript-45"><span class="linenos"> 45</span></a><span class="sd">        :type version: str</span>
</span><span id="Transcript-46"><a href="#Transcript-46"><span class="linenos"> 46</span></a><span class="sd">        :param gene: The `gene` parameter is a string that represents the gene associated with the</span>
</span><span id="Transcript-47"><a href="#Transcript-47"><span class="linenos"> 47</span></a><span class="sd">        coding sequence</span>
</span><span id="Transcript-48"><a href="#Transcript-48"><span class="linenos"> 48</span></a><span class="sd">        :type gene: str</span>
</span><span id="Transcript-49"><a href="#Transcript-49"><span class="linenos"> 49</span></a><span class="sd">        :param tx_position: The `tx_position` parameter represents the position of the transcript. It is</span>
</span><span id="Transcript-50"><a href="#Transcript-50"><span class="linenos"> 50</span></a><span class="sd">        an integer value that indicates the position of the transcript in the genome</span>
</span><span id="Transcript-51"><a href="#Transcript-51"><span class="linenos"> 51</span></a><span class="sd">        :type tx_position: int</span>
</span><span id="Transcript-52"><a href="#Transcript-52"><span class="linenos"> 52</span></a><span class="sd">        :param cds_position: The `cds_position` parameter represents the position of the coding sequence</span>
</span><span id="Transcript-53"><a href="#Transcript-53"><span class="linenos"> 53</span></a><span class="sd">        (CDS) within the transcript. It is an integer value that indicates the starting position of the</span>
</span><span id="Transcript-54"><a href="#Transcript-54"><span class="linenos"> 54</span></a><span class="sd">        CDS within the transcript sequence</span>
</span><span id="Transcript-55"><a href="#Transcript-55"><span class="linenos"> 55</span></a><span class="sd">        :type cds_position: int</span>
</span><span id="Transcript-56"><a href="#Transcript-56"><span class="linenos"> 56</span></a><span class="sd">        :param is_default: The `is_default` parameter is a boolean flag that indicates whether the</span>
</span><span id="Transcript-57"><a href="#Transcript-57"><span class="linenos"> 57</span></a><span class="sd">        instance of the class is the default version of the gene. It is set to `False` by default, but</span>
</span><span id="Transcript-58"><a href="#Transcript-58"><span class="linenos"> 58</span></a><span class="sd">        can be set to `True` if the instance is the default version, defaults to False</span>
</span><span id="Transcript-59"><a href="#Transcript-59"><span class="linenos"> 59</span></a><span class="sd">        :type is_default: bool (optional)</span>
</span><span id="Transcript-60"><a href="#Transcript-60"><span class="linenos"> 60</span></a><span class="sd">        :param cdna_match: The `cdna_match` parameter is a list that contains the positions of the</span>
</span><span id="Transcript-61"><a href="#Transcript-61"><span class="linenos"> 61</span></a><span class="sd">        matching cDNA sequences. It is an optional parameter and if not provided, it defaults to an</span>
</span><span id="Transcript-62"><a href="#Transcript-62"><span class="linenos"> 62</span></a><span class="sd">        empty list</span>
</span><span id="Transcript-63"><a href="#Transcript-63"><span class="linenos"> 63</span></a><span class="sd">        :type cdna_match: list</span>
</span><span id="Transcript-64"><a href="#Transcript-64"><span class="linenos"> 64</span></a><span class="sd">        :param start_codon_transcript_pos: The parameter &quot;start_codon_transcript_pos&quot; is an optional</span>
</span><span id="Transcript-65"><a href="#Transcript-65"><span class="linenos"> 65</span></a><span class="sd">        parameter that represents the transcript position of the start codon. It is used to store the</span>
</span><span id="Transcript-66"><a href="#Transcript-66"><span class="linenos"> 66</span></a><span class="sd">        pre-calculated transcript position of the start codon for a specific gene</span>
</span><span id="Transcript-67"><a href="#Transcript-67"><span class="linenos"> 67</span></a><span class="sd">        :type start_codon_transcript_pos: int</span>
</span><span id="Transcript-68"><a href="#Transcript-68"><span class="linenos"> 68</span></a><span class="sd">        :param stop_codon_transcript_pos: The parameter `stop_codon_transcript_pos` is an optional</span>
</span><span id="Transcript-69"><a href="#Transcript-69"><span class="linenos"> 69</span></a><span class="sd">        integer that represents the transcript position of the stop codon. It is used to store the</span>
</span><span id="Transcript-70"><a href="#Transcript-70"><span class="linenos"> 70</span></a><span class="sd">        pre-calculated transcript coordinate of the stop codon. If not provided, it will be set to</span>
</span><span id="Transcript-71"><a href="#Transcript-71"><span class="linenos"> 71</span></a><span class="sd">        `None`</span>
</span><span id="Transcript-72"><a href="#Transcript-72"><span class="linenos"> 72</span></a><span class="sd">        :type stop_codon_transcript_pos: int</span>
</span><span id="Transcript-73"><a href="#Transcript-73"><span class="linenos"> 73</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="Transcript-74"><a href="#Transcript-74"><span class="linenos"> 74</span></a>
</span><span id="Transcript-75"><a href="#Transcript-75"><span class="linenos"> 75</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">name</span> <span class="o">=</span> <span class="n">name</span>
</span><span id="Transcript-76"><a href="#Transcript-76"><span class="linenos"> 76</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">version</span> <span class="o">=</span> <span class="n">version</span>
</span><span id="Transcript-77"><a href="#Transcript-77"><span class="linenos"> 77</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">gene</span> <span class="o">=</span> <span class="n">Gene</span><span class="p">(</span><span class="n">gene</span><span class="p">)</span>
</span><span id="Transcript-78"><a href="#Transcript-78"><span class="linenos"> 78</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span> <span class="o">=</span> <span class="n">tx_position</span>
</span><span id="Transcript-79"><a href="#Transcript-79"><span class="linenos"> 79</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">cds_position</span> <span class="o">=</span> <span class="n">cds_position</span>
</span><span id="Transcript-80"><a href="#Transcript-80"><span class="linenos"> 80</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">is_default</span> <span class="o">=</span> <span class="n">is_default</span>
</span><span id="Transcript-81"><a href="#Transcript-81"><span class="linenos"> 81</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">cdna_match</span> <span class="o">=</span> <span class="n">cdna_match</span> <span class="ow">or</span> <span class="p">[]</span>
</span><span id="Transcript-82"><a href="#Transcript-82"><span class="linenos"> 82</span></a>        <span class="c1"># Optional pre-calculated transcript coordinates</span>
</span><span id="Transcript-83"><a href="#Transcript-83"><span class="linenos"> 83</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">_start_codon_transcript_pos</span> <span class="o">=</span> <span class="n">start_codon_transcript_pos</span>
</span><span id="Transcript-84"><a href="#Transcript-84"><span class="linenos"> 84</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">_stop_codon_transcript_pos</span> <span class="o">=</span> <span class="n">stop_codon_transcript_pos</span>
</span><span id="Transcript-85"><a href="#Transcript-85"><span class="linenos"> 85</span></a>
</span><span id="Transcript-86"><a href="#Transcript-86"><span class="linenos"> 86</span></a>    <span class="nd">@property</span>
</span><span id="Transcript-87"><a href="#Transcript-87"><span class="linenos"> 87</span></a>    <span class="k">def</span> <span class="nf">full_name</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
</span><span id="Transcript-88"><a href="#Transcript-88"><span class="linenos"> 88</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="Transcript-89"><a href="#Transcript-89"><span class="linenos"> 89</span></a><span class="sd">        The function `full_name` returns the full name of an object, including its version if it exists.</span>
</span><span id="Transcript-90"><a href="#Transcript-90"><span class="linenos"> 90</span></a><span class="sd">        :return: a string. If the `version` attribute of the object is not `None`, it returns a string</span>
</span><span id="Transcript-91"><a href="#Transcript-91"><span class="linenos"> 91</span></a><span class="sd">        in the format `name.version`. Otherwise, it returns just the `name` attribute.</span>
</span><span id="Transcript-92"><a href="#Transcript-92"><span class="linenos"> 92</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="Transcript-93"><a href="#Transcript-93"><span class="linenos"> 93</span></a>
</span><span id="Transcript-94"><a href="#Transcript-94"><span class="linenos"> 94</span></a>        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">version</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
</span><span id="Transcript-95"><a href="#Transcript-95"><span class="linenos"> 95</span></a>            <span class="k">return</span> <span class="s2">&quot;</span><span class="si">%s</span><span class="s2">.</span><span class="si">%d</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">name</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">version</span><span class="p">)</span>
</span><span id="Transcript-96"><a href="#Transcript-96"><span class="linenos"> 96</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="Transcript-97"><a href="#Transcript-97"><span class="linenos"> 97</span></a>            <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">name</span>
</span><span id="Transcript-98"><a href="#Transcript-98"><span class="linenos"> 98</span></a>
</span><span id="Transcript-99"><a href="#Transcript-99"><span class="linenos"> 99</span></a>    <span class="nd">@property</span>
</span><span id="Transcript-100"><a href="#Transcript-100"><span class="linenos">100</span></a>    <span class="k">def</span> <span class="nf">is_coding</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">bool</span><span class="p">:</span>
</span><span id="Transcript-101"><a href="#Transcript-101"><span class="linenos">101</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="Transcript-102"><a href="#Transcript-102"><span class="linenos">102</span></a><span class="sd">        The function checks if a coding transcript has a non-zero length coding sequence.</span>
</span><span id="Transcript-103"><a href="#Transcript-103"><span class="linenos">103</span></a><span class="sd">        :return: a boolean value indicating whether the coding transcript has a coding sequence (CDS)</span>
</span><span id="Transcript-104"><a href="#Transcript-104"><span class="linenos">104</span></a><span class="sd">        with a non-zero length.</span>
</span><span id="Transcript-105"><a href="#Transcript-105"><span class="linenos">105</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="Transcript-106"><a href="#Transcript-106"><span class="linenos">106</span></a>
</span><span id="Transcript-107"><a href="#Transcript-107"><span class="linenos">107</span></a>        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">cds_position</span><span class="o">.</span><span class="n">chrom_stop</span> <span class="o">-</span> <span class="bp">self</span><span class="o">.</span><span class="n">cds_position</span><span class="o">.</span><span class="n">chrom_start</span> <span class="o">&gt;</span> <span class="mi">0</span>
</span><span id="Transcript-108"><a href="#Transcript-108"><span class="linenos">108</span></a>
</span><span id="Transcript-109"><a href="#Transcript-109"><span class="linenos">109</span></a>    <span class="nd">@property</span>
</span><span id="Transcript-110"><a href="#Transcript-110"><span class="linenos">110</span></a>    <span class="k">def</span> <span class="nf">strand</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
</span><span id="Transcript-111"><a href="#Transcript-111"><span class="linenos">111</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="Transcript-112"><a href="#Transcript-112"><span class="linenos">112</span></a><span class="sd">        The function returns a string &#39;+&#39; if the tx_position is on the forward strand, and &#39;-&#39; if it is</span>
</span><span id="Transcript-113"><a href="#Transcript-113"><span class="linenos">113</span></a><span class="sd">        on the reverse strand.</span>
</span><span id="Transcript-114"><a href="#Transcript-114"><span class="linenos">114</span></a><span class="sd">        :return: a string that represents the strand of the given `self.tx_position`. If</span>
</span><span id="Transcript-115"><a href="#Transcript-115"><span class="linenos">115</span></a><span class="sd">        `self.tx_position.is_forward_strand` is `True`, then the string returned is &#39;+&#39;. Otherwise, the</span>
</span><span id="Transcript-116"><a href="#Transcript-116"><span class="linenos">116</span></a><span class="sd">        string returned is &#39;-&#39;.</span>
</span><span id="Transcript-117"><a href="#Transcript-117"><span class="linenos">117</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="Transcript-118"><a href="#Transcript-118"><span class="linenos">118</span></a>
</span><span id="Transcript-119"><a href="#Transcript-119"><span class="linenos">119</span></a>        <span class="k">return</span> <span class="s2">&quot;+&quot;</span> <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">is_forward_strand</span> <span class="k">else</span> <span class="s2">&quot;-&quot;</span>
</span><span id="Transcript-120"><a href="#Transcript-120"><span class="linenos">120</span></a>
</span><span id="Transcript-121"><a href="#Transcript-121"><span class="linenos">121</span></a>    <span class="nd">@lazy</span>
</span><span id="Transcript-122"><a href="#Transcript-122"><span class="linenos">122</span></a>    <span class="k">def</span> <span class="nf">ordered_cdna_match</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
</span><span id="Transcript-123"><a href="#Transcript-123"><span class="linenos">123</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="Transcript-124"><a href="#Transcript-124"><span class="linenos">124</span></a><span class="sd">        The function &quot;ordered_cdna_match&quot; sorts a list of cdna_match objects based on their</span>
</span><span id="Transcript-125"><a href="#Transcript-125"><span class="linenos">125</span></a><span class="sd">        tx_position.chrom_start attribute and returns the sorted list.</span>
</span><span id="Transcript-126"><a href="#Transcript-126"><span class="linenos">126</span></a><span class="sd">        :return: a sorted list of cdna_match objects.</span>
</span><span id="Transcript-127"><a href="#Transcript-127"><span class="linenos">127</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="Transcript-128"><a href="#Transcript-128"><span class="linenos">128</span></a>
</span><span id="Transcript-129"><a href="#Transcript-129"><span class="linenos">129</span></a>        <span class="n">transcript_strand</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">is_forward_strand</span>
</span><span id="Transcript-130"><a href="#Transcript-130"><span class="linenos">130</span></a>        <span class="n">cdna_match</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">cdna_match</span><span class="p">)</span>
</span><span id="Transcript-131"><a href="#Transcript-131"><span class="linenos">131</span></a>        <span class="n">cdna_match</span><span class="o">.</span><span class="n">sort</span><span class="p">(</span><span class="n">key</span><span class="o">=</span><span class="k">lambda</span> <span class="n">cm</span><span class="p">:</span> <span class="n">cm</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_start</span><span class="p">)</span>
</span><span id="Transcript-132"><a href="#Transcript-132"><span class="linenos">132</span></a>        <span class="k">if</span> <span class="ow">not</span> <span class="n">transcript_strand</span><span class="p">:</span>
</span><span id="Transcript-133"><a href="#Transcript-133"><span class="linenos">133</span></a>            <span class="n">cdna_match</span><span class="o">.</span><span class="n">reverse</span><span class="p">()</span>
</span><span id="Transcript-134"><a href="#Transcript-134"><span class="linenos">134</span></a>        <span class="k">return</span> <span class="n">cdna_match</span>
</span><span id="Transcript-135"><a href="#Transcript-135"><span class="linenos">135</span></a>
</span><span id="Transcript-136"><a href="#Transcript-136"><span class="linenos">136</span></a>    <span class="k">def</span> <span class="nf">get_cds_start_stop</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">tuple</span><span class="p">:</span>
</span><span id="Transcript-137"><a href="#Transcript-137"><span class="linenos">137</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="Transcript-138"><a href="#Transcript-138"><span class="linenos">138</span></a><span class="sd">        The function &quot;get_cds_start_stop&quot; returns the start and stop positions of a coding sequence,</span>
</span><span id="Transcript-139"><a href="#Transcript-139"><span class="linenos">139</span></a><span class="sd">        taking into account the direction of the strand.</span>
</span><span id="Transcript-140"><a href="#Transcript-140"><span class="linenos">140</span></a><span class="sd">        :return: a tuple containing the start and stop positions of the coding sequence (CDS).</span>
</span><span id="Transcript-141"><a href="#Transcript-141"><span class="linenos">141</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="Transcript-142"><a href="#Transcript-142"><span class="linenos">142</span></a>
</span><span id="Transcript-143"><a href="#Transcript-143"><span class="linenos">143</span></a>        <span class="p">(</span><span class="n">start</span><span class="p">,</span> <span class="n">stop</span><span class="p">)</span> <span class="o">=</span> <span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">cds_position</span><span class="o">.</span><span class="n">chrom_start</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">cds_position</span><span class="o">.</span><span class="n">chrom_stop</span><span class="p">)</span>
</span><span id="Transcript-144"><a href="#Transcript-144"><span class="linenos">144</span></a>        <span class="k">if</span> <span class="ow">not</span> <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">is_forward_strand</span><span class="p">:</span>
</span><span id="Transcript-145"><a href="#Transcript-145"><span class="linenos">145</span></a>            <span class="p">(</span><span class="n">stop</span><span class="p">,</span> <span class="n">start</span><span class="p">)</span> <span class="o">=</span> <span class="p">(</span><span class="n">start</span><span class="p">,</span> <span class="n">stop</span><span class="p">)</span>
</span><span id="Transcript-146"><a href="#Transcript-146"><span class="linenos">146</span></a>        <span class="k">return</span> <span class="n">start</span><span class="p">,</span> <span class="n">stop</span>
</span><span id="Transcript-147"><a href="#Transcript-147"><span class="linenos">147</span></a>
</span><span id="Transcript-148"><a href="#Transcript-148"><span class="linenos">148</span></a>    <span class="nd">@lazy</span>
</span><span id="Transcript-149"><a href="#Transcript-149"><span class="linenos">149</span></a>    <span class="k">def</span> <span class="nf">start_codon</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">int</span><span class="p">:</span>
</span><span id="Transcript-150"><a href="#Transcript-150"><span class="linenos">150</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="Transcript-151"><a href="#Transcript-151"><span class="linenos">151</span></a><span class="sd">        The function returns the transcript position of the start codon.</span>
</span><span id="Transcript-152"><a href="#Transcript-152"><span class="linenos">152</span></a><span class="sd">        :return: the transcript position of the start codon.</span>
</span><span id="Transcript-153"><a href="#Transcript-153"><span class="linenos">153</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="Transcript-154"><a href="#Transcript-154"><span class="linenos">154</span></a>
</span><span id="Transcript-155"><a href="#Transcript-155"><span class="linenos">155</span></a>        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">_start_codon_transcript_pos</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
</span><span id="Transcript-156"><a href="#Transcript-156"><span class="linenos">156</span></a>            <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">_start_codon_transcript_pos</span>
</span><span id="Transcript-157"><a href="#Transcript-157"><span class="linenos">157</span></a>
</span><span id="Transcript-158"><a href="#Transcript-158"><span class="linenos">158</span></a>        <span class="n">start_genomic_coordinate</span><span class="p">,</span> <span class="n">_</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">get_cds_start_stop</span><span class="p">()</span>
</span><span id="Transcript-159"><a href="#Transcript-159"><span class="linenos">159</span></a>        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">_get_transcript_position</span><span class="p">(</span><span class="n">start_genomic_coordinate</span><span class="p">)</span>
</span><span id="Transcript-160"><a href="#Transcript-160"><span class="linenos">160</span></a>
</span><span id="Transcript-161"><a href="#Transcript-161"><span class="linenos">161</span></a>    <span class="nd">@lazy</span>
</span><span id="Transcript-162"><a href="#Transcript-162"><span class="linenos">162</span></a>    <span class="k">def</span> <span class="nf">stop_codon</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">int</span><span class="p">:</span>
</span><span id="Transcript-163"><a href="#Transcript-163"><span class="linenos">163</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="Transcript-164"><a href="#Transcript-164"><span class="linenos">164</span></a><span class="sd">        The function returns the transcript position of the stop codon.</span>
</span><span id="Transcript-165"><a href="#Transcript-165"><span class="linenos">165</span></a><span class="sd">        :return: The method `stop_codon` returns an integer, which represents the transcript position of</span>
</span><span id="Transcript-166"><a href="#Transcript-166"><span class="linenos">166</span></a><span class="sd">        the stop codon.</span>
</span><span id="Transcript-167"><a href="#Transcript-167"><span class="linenos">167</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="Transcript-168"><a href="#Transcript-168"><span class="linenos">168</span></a>
</span><span id="Transcript-169"><a href="#Transcript-169"><span class="linenos">169</span></a>        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">_stop_codon_transcript_pos</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
</span><span id="Transcript-170"><a href="#Transcript-170"><span class="linenos">170</span></a>            <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">_stop_codon_transcript_pos</span>
</span><span id="Transcript-171"><a href="#Transcript-171"><span class="linenos">171</span></a>
</span><span id="Transcript-172"><a href="#Transcript-172"><span class="linenos">172</span></a>        <span class="n">_</span><span class="p">,</span> <span class="n">stop_genomic_coordinate</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">get_cds_start_stop</span><span class="p">()</span>
</span><span id="Transcript-173"><a href="#Transcript-173"><span class="linenos">173</span></a>        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">_get_transcript_position</span><span class="p">(</span><span class="n">stop_genomic_coordinate</span><span class="p">)</span>
</span><span id="Transcript-174"><a href="#Transcript-174"><span class="linenos">174</span></a>
</span><span id="Transcript-175"><a href="#Transcript-175"><span class="linenos">175</span></a>    <span class="k">def</span> <span class="nf">_get_transcript_position</span><span class="p">(</span>
</span><span id="Transcript-176"><a href="#Transcript-176"><span class="linenos">176</span></a>        <span class="bp">self</span><span class="p">,</span> <span class="n">genomic_coordinate</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span> <span class="n">label</span><span class="p">:</span> <span class="nb">str</span> <span class="o">=</span> <span class="kc">None</span>
</span><span id="Transcript-177"><a href="#Transcript-177"><span class="linenos">177</span></a>    <span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">int</span><span class="p">:</span>
</span><span id="Transcript-178"><a href="#Transcript-178"><span class="linenos">178</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="Transcript-179"><a href="#Transcript-179"><span class="linenos">179</span></a><span class="sd">        The function `_get_transcript_position` calculates the position of a genomic coordinate within a</span>
</span><span id="Transcript-180"><a href="#Transcript-180"><span class="linenos">180</span></a><span class="sd">        transcript.</span>
</span><span id="Transcript-181"><a href="#Transcript-181"><span class="linenos">181</span></a>
</span><span id="Transcript-182"><a href="#Transcript-182"><span class="linenos">182</span></a><span class="sd">        :param genomic_coordinate: The `genomic_coordinate` parameter is an integer representing a</span>
</span><span id="Transcript-183"><a href="#Transcript-183"><span class="linenos">183</span></a><span class="sd">        specific position on the genome. It is the coordinate for which we want to determine the</span>
</span><span id="Transcript-184"><a href="#Transcript-184"><span class="linenos">184</span></a><span class="sd">        corresponding position in the transcript</span>
</span><span id="Transcript-185"><a href="#Transcript-185"><span class="linenos">185</span></a><span class="sd">        :type genomic_coordinate: int</span>
</span><span id="Transcript-186"><a href="#Transcript-186"><span class="linenos">186</span></a><span class="sd">        :param label: The `label` parameter is an optional string that represents a label or identifier</span>
</span><span id="Transcript-187"><a href="#Transcript-187"><span class="linenos">187</span></a><span class="sd">        for the genomic coordinate. It is used in the error message if the genomic coordinate is not</span>
</span><span id="Transcript-188"><a href="#Transcript-188"><span class="linenos">188</span></a><span class="sd">        found in any of the exons. If no label is provided, a default label is generated using the value</span>
</span><span id="Transcript-189"><a href="#Transcript-189"><span class="linenos">189</span></a><span class="sd">        of the `genomic</span>
</span><span id="Transcript-190"><a href="#Transcript-190"><span class="linenos">190</span></a><span class="sd">        :type label: str</span>
</span><span id="Transcript-191"><a href="#Transcript-191"><span class="linenos">191</span></a><span class="sd">        :return: an integer representing the position of the genomic coordinate within the transcript.</span>
</span><span id="Transcript-192"><a href="#Transcript-192"><span class="linenos">192</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="Transcript-193"><a href="#Transcript-193"><span class="linenos">193</span></a>
</span><span id="Transcript-194"><a href="#Transcript-194"><span class="linenos">194</span></a>        <span class="n">transcript_strand</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">is_forward_strand</span>
</span><span id="Transcript-195"><a href="#Transcript-195"><span class="linenos">195</span></a>
</span><span id="Transcript-196"><a href="#Transcript-196"><span class="linenos">196</span></a>        <span class="n">cdna_offset</span> <span class="o">=</span> <span class="mi">0</span>
</span><span id="Transcript-197"><a href="#Transcript-197"><span class="linenos">197</span></a>        <span class="k">for</span> <span class="n">cdna_match</span> <span class="ow">in</span> <span class="bp">self</span><span class="o">.</span><span class="n">ordered_cdna_match</span><span class="p">:</span>
</span><span id="Transcript-198"><a href="#Transcript-198"><span class="linenos">198</span></a>            <span class="n">start</span> <span class="o">=</span> <span class="n">cdna_match</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_start</span>
</span><span id="Transcript-199"><a href="#Transcript-199"><span class="linenos">199</span></a>            <span class="n">stop</span> <span class="o">=</span> <span class="n">cdna_match</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_stop</span>
</span><span id="Transcript-200"><a href="#Transcript-200"><span class="linenos">200</span></a>
</span><span id="Transcript-201"><a href="#Transcript-201"><span class="linenos">201</span></a>            <span class="k">if</span> <span class="n">start</span> <span class="o">&lt;=</span> <span class="n">genomic_coordinate</span> <span class="o">&lt;=</span> <span class="n">stop</span><span class="p">:</span>
</span><span id="Transcript-202"><a href="#Transcript-202"><span class="linenos">202</span></a>                <span class="c1"># We&#39;re inside this match</span>
</span><span id="Transcript-203"><a href="#Transcript-203"><span class="linenos">203</span></a>                <span class="k">if</span> <span class="n">transcript_strand</span><span class="p">:</span>
</span><span id="Transcript-204"><a href="#Transcript-204"><span class="linenos">204</span></a>                    <span class="n">position</span> <span class="o">=</span> <span class="n">genomic_coordinate</span> <span class="o">-</span> <span class="n">start</span>
</span><span id="Transcript-205"><a href="#Transcript-205"><span class="linenos">205</span></a>                <span class="k">else</span><span class="p">:</span>
</span><span id="Transcript-206"><a href="#Transcript-206"><span class="linenos">206</span></a>                    <span class="n">position</span> <span class="o">=</span> <span class="n">stop</span> <span class="o">-</span> <span class="n">genomic_coordinate</span>
</span><span id="Transcript-207"><a href="#Transcript-207"><span class="linenos">207</span></a>                <span class="k">return</span> <span class="n">cdna_offset</span> <span class="o">+</span> <span class="n">position</span> <span class="o">+</span> <span class="n">cdna_match</span><span class="o">.</span><span class="n">get_offset</span><span class="p">(</span><span class="n">position</span><span class="p">)</span>
</span><span id="Transcript-208"><a href="#Transcript-208"><span class="linenos">208</span></a>            <span class="k">else</span><span class="p">:</span>
</span><span id="Transcript-209"><a href="#Transcript-209"><span class="linenos">209</span></a>                <span class="n">cdna_offset</span> <span class="o">+=</span> <span class="n">cdna_match</span><span class="o">.</span><span class="n">length</span>
</span><span id="Transcript-210"><a href="#Transcript-210"><span class="linenos">210</span></a>        <span class="k">if</span> <span class="n">label</span> <span class="ow">is</span> <span class="kc">None</span><span class="p">:</span>
</span><span id="Transcript-211"><a href="#Transcript-211"><span class="linenos">211</span></a>            <span class="n">label</span> <span class="o">=</span> <span class="s2">&quot;Genomic coordinate: </span><span class="si">%d</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="n">genomic_coordinate</span>
</span><span id="Transcript-212"><a href="#Transcript-212"><span class="linenos">212</span></a>        <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="s2">&quot;</span><span class="si">%s</span><span class="s2"> is not in any of the exons&quot;</span> <span class="o">%</span> <span class="n">label</span><span class="p">)</span>
</span><span id="Transcript-213"><a href="#Transcript-213"><span class="linenos">213</span></a>
</span><span id="Transcript-214"><a href="#Transcript-214"><span class="linenos">214</span></a>    <span class="k">def</span> <span class="nf">cdna_to_genomic_coord</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">coord</span><span class="p">:</span> <span class="nb">object</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">int</span><span class="p">:</span>
</span><span id="Transcript-215"><a href="#Transcript-215"><span class="linenos">215</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="Transcript-216"><a href="#Transcript-216"><span class="linenos">216</span></a><span class="sd">        The function `cdna_to_genomic_coord` converts a HGVS cDNA coordinate to a genomic coordinate.</span>
</span><span id="Transcript-217"><a href="#Transcript-217"><span class="linenos">217</span></a>
</span><span id="Transcript-218"><a href="#Transcript-218"><span class="linenos">218</span></a><span class="sd">        :param coord: The parameter `coord` is an object that represents a cDNA coordinate. It is used</span>
</span><span id="Transcript-219"><a href="#Transcript-219"><span class="linenos">219</span></a><span class="sd">        to specify a position along a cDNA sequence</span>
</span><span id="Transcript-220"><a href="#Transcript-220"><span class="linenos">220</span></a><span class="sd">        :type coord: object</span>
</span><span id="Transcript-221"><a href="#Transcript-221"><span class="linenos">221</span></a><span class="sd">        :return: an integer value, which represents the genomic coordinate corresponding to the given</span>
</span><span id="Transcript-222"><a href="#Transcript-222"><span class="linenos">222</span></a><span class="sd">        cDNA coordinate.</span>
</span><span id="Transcript-223"><a href="#Transcript-223"><span class="linenos">223</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="Transcript-224"><a href="#Transcript-224"><span class="linenos">224</span></a>
</span><span id="Transcript-225"><a href="#Transcript-225"><span class="linenos">225</span></a>        <span class="n">transcript_strand</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">is_forward_strand</span>
</span><span id="Transcript-226"><a href="#Transcript-226"><span class="linenos">226</span></a>
</span><span id="Transcript-227"><a href="#Transcript-227"><span class="linenos">227</span></a>        <span class="c1"># compute starting position along spliced transcript.</span>
</span><span id="Transcript-228"><a href="#Transcript-228"><span class="linenos">228</span></a>        <span class="k">if</span> <span class="n">coord</span><span class="o">.</span><span class="n">landmark</span> <span class="o">==</span> <span class="n">CDNA_START_CODON</span><span class="p">:</span>
</span><span id="Transcript-229"><a href="#Transcript-229"><span class="linenos">229</span></a>            <span class="n">utr5p</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">start_codon</span> <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">is_coding</span> <span class="k">else</span> <span class="mi">0</span>
</span><span id="Transcript-230"><a href="#Transcript-230"><span class="linenos">230</span></a>
</span><span id="Transcript-231"><a href="#Transcript-231"><span class="linenos">231</span></a>            <span class="k">if</span> <span class="n">coord</span><span class="o">.</span><span class="n">coord</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">:</span>
</span><span id="Transcript-232"><a href="#Transcript-232"><span class="linenos">232</span></a>                <span class="n">cdna_pos</span> <span class="o">=</span> <span class="n">utr5p</span> <span class="o">+</span> <span class="n">coord</span><span class="o">.</span><span class="n">coord</span>
</span><span id="Transcript-233"><a href="#Transcript-233"><span class="linenos">233</span></a>            <span class="k">else</span><span class="p">:</span>
</span><span id="Transcript-234"><a href="#Transcript-234"><span class="linenos">234</span></a>                <span class="n">cdna_pos</span> <span class="o">=</span> <span class="n">utr5p</span> <span class="o">+</span> <span class="n">coord</span><span class="o">.</span><span class="n">coord</span> <span class="o">+</span> <span class="mi">1</span>
</span><span id="Transcript-235"><a href="#Transcript-235"><span class="linenos">235</span></a>        <span class="k">elif</span> <span class="n">coord</span><span class="o">.</span><span class="n">landmark</span> <span class="o">==</span> <span class="n">CDNA_STOP_CODON</span><span class="p">:</span>
</span><span id="Transcript-236"><a href="#Transcript-236"><span class="linenos">236</span></a>            <span class="k">if</span> <span class="n">coord</span><span class="o">.</span><span class="n">coord</span> <span class="o">&lt;</span> <span class="mi">0</span><span class="p">:</span>
</span><span id="Transcript-237"><a href="#Transcript-237"><span class="linenos">237</span></a>                <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span>
</span><span id="Transcript-238"><a href="#Transcript-238"><span class="linenos">238</span></a>                    <span class="s2">&quot;CDNACoord cannot have a negative coord and &quot;</span>
</span><span id="Transcript-239"><a href="#Transcript-239"><span class="linenos">239</span></a>                    <span class="s2">&quot;landmark CDNA_STOP_CODON&quot;</span>
</span><span id="Transcript-240"><a href="#Transcript-240"><span class="linenos">240</span></a>                <span class="p">)</span>
</span><span id="Transcript-241"><a href="#Transcript-241"><span class="linenos">241</span></a>            <span class="n">cdna_pos</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">stop_codon</span> <span class="o">+</span> <span class="n">coord</span><span class="o">.</span><span class="n">coord</span>
</span><span id="Transcript-242"><a href="#Transcript-242"><span class="linenos">242</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="Transcript-243"><a href="#Transcript-243"><span class="linenos">243</span></a>            <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="s1">&#39;unknown CDNACoord landmark &quot;</span><span class="si">%s</span><span class="s1">&quot;&#39;</span> <span class="o">%</span> <span class="n">coord</span><span class="o">.</span><span class="n">landmark</span><span class="p">)</span>
</span><span id="Transcript-244"><a href="#Transcript-244"><span class="linenos">244</span></a>
</span><span id="Transcript-245"><a href="#Transcript-245"><span class="linenos">245</span></a>        <span class="c1"># 5&#39; flanking sequence (no need to account for gaps)</span>
</span><span id="Transcript-246"><a href="#Transcript-246"><span class="linenos">246</span></a>        <span class="k">if</span> <span class="n">cdna_pos</span> <span class="o">&lt;</span> <span class="mi">1</span><span class="p">:</span>
</span><span id="Transcript-247"><a href="#Transcript-247"><span class="linenos">247</span></a>            <span class="k">if</span> <span class="n">transcript_strand</span><span class="p">:</span>
</span><span id="Transcript-248"><a href="#Transcript-248"><span class="linenos">248</span></a>                <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_start</span> <span class="o">+</span> <span class="n">cdna_pos</span>
</span><span id="Transcript-249"><a href="#Transcript-249"><span class="linenos">249</span></a>            <span class="k">else</span><span class="p">:</span>
</span><span id="Transcript-250"><a href="#Transcript-250"><span class="linenos">250</span></a>                <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_stop</span> <span class="o">-</span> <span class="n">cdna_pos</span> <span class="o">+</span> <span class="mi">1</span>
</span><span id="Transcript-251"><a href="#Transcript-251"><span class="linenos">251</span></a>
</span><span id="Transcript-252"><a href="#Transcript-252"><span class="linenos">252</span></a>        <span class="c1"># Walk along transcript until we find an exon that contains cdna_pos.</span>
</span><span id="Transcript-253"><a href="#Transcript-253"><span class="linenos">253</span></a>        <span class="k">for</span> <span class="n">cdna_match</span> <span class="ow">in</span> <span class="bp">self</span><span class="o">.</span><span class="n">ordered_cdna_match</span><span class="p">:</span>
</span><span id="Transcript-254"><a href="#Transcript-254"><span class="linenos">254</span></a>            <span class="k">if</span> <span class="n">cdna_match</span><span class="o">.</span><span class="n">cdna_start</span> <span class="o">&lt;=</span> <span class="n">cdna_pos</span> <span class="o">&lt;=</span> <span class="n">cdna_match</span><span class="o">.</span><span class="n">cdna_end</span><span class="p">:</span>
</span><span id="Transcript-255"><a href="#Transcript-255"><span class="linenos">255</span></a>                <span class="n">match_pos</span> <span class="o">=</span> <span class="n">cdna_pos</span> <span class="o">-</span> <span class="n">cdna_match</span><span class="o">.</span><span class="n">cdna_start</span>
</span><span id="Transcript-256"><a href="#Transcript-256"><span class="linenos">256</span></a>                <span class="n">match_pos</span> <span class="o">-=</span> <span class="n">cdna_match</span><span class="o">.</span><span class="n">get_offset</span><span class="p">(</span><span class="n">match_pos</span><span class="p">)</span>
</span><span id="Transcript-257"><a href="#Transcript-257"><span class="linenos">257</span></a>                <span class="c1"># Compute genomic coordinate using offset.</span>
</span><span id="Transcript-258"><a href="#Transcript-258"><span class="linenos">258</span></a>                <span class="k">if</span> <span class="n">transcript_strand</span><span class="p">:</span>
</span><span id="Transcript-259"><a href="#Transcript-259"><span class="linenos">259</span></a>                    <span class="c1"># Plus strand.</span>
</span><span id="Transcript-260"><a href="#Transcript-260"><span class="linenos">260</span></a>                    <span class="n">start</span> <span class="o">=</span> <span class="n">cdna_match</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_start</span> <span class="o">+</span> <span class="mi">1</span>
</span><span id="Transcript-261"><a href="#Transcript-261"><span class="linenos">261</span></a>                    <span class="k">return</span> <span class="n">start</span> <span class="o">+</span> <span class="n">match_pos</span> <span class="o">+</span> <span class="n">coord</span><span class="o">.</span><span class="n">offset</span>
</span><span id="Transcript-262"><a href="#Transcript-262"><span class="linenos">262</span></a>                <span class="k">else</span><span class="p">:</span>
</span><span id="Transcript-263"><a href="#Transcript-263"><span class="linenos">263</span></a>                    <span class="c1"># Minus strand.</span>
</span><span id="Transcript-264"><a href="#Transcript-264"><span class="linenos">264</span></a>                    <span class="n">end</span> <span class="o">=</span> <span class="n">cdna_match</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_stop</span>
</span><span id="Transcript-265"><a href="#Transcript-265"><span class="linenos">265</span></a>                    <span class="k">return</span> <span class="n">end</span> <span class="o">-</span> <span class="n">match_pos</span> <span class="o">-</span> <span class="n">coord</span><span class="o">.</span><span class="n">offset</span>
</span><span id="Transcript-266"><a href="#Transcript-266"><span class="linenos">266</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="Transcript-267"><a href="#Transcript-267"><span class="linenos">267</span></a>            <span class="c1"># 3&#39; flanking sequence (no need to account for gaps)</span>
</span><span id="Transcript-268"><a href="#Transcript-268"><span class="linenos">268</span></a>            <span class="k">if</span> <span class="n">transcript_strand</span><span class="p">:</span>
</span><span id="Transcript-269"><a href="#Transcript-269"><span class="linenos">269</span></a>                <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">cds_position</span><span class="o">.</span><span class="n">chrom_stop</span> <span class="o">+</span> <span class="n">coord</span><span class="o">.</span><span class="n">coord</span>
</span><span id="Transcript-270"><a href="#Transcript-270"><span class="linenos">270</span></a>            <span class="k">else</span><span class="p">:</span>
</span><span id="Transcript-271"><a href="#Transcript-271"><span class="linenos">271</span></a>                <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">cds_position</span><span class="o">.</span><span class="n">chrom_start</span> <span class="o">+</span> <span class="mi">1</span> <span class="o">-</span> <span class="n">coord</span><span class="o">.</span><span class="n">coord</span>
</span><span id="Transcript-272"><a href="#Transcript-272"><span class="linenos">272</span></a>
</span><span id="Transcript-273"><a href="#Transcript-273"><span class="linenos">273</span></a>    <span class="k">def</span> <span class="nf">genomic_to_cdna_coord</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">genomic_coord</span><span class="p">:</span> <span class="nb">int</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">object</span><span class="p">:</span>
</span><span id="Transcript-274"><a href="#Transcript-274"><span class="linenos">274</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="Transcript-275"><a href="#Transcript-275"><span class="linenos">275</span></a><span class="sd">        The function `genomic_to_cdna_coord` converts a genomic coordinate to a cDNA coordinate and</span>
</span><span id="Transcript-276"><a href="#Transcript-276"><span class="linenos">276</span></a><span class="sd">        offset, taking into account exons, strand, and coding transcript information.</span>
</span><span id="Transcript-277"><a href="#Transcript-277"><span class="linenos">277</span></a>
</span><span id="Transcript-278"><a href="#Transcript-278"><span class="linenos">278</span></a><span class="sd">        :param genomic_coord: The `genomic_coord` parameter is an integer representing a genomic</span>
</span><span id="Transcript-279"><a href="#Transcript-279"><span class="linenos">279</span></a><span class="sd">        coordinate</span>
</span><span id="Transcript-280"><a href="#Transcript-280"><span class="linenos">280</span></a><span class="sd">        :type genomic_coord: int</span>
</span><span id="Transcript-281"><a href="#Transcript-281"><span class="linenos">281</span></a><span class="sd">        :return: an object of type `CDNACoord`.</span>
</span><span id="Transcript-282"><a href="#Transcript-282"><span class="linenos">282</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="Transcript-283"><a href="#Transcript-283"><span class="linenos">283</span></a>
</span><span id="Transcript-284"><a href="#Transcript-284"><span class="linenos">284</span></a>        <span class="n">exons</span> <span class="o">=</span> <span class="p">[</span><span class="n">exon</span><span class="o">.</span><span class="n">get_as_interval</span><span class="p">()</span> <span class="k">for</span> <span class="n">exon</span> <span class="ow">in</span> <span class="bp">self</span><span class="o">.</span><span class="n">ordered_cdna_match</span><span class="p">]</span>
</span><span id="Transcript-285"><a href="#Transcript-285"><span class="linenos">285</span></a>
</span><span id="Transcript-286"><a href="#Transcript-286"><span class="linenos">286</span></a>        <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">exons</span><span class="p">)</span> <span class="o">==</span> <span class="mi">0</span><span class="p">:</span>
</span><span id="Transcript-287"><a href="#Transcript-287"><span class="linenos">287</span></a>            <span class="k">return</span> <span class="kc">None</span>
</span><span id="Transcript-288"><a href="#Transcript-288"><span class="linenos">288</span></a>
</span><span id="Transcript-289"><a href="#Transcript-289"><span class="linenos">289</span></a>        <span class="n">strand</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">strand</span>
</span><span id="Transcript-290"><a href="#Transcript-290"><span class="linenos">290</span></a>        <span class="n">distances</span> <span class="o">=</span> <span class="p">[</span><span class="n">exon</span><span class="o">.</span><span class="n">distance</span><span class="p">(</span><span class="n">genomic_coord</span><span class="p">)</span> <span class="k">for</span> <span class="n">exon</span> <span class="ow">in</span> <span class="n">exons</span><span class="p">]</span>
</span><span id="Transcript-291"><a href="#Transcript-291"><span class="linenos">291</span></a>        <span class="n">min_distance_to_exon</span> <span class="o">=</span> <span class="nb">min</span><span class="p">(</span><span class="nb">map</span><span class="p">(</span><span class="nb">abs</span><span class="p">,</span> <span class="n">distances</span><span class="p">))</span>
</span><span id="Transcript-292"><a href="#Transcript-292"><span class="linenos">292</span></a>        <span class="k">if</span> <span class="n">min_distance_to_exon</span><span class="p">:</span>
</span><span id="Transcript-293"><a href="#Transcript-293"><span class="linenos">293</span></a>            <span class="c1"># We&#39;re outside of exon - so need to find closest point</span>
</span><span id="Transcript-294"><a href="#Transcript-294"><span class="linenos">294</span></a>            <span class="k">for</span> <span class="n">exon</span> <span class="ow">in</span> <span class="n">exons</span><span class="p">:</span>
</span><span id="Transcript-295"><a href="#Transcript-295"><span class="linenos">295</span></a>                <span class="n">distance</span> <span class="o">=</span> <span class="n">exon</span><span class="o">.</span><span class="n">distance</span><span class="p">(</span><span class="n">genomic_coord</span><span class="p">)</span>
</span><span id="Transcript-296"><a href="#Transcript-296"><span class="linenos">296</span></a>                <span class="k">if</span> <span class="nb">abs</span><span class="p">(</span><span class="n">distance</span><span class="p">)</span> <span class="o">==</span> <span class="n">min_distance_to_exon</span><span class="p">:</span>
</span><span id="Transcript-297"><a href="#Transcript-297"><span class="linenos">297</span></a>
</span><span id="Transcript-298"><a href="#Transcript-298"><span class="linenos">298</span></a>                    <span class="c1"># Outside the exon.</span>
</span><span id="Transcript-299"><a href="#Transcript-299"><span class="linenos">299</span></a>                    <span class="k">if</span> <span class="n">distance</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">:</span>
</span><span id="Transcript-300"><a href="#Transcript-300"><span class="linenos">300</span></a>                        <span class="n">genomic_nearest_exon</span> <span class="o">=</span> <span class="n">exon</span><span class="o">.</span><span class="n">chrom_start</span> <span class="o">+</span> <span class="mi">1</span>
</span><span id="Transcript-301"><a href="#Transcript-301"><span class="linenos">301</span></a>                    <span class="k">else</span><span class="p">:</span>
</span><span id="Transcript-302"><a href="#Transcript-302"><span class="linenos">302</span></a>                        <span class="n">genomic_nearest_exon</span> <span class="o">=</span> <span class="n">exon</span><span class="o">.</span><span class="n">chrom_end</span>
</span><span id="Transcript-303"><a href="#Transcript-303"><span class="linenos">303</span></a>
</span><span id="Transcript-304"><a href="#Transcript-304"><span class="linenos">304</span></a>                    <span class="k">if</span> <span class="n">strand</span> <span class="o">==</span> <span class="s2">&quot;+&quot;</span><span class="p">:</span>
</span><span id="Transcript-305"><a href="#Transcript-305"><span class="linenos">305</span></a>                        <span class="n">distance</span> <span class="o">*=</span> <span class="o">-</span><span class="mi">1</span>
</span><span id="Transcript-306"><a href="#Transcript-306"><span class="linenos">306</span></a>
</span><span id="Transcript-307"><a href="#Transcript-307"><span class="linenos">307</span></a>                    <span class="n">nearest_exon_coord</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_exon_genomic_to_cdna_coord</span><span class="p">(</span>
</span><span id="Transcript-308"><a href="#Transcript-308"><span class="linenos">308</span></a>                        <span class="n">genomic_nearest_exon</span>
</span><span id="Transcript-309"><a href="#Transcript-309"><span class="linenos">309</span></a>                    <span class="p">)</span>
</span><span id="Transcript-310"><a href="#Transcript-310"><span class="linenos">310</span></a>
</span><span id="Transcript-311"><a href="#Transcript-311"><span class="linenos">311</span></a>                    <span class="c1"># If outside transcript, don&#39;t use offset.</span>
</span><span id="Transcript-312"><a href="#Transcript-312"><span class="linenos">312</span></a>                    <span class="k">if</span> <span class="p">(</span>
</span><span id="Transcript-313"><a href="#Transcript-313"><span class="linenos">313</span></a>                        <span class="n">genomic_coord</span> <span class="o">&lt;</span> <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_start</span> <span class="o">+</span> <span class="mi">1</span>
</span><span id="Transcript-314"><a href="#Transcript-314"><span class="linenos">314</span></a>                        <span class="ow">or</span> <span class="n">genomic_coord</span> <span class="o">&gt;</span> <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_stop</span>
</span><span id="Transcript-315"><a href="#Transcript-315"><span class="linenos">315</span></a>                    <span class="p">):</span>
</span><span id="Transcript-316"><a href="#Transcript-316"><span class="linenos">316</span></a>                        <span class="n">nearest_exon_coord</span> <span class="o">+=</span> <span class="n">distance</span>
</span><span id="Transcript-317"><a href="#Transcript-317"><span class="linenos">317</span></a>                        <span class="n">distance</span> <span class="o">=</span> <span class="mi">0</span>
</span><span id="Transcript-318"><a href="#Transcript-318"><span class="linenos">318</span></a>                    <span class="n">cdna_coord</span> <span class="o">=</span> <span class="n">CDNACoord</span><span class="p">(</span><span class="n">nearest_exon_coord</span><span class="p">,</span> <span class="n">distance</span><span class="p">)</span>
</span><span id="Transcript-319"><a href="#Transcript-319"><span class="linenos">319</span></a>                    <span class="k">break</span>
</span><span id="Transcript-320"><a href="#Transcript-320"><span class="linenos">320</span></a>            <span class="k">else</span><span class="p">:</span>
</span><span id="Transcript-321"><a href="#Transcript-321"><span class="linenos">321</span></a>                <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="s2">&quot;Could not find closest exon!&quot;</span><span class="p">)</span>  <span class="c1"># Should never happen</span>
</span><span id="Transcript-322"><a href="#Transcript-322"><span class="linenos">322</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="Transcript-323"><a href="#Transcript-323"><span class="linenos">323</span></a>            <span class="n">coord</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_exon_genomic_to_cdna_coord</span><span class="p">(</span><span class="n">genomic_coord</span><span class="p">)</span>
</span><span id="Transcript-324"><a href="#Transcript-324"><span class="linenos">324</span></a>            <span class="n">cdna_coord</span> <span class="o">=</span> <span class="n">CDNACoord</span><span class="p">(</span><span class="n">coord</span><span class="p">,</span> <span class="mi">0</span><span class="p">)</span>
</span><span id="Transcript-325"><a href="#Transcript-325"><span class="linenos">325</span></a>
</span><span id="Transcript-326"><a href="#Transcript-326"><span class="linenos">326</span></a>        <span class="c1"># Adjust coordinates for coding transcript.</span>
</span><span id="Transcript-327"><a href="#Transcript-327"><span class="linenos">327</span></a>        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">is_coding</span><span class="p">:</span>
</span><span id="Transcript-328"><a href="#Transcript-328"><span class="linenos">328</span></a>            <span class="c1"># Detect if position before start codon.</span>
</span><span id="Transcript-329"><a href="#Transcript-329"><span class="linenos">329</span></a>            <span class="n">utr5p</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">start_codon</span>
</span><span id="Transcript-330"><a href="#Transcript-330"><span class="linenos">330</span></a>            <span class="n">cdna_coord</span><span class="o">.</span><span class="n">coord</span> <span class="o">-=</span> <span class="n">utr5p</span>
</span><span id="Transcript-331"><a href="#Transcript-331"><span class="linenos">331</span></a>            <span class="k">if</span> <span class="n">cdna_coord</span><span class="o">.</span><span class="n">coord</span> <span class="o">&lt;=</span> <span class="mi">0</span><span class="p">:</span>
</span><span id="Transcript-332"><a href="#Transcript-332"><span class="linenos">332</span></a>                <span class="n">cdna_coord</span><span class="o">.</span><span class="n">coord</span> <span class="o">-=</span> <span class="mi">1</span>
</span><span id="Transcript-333"><a href="#Transcript-333"><span class="linenos">333</span></a>            <span class="k">else</span><span class="p">:</span>
</span><span id="Transcript-334"><a href="#Transcript-334"><span class="linenos">334</span></a>                <span class="c1"># Detect if position is after stop_codon.</span>
</span><span id="Transcript-335"><a href="#Transcript-335"><span class="linenos">335</span></a>                <span class="n">stop_codon</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">stop_codon</span>
</span><span id="Transcript-336"><a href="#Transcript-336"><span class="linenos">336</span></a>                <span class="n">stop_codon</span> <span class="o">-=</span> <span class="n">utr5p</span>
</span><span id="Transcript-337"><a href="#Transcript-337"><span class="linenos">337</span></a>                <span class="k">if</span> <span class="p">(</span>
</span><span id="Transcript-338"><a href="#Transcript-338"><span class="linenos">338</span></a>                    <span class="n">cdna_coord</span><span class="o">.</span><span class="n">coord</span> <span class="o">&gt;</span> <span class="n">stop_codon</span>
</span><span id="Transcript-339"><a href="#Transcript-339"><span class="linenos">339</span></a>                    <span class="ow">or</span> <span class="n">cdna_coord</span><span class="o">.</span><span class="n">coord</span> <span class="o">==</span> <span class="n">stop_codon</span>
</span><span id="Transcript-340"><a href="#Transcript-340"><span class="linenos">340</span></a>                    <span class="ow">and</span> <span class="n">cdna_coord</span><span class="o">.</span><span class="n">offset</span> <span class="o">&gt;</span> <span class="mi">0</span>
</span><span id="Transcript-341"><a href="#Transcript-341"><span class="linenos">341</span></a>                <span class="p">):</span>
</span><span id="Transcript-342"><a href="#Transcript-342"><span class="linenos">342</span></a>                    <span class="n">cdna_coord</span><span class="o">.</span><span class="n">coord</span> <span class="o">-=</span> <span class="n">stop_codon</span>
</span><span id="Transcript-343"><a href="#Transcript-343"><span class="linenos">343</span></a>                    <span class="n">cdna_coord</span><span class="o">.</span><span class="n">landmark</span> <span class="o">=</span> <span class="n">CDNA_STOP_CODON</span>
</span><span id="Transcript-344"><a href="#Transcript-344"><span class="linenos">344</span></a>
</span><span id="Transcript-345"><a href="#Transcript-345"><span class="linenos">345</span></a>        <span class="k">return</span> <span class="n">cdna_coord</span>
</span><span id="Transcript-346"><a href="#Transcript-346"><span class="linenos">346</span></a>
</span><span id="Transcript-347"><a href="#Transcript-347"><span class="linenos">347</span></a>    <span class="k">def</span> <span class="nf">_exon_genomic_to_cdna_coord</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">genomic_coord</span><span class="p">:</span> <span class="nb">int</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">int</span><span class="p">:</span>
</span><span id="Transcript-348"><a href="#Transcript-348"><span class="linenos">348</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="Transcript-349"><a href="#Transcript-349"><span class="linenos">349</span></a><span class="sd">        The function `_exon_genomic_to_cdna_coord` converts a genomic coordinate to a cDNA coordinate</span>
</span><span id="Transcript-350"><a href="#Transcript-350"><span class="linenos">350</span></a><span class="sd">        within an exon.</span>
</span><span id="Transcript-351"><a href="#Transcript-351"><span class="linenos">351</span></a>
</span><span id="Transcript-352"><a href="#Transcript-352"><span class="linenos">352</span></a><span class="sd">        :param genomic_coord: The `genomic_coord` parameter represents the genomic coordinate that you</span>
</span><span id="Transcript-353"><a href="#Transcript-353"><span class="linenos">353</span></a><span class="sd">        want to convert to cDNA coordinate. It is an integer value that specifies the position on the</span>
</span><span id="Transcript-354"><a href="#Transcript-354"><span class="linenos">354</span></a><span class="sd">        genome</span>
</span><span id="Transcript-355"><a href="#Transcript-355"><span class="linenos">355</span></a><span class="sd">        :type genomic_coord: int</span>
</span><span id="Transcript-356"><a href="#Transcript-356"><span class="linenos">356</span></a><span class="sd">        :return: the cDNA coordinate corresponding to the given genomic coordinate.</span>
</span><span id="Transcript-357"><a href="#Transcript-357"><span class="linenos">357</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="Transcript-358"><a href="#Transcript-358"><span class="linenos">358</span></a>
</span><span id="Transcript-359"><a href="#Transcript-359"><span class="linenos">359</span></a>        <span class="n">cdna_offset</span> <span class="o">=</span> <span class="mi">0</span>
</span><span id="Transcript-360"><a href="#Transcript-360"><span class="linenos">360</span></a>        <span class="k">for</span> <span class="n">cdna_match</span> <span class="ow">in</span> <span class="bp">self</span><span class="o">.</span><span class="n">ordered_cdna_match</span><span class="p">:</span>
</span><span id="Transcript-361"><a href="#Transcript-361"><span class="linenos">361</span></a>            <span class="c1"># Inside the exon.</span>
</span><span id="Transcript-362"><a href="#Transcript-362"><span class="linenos">362</span></a>            <span class="k">if</span> <span class="p">(</span>
</span><span id="Transcript-363"><a href="#Transcript-363"><span class="linenos">363</span></a>                <span class="n">cdna_match</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_start</span>
</span><span id="Transcript-364"><a href="#Transcript-364"><span class="linenos">364</span></a>                <span class="o">&lt;=</span> <span class="n">genomic_coord</span>
</span><span id="Transcript-365"><a href="#Transcript-365"><span class="linenos">365</span></a>                <span class="o">&lt;=</span> <span class="n">cdna_match</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_stop</span>
</span><span id="Transcript-366"><a href="#Transcript-366"><span class="linenos">366</span></a>            <span class="p">):</span>
</span><span id="Transcript-367"><a href="#Transcript-367"><span class="linenos">367</span></a>                <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">strand</span> <span class="o">==</span> <span class="s2">&quot;+&quot;</span><span class="p">:</span>
</span><span id="Transcript-368"><a href="#Transcript-368"><span class="linenos">368</span></a>                    <span class="n">position</span> <span class="o">=</span> <span class="n">genomic_coord</span> <span class="o">-</span> <span class="p">(</span><span class="n">cdna_match</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_start</span> <span class="o">+</span> <span class="mi">1</span><span class="p">)</span>
</span><span id="Transcript-369"><a href="#Transcript-369"><span class="linenos">369</span></a>                <span class="k">else</span><span class="p">:</span>
</span><span id="Transcript-370"><a href="#Transcript-370"><span class="linenos">370</span></a>                    <span class="n">position</span> <span class="o">=</span> <span class="n">cdna_match</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_stop</span> <span class="o">-</span> <span class="n">genomic_coord</span>
</span><span id="Transcript-371"><a href="#Transcript-371"><span class="linenos">371</span></a>                <span class="n">offset</span> <span class="o">=</span> <span class="n">cdna_match</span><span class="o">.</span><span class="n">get_offset</span><span class="p">(</span><span class="n">position</span><span class="p">,</span> <span class="n">validate</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
</span><span id="Transcript-372"><a href="#Transcript-372"><span class="linenos">372</span></a>                <span class="k">return</span> <span class="n">cdna_offset</span> <span class="o">+</span> <span class="n">position</span> <span class="o">+</span> <span class="n">offset</span> <span class="o">+</span> <span class="mi">1</span>
</span><span id="Transcript-373"><a href="#Transcript-373"><span class="linenos">373</span></a>            <span class="n">cdna_offset</span> <span class="o">+=</span> <span class="n">cdna_match</span><span class="o">.</span><span class="n">length</span>
</span><span id="Transcript-374"><a href="#Transcript-374"><span class="linenos">374</span></a>
</span><span id="Transcript-375"><a href="#Transcript-375"><span class="linenos">375</span></a>        <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Couldn&#39;t find </span><span class="si">{</span><span class="n">genomic_coord</span><span class="si">=}</span><span class="s2">!&quot;</span><span class="p">)</span>
</span><span id="Transcript-376"><a href="#Transcript-376"><span class="linenos">376</span></a>
</span><span id="Transcript-377"><a href="#Transcript-377"><span class="linenos">377</span></a>    <span class="k">def</span> <span class="nf">find_exon_number</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">offset</span><span class="p">:</span> <span class="nb">int</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">int</span><span class="p">:</span>
</span><span id="Transcript-378"><a href="#Transcript-378"><span class="linenos">378</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="Transcript-379"><a href="#Transcript-379"><span class="linenos">379</span></a><span class="sd">        The function `find_exon_number` returns the exon number for a given position.</span>
</span><span id="Transcript-380"><a href="#Transcript-380"><span class="linenos">380</span></a>
</span><span id="Transcript-381"><a href="#Transcript-381"><span class="linenos">381</span></a><span class="sd">        :param offset: The offset parameter represents a position in the genome. It is an integer value</span>
</span><span id="Transcript-382"><a href="#Transcript-382"><span class="linenos">382</span></a><span class="sd">        that indicates the position of interest within the genome</span>
</span><span id="Transcript-383"><a href="#Transcript-383"><span class="linenos">383</span></a><span class="sd">        :type offset: int</span>
</span><span id="Transcript-384"><a href="#Transcript-384"><span class="linenos">384</span></a><span class="sd">        :return: an integer value, which represents the exon number for a given position.</span>
</span><span id="Transcript-385"><a href="#Transcript-385"><span class="linenos">385</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="Transcript-386"><a href="#Transcript-386"><span class="linenos">386</span></a>
</span><span id="Transcript-387"><a href="#Transcript-387"><span class="linenos">387</span></a>        <span class="n">exon_number</span> <span class="o">=</span> <span class="mi">1</span>
</span><span id="Transcript-388"><a href="#Transcript-388"><span class="linenos">388</span></a>        <span class="k">for</span> <span class="n">cdna_match</span> <span class="ow">in</span> <span class="bp">self</span><span class="o">.</span><span class="n">ordered_cdna_match</span><span class="p">:</span>
</span><span id="Transcript-389"><a href="#Transcript-389"><span class="linenos">389</span></a>            <span class="k">if</span> <span class="p">(</span>
</span><span id="Transcript-390"><a href="#Transcript-390"><span class="linenos">390</span></a>                <span class="n">cdna_match</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_start</span>
</span><span id="Transcript-391"><a href="#Transcript-391"><span class="linenos">391</span></a>                <span class="o">&lt;=</span> <span class="n">offset</span>
</span><span id="Transcript-392"><a href="#Transcript-392"><span class="linenos">392</span></a>                <span class="o">&lt;=</span> <span class="n">cdna_match</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_stop</span>
</span><span id="Transcript-393"><a href="#Transcript-393"><span class="linenos">393</span></a>            <span class="p">):</span>
</span><span id="Transcript-394"><a href="#Transcript-394"><span class="linenos">394</span></a>                <span class="k">return</span> <span class="n">exon_number</span>
</span><span id="Transcript-395"><a href="#Transcript-395"><span class="linenos">395</span></a>            <span class="n">exon_number</span> <span class="o">+=</span> <span class="mi">1</span>
</span><span id="Transcript-396"><a href="#Transcript-396"><span class="linenos">396</span></a>        <span class="k">return</span> <span class="kc">None</span>
</span></pre></div>


    

                            <div id="Transcript.__init__" class="classattr">
                                        <input id="Transcript.__init__-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr function">
            
        <span class="name">Transcript</span><span class="signature pdoc-code multiline">(<span class="param">	<span class="n">name</span><span class="p">:</span> <span class="nb">str</span>,</span><span class="param">	<span class="n">version</span><span class="p">:</span> <span class="nb">str</span>,</span><span class="param">	<span class="n">gene</span><span class="p">:</span> <span class="nb">str</span>,</span><span class="param">	<span class="n">tx_position</span><span class="p">:</span> <span class="nb">int</span>,</span><span class="param">	<span class="n">cds_position</span><span class="p">:</span> <span class="nb">int</span>,</span><span class="param">	<span class="n">is_default</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span>,</span><span class="param">	<span class="n">cdna_match</span><span class="p">:</span> <span class="nb">list</span> <span class="o">=</span> <span class="kc">None</span>,</span><span class="param">	<span class="n">start_codon_transcript_pos</span><span class="p">:</span> <span class="nb">int</span> <span class="o">=</span> <span class="kc">None</span>,</span><span class="param">	<span class="n">stop_codon_transcript_pos</span><span class="p">:</span> <span class="nb">int</span> <span class="o">=</span> <span class="kc">None</span></span>)</span>

                <label class="view-source-button" for="Transcript.__init__-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#Transcript.__init__"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="Transcript.__init__-26"><a href="#Transcript.__init__-26"><span class="linenos">26</span></a>    <span class="k">def</span> <span class="fm">__init__</span><span class="p">(</span>
</span><span id="Transcript.__init__-27"><a href="#Transcript.__init__-27"><span class="linenos">27</span></a>        <span class="bp">self</span><span class="p">,</span>
</span><span id="Transcript.__init__-28"><a href="#Transcript.__init__-28"><span class="linenos">28</span></a>        <span class="n">name</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
</span><span id="Transcript.__init__-29"><a href="#Transcript.__init__-29"><span class="linenos">29</span></a>        <span class="n">version</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
</span><span id="Transcript.__init__-30"><a href="#Transcript.__init__-30"><span class="linenos">30</span></a>        <span class="n">gene</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span>
</span><span id="Transcript.__init__-31"><a href="#Transcript.__init__-31"><span class="linenos">31</span></a>        <span class="n">tx_position</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span>
</span><span id="Transcript.__init__-32"><a href="#Transcript.__init__-32"><span class="linenos">32</span></a>        <span class="n">cds_position</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span>
</span><span id="Transcript.__init__-33"><a href="#Transcript.__init__-33"><span class="linenos">33</span></a>        <span class="n">is_default</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span><span class="p">,</span>
</span><span id="Transcript.__init__-34"><a href="#Transcript.__init__-34"><span class="linenos">34</span></a>        <span class="n">cdna_match</span><span class="p">:</span> <span class="nb">list</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
</span><span id="Transcript.__init__-35"><a href="#Transcript.__init__-35"><span class="linenos">35</span></a>        <span class="n">start_codon_transcript_pos</span><span class="p">:</span> <span class="nb">int</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
</span><span id="Transcript.__init__-36"><a href="#Transcript.__init__-36"><span class="linenos">36</span></a>        <span class="n">stop_codon_transcript_pos</span><span class="p">:</span> <span class="nb">int</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span>
</span><span id="Transcript.__init__-37"><a href="#Transcript.__init__-37"><span class="linenos">37</span></a>    <span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
</span><span id="Transcript.__init__-38"><a href="#Transcript.__init__-38"><span class="linenos">38</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="Transcript.__init__-39"><a href="#Transcript.__init__-39"><span class="linenos">39</span></a><span class="sd">        The function initializes an object with various attributes related to a gene and its transcript.</span>
</span><span id="Transcript.__init__-40"><a href="#Transcript.__init__-40"><span class="linenos">40</span></a>
</span><span id="Transcript.__init__-41"><a href="#Transcript.__init__-41"><span class="linenos">41</span></a><span class="sd">        :param name: A string representing the name of the coding sequence</span>
</span><span id="Transcript.__init__-42"><a href="#Transcript.__init__-42"><span class="linenos">42</span></a><span class="sd">        :type name: str</span>
</span><span id="Transcript.__init__-43"><a href="#Transcript.__init__-43"><span class="linenos">43</span></a><span class="sd">        :param version: The `version` parameter is a string that represents the version of the object.</span>
</span><span id="Transcript.__init__-44"><a href="#Transcript.__init__-44"><span class="linenos">44</span></a><span class="sd">        It is used to track changes or updates to the object over time</span>
</span><span id="Transcript.__init__-45"><a href="#Transcript.__init__-45"><span class="linenos">45</span></a><span class="sd">        :type version: str</span>
</span><span id="Transcript.__init__-46"><a href="#Transcript.__init__-46"><span class="linenos">46</span></a><span class="sd">        :param gene: The `gene` parameter is a string that represents the gene associated with the</span>
</span><span id="Transcript.__init__-47"><a href="#Transcript.__init__-47"><span class="linenos">47</span></a><span class="sd">        coding sequence</span>
</span><span id="Transcript.__init__-48"><a href="#Transcript.__init__-48"><span class="linenos">48</span></a><span class="sd">        :type gene: str</span>
</span><span id="Transcript.__init__-49"><a href="#Transcript.__init__-49"><span class="linenos">49</span></a><span class="sd">        :param tx_position: The `tx_position` parameter represents the position of the transcript. It is</span>
</span><span id="Transcript.__init__-50"><a href="#Transcript.__init__-50"><span class="linenos">50</span></a><span class="sd">        an integer value that indicates the position of the transcript in the genome</span>
</span><span id="Transcript.__init__-51"><a href="#Transcript.__init__-51"><span class="linenos">51</span></a><span class="sd">        :type tx_position: int</span>
</span><span id="Transcript.__init__-52"><a href="#Transcript.__init__-52"><span class="linenos">52</span></a><span class="sd">        :param cds_position: The `cds_position` parameter represents the position of the coding sequence</span>
</span><span id="Transcript.__init__-53"><a href="#Transcript.__init__-53"><span class="linenos">53</span></a><span class="sd">        (CDS) within the transcript. It is an integer value that indicates the starting position of the</span>
</span><span id="Transcript.__init__-54"><a href="#Transcript.__init__-54"><span class="linenos">54</span></a><span class="sd">        CDS within the transcript sequence</span>
</span><span id="Transcript.__init__-55"><a href="#Transcript.__init__-55"><span class="linenos">55</span></a><span class="sd">        :type cds_position: int</span>
</span><span id="Transcript.__init__-56"><a href="#Transcript.__init__-56"><span class="linenos">56</span></a><span class="sd">        :param is_default: The `is_default` parameter is a boolean flag that indicates whether the</span>
</span><span id="Transcript.__init__-57"><a href="#Transcript.__init__-57"><span class="linenos">57</span></a><span class="sd">        instance of the class is the default version of the gene. It is set to `False` by default, but</span>
</span><span id="Transcript.__init__-58"><a href="#Transcript.__init__-58"><span class="linenos">58</span></a><span class="sd">        can be set to `True` if the instance is the default version, defaults to False</span>
</span><span id="Transcript.__init__-59"><a href="#Transcript.__init__-59"><span class="linenos">59</span></a><span class="sd">        :type is_default: bool (optional)</span>
</span><span id="Transcript.__init__-60"><a href="#Transcript.__init__-60"><span class="linenos">60</span></a><span class="sd">        :param cdna_match: The `cdna_match` parameter is a list that contains the positions of the</span>
</span><span id="Transcript.__init__-61"><a href="#Transcript.__init__-61"><span class="linenos">61</span></a><span class="sd">        matching cDNA sequences. It is an optional parameter and if not provided, it defaults to an</span>
</span><span id="Transcript.__init__-62"><a href="#Transcript.__init__-62"><span class="linenos">62</span></a><span class="sd">        empty list</span>
</span><span id="Transcript.__init__-63"><a href="#Transcript.__init__-63"><span class="linenos">63</span></a><span class="sd">        :type cdna_match: list</span>
</span><span id="Transcript.__init__-64"><a href="#Transcript.__init__-64"><span class="linenos">64</span></a><span class="sd">        :param start_codon_transcript_pos: The parameter &quot;start_codon_transcript_pos&quot; is an optional</span>
</span><span id="Transcript.__init__-65"><a href="#Transcript.__init__-65"><span class="linenos">65</span></a><span class="sd">        parameter that represents the transcript position of the start codon. It is used to store the</span>
</span><span id="Transcript.__init__-66"><a href="#Transcript.__init__-66"><span class="linenos">66</span></a><span class="sd">        pre-calculated transcript position of the start codon for a specific gene</span>
</span><span id="Transcript.__init__-67"><a href="#Transcript.__init__-67"><span class="linenos">67</span></a><span class="sd">        :type start_codon_transcript_pos: int</span>
</span><span id="Transcript.__init__-68"><a href="#Transcript.__init__-68"><span class="linenos">68</span></a><span class="sd">        :param stop_codon_transcript_pos: The parameter `stop_codon_transcript_pos` is an optional</span>
</span><span id="Transcript.__init__-69"><a href="#Transcript.__init__-69"><span class="linenos">69</span></a><span class="sd">        integer that represents the transcript position of the stop codon. It is used to store the</span>
</span><span id="Transcript.__init__-70"><a href="#Transcript.__init__-70"><span class="linenos">70</span></a><span class="sd">        pre-calculated transcript coordinate of the stop codon. If not provided, it will be set to</span>
</span><span id="Transcript.__init__-71"><a href="#Transcript.__init__-71"><span class="linenos">71</span></a><span class="sd">        `None`</span>
</span><span id="Transcript.__init__-72"><a href="#Transcript.__init__-72"><span class="linenos">72</span></a><span class="sd">        :type stop_codon_transcript_pos: int</span>
</span><span id="Transcript.__init__-73"><a href="#Transcript.__init__-73"><span class="linenos">73</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="Transcript.__init__-74"><a href="#Transcript.__init__-74"><span class="linenos">74</span></a>
</span><span id="Transcript.__init__-75"><a href="#Transcript.__init__-75"><span class="linenos">75</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">name</span> <span class="o">=</span> <span class="n">name</span>
</span><span id="Transcript.__init__-76"><a href="#Transcript.__init__-76"><span class="linenos">76</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">version</span> <span class="o">=</span> <span class="n">version</span>
</span><span id="Transcript.__init__-77"><a href="#Transcript.__init__-77"><span class="linenos">77</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">gene</span> <span class="o">=</span> <span class="n">Gene</span><span class="p">(</span><span class="n">gene</span><span class="p">)</span>
</span><span id="Transcript.__init__-78"><a href="#Transcript.__init__-78"><span class="linenos">78</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span> <span class="o">=</span> <span class="n">tx_position</span>
</span><span id="Transcript.__init__-79"><a href="#Transcript.__init__-79"><span class="linenos">79</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">cds_position</span> <span class="o">=</span> <span class="n">cds_position</span>
</span><span id="Transcript.__init__-80"><a href="#Transcript.__init__-80"><span class="linenos">80</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">is_default</span> <span class="o">=</span> <span class="n">is_default</span>
</span><span id="Transcript.__init__-81"><a href="#Transcript.__init__-81"><span class="linenos">81</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">cdna_match</span> <span class="o">=</span> <span class="n">cdna_match</span> <span class="ow">or</span> <span class="p">[]</span>
</span><span id="Transcript.__init__-82"><a href="#Transcript.__init__-82"><span class="linenos">82</span></a>        <span class="c1"># Optional pre-calculated transcript coordinates</span>
</span><span id="Transcript.__init__-83"><a href="#Transcript.__init__-83"><span class="linenos">83</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">_start_codon_transcript_pos</span> <span class="o">=</span> <span class="n">start_codon_transcript_pos</span>
</span><span id="Transcript.__init__-84"><a href="#Transcript.__init__-84"><span class="linenos">84</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">_stop_codon_transcript_pos</span> <span class="o">=</span> <span class="n">stop_codon_transcript_pos</span>
</span></pre></div>


            <div class="docstring"><p>The function initializes an object with various attributes related to a gene and its transcript.</p>

<h6 id="parameters">Parameters</h6>

<ul>
<li><strong>name</strong>:  A string representing the name of the coding sequence</li>
<li><strong>version</strong>:  The <code><a href="#Transcript.version">version</a></code> parameter is a string that represents the version of the object.
It is used to track changes or updates to the object over time</li>
<li><strong>gene</strong>:  The <code><a href="#Transcript.gene">gene</a></code> parameter is a string that represents the gene associated with the
coding sequence</li>
<li><strong>tx_position</strong>:  The <code><a href="#Transcript.tx_position">tx_position</a></code> parameter represents the position of the transcript. It is
an integer value that indicates the position of the transcript in the genome</li>
<li><strong>cds_position</strong>:  The <code><a href="#Transcript.cds_position">cds_position</a></code> parameter represents the position of the coding sequence
(CDS) within the transcript. It is an integer value that indicates the starting position of the
CDS within the transcript sequence</li>
<li><strong>is_default</strong>:  The <code><a href="#Transcript.is_default">is_default</a></code> parameter is a boolean flag that indicates whether the
instance of the class is the default version of the gene. It is set to <code>False</code> by default, but
can be set to <code>True</code> if the instance is the default version, defaults to False</li>
<li><strong>cdna_match</strong>:  The <code><a href="#Transcript.cdna_match">cdna_match</a></code> parameter is a list that contains the positions of the
matching cDNA sequences. It is an optional parameter and if not provided, it defaults to an
empty list</li>
<li><strong>start_codon_transcript_pos</strong>:  The parameter "start_codon_transcript_pos" is an optional
parameter that represents the transcript position of the start codon. It is used to store the
pre-calculated transcript position of the start codon for a specific gene</li>
<li><strong>stop_codon_transcript_pos</strong>:  The parameter <code>stop_codon_transcript_pos</code> is an optional
integer that represents the transcript position of the stop codon. It is used to store the
pre-calculated transcript coordinate of the stop codon. If not provided, it will be set to
<code>None</code></li>
</ul>
</div>


                            </div>
                            <div id="Transcript.name" class="classattr">
                                <div class="attr variable">
            <span class="name">name</span>

        
    </div>
    <a class="headerlink" href="#Transcript.name"></a>
    
    

                            </div>
                            <div id="Transcript.version" class="classattr">
                                <div class="attr variable">
            <span class="name">version</span>

        
    </div>
    <a class="headerlink" href="#Transcript.version"></a>
    
    

                            </div>
                            <div id="Transcript.gene" class="classattr">
                                <div class="attr variable">
            <span class="name">gene</span>

        
    </div>
    <a class="headerlink" href="#Transcript.gene"></a>
    
    

                            </div>
                            <div id="Transcript.tx_position" class="classattr">
                                <div class="attr variable">
            <span class="name">tx_position</span>

        
    </div>
    <a class="headerlink" href="#Transcript.tx_position"></a>
    
    

                            </div>
                            <div id="Transcript.cds_position" class="classattr">
                                <div class="attr variable">
            <span class="name">cds_position</span>

        
    </div>
    <a class="headerlink" href="#Transcript.cds_position"></a>
    
    

                            </div>
                            <div id="Transcript.is_default" class="classattr">
                                <div class="attr variable">
            <span class="name">is_default</span>

        
    </div>
    <a class="headerlink" href="#Transcript.is_default"></a>
    
    

                            </div>
                            <div id="Transcript.cdna_match" class="classattr">
                                <div class="attr variable">
            <span class="name">cdna_match</span>

        
    </div>
    <a class="headerlink" href="#Transcript.cdna_match"></a>
    
    

                            </div>
                            <div id="Transcript.full_name" class="classattr">
                                        <input id="Transcript.full_name-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr variable">
            <span class="name">full_name</span><span class="annotation">: str</span>

                <label class="view-source-button" for="Transcript.full_name-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#Transcript.full_name"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="Transcript.full_name-86"><a href="#Transcript.full_name-86"><span class="linenos">86</span></a>    <span class="nd">@property</span>
</span><span id="Transcript.full_name-87"><a href="#Transcript.full_name-87"><span class="linenos">87</span></a>    <span class="k">def</span> <span class="nf">full_name</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
</span><span id="Transcript.full_name-88"><a href="#Transcript.full_name-88"><span class="linenos">88</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="Transcript.full_name-89"><a href="#Transcript.full_name-89"><span class="linenos">89</span></a><span class="sd">        The function `full_name` returns the full name of an object, including its version if it exists.</span>
</span><span id="Transcript.full_name-90"><a href="#Transcript.full_name-90"><span class="linenos">90</span></a><span class="sd">        :return: a string. If the `version` attribute of the object is not `None`, it returns a string</span>
</span><span id="Transcript.full_name-91"><a href="#Transcript.full_name-91"><span class="linenos">91</span></a><span class="sd">        in the format `name.version`. Otherwise, it returns just the `name` attribute.</span>
</span><span id="Transcript.full_name-92"><a href="#Transcript.full_name-92"><span class="linenos">92</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="Transcript.full_name-93"><a href="#Transcript.full_name-93"><span class="linenos">93</span></a>
</span><span id="Transcript.full_name-94"><a href="#Transcript.full_name-94"><span class="linenos">94</span></a>        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">version</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
</span><span id="Transcript.full_name-95"><a href="#Transcript.full_name-95"><span class="linenos">95</span></a>            <span class="k">return</span> <span class="s2">&quot;</span><span class="si">%s</span><span class="s2">.</span><span class="si">%d</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">name</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">version</span><span class="p">)</span>
</span><span id="Transcript.full_name-96"><a href="#Transcript.full_name-96"><span class="linenos">96</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="Transcript.full_name-97"><a href="#Transcript.full_name-97"><span class="linenos">97</span></a>            <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">name</span>
</span></pre></div>


            <div class="docstring"><p>The function <code><a href="#Transcript.full_name">full_name</a></code> returns the full name of an object, including its version if it exists.</p>

<h6 id="returns">Returns</h6>

<blockquote>
  <p>a string. If the <code><a href="#Transcript.version">version</a></code> attribute of the object is not <code>None</code>, it returns a string
  in the format <code>name.version</code>. Otherwise, it returns just the <code><a href="#Transcript.name">name</a></code> attribute.</p>
</blockquote>
</div>


                            </div>
                            <div id="Transcript.is_coding" class="classattr">
                                        <input id="Transcript.is_coding-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr variable">
            <span class="name">is_coding</span><span class="annotation">: bool</span>

                <label class="view-source-button" for="Transcript.is_coding-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#Transcript.is_coding"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="Transcript.is_coding-99"><a href="#Transcript.is_coding-99"><span class="linenos"> 99</span></a>    <span class="nd">@property</span>
</span><span id="Transcript.is_coding-100"><a href="#Transcript.is_coding-100"><span class="linenos">100</span></a>    <span class="k">def</span> <span class="nf">is_coding</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">bool</span><span class="p">:</span>
</span><span id="Transcript.is_coding-101"><a href="#Transcript.is_coding-101"><span class="linenos">101</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="Transcript.is_coding-102"><a href="#Transcript.is_coding-102"><span class="linenos">102</span></a><span class="sd">        The function checks if a coding transcript has a non-zero length coding sequence.</span>
</span><span id="Transcript.is_coding-103"><a href="#Transcript.is_coding-103"><span class="linenos">103</span></a><span class="sd">        :return: a boolean value indicating whether the coding transcript has a coding sequence (CDS)</span>
</span><span id="Transcript.is_coding-104"><a href="#Transcript.is_coding-104"><span class="linenos">104</span></a><span class="sd">        with a non-zero length.</span>
</span><span id="Transcript.is_coding-105"><a href="#Transcript.is_coding-105"><span class="linenos">105</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="Transcript.is_coding-106"><a href="#Transcript.is_coding-106"><span class="linenos">106</span></a>
</span><span id="Transcript.is_coding-107"><a href="#Transcript.is_coding-107"><span class="linenos">107</span></a>        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">cds_position</span><span class="o">.</span><span class="n">chrom_stop</span> <span class="o">-</span> <span class="bp">self</span><span class="o">.</span><span class="n">cds_position</span><span class="o">.</span><span class="n">chrom_start</span> <span class="o">&gt;</span> <span class="mi">0</span>
</span></pre></div>


            <div class="docstring"><p>The function checks if a coding transcript has a non-zero length coding sequence.</p>

<h6 id="returns">Returns</h6>

<blockquote>
  <p>a boolean value indicating whether the coding transcript has a coding sequence (CDS)
  with a non-zero length.</p>
</blockquote>
</div>


                            </div>
                            <div id="Transcript.strand" class="classattr">
                                        <input id="Transcript.strand-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr variable">
            <span class="name">strand</span><span class="annotation">: str</span>

                <label class="view-source-button" for="Transcript.strand-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#Transcript.strand"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="Transcript.strand-109"><a href="#Transcript.strand-109"><span class="linenos">109</span></a>    <span class="nd">@property</span>
</span><span id="Transcript.strand-110"><a href="#Transcript.strand-110"><span class="linenos">110</span></a>    <span class="k">def</span> <span class="nf">strand</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
</span><span id="Transcript.strand-111"><a href="#Transcript.strand-111"><span class="linenos">111</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="Transcript.strand-112"><a href="#Transcript.strand-112"><span class="linenos">112</span></a><span class="sd">        The function returns a string &#39;+&#39; if the tx_position is on the forward strand, and &#39;-&#39; if it is</span>
</span><span id="Transcript.strand-113"><a href="#Transcript.strand-113"><span class="linenos">113</span></a><span class="sd">        on the reverse strand.</span>
</span><span id="Transcript.strand-114"><a href="#Transcript.strand-114"><span class="linenos">114</span></a><span class="sd">        :return: a string that represents the strand of the given `self.tx_position`. If</span>
</span><span id="Transcript.strand-115"><a href="#Transcript.strand-115"><span class="linenos">115</span></a><span class="sd">        `self.tx_position.is_forward_strand` is `True`, then the string returned is &#39;+&#39;. Otherwise, the</span>
</span><span id="Transcript.strand-116"><a href="#Transcript.strand-116"><span class="linenos">116</span></a><span class="sd">        string returned is &#39;-&#39;.</span>
</span><span id="Transcript.strand-117"><a href="#Transcript.strand-117"><span class="linenos">117</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="Transcript.strand-118"><a href="#Transcript.strand-118"><span class="linenos">118</span></a>
</span><span id="Transcript.strand-119"><a href="#Transcript.strand-119"><span class="linenos">119</span></a>        <span class="k">return</span> <span class="s2">&quot;+&quot;</span> <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">is_forward_strand</span> <span class="k">else</span> <span class="s2">&quot;-&quot;</span>
</span></pre></div>


            <div class="docstring"><p>The function returns a string '+' if the tx_position is on the forward strand, and '-' if it is
on the reverse strand.</p>

<h6 id="returns">Returns</h6>

<blockquote>
  <p>a string that represents the strand of the given <code>self.tx_position</code>. If
  <code>self.tx_position.is_forward_strand</code> is <code>True</code>, then the string returned is '+'. Otherwise, the
  string returned is '-'.</p>
</blockquote>
</div>


                            </div>
                            <div id="Transcript.ordered_cdna_match" class="classattr">
                                        <input id="Transcript.ordered_cdna_match-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr function">
                    <div class="decorator">@lazy</div>

        <span class="def">def</span>
        <span class="name">ordered_cdna_match</span><span class="signature pdoc-code condensed">(<span class="param"><span class="bp">self</span></span><span class="return-annotation">) -> <span class="nb">str</span>:</span></span>

                <label class="view-source-button" for="Transcript.ordered_cdna_match-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#Transcript.ordered_cdna_match"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="Transcript.ordered_cdna_match-121"><a href="#Transcript.ordered_cdna_match-121"><span class="linenos">121</span></a>    <span class="nd">@lazy</span>
</span><span id="Transcript.ordered_cdna_match-122"><a href="#Transcript.ordered_cdna_match-122"><span class="linenos">122</span></a>    <span class="k">def</span> <span class="nf">ordered_cdna_match</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
</span><span id="Transcript.ordered_cdna_match-123"><a href="#Transcript.ordered_cdna_match-123"><span class="linenos">123</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="Transcript.ordered_cdna_match-124"><a href="#Transcript.ordered_cdna_match-124"><span class="linenos">124</span></a><span class="sd">        The function &quot;ordered_cdna_match&quot; sorts a list of cdna_match objects based on their</span>
</span><span id="Transcript.ordered_cdna_match-125"><a href="#Transcript.ordered_cdna_match-125"><span class="linenos">125</span></a><span class="sd">        tx_position.chrom_start attribute and returns the sorted list.</span>
</span><span id="Transcript.ordered_cdna_match-126"><a href="#Transcript.ordered_cdna_match-126"><span class="linenos">126</span></a><span class="sd">        :return: a sorted list of cdna_match objects.</span>
</span><span id="Transcript.ordered_cdna_match-127"><a href="#Transcript.ordered_cdna_match-127"><span class="linenos">127</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="Transcript.ordered_cdna_match-128"><a href="#Transcript.ordered_cdna_match-128"><span class="linenos">128</span></a>
</span><span id="Transcript.ordered_cdna_match-129"><a href="#Transcript.ordered_cdna_match-129"><span class="linenos">129</span></a>        <span class="n">transcript_strand</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">is_forward_strand</span>
</span><span id="Transcript.ordered_cdna_match-130"><a href="#Transcript.ordered_cdna_match-130"><span class="linenos">130</span></a>        <span class="n">cdna_match</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">cdna_match</span><span class="p">)</span>
</span><span id="Transcript.ordered_cdna_match-131"><a href="#Transcript.ordered_cdna_match-131"><span class="linenos">131</span></a>        <span class="n">cdna_match</span><span class="o">.</span><span class="n">sort</span><span class="p">(</span><span class="n">key</span><span class="o">=</span><span class="k">lambda</span> <span class="n">cm</span><span class="p">:</span> <span class="n">cm</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_start</span><span class="p">)</span>
</span><span id="Transcript.ordered_cdna_match-132"><a href="#Transcript.ordered_cdna_match-132"><span class="linenos">132</span></a>        <span class="k">if</span> <span class="ow">not</span> <span class="n">transcript_strand</span><span class="p">:</span>
</span><span id="Transcript.ordered_cdna_match-133"><a href="#Transcript.ordered_cdna_match-133"><span class="linenos">133</span></a>            <span class="n">cdna_match</span><span class="o">.</span><span class="n">reverse</span><span class="p">()</span>
</span><span id="Transcript.ordered_cdna_match-134"><a href="#Transcript.ordered_cdna_match-134"><span class="linenos">134</span></a>        <span class="k">return</span> <span class="n">cdna_match</span>
</span></pre></div>


            <div class="docstring"><p>The function "ordered_cdna_match" sorts a list of cdna_match objects based on their
tx_position.chrom_start attribute and returns the sorted list.</p>

<h6 id="returns">Returns</h6>

<blockquote>
  <p>a sorted list of cdna_match objects.</p>
</blockquote>
</div>


                            </div>
                            <div id="Transcript.get_cds_start_stop" class="classattr">
                                        <input id="Transcript.get_cds_start_stop-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr function">
            
        <span class="def">def</span>
        <span class="name">get_cds_start_stop</span><span class="signature pdoc-code condensed">(<span class="param"><span class="bp">self</span></span><span class="return-annotation">) -> <span class="nb">tuple</span>:</span></span>

                <label class="view-source-button" for="Transcript.get_cds_start_stop-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#Transcript.get_cds_start_stop"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="Transcript.get_cds_start_stop-136"><a href="#Transcript.get_cds_start_stop-136"><span class="linenos">136</span></a>    <span class="k">def</span> <span class="nf">get_cds_start_stop</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">tuple</span><span class="p">:</span>
</span><span id="Transcript.get_cds_start_stop-137"><a href="#Transcript.get_cds_start_stop-137"><span class="linenos">137</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="Transcript.get_cds_start_stop-138"><a href="#Transcript.get_cds_start_stop-138"><span class="linenos">138</span></a><span class="sd">        The function &quot;get_cds_start_stop&quot; returns the start and stop positions of a coding sequence,</span>
</span><span id="Transcript.get_cds_start_stop-139"><a href="#Transcript.get_cds_start_stop-139"><span class="linenos">139</span></a><span class="sd">        taking into account the direction of the strand.</span>
</span><span id="Transcript.get_cds_start_stop-140"><a href="#Transcript.get_cds_start_stop-140"><span class="linenos">140</span></a><span class="sd">        :return: a tuple containing the start and stop positions of the coding sequence (CDS).</span>
</span><span id="Transcript.get_cds_start_stop-141"><a href="#Transcript.get_cds_start_stop-141"><span class="linenos">141</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="Transcript.get_cds_start_stop-142"><a href="#Transcript.get_cds_start_stop-142"><span class="linenos">142</span></a>
</span><span id="Transcript.get_cds_start_stop-143"><a href="#Transcript.get_cds_start_stop-143"><span class="linenos">143</span></a>        <span class="p">(</span><span class="n">start</span><span class="p">,</span> <span class="n">stop</span><span class="p">)</span> <span class="o">=</span> <span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">cds_position</span><span class="o">.</span><span class="n">chrom_start</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">cds_position</span><span class="o">.</span><span class="n">chrom_stop</span><span class="p">)</span>
</span><span id="Transcript.get_cds_start_stop-144"><a href="#Transcript.get_cds_start_stop-144"><span class="linenos">144</span></a>        <span class="k">if</span> <span class="ow">not</span> <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">is_forward_strand</span><span class="p">:</span>
</span><span id="Transcript.get_cds_start_stop-145"><a href="#Transcript.get_cds_start_stop-145"><span class="linenos">145</span></a>            <span class="p">(</span><span class="n">stop</span><span class="p">,</span> <span class="n">start</span><span class="p">)</span> <span class="o">=</span> <span class="p">(</span><span class="n">start</span><span class="p">,</span> <span class="n">stop</span><span class="p">)</span>
</span><span id="Transcript.get_cds_start_stop-146"><a href="#Transcript.get_cds_start_stop-146"><span class="linenos">146</span></a>        <span class="k">return</span> <span class="n">start</span><span class="p">,</span> <span class="n">stop</span>
</span></pre></div>


            <div class="docstring"><p>The function "get_cds_start_stop" returns the start and stop positions of a coding sequence,
taking into account the direction of the strand.</p>

<h6 id="returns">Returns</h6>

<blockquote>
  <p>a tuple containing the start and stop positions of the coding sequence (CDS).</p>
</blockquote>
</div>


                            </div>
                            <div id="Transcript.start_codon" class="classattr">
                                        <input id="Transcript.start_codon-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr function">
                    <div class="decorator">@lazy</div>

        <span class="def">def</span>
        <span class="name">start_codon</span><span class="signature pdoc-code condensed">(<span class="param"><span class="bp">self</span></span><span class="return-annotation">) -> <span class="nb">int</span>:</span></span>

                <label class="view-source-button" for="Transcript.start_codon-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#Transcript.start_codon"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="Transcript.start_codon-148"><a href="#Transcript.start_codon-148"><span class="linenos">148</span></a>    <span class="nd">@lazy</span>
</span><span id="Transcript.start_codon-149"><a href="#Transcript.start_codon-149"><span class="linenos">149</span></a>    <span class="k">def</span> <span class="nf">start_codon</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">int</span><span class="p">:</span>
</span><span id="Transcript.start_codon-150"><a href="#Transcript.start_codon-150"><span class="linenos">150</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="Transcript.start_codon-151"><a href="#Transcript.start_codon-151"><span class="linenos">151</span></a><span class="sd">        The function returns the transcript position of the start codon.</span>
</span><span id="Transcript.start_codon-152"><a href="#Transcript.start_codon-152"><span class="linenos">152</span></a><span class="sd">        :return: the transcript position of the start codon.</span>
</span><span id="Transcript.start_codon-153"><a href="#Transcript.start_codon-153"><span class="linenos">153</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="Transcript.start_codon-154"><a href="#Transcript.start_codon-154"><span class="linenos">154</span></a>
</span><span id="Transcript.start_codon-155"><a href="#Transcript.start_codon-155"><span class="linenos">155</span></a>        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">_start_codon_transcript_pos</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
</span><span id="Transcript.start_codon-156"><a href="#Transcript.start_codon-156"><span class="linenos">156</span></a>            <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">_start_codon_transcript_pos</span>
</span><span id="Transcript.start_codon-157"><a href="#Transcript.start_codon-157"><span class="linenos">157</span></a>
</span><span id="Transcript.start_codon-158"><a href="#Transcript.start_codon-158"><span class="linenos">158</span></a>        <span class="n">start_genomic_coordinate</span><span class="p">,</span> <span class="n">_</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">get_cds_start_stop</span><span class="p">()</span>
</span><span id="Transcript.start_codon-159"><a href="#Transcript.start_codon-159"><span class="linenos">159</span></a>        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">_get_transcript_position</span><span class="p">(</span><span class="n">start_genomic_coordinate</span><span class="p">)</span>
</span></pre></div>


            <div class="docstring"><p>The function returns the transcript position of the start codon.</p>

<h6 id="returns">Returns</h6>

<blockquote>
  <p>the transcript position of the start codon.</p>
</blockquote>
</div>


                            </div>
                            <div id="Transcript.stop_codon" class="classattr">
                                        <input id="Transcript.stop_codon-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr function">
                    <div class="decorator">@lazy</div>

        <span class="def">def</span>
        <span class="name">stop_codon</span><span class="signature pdoc-code condensed">(<span class="param"><span class="bp">self</span></span><span class="return-annotation">) -> <span class="nb">int</span>:</span></span>

                <label class="view-source-button" for="Transcript.stop_codon-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#Transcript.stop_codon"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="Transcript.stop_codon-161"><a href="#Transcript.stop_codon-161"><span class="linenos">161</span></a>    <span class="nd">@lazy</span>
</span><span id="Transcript.stop_codon-162"><a href="#Transcript.stop_codon-162"><span class="linenos">162</span></a>    <span class="k">def</span> <span class="nf">stop_codon</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">int</span><span class="p">:</span>
</span><span id="Transcript.stop_codon-163"><a href="#Transcript.stop_codon-163"><span class="linenos">163</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="Transcript.stop_codon-164"><a href="#Transcript.stop_codon-164"><span class="linenos">164</span></a><span class="sd">        The function returns the transcript position of the stop codon.</span>
</span><span id="Transcript.stop_codon-165"><a href="#Transcript.stop_codon-165"><span class="linenos">165</span></a><span class="sd">        :return: The method `stop_codon` returns an integer, which represents the transcript position of</span>
</span><span id="Transcript.stop_codon-166"><a href="#Transcript.stop_codon-166"><span class="linenos">166</span></a><span class="sd">        the stop codon.</span>
</span><span id="Transcript.stop_codon-167"><a href="#Transcript.stop_codon-167"><span class="linenos">167</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="Transcript.stop_codon-168"><a href="#Transcript.stop_codon-168"><span class="linenos">168</span></a>
</span><span id="Transcript.stop_codon-169"><a href="#Transcript.stop_codon-169"><span class="linenos">169</span></a>        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">_stop_codon_transcript_pos</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
</span><span id="Transcript.stop_codon-170"><a href="#Transcript.stop_codon-170"><span class="linenos">170</span></a>            <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">_stop_codon_transcript_pos</span>
</span><span id="Transcript.stop_codon-171"><a href="#Transcript.stop_codon-171"><span class="linenos">171</span></a>
</span><span id="Transcript.stop_codon-172"><a href="#Transcript.stop_codon-172"><span class="linenos">172</span></a>        <span class="n">_</span><span class="p">,</span> <span class="n">stop_genomic_coordinate</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">get_cds_start_stop</span><span class="p">()</span>
</span><span id="Transcript.stop_codon-173"><a href="#Transcript.stop_codon-173"><span class="linenos">173</span></a>        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">_get_transcript_position</span><span class="p">(</span><span class="n">stop_genomic_coordinate</span><span class="p">)</span>
</span></pre></div>


            <div class="docstring"><p>The function returns the transcript position of the stop codon.</p>

<h6 id="returns">Returns</h6>

<blockquote>
  <p>The method <code><a href="#Transcript.stop_codon">stop_codon</a></code> returns an integer, which represents the transcript position of
  the stop codon.</p>
</blockquote>
</div>


                            </div>
                            <div id="Transcript.cdna_to_genomic_coord" class="classattr">
                                        <input id="Transcript.cdna_to_genomic_coord-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr function">
            
        <span class="def">def</span>
        <span class="name">cdna_to_genomic_coord</span><span class="signature pdoc-code condensed">(<span class="param"><span class="bp">self</span>, </span><span class="param"><span class="n">coord</span><span class="p">:</span> <span class="nb">object</span></span><span class="return-annotation">) -> <span class="nb">int</span>:</span></span>

                <label class="view-source-button" for="Transcript.cdna_to_genomic_coord-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#Transcript.cdna_to_genomic_coord"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="Transcript.cdna_to_genomic_coord-214"><a href="#Transcript.cdna_to_genomic_coord-214"><span class="linenos">214</span></a>    <span class="k">def</span> <span class="nf">cdna_to_genomic_coord</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">coord</span><span class="p">:</span> <span class="nb">object</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">int</span><span class="p">:</span>
</span><span id="Transcript.cdna_to_genomic_coord-215"><a href="#Transcript.cdna_to_genomic_coord-215"><span class="linenos">215</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="Transcript.cdna_to_genomic_coord-216"><a href="#Transcript.cdna_to_genomic_coord-216"><span class="linenos">216</span></a><span class="sd">        The function `cdna_to_genomic_coord` converts a HGVS cDNA coordinate to a genomic coordinate.</span>
</span><span id="Transcript.cdna_to_genomic_coord-217"><a href="#Transcript.cdna_to_genomic_coord-217"><span class="linenos">217</span></a>
</span><span id="Transcript.cdna_to_genomic_coord-218"><a href="#Transcript.cdna_to_genomic_coord-218"><span class="linenos">218</span></a><span class="sd">        :param coord: The parameter `coord` is an object that represents a cDNA coordinate. It is used</span>
</span><span id="Transcript.cdna_to_genomic_coord-219"><a href="#Transcript.cdna_to_genomic_coord-219"><span class="linenos">219</span></a><span class="sd">        to specify a position along a cDNA sequence</span>
</span><span id="Transcript.cdna_to_genomic_coord-220"><a href="#Transcript.cdna_to_genomic_coord-220"><span class="linenos">220</span></a><span class="sd">        :type coord: object</span>
</span><span id="Transcript.cdna_to_genomic_coord-221"><a href="#Transcript.cdna_to_genomic_coord-221"><span class="linenos">221</span></a><span class="sd">        :return: an integer value, which represents the genomic coordinate corresponding to the given</span>
</span><span id="Transcript.cdna_to_genomic_coord-222"><a href="#Transcript.cdna_to_genomic_coord-222"><span class="linenos">222</span></a><span class="sd">        cDNA coordinate.</span>
</span><span id="Transcript.cdna_to_genomic_coord-223"><a href="#Transcript.cdna_to_genomic_coord-223"><span class="linenos">223</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="Transcript.cdna_to_genomic_coord-224"><a href="#Transcript.cdna_to_genomic_coord-224"><span class="linenos">224</span></a>
</span><span id="Transcript.cdna_to_genomic_coord-225"><a href="#Transcript.cdna_to_genomic_coord-225"><span class="linenos">225</span></a>        <span class="n">transcript_strand</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">is_forward_strand</span>
</span><span id="Transcript.cdna_to_genomic_coord-226"><a href="#Transcript.cdna_to_genomic_coord-226"><span class="linenos">226</span></a>
</span><span id="Transcript.cdna_to_genomic_coord-227"><a href="#Transcript.cdna_to_genomic_coord-227"><span class="linenos">227</span></a>        <span class="c1"># compute starting position along spliced transcript.</span>
</span><span id="Transcript.cdna_to_genomic_coord-228"><a href="#Transcript.cdna_to_genomic_coord-228"><span class="linenos">228</span></a>        <span class="k">if</span> <span class="n">coord</span><span class="o">.</span><span class="n">landmark</span> <span class="o">==</span> <span class="n">CDNA_START_CODON</span><span class="p">:</span>
</span><span id="Transcript.cdna_to_genomic_coord-229"><a href="#Transcript.cdna_to_genomic_coord-229"><span class="linenos">229</span></a>            <span class="n">utr5p</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">start_codon</span> <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">is_coding</span> <span class="k">else</span> <span class="mi">0</span>
</span><span id="Transcript.cdna_to_genomic_coord-230"><a href="#Transcript.cdna_to_genomic_coord-230"><span class="linenos">230</span></a>
</span><span id="Transcript.cdna_to_genomic_coord-231"><a href="#Transcript.cdna_to_genomic_coord-231"><span class="linenos">231</span></a>            <span class="k">if</span> <span class="n">coord</span><span class="o">.</span><span class="n">coord</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">:</span>
</span><span id="Transcript.cdna_to_genomic_coord-232"><a href="#Transcript.cdna_to_genomic_coord-232"><span class="linenos">232</span></a>                <span class="n">cdna_pos</span> <span class="o">=</span> <span class="n">utr5p</span> <span class="o">+</span> <span class="n">coord</span><span class="o">.</span><span class="n">coord</span>
</span><span id="Transcript.cdna_to_genomic_coord-233"><a href="#Transcript.cdna_to_genomic_coord-233"><span class="linenos">233</span></a>            <span class="k">else</span><span class="p">:</span>
</span><span id="Transcript.cdna_to_genomic_coord-234"><a href="#Transcript.cdna_to_genomic_coord-234"><span class="linenos">234</span></a>                <span class="n">cdna_pos</span> <span class="o">=</span> <span class="n">utr5p</span> <span class="o">+</span> <span class="n">coord</span><span class="o">.</span><span class="n">coord</span> <span class="o">+</span> <span class="mi">1</span>
</span><span id="Transcript.cdna_to_genomic_coord-235"><a href="#Transcript.cdna_to_genomic_coord-235"><span class="linenos">235</span></a>        <span class="k">elif</span> <span class="n">coord</span><span class="o">.</span><span class="n">landmark</span> <span class="o">==</span> <span class="n">CDNA_STOP_CODON</span><span class="p">:</span>
</span><span id="Transcript.cdna_to_genomic_coord-236"><a href="#Transcript.cdna_to_genomic_coord-236"><span class="linenos">236</span></a>            <span class="k">if</span> <span class="n">coord</span><span class="o">.</span><span class="n">coord</span> <span class="o">&lt;</span> <span class="mi">0</span><span class="p">:</span>
</span><span id="Transcript.cdna_to_genomic_coord-237"><a href="#Transcript.cdna_to_genomic_coord-237"><span class="linenos">237</span></a>                <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span>
</span><span id="Transcript.cdna_to_genomic_coord-238"><a href="#Transcript.cdna_to_genomic_coord-238"><span class="linenos">238</span></a>                    <span class="s2">&quot;CDNACoord cannot have a negative coord and &quot;</span>
</span><span id="Transcript.cdna_to_genomic_coord-239"><a href="#Transcript.cdna_to_genomic_coord-239"><span class="linenos">239</span></a>                    <span class="s2">&quot;landmark CDNA_STOP_CODON&quot;</span>
</span><span id="Transcript.cdna_to_genomic_coord-240"><a href="#Transcript.cdna_to_genomic_coord-240"><span class="linenos">240</span></a>                <span class="p">)</span>
</span><span id="Transcript.cdna_to_genomic_coord-241"><a href="#Transcript.cdna_to_genomic_coord-241"><span class="linenos">241</span></a>            <span class="n">cdna_pos</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">stop_codon</span> <span class="o">+</span> <span class="n">coord</span><span class="o">.</span><span class="n">coord</span>
</span><span id="Transcript.cdna_to_genomic_coord-242"><a href="#Transcript.cdna_to_genomic_coord-242"><span class="linenos">242</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="Transcript.cdna_to_genomic_coord-243"><a href="#Transcript.cdna_to_genomic_coord-243"><span class="linenos">243</span></a>            <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="s1">&#39;unknown CDNACoord landmark &quot;</span><span class="si">%s</span><span class="s1">&quot;&#39;</span> <span class="o">%</span> <span class="n">coord</span><span class="o">.</span><span class="n">landmark</span><span class="p">)</span>
</span><span id="Transcript.cdna_to_genomic_coord-244"><a href="#Transcript.cdna_to_genomic_coord-244"><span class="linenos">244</span></a>
</span><span id="Transcript.cdna_to_genomic_coord-245"><a href="#Transcript.cdna_to_genomic_coord-245"><span class="linenos">245</span></a>        <span class="c1"># 5&#39; flanking sequence (no need to account for gaps)</span>
</span><span id="Transcript.cdna_to_genomic_coord-246"><a href="#Transcript.cdna_to_genomic_coord-246"><span class="linenos">246</span></a>        <span class="k">if</span> <span class="n">cdna_pos</span> <span class="o">&lt;</span> <span class="mi">1</span><span class="p">:</span>
</span><span id="Transcript.cdna_to_genomic_coord-247"><a href="#Transcript.cdna_to_genomic_coord-247"><span class="linenos">247</span></a>            <span class="k">if</span> <span class="n">transcript_strand</span><span class="p">:</span>
</span><span id="Transcript.cdna_to_genomic_coord-248"><a href="#Transcript.cdna_to_genomic_coord-248"><span class="linenos">248</span></a>                <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_start</span> <span class="o">+</span> <span class="n">cdna_pos</span>
</span><span id="Transcript.cdna_to_genomic_coord-249"><a href="#Transcript.cdna_to_genomic_coord-249"><span class="linenos">249</span></a>            <span class="k">else</span><span class="p">:</span>
</span><span id="Transcript.cdna_to_genomic_coord-250"><a href="#Transcript.cdna_to_genomic_coord-250"><span class="linenos">250</span></a>                <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_stop</span> <span class="o">-</span> <span class="n">cdna_pos</span> <span class="o">+</span> <span class="mi">1</span>
</span><span id="Transcript.cdna_to_genomic_coord-251"><a href="#Transcript.cdna_to_genomic_coord-251"><span class="linenos">251</span></a>
</span><span id="Transcript.cdna_to_genomic_coord-252"><a href="#Transcript.cdna_to_genomic_coord-252"><span class="linenos">252</span></a>        <span class="c1"># Walk along transcript until we find an exon that contains cdna_pos.</span>
</span><span id="Transcript.cdna_to_genomic_coord-253"><a href="#Transcript.cdna_to_genomic_coord-253"><span class="linenos">253</span></a>        <span class="k">for</span> <span class="n">cdna_match</span> <span class="ow">in</span> <span class="bp">self</span><span class="o">.</span><span class="n">ordered_cdna_match</span><span class="p">:</span>
</span><span id="Transcript.cdna_to_genomic_coord-254"><a href="#Transcript.cdna_to_genomic_coord-254"><span class="linenos">254</span></a>            <span class="k">if</span> <span class="n">cdna_match</span><span class="o">.</span><span class="n">cdna_start</span> <span class="o">&lt;=</span> <span class="n">cdna_pos</span> <span class="o">&lt;=</span> <span class="n">cdna_match</span><span class="o">.</span><span class="n">cdna_end</span><span class="p">:</span>
</span><span id="Transcript.cdna_to_genomic_coord-255"><a href="#Transcript.cdna_to_genomic_coord-255"><span class="linenos">255</span></a>                <span class="n">match_pos</span> <span class="o">=</span> <span class="n">cdna_pos</span> <span class="o">-</span> <span class="n">cdna_match</span><span class="o">.</span><span class="n">cdna_start</span>
</span><span id="Transcript.cdna_to_genomic_coord-256"><a href="#Transcript.cdna_to_genomic_coord-256"><span class="linenos">256</span></a>                <span class="n">match_pos</span> <span class="o">-=</span> <span class="n">cdna_match</span><span class="o">.</span><span class="n">get_offset</span><span class="p">(</span><span class="n">match_pos</span><span class="p">)</span>
</span><span id="Transcript.cdna_to_genomic_coord-257"><a href="#Transcript.cdna_to_genomic_coord-257"><span class="linenos">257</span></a>                <span class="c1"># Compute genomic coordinate using offset.</span>
</span><span id="Transcript.cdna_to_genomic_coord-258"><a href="#Transcript.cdna_to_genomic_coord-258"><span class="linenos">258</span></a>                <span class="k">if</span> <span class="n">transcript_strand</span><span class="p">:</span>
</span><span id="Transcript.cdna_to_genomic_coord-259"><a href="#Transcript.cdna_to_genomic_coord-259"><span class="linenos">259</span></a>                    <span class="c1"># Plus strand.</span>
</span><span id="Transcript.cdna_to_genomic_coord-260"><a href="#Transcript.cdna_to_genomic_coord-260"><span class="linenos">260</span></a>                    <span class="n">start</span> <span class="o">=</span> <span class="n">cdna_match</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_start</span> <span class="o">+</span> <span class="mi">1</span>
</span><span id="Transcript.cdna_to_genomic_coord-261"><a href="#Transcript.cdna_to_genomic_coord-261"><span class="linenos">261</span></a>                    <span class="k">return</span> <span class="n">start</span> <span class="o">+</span> <span class="n">match_pos</span> <span class="o">+</span> <span class="n">coord</span><span class="o">.</span><span class="n">offset</span>
</span><span id="Transcript.cdna_to_genomic_coord-262"><a href="#Transcript.cdna_to_genomic_coord-262"><span class="linenos">262</span></a>                <span class="k">else</span><span class="p">:</span>
</span><span id="Transcript.cdna_to_genomic_coord-263"><a href="#Transcript.cdna_to_genomic_coord-263"><span class="linenos">263</span></a>                    <span class="c1"># Minus strand.</span>
</span><span id="Transcript.cdna_to_genomic_coord-264"><a href="#Transcript.cdna_to_genomic_coord-264"><span class="linenos">264</span></a>                    <span class="n">end</span> <span class="o">=</span> <span class="n">cdna_match</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_stop</span>
</span><span id="Transcript.cdna_to_genomic_coord-265"><a href="#Transcript.cdna_to_genomic_coord-265"><span class="linenos">265</span></a>                    <span class="k">return</span> <span class="n">end</span> <span class="o">-</span> <span class="n">match_pos</span> <span class="o">-</span> <span class="n">coord</span><span class="o">.</span><span class="n">offset</span>
</span><span id="Transcript.cdna_to_genomic_coord-266"><a href="#Transcript.cdna_to_genomic_coord-266"><span class="linenos">266</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="Transcript.cdna_to_genomic_coord-267"><a href="#Transcript.cdna_to_genomic_coord-267"><span class="linenos">267</span></a>            <span class="c1"># 3&#39; flanking sequence (no need to account for gaps)</span>
</span><span id="Transcript.cdna_to_genomic_coord-268"><a href="#Transcript.cdna_to_genomic_coord-268"><span class="linenos">268</span></a>            <span class="k">if</span> <span class="n">transcript_strand</span><span class="p">:</span>
</span><span id="Transcript.cdna_to_genomic_coord-269"><a href="#Transcript.cdna_to_genomic_coord-269"><span class="linenos">269</span></a>                <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">cds_position</span><span class="o">.</span><span class="n">chrom_stop</span> <span class="o">+</span> <span class="n">coord</span><span class="o">.</span><span class="n">coord</span>
</span><span id="Transcript.cdna_to_genomic_coord-270"><a href="#Transcript.cdna_to_genomic_coord-270"><span class="linenos">270</span></a>            <span class="k">else</span><span class="p">:</span>
</span><span id="Transcript.cdna_to_genomic_coord-271"><a href="#Transcript.cdna_to_genomic_coord-271"><span class="linenos">271</span></a>                <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">cds_position</span><span class="o">.</span><span class="n">chrom_start</span> <span class="o">+</span> <span class="mi">1</span> <span class="o">-</span> <span class="n">coord</span><span class="o">.</span><span class="n">coord</span>
</span></pre></div>


            <div class="docstring"><p>The function <code><a href="#Transcript.cdna_to_genomic_coord">cdna_to_genomic_coord</a></code> converts a HGVS cDNA coordinate to a genomic coordinate.</p>

<h6 id="parameters">Parameters</h6>

<ul>
<li><strong>coord</strong>:  The parameter <code>coord</code> is an object that represents a cDNA coordinate. It is used
to specify a position along a cDNA sequence</li>
</ul>

<h6 id="returns">Returns</h6>

<blockquote>
  <p>an integer value, which represents the genomic coordinate corresponding to the given
  cDNA coordinate.</p>
</blockquote>
</div>


                            </div>
                            <div id="Transcript.genomic_to_cdna_coord" class="classattr">
                                        <input id="Transcript.genomic_to_cdna_coord-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr function">
            
        <span class="def">def</span>
        <span class="name">genomic_to_cdna_coord</span><span class="signature pdoc-code condensed">(<span class="param"><span class="bp">self</span>, </span><span class="param"><span class="n">genomic_coord</span><span class="p">:</span> <span class="nb">int</span></span><span class="return-annotation">) -> <span class="nb">object</span>:</span></span>

                <label class="view-source-button" for="Transcript.genomic_to_cdna_coord-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#Transcript.genomic_to_cdna_coord"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="Transcript.genomic_to_cdna_coord-273"><a href="#Transcript.genomic_to_cdna_coord-273"><span class="linenos">273</span></a>    <span class="k">def</span> <span class="nf">genomic_to_cdna_coord</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">genomic_coord</span><span class="p">:</span> <span class="nb">int</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">object</span><span class="p">:</span>
</span><span id="Transcript.genomic_to_cdna_coord-274"><a href="#Transcript.genomic_to_cdna_coord-274"><span class="linenos">274</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="Transcript.genomic_to_cdna_coord-275"><a href="#Transcript.genomic_to_cdna_coord-275"><span class="linenos">275</span></a><span class="sd">        The function `genomic_to_cdna_coord` converts a genomic coordinate to a cDNA coordinate and</span>
</span><span id="Transcript.genomic_to_cdna_coord-276"><a href="#Transcript.genomic_to_cdna_coord-276"><span class="linenos">276</span></a><span class="sd">        offset, taking into account exons, strand, and coding transcript information.</span>
</span><span id="Transcript.genomic_to_cdna_coord-277"><a href="#Transcript.genomic_to_cdna_coord-277"><span class="linenos">277</span></a>
</span><span id="Transcript.genomic_to_cdna_coord-278"><a href="#Transcript.genomic_to_cdna_coord-278"><span class="linenos">278</span></a><span class="sd">        :param genomic_coord: The `genomic_coord` parameter is an integer representing a genomic</span>
</span><span id="Transcript.genomic_to_cdna_coord-279"><a href="#Transcript.genomic_to_cdna_coord-279"><span class="linenos">279</span></a><span class="sd">        coordinate</span>
</span><span id="Transcript.genomic_to_cdna_coord-280"><a href="#Transcript.genomic_to_cdna_coord-280"><span class="linenos">280</span></a><span class="sd">        :type genomic_coord: int</span>
</span><span id="Transcript.genomic_to_cdna_coord-281"><a href="#Transcript.genomic_to_cdna_coord-281"><span class="linenos">281</span></a><span class="sd">        :return: an object of type `CDNACoord`.</span>
</span><span id="Transcript.genomic_to_cdna_coord-282"><a href="#Transcript.genomic_to_cdna_coord-282"><span class="linenos">282</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="Transcript.genomic_to_cdna_coord-283"><a href="#Transcript.genomic_to_cdna_coord-283"><span class="linenos">283</span></a>
</span><span id="Transcript.genomic_to_cdna_coord-284"><a href="#Transcript.genomic_to_cdna_coord-284"><span class="linenos">284</span></a>        <span class="n">exons</span> <span class="o">=</span> <span class="p">[</span><span class="n">exon</span><span class="o">.</span><span class="n">get_as_interval</span><span class="p">()</span> <span class="k">for</span> <span class="n">exon</span> <span class="ow">in</span> <span class="bp">self</span><span class="o">.</span><span class="n">ordered_cdna_match</span><span class="p">]</span>
</span><span id="Transcript.genomic_to_cdna_coord-285"><a href="#Transcript.genomic_to_cdna_coord-285"><span class="linenos">285</span></a>
</span><span id="Transcript.genomic_to_cdna_coord-286"><a href="#Transcript.genomic_to_cdna_coord-286"><span class="linenos">286</span></a>        <span class="k">if</span> <span class="nb">len</span><span class="p">(</span><span class="n">exons</span><span class="p">)</span> <span class="o">==</span> <span class="mi">0</span><span class="p">:</span>
</span><span id="Transcript.genomic_to_cdna_coord-287"><a href="#Transcript.genomic_to_cdna_coord-287"><span class="linenos">287</span></a>            <span class="k">return</span> <span class="kc">None</span>
</span><span id="Transcript.genomic_to_cdna_coord-288"><a href="#Transcript.genomic_to_cdna_coord-288"><span class="linenos">288</span></a>
</span><span id="Transcript.genomic_to_cdna_coord-289"><a href="#Transcript.genomic_to_cdna_coord-289"><span class="linenos">289</span></a>        <span class="n">strand</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">strand</span>
</span><span id="Transcript.genomic_to_cdna_coord-290"><a href="#Transcript.genomic_to_cdna_coord-290"><span class="linenos">290</span></a>        <span class="n">distances</span> <span class="o">=</span> <span class="p">[</span><span class="n">exon</span><span class="o">.</span><span class="n">distance</span><span class="p">(</span><span class="n">genomic_coord</span><span class="p">)</span> <span class="k">for</span> <span class="n">exon</span> <span class="ow">in</span> <span class="n">exons</span><span class="p">]</span>
</span><span id="Transcript.genomic_to_cdna_coord-291"><a href="#Transcript.genomic_to_cdna_coord-291"><span class="linenos">291</span></a>        <span class="n">min_distance_to_exon</span> <span class="o">=</span> <span class="nb">min</span><span class="p">(</span><span class="nb">map</span><span class="p">(</span><span class="nb">abs</span><span class="p">,</span> <span class="n">distances</span><span class="p">))</span>
</span><span id="Transcript.genomic_to_cdna_coord-292"><a href="#Transcript.genomic_to_cdna_coord-292"><span class="linenos">292</span></a>        <span class="k">if</span> <span class="n">min_distance_to_exon</span><span class="p">:</span>
</span><span id="Transcript.genomic_to_cdna_coord-293"><a href="#Transcript.genomic_to_cdna_coord-293"><span class="linenos">293</span></a>            <span class="c1"># We&#39;re outside of exon - so need to find closest point</span>
</span><span id="Transcript.genomic_to_cdna_coord-294"><a href="#Transcript.genomic_to_cdna_coord-294"><span class="linenos">294</span></a>            <span class="k">for</span> <span class="n">exon</span> <span class="ow">in</span> <span class="n">exons</span><span class="p">:</span>
</span><span id="Transcript.genomic_to_cdna_coord-295"><a href="#Transcript.genomic_to_cdna_coord-295"><span class="linenos">295</span></a>                <span class="n">distance</span> <span class="o">=</span> <span class="n">exon</span><span class="o">.</span><span class="n">distance</span><span class="p">(</span><span class="n">genomic_coord</span><span class="p">)</span>
</span><span id="Transcript.genomic_to_cdna_coord-296"><a href="#Transcript.genomic_to_cdna_coord-296"><span class="linenos">296</span></a>                <span class="k">if</span> <span class="nb">abs</span><span class="p">(</span><span class="n">distance</span><span class="p">)</span> <span class="o">==</span> <span class="n">min_distance_to_exon</span><span class="p">:</span>
</span><span id="Transcript.genomic_to_cdna_coord-297"><a href="#Transcript.genomic_to_cdna_coord-297"><span class="linenos">297</span></a>
</span><span id="Transcript.genomic_to_cdna_coord-298"><a href="#Transcript.genomic_to_cdna_coord-298"><span class="linenos">298</span></a>                    <span class="c1"># Outside the exon.</span>
</span><span id="Transcript.genomic_to_cdna_coord-299"><a href="#Transcript.genomic_to_cdna_coord-299"><span class="linenos">299</span></a>                    <span class="k">if</span> <span class="n">distance</span> <span class="o">&gt;</span> <span class="mi">0</span><span class="p">:</span>
</span><span id="Transcript.genomic_to_cdna_coord-300"><a href="#Transcript.genomic_to_cdna_coord-300"><span class="linenos">300</span></a>                        <span class="n">genomic_nearest_exon</span> <span class="o">=</span> <span class="n">exon</span><span class="o">.</span><span class="n">chrom_start</span> <span class="o">+</span> <span class="mi">1</span>
</span><span id="Transcript.genomic_to_cdna_coord-301"><a href="#Transcript.genomic_to_cdna_coord-301"><span class="linenos">301</span></a>                    <span class="k">else</span><span class="p">:</span>
</span><span id="Transcript.genomic_to_cdna_coord-302"><a href="#Transcript.genomic_to_cdna_coord-302"><span class="linenos">302</span></a>                        <span class="n">genomic_nearest_exon</span> <span class="o">=</span> <span class="n">exon</span><span class="o">.</span><span class="n">chrom_end</span>
</span><span id="Transcript.genomic_to_cdna_coord-303"><a href="#Transcript.genomic_to_cdna_coord-303"><span class="linenos">303</span></a>
</span><span id="Transcript.genomic_to_cdna_coord-304"><a href="#Transcript.genomic_to_cdna_coord-304"><span class="linenos">304</span></a>                    <span class="k">if</span> <span class="n">strand</span> <span class="o">==</span> <span class="s2">&quot;+&quot;</span><span class="p">:</span>
</span><span id="Transcript.genomic_to_cdna_coord-305"><a href="#Transcript.genomic_to_cdna_coord-305"><span class="linenos">305</span></a>                        <span class="n">distance</span> <span class="o">*=</span> <span class="o">-</span><span class="mi">1</span>
</span><span id="Transcript.genomic_to_cdna_coord-306"><a href="#Transcript.genomic_to_cdna_coord-306"><span class="linenos">306</span></a>
</span><span id="Transcript.genomic_to_cdna_coord-307"><a href="#Transcript.genomic_to_cdna_coord-307"><span class="linenos">307</span></a>                    <span class="n">nearest_exon_coord</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_exon_genomic_to_cdna_coord</span><span class="p">(</span>
</span><span id="Transcript.genomic_to_cdna_coord-308"><a href="#Transcript.genomic_to_cdna_coord-308"><span class="linenos">308</span></a>                        <span class="n">genomic_nearest_exon</span>
</span><span id="Transcript.genomic_to_cdna_coord-309"><a href="#Transcript.genomic_to_cdna_coord-309"><span class="linenos">309</span></a>                    <span class="p">)</span>
</span><span id="Transcript.genomic_to_cdna_coord-310"><a href="#Transcript.genomic_to_cdna_coord-310"><span class="linenos">310</span></a>
</span><span id="Transcript.genomic_to_cdna_coord-311"><a href="#Transcript.genomic_to_cdna_coord-311"><span class="linenos">311</span></a>                    <span class="c1"># If outside transcript, don&#39;t use offset.</span>
</span><span id="Transcript.genomic_to_cdna_coord-312"><a href="#Transcript.genomic_to_cdna_coord-312"><span class="linenos">312</span></a>                    <span class="k">if</span> <span class="p">(</span>
</span><span id="Transcript.genomic_to_cdna_coord-313"><a href="#Transcript.genomic_to_cdna_coord-313"><span class="linenos">313</span></a>                        <span class="n">genomic_coord</span> <span class="o">&lt;</span> <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_start</span> <span class="o">+</span> <span class="mi">1</span>
</span><span id="Transcript.genomic_to_cdna_coord-314"><a href="#Transcript.genomic_to_cdna_coord-314"><span class="linenos">314</span></a>                        <span class="ow">or</span> <span class="n">genomic_coord</span> <span class="o">&gt;</span> <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_stop</span>
</span><span id="Transcript.genomic_to_cdna_coord-315"><a href="#Transcript.genomic_to_cdna_coord-315"><span class="linenos">315</span></a>                    <span class="p">):</span>
</span><span id="Transcript.genomic_to_cdna_coord-316"><a href="#Transcript.genomic_to_cdna_coord-316"><span class="linenos">316</span></a>                        <span class="n">nearest_exon_coord</span> <span class="o">+=</span> <span class="n">distance</span>
</span><span id="Transcript.genomic_to_cdna_coord-317"><a href="#Transcript.genomic_to_cdna_coord-317"><span class="linenos">317</span></a>                        <span class="n">distance</span> <span class="o">=</span> <span class="mi">0</span>
</span><span id="Transcript.genomic_to_cdna_coord-318"><a href="#Transcript.genomic_to_cdna_coord-318"><span class="linenos">318</span></a>                    <span class="n">cdna_coord</span> <span class="o">=</span> <span class="n">CDNACoord</span><span class="p">(</span><span class="n">nearest_exon_coord</span><span class="p">,</span> <span class="n">distance</span><span class="p">)</span>
</span><span id="Transcript.genomic_to_cdna_coord-319"><a href="#Transcript.genomic_to_cdna_coord-319"><span class="linenos">319</span></a>                    <span class="k">break</span>
</span><span id="Transcript.genomic_to_cdna_coord-320"><a href="#Transcript.genomic_to_cdna_coord-320"><span class="linenos">320</span></a>            <span class="k">else</span><span class="p">:</span>
</span><span id="Transcript.genomic_to_cdna_coord-321"><a href="#Transcript.genomic_to_cdna_coord-321"><span class="linenos">321</span></a>                <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="s2">&quot;Could not find closest exon!&quot;</span><span class="p">)</span>  <span class="c1"># Should never happen</span>
</span><span id="Transcript.genomic_to_cdna_coord-322"><a href="#Transcript.genomic_to_cdna_coord-322"><span class="linenos">322</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="Transcript.genomic_to_cdna_coord-323"><a href="#Transcript.genomic_to_cdna_coord-323"><span class="linenos">323</span></a>            <span class="n">coord</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">_exon_genomic_to_cdna_coord</span><span class="p">(</span><span class="n">genomic_coord</span><span class="p">)</span>
</span><span id="Transcript.genomic_to_cdna_coord-324"><a href="#Transcript.genomic_to_cdna_coord-324"><span class="linenos">324</span></a>            <span class="n">cdna_coord</span> <span class="o">=</span> <span class="n">CDNACoord</span><span class="p">(</span><span class="n">coord</span><span class="p">,</span> <span class="mi">0</span><span class="p">)</span>
</span><span id="Transcript.genomic_to_cdna_coord-325"><a href="#Transcript.genomic_to_cdna_coord-325"><span class="linenos">325</span></a>
</span><span id="Transcript.genomic_to_cdna_coord-326"><a href="#Transcript.genomic_to_cdna_coord-326"><span class="linenos">326</span></a>        <span class="c1"># Adjust coordinates for coding transcript.</span>
</span><span id="Transcript.genomic_to_cdna_coord-327"><a href="#Transcript.genomic_to_cdna_coord-327"><span class="linenos">327</span></a>        <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">is_coding</span><span class="p">:</span>
</span><span id="Transcript.genomic_to_cdna_coord-328"><a href="#Transcript.genomic_to_cdna_coord-328"><span class="linenos">328</span></a>            <span class="c1"># Detect if position before start codon.</span>
</span><span id="Transcript.genomic_to_cdna_coord-329"><a href="#Transcript.genomic_to_cdna_coord-329"><span class="linenos">329</span></a>            <span class="n">utr5p</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">start_codon</span>
</span><span id="Transcript.genomic_to_cdna_coord-330"><a href="#Transcript.genomic_to_cdna_coord-330"><span class="linenos">330</span></a>            <span class="n">cdna_coord</span><span class="o">.</span><span class="n">coord</span> <span class="o">-=</span> <span class="n">utr5p</span>
</span><span id="Transcript.genomic_to_cdna_coord-331"><a href="#Transcript.genomic_to_cdna_coord-331"><span class="linenos">331</span></a>            <span class="k">if</span> <span class="n">cdna_coord</span><span class="o">.</span><span class="n">coord</span> <span class="o">&lt;=</span> <span class="mi">0</span><span class="p">:</span>
</span><span id="Transcript.genomic_to_cdna_coord-332"><a href="#Transcript.genomic_to_cdna_coord-332"><span class="linenos">332</span></a>                <span class="n">cdna_coord</span><span class="o">.</span><span class="n">coord</span> <span class="o">-=</span> <span class="mi">1</span>
</span><span id="Transcript.genomic_to_cdna_coord-333"><a href="#Transcript.genomic_to_cdna_coord-333"><span class="linenos">333</span></a>            <span class="k">else</span><span class="p">:</span>
</span><span id="Transcript.genomic_to_cdna_coord-334"><a href="#Transcript.genomic_to_cdna_coord-334"><span class="linenos">334</span></a>                <span class="c1"># Detect if position is after stop_codon.</span>
</span><span id="Transcript.genomic_to_cdna_coord-335"><a href="#Transcript.genomic_to_cdna_coord-335"><span class="linenos">335</span></a>                <span class="n">stop_codon</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">stop_codon</span>
</span><span id="Transcript.genomic_to_cdna_coord-336"><a href="#Transcript.genomic_to_cdna_coord-336"><span class="linenos">336</span></a>                <span class="n">stop_codon</span> <span class="o">-=</span> <span class="n">utr5p</span>
</span><span id="Transcript.genomic_to_cdna_coord-337"><a href="#Transcript.genomic_to_cdna_coord-337"><span class="linenos">337</span></a>                <span class="k">if</span> <span class="p">(</span>
</span><span id="Transcript.genomic_to_cdna_coord-338"><a href="#Transcript.genomic_to_cdna_coord-338"><span class="linenos">338</span></a>                    <span class="n">cdna_coord</span><span class="o">.</span><span class="n">coord</span> <span class="o">&gt;</span> <span class="n">stop_codon</span>
</span><span id="Transcript.genomic_to_cdna_coord-339"><a href="#Transcript.genomic_to_cdna_coord-339"><span class="linenos">339</span></a>                    <span class="ow">or</span> <span class="n">cdna_coord</span><span class="o">.</span><span class="n">coord</span> <span class="o">==</span> <span class="n">stop_codon</span>
</span><span id="Transcript.genomic_to_cdna_coord-340"><a href="#Transcript.genomic_to_cdna_coord-340"><span class="linenos">340</span></a>                    <span class="ow">and</span> <span class="n">cdna_coord</span><span class="o">.</span><span class="n">offset</span> <span class="o">&gt;</span> <span class="mi">0</span>
</span><span id="Transcript.genomic_to_cdna_coord-341"><a href="#Transcript.genomic_to_cdna_coord-341"><span class="linenos">341</span></a>                <span class="p">):</span>
</span><span id="Transcript.genomic_to_cdna_coord-342"><a href="#Transcript.genomic_to_cdna_coord-342"><span class="linenos">342</span></a>                    <span class="n">cdna_coord</span><span class="o">.</span><span class="n">coord</span> <span class="o">-=</span> <span class="n">stop_codon</span>
</span><span id="Transcript.genomic_to_cdna_coord-343"><a href="#Transcript.genomic_to_cdna_coord-343"><span class="linenos">343</span></a>                    <span class="n">cdna_coord</span><span class="o">.</span><span class="n">landmark</span> <span class="o">=</span> <span class="n">CDNA_STOP_CODON</span>
</span><span id="Transcript.genomic_to_cdna_coord-344"><a href="#Transcript.genomic_to_cdna_coord-344"><span class="linenos">344</span></a>
</span><span id="Transcript.genomic_to_cdna_coord-345"><a href="#Transcript.genomic_to_cdna_coord-345"><span class="linenos">345</span></a>        <span class="k">return</span> <span class="n">cdna_coord</span>
</span></pre></div>


            <div class="docstring"><p>The function <code><a href="#Transcript.genomic_to_cdna_coord">genomic_to_cdna_coord</a></code> converts a genomic coordinate to a cDNA coordinate and
offset, taking into account exons, strand, and coding transcript information.</p>

<h6 id="parameters">Parameters</h6>

<ul>
<li><strong>genomic_coord</strong>:  The <code>genomic_coord</code> parameter is an integer representing a genomic
coordinate</li>
</ul>

<h6 id="returns">Returns</h6>

<blockquote>
  <p>an object of type <code>CDNACoord</code>.</p>
</blockquote>
</div>


                            </div>
                            <div id="Transcript.find_exon_number" class="classattr">
                                        <input id="Transcript.find_exon_number-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr function">
            
        <span class="def">def</span>
        <span class="name">find_exon_number</span><span class="signature pdoc-code condensed">(<span class="param"><span class="bp">self</span>, </span><span class="param"><span class="n">offset</span><span class="p">:</span> <span class="nb">int</span></span><span class="return-annotation">) -> <span class="nb">int</span>:</span></span>

                <label class="view-source-button" for="Transcript.find_exon_number-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#Transcript.find_exon_number"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="Transcript.find_exon_number-377"><a href="#Transcript.find_exon_number-377"><span class="linenos">377</span></a>    <span class="k">def</span> <span class="nf">find_exon_number</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">offset</span><span class="p">:</span> <span class="nb">int</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">int</span><span class="p">:</span>
</span><span id="Transcript.find_exon_number-378"><a href="#Transcript.find_exon_number-378"><span class="linenos">378</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="Transcript.find_exon_number-379"><a href="#Transcript.find_exon_number-379"><span class="linenos">379</span></a><span class="sd">        The function `find_exon_number` returns the exon number for a given position.</span>
</span><span id="Transcript.find_exon_number-380"><a href="#Transcript.find_exon_number-380"><span class="linenos">380</span></a>
</span><span id="Transcript.find_exon_number-381"><a href="#Transcript.find_exon_number-381"><span class="linenos">381</span></a><span class="sd">        :param offset: The offset parameter represents a position in the genome. It is an integer value</span>
</span><span id="Transcript.find_exon_number-382"><a href="#Transcript.find_exon_number-382"><span class="linenos">382</span></a><span class="sd">        that indicates the position of interest within the genome</span>
</span><span id="Transcript.find_exon_number-383"><a href="#Transcript.find_exon_number-383"><span class="linenos">383</span></a><span class="sd">        :type offset: int</span>
</span><span id="Transcript.find_exon_number-384"><a href="#Transcript.find_exon_number-384"><span class="linenos">384</span></a><span class="sd">        :return: an integer value, which represents the exon number for a given position.</span>
</span><span id="Transcript.find_exon_number-385"><a href="#Transcript.find_exon_number-385"><span class="linenos">385</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="Transcript.find_exon_number-386"><a href="#Transcript.find_exon_number-386"><span class="linenos">386</span></a>
</span><span id="Transcript.find_exon_number-387"><a href="#Transcript.find_exon_number-387"><span class="linenos">387</span></a>        <span class="n">exon_number</span> <span class="o">=</span> <span class="mi">1</span>
</span><span id="Transcript.find_exon_number-388"><a href="#Transcript.find_exon_number-388"><span class="linenos">388</span></a>        <span class="k">for</span> <span class="n">cdna_match</span> <span class="ow">in</span> <span class="bp">self</span><span class="o">.</span><span class="n">ordered_cdna_match</span><span class="p">:</span>
</span><span id="Transcript.find_exon_number-389"><a href="#Transcript.find_exon_number-389"><span class="linenos">389</span></a>            <span class="k">if</span> <span class="p">(</span>
</span><span id="Transcript.find_exon_number-390"><a href="#Transcript.find_exon_number-390"><span class="linenos">390</span></a>                <span class="n">cdna_match</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_start</span>
</span><span id="Transcript.find_exon_number-391"><a href="#Transcript.find_exon_number-391"><span class="linenos">391</span></a>                <span class="o">&lt;=</span> <span class="n">offset</span>
</span><span id="Transcript.find_exon_number-392"><a href="#Transcript.find_exon_number-392"><span class="linenos">392</span></a>                <span class="o">&lt;=</span> <span class="n">cdna_match</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_stop</span>
</span><span id="Transcript.find_exon_number-393"><a href="#Transcript.find_exon_number-393"><span class="linenos">393</span></a>            <span class="p">):</span>
</span><span id="Transcript.find_exon_number-394"><a href="#Transcript.find_exon_number-394"><span class="linenos">394</span></a>                <span class="k">return</span> <span class="n">exon_number</span>
</span><span id="Transcript.find_exon_number-395"><a href="#Transcript.find_exon_number-395"><span class="linenos">395</span></a>            <span class="n">exon_number</span> <span class="o">+=</span> <span class="mi">1</span>
</span><span id="Transcript.find_exon_number-396"><a href="#Transcript.find_exon_number-396"><span class="linenos">396</span></a>        <span class="k">return</span> <span class="kc">None</span>
</span></pre></div>


            <div class="docstring"><p>The function <code><a href="#Transcript.find_exon_number">find_exon_number</a></code> returns the exon number for a given position.</p>

<h6 id="parameters">Parameters</h6>

<ul>
<li><strong>offset</strong>:  The offset parameter represents a position in the genome. It is an integer value
that indicates the position of interest within the genome</li>
</ul>

<h6 id="returns">Returns</h6>

<blockquote>
  <p>an integer value, which represents the exon number for a given position.</p>
</blockquote>
</div>


                            </div>
                </section>
                <section id="BED6Interval_base">
                    <div class="attr class">
            
    <span class="def">class</span>
    <span class="name">BED6Interval_base</span><wbr>(<span class="base">builtins.tuple</span>):

        
    </div>
    <a class="headerlink" href="#BED6Interval_base"></a>
    
            <div class="docstring"><p>BED6Interval_base(chrom, chrom_start, chrom_end, name, score, strand)</p>
</div>


                            <div id="BED6Interval_base.__init__" class="classattr">
                                <div class="attr function">
            
        <span class="name">BED6Interval_base</span><span class="signature pdoc-code condensed">(<span class="param"><span class="n">chrom</span>, </span><span class="param"><span class="n">chrom_start</span>, </span><span class="param"><span class="n">chrom_end</span>, </span><span class="param"><span class="n">name</span>, </span><span class="param"><span class="n">score</span>, </span><span class="param"><span class="n">strand</span></span>)</span>

        
    </div>
    <a class="headerlink" href="#BED6Interval_base.__init__"></a>
    
            <div class="docstring"><p>Create new instance of BED6Interval_base(chrom, chrom_start, chrom_end, name, score, strand)</p>
</div>


                            </div>
                            <div id="BED6Interval_base.chrom" class="classattr">
                                <div class="attr variable">
            <span class="name">chrom</span>

        
    </div>
    <a class="headerlink" href="#BED6Interval_base.chrom"></a>
    
            <div class="docstring"><p>Alias for field number 0</p>
</div>


                            </div>
                            <div id="BED6Interval_base.chrom_start" class="classattr">
                                <div class="attr variable">
            <span class="name">chrom_start</span>

        
    </div>
    <a class="headerlink" href="#BED6Interval_base.chrom_start"></a>
    
            <div class="docstring"><p>Alias for field number 1</p>
</div>


                            </div>
                            <div id="BED6Interval_base.chrom_end" class="classattr">
                                <div class="attr variable">
            <span class="name">chrom_end</span>

        
    </div>
    <a class="headerlink" href="#BED6Interval_base.chrom_end"></a>
    
            <div class="docstring"><p>Alias for field number 2</p>
</div>


                            </div>
                            <div id="BED6Interval_base.name" class="classattr">
                                <div class="attr variable">
            <span class="name">name</span>

        
    </div>
    <a class="headerlink" href="#BED6Interval_base.name"></a>
    
            <div class="docstring"><p>Alias for field number 3</p>
</div>


                            </div>
                            <div id="BED6Interval_base.score" class="classattr">
                                <div class="attr variable">
            <span class="name">score</span>

        
    </div>
    <a class="headerlink" href="#BED6Interval_base.score"></a>
    
            <div class="docstring"><p>Alias for field number 4</p>
</div>


                            </div>
                            <div id="BED6Interval_base.strand" class="classattr">
                                <div class="attr variable">
            <span class="name">strand</span>

        
    </div>
    <a class="headerlink" href="#BED6Interval_base.strand"></a>
    
            <div class="docstring"><p>Alias for field number 5</p>
</div>


                            </div>
                            <div class="inherited">
                                <h5>Inherited Members</h5>
                                <dl>
                                    <div><dt>builtins.tuple</dt>
                                <dd id="BED6Interval_base.index" class="function">index</dd>
                <dd id="BED6Interval_base.count" class="function">count</dd>

            </div>
                                </dl>
                            </div>
                </section>
                <section id="BED6Interval">
                            <input id="BED6Interval-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr class">
            
    <span class="def">class</span>
    <span class="name">BED6Interval</span><wbr>(<span class="base"><a href="#BED6Interval_base">BED6Interval_base</a></span>):

                <label class="view-source-button" for="BED6Interval-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#BED6Interval"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="BED6Interval-406"><a href="#BED6Interval-406"><span class="linenos">406</span></a><span class="k">class</span> <span class="nc">BED6Interval</span><span class="p">(</span><span class="n">BED6Interval_base</span><span class="p">):</span>
</span><span id="BED6Interval-407"><a href="#BED6Interval-407"><span class="linenos">407</span></a>
</span><span id="BED6Interval-408"><a href="#BED6Interval-408"><span class="linenos">408</span></a>    <span class="k">def</span> <span class="nf">distance</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">offset</span><span class="p">:</span> <span class="nb">int</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">int</span><span class="p">:</span>
</span><span id="BED6Interval-409"><a href="#BED6Interval-409"><span class="linenos">409</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="BED6Interval-410"><a href="#BED6Interval-410"><span class="linenos">410</span></a><span class="sd">        The `distance` function calculates the distance between an offset and an interval, returning</span>
</span><span id="BED6Interval-411"><a href="#BED6Interval-411"><span class="linenos">411</span></a><span class="sd">        zero if the offset is inside the interval, a positive value if the interval comes after the</span>
</span><span id="BED6Interval-412"><a href="#BED6Interval-412"><span class="linenos">412</span></a><span class="sd">        offset, and a negative value if the interval comes before the offset.</span>
</span><span id="BED6Interval-413"><a href="#BED6Interval-413"><span class="linenos">413</span></a><span class="sd">        if offset is inside the exon, distance is zero.</span>
</span><span id="BED6Interval-414"><a href="#BED6Interval-414"><span class="linenos">414</span></a><span class="sd">        otherwise, distance is the distance to the nearest edge.</span>
</span><span id="BED6Interval-415"><a href="#BED6Interval-415"><span class="linenos">415</span></a><span class="sd">        distance is positive if the exon comes after the offset.</span>
</span><span id="BED6Interval-416"><a href="#BED6Interval-416"><span class="linenos">416</span></a><span class="sd">        distance is negative if the exon comes before the offset.</span>
</span><span id="BED6Interval-417"><a href="#BED6Interval-417"><span class="linenos">417</span></a>
</span><span id="BED6Interval-418"><a href="#BED6Interval-418"><span class="linenos">418</span></a><span class="sd">        :param offset: The offset parameter represents a position or point in the genome. It is an</span>
</span><span id="BED6Interval-419"><a href="#BED6Interval-419"><span class="linenos">419</span></a><span class="sd">        integer value that indicates the position within the genome sequence</span>
</span><span id="BED6Interval-420"><a href="#BED6Interval-420"><span class="linenos">420</span></a><span class="sd">        :type offset: int</span>
</span><span id="BED6Interval-421"><a href="#BED6Interval-421"><span class="linenos">421</span></a><span class="sd">        :return: an integer value, which represents the distance to the interval.</span>
</span><span id="BED6Interval-422"><a href="#BED6Interval-422"><span class="linenos">422</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="BED6Interval-423"><a href="#BED6Interval-423"><span class="linenos">423</span></a>
</span><span id="BED6Interval-424"><a href="#BED6Interval-424"><span class="linenos">424</span></a>        <span class="n">start</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">chrom_start</span> <span class="o">+</span> <span class="mi">1</span>
</span><span id="BED6Interval-425"><a href="#BED6Interval-425"><span class="linenos">425</span></a>        <span class="n">end</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">chrom_end</span>
</span><span id="BED6Interval-426"><a href="#BED6Interval-426"><span class="linenos">426</span></a>
</span><span id="BED6Interval-427"><a href="#BED6Interval-427"><span class="linenos">427</span></a>        <span class="k">if</span> <span class="n">start</span> <span class="o">&lt;=</span> <span class="n">offset</span> <span class="o">&lt;=</span> <span class="n">end</span><span class="p">:</span>
</span><span id="BED6Interval-428"><a href="#BED6Interval-428"><span class="linenos">428</span></a>            <span class="k">return</span> <span class="mi">0</span>
</span><span id="BED6Interval-429"><a href="#BED6Interval-429"><span class="linenos">429</span></a>
</span><span id="BED6Interval-430"><a href="#BED6Interval-430"><span class="linenos">430</span></a>        <span class="n">start_distance</span> <span class="o">=</span> <span class="n">start</span> <span class="o">-</span> <span class="n">offset</span>
</span><span id="BED6Interval-431"><a href="#BED6Interval-431"><span class="linenos">431</span></a>        <span class="n">end_distance</span> <span class="o">=</span> <span class="n">offset</span> <span class="o">-</span> <span class="n">end</span>
</span><span id="BED6Interval-432"><a href="#BED6Interval-432"><span class="linenos">432</span></a>
</span><span id="BED6Interval-433"><a href="#BED6Interval-433"><span class="linenos">433</span></a>        <span class="k">if</span> <span class="nb">abs</span><span class="p">(</span><span class="n">start_distance</span><span class="p">)</span> <span class="o">&lt;</span> <span class="nb">abs</span><span class="p">(</span><span class="n">end_distance</span><span class="p">):</span>
</span><span id="BED6Interval-434"><a href="#BED6Interval-434"><span class="linenos">434</span></a>            <span class="k">return</span> <span class="n">start_distance</span>
</span><span id="BED6Interval-435"><a href="#BED6Interval-435"><span class="linenos">435</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="BED6Interval-436"><a href="#BED6Interval-436"><span class="linenos">436</span></a>            <span class="k">return</span> <span class="o">-</span><span class="n">end_distance</span>
</span></pre></div>


            <div class="docstring"><p>BED6Interval_base(chrom, chrom_start, chrom_end, name, score, strand)</p>
</div>


                            <div id="BED6Interval.__init__" class="classattr">
                                <div class="attr function">
            
        <span class="name">BED6Interval</span><span class="signature pdoc-code condensed">(<span class="param"><span class="n">chrom</span>, </span><span class="param"><span class="n">chrom_start</span>, </span><span class="param"><span class="n">chrom_end</span>, </span><span class="param"><span class="n">name</span>, </span><span class="param"><span class="n">score</span>, </span><span class="param"><span class="n">strand</span></span>)</span>

        
    </div>
    <a class="headerlink" href="#BED6Interval.__init__"></a>
    
            <div class="docstring"><p>Create new instance of BED6Interval_base(chrom, chrom_start, chrom_end, name, score, strand)</p>
</div>


                            </div>
                            <div id="BED6Interval.distance" class="classattr">
                                        <input id="BED6Interval.distance-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr function">
            
        <span class="def">def</span>
        <span class="name">distance</span><span class="signature pdoc-code condensed">(<span class="param"><span class="bp">self</span>, </span><span class="param"><span class="n">offset</span><span class="p">:</span> <span class="nb">int</span></span><span class="return-annotation">) -> <span class="nb">int</span>:</span></span>

                <label class="view-source-button" for="BED6Interval.distance-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#BED6Interval.distance"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="BED6Interval.distance-408"><a href="#BED6Interval.distance-408"><span class="linenos">408</span></a>    <span class="k">def</span> <span class="nf">distance</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">offset</span><span class="p">:</span> <span class="nb">int</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">int</span><span class="p">:</span>
</span><span id="BED6Interval.distance-409"><a href="#BED6Interval.distance-409"><span class="linenos">409</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="BED6Interval.distance-410"><a href="#BED6Interval.distance-410"><span class="linenos">410</span></a><span class="sd">        The `distance` function calculates the distance between an offset and an interval, returning</span>
</span><span id="BED6Interval.distance-411"><a href="#BED6Interval.distance-411"><span class="linenos">411</span></a><span class="sd">        zero if the offset is inside the interval, a positive value if the interval comes after the</span>
</span><span id="BED6Interval.distance-412"><a href="#BED6Interval.distance-412"><span class="linenos">412</span></a><span class="sd">        offset, and a negative value if the interval comes before the offset.</span>
</span><span id="BED6Interval.distance-413"><a href="#BED6Interval.distance-413"><span class="linenos">413</span></a><span class="sd">        if offset is inside the exon, distance is zero.</span>
</span><span id="BED6Interval.distance-414"><a href="#BED6Interval.distance-414"><span class="linenos">414</span></a><span class="sd">        otherwise, distance is the distance to the nearest edge.</span>
</span><span id="BED6Interval.distance-415"><a href="#BED6Interval.distance-415"><span class="linenos">415</span></a><span class="sd">        distance is positive if the exon comes after the offset.</span>
</span><span id="BED6Interval.distance-416"><a href="#BED6Interval.distance-416"><span class="linenos">416</span></a><span class="sd">        distance is negative if the exon comes before the offset.</span>
</span><span id="BED6Interval.distance-417"><a href="#BED6Interval.distance-417"><span class="linenos">417</span></a>
</span><span id="BED6Interval.distance-418"><a href="#BED6Interval.distance-418"><span class="linenos">418</span></a><span class="sd">        :param offset: The offset parameter represents a position or point in the genome. It is an</span>
</span><span id="BED6Interval.distance-419"><a href="#BED6Interval.distance-419"><span class="linenos">419</span></a><span class="sd">        integer value that indicates the position within the genome sequence</span>
</span><span id="BED6Interval.distance-420"><a href="#BED6Interval.distance-420"><span class="linenos">420</span></a><span class="sd">        :type offset: int</span>
</span><span id="BED6Interval.distance-421"><a href="#BED6Interval.distance-421"><span class="linenos">421</span></a><span class="sd">        :return: an integer value, which represents the distance to the interval.</span>
</span><span id="BED6Interval.distance-422"><a href="#BED6Interval.distance-422"><span class="linenos">422</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="BED6Interval.distance-423"><a href="#BED6Interval.distance-423"><span class="linenos">423</span></a>
</span><span id="BED6Interval.distance-424"><a href="#BED6Interval.distance-424"><span class="linenos">424</span></a>        <span class="n">start</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">chrom_start</span> <span class="o">+</span> <span class="mi">1</span>
</span><span id="BED6Interval.distance-425"><a href="#BED6Interval.distance-425"><span class="linenos">425</span></a>        <span class="n">end</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">chrom_end</span>
</span><span id="BED6Interval.distance-426"><a href="#BED6Interval.distance-426"><span class="linenos">426</span></a>
</span><span id="BED6Interval.distance-427"><a href="#BED6Interval.distance-427"><span class="linenos">427</span></a>        <span class="k">if</span> <span class="n">start</span> <span class="o">&lt;=</span> <span class="n">offset</span> <span class="o">&lt;=</span> <span class="n">end</span><span class="p">:</span>
</span><span id="BED6Interval.distance-428"><a href="#BED6Interval.distance-428"><span class="linenos">428</span></a>            <span class="k">return</span> <span class="mi">0</span>
</span><span id="BED6Interval.distance-429"><a href="#BED6Interval.distance-429"><span class="linenos">429</span></a>
</span><span id="BED6Interval.distance-430"><a href="#BED6Interval.distance-430"><span class="linenos">430</span></a>        <span class="n">start_distance</span> <span class="o">=</span> <span class="n">start</span> <span class="o">-</span> <span class="n">offset</span>
</span><span id="BED6Interval.distance-431"><a href="#BED6Interval.distance-431"><span class="linenos">431</span></a>        <span class="n">end_distance</span> <span class="o">=</span> <span class="n">offset</span> <span class="o">-</span> <span class="n">end</span>
</span><span id="BED6Interval.distance-432"><a href="#BED6Interval.distance-432"><span class="linenos">432</span></a>
</span><span id="BED6Interval.distance-433"><a href="#BED6Interval.distance-433"><span class="linenos">433</span></a>        <span class="k">if</span> <span class="nb">abs</span><span class="p">(</span><span class="n">start_distance</span><span class="p">)</span> <span class="o">&lt;</span> <span class="nb">abs</span><span class="p">(</span><span class="n">end_distance</span><span class="p">):</span>
</span><span id="BED6Interval.distance-434"><a href="#BED6Interval.distance-434"><span class="linenos">434</span></a>            <span class="k">return</span> <span class="n">start_distance</span>
</span><span id="BED6Interval.distance-435"><a href="#BED6Interval.distance-435"><span class="linenos">435</span></a>        <span class="k">else</span><span class="p">:</span>
</span><span id="BED6Interval.distance-436"><a href="#BED6Interval.distance-436"><span class="linenos">436</span></a>            <span class="k">return</span> <span class="o">-</span><span class="n">end_distance</span>
</span></pre></div>


            <div class="docstring"><p>The <code><a href="#BED6Interval.distance">distance</a></code> function calculates the distance between an offset and an interval, returning
zero if the offset is inside the interval, a positive value if the interval comes after the
offset, and a negative value if the interval comes before the offset.
if offset is inside the exon, distance is zero.
otherwise, distance is the distance to the nearest edge.
distance is positive if the exon comes after the offset.
distance is negative if the exon comes before the offset.</p>

<h6 id="parameters">Parameters</h6>

<ul>
<li><strong>offset</strong>:  The offset parameter represents a position or point in the genome. It is an
integer value that indicates the position within the genome sequence</li>
</ul>

<h6 id="returns">Returns</h6>

<blockquote>
  <p>an integer value, which represents the distance to the interval.</p>
</blockquote>
</div>


                            </div>
                            <div class="inherited">
                                <h5>Inherited Members</h5>
                                <dl>
                                    <div><dt><a href="#BED6Interval_base">BED6Interval_base</a></dt>
                                <dd id="BED6Interval.chrom" class="variable"><a href="#BED6Interval_base.chrom">chrom</a></dd>
                <dd id="BED6Interval.chrom_start" class="variable"><a href="#BED6Interval_base.chrom_start">chrom_start</a></dd>
                <dd id="BED6Interval.chrom_end" class="variable"><a href="#BED6Interval_base.chrom_end">chrom_end</a></dd>
                <dd id="BED6Interval.name" class="variable"><a href="#BED6Interval_base.name">name</a></dd>
                <dd id="BED6Interval.score" class="variable"><a href="#BED6Interval_base.score">score</a></dd>
                <dd id="BED6Interval.strand" class="variable"><a href="#BED6Interval_base.strand">strand</a></dd>

            </div>
            <div><dt>builtins.tuple</dt>
                                <dd id="BED6Interval.index" class="function">index</dd>
                <dd id="BED6Interval.count" class="function">count</dd>

            </div>
                                </dl>
                            </div>
                </section>
                <section id="Exon">
                            <input id="Exon-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr class">
            
    <span class="def">class</span>
    <span class="name">Exon</span>:

                <label class="view-source-button" for="Exon-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#Exon"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="Exon-441"><a href="#Exon-441"><span class="linenos">441</span></a><span class="k">class</span> <span class="nc">Exon</span><span class="p">(</span><span class="nb">object</span><span class="p">):</span>
</span><span id="Exon-442"><a href="#Exon-442"><span class="linenos">442</span></a>
</span><span id="Exon-443"><a href="#Exon-443"><span class="linenos">443</span></a>    <span class="k">def</span> <span class="fm">__init__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">transcript</span><span class="p">:</span> <span class="n">Transcript</span><span class="p">,</span> <span class="n">tx_position</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span> <span class="n">number</span><span class="p">:</span> <span class="nb">int</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
</span><span id="Exon-444"><a href="#Exon-444"><span class="linenos">444</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="Exon-445"><a href="#Exon-445"><span class="linenos">445</span></a><span class="sd">        The function initializes an object with a transcript, a position in the transcript, and a</span>
</span><span id="Exon-446"><a href="#Exon-446"><span class="linenos">446</span></a><span class="sd">        number.</span>
</span><span id="Exon-447"><a href="#Exon-447"><span class="linenos">447</span></a>
</span><span id="Exon-448"><a href="#Exon-448"><span class="linenos">448</span></a><span class="sd">        :param transcript: The `transcript` parameter is of type `Transcript`. It represents a</span>
</span><span id="Exon-449"><a href="#Exon-449"><span class="linenos">449</span></a><span class="sd">        transcript object that contains information about a conversation or dialogue</span>
</span><span id="Exon-450"><a href="#Exon-450"><span class="linenos">450</span></a><span class="sd">        :type transcript: Transcript</span>
</span><span id="Exon-451"><a href="#Exon-451"><span class="linenos">451</span></a><span class="sd">        :param tx_position: The `tx_position` parameter represents the position of the transcript in a</span>
</span><span id="Exon-452"><a href="#Exon-452"><span class="linenos">452</span></a><span class="sd">        list or array. It is an integer value that indicates the index of the transcript in the list or</span>
</span><span id="Exon-453"><a href="#Exon-453"><span class="linenos">453</span></a><span class="sd">        array</span>
</span><span id="Exon-454"><a href="#Exon-454"><span class="linenos">454</span></a><span class="sd">        :type tx_position: int</span>
</span><span id="Exon-455"><a href="#Exon-455"><span class="linenos">455</span></a><span class="sd">        :param number: The &quot;number&quot; parameter is an integer that represents a specific number. It is</span>
</span><span id="Exon-456"><a href="#Exon-456"><span class="linenos">456</span></a><span class="sd">        used as a parameter in the constructor of a class</span>
</span><span id="Exon-457"><a href="#Exon-457"><span class="linenos">457</span></a><span class="sd">        :type number: int</span>
</span><span id="Exon-458"><a href="#Exon-458"><span class="linenos">458</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="Exon-459"><a href="#Exon-459"><span class="linenos">459</span></a>
</span><span id="Exon-460"><a href="#Exon-460"><span class="linenos">460</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span> <span class="o">=</span> <span class="n">transcript</span>
</span><span id="Exon-461"><a href="#Exon-461"><span class="linenos">461</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span> <span class="o">=</span> <span class="n">tx_position</span>
</span><span id="Exon-462"><a href="#Exon-462"><span class="linenos">462</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">number</span> <span class="o">=</span> <span class="n">number</span>
</span><span id="Exon-463"><a href="#Exon-463"><span class="linenos">463</span></a>
</span><span id="Exon-464"><a href="#Exon-464"><span class="linenos">464</span></a>    <span class="nd">@property</span>
</span><span id="Exon-465"><a href="#Exon-465"><span class="linenos">465</span></a>    <span class="k">def</span> <span class="nf">name</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
</span><span id="Exon-466"><a href="#Exon-466"><span class="linenos">466</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="Exon-467"><a href="#Exon-467"><span class="linenos">467</span></a><span class="sd">        The function returns a string that combines the name of the transcript and a number.</span>
</span><span id="Exon-468"><a href="#Exon-468"><span class="linenos">468</span></a><span class="sd">        :return: a string that combines the name of the transcript with the number. The format of the</span>
</span><span id="Exon-469"><a href="#Exon-469"><span class="linenos">469</span></a><span class="sd">        string is &quot;{transcript name}.{number}&quot;.</span>
</span><span id="Exon-470"><a href="#Exon-470"><span class="linenos">470</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="Exon-471"><a href="#Exon-471"><span class="linenos">471</span></a>
</span><span id="Exon-472"><a href="#Exon-472"><span class="linenos">472</span></a>        <span class="k">return</span> <span class="s2">&quot;</span><span class="si">%s</span><span class="s2">.</span><span class="si">%d</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">transcript</span><span class="o">.</span><span class="n">name</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">number</span><span class="p">)</span>
</span><span id="Exon-473"><a href="#Exon-473"><span class="linenos">473</span></a>
</span><span id="Exon-474"><a href="#Exon-474"><span class="linenos">474</span></a>    <span class="k">def</span> <span class="nf">get_as_interval</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">coding_only</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">object</span><span class="p">:</span>
</span><span id="Exon-475"><a href="#Exon-475"><span class="linenos">475</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="Exon-476"><a href="#Exon-476"><span class="linenos">476</span></a><span class="sd">        The function `get_as_interval` returns the coding region for an exon as a BED6Interval object.</span>
</span><span id="Exon-477"><a href="#Exon-477"><span class="linenos">477</span></a><span class="sd">        This function returns a BED6Interval objects containing  position</span>
</span><span id="Exon-478"><a href="#Exon-478"><span class="linenos">478</span></a><span class="sd">        information for this exon. This may be used as input for</span>
</span><span id="Exon-479"><a href="#Exon-479"><span class="linenos">479</span></a><span class="sd">        pybedtools.create_interval_from_list() after casting chrom_start</span>
</span><span id="Exon-480"><a href="#Exon-480"><span class="linenos">480</span></a><span class="sd">        and chrom_end as strings.</span>
</span><span id="Exon-481"><a href="#Exon-481"><span class="linenos">481</span></a>
</span><span id="Exon-482"><a href="#Exon-482"><span class="linenos">482</span></a><span class="sd">        :param coding_only: The `coding_only` parameter is a boolean flag that determines whether to</span>
</span><span id="Exon-483"><a href="#Exon-483"><span class="linenos">483</span></a><span class="sd">        include only exons in the coding region. If `coding_only` is set to `True`, the function will</span>
</span><span id="Exon-484"><a href="#Exon-484"><span class="linenos">484</span></a><span class="sd">        check if the exon is completely outside the coding region defined by the transcript&#39;s CDS</span>
</span><span id="Exon-485"><a href="#Exon-485"><span class="linenos">485</span></a><span class="sd">        (coding sequence) position, defaults to False</span>
</span><span id="Exon-486"><a href="#Exon-486"><span class="linenos">486</span></a><span class="sd">        :type coding_only: bool (optional)</span>
</span><span id="Exon-487"><a href="#Exon-487"><span class="linenos">487</span></a><span class="sd">        :return: a BED6Interval object.</span>
</span><span id="Exon-488"><a href="#Exon-488"><span class="linenos">488</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="Exon-489"><a href="#Exon-489"><span class="linenos">489</span></a>
</span><span id="Exon-490"><a href="#Exon-490"><span class="linenos">490</span></a>        <span class="n">exon_start</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_start</span>
</span><span id="Exon-491"><a href="#Exon-491"><span class="linenos">491</span></a>        <span class="n">exon_stop</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_stop</span>
</span><span id="Exon-492"><a href="#Exon-492"><span class="linenos">492</span></a>
</span><span id="Exon-493"><a href="#Exon-493"><span class="linenos">493</span></a>        <span class="c1"># Get only exon coding region if requested</span>
</span><span id="Exon-494"><a href="#Exon-494"><span class="linenos">494</span></a>        <span class="k">if</span> <span class="n">coding_only</span><span class="p">:</span>
</span><span id="Exon-495"><a href="#Exon-495"><span class="linenos">495</span></a>            <span class="k">if</span> <span class="p">(</span>
</span><span id="Exon-496"><a href="#Exon-496"><span class="linenos">496</span></a>                <span class="n">exon_stop</span> <span class="o">&lt;=</span> <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span><span class="o">.</span><span class="n">cds_position</span><span class="o">.</span><span class="n">chrom_start</span>
</span><span id="Exon-497"><a href="#Exon-497"><span class="linenos">497</span></a>                <span class="ow">or</span> <span class="n">exon_start</span> <span class="o">&gt;=</span> <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span><span class="o">.</span><span class="n">cds_position</span><span class="o">.</span><span class="n">chrom_stop</span>
</span><span id="Exon-498"><a href="#Exon-498"><span class="linenos">498</span></a>            <span class="p">):</span>
</span><span id="Exon-499"><a href="#Exon-499"><span class="linenos">499</span></a>                <span class="k">return</span> <span class="kc">None</span>
</span><span id="Exon-500"><a href="#Exon-500"><span class="linenos">500</span></a>            <span class="n">exon_start</span> <span class="o">=</span> <span class="nb">max</span><span class="p">(</span><span class="n">exon_start</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span><span class="o">.</span><span class="n">cds_position</span><span class="o">.</span><span class="n">chrom_start</span><span class="p">)</span>
</span><span id="Exon-501"><a href="#Exon-501"><span class="linenos">501</span></a>            <span class="n">exon_stop</span> <span class="o">=</span> <span class="nb">min</span><span class="p">(</span>
</span><span id="Exon-502"><a href="#Exon-502"><span class="linenos">502</span></a>                <span class="nb">max</span><span class="p">(</span><span class="n">exon_stop</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span><span class="o">.</span><span class="n">cds_position</span><span class="o">.</span><span class="n">chrom_start</span><span class="p">),</span>
</span><span id="Exon-503"><a href="#Exon-503"><span class="linenos">503</span></a>                <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span><span class="o">.</span><span class="n">cds_position</span><span class="o">.</span><span class="n">chrom_stop</span><span class="p">,</span>
</span><span id="Exon-504"><a href="#Exon-504"><span class="linenos">504</span></a>            <span class="p">)</span>
</span><span id="Exon-505"><a href="#Exon-505"><span class="linenos">505</span></a>
</span><span id="Exon-506"><a href="#Exon-506"><span class="linenos">506</span></a>        <span class="k">return</span> <span class="n">BED6Interval</span><span class="p">(</span>
</span><span id="Exon-507"><a href="#Exon-507"><span class="linenos">507</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom</span><span class="p">,</span>
</span><span id="Exon-508"><a href="#Exon-508"><span class="linenos">508</span></a>            <span class="n">exon_start</span><span class="p">,</span>
</span><span id="Exon-509"><a href="#Exon-509"><span class="linenos">509</span></a>            <span class="n">exon_stop</span><span class="p">,</span>
</span><span id="Exon-510"><a href="#Exon-510"><span class="linenos">510</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">name</span><span class="p">,</span>
</span><span id="Exon-511"><a href="#Exon-511"><span class="linenos">511</span></a>            <span class="s2">&quot;.&quot;</span><span class="p">,</span>
</span><span id="Exon-512"><a href="#Exon-512"><span class="linenos">512</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">strand</span><span class="p">,</span>
</span><span id="Exon-513"><a href="#Exon-513"><span class="linenos">513</span></a>        <span class="p">)</span>
</span><span id="Exon-514"><a href="#Exon-514"><span class="linenos">514</span></a>
</span><span id="Exon-515"><a href="#Exon-515"><span class="linenos">515</span></a>    <span class="nd">@property</span>
</span><span id="Exon-516"><a href="#Exon-516"><span class="linenos">516</span></a>    <span class="k">def</span> <span class="nf">strand</span><span class="p">(</span><span class="bp">self</span><span class="p">):</span>
</span><span id="Exon-517"><a href="#Exon-517"><span class="linenos">517</span></a>        <span class="n">strand</span> <span class="o">=</span> <span class="s2">&quot;+&quot;</span> <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">is_forward_strand</span> <span class="k">else</span> <span class="s2">&quot;-&quot;</span>
</span><span id="Exon-518"><a href="#Exon-518"><span class="linenos">518</span></a>        <span class="k">return</span> <span class="n">strand</span>
</span></pre></div>


    

                            <div id="Exon.__init__" class="classattr">
                                        <input id="Exon.__init__-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr function">
            
        <span class="name">Exon</span><span class="signature pdoc-code multiline">(<span class="param">	<span class="n">transcript</span><span class="p">:</span> <span class="n"><a href="#Transcript">Transcript</a></span>,</span><span class="param">	<span class="n">tx_position</span><span class="p">:</span> <span class="nb">int</span>,</span><span class="param">	<span class="n">number</span><span class="p">:</span> <span class="nb">int</span></span>)</span>

                <label class="view-source-button" for="Exon.__init__-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#Exon.__init__"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="Exon.__init__-443"><a href="#Exon.__init__-443"><span class="linenos">443</span></a>    <span class="k">def</span> <span class="fm">__init__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">transcript</span><span class="p">:</span> <span class="n">Transcript</span><span class="p">,</span> <span class="n">tx_position</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span> <span class="n">number</span><span class="p">:</span> <span class="nb">int</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
</span><span id="Exon.__init__-444"><a href="#Exon.__init__-444"><span class="linenos">444</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="Exon.__init__-445"><a href="#Exon.__init__-445"><span class="linenos">445</span></a><span class="sd">        The function initializes an object with a transcript, a position in the transcript, and a</span>
</span><span id="Exon.__init__-446"><a href="#Exon.__init__-446"><span class="linenos">446</span></a><span class="sd">        number.</span>
</span><span id="Exon.__init__-447"><a href="#Exon.__init__-447"><span class="linenos">447</span></a>
</span><span id="Exon.__init__-448"><a href="#Exon.__init__-448"><span class="linenos">448</span></a><span class="sd">        :param transcript: The `transcript` parameter is of type `Transcript`. It represents a</span>
</span><span id="Exon.__init__-449"><a href="#Exon.__init__-449"><span class="linenos">449</span></a><span class="sd">        transcript object that contains information about a conversation or dialogue</span>
</span><span id="Exon.__init__-450"><a href="#Exon.__init__-450"><span class="linenos">450</span></a><span class="sd">        :type transcript: Transcript</span>
</span><span id="Exon.__init__-451"><a href="#Exon.__init__-451"><span class="linenos">451</span></a><span class="sd">        :param tx_position: The `tx_position` parameter represents the position of the transcript in a</span>
</span><span id="Exon.__init__-452"><a href="#Exon.__init__-452"><span class="linenos">452</span></a><span class="sd">        list or array. It is an integer value that indicates the index of the transcript in the list or</span>
</span><span id="Exon.__init__-453"><a href="#Exon.__init__-453"><span class="linenos">453</span></a><span class="sd">        array</span>
</span><span id="Exon.__init__-454"><a href="#Exon.__init__-454"><span class="linenos">454</span></a><span class="sd">        :type tx_position: int</span>
</span><span id="Exon.__init__-455"><a href="#Exon.__init__-455"><span class="linenos">455</span></a><span class="sd">        :param number: The &quot;number&quot; parameter is an integer that represents a specific number. It is</span>
</span><span id="Exon.__init__-456"><a href="#Exon.__init__-456"><span class="linenos">456</span></a><span class="sd">        used as a parameter in the constructor of a class</span>
</span><span id="Exon.__init__-457"><a href="#Exon.__init__-457"><span class="linenos">457</span></a><span class="sd">        :type number: int</span>
</span><span id="Exon.__init__-458"><a href="#Exon.__init__-458"><span class="linenos">458</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="Exon.__init__-459"><a href="#Exon.__init__-459"><span class="linenos">459</span></a>
</span><span id="Exon.__init__-460"><a href="#Exon.__init__-460"><span class="linenos">460</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span> <span class="o">=</span> <span class="n">transcript</span>
</span><span id="Exon.__init__-461"><a href="#Exon.__init__-461"><span class="linenos">461</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span> <span class="o">=</span> <span class="n">tx_position</span>
</span><span id="Exon.__init__-462"><a href="#Exon.__init__-462"><span class="linenos">462</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">number</span> <span class="o">=</span> <span class="n">number</span>
</span></pre></div>


            <div class="docstring"><p>The function initializes an object with a transcript, a position in the transcript, and a
number.</p>

<h6 id="parameters">Parameters</h6>

<ul>
<li><strong>transcript</strong>:  The <code><a href="#Exon.transcript">transcript</a></code> parameter is of type <code><a href="#Transcript">Transcript</a></code>. It represents a
transcript object that contains information about a conversation or dialogue</li>
<li><strong>tx_position</strong>:  The <code><a href="#Exon.tx_position">tx_position</a></code> parameter represents the position of the transcript in a
list or array. It is an integer value that indicates the index of the transcript in the list or
array</li>
<li><strong>number</strong>:  The "number" parameter is an integer that represents a specific number. It is
used as a parameter in the constructor of a class</li>
</ul>
</div>


                            </div>
                            <div id="Exon.transcript" class="classattr">
                                <div class="attr variable">
            <span class="name">transcript</span>

        
    </div>
    <a class="headerlink" href="#Exon.transcript"></a>
    
    

                            </div>
                            <div id="Exon.tx_position" class="classattr">
                                <div class="attr variable">
            <span class="name">tx_position</span>

        
    </div>
    <a class="headerlink" href="#Exon.tx_position"></a>
    
    

                            </div>
                            <div id="Exon.number" class="classattr">
                                <div class="attr variable">
            <span class="name">number</span>

        
    </div>
    <a class="headerlink" href="#Exon.number"></a>
    
    

                            </div>
                            <div id="Exon.name" class="classattr">
                                        <input id="Exon.name-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr variable">
            <span class="name">name</span><span class="annotation">: str</span>

                <label class="view-source-button" for="Exon.name-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#Exon.name"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="Exon.name-464"><a href="#Exon.name-464"><span class="linenos">464</span></a>    <span class="nd">@property</span>
</span><span id="Exon.name-465"><a href="#Exon.name-465"><span class="linenos">465</span></a>    <span class="k">def</span> <span class="nf">name</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">str</span><span class="p">:</span>
</span><span id="Exon.name-466"><a href="#Exon.name-466"><span class="linenos">466</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="Exon.name-467"><a href="#Exon.name-467"><span class="linenos">467</span></a><span class="sd">        The function returns a string that combines the name of the transcript and a number.</span>
</span><span id="Exon.name-468"><a href="#Exon.name-468"><span class="linenos">468</span></a><span class="sd">        :return: a string that combines the name of the transcript with the number. The format of the</span>
</span><span id="Exon.name-469"><a href="#Exon.name-469"><span class="linenos">469</span></a><span class="sd">        string is &quot;{transcript name}.{number}&quot;.</span>
</span><span id="Exon.name-470"><a href="#Exon.name-470"><span class="linenos">470</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="Exon.name-471"><a href="#Exon.name-471"><span class="linenos">471</span></a>
</span><span id="Exon.name-472"><a href="#Exon.name-472"><span class="linenos">472</span></a>        <span class="k">return</span> <span class="s2">&quot;</span><span class="si">%s</span><span class="s2">.</span><span class="si">%d</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">transcript</span><span class="o">.</span><span class="n">name</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">number</span><span class="p">)</span>
</span></pre></div>


            <div class="docstring"><p>The function returns a string that combines the name of the transcript and a number.</p>

<h6 id="returns">Returns</h6>

<blockquote>
  <p>a string that combines the name of the transcript with the number. The format of the
  string is "{transcript name}.{number}".</p>
</blockquote>
</div>


                            </div>
                            <div id="Exon.get_as_interval" class="classattr">
                                        <input id="Exon.get_as_interval-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr function">
            
        <span class="def">def</span>
        <span class="name">get_as_interval</span><span class="signature pdoc-code condensed">(<span class="param"><span class="bp">self</span>, </span><span class="param"><span class="n">coding_only</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span></span><span class="return-annotation">) -> <span class="nb">object</span>:</span></span>

                <label class="view-source-button" for="Exon.get_as_interval-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#Exon.get_as_interval"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="Exon.get_as_interval-474"><a href="#Exon.get_as_interval-474"><span class="linenos">474</span></a>    <span class="k">def</span> <span class="nf">get_as_interval</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">coding_only</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">False</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">object</span><span class="p">:</span>
</span><span id="Exon.get_as_interval-475"><a href="#Exon.get_as_interval-475"><span class="linenos">475</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="Exon.get_as_interval-476"><a href="#Exon.get_as_interval-476"><span class="linenos">476</span></a><span class="sd">        The function `get_as_interval` returns the coding region for an exon as a BED6Interval object.</span>
</span><span id="Exon.get_as_interval-477"><a href="#Exon.get_as_interval-477"><span class="linenos">477</span></a><span class="sd">        This function returns a BED6Interval objects containing  position</span>
</span><span id="Exon.get_as_interval-478"><a href="#Exon.get_as_interval-478"><span class="linenos">478</span></a><span class="sd">        information for this exon. This may be used as input for</span>
</span><span id="Exon.get_as_interval-479"><a href="#Exon.get_as_interval-479"><span class="linenos">479</span></a><span class="sd">        pybedtools.create_interval_from_list() after casting chrom_start</span>
</span><span id="Exon.get_as_interval-480"><a href="#Exon.get_as_interval-480"><span class="linenos">480</span></a><span class="sd">        and chrom_end as strings.</span>
</span><span id="Exon.get_as_interval-481"><a href="#Exon.get_as_interval-481"><span class="linenos">481</span></a>
</span><span id="Exon.get_as_interval-482"><a href="#Exon.get_as_interval-482"><span class="linenos">482</span></a><span class="sd">        :param coding_only: The `coding_only` parameter is a boolean flag that determines whether to</span>
</span><span id="Exon.get_as_interval-483"><a href="#Exon.get_as_interval-483"><span class="linenos">483</span></a><span class="sd">        include only exons in the coding region. If `coding_only` is set to `True`, the function will</span>
</span><span id="Exon.get_as_interval-484"><a href="#Exon.get_as_interval-484"><span class="linenos">484</span></a><span class="sd">        check if the exon is completely outside the coding region defined by the transcript&#39;s CDS</span>
</span><span id="Exon.get_as_interval-485"><a href="#Exon.get_as_interval-485"><span class="linenos">485</span></a><span class="sd">        (coding sequence) position, defaults to False</span>
</span><span id="Exon.get_as_interval-486"><a href="#Exon.get_as_interval-486"><span class="linenos">486</span></a><span class="sd">        :type coding_only: bool (optional)</span>
</span><span id="Exon.get_as_interval-487"><a href="#Exon.get_as_interval-487"><span class="linenos">487</span></a><span class="sd">        :return: a BED6Interval object.</span>
</span><span id="Exon.get_as_interval-488"><a href="#Exon.get_as_interval-488"><span class="linenos">488</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="Exon.get_as_interval-489"><a href="#Exon.get_as_interval-489"><span class="linenos">489</span></a>
</span><span id="Exon.get_as_interval-490"><a href="#Exon.get_as_interval-490"><span class="linenos">490</span></a>        <span class="n">exon_start</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_start</span>
</span><span id="Exon.get_as_interval-491"><a href="#Exon.get_as_interval-491"><span class="linenos">491</span></a>        <span class="n">exon_stop</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom_stop</span>
</span><span id="Exon.get_as_interval-492"><a href="#Exon.get_as_interval-492"><span class="linenos">492</span></a>
</span><span id="Exon.get_as_interval-493"><a href="#Exon.get_as_interval-493"><span class="linenos">493</span></a>        <span class="c1"># Get only exon coding region if requested</span>
</span><span id="Exon.get_as_interval-494"><a href="#Exon.get_as_interval-494"><span class="linenos">494</span></a>        <span class="k">if</span> <span class="n">coding_only</span><span class="p">:</span>
</span><span id="Exon.get_as_interval-495"><a href="#Exon.get_as_interval-495"><span class="linenos">495</span></a>            <span class="k">if</span> <span class="p">(</span>
</span><span id="Exon.get_as_interval-496"><a href="#Exon.get_as_interval-496"><span class="linenos">496</span></a>                <span class="n">exon_stop</span> <span class="o">&lt;=</span> <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span><span class="o">.</span><span class="n">cds_position</span><span class="o">.</span><span class="n">chrom_start</span>
</span><span id="Exon.get_as_interval-497"><a href="#Exon.get_as_interval-497"><span class="linenos">497</span></a>                <span class="ow">or</span> <span class="n">exon_start</span> <span class="o">&gt;=</span> <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span><span class="o">.</span><span class="n">cds_position</span><span class="o">.</span><span class="n">chrom_stop</span>
</span><span id="Exon.get_as_interval-498"><a href="#Exon.get_as_interval-498"><span class="linenos">498</span></a>            <span class="p">):</span>
</span><span id="Exon.get_as_interval-499"><a href="#Exon.get_as_interval-499"><span class="linenos">499</span></a>                <span class="k">return</span> <span class="kc">None</span>
</span><span id="Exon.get_as_interval-500"><a href="#Exon.get_as_interval-500"><span class="linenos">500</span></a>            <span class="n">exon_start</span> <span class="o">=</span> <span class="nb">max</span><span class="p">(</span><span class="n">exon_start</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span><span class="o">.</span><span class="n">cds_position</span><span class="o">.</span><span class="n">chrom_start</span><span class="p">)</span>
</span><span id="Exon.get_as_interval-501"><a href="#Exon.get_as_interval-501"><span class="linenos">501</span></a>            <span class="n">exon_stop</span> <span class="o">=</span> <span class="nb">min</span><span class="p">(</span>
</span><span id="Exon.get_as_interval-502"><a href="#Exon.get_as_interval-502"><span class="linenos">502</span></a>                <span class="nb">max</span><span class="p">(</span><span class="n">exon_stop</span><span class="p">,</span> <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span><span class="o">.</span><span class="n">cds_position</span><span class="o">.</span><span class="n">chrom_start</span><span class="p">),</span>
</span><span id="Exon.get_as_interval-503"><a href="#Exon.get_as_interval-503"><span class="linenos">503</span></a>                <span class="bp">self</span><span class="o">.</span><span class="n">transcript</span><span class="o">.</span><span class="n">cds_position</span><span class="o">.</span><span class="n">chrom_stop</span><span class="p">,</span>
</span><span id="Exon.get_as_interval-504"><a href="#Exon.get_as_interval-504"><span class="linenos">504</span></a>            <span class="p">)</span>
</span><span id="Exon.get_as_interval-505"><a href="#Exon.get_as_interval-505"><span class="linenos">505</span></a>
</span><span id="Exon.get_as_interval-506"><a href="#Exon.get_as_interval-506"><span class="linenos">506</span></a>        <span class="k">return</span> <span class="n">BED6Interval</span><span class="p">(</span>
</span><span id="Exon.get_as_interval-507"><a href="#Exon.get_as_interval-507"><span class="linenos">507</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">chrom</span><span class="p">,</span>
</span><span id="Exon.get_as_interval-508"><a href="#Exon.get_as_interval-508"><span class="linenos">508</span></a>            <span class="n">exon_start</span><span class="p">,</span>
</span><span id="Exon.get_as_interval-509"><a href="#Exon.get_as_interval-509"><span class="linenos">509</span></a>            <span class="n">exon_stop</span><span class="p">,</span>
</span><span id="Exon.get_as_interval-510"><a href="#Exon.get_as_interval-510"><span class="linenos">510</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">name</span><span class="p">,</span>
</span><span id="Exon.get_as_interval-511"><a href="#Exon.get_as_interval-511"><span class="linenos">511</span></a>            <span class="s2">&quot;.&quot;</span><span class="p">,</span>
</span><span id="Exon.get_as_interval-512"><a href="#Exon.get_as_interval-512"><span class="linenos">512</span></a>            <span class="bp">self</span><span class="o">.</span><span class="n">strand</span><span class="p">,</span>
</span><span id="Exon.get_as_interval-513"><a href="#Exon.get_as_interval-513"><span class="linenos">513</span></a>        <span class="p">)</span>
</span></pre></div>


            <div class="docstring"><p>The function <code><a href="#Exon.get_as_interval">get_as_interval</a></code> returns the coding region for an exon as a BED6Interval object.
This function returns a BED6Interval objects containing  position
information for this exon. This may be used as input for
pybedtools.create_interval_from_list() after casting chrom_start
and chrom_end as strings.</p>

<h6 id="parameters">Parameters</h6>

<ul>
<li><strong>coding_only</strong>:  The <code>coding_only</code> parameter is a boolean flag that determines whether to
include only exons in the coding region. If <code>coding_only</code> is set to <code>True</code>, the function will
check if the exon is completely outside the coding region defined by the transcript's CDS
(coding sequence) position, defaults to False</li>
</ul>

<h6 id="returns">Returns</h6>

<blockquote>
  <p>a BED6Interval object.</p>
</blockquote>
</div>


                            </div>
                            <div id="Exon.strand" class="classattr">
                                        <input id="Exon.strand-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr variable">
            <span class="name">strand</span>

                <label class="view-source-button" for="Exon.strand-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#Exon.strand"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="Exon.strand-515"><a href="#Exon.strand-515"><span class="linenos">515</span></a>    <span class="nd">@property</span>
</span><span id="Exon.strand-516"><a href="#Exon.strand-516"><span class="linenos">516</span></a>    <span class="k">def</span> <span class="nf">strand</span><span class="p">(</span><span class="bp">self</span><span class="p">):</span>
</span><span id="Exon.strand-517"><a href="#Exon.strand-517"><span class="linenos">517</span></a>        <span class="n">strand</span> <span class="o">=</span> <span class="s2">&quot;+&quot;</span> <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">tx_position</span><span class="o">.</span><span class="n">is_forward_strand</span> <span class="k">else</span> <span class="s2">&quot;-&quot;</span>
</span><span id="Exon.strand-518"><a href="#Exon.strand-518"><span class="linenos">518</span></a>        <span class="k">return</span> <span class="n">strand</span>
</span></pre></div>


    

                            </div>
                </section>
                <section id="CDNA_Match">
                            <input id="CDNA_Match-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr class">
            
    <span class="def">class</span>
    <span class="name">CDNA_Match</span><wbr>(<span class="base"><a href="#Exon">Exon</a></span>):

                <label class="view-source-button" for="CDNA_Match-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#CDNA_Match"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="CDNA_Match-522"><a href="#CDNA_Match-522"><span class="linenos">522</span></a><span class="k">class</span> <span class="nc">CDNA_Match</span><span class="p">(</span><span class="n">Exon</span><span class="p">):</span>
</span><span id="CDNA_Match-523"><a href="#CDNA_Match-523"><span class="linenos">523</span></a>
</span><span id="CDNA_Match-524"><a href="#CDNA_Match-524"><span class="linenos">524</span></a>    <span class="k">def</span> <span class="fm">__init__</span><span class="p">(</span>
</span><span id="CDNA_Match-525"><a href="#CDNA_Match-525"><span class="linenos">525</span></a>        <span class="bp">self</span><span class="p">,</span>
</span><span id="CDNA_Match-526"><a href="#CDNA_Match-526"><span class="linenos">526</span></a>        <span class="n">transcript</span><span class="p">:</span> <span class="n">Transcript</span><span class="p">,</span>
</span><span id="CDNA_Match-527"><a href="#CDNA_Match-527"><span class="linenos">527</span></a>        <span class="n">tx_position</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span>
</span><span id="CDNA_Match-528"><a href="#CDNA_Match-528"><span class="linenos">528</span></a>        <span class="n">cdna_start</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span>
</span><span id="CDNA_Match-529"><a href="#CDNA_Match-529"><span class="linenos">529</span></a>        <span class="n">cdna_end</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span>
</span><span id="CDNA_Match-530"><a href="#CDNA_Match-530"><span class="linenos">530</span></a>        <span class="n">gap</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span>
</span><span id="CDNA_Match-531"><a href="#CDNA_Match-531"><span class="linenos">531</span></a>        <span class="n">number</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span>
</span><span id="CDNA_Match-532"><a href="#CDNA_Match-532"><span class="linenos">532</span></a>    <span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
</span><span id="CDNA_Match-533"><a href="#CDNA_Match-533"><span class="linenos">533</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="CDNA_Match-534"><a href="#CDNA_Match-534"><span class="linenos">534</span></a><span class="sd">        The function initializes a CDNA_Match object with specified attributes.</span>
</span><span id="CDNA_Match-535"><a href="#CDNA_Match-535"><span class="linenos">535</span></a>
</span><span id="CDNA_Match-536"><a href="#CDNA_Match-536"><span class="linenos">536</span></a><span class="sd">        :param transcript: The `transcript` parameter is an instance of the `Transcript` class. It</span>
</span><span id="CDNA_Match-537"><a href="#CDNA_Match-537"><span class="linenos">537</span></a><span class="sd">        represents the transcript that the CDNA match belongs to</span>
</span><span id="CDNA_Match-538"><a href="#CDNA_Match-538"><span class="linenos">538</span></a><span class="sd">        :type transcript: Transcript</span>
</span><span id="CDNA_Match-539"><a href="#CDNA_Match-539"><span class="linenos">539</span></a><span class="sd">        :param tx_position: The `tx_position` parameter represents the position of the transcript in the</span>
</span><span id="CDNA_Match-540"><a href="#CDNA_Match-540"><span class="linenos">540</span></a><span class="sd">        genome. It is an integer value</span>
</span><span id="CDNA_Match-541"><a href="#CDNA_Match-541"><span class="linenos">541</span></a><span class="sd">        :type tx_position: int</span>
</span><span id="CDNA_Match-542"><a href="#CDNA_Match-542"><span class="linenos">542</span></a><span class="sd">        :param cdna_start: The `cdna_start` parameter represents the starting position of the cDNA</span>
</span><span id="CDNA_Match-543"><a href="#CDNA_Match-543"><span class="linenos">543</span></a><span class="sd">        match. It is an integer value</span>
</span><span id="CDNA_Match-544"><a href="#CDNA_Match-544"><span class="linenos">544</span></a><span class="sd">        :type cdna_start: int</span>
</span><span id="CDNA_Match-545"><a href="#CDNA_Match-545"><span class="linenos">545</span></a><span class="sd">        :param cdna_end: The `cdna_end` parameter represents the end position of the cDNA match. It is</span>
</span><span id="CDNA_Match-546"><a href="#CDNA_Match-546"><span class="linenos">546</span></a><span class="sd">        an integer value that indicates the position of the last nucleotide in the cDNA sequence that</span>
</span><span id="CDNA_Match-547"><a href="#CDNA_Match-547"><span class="linenos">547</span></a><span class="sd">        matches the transcript</span>
</span><span id="CDNA_Match-548"><a href="#CDNA_Match-548"><span class="linenos">548</span></a><span class="sd">        :type cdna_end: int</span>
</span><span id="CDNA_Match-549"><a href="#CDNA_Match-549"><span class="linenos">549</span></a><span class="sd">        :param gap: The &quot;gap&quot; parameter represents the number of nucleotides that are missing or</span>
</span><span id="CDNA_Match-550"><a href="#CDNA_Match-550"><span class="linenos">550</span></a><span class="sd">        inserted in the cDNA sequence compared to the reference transcript sequence. It indicates the</span>
</span><span id="CDNA_Match-551"><a href="#CDNA_Match-551"><span class="linenos">551</span></a><span class="sd">        presence of gaps or insertions in the alignment between the cDNA and the reference transcript</span>
</span><span id="CDNA_Match-552"><a href="#CDNA_Match-552"><span class="linenos">552</span></a><span class="sd">        :type gap: int</span>
</span><span id="CDNA_Match-553"><a href="#CDNA_Match-553"><span class="linenos">553</span></a><span class="sd">        :param number: The `number` parameter represents the number of the CDNA match. It is used to</span>
</span><span id="CDNA_Match-554"><a href="#CDNA_Match-554"><span class="linenos">554</span></a><span class="sd">        uniquely identify each CDNA match object</span>
</span><span id="CDNA_Match-555"><a href="#CDNA_Match-555"><span class="linenos">555</span></a><span class="sd">        :type number: int</span>
</span><span id="CDNA_Match-556"><a href="#CDNA_Match-556"><span class="linenos">556</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="CDNA_Match-557"><a href="#CDNA_Match-557"><span class="linenos">557</span></a>
</span><span id="CDNA_Match-558"><a href="#CDNA_Match-558"><span class="linenos">558</span></a>        <span class="nb">super</span><span class="p">(</span><span class="n">CDNA_Match</span><span class="p">,</span> <span class="bp">self</span><span class="p">)</span><span class="o">.</span><span class="fm">__init__</span><span class="p">(</span><span class="n">transcript</span><span class="p">,</span> <span class="n">tx_position</span><span class="p">,</span> <span class="n">number</span><span class="p">)</span>
</span><span id="CDNA_Match-559"><a href="#CDNA_Match-559"><span class="linenos">559</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">cdna_start</span> <span class="o">=</span> <span class="n">cdna_start</span>
</span><span id="CDNA_Match-560"><a href="#CDNA_Match-560"><span class="linenos">560</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">cdna_end</span> <span class="o">=</span> <span class="n">cdna_end</span>
</span><span id="CDNA_Match-561"><a href="#CDNA_Match-561"><span class="linenos">561</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">gap</span> <span class="o">=</span> <span class="n">gap</span>
</span><span id="CDNA_Match-562"><a href="#CDNA_Match-562"><span class="linenos">562</span></a>
</span><span id="CDNA_Match-563"><a href="#CDNA_Match-563"><span class="linenos">563</span></a>    <span class="nd">@property</span>
</span><span id="CDNA_Match-564"><a href="#CDNA_Match-564"><span class="linenos">564</span></a>    <span class="k">def</span> <span class="nf">length</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">int</span><span class="p">:</span>
</span><span id="CDNA_Match-565"><a href="#CDNA_Match-565"><span class="linenos">565</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="CDNA_Match-566"><a href="#CDNA_Match-566"><span class="linenos">566</span></a><span class="sd">        The function calculates the length of a sequence by subtracting the start position from the end</span>
</span><span id="CDNA_Match-567"><a href="#CDNA_Match-567"><span class="linenos">567</span></a><span class="sd">        position and adding 1.</span>
</span><span id="CDNA_Match-568"><a href="#CDNA_Match-568"><span class="linenos">568</span></a><span class="sd">        :return: The length of the sequence, calculated by subtracting the cdna_start from the cdna_end</span>
</span><span id="CDNA_Match-569"><a href="#CDNA_Match-569"><span class="linenos">569</span></a><span class="sd">        and adding 1.</span>
</span><span id="CDNA_Match-570"><a href="#CDNA_Match-570"><span class="linenos">570</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="CDNA_Match-571"><a href="#CDNA_Match-571"><span class="linenos">571</span></a>
</span><span id="CDNA_Match-572"><a href="#CDNA_Match-572"><span class="linenos">572</span></a>        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">cdna_end</span> <span class="o">-</span> <span class="bp">self</span><span class="o">.</span><span class="n">cdna_start</span> <span class="o">+</span> <span class="mi">1</span>
</span><span id="CDNA_Match-573"><a href="#CDNA_Match-573"><span class="linenos">573</span></a>
</span><span id="CDNA_Match-574"><a href="#CDNA_Match-574"><span class="linenos">574</span></a>    <span class="k">def</span> <span class="nf">get_offset</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">position</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span> <span class="n">validate</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">True</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">int</span><span class="p">:</span>
</span><span id="CDNA_Match-575"><a href="#CDNA_Match-575"><span class="linenos">575</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="CDNA_Match-576"><a href="#CDNA_Match-576"><span class="linenos">576</span></a><span class="sd">        The `get_offset` function calculates the offset for a given position in a cDNA sequence based on</span>
</span><span id="CDNA_Match-577"><a href="#CDNA_Match-577"><span class="linenos">577</span></a><span class="sd">        the GAP attribute.</span>
</span><span id="CDNA_Match-578"><a href="#CDNA_Match-578"><span class="linenos">578</span></a><span class="sd">        cdna_match GAP attribute looks like: &#39;M185 I3 M250&#39; which is code/length</span>
</span><span id="CDNA_Match-579"><a href="#CDNA_Match-579"><span class="linenos">579</span></a><span class="sd">        @see https://github.com/The-Sequence-Ontology/Specifications/blob/master/gff3.md#the-gap-attribute</span>
</span><span id="CDNA_Match-580"><a href="#CDNA_Match-580"><span class="linenos">580</span></a><span class="sd">        codes operation</span>
</span><span id="CDNA_Match-581"><a href="#CDNA_Match-581"><span class="linenos">581</span></a><span class="sd">        M 	match</span>
</span><span id="CDNA_Match-582"><a href="#CDNA_Match-582"><span class="linenos">582</span></a><span class="sd">        I 	insert a gap into the reference sequence</span>
</span><span id="CDNA_Match-583"><a href="#CDNA_Match-583"><span class="linenos">583</span></a><span class="sd">        D 	insert a gap into the target (delete from reference)</span>
</span><span id="CDNA_Match-584"><a href="#CDNA_Match-584"><span class="linenos">584</span></a><span class="sd">        If you want the whole exon, then pass the end</span>
</span><span id="CDNA_Match-585"><a href="#CDNA_Match-585"><span class="linenos">585</span></a>
</span><span id="CDNA_Match-586"><a href="#CDNA_Match-586"><span class="linenos">586</span></a><span class="sd">        :param position: The `position` parameter is an integer that represents the position in the</span>
</span><span id="CDNA_Match-587"><a href="#CDNA_Match-587"><span class="linenos">587</span></a><span class="sd">        sequence. It is used to calculate the offset based on the GAP attribute of the cDNA match</span>
</span><span id="CDNA_Match-588"><a href="#CDNA_Match-588"><span class="linenos">588</span></a><span class="sd">        :type position: int</span>
</span><span id="CDNA_Match-589"><a href="#CDNA_Match-589"><span class="linenos">589</span></a><span class="sd">        :param validate: The `validate` parameter is a boolean flag that determines whether to perform</span>
</span><span id="CDNA_Match-590"><a href="#CDNA_Match-590"><span class="linenos">590</span></a><span class="sd">        validation checks during the calculation of the offset. If `validate` is set to `True`, the</span>
</span><span id="CDNA_Match-591"><a href="#CDNA_Match-591"><span class="linenos">591</span></a><span class="sd">        function will raise a `ValueError` if the given position falls within an insertion or deletion</span>
</span><span id="CDNA_Match-592"><a href="#CDNA_Match-592"><span class="linenos">592</span></a><span class="sd">        gap. If `validate` is set, defaults to True</span>
</span><span id="CDNA_Match-593"><a href="#CDNA_Match-593"><span class="linenos">593</span></a><span class="sd">        :type validate: bool (optional)</span>
</span><span id="CDNA_Match-594"><a href="#CDNA_Match-594"><span class="linenos">594</span></a><span class="sd">        :return: an integer value representing the offset for a given position in the cDNA sequence.</span>
</span><span id="CDNA_Match-595"><a href="#CDNA_Match-595"><span class="linenos">595</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="CDNA_Match-596"><a href="#CDNA_Match-596"><span class="linenos">596</span></a>
</span><span id="CDNA_Match-597"><a href="#CDNA_Match-597"><span class="linenos">597</span></a>        <span class="k">if</span> <span class="ow">not</span> <span class="bp">self</span><span class="o">.</span><span class="n">gap</span><span class="p">:</span>
</span><span id="CDNA_Match-598"><a href="#CDNA_Match-598"><span class="linenos">598</span></a>            <span class="k">return</span> <span class="mi">0</span>
</span><span id="CDNA_Match-599"><a href="#CDNA_Match-599"><span class="linenos">599</span></a>
</span><span id="CDNA_Match-600"><a href="#CDNA_Match-600"><span class="linenos">600</span></a>        <span class="n">position_1_based</span> <span class="o">=</span> <span class="n">position</span> <span class="o">+</span> <span class="mi">1</span>
</span><span id="CDNA_Match-601"><a href="#CDNA_Match-601"><span class="linenos">601</span></a>        <span class="n">cdna_match_index</span> <span class="o">=</span> <span class="mi">1</span>
</span><span id="CDNA_Match-602"><a href="#CDNA_Match-602"><span class="linenos">602</span></a>        <span class="n">offset</span> <span class="o">=</span> <span class="mi">0</span>
</span><span id="CDNA_Match-603"><a href="#CDNA_Match-603"><span class="linenos">603</span></a>        <span class="k">for</span> <span class="n">gap_op</span> <span class="ow">in</span> <span class="bp">self</span><span class="o">.</span><span class="n">gap</span><span class="o">.</span><span class="n">split</span><span class="p">():</span>
</span><span id="CDNA_Match-604"><a href="#CDNA_Match-604"><span class="linenos">604</span></a>            <span class="n">code</span> <span class="o">=</span> <span class="n">gap_op</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>
</span><span id="CDNA_Match-605"><a href="#CDNA_Match-605"><span class="linenos">605</span></a>            <span class="n">length</span> <span class="o">=</span> <span class="nb">int</span><span class="p">(</span><span class="n">gap_op</span><span class="p">[</span><span class="mi">1</span><span class="p">:])</span>
</span><span id="CDNA_Match-606"><a href="#CDNA_Match-606"><span class="linenos">606</span></a>            <span class="k">if</span> <span class="n">code</span> <span class="o">==</span> <span class="s2">&quot;M&quot;</span><span class="p">:</span>
</span><span id="CDNA_Match-607"><a href="#CDNA_Match-607"><span class="linenos">607</span></a>                <span class="n">cdna_match_index</span> <span class="o">+=</span> <span class="n">length</span>
</span><span id="CDNA_Match-608"><a href="#CDNA_Match-608"><span class="linenos">608</span></a>            <span class="k">elif</span> <span class="n">code</span> <span class="o">==</span> <span class="s2">&quot;I&quot;</span><span class="p">:</span>
</span><span id="CDNA_Match-609"><a href="#CDNA_Match-609"><span class="linenos">609</span></a>                <span class="k">if</span> <span class="n">validate</span> <span class="ow">and</span> <span class="n">position_1_based</span> <span class="o">&lt;</span> <span class="n">cdna_match_index</span> <span class="o">+</span> <span class="n">length</span><span class="p">:</span>
</span><span id="CDNA_Match-610"><a href="#CDNA_Match-610"><span class="linenos">610</span></a>                    <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span>
</span><span id="CDNA_Match-611"><a href="#CDNA_Match-611"><span class="linenos">611</span></a>                        <span class="s2">&quot;Coordinate (</span><span class="si">%d</span><span class="s2">) inside insertion (</span><span class="si">%s</span><span class="s2">) - no mapping possible!&quot;</span>
</span><span id="CDNA_Match-612"><a href="#CDNA_Match-612"><span class="linenos">612</span></a>                        <span class="o">%</span> <span class="p">(</span><span class="n">position_1_based</span><span class="p">,</span> <span class="n">gap_op</span><span class="p">)</span>
</span><span id="CDNA_Match-613"><a href="#CDNA_Match-613"><span class="linenos">613</span></a>                    <span class="p">)</span>
</span><span id="CDNA_Match-614"><a href="#CDNA_Match-614"><span class="linenos">614</span></a>                <span class="n">offset</span> <span class="o">+=</span> <span class="n">length</span>
</span><span id="CDNA_Match-615"><a href="#CDNA_Match-615"><span class="linenos">615</span></a>            <span class="k">elif</span> <span class="n">code</span> <span class="o">==</span> <span class="s2">&quot;D&quot;</span><span class="p">:</span>
</span><span id="CDNA_Match-616"><a href="#CDNA_Match-616"><span class="linenos">616</span></a>                <span class="k">if</span> <span class="n">validate</span> <span class="ow">and</span> <span class="n">position</span> <span class="o">&lt;</span> <span class="n">cdna_match_index</span> <span class="o">+</span> <span class="n">length</span><span class="p">:</span>
</span><span id="CDNA_Match-617"><a href="#CDNA_Match-617"><span class="linenos">617</span></a>                    <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span>
</span><span id="CDNA_Match-618"><a href="#CDNA_Match-618"><span class="linenos">618</span></a>                        <span class="s2">&quot;Coordinate (</span><span class="si">%d</span><span class="s2">) inside deletion (</span><span class="si">%s</span><span class="s2">) - no mapping possible!&quot;</span>
</span><span id="CDNA_Match-619"><a href="#CDNA_Match-619"><span class="linenos">619</span></a>                        <span class="o">%</span> <span class="p">(</span><span class="n">position_1_based</span><span class="p">,</span> <span class="n">gap_op</span><span class="p">)</span>
</span><span id="CDNA_Match-620"><a href="#CDNA_Match-620"><span class="linenos">620</span></a>                    <span class="p">)</span>
</span><span id="CDNA_Match-621"><a href="#CDNA_Match-621"><span class="linenos">621</span></a>                <span class="n">offset</span> <span class="o">-=</span> <span class="n">length</span>
</span><span id="CDNA_Match-622"><a href="#CDNA_Match-622"><span class="linenos">622</span></a>            <span class="k">else</span><span class="p">:</span>
</span><span id="CDNA_Match-623"><a href="#CDNA_Match-623"><span class="linenos">623</span></a>                <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Unknown code in cDNA GAP: </span><span class="si">{</span><span class="n">gap_op</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
</span><span id="CDNA_Match-624"><a href="#CDNA_Match-624"><span class="linenos">624</span></a>
</span><span id="CDNA_Match-625"><a href="#CDNA_Match-625"><span class="linenos">625</span></a>            <span class="k">if</span> <span class="n">cdna_match_index</span> <span class="o">&gt;</span> <span class="n">position_1_based</span><span class="p">:</span>
</span><span id="CDNA_Match-626"><a href="#CDNA_Match-626"><span class="linenos">626</span></a>                <span class="k">break</span>
</span><span id="CDNA_Match-627"><a href="#CDNA_Match-627"><span class="linenos">627</span></a>
</span><span id="CDNA_Match-628"><a href="#CDNA_Match-628"><span class="linenos">628</span></a>        <span class="k">return</span> <span class="n">offset</span>
</span></pre></div>


    

                            <div id="CDNA_Match.__init__" class="classattr">
                                        <input id="CDNA_Match.__init__-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr function">
            
        <span class="name">CDNA_Match</span><span class="signature pdoc-code multiline">(<span class="param">	<span class="n">transcript</span><span class="p">:</span> <span class="n"><a href="#Transcript">Transcript</a></span>,</span><span class="param">	<span class="n">tx_position</span><span class="p">:</span> <span class="nb">int</span>,</span><span class="param">	<span class="n">cdna_start</span><span class="p">:</span> <span class="nb">int</span>,</span><span class="param">	<span class="n">cdna_end</span><span class="p">:</span> <span class="nb">int</span>,</span><span class="param">	<span class="n">gap</span><span class="p">:</span> <span class="nb">int</span>,</span><span class="param">	<span class="n">number</span><span class="p">:</span> <span class="nb">int</span></span>)</span>

                <label class="view-source-button" for="CDNA_Match.__init__-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#CDNA_Match.__init__"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="CDNA_Match.__init__-524"><a href="#CDNA_Match.__init__-524"><span class="linenos">524</span></a>    <span class="k">def</span> <span class="fm">__init__</span><span class="p">(</span>
</span><span id="CDNA_Match.__init__-525"><a href="#CDNA_Match.__init__-525"><span class="linenos">525</span></a>        <span class="bp">self</span><span class="p">,</span>
</span><span id="CDNA_Match.__init__-526"><a href="#CDNA_Match.__init__-526"><span class="linenos">526</span></a>        <span class="n">transcript</span><span class="p">:</span> <span class="n">Transcript</span><span class="p">,</span>
</span><span id="CDNA_Match.__init__-527"><a href="#CDNA_Match.__init__-527"><span class="linenos">527</span></a>        <span class="n">tx_position</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span>
</span><span id="CDNA_Match.__init__-528"><a href="#CDNA_Match.__init__-528"><span class="linenos">528</span></a>        <span class="n">cdna_start</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span>
</span><span id="CDNA_Match.__init__-529"><a href="#CDNA_Match.__init__-529"><span class="linenos">529</span></a>        <span class="n">cdna_end</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span>
</span><span id="CDNA_Match.__init__-530"><a href="#CDNA_Match.__init__-530"><span class="linenos">530</span></a>        <span class="n">gap</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span>
</span><span id="CDNA_Match.__init__-531"><a href="#CDNA_Match.__init__-531"><span class="linenos">531</span></a>        <span class="n">number</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span>
</span><span id="CDNA_Match.__init__-532"><a href="#CDNA_Match.__init__-532"><span class="linenos">532</span></a>    <span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
</span><span id="CDNA_Match.__init__-533"><a href="#CDNA_Match.__init__-533"><span class="linenos">533</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="CDNA_Match.__init__-534"><a href="#CDNA_Match.__init__-534"><span class="linenos">534</span></a><span class="sd">        The function initializes a CDNA_Match object with specified attributes.</span>
</span><span id="CDNA_Match.__init__-535"><a href="#CDNA_Match.__init__-535"><span class="linenos">535</span></a>
</span><span id="CDNA_Match.__init__-536"><a href="#CDNA_Match.__init__-536"><span class="linenos">536</span></a><span class="sd">        :param transcript: The `transcript` parameter is an instance of the `Transcript` class. It</span>
</span><span id="CDNA_Match.__init__-537"><a href="#CDNA_Match.__init__-537"><span class="linenos">537</span></a><span class="sd">        represents the transcript that the CDNA match belongs to</span>
</span><span id="CDNA_Match.__init__-538"><a href="#CDNA_Match.__init__-538"><span class="linenos">538</span></a><span class="sd">        :type transcript: Transcript</span>
</span><span id="CDNA_Match.__init__-539"><a href="#CDNA_Match.__init__-539"><span class="linenos">539</span></a><span class="sd">        :param tx_position: The `tx_position` parameter represents the position of the transcript in the</span>
</span><span id="CDNA_Match.__init__-540"><a href="#CDNA_Match.__init__-540"><span class="linenos">540</span></a><span class="sd">        genome. It is an integer value</span>
</span><span id="CDNA_Match.__init__-541"><a href="#CDNA_Match.__init__-541"><span class="linenos">541</span></a><span class="sd">        :type tx_position: int</span>
</span><span id="CDNA_Match.__init__-542"><a href="#CDNA_Match.__init__-542"><span class="linenos">542</span></a><span class="sd">        :param cdna_start: The `cdna_start` parameter represents the starting position of the cDNA</span>
</span><span id="CDNA_Match.__init__-543"><a href="#CDNA_Match.__init__-543"><span class="linenos">543</span></a><span class="sd">        match. It is an integer value</span>
</span><span id="CDNA_Match.__init__-544"><a href="#CDNA_Match.__init__-544"><span class="linenos">544</span></a><span class="sd">        :type cdna_start: int</span>
</span><span id="CDNA_Match.__init__-545"><a href="#CDNA_Match.__init__-545"><span class="linenos">545</span></a><span class="sd">        :param cdna_end: The `cdna_end` parameter represents the end position of the cDNA match. It is</span>
</span><span id="CDNA_Match.__init__-546"><a href="#CDNA_Match.__init__-546"><span class="linenos">546</span></a><span class="sd">        an integer value that indicates the position of the last nucleotide in the cDNA sequence that</span>
</span><span id="CDNA_Match.__init__-547"><a href="#CDNA_Match.__init__-547"><span class="linenos">547</span></a><span class="sd">        matches the transcript</span>
</span><span id="CDNA_Match.__init__-548"><a href="#CDNA_Match.__init__-548"><span class="linenos">548</span></a><span class="sd">        :type cdna_end: int</span>
</span><span id="CDNA_Match.__init__-549"><a href="#CDNA_Match.__init__-549"><span class="linenos">549</span></a><span class="sd">        :param gap: The &quot;gap&quot; parameter represents the number of nucleotides that are missing or</span>
</span><span id="CDNA_Match.__init__-550"><a href="#CDNA_Match.__init__-550"><span class="linenos">550</span></a><span class="sd">        inserted in the cDNA sequence compared to the reference transcript sequence. It indicates the</span>
</span><span id="CDNA_Match.__init__-551"><a href="#CDNA_Match.__init__-551"><span class="linenos">551</span></a><span class="sd">        presence of gaps or insertions in the alignment between the cDNA and the reference transcript</span>
</span><span id="CDNA_Match.__init__-552"><a href="#CDNA_Match.__init__-552"><span class="linenos">552</span></a><span class="sd">        :type gap: int</span>
</span><span id="CDNA_Match.__init__-553"><a href="#CDNA_Match.__init__-553"><span class="linenos">553</span></a><span class="sd">        :param number: The `number` parameter represents the number of the CDNA match. It is used to</span>
</span><span id="CDNA_Match.__init__-554"><a href="#CDNA_Match.__init__-554"><span class="linenos">554</span></a><span class="sd">        uniquely identify each CDNA match object</span>
</span><span id="CDNA_Match.__init__-555"><a href="#CDNA_Match.__init__-555"><span class="linenos">555</span></a><span class="sd">        :type number: int</span>
</span><span id="CDNA_Match.__init__-556"><a href="#CDNA_Match.__init__-556"><span class="linenos">556</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="CDNA_Match.__init__-557"><a href="#CDNA_Match.__init__-557"><span class="linenos">557</span></a>
</span><span id="CDNA_Match.__init__-558"><a href="#CDNA_Match.__init__-558"><span class="linenos">558</span></a>        <span class="nb">super</span><span class="p">(</span><span class="n">CDNA_Match</span><span class="p">,</span> <span class="bp">self</span><span class="p">)</span><span class="o">.</span><span class="fm">__init__</span><span class="p">(</span><span class="n">transcript</span><span class="p">,</span> <span class="n">tx_position</span><span class="p">,</span> <span class="n">number</span><span class="p">)</span>
</span><span id="CDNA_Match.__init__-559"><a href="#CDNA_Match.__init__-559"><span class="linenos">559</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">cdna_start</span> <span class="o">=</span> <span class="n">cdna_start</span>
</span><span id="CDNA_Match.__init__-560"><a href="#CDNA_Match.__init__-560"><span class="linenos">560</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">cdna_end</span> <span class="o">=</span> <span class="n">cdna_end</span>
</span><span id="CDNA_Match.__init__-561"><a href="#CDNA_Match.__init__-561"><span class="linenos">561</span></a>        <span class="bp">self</span><span class="o">.</span><span class="n">gap</span> <span class="o">=</span> <span class="n">gap</span>
</span></pre></div>


            <div class="docstring"><p>The function initializes a CDNA_Match object with specified attributes.</p>

<h6 id="parameters">Parameters</h6>

<ul>
<li><strong>transcript</strong>:  The <code><a href="#CDNA_Match.transcript">transcript</a></code> parameter is an instance of the <code><a href="#Transcript">Transcript</a></code> class. It
represents the transcript that the CDNA match belongs to</li>
<li><strong>tx_position</strong>:  The <code><a href="#CDNA_Match.tx_position">tx_position</a></code> parameter represents the position of the transcript in the
genome. It is an integer value</li>
<li><strong>cdna_start</strong>:  The <code><a href="#CDNA_Match.cdna_start">cdna_start</a></code> parameter represents the starting position of the cDNA
match. It is an integer value</li>
<li><strong>cdna_end</strong>:  The <code><a href="#CDNA_Match.cdna_end">cdna_end</a></code> parameter represents the end position of the cDNA match. It is
an integer value that indicates the position of the last nucleotide in the cDNA sequence that
matches the transcript</li>
<li><strong>gap</strong>:  The "gap" parameter represents the number of nucleotides that are missing or
inserted in the cDNA sequence compared to the reference transcript sequence. It indicates the
presence of gaps or insertions in the alignment between the cDNA and the reference transcript</li>
<li><strong>number</strong>:  The <code><a href="#CDNA_Match.number">number</a></code> parameter represents the number of the CDNA match. It is used to
uniquely identify each CDNA match object</li>
</ul>
</div>


                            </div>
                            <div id="CDNA_Match.cdna_start" class="classattr">
                                <div class="attr variable">
            <span class="name">cdna_start</span>

        
    </div>
    <a class="headerlink" href="#CDNA_Match.cdna_start"></a>
    
    

                            </div>
                            <div id="CDNA_Match.cdna_end" class="classattr">
                                <div class="attr variable">
            <span class="name">cdna_end</span>

        
    </div>
    <a class="headerlink" href="#CDNA_Match.cdna_end"></a>
    
    

                            </div>
                            <div id="CDNA_Match.gap" class="classattr">
                                <div class="attr variable">
            <span class="name">gap</span>

        
    </div>
    <a class="headerlink" href="#CDNA_Match.gap"></a>
    
    

                            </div>
                            <div id="CDNA_Match.length" class="classattr">
                                        <input id="CDNA_Match.length-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr variable">
            <span class="name">length</span><span class="annotation">: int</span>

                <label class="view-source-button" for="CDNA_Match.length-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#CDNA_Match.length"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="CDNA_Match.length-563"><a href="#CDNA_Match.length-563"><span class="linenos">563</span></a>    <span class="nd">@property</span>
</span><span id="CDNA_Match.length-564"><a href="#CDNA_Match.length-564"><span class="linenos">564</span></a>    <span class="k">def</span> <span class="nf">length</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">int</span><span class="p">:</span>
</span><span id="CDNA_Match.length-565"><a href="#CDNA_Match.length-565"><span class="linenos">565</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="CDNA_Match.length-566"><a href="#CDNA_Match.length-566"><span class="linenos">566</span></a><span class="sd">        The function calculates the length of a sequence by subtracting the start position from the end</span>
</span><span id="CDNA_Match.length-567"><a href="#CDNA_Match.length-567"><span class="linenos">567</span></a><span class="sd">        position and adding 1.</span>
</span><span id="CDNA_Match.length-568"><a href="#CDNA_Match.length-568"><span class="linenos">568</span></a><span class="sd">        :return: The length of the sequence, calculated by subtracting the cdna_start from the cdna_end</span>
</span><span id="CDNA_Match.length-569"><a href="#CDNA_Match.length-569"><span class="linenos">569</span></a><span class="sd">        and adding 1.</span>
</span><span id="CDNA_Match.length-570"><a href="#CDNA_Match.length-570"><span class="linenos">570</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="CDNA_Match.length-571"><a href="#CDNA_Match.length-571"><span class="linenos">571</span></a>
</span><span id="CDNA_Match.length-572"><a href="#CDNA_Match.length-572"><span class="linenos">572</span></a>        <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">cdna_end</span> <span class="o">-</span> <span class="bp">self</span><span class="o">.</span><span class="n">cdna_start</span> <span class="o">+</span> <span class="mi">1</span>
</span></pre></div>


            <div class="docstring"><p>The function calculates the length of a sequence by subtracting the start position from the end
position and adding 1.</p>

<h6 id="returns">Returns</h6>

<blockquote>
  <p>The length of the sequence, calculated by subtracting the cdna_start from the cdna_end
  and adding 1.</p>
</blockquote>
</div>


                            </div>
                            <div id="CDNA_Match.get_offset" class="classattr">
                                        <input id="CDNA_Match.get_offset-view-source" class="view-source-toggle-state" type="checkbox" aria-hidden="true" tabindex="-1">
<div class="attr function">
            
        <span class="def">def</span>
        <span class="name">get_offset</span><span class="signature pdoc-code condensed">(<span class="param"><span class="bp">self</span>, </span><span class="param"><span class="n">position</span><span class="p">:</span> <span class="nb">int</span>, </span><span class="param"><span class="n">validate</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">True</span></span><span class="return-annotation">) -> <span class="nb">int</span>:</span></span>

                <label class="view-source-button" for="CDNA_Match.get_offset-view-source"><span>View Source</span></label>

    </div>
    <a class="headerlink" href="#CDNA_Match.get_offset"></a>
            <div class="pdoc-code codehilite"><pre><span></span><span id="CDNA_Match.get_offset-574"><a href="#CDNA_Match.get_offset-574"><span class="linenos">574</span></a>    <span class="k">def</span> <span class="nf">get_offset</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">position</span><span class="p">:</span> <span class="nb">int</span><span class="p">,</span> <span class="n">validate</span><span class="p">:</span> <span class="nb">bool</span> <span class="o">=</span> <span class="kc">True</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">int</span><span class="p">:</span>
</span><span id="CDNA_Match.get_offset-575"><a href="#CDNA_Match.get_offset-575"><span class="linenos">575</span></a><span class="w">        </span><span class="sd">&quot;&quot;&quot;</span>
</span><span id="CDNA_Match.get_offset-576"><a href="#CDNA_Match.get_offset-576"><span class="linenos">576</span></a><span class="sd">        The `get_offset` function calculates the offset for a given position in a cDNA sequence based on</span>
</span><span id="CDNA_Match.get_offset-577"><a href="#CDNA_Match.get_offset-577"><span class="linenos">577</span></a><span class="sd">        the GAP attribute.</span>
</span><span id="CDNA_Match.get_offset-578"><a href="#CDNA_Match.get_offset-578"><span class="linenos">578</span></a><span class="sd">        cdna_match GAP attribute looks like: &#39;M185 I3 M250&#39; which is code/length</span>
</span><span id="CDNA_Match.get_offset-579"><a href="#CDNA_Match.get_offset-579"><span class="linenos">579</span></a><span class="sd">        @see https://github.com/The-Sequence-Ontology/Specifications/blob/master/gff3.md#the-gap-attribute</span>
</span><span id="CDNA_Match.get_offset-580"><a href="#CDNA_Match.get_offset-580"><span class="linenos">580</span></a><span class="sd">        codes operation</span>
</span><span id="CDNA_Match.get_offset-581"><a href="#CDNA_Match.get_offset-581"><span class="linenos">581</span></a><span class="sd">        M 	match</span>
</span><span id="CDNA_Match.get_offset-582"><a href="#CDNA_Match.get_offset-582"><span class="linenos">582</span></a><span class="sd">        I 	insert a gap into the reference sequence</span>
</span><span id="CDNA_Match.get_offset-583"><a href="#CDNA_Match.get_offset-583"><span class="linenos">583</span></a><span class="sd">        D 	insert a gap into the target (delete from reference)</span>
</span><span id="CDNA_Match.get_offset-584"><a href="#CDNA_Match.get_offset-584"><span class="linenos">584</span></a><span class="sd">        If you want the whole exon, then pass the end</span>
</span><span id="CDNA_Match.get_offset-585"><a href="#CDNA_Match.get_offset-585"><span class="linenos">585</span></a>
</span><span id="CDNA_Match.get_offset-586"><a href="#CDNA_Match.get_offset-586"><span class="linenos">586</span></a><span class="sd">        :param position: The `position` parameter is an integer that represents the position in the</span>
</span><span id="CDNA_Match.get_offset-587"><a href="#CDNA_Match.get_offset-587"><span class="linenos">587</span></a><span class="sd">        sequence. It is used to calculate the offset based on the GAP attribute of the cDNA match</span>
</span><span id="CDNA_Match.get_offset-588"><a href="#CDNA_Match.get_offset-588"><span class="linenos">588</span></a><span class="sd">        :type position: int</span>
</span><span id="CDNA_Match.get_offset-589"><a href="#CDNA_Match.get_offset-589"><span class="linenos">589</span></a><span class="sd">        :param validate: The `validate` parameter is a boolean flag that determines whether to perform</span>
</span><span id="CDNA_Match.get_offset-590"><a href="#CDNA_Match.get_offset-590"><span class="linenos">590</span></a><span class="sd">        validation checks during the calculation of the offset. If `validate` is set to `True`, the</span>
</span><span id="CDNA_Match.get_offset-591"><a href="#CDNA_Match.get_offset-591"><span class="linenos">591</span></a><span class="sd">        function will raise a `ValueError` if the given position falls within an insertion or deletion</span>
</span><span id="CDNA_Match.get_offset-592"><a href="#CDNA_Match.get_offset-592"><span class="linenos">592</span></a><span class="sd">        gap. If `validate` is set, defaults to True</span>
</span><span id="CDNA_Match.get_offset-593"><a href="#CDNA_Match.get_offset-593"><span class="linenos">593</span></a><span class="sd">        :type validate: bool (optional)</span>
</span><span id="CDNA_Match.get_offset-594"><a href="#CDNA_Match.get_offset-594"><span class="linenos">594</span></a><span class="sd">        :return: an integer value representing the offset for a given position in the cDNA sequence.</span>
</span><span id="CDNA_Match.get_offset-595"><a href="#CDNA_Match.get_offset-595"><span class="linenos">595</span></a><span class="sd">        &quot;&quot;&quot;</span>
</span><span id="CDNA_Match.get_offset-596"><a href="#CDNA_Match.get_offset-596"><span class="linenos">596</span></a>
</span><span id="CDNA_Match.get_offset-597"><a href="#CDNA_Match.get_offset-597"><span class="linenos">597</span></a>        <span class="k">if</span> <span class="ow">not</span> <span class="bp">self</span><span class="o">.</span><span class="n">gap</span><span class="p">:</span>
</span><span id="CDNA_Match.get_offset-598"><a href="#CDNA_Match.get_offset-598"><span class="linenos">598</span></a>            <span class="k">return</span> <span class="mi">0</span>
</span><span id="CDNA_Match.get_offset-599"><a href="#CDNA_Match.get_offset-599"><span class="linenos">599</span></a>
</span><span id="CDNA_Match.get_offset-600"><a href="#CDNA_Match.get_offset-600"><span class="linenos">600</span></a>        <span class="n">position_1_based</span> <span class="o">=</span> <span class="n">position</span> <span class="o">+</span> <span class="mi">1</span>
</span><span id="CDNA_Match.get_offset-601"><a href="#CDNA_Match.get_offset-601"><span class="linenos">601</span></a>        <span class="n">cdna_match_index</span> <span class="o">=</span> <span class="mi">1</span>
</span><span id="CDNA_Match.get_offset-602"><a href="#CDNA_Match.get_offset-602"><span class="linenos">602</span></a>        <span class="n">offset</span> <span class="o">=</span> <span class="mi">0</span>
</span><span id="CDNA_Match.get_offset-603"><a href="#CDNA_Match.get_offset-603"><span class="linenos">603</span></a>        <span class="k">for</span> <span class="n">gap_op</span> <span class="ow">in</span> <span class="bp">self</span><span class="o">.</span><span class="n">gap</span><span class="o">.</span><span class="n">split</span><span class="p">():</span>
</span><span id="CDNA_Match.get_offset-604"><a href="#CDNA_Match.get_offset-604"><span class="linenos">604</span></a>            <span class="n">code</span> <span class="o">=</span> <span class="n">gap_op</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>
</span><span id="CDNA_Match.get_offset-605"><a href="#CDNA_Match.get_offset-605"><span class="linenos">605</span></a>            <span class="n">length</span> <span class="o">=</span> <span class="nb">int</span><span class="p">(</span><span class="n">gap_op</span><span class="p">[</span><span class="mi">1</span><span class="p">:])</span>
</span><span id="CDNA_Match.get_offset-606"><a href="#CDNA_Match.get_offset-606"><span class="linenos">606</span></a>            <span class="k">if</span> <span class="n">code</span> <span class="o">==</span> <span class="s2">&quot;M&quot;</span><span class="p">:</span>
</span><span id="CDNA_Match.get_offset-607"><a href="#CDNA_Match.get_offset-607"><span class="linenos">607</span></a>                <span class="n">cdna_match_index</span> <span class="o">+=</span> <span class="n">length</span>
</span><span id="CDNA_Match.get_offset-608"><a href="#CDNA_Match.get_offset-608"><span class="linenos">608</span></a>            <span class="k">elif</span> <span class="n">code</span> <span class="o">==</span> <span class="s2">&quot;I&quot;</span><span class="p">:</span>
</span><span id="CDNA_Match.get_offset-609"><a href="#CDNA_Match.get_offset-609"><span class="linenos">609</span></a>                <span class="k">if</span> <span class="n">validate</span> <span class="ow">and</span> <span class="n">position_1_based</span> <span class="o">&lt;</span> <span class="n">cdna_match_index</span> <span class="o">+</span> <span class="n">length</span><span class="p">:</span>
</span><span id="CDNA_Match.get_offset-610"><a href="#CDNA_Match.get_offset-610"><span class="linenos">610</span></a>                    <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span>
</span><span id="CDNA_Match.get_offset-611"><a href="#CDNA_Match.get_offset-611"><span class="linenos">611</span></a>                        <span class="s2">&quot;Coordinate (</span><span class="si">%d</span><span class="s2">) inside insertion (</span><span class="si">%s</span><span class="s2">) - no mapping possible!&quot;</span>
</span><span id="CDNA_Match.get_offset-612"><a href="#CDNA_Match.get_offset-612"><span class="linenos">612</span></a>                        <span class="o">%</span> <span class="p">(</span><span class="n">position_1_based</span><span class="p">,</span> <span class="n">gap_op</span><span class="p">)</span>
</span><span id="CDNA_Match.get_offset-613"><a href="#CDNA_Match.get_offset-613"><span class="linenos">613</span></a>                    <span class="p">)</span>
</span><span id="CDNA_Match.get_offset-614"><a href="#CDNA_Match.get_offset-614"><span class="linenos">614</span></a>                <span class="n">offset</span> <span class="o">+=</span> <span class="n">length</span>
</span><span id="CDNA_Match.get_offset-615"><a href="#CDNA_Match.get_offset-615"><span class="linenos">615</span></a>            <span class="k">elif</span> <span class="n">code</span> <span class="o">==</span> <span class="s2">&quot;D&quot;</span><span class="p">:</span>
</span><span id="CDNA_Match.get_offset-616"><a href="#CDNA_Match.get_offset-616"><span class="linenos">616</span></a>                <span class="k">if</span> <span class="n">validate</span> <span class="ow">and</span> <span class="n">position</span> <span class="o">&lt;</span> <span class="n">cdna_match_index</span> <span class="o">+</span> <span class="n">length</span><span class="p">:</span>
</span><span id="CDNA_Match.get_offset-617"><a href="#CDNA_Match.get_offset-617"><span class="linenos">617</span></a>                    <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span>
</span><span id="CDNA_Match.get_offset-618"><a href="#CDNA_Match.get_offset-618"><span class="linenos">618</span></a>                        <span class="s2">&quot;Coordinate (</span><span class="si">%d</span><span class="s2">) inside deletion (</span><span class="si">%s</span><span class="s2">) - no mapping possible!&quot;</span>
</span><span id="CDNA_Match.get_offset-619"><a href="#CDNA_Match.get_offset-619"><span class="linenos">619</span></a>                        <span class="o">%</span> <span class="p">(</span><span class="n">position_1_based</span><span class="p">,</span> <span class="n">gap_op</span><span class="p">)</span>
</span><span id="CDNA_Match.get_offset-620"><a href="#CDNA_Match.get_offset-620"><span class="linenos">620</span></a>                    <span class="p">)</span>
</span><span id="CDNA_Match.get_offset-621"><a href="#CDNA_Match.get_offset-621"><span class="linenos">621</span></a>                <span class="n">offset</span> <span class="o">-=</span> <span class="n">length</span>
</span><span id="CDNA_Match.get_offset-622"><a href="#CDNA_Match.get_offset-622"><span class="linenos">622</span></a>            <span class="k">else</span><span class="p">:</span>
</span><span id="CDNA_Match.get_offset-623"><a href="#CDNA_Match.get_offset-623"><span class="linenos">623</span></a>                <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Unknown code in cDNA GAP: </span><span class="si">{</span><span class="n">gap_op</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
</span><span id="CDNA_Match.get_offset-624"><a href="#CDNA_Match.get_offset-624"><span class="linenos">624</span></a>
</span><span id="CDNA_Match.get_offset-625"><a href="#CDNA_Match.get_offset-625"><span class="linenos">625</span></a>            <span class="k">if</span> <span class="n">cdna_match_index</span> <span class="o">&gt;</span> <span class="n">position_1_based</span><span class="p">:</span>
</span><span id="CDNA_Match.get_offset-626"><a href="#CDNA_Match.get_offset-626"><span class="linenos">626</span></a>                <span class="k">break</span>
</span><span id="CDNA_Match.get_offset-627"><a href="#CDNA_Match.get_offset-627"><span class="linenos">627</span></a>
</span><span id="CDNA_Match.get_offset-628"><a href="#CDNA_Match.get_offset-628"><span class="linenos">628</span></a>        <span class="k">return</span> <span class="n">offset</span>
</span></pre></div>


            <div class="docstring"><p>The <code><a href="#CDNA_Match.get_offset">get_offset</a></code> function calculates the offset for a given position in a cDNA sequence based on
the GAP attribute.
cdna_match GAP attribute looks like: 'M185 I3 M250' which is code/length
@see <a href="https://github.com/The-Sequence-Ontology/Specifications/blob/master/gff3.md#the-gap-attribute">https://github.com/The-Sequence-Ontology/Specifications/blob/master/gff3.md#the-gap-attribute</a>
codes operation
M       match
I       insert a gap into the reference sequence
D       insert a gap into the target (delete from reference)
If you want the whole exon, then pass the end</p>

<h6 id="parameters">Parameters</h6>

<ul>
<li><strong>position</strong>:  The <code>position</code> parameter is an integer that represents the position in the
sequence. It is used to calculate the offset based on the GAP attribute of the cDNA match</li>
<li><strong>validate</strong>:  The <code>validate</code> parameter is a boolean flag that determines whether to perform
validation checks during the calculation of the offset. If <code>validate</code> is set to <code>True</code>, the
function will raise a <code>ValueError</code> if the given position falls within an insertion or deletion
gap. If <code>validate</code> is set, defaults to True</li>
</ul>

<h6 id="returns">Returns</h6>

<blockquote>
  <p>an integer value representing the offset for a given position in the cDNA sequence.</p>
</blockquote>
</div>


                            </div>
                            <div class="inherited">
                                <h5>Inherited Members</h5>
                                <dl>
                                    <div><dt><a href="#Exon">Exon</a></dt>
                                <dd id="CDNA_Match.transcript" class="variable"><a href="#Exon.transcript">transcript</a></dd>
                <dd id="CDNA_Match.tx_position" class="variable"><a href="#Exon.tx_position">tx_position</a></dd>
                <dd id="CDNA_Match.number" class="variable"><a href="#Exon.number">number</a></dd>
                <dd id="CDNA_Match.name" class="variable"><a href="#Exon.name">name</a></dd>
                <dd id="CDNA_Match.get_as_interval" class="function"><a href="#Exon.get_as_interval">get_as_interval</a></dd>
                <dd id="CDNA_Match.strand" class="variable"><a href="#Exon.strand">strand</a></dd>

            </div>
                                </dl>
                            </div>
                </section>
    </main>
<script>
    function escapeHTML(html) {
        return document.createElement('div').appendChild(document.createTextNode(html)).parentNode.innerHTML;
    }

    const originalContent = document.querySelector("main.pdoc");
    let currentContent = originalContent;

    function setContent(innerHTML) {
        let elem;
        if (innerHTML) {
            elem = document.createElement("main");
            elem.classList.add("pdoc");
            elem.innerHTML = innerHTML;
        } else {
            elem = originalContent;
        }
        if (currentContent !== elem) {
            currentContent.replaceWith(elem);
            currentContent = elem;
        }
    }

    function getSearchTerm() {
        return (new URL(window.location)).searchParams.get("search");
    }

    const searchBox = document.querySelector(".pdoc input[type=search]");
    searchBox.addEventListener("input", function () {
        let url = new URL(window.location);
        if (searchBox.value.trim()) {
            url.hash = "";
            url.searchParams.set("search", searchBox.value);
        } else {
            url.searchParams.delete("search");
        }
        history.replaceState("", "", url.toString());
        onInput();
    });
    window.addEventListener("popstate", onInput);


    let search, searchErr;

    async function initialize() {
        try {
            search = await new Promise((resolve, reject) => {
                const script = document.createElement("script");
                script.type = "text/javascript";
                script.async = true;
                script.onload = () => resolve(window.pdocSearch);
                script.onerror = (e) => reject(e);
                script.src = "../../search.js";
                document.getElementsByTagName("head")[0].appendChild(script);
            });
        } catch (e) {
            console.error("Cannot fetch pdoc search index");
            searchErr = "Cannot fetch search index.";
        }
        onInput();

        document.querySelector("nav.pdoc").addEventListener("click", e => {
            if (e.target.hash) {
                searchBox.value = "";
                searchBox.dispatchEvent(new Event("input"));
            }
        });
    }

    function onInput() {
        setContent((() => {
            const term = getSearchTerm();
            if (!term) {
                return null
            }
            if (searchErr) {
                return `<h3>Error: ${searchErr}</h3>`
            }
            if (!search) {
                return "<h3>Searching...</h3>"
            }

            window.scrollTo({top: 0, left: 0, behavior: 'auto'});

            const results = search(term);

            let html;
            if (results.length === 0) {
                html = `No search results for '${escapeHTML(term)}'.`
            } else {
                html = `<h4>${results.length} search result${results.length > 1 ? "s" : ""} for '${escapeHTML(term)}'.</h4>`;
            }
            for (let result of results.slice(0, 10)) {
                let doc = result.doc;
                let url = `../../${doc.modulename.replaceAll(".", "/")}.html`;
                if (doc.qualname) {
                    url += `#${doc.qualname}`;
                }

                let heading;
                switch (result.doc.kind) {
                    case "function":
                        if (doc.fullname.endsWith(".__init__")) {
                            heading = `<span class="name">${doc.fullname.replace(/\.__init__$/, "")}</span>${doc.signature}`;
                        } else {
                            heading = `<span class="def">${doc.funcdef}</span> <span class="name">${doc.fullname}</span>${doc.signature}`;
                        }
                        break;
                    case "class":
                        heading = `<span class="def">class</span> <span class="name">${doc.fullname}</span>`;
                        if (doc.bases)
                            heading += `<wbr>(<span class="base">${doc.bases}</span>)`;
                        heading += `:`;
                        break;
                    case "variable":
                        heading = `<span class="name">${doc.fullname}</span>`;
                        if (doc.annotation)
                            heading += `<span class="annotation">${doc.annotation}</span>`;
                        if (doc.default_value)
                            heading += `<span class="default_value"> = ${doc.default_value}</span>`;
                        break;
                    default:
                        heading = `<span class="name">${doc.fullname}</span>`;
                        break;
                }
                html += `
                        <section class="search-result">
                        <a href="${url}" class="attr ${doc.kind}">${heading}</a>
                        <div class="docstring">${doc.doc}</div>
                        </section>
                    `;

            }
            return html;
        })());
    }

    if (getSearchTerm()) {
        initialize();
        searchBox.value = getSearchTerm();
        onInput();
    } else {
        searchBox.addEventListener("focus", initialize, {once: true});
    }

    searchBox.addEventListener("keydown", e => {
        if (["ArrowDown", "ArrowUp", "Enter"].includes(e.key)) {
            let focused = currentContent.querySelector(".search-result.focused");
            if (!focused) {
                currentContent.querySelector(".search-result").classList.add("focused");
            } else if (
                e.key === "ArrowDown"
                && focused.nextElementSibling
                && focused.nextElementSibling.classList.contains("search-result")
            ) {
                focused.classList.remove("focused");
                focused.nextElementSibling.classList.add("focused");
                focused.nextElementSibling.scrollIntoView({
                    behavior: "smooth",
                    block: "nearest",
                    inline: "nearest"
                });
            } else if (
                e.key === "ArrowUp"
                && focused.previousElementSibling
                && focused.previousElementSibling.classList.contains("search-result")
            ) {
                focused.classList.remove("focused");
                focused.previousElementSibling.classList.add("focused");
                focused.previousElementSibling.scrollIntoView({
                    behavior: "smooth",
                    block: "nearest",
                    inline: "nearest"
                });
            } else if (
                e.key === "Enter"
            ) {
                focused.querySelector("a").click();
            }
        }
    });
</script></body>
</html>