<!doctype html>
<html lang="en">
<head>
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, minimum-scale=1" />
<meta name="generator" content="pdoc 0.10.0" />
<title>howard.objects.variant API documentation</title>
<meta name="description" content="Methods for manipulating genetic variants." />
<link rel="preload stylesheet" as="style" href="https://cdnjs.cloudflare.com/ajax/libs/10up-sanitize.css/11.0.1/sanitize.min.css" integrity="sha256-PK9q560IAAa6WVRRh76LtCaI8pjTJ2z11v0miyNNjrs=" crossorigin>
<link rel="preload stylesheet" as="style" href="https://cdnjs.cloudflare.com/ajax/libs/10up-sanitize.css/11.0.1/typography.min.css" integrity="sha256-7l/o7C8jubJiy74VsKTidCy1yBkRtiUGbVkYBylBqUg=" crossorigin>
<link rel="stylesheet preload" as="style" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.1.1/styles/github.min.css" crossorigin>
<style>:root{--highlight-color:#fe9}.flex{display:flex !important}body{line-height:1.5em}#content{padding:20px}#sidebar{padding:30px;overflow:hidden}#sidebar > *:last-child{margin-bottom:2cm}.http-server-breadcrumbs{font-size:130%;margin:0 0 15px 0}#footer{font-size:.75em;padding:5px 30px;border-top:1px solid #ddd;text-align:right}#footer p{margin:0 0 0 1em;display:inline-block}#footer p:last-child{margin-right:30px}h1,h2,h3,h4,h5{font-weight:300}h1{font-size:2.5em;line-height:1.1em}h2{font-size:1.75em;margin:1em 0 .50em 0}h3{font-size:1.4em;margin:25px 0 10px 0}h4{margin:0;font-size:105%}h1:target,h2:target,h3:target,h4:target,h5:target,h6:target{background:var(--highlight-color);padding:.2em 0}a{color:#058;text-decoration:none;transition:color .3s ease-in-out}a:hover{color:#e82}.title code{font-weight:bold}h2[id^="header-"]{margin-top:2em}.ident{color:#900}pre code{background:#f8f8f8;font-size:.8em;line-height:1.4em}code{background:#f2f2f1;padding:1px 4px;overflow-wrap:break-word}h1 code{background:transparent}pre{background:#f8f8f8;border:0;border-top:1px solid #ccc;border-bottom:1px solid #ccc;margin:1em 0;padding:1ex}#http-server-module-list{display:flex;flex-flow:column}#http-server-module-list div{display:flex}#http-server-module-list dt{min-width:10%}#http-server-module-list p{margin-top:0}.toc ul,#index{list-style-type:none;margin:0;padding:0}#index code{background:transparent}#index h3{border-bottom:1px solid #ddd}#index ul{padding:0}#index h4{margin-top:.6em;font-weight:bold}@media (min-width:200ex){#index .two-column{column-count:2}}@media (min-width:300ex){#index .two-column{column-count:3}}dl{margin-bottom:2em}dl dl:last-child{margin-bottom:4em}dd{margin:0 0 1em 3em}#header-classes + dl > dd{margin-bottom:3em}dd dd{margin-left:2em}dd p{margin:10px 0}.name{background:#eee;font-weight:bold;font-size:.85em;padding:5px 10px;display:inline-block;min-width:40%}.name:hover{background:#e0e0e0}dt:target .name{background:var(--highlight-color)}.name > span:first-child{white-space:nowrap}.name.class > span:nth-child(2){margin-left:.4em}.inherited{color:#999;border-left:5px solid #eee;padding-left:1em}.inheritance em{font-style:normal;font-weight:bold}.desc h2{font-weight:400;font-size:1.25em}.desc h3{font-size:1em}.desc dt code{background:inherit}.source summary,.git-link-div{color:#666;text-align:right;font-weight:400;font-size:.8em;text-transform:uppercase}.source summary > *{white-space:nowrap;cursor:pointer}.git-link{color:inherit;margin-left:1em}.source pre{max-height:500px;overflow:auto;margin:0}.source pre code{font-size:12px;overflow:visible}.hlist{list-style:none}.hlist li{display:inline}.hlist li:after{content:',\2002'}.hlist li:last-child:after{content:none}.hlist .hlist{display:inline;padding-left:1em}img{max-width:100%}td{padding:0 .5em}.admonition{padding:.1em .5em;margin-bottom:1em}.admonition-title{font-weight:bold}.admonition.note,.admonition.info,.admonition.important{background:#aef}.admonition.todo,.admonition.versionadded,.admonition.tip,.admonition.hint{background:#dfd}.admonition.warning,.admonition.versionchanged,.admonition.deprecated{background:#fd4}.admonition.error,.admonition.danger,.admonition.caution{background:lightpink}</style>
<style media="screen and (min-width: 700px)">@media screen and (min-width:700px){#sidebar{width:30%;height:100vh;overflow:auto;position:sticky;top:0}#content{width:70%;max-width:100ch;padding:3em 4em;border-left:1px solid #ddd}pre code{font-size:1em}.item .name{font-size:1em}main{display:flex;flex-direction:row-reverse;justify-content:flex-end}.toc ul ul,#index ul{padding-left:1.5em}.toc > ul > li{margin-top:.5em}}</style>
<style media="print">@media print{#sidebar h1{page-break-before:always}.source{display:none}}@media print{*{background:transparent !important;color:#000 !important;box-shadow:none !important;text-shadow:none !important}a[href]:after{content:" (" attr(href) ")";font-size:90%}a[href][title]:after{content:none}abbr[title]:after{content:" (" attr(title) ")"}.ir a:after,a[href^="javascript:"]:after,a[href^="#"]:after{content:""}pre,blockquote{border:1px solid #999;page-break-inside:avoid}thead{display:table-header-group}tr,img{page-break-inside:avoid}img{max-width:100% !important}@page{margin:0.5cm}p,h2,h3{orphans:3;widows:3}h1,h2,h3,h4,h5,h6{page-break-after:avoid}}</style>
<script defer src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.1.1/highlight.min.js" integrity="sha256-Uv3H6lx7dJmRfRvH8TH6kJD1TSK1aFcwgx+mdg3epi8=" crossorigin></script>
<script>window.addEventListener('DOMContentLoaded', () => hljs.initHighlighting())</script>
</head>
<body>
<main>
<article id="content">
<header>
<h1 class="title">Module <code>howard.objects.variant</code></h1>
</header>
<section id="section-intro">
<p>Methods for manipulating genetic variants.</p>
</section>
<section>
</section>
<section>
</section>
<section>
<h2 class="section-title" id="header-functions">Functions</h2>
<dl>
<dt id="howard.objects.variant.get_sequence"><code class="name flex">
<span>def <span class="ident">get_sequence</span></span>(<span>genome, chrom, start, end, is_forward_strand=True)</span>
</code></dt>
<dd>
<div class="desc"><p>Return a sequence for the genomic region.</p>
<p>Coordinates are 0-based, end-exclusive.</p></div>
</dd>
<dt id="howard.objects.variant.get_sequence_from_position"><code class="name flex">
<span>def <span class="ident">get_sequence_from_position</span></span>(<span>genome, position)</span>
</code></dt>
<dd>
<div class="desc"><p>Return a sequence for the genomic region</p>
<p>Position is 0-based, end-exclusive.</p></div>
</dd>
<dt id="howard.objects.variant.justify_genomic_indel"><code class="name flex">
<span>def <span class="ident">justify_genomic_indel</span></span>(<span>genome, chrom, start, end, indel, justify, flank_length=20)</span>
</code></dt>
<dd>
<div class="desc"><p>start, end: 0-based, end-exclusive coordinates of 'indel'.</p></div>
</dd>
<dt id="howard.objects.variant.justify_indel"><code class="name flex">
<span>def <span class="ident">justify_indel</span></span>(<span>start, end, indel, seq, justify)</span>
</code></dt>
<dd>
<div class="desc"><p>Justify an indel to the left or right along a sequence 'seq'.</p>
<p>start, end: 0-based, end-exclusive coordinates of 'indel' within the
sequence 'seq'. Inserts denote the insertion point using start=end
and deletions indicate the deleted region with (start,end).
indel: indel sequence, can be insertion or deletion.
seq: a larger sequence containing the indel. Can be a fragment from the
genome.
justify: Which direction to justify the indel ('left', 'right').</p></div>
</dd>
<dt id="howard.objects.variant.normalize_variant"><code class="name flex">
<span>def <span class="ident">normalize_variant</span></span>(<span>chrom, offset, ref_sequence, alt_sequences, genome, justify='left', flank_length=30, indels_start_with_same_base=True)</span>
</code></dt>
<dd>
<div class="desc"><p>Normalize variant according to the GATK/VCF standard.</p>
<p>chrom: chromsome containing variant.
offset: 1-based coordinate of reference allele in the genome.
ref_sequence: reference allele.
alt_sequences: list of all alternate sequences.
genome: pygr-compatiable genome object.</p></div>
</dd>
<dt id="howard.objects.variant.revcomp"><code class="name flex">
<span>def <span class="ident">revcomp</span></span>(<span>seq)</span>
</code></dt>
<dd>
<div class="desc"><p>Reverse complement.</p></div>
</dd>
</dl>
</section>
<section>
<h2 class="section-title" id="header-classes">Classes</h2>
<dl>
<dt id="howard.objects.variant.NormalizedVariant"><code class="flex name class">
<span>class <span class="ident">NormalizedVariant</span></span>
<span>(</span><span>position, ref_allele, alt_alleles, seq_5p='', seq_3p='', genome=None, justify='left', indels_start_with_same_base=True)</span>
</code></dt>
<dd>
<div class="desc"><p>Normalizes variant representation to match GATK/VCF.</p>
<p>position: a 0-index genomic Position.
ref_allele: the reference allele sequence.
alt_alleles: a list of alternate allele sequences.
seq_5p: 5 prime flanking sequence of variant.
seq_3p: 3 prime flanking sequence of variant.
genome: a pygr compatible genome object (optional).</p>
<p>indels_start_with_same_base: DML - I have no idea why this is required
but am keeping for backwards compat</p></div>
<h3>Instance variables</h3>
<dl>
<dt id="howard.objects.variant.NormalizedVariant.alt_alleles"><code class="name">var <span class="ident">alt_alleles</span></code></dt>
<dd>
<div class="desc"></div>
</dd>
<dt id="howard.objects.variant.NormalizedVariant.molecular_class"><code class="name">var <span class="ident">molecular_class</span></code></dt>
<dd>
<div class="desc"></div>
</dd>
<dt id="howard.objects.variant.NormalizedVariant.ref_allele"><code class="name">var <span class="ident">ref_allele</span></code></dt>
<dd>
<div class="desc"></div>
</dd>
<dt id="howard.objects.variant.NormalizedVariant.variant"><code class="name">var <span class="ident">variant</span></code></dt>
<dd>
<div class="desc"></div>
</dd>
</dl>
</dd>
<dt id="howard.objects.variant.Position"><code class="flex name class">
<span>class <span class="ident">Position</span></span>
<span>(</span><span>chrom, chrom_start, chrom_stop, is_forward_strand)</span>
</code></dt>
<dd>
<div class="desc"><p>A position in the genome.</p></div>
</dd>
</dl>
</section>
</article>
<nav id="sidebar">
<h1>Index</h1>
<div class="toc">
<ul></ul>
</div>
<ul id="index">
<li><h3><a href="#header-functions">Functions</a></h3>
<ul class="">
<li><code><a title="howard.objects.variant.get_sequence" href="#howard.objects.variant.get_sequence">get_sequence</a></code></li>
<li><code><a title="howard.objects.variant.get_sequence_from_position" href="#howard.objects.variant.get_sequence_from_position">get_sequence_from_position</a></code></li>
<li><code><a title="howard.objects.variant.justify_genomic_indel" href="#howard.objects.variant.justify_genomic_indel">justify_genomic_indel</a></code></li>
<li><code><a title="howard.objects.variant.justify_indel" href="#howard.objects.variant.justify_indel">justify_indel</a></code></li>
<li><code><a title="howard.objects.variant.normalize_variant" href="#howard.objects.variant.normalize_variant">normalize_variant</a></code></li>
<li><code><a title="howard.objects.variant.revcomp" href="#howard.objects.variant.revcomp">revcomp</a></code></li>
</ul>
</li>
<li><h3><a href="#header-classes">Classes</a></h3>
<ul>
<li>
<h4><code><a title="howard.objects.variant.NormalizedVariant" href="#howard.objects.variant.NormalizedVariant">NormalizedVariant</a></code></h4>
<ul class="">
<li><code><a title="howard.objects.variant.NormalizedVariant.alt_alleles" href="#howard.objects.variant.NormalizedVariant.alt_alleles">alt_alleles</a></code></li>
<li><code><a title="howard.objects.variant.NormalizedVariant.molecular_class" href="#howard.objects.variant.NormalizedVariant.molecular_class">molecular_class</a></code></li>
<li><code><a title="howard.objects.variant.NormalizedVariant.ref_allele" href="#howard.objects.variant.NormalizedVariant.ref_allele">ref_allele</a></code></li>
<li><code><a title="howard.objects.variant.NormalizedVariant.variant" href="#howard.objects.variant.NormalizedVariant.variant">variant</a></code></li>
</ul>
</li>
<li>
<h4><code><a title="howard.objects.variant.Position" href="#howard.objects.variant.Position">Position</a></code></h4>
</li>
</ul>
</li>
</ul>
</nav>
</main>
<footer id="footer">
<p>Generated by <a href="https://pdoc3.github.io/pdoc" title="pdoc: Python API documentation generator"><cite>pdoc</cite> 0.10.0</a>.</p>
</footer>
</body>
</html>