<!DOCTYPE html>
<html>
<head>
<title>help.param.databases.md</title>
<meta http-equiv="Content-type" content="text/html;charset=UTF-8">

<style>
/* https://github.com/microsoft/vscode/blob/master/extensions/markdown-language-features/media/markdown.css */
/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Microsoft Corporation. All rights reserved.
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/

body {
	font-family: var(--vscode-markdown-font-family, -apple-system, BlinkMacSystemFont, "Segoe WPC", "Segoe UI", "Ubuntu", "Droid Sans", sans-serif);
	font-size: var(--vscode-markdown-font-size, 14px);
	padding: 0 26px;
	line-height: var(--vscode-markdown-line-height, 22px);
	word-wrap: break-word;
}

#code-csp-warning {
	position: fixed;
	top: 0;
	right: 0;
	color: white;
	margin: 16px;
	text-align: center;
	font-size: 12px;
	font-family: sans-serif;
	background-color:#444444;
	cursor: pointer;
	padding: 6px;
	box-shadow: 1px 1px 1px rgba(0,0,0,.25);
}

#code-csp-warning:hover {
	text-decoration: none;
	background-color:#007acc;
	box-shadow: 2px 2px 2px rgba(0,0,0,.25);
}

body.scrollBeyondLastLine {
	margin-bottom: calc(100vh - 22px);
}

body.showEditorSelection .code-line {
	position: relative;
}

body.showEditorSelection .code-active-line:before,
body.showEditorSelection .code-line:hover:before {
	content: "";
	display: block;
	position: absolute;
	top: 0;
	left: -12px;
	height: 100%;
}

body.showEditorSelection li.code-active-line:before,
body.showEditorSelection li.code-line:hover:before {
	left: -30px;
}

.vscode-light.showEditorSelection .code-active-line:before {
	border-left: 3px solid rgba(0, 0, 0, 0.15);
}

.vscode-light.showEditorSelection .code-line:hover:before {
	border-left: 3px solid rgba(0, 0, 0, 0.40);
}

.vscode-light.showEditorSelection .code-line .code-line:hover:before {
	border-left: none;
}

.vscode-dark.showEditorSelection .code-active-line:before {
	border-left: 3px solid rgba(255, 255, 255, 0.4);
}

.vscode-dark.showEditorSelection .code-line:hover:before {
	border-left: 3px solid rgba(255, 255, 255, 0.60);
}

.vscode-dark.showEditorSelection .code-line .code-line:hover:before {
	border-left: none;
}

.vscode-high-contrast.showEditorSelection .code-active-line:before {
	border-left: 3px solid rgba(255, 160, 0, 0.7);
}

.vscode-high-contrast.showEditorSelection .code-line:hover:before {
	border-left: 3px solid rgba(255, 160, 0, 1);
}

.vscode-high-contrast.showEditorSelection .code-line .code-line:hover:before {
	border-left: none;
}

img {
	max-width: 100%;
	max-height: 100%;
}

a {
	text-decoration: none;
}

a:hover {
	text-decoration: underline;
}

a:focus,
input:focus,
select:focus,
textarea:focus {
	outline: 1px solid -webkit-focus-ring-color;
	outline-offset: -1px;
}

hr {
	border: 0;
	height: 2px;
	border-bottom: 2px solid;
}

h1 {
	padding-bottom: 0.3em;
	line-height: 1.2;
	border-bottom-width: 1px;
	border-bottom-style: solid;
}

h1, h2, h3 {
	font-weight: normal;
}

table {
	border-collapse: collapse;
}

table > thead > tr > th {
	text-align: left;
	border-bottom: 1px solid;
}

table > thead > tr > th,
table > thead > tr > td,
table > tbody > tr > th,
table > tbody > tr > td {
	padding: 5px 10px;
}

table > tbody > tr + tr > td {
	border-top: 1px solid;
}

blockquote {
	margin: 0 7px 0 5px;
	padding: 0 16px 0 10px;
	border-left-width: 5px;
	border-left-style: solid;
}

code {
	font-family: Menlo, Monaco, Consolas, "Droid Sans Mono", "Courier New", monospace, "Droid Sans Fallback";
	font-size: 1em;
	line-height: 1.357em;
}

body.wordWrap pre {
	white-space: pre-wrap;
}

pre:not(.hljs),
pre.hljs code > div {
	padding: 16px;
	border-radius: 3px;
	overflow: auto;
}

pre code {
	color: var(--vscode-editor-foreground);
	tab-size: 4;
}

/** Theming */

.vscode-light pre {
	background-color: rgba(220, 220, 220, 0.4);
}

.vscode-dark pre {
	background-color: rgba(10, 10, 10, 0.4);
}

.vscode-high-contrast pre {
	background-color: rgb(0, 0, 0);
}

.vscode-high-contrast h1 {
	border-color: rgb(0, 0, 0);
}

.vscode-light table > thead > tr > th {
	border-color: rgba(0, 0, 0, 0.69);
}

.vscode-dark table > thead > tr > th {
	border-color: rgba(255, 255, 255, 0.69);
}

.vscode-light h1,
.vscode-light hr,
.vscode-light table > tbody > tr + tr > td {
	border-color: rgba(0, 0, 0, 0.18);
}

.vscode-dark h1,
.vscode-dark hr,
.vscode-dark table > tbody > tr + tr > td {
	border-color: rgba(255, 255, 255, 0.18);
}

</style>

<style>
/* Tomorrow Theme */
/* http://jmblog.github.com/color-themes-for-google-code-highlightjs */
/* Original theme - https://github.com/chriskempson/tomorrow-theme */

/* Tomorrow Comment */
.hljs-comment,
.hljs-quote {
	color: #8e908c;
}

/* Tomorrow Red */
.hljs-variable,
.hljs-template-variable,
.hljs-tag,
.hljs-name,
.hljs-selector-id,
.hljs-selector-class,
.hljs-regexp,
.hljs-deletion {
	color: #c82829;
}

/* Tomorrow Orange */
.hljs-number,
.hljs-built_in,
.hljs-builtin-name,
.hljs-literal,
.hljs-type,
.hljs-params,
.hljs-meta,
.hljs-link {
	color: #f5871f;
}

/* Tomorrow Yellow */
.hljs-attribute {
	color: #eab700;
}

/* Tomorrow Green */
.hljs-string,
.hljs-symbol,
.hljs-bullet,
.hljs-addition {
	color: #718c00;
}

/* Tomorrow Blue */
.hljs-title,
.hljs-section {
	color: #4271ae;
}

/* Tomorrow Purple */
.hljs-keyword,
.hljs-selector-tag {
	color: #8959a8;
}

.hljs {
	display: block;
	overflow-x: auto;
	color: #4d4d4c;
	padding: 0.5em;
}

.hljs-emphasis {
	font-style: italic;
}

.hljs-strong {
	font-weight: bold;
}
</style>

<style>
/*
 * Markdown PDF CSS
 */

 body {
	font-family: -apple-system, BlinkMacSystemFont, "Segoe WPC", "Segoe UI", "Ubuntu", "Droid Sans", sans-serif, "Meiryo";
	padding: 0 12px;
}

pre {
	background-color: #f8f8f8;
	border: 1px solid #cccccc;
	border-radius: 3px;
	overflow-x: auto;
	white-space: pre-wrap;
	overflow-wrap: break-word;
}

pre:not(.hljs) {
	padding: 23px;
	line-height: 19px;
}

blockquote {
	background: rgba(127, 127, 127, 0.1);
	border-color: rgba(0, 122, 204, 0.5);
}

.emoji {
	height: 1.4em;
}

code {
	font-size: 14px;
	line-height: 19px;
}

/* for inline code */
:not(pre):not(.hljs) > code {
	color: #C9AE75; /* Change the old color so it seems less like an error */
	font-size: inherit;
}

/* Page Break : use <div class="page"/> to insert page break
-------------------------------------------------------- */
.page {
	page-break-after: always;
}

</style>

<script src="https://unpkg.com/mermaid/dist/mermaid.min.js"></script>
</head>
<body>
  <script>
    mermaid.initialize({
      startOnLoad: true,
      theme: document.body.classList.contains('vscode-dark') || document.body.classList.contains('vscode-high-contrast')
          ? 'dark'
          : 'default'
    });
  </script>
<h1 id="howard-parameters-databases">HOWARD Parameters Databases</h1>
<p>HOWARD Parameters JSON file defines parameters for databases' downloads.</p>
<h2 id="table-of-contents">Table of contents</h2>
<ul>
<li><a href="#howard-parameters-databases">HOWARD Parameters Databases</a>
<ul>
<li><a href="#databases">databases</a>
<ul>
<li><a href="#databasesassembly">assembly</a></li>
<li><a href="#databasesgenomes_folder">genomes_folder</a></li>
<li><a href="#databasesgenome">genome</a></li>
<li><a href="#databasesgenomes">genomes</a>
<ul>
<li><a href="#databasesgenomesdownload_genomes">download_genomes</a></li>
<li><a href="#databasesgenomesdownload_genomes_provider">download_genomes_provider</a></li>
<li><a href="#databasesgenomesdownload_genomes_contig_regex">download_genomes_contig_regex</a></li>
</ul>
</li>
<li><a href="#databasessnpeff">snpeff</a>
<ul>
<li><a href="#databasessnpeffdownload_snpeff">download_snpeff</a></li>
</ul>
</li>
<li><a href="#databasesannovar">annovar</a>
<ul>
<li><a href="#databasesannovardownload_annovar">download_annovar</a></li>
<li><a href="#databasesannovardownload_annovar_files">download_annovar_files</a></li>
<li><a href="#databasesannovardownload_annovar_url">download_annovar_url</a></li>
</ul>
</li>
<li><a href="#databasesrefseq">refseq</a>
<ul>
<li><a href="#databasesrefseqdownload_refseq">download_refseq</a></li>
<li><a href="#databasesrefseqdownload_refseq_url">download_refseq_url</a></li>
<li><a href="#databasesrefseqdownload_refseq_prefix">download_refseq_prefix</a></li>
<li><a href="#databasesrefseqdownload_refseq_files">download_refseq_files</a></li>
<li><a href="#databasesrefseqdownload_refseq_format_file">download_refseq_format_file</a></li>
<li><a href="#databasesrefseqdownload_refseq_include_utr5">download_refseq_include_utr5</a></li>
<li><a href="#databasesrefseqdownload_refseq_include_utr3">download_refseq_include_utr3</a></li>
<li><a href="#databasesrefseqdownload_refseq_include_chrm">download_refseq_include_chrM</a></li>
<li><a href="#databasesrefseqdownload_refseq_include_non_canonical_chr">download_refseq_include_non_canonical_chr</a></li>
<li><a href="#databasesrefseqdownload_refseq_include_non_coding_transcripts">download_refseq_include_non_coding_transcripts</a></li>
<li><a href="#databasesrefseqdownload_refseq_include_transcript_version">download_refseq_include_transcript_version</a></li>
</ul>
</li>
<li><a href="#databasesdbnsfp">dbnsfp</a>
<ul>
<li><a href="#databasesdbnsfpdownload_dbnsfp">download_dbnsfp</a></li>
<li><a href="#databasesdbnsfpdownload_dbnsfp_url">download_dbnsfp_url</a></li>
<li><a href="#databasesdbnsfpdownload_dbnsfp_release">download_dbnsfp_release</a></li>
<li><a href="#databasesdbnsfpdownload_dbnsfp_parquet_size">download_dbnsfp_parquet_size</a></li>
<li><a href="#databasesdbnsfpdownload_dbnsfp_subdatabases">download_dbnsfp_subdatabases</a></li>
<li><a href="#databasesdbnsfpdownload_dbnsfp_parquet">download_dbnsfp_parquet</a></li>
<li><a href="#databasesdbnsfpdownload_dbnsfp_vcf">download_dbnsfp_vcf</a></li>
<li><a href="#databasesdbnsfpdownload_dbnsfp_no_files_all">download_dbnsfp_no_files_all</a></li>
<li><a href="#databasesdbnsfpdownload_dbnsfp_add_info">download_dbnsfp_add_info</a></li>
<li><a href="#databasesdbnsfpdownload_dbnsfp_row_group_size">download_dbnsfp_row_group_size</a></li>
</ul>
</li>
<li><a href="#databasesalphamissense">alphamissense</a>
<ul>
<li><a href="#databasesalphamissensedownload_alphamissense">download_alphamissense</a></li>
<li><a href="#databasesalphamissensedownload_alphamissense_url">download_alphamissense_url</a></li>
</ul>
</li>
<li><a href="#databasesexomiser">exomiser</a>
<ul>
<li><a href="#databasesexomiserdownload_exomiser">download_exomiser</a></li>
<li><a href="#databasesexomiserdownload_exomiser_application_properties">download_exomiser_application_properties</a></li>
<li><a href="#databasesexomiserdownload_exomiser_url">download_exomiser_url</a></li>
<li><a href="#databasesexomiserdownload_exomiser_release">download_exomiser_release</a></li>
<li><a href="#databasesexomiserdownload_exomiser_phenotype_release">download_exomiser_phenotype_release</a></li>
<li><a href="#databasesexomiserdownload_exomiser_remm_release">download_exomiser_remm_release</a></li>
<li><a href="#databasesexomiserdownload_exomiser_remm_url">download_exomiser_remm_url</a></li>
<li><a href="#databasesexomiserdownload_exomiser_cadd_release">download_exomiser_cadd_release</a></li>
<li><a href="#databasesexomiserdownload_exomiser_cadd_url">download_exomiser_cadd_url</a></li>
<li><a href="#databasesexomiserdownload_exomiser_cadd_url_snv_file">download_exomiser_cadd_url_snv_file</a></li>
<li><a href="#databasesexomiserdownload_exomiser_cadd_url_indel_file">download_exomiser_cadd_url_indel_file</a></li>
</ul>
</li>
<li><a href="#databasesdbsnp">dbsnp</a>
<ul>
<li><a href="#databasesdbsnpdownload_dbsnp">download_dbsnp</a></li>
<li><a href="#databasesdbsnpdownload_dbsnp_releases">download_dbsnp_releases</a></li>
<li><a href="#databasesdbsnpdownload_dbsnp_release_default">download_dbsnp_release_default</a></li>
<li><a href="#databasesdbsnpdownload_dbsnp_url">download_dbsnp_url</a></li>
<li><a href="#databasesdbsnpdownload_dbsnp_url_files">download_dbsnp_url_files</a></li>
<li><a href="#databasesdbsnpdownload_dbsnp_url_files_prefix">download_dbsnp_url_files_prefix</a></li>
<li><a href="#databasesdbsnpdownload_dbsnp_assemblies_map">download_dbsnp_assemblies_map</a></li>
<li><a href="#databasesdbsnpdownload_dbsnp_vcf">download_dbsnp_vcf</a></li>
<li><a href="#databasesdbsnpdownload_dbsnp_parquet">download_dbsnp_parquet</a></li>
</ul>
</li>
<li><a href="#databaseshgmd">hgmd</a>
<ul>
<li><a href="#databaseshgmdconvert_hgmd">convert_hgmd</a></li>
<li><a href="#databaseshgmdconvert_hgmd_file">convert_hgmd_file</a></li>
<li><a href="#databaseshgmdconvert_hgmd_basename">convert_hgmd_basename</a></li>
</ul>
</li>
<li><a href="#databasesfrom_annovar">from_Annovar</a>
<ul>
<li><a href="#databasesfrom_annovarinput_annovar">input_annovar</a></li>
<li><a href="#databasesfrom_annovaroutput_annovar">output_annovar</a></li>
<li><a href="#databasesfrom_annovarannovar_code">annovar_code</a></li>
<li><a href="#databasesfrom_annovarannovar_to_parquet">annovar_to_parquet</a></li>
<li><a href="#databasesfrom_annovarannovar_reduce_memory">annovar_reduce_memory</a></li>
<li><a href="#databasesfrom_annovarannovar_multi_variant">annovar_multi_variant</a></li>
</ul>
</li>
<li><a href="#databasesfrom_extann">from_extann</a>
<ul>
<li><a href="#databasesfrom_extanninput_extann">input_extann</a></li>
<li><a href="#databasesfrom_extannoutput_extann">output_extann</a></li>
<li><a href="#databasesfrom_extannrefgene">refgene</a></li>
<li><a href="#databasesfrom_extanntranscripts">transcripts</a></li>
<li><a href="#databasesfrom_extannparam_extann">param_extann</a></li>
<li><a href="#databasesfrom_extannmode_extann">mode_extann</a></li>
</ul>
</li>
<li><a href="#databasesparameters">Parameters</a>
<ul>
<li><a href="#databasesparametersgenerate_param">generate_param</a></li>
<li><a href="#databasesparametersgenerate_param_description">generate_param_description</a></li>
<li><a href="#databasesparametersgenerate_param_releases">generate_param_releases</a></li>
<li><a href="#databasesparametersgenerate_param_formats">generate_param_formats</a></li>
<li><a href="#databasesparametersgenerate_param_bcftools">generate_param_bcftools</a></li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
<p>Examples:</p>
<blockquote>
<p>Example of simple databases parameters JSON file for downloads</p>
</blockquote>
<pre class="hljs"><code><div><span class="hljs-string">"databases"</span>: {
      <span class="hljs-attr">"genomes"</span>: {
          <span class="hljs-attr">"download_genomes"</span>: <span class="hljs-string">"~/howard/databases/genomes/current"</span>,
          <span class="hljs-attr">"download_genomes_contig_regex"</span>: <span class="hljs-string">"chr[0-9XYM]+$"</span>
      },
      <span class="hljs-attr">"snpeff"</span>: {
          <span class="hljs-attr">"download_snpeff"</span>: <span class="hljs-string">"~/howard/databases/snpeff/current"</span>
      },
      <span class="hljs-attr">"annovar"</span>: {
          <span class="hljs-attr">"download_annovar"</span>: <span class="hljs-string">"~/howard/databases/annovar/current"</span>,
          <span class="hljs-attr">"download_annovar_files"</span>: <span class="hljs-string">"refGene,cosmic70,nci60"</span>
      },
      <span class="hljs-attr">"refseq"</span>: {
          <span class="hljs-attr">"download_refseq"</span>: <span class="hljs-string">"~/howard/databases/refseq/current"</span>
      },
      <span class="hljs-attr">"dbnsfp"</span>: {
          <span class="hljs-attr">"download_dbnsfp"</span>: <span class="hljs-string">"~/howard/databases/dbnsfp/current"</span>,
          <span class="hljs-attr">"download_dbnsfp_release"</span>: <span class="hljs-string">"4.4a"</span>
      },
      <span class="hljs-attr">"alphamissense"</span>: {
          <span class="hljs-attr">"download_alphamissense"</span>: <span class="hljs-string">"~/howard/databases/alphamissense/current"</span>
      },
      <span class="hljs-attr">"exomiser"</span>: {
          <span class="hljs-attr">"download_exomiser"</span>: <span class="hljs-string">"~/howard/databases/exomiser/current"</span>
      },
      <span class="hljs-attr">"dbsnp"</span>: {
          <span class="hljs-attr">"download_dbsnp"</span>: <span class="hljs-string">"~/howard/databases/dbsnp/current"</span>,
          <span class="hljs-attr">"download_dbsnp_releases"</span>: <span class="hljs-string">"b156"</span>,
          <span class="hljs-attr">"download_dbsnp_vcf"</span>: <span class="hljs-literal">true</span>,
          <span class="hljs-attr">"download_dbsnp_parquet"</span>: <span class="hljs-literal">true</span>
      },
      <span class="hljs-attr">"assemblies"</span>: [
          <span class="hljs-string">"hg19"</span>
      ]
}
</div></code></pre>
<blockquote>
<p>Example of a full Databases parameters JSON file for downloads</p>
</blockquote>
<pre class="hljs"><code><div><span class="hljs-string">"databases"</span>: {
      <span class="hljs-attr">"genomes_folder"</span>: <span class="hljs-string">"~/howard/databases/genomes/current"</span>,
      <span class="hljs-attr">"genome"</span>: <span class="hljs-string">"~/howard/databases/genomes/current/hg19/hg19.fa"</span>,
      <span class="hljs-attr">"genomes"</span>: {
          <span class="hljs-attr">"download_genomes"</span>: <span class="hljs-string">"~/howard/databases/genomes/current"</span>,
          <span class="hljs-attr">"download_genomes_provider"</span>: <span class="hljs-string">"UCSC"</span>,
          <span class="hljs-attr">"download_genomes_contig_regex"</span>: <span class="hljs-string">"chr[0-9XYM]+$"</span>
      },
      <span class="hljs-attr">"snpeff"</span>: {
          <span class="hljs-attr">"download_snpeff"</span>: <span class="hljs-string">"~/howard/databases/snpeff/current"</span>
      },
      <span class="hljs-attr">"annovar"</span>: {
          <span class="hljs-attr">"download_annovar"</span>: <span class="hljs-string">"~/howard/databases/annovar/current"</span>,
          <span class="hljs-attr">"download_annovar_files"</span>: <span class="hljs-string">"refGene,cosmic70,nci60"</span>,
          <span class="hljs-attr">"download_annovar_url"</span>: <span class="hljs-string">"http://www.openbioinformatics.org/annovar/download"</span>
      },
      <span class="hljs-attr">"refseq"</span>: {
          <span class="hljs-attr">"download_refseq"</span>: <span class="hljs-string">"~/howard/databases/refseq/current"</span>,
          <span class="hljs-attr">"download_refseq_url"</span>: <span class="hljs-string">"http://hgdownload.soe.ucsc.edu/goldenPath"</span>,
          <span class="hljs-attr">"download_refseq_prefix"</span>: <span class="hljs-string">"ncbiRefSeq"</span>,
          <span class="hljs-attr">"download_refseq_files"</span>: <span class="hljs-string">"ncbiRefSeq.txt,ncbiRefSeqLink.txt"</span>,
          <span class="hljs-attr">"download_refseq_format_file"</span>: <span class="hljs-string">"ncbiRefSeq.txt"</span>,
          <span class="hljs-attr">"download_refseq_include_utr5"</span>: <span class="hljs-literal">false</span>,
          <span class="hljs-attr">"download_refseq_include_utr3"</span>: <span class="hljs-literal">false</span>,
          <span class="hljs-attr">"download_refseq_include_chrM"</span>: <span class="hljs-literal">false</span>,
          <span class="hljs-attr">"download_refseq_include_non_canonical_chr"</span>: <span class="hljs-literal">false</span>,
          <span class="hljs-attr">"download_refseq_include_non_coding_transcripts"</span>: <span class="hljs-literal">false</span>,
          <span class="hljs-attr">"download_refseq_include_transcript_version"</span>: <span class="hljs-literal">false</span>
      },
      <span class="hljs-attr">"dbnsfp"</span>: {
          <span class="hljs-attr">"download_dbnsfp"</span>: <span class="hljs-string">"~/howard/databases/dbnsfp/current"</span>,
          <span class="hljs-attr">"download_dbnsfp_url"</span>: <span class="hljs-string">"https://dbnsfp.s3.amazonaws.com"</span>,
          <span class="hljs-attr">"download_dbnsfp_release"</span>: <span class="hljs-string">"4.4a"</span>,
          <span class="hljs-attr">"download_dbnsfp_parquet_size"</span>: <span class="hljs-number">100</span>,
          <span class="hljs-attr">"download_dbnsfp_subdatabases"</span>: <span class="hljs-literal">true</span>,
          <span class="hljs-attr">"download_dbnsfp_parquet"</span>: <span class="hljs-literal">false</span>,
          <span class="hljs-attr">"download_dbnsfp_vcf"</span>: <span class="hljs-literal">false</span>,
          <span class="hljs-attr">"download_dbnsfp_no_files_all"</span>: <span class="hljs-literal">false</span>,
          <span class="hljs-attr">"download_dbnsfp_add_info"</span>: <span class="hljs-literal">false</span>,
          <span class="hljs-attr">"download_dbnsfp_row_group_size"</span>: <span class="hljs-number">100000</span>
      },
      <span class="hljs-attr">"alphamissense"</span>: {
          <span class="hljs-attr">"download_alphamissense"</span>: <span class="hljs-string">"~/howard/databases/alphamissense/current"</span>,
          <span class="hljs-attr">"download_alphamissense_url"</span>: <span class="hljs-string">"https://storage.googleapis.com/dm_alphamissense"</span>
      },
      <span class="hljs-attr">"exomiser"</span>: {
          <span class="hljs-attr">"download_exomiser"</span>: <span class="hljs-string">"~/howard/databases/exomiser/current"</span>,
          <span class="hljs-attr">"download_exomiser_url"</span>: <span class="hljs-string">"http://data.monarchinitiative.org/exomiser"</span>,
          <span class="hljs-attr">"download_exomiser_remm_url"</span>: <span class="hljs-string">"https://kircherlab.bihealth.org/download/ReMM"</span>,
          <span class="hljs-attr">"download_exomiser_cadd_url"</span>: <span class="hljs-string">"https://kircherlab.bihealth.org/download/CADD"</span>,
          <span class="hljs-attr">"download_exomiser_cadd_url_snv_file"</span>: <span class="hljs-string">"whole_genome_SNVs.tsv.gz"</span>,
          <span class="hljs-attr">"download_exomiser_cadd_url_indel_file"</span>: <span class="hljs-string">"InDels.tsv.gz"</span>
      },
      <span class="hljs-attr">"dbsnp"</span>: {
          <span class="hljs-attr">"download_dbsnp"</span>: <span class="hljs-string">"~/howard/databases/dbsnp/current"</span>,
          <span class="hljs-attr">"download_dbsnp_releases"</span>: <span class="hljs-string">"b156"</span>,
          <span class="hljs-attr">"download_dbsnp_url"</span>: <span class="hljs-string">"https://ftp.ncbi.nih.gov/snp/archive"</span>,
          <span class="hljs-attr">"download_dbsnp_url_files_prefix"</span>: <span class="hljs-string">"GCF_000001405"</span>,
          <span class="hljs-attr">"download_dbsnp_assemblies_map"</span>: {
              <span class="hljs-attr">"hg19"</span>: <span class="hljs-string">"25"</span>,
              <span class="hljs-attr">"hg38"</span>: <span class="hljs-string">"40"</span>
          },
          <span class="hljs-attr">"download_dbsnp_vcf"</span>: <span class="hljs-literal">true</span>,
          <span class="hljs-attr">"download_dbsnp_parquet"</span>: <span class="hljs-literal">false</span>
      },
      <span class="hljs-attr">"parameters"</span>: {
          <span class="hljs-attr">"generate_param_releases"</span>: <span class="hljs-string">"current"</span>,
          <span class="hljs-attr">"generate_param_formats"</span>: <span class="hljs-string">"parquet"</span>,
          <span class="hljs-attr">"generate_param_bcftools"</span>: <span class="hljs-literal">false</span>
      },
      <span class="hljs-attr">"assemblies"</span>: [
          <span class="hljs-string">"hg19"</span>
      ]
}
</div></code></pre>
<h2 id="databases">databases</h2>
<p>Databases download options</p>
<h3 id="databasesassembly">databases::assembly</h3>
<p>Genome Assembly (e.g. 'hg19', 'hg38').</p>
<p>Type: <code>str</code></p>
<p>Default: <code>hg19</code></p>
<p>Examples:</p>
<blockquote>
<p>Default assembly for all analysis tools</p>
</blockquote>
<pre class="hljs"><code><div><span class="hljs-string">"assembly"</span>: <span class="hljs-string">"hg19"</span> 
</div></code></pre>
<blockquote>
<p>List of assemblies for databases download tool</p>
</blockquote>
<pre class="hljs"><code><div><span class="hljs-string">"assembly"</span>: <span class="hljs-string">"hg19,hg38"</span> 
</div></code></pre>
<h3 id="databasesgenomesfolder">databases::genomes_folder</h3>
<p>Folder containing genomes. (e.g. '/Users/lebechea/howard/databases/genomes/current'</p>
<p>Type: <code>Path</code></p>
<p>Default: <code>/Users/lebechea/howard/databases/genomes/current</code></p>
<h3 id="databasesgenome">databases::genome</h3>
<p>Genome file in fasta format (e.g. 'hg19.fa', 'hg38.fa').</p>
<p>Type: <code>Path</code></p>
<p>Default: <code>~/howard/databases/genomes/current/hg19/hg19.fa</code></p>
<h3 id="databasesgenomes">databases::genomes</h3>
<p>Genomes download.</p>
<h4 id="databasesgenomesdownloadgenomes">databases::genomes::download_genomes</h4>
<p>Path to genomes folder with Fasta files, indexes, and all files generated by pygenome module. (e.g. '/Users/lebechea/howard/databases/genomes/current').</p>
<p>Type: <code>Path</code></p>
<p>Default: <code>None</code></p>
<h4 id="databasesgenomesdownloadgenomesprovider">databases::genomes::download_genomes_provider</h4>
<p>Download Genome from an external provider. Available: GENCODE, Ensembl, UCSC, NCBI.</p>
<p>Type: <code>str</code></p>
<p>Choices: <code>['GENCODE', 'Ensembl', 'UCSC', 'NCBI']</code></p>
<p>Default: <code>UCSC</code></p>
<h4 id="databasesgenomesdownloadgenomescontigregex">databases::genomes::download_genomes_contig_regex</h4>
<p>Regular expression to select specific chromosome (e.g 'chr[0-9XYM]+$').</p>
<p>Type: <code>str</code></p>
<p>Default: <code>None</code></p>
<h3 id="databasessnpeff">databases::snpeff</h3>
<p>snpEff download.</p>
<h4 id="databasessnpeffdownloadsnpeff">databases::snpeff::download_snpeff</h4>
<p>Download snpEff databases within snpEff folder</p>
<p>Type: <code>Path</code></p>
<p>Default: <code>None</code></p>
<h3 id="databasesannovar">databases::annovar</h3>
<p>Annovar download.</p>
<h4 id="databasesannovardownloadannovar">databases::annovar::download_annovar</h4>
<p>Path to Annovar databases (e.g. '/Users/lebechea/howard/databases/annovar/current').</p>
<p>Type: <code>Path</code></p>
<p>Default: <code>None</code></p>
<h4 id="databasesannovardownloadannovarfiles">databases::annovar::download_annovar_files</h4>
<p>Download Annovar databases for a list of Annovar file code (see Annovar Doc). Use None to donwload all available files, or Annovar keyword (e.g. 'refGene', 'cosmic70', 'clinvar_202*'). Note that refGene will at least be downloaded, and only files that not already exist or changed will be downloaded.</p>
<p>Type: <code>str</code></p>
<p>Default: <code>None</code></p>
<h4 id="databasesannovardownloadannovarurl">databases::annovar::download_annovar_url</h4>
<p>Annovar databases URL (see Annovar Doc).</p>
<p>Type: <code>str</code></p>
<p>Default: <code>http://www.openbioinformatics.org/annovar/download</code></p>
<h3 id="databasesrefseq">databases::refseq</h3>
<p>refSeq download.</p>
<h4 id="databasesrefseqdownloadrefseq">databases::refseq::download_refseq</h4>
<p>Path to refSeq databases (e.g. '/Users/lebechea/howard/databases/refseq/current').</p>
<p>Type: <code>Path</code></p>
<p>Default: <code>None</code></p>
<h4 id="databasesrefseqdownloadrefsequrl">databases::refseq::download_refseq_url</h4>
<p>refSeq databases URL (see refSeq WebSite) (e.g. 'http://hgdownload.soe.ucsc.edu/goldenPath')â€¢/n</p>
<p>Type: <code>str</code></p>
<p>Default: <code>http://hgdownload.soe.ucsc.edu/goldenPath</code></p>
<h4 id="databasesrefseqdownloadrefseqprefix">databases::refseq::download_refseq_prefix</h4>
<p>Check existing refSeq files in refSeq folder.</p>
<p>Type: <code>str</code></p>
<p>Default: <code>ncbiRefSeq</code></p>
<h4 id="databasesrefseqdownloadrefseqfiles">databases::refseq::download_refseq_files</h4>
<p>List of refSeq files to download.</p>
<p>Type: <code>str</code></p>
<p>Default: <code>ncbiRefSeq.txt,ncbiRefSeqLink.txt</code></p>
<h4 id="databasesrefseqdownloadrefseqformatfile">databases::refseq::download_refseq_format_file</h4>
<p>Name of refSeq file to convert in BED format (e.g. 'ncbiRefSeq.txt'). Process only if not None.</p>
<p>Type: <code>str</code></p>
<p>Default: <code>None</code></p>
<h4 id="databasesrefseqdownloadrefseqincludeutr5">databases::refseq::download_refseq_include_utr5</h4>
<p>Formating BED refSeq file including 5'UTR.</p>
<p>Default: <code>False</code></p>
<h4 id="databasesrefseqdownloadrefseqincludeutr3">databases::refseq::download_refseq_include_utr3</h4>
<p>Formating BED refSeq file including 3'UTR.</p>
<p>Default: <code>False</code></p>
<h4 id="databasesrefseqdownloadrefseqincludechrm">databases::refseq::download_refseq_include_chrM</h4>
<p>Formating BED refSeq file including Mitochondiral chromosome 'chrM' or 'chrMT'.</p>
<p>Default: <code>False</code></p>
<h4 id="databasesrefseqdownloadrefseqincludenoncanonicalchr">databases::refseq::download_refseq_include_non_canonical_chr</h4>
<p>Formating BED refSeq file including non canonical chromosomes.</p>
<p>Default: <code>False</code></p>
<h4 id="databasesrefseqdownloadrefseqincludenoncodingtranscripts">databases::refseq::download_refseq_include_non_coding_transcripts</h4>
<p>Formating BED refSeq file including non coding transcripts.</p>
<p>Default: <code>False</code></p>
<h4 id="databasesrefseqdownloadrefseqincludetranscriptversion">databases::refseq::download_refseq_include_transcript_version</h4>
<p>Formating BED refSeq file including transcript version.</p>
<p>Default: <code>False</code></p>
<h3 id="databasesdbnsfp">databases::dbnsfp</h3>
<p>dbNSFP download.</p>
<h4 id="databasesdbnsfpdownloaddbnsfp">databases::dbnsfp::download_dbnsfp</h4>
<p>Download dbNSFP databases within dbNSFP folder(e.g. '/Users/lebechea/howard/databases').</p>
<p>Type: <code>Path</code></p>
<p>Default: <code>None</code></p>
<h4 id="databasesdbnsfpdownloaddbnsfpurl">databases::dbnsfp::download_dbnsfp_url</h4>
<p>Download dbNSFP databases URL (see dbNSFP website) (e.g. https://dbnsfp.s3.amazonaws.com').</p>
<p>Type: <code>str</code></p>
<p>Default: <code>https://dbnsfp.s3.amazonaws.com</code></p>
<h4 id="databasesdbnsfpdownloaddbnsfprelease">databases::dbnsfp::download_dbnsfp_release</h4>
<p>Release of dbNSFP to download (see dbNSFP website) (e.g. '4.4a').</p>
<p>Default: <code>4.4a</code></p>
<h4 id="databasesdbnsfpdownloaddbnsfpparquetsize">databases::dbnsfp::download_dbnsfp_parquet_size</h4>
<p>Maximum size (Mb) of data files in Parquet folder. Parquet folder are partitioned (hive) by chromosome (sub-folder), which contain N data files.</p>
<p>Type: <code>int</code></p>
<p>Default: <code>100</code></p>
<h4 id="databasesdbnsfpdownloaddbnsfpsubdatabases">databases::dbnsfp::download_dbnsfp_subdatabases</h4>
<p>Generate dbNSFP sub-databases. dbNSFP provides multiple databases which are split onto multiple columns. This option create a Parquet folder for each sub-database (based on columns names).</p>
<p>Default: <code>False</code></p>
<h4 id="databasesdbnsfpdownloaddbnsfpparquet">databases::dbnsfp::download_dbnsfp_parquet</h4>
<p>Generate a Parquet file for each Parquet folder.</p>
<p>Default: <code>False</code></p>
<h4 id="databasesdbnsfpdownloaddbnsfpvcf">databases::dbnsfp::download_dbnsfp_vcf</h4>
<p>Generate a VCF file for each Parquet folder. Need genome FASTA file (see --download-genome).</p>
<p>Default: <code>False</code></p>
<h4 id="databasesdbnsfpdownloaddbnsfpnofilesall">databases::dbnsfp::download_dbnsfp_no_files_all</h4>
<p>Not generate database Parquet/VCF file for the entire database ('ALL'). Only sub-databases files will be generated. (see '--download-dbnsfp-subdatabases').</p>
<p>Default: <code>False</code></p>
<h4 id="databasesdbnsfpdownloaddbnsfpaddinfo">databases::dbnsfp::download_dbnsfp_add_info</h4>
<p>Add INFO column (VCF format) in Parquet folder and file. Useful for speed up full annotation (all available columns). Increase memory and space during generation of files.</p>
<p>Default: <code>False</code></p>
<h4 id="databasesdbnsfpdownloaddbnsfprowgroupsize">databases::dbnsfp::download_dbnsfp_row_group_size</h4>
<p>Minimum number of rows in a parquet row group (see duckDB doc). Lower can reduce memory usage and slightly increase space during generation, speed up highly selective queries, slow down whole file queries (e.g. aggregations).</p>
<p>Type: <code>int</code></p>
<p>Default: <code>100000</code></p>
<h3 id="databasesalphamissense">databases::alphamissense</h3>
<p>AlphaMissense download.</p>
<h4 id="databasesalphamissensedownloadalphamissense">databases::alphamissense::download_alphamissense</h4>
<p>Path to AlphaMissense databases</p>
<p>Type: <code>Path</code></p>
<p>Default: <code>None</code></p>
<h4 id="databasesalphamissensedownloadalphamissenseurl">databases::alphamissense::download_alphamissense_url</h4>
<p>Download AlphaMissense databases URL (see AlphaMissense website) (e.g. 'https://storage.googleapis.com/dm_alphamissense').</p>
<p>Type: <code>str</code></p>
<p>Default: <code>https://storage.googleapis.com/dm_alphamissense</code></p>
<h3 id="databasesexomiser">databases::exomiser</h3>
<p>Exomiser download.</p>
<h4 id="databasesexomiserdownloadexomiser">databases::exomiser::download_exomiser</h4>
<p>Path to Exomiser databases (e.g. /Users/lebechea/howard/databases/exomiser/current).</p>
<p>Type: <code>Path</code></p>
<p>Default: <code>None</code></p>
<h4 id="databasesexomiserdownloadexomiserapplicationproperties">databases::exomiser::download_exomiser_application_properties</h4>
<p>Exomiser Application Properties configuration file (see Exomiser website). This file contains configuration settings for the Exomiser tool. If this parameter is not provided, the function will attempt to locate the application properties file automatically based on the Exomiser. Configuration information will be used to download expected releases (if no other parameters). CADD and REMM will be downloaded only if 'path' are provided.</p>
<p>Type: <code>Path</code></p>
<p>Default: <code>None</code></p>
<h4 id="databasesexomiserdownloadexomiserurl">databases::exomiser::download_exomiser_url</h4>
<p>URL where Exomiser database files can be downloaded from (e.g. 'http://data.monarchinitiative.org/exomiser').</p>
<p>Type: <code>str</code></p>
<p>Default: <code>http://data.monarchinitiative.org/exomiser</code></p>
<h4 id="databasesexomiserdownloadexomiserrelease">databases::exomiser::download_exomiser_release</h4>
<p>Release of Exomiser data to download. If &quot;default&quot;, &quot;auto&quot;, or &quot;config&quot;, retrieve from Application Properties file. If not provided (None), from Application Properties file (Exomiser data-version)  or default '2109'.</p>
<p>Type: <code>str</code></p>
<p>Default: <code>None</code></p>
<h4 id="databasesexomiserdownloadexomiserphenotyperelease">databases::exomiser::download_exomiser_phenotype_release</h4>
<p>Release of Exomiser phenotype to download. If not provided (None), from Application Properties file (Exomiser Phenotype data-version) or Exomiser release.</p>
<p>Type: <code>str</code></p>
<p>Default: <code>None</code></p>
<h4 id="databasesexomiserdownloadexomiserremmrelease">databases::exomiser::download_exomiser_remm_release</h4>
<p>Release of ReMM (Regulatory Mendelian Mutation) database to download. If &quot;default&quot;, &quot;auto&quot;, or &quot;config&quot;, retrieve from Application Properties file.</p>
<p>Type: <code>str</code></p>
<p>Default: <code>None</code></p>
<h4 id="databasesexomiserdownloadexomiserremmurl">databases::exomiser::download_exomiser_remm_url</h4>
<p>URL where ReMM (Regulatory Mendelian Mutation) database files can be downloaded from (e.g. 'https://kircherlab.bihealth.org/download/ReMM').</p>
<p>Type: <code>str</code></p>
<p>Default: <code>https://kircherlab.bihealth.org/download/ReMM</code></p>
<h4 id="databasesexomiserdownloadexomisercaddrelease">databases::exomiser::download_exomiser_cadd_release</h4>
<p>Release of CADD (Combined Annotation Dependent Depletion) database to download. If &quot;default&quot;, &quot;auto&quot;, or &quot;config&quot;, retrieve from Application Properties file.</p>
<p>Type: <code>str</code></p>
<p>Default: <code>None</code></p>
<h4 id="databasesexomiserdownloadexomisercaddurl">databases::exomiser::download_exomiser_cadd_url</h4>
<p>URL where CADD (Combined Annotation Dependent Depletion) database files can be downloaded from (e.g. 'https://kircherlab.bihealth.org/download/CADD').</p>
<p>Type: <code>str</code></p>
<p>Default: <code>https://kircherlab.bihealth.org/download/CADD</code></p>
<h4 id="databasesexomiserdownloadexomisercaddurlsnvfile">databases::exomiser::download_exomiser_cadd_url_snv_file</h4>
<p>Name of the file containing the SNV (Single Nucleotide Variant) data for the CADD (Combined Annotation Dependent Depletion) database.</p>
<p>Type: <code>str</code></p>
<p>Default: <code>whole_genome_SNVs.tsv.gz</code></p>
<h4 id="databasesexomiserdownloadexomisercaddurlindelfile">databases::exomiser::download_exomiser_cadd_url_indel_file</h4>
<p>Name of the file containing the INDEL (Insertion-Deletion) data for the CADD (Combined Annotation Dependent Depletion) database.</p>
<p>Type: <code>str</code></p>
<p>Default: <code>InDels.tsv.gz</code></p>
<h3 id="databasesdbsnp">databases::dbsnp</h3>
<p>dbSNP download.</p>
<h4 id="databasesdbsnpdownloaddbsnp">databases::dbsnp::download_dbsnp</h4>
<p>Path to dbSNP databases (e.g. '/Users/lebechea/howard/databases/exomiser/dbsnp').</p>
<p>Type: <code>Path</code></p>
<p>Default: <code>None</code></p>
<h4 id="databasesdbsnpdownloaddbsnpreleases">databases::dbsnp::download_dbsnp_releases</h4>
<p>Release of dbSNP to download (e.g. 'b152', 'b152,b156').</p>
<p>Type: <code>str</code></p>
<p>Default: <code>b156</code></p>
<h4 id="databasesdbsnpdownloaddbsnpreleasedefault">databases::dbsnp::download_dbsnp_release_default</h4>
<p>Default Release of dbSNP ('default' symlink) (e.g. 'b156'). If None, first release to download will be assigned as default only if it does not exists.</p>
<p>Type: <code>str</code></p>
<p>Default: <code>None</code></p>
<h4 id="databasesdbsnpdownloaddbsnpurl">databases::dbsnp::download_dbsnp_url</h4>
<p>URL where dbSNP database files can be downloaded from. (e.g. 'https://ftp.ncbi.nih.gov/snp/archive').</p>
<p>Type: <code>str</code></p>
<p>Default: <code>https://ftp.ncbi.nih.gov/snp/archive</code></p>
<h4 id="databasesdbsnpdownloaddbsnpurlfiles">databases::dbsnp::download_dbsnp_url_files</h4>
<p>Dictionary that maps assembly names to specific dbSNP URL files. It allows you to provide custom dbSNP URL files for specific assemblies instead of using the default file naming convention.</p>
<p>Type: <code>str</code></p>
<p>Default: <code>None</code></p>
<h4 id="databasesdbsnpdownloaddbsnpurlfilesprefix">databases::dbsnp::download_dbsnp_url_files_prefix</h4>
<p>String that represents the prefix of the dbSNP file name for a specific assembly. It is used to construct the full URL of the dbSNP file to be downloaded.</p>
<p>Type: <code>str</code></p>
<p>Default: <code>GCF_000001405</code></p>
<h4 id="databasesdbsnpdownloaddbsnpassembliesmap">databases::dbsnp::download_dbsnp_assemblies_map</h4>
<p>dictionary that maps assembly names to their corresponding dbSNP versions. It is used to construct the dbSNP file name based on the assembly name.</p>
<p>Type: <code>str</code></p>
<p>Default: <code>{'hg19': '25', 'hg38': '40'}</code></p>
<h4 id="databasesdbsnpdownloaddbsnpvcf">databases::dbsnp::download_dbsnp_vcf</h4>
<p>Generate well-formatted VCF from downloaded file:</p>
<ul>
<li>
<p>Add and filter contigs associated to assembly</p>
</li>
<li>
<p>Normalize by splitting multiallelics</p>
</li>
<li>
<p>Need genome (see --download-genome)</p>
</li>
</ul>
<p>Default: <code>False</code></p>
<h4 id="databasesdbsnpdownloaddbsnpparquet">databases::dbsnp::download_dbsnp_parquet</h4>
<p>Generate Parquet file from VCF.</p>
<p>Default: <code>False</code></p>
<h3 id="databaseshgmd">databases::hgmd</h3>
<p>HGMD convert.</p>
<h4 id="databaseshgmdconverthgmd">databases::hgmd::convert_hgmd</h4>
<p>Convert HGMD databases. Folder where the HGMD databases will be stored. Fields in VCF, Parquet and TSV will be generated. If the folder does not exist, it will be created.</p>
<p>Type: <code>Path</code></p>
<p>Default: <code>None</code></p>
<h4 id="databaseshgmdconverthgmdfile">databases::hgmd::convert_hgmd_file</h4>
<p>File from HGMD. Name format 'HGMD_Pro_<release>_<assembly>.vcf.gz'.</p>
<p>Type: <code>Path</code></p>
<p>Default: <code>None</code></p>
<h4 id="databaseshgmdconverthgmdbasename">databases::hgmd::convert_hgmd_basename</h4>
<p>File output basename. Generated files will be prefixed by basename (e.g. 'HGMD_Pro_MY_RELEASE') By default (None), input file name without '.vcf.gz'.</p>
<p>Type: <code>str</code></p>
<p>Default: <code>None</code></p>
<h3 id="databasesfromannovar">databases::from_Annovar</h3>
<p>Annovar convert.</p>
<h4 id="databasesfromannovarinputannovar">databases::from_Annovar::input_annovar</h4>
<p>Input Annovar file path. Format file must be a Annovar TXT file, associated with '.idx'.</p>
<p>Type: <code>Path</code></p>
<p>Default: <code>None</code></p>
<h4 id="databasesfromannovaroutputannovar">databases::from_Annovar::output_annovar</h4>
<p>Output Annovar file path. Format file must be either VCF compressesd file '.vcf.gz'.</p>
<p>Type: <code>Path</code></p>
<p>Default: <code>None</code></p>
<h4 id="databasesfromannovarannovarcode">databases::from_Annovar::annovar_code</h4>
<p>Annovar code, or database name. Usefull to name databases columns.</p>
<p>Type: <code>str</code></p>
<p>Default: <code>None</code></p>
<h4 id="databasesfromannovarannovartoparquet">databases::from_Annovar::annovar_to_parquet</h4>
<p>Parquet file conversion.</p>
<p>Type: <code>Path</code></p>
<p>Default: <code>None</code></p>
<h4 id="databasesfromannovarannovarreducememory">databases::from_Annovar::annovar_reduce_memory</h4>
<p>Reduce memory option for Annovar convert, either 'auto' (auto-detection), 'enable' or 'disable'.</p>
<p>Type: <code>str</code></p>
<p>Choices: <code>['auto', 'enable', 'disable']</code></p>
<p>Default: <code>auto</code></p>
<h4 id="databasesfromannovarannovarmultivariant">databases::from_Annovar::annovar_multi_variant</h4>
<p>Variant with multiple annotation lines on Annovar file. Either 'auto' (auto-detection), 'enable' or 'disable'.</p>
<p>Type: <code>str</code></p>
<p>Choices: <code>['auto', 'enable', 'disable']</code></p>
<p>Default: <code>auto</code></p>
<h3 id="databasesfromextann">databases::from_extann</h3>
<p>Extann convert (gene annotation).</p>
<h4 id="databasesfromextanninputextann">databases::from_extann::input_extann</h4>
<p>Input Extann file path. Format file must be a Extann TXT file or TSV file. File need to have at least the genes column.</p>
<p>Type: <code>Path</code></p>
<p>Default: <code>None</code></p>
<h4 id="databasesfromextannoutputextann">databases::from_extann::output_extann</h4>
<p>Output Extann file path. Output extann file, should be BED or BED.gz.</p>
<p>Type: <code>Path</code></p>
<p>Default: <code>None</code></p>
<h4 id="databasesfromextannrefgene">databases::from_extann::refgene</h4>
<p>Path to refGene annotation file.</p>
<p>Type: <code>Path</code></p>
<p>Default: <code>None</code></p>
<h4 id="databasesfromextanntranscripts">databases::from_extann::transcripts</h4>
<p>Transcripts TSV file, with Transcript in first column, optional Gene in second column.</p>
<p>Type: <code>Path</code></p>
<p>Default: <code>None</code></p>
<h4 id="databasesfromextannparamextann">databases::from_extann::param_extann</h4>
<p>Param extann file path. Param containing configuration, options to replace chars and bedlike header description, conf vcf specs.</p>
<p>Type: <code>Path</code></p>
<p>Default: <code>~/howard/config/param.extann.json</code></p>
<h4 id="databasesfromextannmodeextann">databases::from_extann::mode_extann</h4>
<p>Mode extann selection. How to pick transcript from ncbi, keep all, keep the longest, or keep the chosen one (transcript_extann).</p>
<p>Type: <code>str</code></p>
<p>Choices: <code>['all', 'longest', 'chosen']</code></p>
<p>Default: <code>longest</code></p>
<h3 id="databasesparameters">databases::Parameters</h3>
<p>Parameters generation.</p>
<h4 id="databasesparametersgenerateparam">databases::Parameters::generate_param</h4>
<p>Parameter file (JSON) with all databases found. Databases folders scanned are defined in config file. Structure of databases follow this structure (see doc): .../<database>/<release>/<assembly>/*.[parquet|vcf.gz|...]</p>
<p>Type: <code>Path</code></p>
<p>Default: <code>None</code></p>
<h4 id="databasesparametersgenerateparamdescription">databases::Parameters::generate_param_description</h4>
<p>Description file (JSON) with all databases found. Contains all databases with description of format, assembly, fields...</p>
<p>Type: <code>Path</code></p>
<p>Default: <code>None</code></p>
<h4 id="databasesparametersgenerateparamreleases">databases::Parameters::generate_param_releases</h4>
<p>List of database folder releases to check (e.g. 'current', 'latest').</p>
<p>Type: <code>str</code></p>
<p>Default: <code>current</code></p>
<h4 id="databasesparametersgenerateparamformats">databases::Parameters::generate_param_formats</h4>
<p>List of database formats to check (e.g. 'parquet', 'parquet,vcf,bed,tsv').</p>
<p>Type: <code>str</code></p>
<p>Default: <code>parquet</code></p>
<h4 id="databasesparametersgenerateparambcftools">databases::Parameters::generate_param_bcftools</h4>
<p>Generate parameter JSON file with BCFTools annotation for allowed formats (i.e. 'vcf', 'bed').</p>
<p>Default: <code>False</code></p>

</body>
</html>
