<H1 id='HOWARD Parameters'>HOWARD Parameters</H1>HOWARD Parameters JSON file defines parameters to process annotations, calculations, prioritizations, convertions and queries.<br>
                    <H2>Table of contents</H1>
                    - <a href='#HOWARD Parameters'>HOWARD Parameters</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#hgvs'>hgvs</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#hgvs::use_gene'>use_gene</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#hgvs::use_exon'>use_exon</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#hgvs::use_protein'>use_protein</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#hgvs::add_protein'>add_protein</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#hgvs::full_format'>full_format</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#hgvs::codon_type'>codon_type</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#hgvs::refgene'>refgene</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#hgvs::refseqlink'>refseqlink</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#annotation'>annotation</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#annotation::parquet'>parquet</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#annotation::parquet::annotations'>annotations</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#annotation::bcftools'>bcftools</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#annotation::bcftools::annotations'>annotations</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#annotation::annovar'>annovar</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#annotation::annovar::annotations'>annotations</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#annotation::annovar::options'>options</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#annotation::snpeff'>snpeff</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#annotation::snpeff::options'>options</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#annotation::snpeff::stats'>stats</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#annotation::snpeff::csvStats'>csvStats</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#annotation::exomiser'>exomiser</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#annotation::exomiser::release'>release</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#annotation::options'>options</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#annotation::options::annotations_update'>annotations_update</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#annotation::options::annotations_append'>annotations_append</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#calculation'>calculation</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#calculation::calculations'>calculations</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#calculation::calculation_config'>calculation_config</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#prioritization'>prioritization</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#prioritization::prioritizations'>prioritizations</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#prioritization::profiles'>profiles</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#prioritization::default_profile'>default_profile</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#prioritization::pzfields'>pzfields</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#prioritization::prioritization_score_mode'>prioritization_score_mode</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#stats'>stats</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#stats::stats_md'>stats_md</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#stats::stats_json'>stats_json</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#query'>query</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#query::query'>query</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#query::query_limit'>query_limit</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#query::query_print_mode'>query_print_mode</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#export'>export</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#export::order_by'>order_by</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#export::include_header'>include_header</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#export::parquet_partitions'>parquet_partitions</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#explode'>explode</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#explode::explode_infos'>explode_infos</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#explode::explode_infos_prefix'>explode_infos_prefix</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#explode::explode_infos_fields'>explode_infos_fields</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#threads'>threads</a><BR>
                    <br>
                    Examples: <xmp># Parameters for annotation, calculation, prioritization, HGVS annotation and export{
   "hgvs": {
      "full_format": true,
      "use_exon": true
   }
   "annotation": {
      "parquet": {
         "annotations": {
            "/path/to/database3.parquet": {
               "field1": null,
               "field2": "field2_renamed"
            },
            "/path/to/database4.vcf.gz": {
               "INFO": null
            },
            "/path/to/database5.bed.gz": {
               "INFO": null
            }
         }
      },
      "bcftools": {
         "annotations": {
            "/path/to/database6.vcf.gz": {
               "field1": null,
               "field2": "field2_renamed",
            },
            "/path/to/database7.bed": {
               "INFO": null
            }
         }
      },
      "annovar": {
         "annotations": {
            "annovar_keyword2": {
               "field1": null,
               "field2": "field2_renamed",
            },
            "annovar_keyword3": {
               "INFO": null
            }
         }
      },
      "snpeff": {
         "options": {
            " -hgvs -noShiftHgvs -spliceSiteSize 3 -lof -oicr "}
         }
      },
      "exomiser": {
         "release": "2109",
         "transcript_source": "refseq",
         "hpo": ["HP:0001156", "HP:0001363", "HP:0011304", "HP:0010055"]
      },
      "options": {
         "append": true
      }
   },
   "calculation": {
      "operation1": null,
      "operation2": {
        "options": {
          "option1": "value1",
          "option2": "value2"
        }
      }
   },
   "prioritization": {
      "prioritizations": "config/prioritization_profiles.json",
      "profiles": ["GENOME", "GERMLINE"],
      "default_profile": "GERMLINE",
      "pzfields": ["PZScore", "PZFlag", "PZComment"],
      "prioritization_score_mode": "VaRank"
   },
   "export": {
      "include_header": true
   }
}</xmp><br><H2 id='hgvs'>hgvs</H2>HOWARD annotates variants with HGVS annotation using HUGO HGVS internation Sequence Variant Nomenclature (http://varnomen.hgvs.org/). Annotation refere to refGene and genome to generate HGVS nomenclature for all available transcripts. This annotation add 'hgvs' field into VCF INFO column of a VCF file. Several options are available, to add gene, exon and protein information, to generate a 'full format' detailed annotation, to choose codon format.<br>Examples: <xmp># HGVS annotation  with operations for generate variant_id and variant type, extract HGVS from snpEff annotation, select NOMEN from snpEff HGVS with a list of transcripts of preference
"hgvs": {
  "full_format": true,
  "use_exon": true
}</xmp><br><H3 id='hgvs::use_gene'>hgvs::use_gene</H3>Add Gene information to generate HGVS annotation (e.g. 'NM_152232**(TAS1R2)**:c.231T>C').<br>Default: False<br>Examples: <xmp># Use Gene in HGVS annotation
"use_gene": true</xmp><br><H3 id='hgvs::use_exon'>hgvs::use_exon</H3>Add Exon information to generate HGVS annotation (e.g. 'NM_152232(exon2):c.231T>C'). Only if 'use_gene' is not enabled.<br>Default: False<br>Examples: <xmp># Use Exon in HGVS annotation
"use_exon": true</xmp><br><H3 id='hgvs::use_protein'>hgvs::use_protein</H3>Use Protein level to generate HGVS annotation (e.g. 'NP_689418:p.Cys77Arg'). Can be used with 'use_exon' or 'use_gene'.<br>Default: False<br>Examples: <xmp># Use Protein in HGVS annotation
"use_protein": true</xmp><br><H3 id='hgvs::add_protein'>hgvs::add_protein</H3>Add Protein level to DNA HGVS annotation (e.g. 'NM_152232:c.231T>C,NP_689418:p.Cys77Arg').<br>Default: False<br>Examples: <xmp># Add Protein level to DNA HGVS annotation
"add_protein": true</xmp><br><H3 id='hgvs::full_format'>hgvs::full_format</H3>Generates HGVS annotation in a full format (non-standard, e.g. 'TAS1R2:NM_152232:NP_689418:c.231T>C:p.Cys77Arg', 'TAS1R2:NM_152232:NP_689418:exon2:c.231T>C:p.Cys77Arg'). Full format use all information to generates an exhaustive annotation. Use specifically 'use_exon' to add exon information.<br>Default: False<br>Examples: <xmp># Use full format for HGVS annotation
"full_format": true</xmp><br><H3 id='hgvs::codon_type'>hgvs::codon_type</H3>Amino Acide Codon format type to use to generate HGVS annotation (default '3'):<br>- '1': codon in 1 character (e.g. 'C', 'R')<br>- '3': codon in 3 character (e.g. 'Cys', 'Arg')<br>- 'FULL': codon in full name (e.g. 'Cysteine', 'Arginine')<br>Type: str<br>Choices: <br>1<br>3<br>FULL<br>Default: 3<br>Examples: <xmp># Amino Acide Codon format with 1 character
"codon_type": "1"
# Amino Acide Codon format with 3 character
"codon_type": "3"
# Amino Acide Codon format with full name
"codon_type": "FULL"</xmp><br><H3 id='hgvs::refgene'>hgvs::refgene</H3>Path to refGene annotation file (see [HOWARD User Guide](user_guide.md#databases-tool)).<br>Type: Path<br>Default: None<br>Examples: <xmp># Path to refSeq file
"refgene": "~/howard/databases/refseq/current/hg19/ncbiRefSeq.txt"</xmp><br><H3 id='hgvs::refseqlink'>hgvs::refseqlink</H3>Path to refGeneLink annotation file (see [HOWARD User Guide](user_guide.md#databases-tool)).<br>Type: Path<br>Default: None<br>Examples: <xmp># Path to refSeq file
"refseqlink": "~/howard/databases/refseq/current/hg19/ncbiRefSeqLink.txt"</xmp><br><H2 id='annotation'>annotation</H2>Annotation process using HOWARD algorithms or external tools.
<br>For HOWARD Parquet algorithm, provide the list of database files available (formats such as Parquet, VCF, TSV, duckDB, JSON) and select fields (rename possible, 'INFO' keyword for all fields), or use 'ALL' keyword to detect available databases.
<br>For external tools, such as Annovar, snpEff and Exomiser, specify parameters such as annotation keywords (Annovar) and options (depending on the tool), and select fields (BCFtools and Annovar, field rename available).<br>Examples: <xmp># Annotation with multiple tools in multiple formats with multiple options
"annotation": {
   "parquet": {
      "annotations": {
         "/path/to/database3.parquet": {
            "field1": null,
            "field2": "field2_renamed"
         },
         "/path/to/database4.vcf.gz": {
            "INFO": null
         }
         "/path/to/database5.bed.gz": {
            "INFO": null
         }
      }
   }
   "bcftools": {
      "annotations": {
         "/path/to/database6.vcf.gz": {
            "field1": null,
            "field2": "field2_renamed"
         },
         "/path/to/database7.bed": {
            "INFO": null
         }
      }
   }
   "annovar": {
      "annotations": {
         "annovar_keyword2": {
            "field1": null,
            "field2": "field2_renamed"
         },
         "annovar_keyword3": {
            "INFO": null
         }
      }
   }
   "snpeff": {
      "options": {
         " -hgvs -noShiftHgvs -spliceSiteSize 3 -lof -oicr "}
      }
   }
   "exomiser": {
      "release": "2109",
      "transcript_source": "refseq",
      "hpo": ["HP:0001156", "HP:0001363", "HP:0011304", "HP:0010055"]
   }
   "options": {
      "append": true
   }
}</xmp><br><H3 id='annotation::parquet'>annotation::parquet</H3>Annotation process using HOWARD Parquet algorithm, for the list of databases available (formats such as Parquet, VCF, TSV, duckDB, JSON).<br>Examples: <xmp>Annotation with multiple databases in multiple formats"parquet": {
   "annotations": {
      "/path/to/database3.parquet": {
         "field1": null,
         "field2": "field2_renamed",
      },
      "/path/to/database4.vcf.gz": {
         "INFO": null
      }
      "/path/to/database5.bed.gz": {
         "INFO": null
      }
   }
}</xmp><br><H4 id='annotation::parquet::annotations'>annotation::parquet::annotations</H4>Specify a list of databases files available (formats such as Parquet, VCF, TSV, duckDB, JSON). This parameter enables users to select specific database fields and optionally rename them (e.g. '"field": null' to keep field name, '"field": "new_name"' to rename field). Use 'INFO' keyword to select all fields within the database INFO/Tags header (e.g. '"INFO": null'). Use 'ALL' keyword to select all fields within the database regardless INFO/Tags header (e.g. '"ALL": null').
<br>For add all availalbe databases files, use 'ALL' keyword, with filters on type and release (e.g. 'ALL', 'ALL:parquet:current', 'ALL:parquet,vcf:current,devel').
<br>If a full path is not provided, the system will automatically detect files within database folders (see Configuration doc) and assembly (see Parameter option).<br>Examples: <xmp>Annotation with all available databases in Parquet for ''current release"parquet": {
   "annotations": {
      "ALL": {
         "formats": ["parquet"],
         "releases": ["current"]
      }
   }
}</xmp><br><H3 id='annotation::bcftools'>annotation::bcftools</H3>Annotation process using BCFTools. Provide a list of database files and annotation fields.<br>Examples: <xmp># Annotation with multiple databases in multiple formats
"parquet": {
   "bcftools": {
      "/path/to/database1.vcf.gz": {
         "field1": null
         "field2": "field2_renamed",
      },
      "database2.bed.gz": {
         "INFO": null
      }
   }
}</xmp><br><H4 id='annotation::bcftools::annotations'>annotation::bcftools::annotations</H4>Specify the list of database files in formats VCF or BED. Files need to be compressed and indexed.<br>This parameter enables users to select specific database fields and optionally rename them (e.g. '"field": null' to keep field name, '"field": "new_name"' to rename field). Use 'INFO' or 'ALL' keyword to select all fields within the database INFO/Tags header (e.g. '"INFO": null', '"ALL": null').
<br>If a full path is not provided, the system will automatically detect files within database folders (see Configuration doc) and assembly (see Parameter option).<br>Examples: <xmp># Annotation with dbSNP database  with all fields
"annotations": {
   "tests/databases/annotations/current/hg19/avsnp150.vcf.gz": {
      "INFO": null
   }
}
# Annotation with dbNSFP (only PolyPhen, ClinVar and REVEL score), and rename fields
"annotations": {
   "tests/databases/annotations/current/hg19/dbnsfp42a.vcf.gz": {
      "Polyphen2_HDIV_pred": "PolyPhen",
      "ClinPred_pred": "ClinVar",
      "REVEL_score": null
   }
}
# Annotation with refSeq as a BED file
"annotations": {
   "tests/databases/annotations/current/hg19/refGene.bed": {
      "INFO": null
   }
}
# Annotation with dbNSFP REVEL annotation (as a VCF file) within configured annotation databases folders (default: '~/howard/databases/annotations/current') and assembly (default: 'hg19')
"annotations": {
   "dbnsfp42a.REVEL.vcf.gz": {
      "REVEL_score": null,
      "REVEL_rankscore": null
   }
}</xmp><br><H3 id='annotation::annovar'>annotation::annovar</H3>Annotation process using Annovar tool. Provides a list of keywords to select Annovar databases, and defines Annovar options (see [Annovar documentation](https://annovar.openbioinformatics.org)).<br>Examples: <xmp># Annotation with multiple Annovar databases, with fields selection, and Annovar options
"annovar": {
   "annotations": {
      "annovar_keyword2": {
         "field1": null,
         "field2": "field2_renamed",
      },
      "annovar_keyword3": {
         "INFO": null
      }
   }
}</xmp><br><H4 id='annotation::annovar::annotations'>annotation::annovar::annotations</H4>List of keywords refering to Annovar databases (see [Annovar Databases documentation](https://annovar.openbioinformatics.org/en/latest/user-guide/download/)), with a list of selected fields for each of them (rename available)<br>Examples: <xmp># Annotation with ClinVar (fields CLNSIG and CLNDN renamed) and Cosmic (all fields)
"annotations": {
   "clinvar_20221231": {
      "CLNSIG": "ClinVar_class"
      "CLNDN": "ClinVar_desease",
   },
   "cosmic70": {
      "INFO": null
   },
}</xmp><br><H4 id='annotation::annovar::options'>annotation::annovar::options</H4>List of options available with Annovar tool (see Annovar documentation). As example, these options allows to define splicing threshold or HGVS annotation with refGene database<br>Examples: <xmp># HGVS Annotation with refGene (add 'refGene' to 'annotations') and a splicing threshold as 3
"options": {
   "splicing_threshold": 3,
   "argument": "'-hgvs'"
   }
}</xmp><br><H3 id='annotation::snpeff'>annotation::snpeff</H3>Annotation process using snpEff tool and options (see [snpEff documentation](https://pcingola.github.io/SnpEff/snpeff/commandline/)).<br>Examples: <xmp># Annotation with snpEff databases, with options for HGVS annotation and additional tags.
"snpeff": {
   "options": {
      " -hgvs -noShiftHgvs -spliceSiteSize 3 -lof -oicr "
   }
}</xmp><br><H4 id='annotation::snpeff::options'>annotation::snpeff::options</H4>String (as command line) of options available such as:<br> - filters on variants (regions filter, specific changes as intronic or downstream)<br> - annotation (e.g. HGVS, loss of function) <br> - database (e.g. only protein coding transcripts, splice sites size)<br>Examples: <xmp># Annotation with snpEff databases, with options to generate HGVS annotation, specify to not shift variants according to HGVS notation, define splice sites size to 3, add loss of function (LOF), Nonsense mediated decay and OICR tags.
"options": " -hgvs -noShiftHgvs -spliceSiteSize 3 -lof -oicr "</xmp><br><H4 id='annotation::snpeff::stats'>annotation::snpeff::stats</H4>HTML file for snpEff stats. Use keyword 'OUTPUT' to generate file according to output file.<br>Examples: <xmp># Annotation with snpEff databases, and generate a specific stats in HTML format.
"stats": "/path/to/stats.html"
# Annotation with snpEff databases, and generate stats in HTML format associated with output file.
"stats": "OUTPUT.html"</xmp><br><H4 id='annotation::snpeff::csvStats'>annotation::snpeff::csvStats</H4>CSV file for snpEff stats. Use keyword 'OUTPUT' to generate file according to output file.<br>Examples: <xmp># Annotation with snpEff databases, and generate a specific stats in CSV format.
"csvStats": "/path/to/stats.csv"
# Annotation with snpEff databases, and generate stats in CSV format associated with output file.
"csvStats": "OUTPUT.csv"</xmp><br><H3 id='annotation::exomiser'>annotation::exomiser</H3>Annotation process using Exomiser tool and options (see [Exomiser website documentation](https://www.sanger.ac.uk/tool/exomiser/)).<br>Examples: <xmp># Annotation with Exomiser, using database relse '2109', transcripts source as UCSC and a list of HPO terms.
"exomiser": {
   "release": "2109"
   "transcript_source": "refseq"
   "hpo": ["HP:0001156", "HP:0001363", "HP:0011304", "HP:0010055"]
}</xmp><br><H4 id='annotation::exomiser::release'>annotation::exomiser::release</H4>Release of Exomiser database. This option replace the release variable in 'application.properties' file (see 'exomiser_application_properties' option). The release will be downloaded if it is not available locally. <br>Examples: <xmp># Annotation with release '2109' of Exomiser database.
"release": "2109"</xmp><br><H3 id='annotation::options'>annotation::options</H3>Options for annotations, such as annotation strategy (skip if exists, update, append)<br>Examples: <xmp># Annotation with Parquet databases, with update annotation strategy.
"options": {
   "update": true
}</xmp><br><H4 id='annotation::options::annotations_update'>annotation::options::annotations_update</H4>Update option for annotation (only for Parquet annotation). If True, annotation fields will be removed and re-annotated. These options will be applied to all annotation databases.<br>Default: False<br>Examples: <xmp># Apply update on all annotation fields for all databases.
"update": true</xmp><br><H4 id='annotation::options::annotations_append'>annotation::options::annotations_append</H4>Append option for annotation (only for Parquet annotation). If True, annotation fields will be annotated only if not annotation exists for the variant. These options will be applied to all annotation databases.<br>Default: False<br>Examples: <xmp># Apply append on all annotation fields for all databases.
"append": true</xmp><br><H2 id='calculation'>calculation</H2>Calculation process operations that are defiend in a Calculation Configuration JSON file. List available calculation operations with possible options (see [Calculation JSON file](help.calculation.md) help).<br>Examples: <xmp># Calculation of operations 'operation1' and 'operation2' (with options) defined in 'calculation_config.json' file
"calculation": {
  "calculations": {
    "operation1": null,
    "operation2": {
      "options": {
        "option1": "value1",
        "option2": "value2"
      }
    }
  },
  "calculation_config": "calculation_config.json"
}</xmp><br><H3 id='calculation::calculations'>calculation::calculations</H3>List of operations to process with possible options (see [Calculation JSON file](help.calculation.md) help).<br>Examples: <xmp># Calculation with operations for generate variant_id and variant type, extract HGVS from snpEff annotation, select NOMEN from snpEff HGVS with a list of transcripts of preference
"calculations": {
  "variant_id": null,
  "vartype": null,
  "snpeff_hgvs": null,
  "NOMEN": {
    "options": {
      "hgvs_field": "snpeff_hgvs",
      "transcripts": "tests/data/transcripts.tsv"
    }
  }
}</xmp><br><H3 id='calculation::calculation_config'>calculation::calculation_config</H3>Calculation configuration JSON file. <br>Type: Path<br>Default: None<br>Examples: <xmp>Calculation configuration JSON file as an option"calculation_config": "calculation_config.json" </xmp><br><H2 id='prioritization'>prioritization</H2>Prioritization process use a JSON configuration file that defines all profiles that can be used. By default, all profiles will be calculated from the JSON configuration file, and the first profile will be considered as default. Proritization annotations (INFO/tags) will be generated depending of a input list (default 'PZScore' and 'PZFlag'), for all profiles (e.g. 'PZScore_GERMLINE' for 'GERMLINE' profile) and for default profile (e.g. 'PZScore' for default). Prioritization score mode is 'HOWARD' by default.<br>Examples: <xmp># Prioritization with 'GENOME' and 'GERMLINE' (default) profiles, from a list of configured profiles, only 3 prioritization fields returned, and score calculated in 'VaRank' mode
"prioritization": {
  "prioritizations": "config/prioritization_profiles.json",
  "profiles": ["GENOME", "GERMLINE"],
  "default_profile": "GERMLINE",
  "pzfields": ["PZScore", "PZFlag", "PZComment"],
  "prioritization_score_mode": "VaRank"
}</xmp><br><H3 id='prioritization::prioritizations'>prioritization::prioritizations</H3>Prioritization configuration profiles JSON file defining profiles to calculate. All configured profiles will be calculated by default (see 'profiles' parameter). First profile will be considered as 'default' if none are provided (see 'default_profile' parameter). Default score calculation mode is 'HOWARD'. This option refers to the quick prioritization command line parameter `--prioritizations`.<br>Type: str<br>Default: None<br>Examples: <xmp># Prioritization configuration profile JSON file
"prioritizations": "config/prioritization_profiles.json"</xmp><br><H3 id='prioritization::profiles'>prioritization::profiles</H3>Prioritization profiles to consider, from the list of configured profiles. If empty, all configured profiles will be calculated. First profile will be considered as 'default' if none are provided (see 'default_profile' parameter). Prioritization annotations (INFO/tags) will be generated for all these profiles (e.g. 'PZScore_GERMLINE' for 'GERMLINE' profile).<br>Type: str<br>Default: None<br>Examples: <xmp># Prioritization with 'GERMLINE' profile only
"profiles": ["GERMLINE"]
# Prioritization with 'GENOME' and 'GERMLINE' profiles
"profiles": ["GENOME", "GERMLINE"]</xmp><br><H3 id='prioritization::default_profile'>prioritization::default_profile</H3>Prioritization default profile from the list of processed profiles. Prioritization annotations (INFO/tags) will be generated for this default profile (e.g. 'PZScore', 'PZFlags').<br>Type: str<br>Default: None<br>Examples: <xmp># Prioritization default profile 'GERMLINE'
"default_profile": "GERMLINE"</xmp><br><H3 id='prioritization::pzfields'>prioritization::pzfields</H3>Prioritization annotations (INFO/tags) to generate. By default 'PZScore', 'PZFlags'.<br>Prioritization fields can be selected from:<br>- PZScore: calculated score from all passing filters, depending of the mode<br>- PZFlag: final flag ('PASS' or 'FILTERED'), with strategy that consider a variant is filtered as soon as at least one filter do not pass. By default, the variant is considered as 'PASS' (no filter pass)<br>- PZComment: concatenation of all passing filter comments<br>- PZTags: combinason of score, flags and comments in a tags format (e.g. 'PZFlag#PASS|PZScore#15|PZComment#Described on ...')<br>- PZInfos: information about passing filter criteria<br>Type: str<br>Default: PZScore,PZFlag<br>Examples: <xmp># Prioritization annotations (INFO/tags) list
"pzfields": ["PZScore", "PZFlag", "PZComment"]</xmp><br><H3 id='prioritization::prioritization_score_mode'>prioritization::prioritization_score_mode</H3>Prioritization score can be calculated following multiple mode. The HOWARD mode will increment scores of all passing filters (default). The VaRank mode will select the maximum score from all passing filters.<br>Type: str<br>Choices: <br>HOWARD<br>VaRank<br>Default: HOWARD<br>Examples: <xmp># Prioritization score calculation mode 'HOWARD'
"prioritization_score_mode": "HOWARD"
# Prioritization score calculation mode 'VaRank'
"prioritization_score_mode": "VaRank"</xmp><br><H2 id='stats'>stats</H2>Statistics on loaded variants.<br><H3 id='stats::stats_md'>stats::stats_md</H3>Stats Output file in MarkDown format. <br>Type: Path<br>Default: None<br>Examples: <xmp>Export statistics in Markdown format"stats_md": "/tmp/stats.md" </xmp><br><H3 id='stats::stats_json'>stats::stats_json</H3>Stats Output file in JSON format. <br>Type: Path<br>Default: None<br>Examples: <xmp>Export statistics in JSON format"stats_json": "/tmp/stats.json" </xmp><br><H2 id='query'>query</H2>Query options tools. Mainly a SQL query, based on 'variants' table corresponding on input file data, or a independant query. Print options for 'query' tool allow limiting number of lines and choose printing mode.<br>Type: str<br>Default: None<br><H3 id='query::query'>query::query</H3>Query in SQL format (e.g. 'SELECT * FROM variants LIMIT 50'). <br>Type: str<br>Default: None<br>Examples: <xmp># Simple query to show all variants file
SELECT "#CHROM", POS, REF, ALT, INFO 
FROM variants</xmp><br><H3 id='query::query_limit'>query::query_limit</H3>Limit of number of row for query (only for print result, not output). <br>Type: int<br>Default: 10<br><H3 id='query::query_print_mode'>query::query_print_mode</H3>Print mode of query result (only for print result, not output). Either None (native), 'markdown' or 'tabulate'. <br>Type: str<br>Choices: <br>None<br>markdown<br>tabulate<br>Default: None<br><H2 id='export'>export</H2>Export options for output files, such as data order, include header in output and hive partitioning.<br><H3 id='export::order_by'>export::order_by</H3>List of columns to sort the result-set in ascending or descending order. Use SQL format, and keywords ASC (ascending) and DESC (descending). If a column is not available, order will not be considered. Order is enable only for compatible format (e.g. TSV, CSV, JSON). Examples: 'ACMG_score DESC', 'PZFlag DESC, PZScore DESC'. <br>Type: str<br>Default: <br>Examples: <xmp>Order by PZFlag and PZScore in descending order"order_by": PZFlag DESC, PZScore DESC" </xmp><br><H3 id='export::include_header'>export::include_header</H3>Include header (in VCF format) in output file. Only for compatible formats (tab-delimiter format as TSV or BED). <br>Default: False<br><H3 id='export::parquet_partitions'>export::parquet_partitions</H3>Parquet partitioning using hive (available for any format). This option is faster parallel writing, but memory consuming. Use 'None' (string) for NO partition but split parquet files into a folder. Examples: '#CHROM', '#CHROM,REF', 'None'. <br>Type: str<br>Default: None<br><H2 id='explode'>explode</H2>Explode options for INFO/tags annotations within VCF files.<br><H3 id='explode::explode_infos'>explode::explode_infos</H3>Explode VCF INFO/Tag into 'variants' table columns. <br>Default: False<br><H3 id='explode::explode_infos_prefix'>explode::explode_infos_prefix</H3>Explode VCF INFO/Tag with a specific prefix. <br>Type: str<br>Default: <br><H3 id='explode::explode_infos_fields'>explode::explode_infos_fields</H3>Explode VCF INFO/Tag specific fields/tags. Keyword `*` specify all available fields, except those already specified. Pattern (regex) can be used, such as `.*_score` for fields named with '_score' at the end. Examples:<br>- 'HGVS,SIFT,Clinvar' (list of fields)<br>- 'HGVS,*,Clinvar' (list of fields with all other fields at the end)<br>- 'HGVS,.*_score,Clinvar' (list of 2 fields with all scores in the middle)<br>- 'HGVS,.*_score,*' (1 field, scores, all other fields)<br>- 'HGVS,*,.*_score' (1 field, all other fields, all scores) <br>Type: str<br>Default: *<br><H2 id='threads'>threads</H2>Specify the number of threads to use for processing HOWARD. It determines the level of parallelism, either on python scripts, duckdb engine and external tools. It and can help speed up the process/tool. Use -1 to use all available CPU/cores. Either non valid value is 1 CPU/core. <br>Type: int<br>Default: -1<br>Examples: <xmp># Define 8 CPU/cores"threads": 8</xmp><br>