<!DOCTYPE html>
<html>
<head>
<title>help.md</title>
<meta http-equiv="Content-type" content="text/html;charset=UTF-8">

<style>
/* https://github.com/microsoft/vscode/blob/master/extensions/markdown-language-features/media/markdown.css */
/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Microsoft Corporation. All rights reserved.
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/

body {
	font-family: var(--vscode-markdown-font-family, -apple-system, BlinkMacSystemFont, "Segoe WPC", "Segoe UI", "Ubuntu", "Droid Sans", sans-serif);
	font-size: var(--vscode-markdown-font-size, 14px);
	padding: 0 26px;
	line-height: var(--vscode-markdown-line-height, 22px);
	word-wrap: break-word;
}

#code-csp-warning {
	position: fixed;
	top: 0;
	right: 0;
	color: white;
	margin: 16px;
	text-align: center;
	font-size: 12px;
	font-family: sans-serif;
	background-color:#444444;
	cursor: pointer;
	padding: 6px;
	box-shadow: 1px 1px 1px rgba(0,0,0,.25);
}

#code-csp-warning:hover {
	text-decoration: none;
	background-color:#007acc;
	box-shadow: 2px 2px 2px rgba(0,0,0,.25);
}

body.scrollBeyondLastLine {
	margin-bottom: calc(100vh - 22px);
}

body.showEditorSelection .code-line {
	position: relative;
}

body.showEditorSelection .code-active-line:before,
body.showEditorSelection .code-line:hover:before {
	content: "";
	display: block;
	position: absolute;
	top: 0;
	left: -12px;
	height: 100%;
}

body.showEditorSelection li.code-active-line:before,
body.showEditorSelection li.code-line:hover:before {
	left: -30px;
}

.vscode-light.showEditorSelection .code-active-line:before {
	border-left: 3px solid rgba(0, 0, 0, 0.15);
}

.vscode-light.showEditorSelection .code-line:hover:before {
	border-left: 3px solid rgba(0, 0, 0, 0.40);
}

.vscode-light.showEditorSelection .code-line .code-line:hover:before {
	border-left: none;
}

.vscode-dark.showEditorSelection .code-active-line:before {
	border-left: 3px solid rgba(255, 255, 255, 0.4);
}

.vscode-dark.showEditorSelection .code-line:hover:before {
	border-left: 3px solid rgba(255, 255, 255, 0.60);
}

.vscode-dark.showEditorSelection .code-line .code-line:hover:before {
	border-left: none;
}

.vscode-high-contrast.showEditorSelection .code-active-line:before {
	border-left: 3px solid rgba(255, 160, 0, 0.7);
}

.vscode-high-contrast.showEditorSelection .code-line:hover:before {
	border-left: 3px solid rgba(255, 160, 0, 1);
}

.vscode-high-contrast.showEditorSelection .code-line .code-line:hover:before {
	border-left: none;
}

img {
	max-width: 100%;
	max-height: 100%;
}

a {
	text-decoration: none;
}

a:hover {
	text-decoration: underline;
}

a:focus,
input:focus,
select:focus,
textarea:focus {
	outline: 1px solid -webkit-focus-ring-color;
	outline-offset: -1px;
}

hr {
	border: 0;
	height: 2px;
	border-bottom: 2px solid;
}

h1 {
	padding-bottom: 0.3em;
	line-height: 1.2;
	border-bottom-width: 1px;
	border-bottom-style: solid;
}

h1, h2, h3 {
	font-weight: normal;
}

table {
	border-collapse: collapse;
}

table > thead > tr > th {
	text-align: left;
	border-bottom: 1px solid;
}

table > thead > tr > th,
table > thead > tr > td,
table > tbody > tr > th,
table > tbody > tr > td {
	padding: 5px 10px;
}

table > tbody > tr + tr > td {
	border-top: 1px solid;
}

blockquote {
	margin: 0 7px 0 5px;
	padding: 0 16px 0 10px;
	border-left-width: 5px;
	border-left-style: solid;
}

code {
	font-family: Menlo, Monaco, Consolas, "Droid Sans Mono", "Courier New", monospace, "Droid Sans Fallback";
	font-size: 1em;
	line-height: 1.357em;
}

body.wordWrap pre {
	white-space: pre-wrap;
}

pre:not(.hljs),
pre.hljs code > div {
	padding: 16px;
	border-radius: 3px;
	overflow: auto;
}

pre code {
	color: var(--vscode-editor-foreground);
	tab-size: 4;
}

/** Theming */

.vscode-light pre {
	background-color: rgba(220, 220, 220, 0.4);
}

.vscode-dark pre {
	background-color: rgba(10, 10, 10, 0.4);
}

.vscode-high-contrast pre {
	background-color: rgb(0, 0, 0);
}

.vscode-high-contrast h1 {
	border-color: rgb(0, 0, 0);
}

.vscode-light table > thead > tr > th {
	border-color: rgba(0, 0, 0, 0.69);
}

.vscode-dark table > thead > tr > th {
	border-color: rgba(255, 255, 255, 0.69);
}

.vscode-light h1,
.vscode-light hr,
.vscode-light table > tbody > tr + tr > td {
	border-color: rgba(0, 0, 0, 0.18);
}

.vscode-dark h1,
.vscode-dark hr,
.vscode-dark table > tbody > tr + tr > td {
	border-color: rgba(255, 255, 255, 0.18);
}

</style>

<style>
/*

github.com style (c) Vasily Polovnyov <vast@whiteants.net>

*/

.hljs {
  display: block;
  overflow-x: auto;
  padding: 0.5em;
  color: #333;
  background: #f8f8f8;
}

.hljs-comment,
.hljs-quote {
  color: #998;
  font-style: italic;
}

.hljs-keyword,
.hljs-selector-tag,
.hljs-subst {
  color: #333;
  font-weight: bold;
}

.hljs-number,
.hljs-literal,
.hljs-variable,
.hljs-template-variable,
.hljs-tag .hljs-attr {
  color: #008080;
}

.hljs-string,
.hljs-doctag {
  color: #d14;
}

.hljs-title,
.hljs-section,
.hljs-selector-id {
  color: #900;
  font-weight: bold;
}

.hljs-subst {
  font-weight: normal;
}

.hljs-type,
.hljs-class .hljs-title {
  color: #458;
  font-weight: bold;
}

.hljs-tag,
.hljs-name,
.hljs-attribute {
  color: #000080;
  font-weight: normal;
}

.hljs-regexp,
.hljs-link {
  color: #009926;
}

.hljs-symbol,
.hljs-bullet {
  color: #990073;
}

.hljs-built_in,
.hljs-builtin-name {
  color: #0086b3;
}

.hljs-meta {
  color: #999;
  font-weight: bold;
}

.hljs-deletion {
  background: #fdd;
}

.hljs-addition {
  background: #dfd;
}

.hljs-emphasis {
  font-style: italic;
}

.hljs-strong {
  font-weight: bold;
}

</style>

<style>
/*
 * Markdown PDF CSS
 */

 body {
	font-family: -apple-system, BlinkMacSystemFont, "Segoe WPC", "Segoe UI", "Ubuntu", "Droid Sans", sans-serif, "Meiryo";
	padding: 0 12px;
}

pre {
	background-color: #f8f8f8;
	border: 1px solid #cccccc;
	border-radius: 3px;
	overflow-x: auto;
	white-space: pre-wrap;
	overflow-wrap: break-word;
}

pre:not(.hljs) {
	padding: 23px;
	line-height: 19px;
}

blockquote {
	background: rgba(127, 127, 127, 0.1);
	border-color: rgba(0, 122, 204, 0.5);
}

.emoji {
	height: 1.4em;
}

code {
	font-size: 14px;
	line-height: 19px;
}

/* for inline code */
:not(pre):not(.hljs) > code {
	color: #C9AE75; /* Change the old color so it seems less like an error */
	font-size: inherit;
}

/* Page Break : use <div class="page"/> to insert page break
-------------------------------------------------------- */
.page {
	page-break-after: always;
}

</style>

<script src="https://unpkg.com/mermaid/dist/mermaid.min.js"></script>
</head>
<body>
  <script>
    mermaid.initialize({
      startOnLoad: true,
      theme: document.body.classList.contains('vscode-dark') || document.body.classList.contains('vscode-high-contrast')
          ? 'dark'
          : 'default'
    });
  </script>
<h1 id="howard-help">HOWARD Help</h1>
<!--TOC-->
<ul>
<li><a href="#howard-help">HOWARD Help</a>
<ul>
<li><a href="#query-tool">QUERY tool</a>
<ul>
<li><a href="#main-options">Main options</a></li>
<li><a href="#explode">Explode</a></li>
<li><a href="#query">Query</a></li>
<li><a href="#export">Export</a></li>
</ul>
</li>
<li><a href="#stats-tool">STATS tool</a>
<ul>
<li><a href="#main-options-1">Main options</a></li>
<li><a href="#stats">Stats</a></li>
</ul>
</li>
<li><a href="#convert-tool">CONVERT tool</a>
<ul>
<li><a href="#main-options-2">Main options</a></li>
<li><a href="#explode-1">Explode</a></li>
<li><a href="#export-1">Export</a></li>
</ul>
</li>
<li><a href="#hgvs-tool">HGVS tool</a>
<ul>
<li><a href="#main-options-3">Main options</a></li>
<li><a href="#hgvs">HGVS</a></li>
</ul>
</li>
<li><a href="#annotation-tool">ANNOTATION tool</a>
<ul>
<li><a href="#main-options-4">Main options</a></li>
<li><a href="#annotation">Annotation</a></li>
</ul>
</li>
<li><a href="#calculation-tool">CALCULATION tool</a>
<ul>
<li><a href="#main-options-5">Main options</a></li>
<li><a href="#calculation">Calculation</a></li>
<li><a href="#nomen">NOMEN</a></li>
<li><a href="#trio">TRIO</a></li>
<li><a href="#barcodefamily">BARCODEFAMILY</a></li>
</ul>
</li>
<li><a href="#prioritization-tool">PRIORITIZATION tool</a>
<ul>
<li><a href="#main-options-6">Main options</a></li>
<li><a href="#prioritization">Prioritization</a></li>
</ul>
</li>
<li><a href="#process-tool">PROCESS tool</a>
<ul>
<li><a href="#main-options-7">Main options</a></li>
<li><a href="#hgvs-1">HGVS</a></li>
<li><a href="#annotation-1">Annotation</a></li>
<li><a href="#calculation-1">Calculation</a></li>
<li><a href="#prioritization-1">Prioritization</a></li>
<li><a href="#query-1">Query</a></li>
<li><a href="#explode-2">Explode</a></li>
<li><a href="#export-2">Export</a></li>
</ul>
</li>
<li><a href="#databases-tool">DATABASES tool</a>
<ul>
<li><a href="#main-options-8">Main options</a></li>
<li><a href="#genomes">Genomes</a></li>
<li><a href="#snpeff">snpEff</a></li>
<li><a href="#annovar">Annovar</a></li>
<li><a href="#refseq">refSeq</a></li>
<li><a href="#dbnsfp">dbNSFP</a></li>
<li><a href="#alphamissense">AlphaMissense</a></li>
<li><a href="#exomiser">Exomiser</a></li>
<li><a href="#dbsnp">dbSNP</a></li>
<li><a href="#hgmd">HGMD</a></li>
<li><a href="#from_annovar">from_Annovar</a></li>
<li><a href="#from_extann">from_extann</a></li>
<li><a href="#parameters">Parameters</a></li>
</ul>
</li>
<li><a href="#gui-tool">GUI tool</a></li>
<li><a href="#help-tool">HELP tool</a>
<ul>
<li><a href="#main-options-9">Main options</a></li>
</ul>
</li>
<li><a href="#genebe-tool">GENEBE tool</a>
<ul>
<li><a href="#main-options-10">Main options</a></li>
<li><a href="#genebe">GeneBe</a></li>
<li><a href="#explode-3">Explode</a></li>
<li><a href="#export-3">Export</a></li>
</ul>
</li>
<li><a href="#minimalize-tool">MINIMALIZE tool</a>
<ul>
<li><a href="#main-options-11">Main options</a></li>
<li><a href="#minimalize">Minimalize</a></li>
<li><a href="#explode-4">Explode</a></li>
<li><a href="#export-4">Export</a></li>
</ul>
</li>
<li><a href="#shared-arguments">Shared arguments</a></li>
</ul>
</li>
</ul>
<!--TOC-->
<p>HOWARD:0.10.0</p>
<p>Highly Open Workflow for Annotation &amp; Ranking toward genomic variant Discovery</p>
<p>HOWARD annotates and prioritizes genetic variations, calculates and normalizes annotations, convert on multiple formats, query variations and generates statisticsUsage examples:</p>
<blockquote>
<p>howard process --input=tests/data/example.vcf.gz --output=/tmp/example.annotated.vcf.gz --param=config/param.json</p>
</blockquote>
<blockquote>
<p>howard annotation --input=tests/data/example.vcf.gz --output=/tmp/example.howard.vcf.gz --annotations='tests/databases/annotations/current/hg19/dbnsfp42a.parquet,tests/databases/annotations/current/hg19/gnomad211_genome.parquet'</p>
</blockquote>
<blockquote>
<p>howard calculation --input=tests/data/example.full.vcf --output=/tmp/example.calculation.tsv --calculations='vartype'</p>
</blockquote>
<blockquote>
<p>howard prioritization --input=tests/data/example.vcf.gz --output=/tmp/example.prioritized.vcf.gz --prioritizations=config/prioritization_profiles.json --profiles='default,GERMLINE'</p>
</blockquote>
<blockquote>
<p>howard query --input=tests/data/example.vcf.gz --explode_infos --query='SELECT &quot;#CHROM&quot;, POS, REF, ALT, &quot;DP&quot;, &quot;CLNSIG&quot;, sample2, sample3 FROM variants WHERE &quot;DP&quot; &gt;= 50 OR &quot;CLNSIG&quot; NOT NULL ORDER BY &quot;CLNSIG&quot; DESC, &quot;DP&quot; DESC'</p>
</blockquote>
<blockquote>
<p>howard stats --input=tests/data/example.vcf.gz</p>
</blockquote>
<blockquote>
<p>howard convert --input=tests/data/example.vcf.gz --output=/tmp/example.tsv --explode_infos &amp;&amp; cat /tmp/example.tsv</p>
</blockquote>
<h2 id="query-tool">QUERY tool</h2>
<p>Query genetic variations in SQL format. Data can be loaded into 'variants' table from various formats (e.g. VCF, TSV, Parquet...). Using --explode_infos allow query on INFO/tag annotations. SQL query can also use external data within the request, such as a Parquet file(s).</p>
<p>Usage examples:</p>
<blockquote>
<p>howard query --input=tests/data/example.vcf.gz --query=&quot;SELECT * FROM variants WHERE REF = 'A' AND POS &lt; 100000&quot;</p>
</blockquote>
<blockquote>
<p>howard query --input=tests/data/example.vcf.gz --explode_infos --query='SELECT &quot;#CHROM&quot;, POS, REF, ALT, DP, CLNSIG, sample2, sample3 FROM variants WHERE DP &gt;= 50 OR CLNSIG NOT NULL ORDER BY DP DESC'</p>
</blockquote>
<blockquote>
<p>howard query --query=&quot;SELECT &quot;#CHROM&quot;, POS, REF, ALT, &quot;INFO/Interpro_domain&quot; FROM 'tests/databases/annotations/current/hg19/dbnsfp42a.parquet' WHERE &quot;INFO/Interpro_domain&quot; NOT NULL ORDER BY &quot;INFO/SiPhy_29way_logOdds_rankscore&quot; DESC LIMIT 10&quot;</p>
</blockquote>
<blockquote>
<p>howard query --explode_infos --explode_infos_prefix='INFO/' --query=&quot;SELECT &quot;#CHROM&quot;, POS, REF, ALT, STRING_AGG(INFO, ';') AS INFO FROM 'tests/databases/annotations/current/hg19/*.parquet' GROUP BY &quot;#CHROM&quot;, POS, REF, ALT&quot; --output=/tmp/full_annotation.tsv  &amp;&amp; head -n2 /tmp/full_annotation.tsv</p>
</blockquote>
<blockquote>
<p>howard query --input=tests/data/example.vcf.gz --param=config/param.json</p>
</blockquote>
<blockquote></blockquote>
<h3 id="main-options">Main options</h3>
<pre class="hljs"><code><div>--input=&lt;input&gt;

Input file path.
Format file must be either VCF, Parquet, TSV, CSV, PSV or duckDB.
Files can be compressesd (e.g. vcf.gz, tsv.gz).

</div></code></pre>
<pre class="hljs"><code><div>--output=&lt;output&gt;

Output file path.
Format file must be either VCF, Parquet, TSV, CSV, PSV or duckDB.
Files can be compressesd (e.g. vcf.gz, tsv.gz).

</div></code></pre>
<pre class="hljs"><code><div>--param=&lt;param&gt; (default: {})

Parameters JSON file (or string) defines parameters to process 
annotations, calculations, prioritizations, convertions and queries.

</div></code></pre>
<pre class="hljs"><code><div>--query=&lt;query&gt;

Query in SQL format
(e.g. 'SELECT * FROM variants LIMIT 50').

</div></code></pre>
<h3 id="explode">Explode</h3>
<pre class="hljs"><code><div>--explode_infos

Explode VCF INFO/Tag into 'variants' table columns.

</div></code></pre>
<pre class="hljs"><code><div>--explode_infos_prefix=&lt;explode infos prefix&gt;

Explode VCF INFO/Tag with a specific prefix.

</div></code></pre>
<pre class="hljs"><code><div>--explode_infos_fields=&lt;explode infos list&gt; (default: *)

Explode VCF INFO/Tag specific fields/tags.
Keyword `*` specify all available fields, except those already specified.
Pattern (regex) can be used, such as `.*_score` for fields named with '_score' at the end.
Examples:
- 'HGVS,SIFT,Clinvar' (list of fields)
- 'HGVS,*,Clinvar' (list of fields with all other fields at the end)
- 'HGVS,.*_score,Clinvar' (list of 2 fields with all scores in the middle)
- 'HGVS,.*_score,*' (1 field, scores, all other fields)
- 'HGVS,*,.*_score' (1 field, all other fields, all scores)

</div></code></pre>
<h3 id="query">Query</h3>
<pre class="hljs"><code><div>--query_limit=&lt;query limit&gt; (default: 10)

Limit of number of row for query (only for print result, not output).

</div></code></pre>
<pre class="hljs"><code><div>--query_print_mode=&lt;print mode&gt; [None, 'markdown', 'tabulate']

Print mode of query result (only for print result, not output).
Either None (native), 'markdown' or 'tabulate'.

</div></code></pre>
<h3 id="export">Export</h3>
<pre class="hljs"><code><div>--include_header

Include header (in VCF format) in output file.
Only for compatible formats (tab-delimiter format as TSV or BED).

</div></code></pre>
<pre class="hljs"><code><div>--parquet_partitions=&lt;parquet partitions&gt;

Parquet partitioning using hive (available for any format).
This option is faster parallel writing, but memory consuming.
Use 'None' (string) for NO partition but split parquet files into a folder.
Examples: '#CHROM', '#CHROM,REF', 'None'.

</div></code></pre>
<h2 id="stats-tool">STATS tool</h2>
<p>Statistics on genetic variations, such as: number of variants, number of samples, statistics by chromosome, genotypes by samples...</p>
<p>Usage examples:</p>
<blockquote>
<p>howard stats --input=tests/data/example.vcf.gz</p>
</blockquote>
<blockquote>
<p>howard stats --input=tests/data/example.vcf.gz --stats_md=/tmp/stats.md</p>
</blockquote>
<blockquote>
<p>howard stats --input=tests/data/example.vcf.gz --param=config/param.json</p>
</blockquote>
<blockquote></blockquote>
<h3 id="main-options">Main options</h3>
<pre class="hljs"><code><div>--input=&lt;input&gt; | required

Input file path.
Format file must be either VCF, Parquet, TSV, CSV, PSV or duckDB.
Files can be compressesd (e.g. vcf.gz, tsv.gz).

</div></code></pre>
<pre class="hljs"><code><div>--param=&lt;param&gt; (default: {})

Parameters JSON file (or string) defines parameters to process 
annotations, calculations, prioritizations, convertions and queries.

</div></code></pre>
<h3 id="stats">Stats</h3>
<pre class="hljs"><code><div>--stats_md=&lt;stats markdown&gt;

Stats Output file in MarkDown format.

</div></code></pre>
<pre class="hljs"><code><div>--stats_json=&lt;stats json&gt;

Stats Output file in JSON format.

</div></code></pre>
<h2 id="convert-tool">CONVERT tool</h2>
<p>Convert genetic variations file to another format. Multiple format are available, such as usual and official VCF and BCF format, but also other formats such as TSV, CSV, PSV and Parquet/duckDB. These formats need a header '.hdr' file to take advantage of the power of howard (especially through INFO/tag definition), and using howard convert tool automatically generate header file fo futher use.</p>
<p>Usage examples:</p>
<blockquote>
<p>howard convert --input=tests/data/example.vcf.gz --output=/tmp/example.tsv</p>
</blockquote>
<blockquote>
<p>howard convert --input=tests/data/example.vcf.gz --output=/tmp/example.parquet</p>
</blockquote>
<blockquote>
<p>howard convert --input=tests/data/example.vcf.gz --output=/tmp/example.tsv --explode_infos --explode_infos_fields='CLNSIG,SIFT,DP' --order_by='CLNSIG DESC, DP DESC'</p>
</blockquote>
<blockquote>
<p>howard convert --input=tests/data/example.vcf.gz --output=/tmp/example.tsv --explode_infos --explode_infos_prefix='INFO/' --explode_infos_fields='CLNSIG,SIFT,DP,*' --order_by='&quot;INFO/CLNSIG&quot; DESC, &quot;INFO/DP&quot; DESC' --include_header</p>
</blockquote>
<blockquote>
<p>howard convert --input=tests/data/example.vcf.gz --output=/tmp/example.tsv --param=config/param.json</p>
</blockquote>
<blockquote></blockquote>
<h3 id="main-options">Main options</h3>
<pre class="hljs"><code><div>--input=&lt;input&gt; | required

Input file path.
Format file must be either VCF, Parquet, TSV, CSV, PSV or duckDB.
Files can be compressesd (e.g. vcf.gz, tsv.gz).

</div></code></pre>
<pre class="hljs"><code><div>--output=&lt;output&gt; | required

Output file path.
Format file must be either VCF, Parquet, TSV, CSV, PSV or duckDB.
Files can be compressesd (e.g. vcf.gz, tsv.gz).

</div></code></pre>
<pre class="hljs"><code><div>--param=&lt;param&gt; (default: {})

Parameters JSON file (or string) defines parameters to process 
annotations, calculations, prioritizations, convertions and queries.

</div></code></pre>
<h3 id="explode">Explode</h3>
<pre class="hljs"><code><div>--explode_infos

Explode VCF INFO/Tag into 'variants' table columns.

</div></code></pre>
<pre class="hljs"><code><div>--explode_infos_prefix=&lt;explode infos prefix&gt;

Explode VCF INFO/Tag with a specific prefix.

</div></code></pre>
<pre class="hljs"><code><div>--explode_infos_fields=&lt;explode infos list&gt; (default: *)

Explode VCF INFO/Tag specific fields/tags.
Keyword `*` specify all available fields, except those already specified.
Pattern (regex) can be used, such as `.*_score` for fields named with '_score' at the end.
Examples:
- 'HGVS,SIFT,Clinvar' (list of fields)
- 'HGVS,*,Clinvar' (list of fields with all other fields at the end)
- 'HGVS,.*_score,Clinvar' (list of 2 fields with all scores in the middle)
- 'HGVS,.*_score,*' (1 field, scores, all other fields)
- 'HGVS,*,.*_score' (1 field, all other fields, all scores)

</div></code></pre>
<h3 id="export">Export</h3>
<pre class="hljs"><code><div>--include_header

Include header (in VCF format) in output file.
Only for compatible formats (tab-delimiter format as TSV or BED).

</div></code></pre>
<pre class="hljs"><code><div>--order_by=&lt;order by&gt;

List of columns to sort the result-set in ascending or descending order.
Use SQL format, and keywords ASC (ascending) and DESC (descending).
If a column is not available, order will not be considered.
Order is enable only for compatible format (e.g. TSV, CSV, JSON).
Examples: 'ACMG_score DESC', 'PZFlag DESC, PZScore DESC'.

</div></code></pre>
<pre class="hljs"><code><div>--parquet_partitions=&lt;parquet partitions&gt;

Parquet partitioning using hive (available for any format).
This option is faster parallel writing, but memory consuming.
Use 'None' (string) for NO partition but split parquet files into a folder.
Examples: '#CHROM', '#CHROM,REF', 'None'.

</div></code></pre>
<h2 id="hgvs-tool">HGVS tool</h2>
<p>HGVS annotation using HUGO HGVS internation Sequence Variant Nomenclature (http://varnomen.hgvs.org/). Annotation refere to refGene and genome to generate HGVS nomenclature for all available transcripts. This annotation add 'hgvs' field into VCF INFO column of a VCF file.</p>
<p>Usage examples:</p>
<blockquote>
<p>howard hgvs --input=tests/data/example.full.vcf --output=/tmp/example.hgvs.vcf</p>
</blockquote>
<blockquote>
<p>howard hgvs --input=tests/data/example.full.vcf --output=/tmp/example.hgvs.tsv --param=config/param.json</p>
</blockquote>
<blockquote>
<p>howard hgvs --input=tests/data/example.full.vcf --output=/tmp/example.hgvs.vcf --full_format --use_exon</p>
</blockquote>
<blockquote></blockquote>
<h3 id="main-options">Main options</h3>
<pre class="hljs"><code><div>--input=&lt;input&gt; | required

Input file path.
Format file must be either VCF, Parquet, TSV, CSV, PSV or duckDB.
Files can be compressesd (e.g. vcf.gz, tsv.gz).

</div></code></pre>
<pre class="hljs"><code><div>--output=&lt;output&gt; | required

Output file path.
Format file must be either VCF, Parquet, TSV, CSV, PSV or duckDB.
Files can be compressesd (e.g. vcf.gz, tsv.gz).

</div></code></pre>
<pre class="hljs"><code><div>--param=&lt;param&gt; (default: {})

Parameters JSON file (or string) defines parameters to process 
annotations, calculations, prioritizations, convertions and queries.

</div></code></pre>
<pre class="hljs"><code><div>--hgvs_options=&lt;HGVS options&gt;

Quick HGVS annotation options.
This option will skip all other hgvs options.
Examples:
- 'default' (for default options)
- 'full_format' (for full format HGVS annotation)
- 'use_gene=True:add_protein=true:codon_type=FULL'

</div></code></pre>
<pre class="hljs"><code><div>--assembly=&lt;assembly&gt; (default: hg19)

Genome Assembly (e.g. 'hg19', 'hg38').

</div></code></pre>
<h3 id="hgvs">HGVS</h3>
<pre class="hljs"><code><div>--use_gene

Use Gene information to generate HGVS annotation
(e.g. 'NM_152232(TAS1R2):c.231T&gt;C')
</div></code></pre>
<pre class="hljs"><code><div>--use_exon

Use Exon information to generate HGVS annotation
(e.g. 'NM_152232(exon2):c.231T&gt;C').
Only if 'use_gene' is not enabled.

</div></code></pre>
<pre class="hljs"><code><div>--use_protein

Use Protein level to generate HGVS annotation
(e.g. 'NP_689418:p.Cys77Arg').
Can be used with 'use_exon' or 'use_gene'.

</div></code></pre>
<pre class="hljs"><code><div>--add_protein

Add Protein level to DNA HGVS annotation (e.g 'NM_152232:c.231T&gt;C,NP_689418:p.Cys77Arg').

</div></code></pre>
<pre class="hljs"><code><div>--full_format

Generates HGVS annotation in a full format
by using all information to generates an exhaustive annotation
(non-standard, e.g. 'TAS1R2:NM_152232:NP_689418:c.231T&gt;C:p.Cys77Arg').
Use 'use_exon' to add exon information
(e.g 'TAS1R2:NM_152232:NP_689418:exon2:c.231T&gt;C:p.Cys77Arg').

</div></code></pre>
<pre class="hljs"><code><div>--codon_type=&lt;Codon type&gt; ['1', '3', 'FULL'] (default: 3)

Amino Acide Codon format type to use to generate HGVS annotation.
Available:
- '1': codon in 1 character (e.g. 'C', 'R')
- '3': codon in 3 character (e.g. 'Cys', 'Arg')
-'FULL': codon in full name (e.g. 'Cysteine', 'Arginine')

</div></code></pre>
<pre class="hljs"><code><div>--refgene=&lt;refGene&gt;

Path to refGene annotation file.

</div></code></pre>
<pre class="hljs"><code><div>--refseqlink=&lt;refSeqLink&gt;

Path to refSeqLink annotation file.

</div></code></pre>
<h2 id="annotation-tool">ANNOTATION tool</h2>
<p>Annotation is mainly based on a build-in Parquet annotation method, and tools such as BCFTOOLS, Annovar and snpEff. It uses available databases (see Annovar and snpEff) and homemade databases. Format of databases are: parquet, duckdb, vcf, bed, Annovar and snpEff (Annovar and snpEff databases are automatically downloaded, see howard databases tool).</p>
<p>Usage examples:</p>
<blockquote>
<p>howard annotation --input=tests/data/example.vcf.gz --output=/tmp/example.howard.vcf.gz --annotations='tests/databases/annotations/current/hg19/avsnp150.parquet,tests/databases/annotations/current/hg19/dbnsfp42a.parquet,tests/databases/annotations/current/hg19/gnomad211_genome.parquet'</p>
</blockquote>
<blockquote>
<p>howard annotation --input=tests/data/example.vcf.gz --output=/tmp/example.howard.tsv --assembly=hg19 --annotations='annovar:refGene,annovar:cosmic70,snpeff,tests/databases/annotations/current/hg19/clinvar_20210123.parquet'</p>
</blockquote>
<blockquote>
<p>howard annotation --input=tests/data/example.vcf.gz --output=/tmp/example.howard.tsv --assembly=hg19 --annotations='ALL:parquet'</p>
</blockquote>
<blockquote>
<p>howard annotation --input=tests/data/example.vcf.gz --output=/tmp/example.howard.tsv --param=config/param.json</p>
</blockquote>
<blockquote></blockquote>
<h3 id="main-options">Main options</h3>
<pre class="hljs"><code><div>--input=&lt;input&gt; | required

Input file path.
Format file must be either VCF, Parquet, TSV, CSV, PSV or duckDB.
Files can be compressesd (e.g. vcf.gz, tsv.gz).

</div></code></pre>
<pre class="hljs"><code><div>--output=&lt;output&gt; | required

Output file path.
Format file must be either VCF, Parquet, TSV, CSV, PSV or duckDB.
Files can be compressesd (e.g. vcf.gz, tsv.gz).

</div></code></pre>
<pre class="hljs"><code><div>--param=&lt;param&gt; (default: {})

Parameters JSON file (or string) defines parameters to process 
annotations, calculations, prioritizations, convertions and queries.

</div></code></pre>
<pre class="hljs"><code><div>--annotations=&lt;annotations&gt;

Annotation with databases files, or with tools,
as a list of files in Parquet, VCF, BED, or keywords
 (e.g. 'file.parquet,bcftools:file2.vcf.gz,annovar:refGene,snpeff').
- For a Parquet/VCF/BED, use file paths
 (e.g. 'file1.parquet,file2.vcf.gz').
- For BCFTools annotation, use keyword 'bcftools' with file paths
 (e.g. 'bcftools:file.vcf.gz:file.bed.gz').
- For Annovar annotation, use keyword 'annovar' with annovar code
 (e.g. 'annovar:refGene', 'annovar:cosmic70').
- For snpeff annotation, use keyword 'snpeff'.
- For Exomiser annotation, use keyword 'exomiser' with options as key=value
 (e.g. 'exomiser:preset=exome:transcript_source=refseq').
- For add all availalbe databases files, use 'ALL' keyword,
 with filters on type and release
 (e.g. 'ALL', 'ALL:parquet:current', 'ALL:parquet,vcf:current,devel').

</div></code></pre>
<pre class="hljs"><code><div>--assembly=&lt;assembly&gt; (default: hg19)

Genome Assembly (e.g. 'hg19', 'hg38').

</div></code></pre>
<h3 id="annotation">Annotation</h3>
<pre class="hljs"><code><div>--annotations_update

Update option for annotation (Only for Parquet annotation).
If True, annotation fields will be removed and re-annotated.
These options will be applied to all annotation databases.

</div></code></pre>
<pre class="hljs"><code><div>--annotations_append

Append option for annotation (Only for Parquet annotation).
If True, annotation fields will be annotated only if not annotation exists for the variant.
These options will be applied to all annotation databases.

</div></code></pre>
<h2 id="calculation-tool">CALCULATION tool</h2>
<p>Calculation processes variants information to generate new information, such as: identify variation type (VarType), harmonizes allele frequency (VAF) and calculate sttistics (VAF_stats), extracts Nomen (transcript, cNomen, pNomen...) from an HGVS field (e.g. snpEff, Annovar) with an optional list of personalized transcripts, generates VaRank format barcode, identify trio inheritance.</p>
<p>Usage examples:</p>
<blockquote>
<p>howard calculation --input=tests/data/example.full.vcf --output=/tmp/example.calculation.tsv --calculations='vartype'</p>
</blockquote>
<blockquote>
<p>howard calculation --input=tests/data/example.ann.vcf.gz --output=/tmp/example.calculated.tsv --calculations='snpeff_hgvs,NOMEN' --hgvs_field=snpeff_hgvs --transcripts=tests/data/transcripts.tsv</p>
</blockquote>
<blockquote>
<p>howard calculation --input=tests/data/example.vcf.gz --output=/tmp/example.calculated.tsv --calculations='TRIO' --trio_pedigree='sample1,sample2,sample4'</p>
</blockquote>
<blockquote>
<p>howard calculation --input=tests/data/example.vcf.gz --output=/tmp/example.calculated.tsv --calculations='BARCODEFAMILY' --family_pedigree='sample1,sample2,sample4'</p>
</blockquote>
<blockquote>
<p>howard calculation --input=tests/data/example.ann.vcf.gz --output=/tmp/example.ann.tsv --param=config/param.json</p>
</blockquote>
<blockquote>
<p>howard calculation --show_calculations</p>
</blockquote>
<blockquote></blockquote>
<h3 id="main-options">Main options</h3>
<pre class="hljs"><code><div>--input=&lt;input&gt;

Input file path.
Format file must be either VCF, Parquet, TSV, CSV, PSV or duckDB.
Files can be compressesd (e.g. vcf.gz, tsv.gz).

</div></code></pre>
<pre class="hljs"><code><div>--output=&lt;output&gt;

Output file path.
Format file must be either VCF, Parquet, TSV, CSV, PSV or duckDB.
Files can be compressesd (e.g. vcf.gz, tsv.gz).

</div></code></pre>
<pre class="hljs"><code><div>--param=&lt;param&gt; (default: {})

Parameters JSON file (or string) defines parameters to process 
annotations, calculations, prioritizations, convertions and queries.

</div></code></pre>
<pre class="hljs"><code><div>--calculations=&lt;operations&gt;

Quick calculations on genetic variants information and genotype information,
as a list of operations (e.g. 'VARTYPE,variant_id').
List of available calculations by default
 (unsensitive case, see doc for more information):
 VARTYPE  snpeff_hgvs  FINDBYPIPELINE  GENOTYPECONCORDANCE  BARCODE  TRIO  VAF  VAF_STATS  DP_STATS 

</div></code></pre>
<h3 id="calculation">Calculation</h3>
<pre class="hljs"><code><div>--calculation_config=&lt;calculation config&gt;

Calculation configuration JSON file.

</div></code></pre>
<pre class="hljs"><code><div>--show_calculations

Show available calculation operations.

</div></code></pre>
<h3 id="nomen">NOMEN</h3>
<pre class="hljs"><code><div>--hgvs_field=&lt;HGVS field&gt; (default: hgvs)

HGVS INFO/tag containing a list o HGVS annotations.

</div></code></pre>
<pre class="hljs"><code><div>--transcripts=&lt;transcripts&gt;

Transcripts TSV file,
with Transcript in first column, optional Gene in second column.

</div></code></pre>
<h3 id="trio">TRIO</h3>
<pre class="hljs"><code><div>--trio_pedigree=&lt;trio pedigree&gt;

Pedigree Trio for trio inheritance calculation.
Either a JSON file or JSON string or a list of samples
(e.g. 'sample1,sample2,sample3' for father, mother and child,
 '{&quot;father&quot;: &quot;sample1&quot;, &quot;mother&quot;: &quot;sample2&quot;, &quot;child&quot;: &quot;sample3&quot;}').

</div></code></pre>
<h3 id="barcodefamily">BARCODEFAMILY</h3>
<pre class="hljs"><code><div>--family_pedigree=&lt;family pedigree&gt;

Pedigree family for barcode calculation on genotype.
Either a JSON file or JSON string or a list of samples
(e.g. 'sample1,sample2,sample3,sample4',
 '{&quot;father&quot;: &quot;sample1&quot;, &quot;mother&quot;: &quot;sample2&quot;, &quot;child1&quot;: &quot;sample3&quot;, &quot;child2&quot;: &quot;sample3&quot;}').

</div></code></pre>
<h2 id="prioritization-tool">PRIORITIZATION tool</h2>
<p>Prioritization algorithm uses profiles to flag variants (as passed or filtered), calculate a prioritization score, and automatically generate a comment for each variants (example: 'polymorphism identified in dbSNP. associated to Lung Cancer. Found in ClinVar database'). Prioritization profiles are defined in a configuration file in JSON format. A profile is defined as a list of annotation/value, using wildcards and comparison options (contains, lower than, greater than, equal...). Annotations fields may be quality values (usually from callers, such as 'DP') or other annotations fields provided by annotations tools, such as HOWARD itself (example: COSMIC, Clinvar, 1000genomes, PolyPhen, SIFT). Multiple profiles can be used simultaneously, which is useful to define multiple validation/prioritization levels (example: 'standard', 'stringent', 'rare variants', 'low allele frequency').</p>
<p>Usage examples:</p>
<blockquote>
<p>howard prioritization --input=tests/data/example.vcf.gz --output=/tmp/example.prioritized.vcf.gz --prioritizations='default'</p>
</blockquote>
<blockquote>
<p>howard prioritization --input=tests/data/example.vcf.gz --output=/tmp/example.prioritized.vcf.gz --prioritizations='default,GERMLINE' --prioritization_config=config/prioritization_profiles.json</p>
</blockquote>
<blockquote>
<p>howard prioritization --input=tests/data/example.vcf.gz --output=/tmp/example.prioritized.tsv --param=config/param.json</p>
</blockquote>
<blockquote></blockquote>
<h3 id="main-options">Main options</h3>
<pre class="hljs"><code><div>--input=&lt;input&gt; | required

Input file path.
Format file must be either VCF, Parquet, TSV, CSV, PSV or duckDB.
Files can be compressesd (e.g. vcf.gz, tsv.gz).

</div></code></pre>
<pre class="hljs"><code><div>--output=&lt;output&gt; | required

Output file path.
Format file must be either VCF, Parquet, TSV, CSV, PSV or duckDB.
Files can be compressesd (e.g. vcf.gz, tsv.gz).

</div></code></pre>
<pre class="hljs"><code><div>--param=&lt;param&gt; (default: {})

Parameters JSON file (or string) defines parameters to process 
annotations, calculations, prioritizations, convertions and queries.

</div></code></pre>
<pre class="hljs"><code><div>--prioritizations=&lt;prioritisations&gt;

List of prioritization profiles to process (based on Prioritization JSON file),
such as 'default', 'rare variants', 'low allele frequency', 'GERMLINE'.
By default, all profiles available will be processed.

</div></code></pre>
<h3 id="prioritization">Prioritization</h3>
<pre class="hljs"><code><div>--default_profile=&lt;default profile&gt;

Prioritization profile by default (see doc).
Default is the first profile in the list of prioritization profiles.

</div></code></pre>
<pre class="hljs"><code><div>--pzfields=&lt;pzfields&gt; (default: PZScore,PZFlag)

Prioritization fields to provide (see doc).
Available: PZScore, PZFlag, PZTags, PZComment, PZInfos

</div></code></pre>
<pre class="hljs"><code><div>--prioritization_score_mode=&lt;prioritization score mode&gt; ['HOWARD', 'VaRank'] (default: HOWARD)

Prioritization Score mode (see doc).
Available: HOWARD (increment score), VaRank (max score)

</div></code></pre>
<pre class="hljs"><code><div>--prioritization_config=&lt;prioritization config&gt;

Prioritization configuration JSON file (defines profiles, see doc).

</div></code></pre>
<h2 id="process-tool">PROCESS tool</h2>
<p>howard process tool manage genetic variations to:</p>
<ul>
<li>
<p>annotates genetic variants with multiple annotation databases/files and tools</p>
</li>
<li>
<p>calculates and normalizes annotations</p>
</li>
<li>
<p>prioritizes variants with profiles (list of citeria) to calculate scores and flags</p>
</li>
<li>
<p>translates into various formats</p>
</li>
<li>
<p>query genetic variants and annotations</p>
</li>
<li>
<p>generates variants statistics</p>
</li>
</ul>
<p>Usage examples:</p>
<blockquote>
<p>howard process --input=tests/data/example.vcf.gz --output=/tmp/example.annotated.vcf.gz --param=config/param.json</p>
</blockquote>
<blockquote>
<p>howard process --input=tests/data/example.vcf.gz --annotations='snpeff' --calculations='snpeff_hgvs' --prioritizations='default' --explode_infos --output=/tmp/example.annotated.tsv --query='SELECT &quot;#CHROM&quot;, POS, ALT, REF, snpeff_hgvs FROM variants'</p>
</blockquote>
<blockquote>
<p>howard process --input=tests/data/example.vcf.gz --hgvs_options='full_format,use_exon' --explode_infos --output=/tmp/example.annotated.tsv --query='SELECT &quot;#CHROM&quot;, POS, ALT, REF, hgvs FROM variants'</p>
</blockquote>
<blockquote>
<p>howard process --input=tests/data/example.vcf.gz --output=/tmp/example.howard.vcf.gz --hgvs='full_format,use_exon' --annotations='tests/databases/annotations/current/hg19/avsnp150.parquet,tests/databases/annotations/current/hg19/dbnsfp42a.parquet,tests/databases/annotations/current/hg19/gnomad211_genome.parquet' --calculations='NOMEN' --explode_infos --query='SELECT NOMEN, REVEL_score, SIFT_score, AF AS 'gnomad_AF', ClinPred_score, ClinPred_pred FROM variants'</p>
</blockquote>
<blockquote></blockquote>
<h3 id="main-options">Main options</h3>
<pre class="hljs"><code><div>--input=&lt;input&gt; | required

Input file path.
Format file must be either VCF, Parquet, TSV, CSV, PSV or duckDB.
Files can be compressesd (e.g. vcf.gz, tsv.gz).

</div></code></pre>
<pre class="hljs"><code><div>--output=&lt;output&gt; | required

Output file path.
Format file must be either VCF, Parquet, TSV, CSV, PSV or duckDB.
Files can be compressesd (e.g. vcf.gz, tsv.gz).

</div></code></pre>
<pre class="hljs"><code><div>--param=&lt;param&gt; (default: {})

Parameters JSON file (or string) defines parameters to process 
annotations, calculations, prioritizations, convertions and queries.

</div></code></pre>
<pre class="hljs"><code><div>--hgvs_options=&lt;HGVS options&gt;

Quick HGVS annotation options.
This option will skip all other hgvs options.
Examples:
- 'default' (for default options)
- 'full_format' (for full format HGVS annotation)
- 'use_gene=True:add_protein=true:codon_type=FULL'

</div></code></pre>
<pre class="hljs"><code><div>--annotations=&lt;annotations&gt;

Annotation with databases files, or with tools,
as a list of files in Parquet, VCF, BED, or keywords
 (e.g. 'file.parquet,bcftools:file2.vcf.gz,annovar:refGene,snpeff').
- For a Parquet/VCF/BED, use file paths
 (e.g. 'file1.parquet,file2.vcf.gz').
- For BCFTools annotation, use keyword 'bcftools' with file paths
 (e.g. 'bcftools:file.vcf.gz:file.bed.gz').
- For Annovar annotation, use keyword 'annovar' with annovar code
 (e.g. 'annovar:refGene', 'annovar:cosmic70').
- For snpeff annotation, use keyword 'snpeff'.
- For Exomiser annotation, use keyword 'exomiser' with options as key=value
 (e.g. 'exomiser:preset=exome:transcript_source=refseq').
- For add all availalbe databases files, use 'ALL' keyword,
 with filters on type and release
 (e.g. 'ALL', 'ALL:parquet:current', 'ALL:parquet,vcf:current,devel').

</div></code></pre>
<pre class="hljs"><code><div>--calculations=&lt;operations&gt;

Quick calculations on genetic variants information and genotype information,
as a list of operations (e.g. 'VARTYPE,variant_id').
List of available calculations by default
 (unsensitive case, see doc for more information):
 VARTYPE  snpeff_hgvs  FINDBYPIPELINE  GENOTYPECONCORDANCE  BARCODE  TRIO  VAF  VAF_STATS  DP_STATS 

</div></code></pre>
<pre class="hljs"><code><div>--prioritizations=&lt;prioritisations&gt;

List of prioritization profiles to process (based on Prioritization JSON file),
such as 'default', 'rare variants', 'low allele frequency', 'GERMLINE'.
By default, all profiles available will be processed.

</div></code></pre>
<pre class="hljs"><code><div>--assembly=&lt;assembly&gt; (default: hg19)

Genome Assembly (e.g. 'hg19', 'hg38').

</div></code></pre>
<h3 id="hgvs">HGVS</h3>
<pre class="hljs"><code><div>--use_gene

Use Gene information to generate HGVS annotation
(e.g. 'NM_152232(TAS1R2):c.231T&gt;C')
</div></code></pre>
<pre class="hljs"><code><div>--use_exon

Use Exon information to generate HGVS annotation
(e.g. 'NM_152232(exon2):c.231T&gt;C').
Only if 'use_gene' is not enabled.

</div></code></pre>
<pre class="hljs"><code><div>--use_protein

Use Protein level to generate HGVS annotation
(e.g. 'NP_689418:p.Cys77Arg').
Can be used with 'use_exon' or 'use_gene'.

</div></code></pre>
<pre class="hljs"><code><div>--add_protein

Add Protein level to DNA HGVS annotation (e.g 'NM_152232:c.231T&gt;C,NP_689418:p.Cys77Arg').

</div></code></pre>
<pre class="hljs"><code><div>--full_format

Generates HGVS annotation in a full format
by using all information to generates an exhaustive annotation
(non-standard, e.g. 'TAS1R2:NM_152232:NP_689418:c.231T&gt;C:p.Cys77Arg').
Use 'use_exon' to add exon information
(e.g 'TAS1R2:NM_152232:NP_689418:exon2:c.231T&gt;C:p.Cys77Arg').

</div></code></pre>
<pre class="hljs"><code><div>--codon_type=&lt;Codon type&gt; ['1', '3', 'FULL'] (default: 3)

Amino Acide Codon format type to use to generate HGVS annotation.
Available:
- '1': codon in 1 character (e.g. 'C', 'R')
- '3': codon in 3 character (e.g. 'Cys', 'Arg')
-'FULL': codon in full name (e.g. 'Cysteine', 'Arginine')

</div></code></pre>
<pre class="hljs"><code><div>--refgene=&lt;refGene&gt;

Path to refGene annotation file.

</div></code></pre>
<pre class="hljs"><code><div>--refseqlink=&lt;refSeqLink&gt;

Path to refSeqLink annotation file.

</div></code></pre>
<h3 id="annotation">Annotation</h3>
<pre class="hljs"><code><div>--annotations_update

Update option for annotation (Only for Parquet annotation).
If True, annotation fields will be removed and re-annotated.
These options will be applied to all annotation databases.

</div></code></pre>
<pre class="hljs"><code><div>--annotations_append

Append option for annotation (Only for Parquet annotation).
If True, annotation fields will be annotated only if not annotation exists for the variant.
These options will be applied to all annotation databases.

</div></code></pre>
<h3 id="calculation">Calculation</h3>
<pre class="hljs"><code><div>--calculation_config=&lt;calculation config&gt;

Calculation configuration JSON file.

</div></code></pre>
<h3 id="prioritization">Prioritization</h3>
<pre class="hljs"><code><div>--default_profile=&lt;default profile&gt;

Prioritization profile by default (see doc).
Default is the first profile in the list of prioritization profiles.

</div></code></pre>
<pre class="hljs"><code><div>--pzfields=&lt;pzfields&gt; (default: PZScore,PZFlag)

Prioritization fields to provide (see doc).
Available: PZScore, PZFlag, PZTags, PZComment, PZInfos

</div></code></pre>
<pre class="hljs"><code><div>--prioritization_score_mode=&lt;prioritization score mode&gt; ['HOWARD', 'VaRank'] (default: HOWARD)

Prioritization Score mode (see doc).
Available: HOWARD (increment score), VaRank (max score)

</div></code></pre>
<pre class="hljs"><code><div>--prioritization_config=&lt;prioritization config&gt;

Prioritization configuration JSON file (defines profiles, see doc).

</div></code></pre>
<h3 id="query">Query</h3>
<pre class="hljs"><code><div>--query=&lt;query&gt;

Query in SQL format
(e.g. 'SELECT * FROM variants LIMIT 50').

</div></code></pre>
<pre class="hljs"><code><div>--query_limit=&lt;query limit&gt; (default: 10)

Limit of number of row for query (only for print result, not output).

</div></code></pre>
<pre class="hljs"><code><div>--query_print_mode=&lt;print mode&gt; [None, 'markdown', 'tabulate']

Print mode of query result (only for print result, not output).
Either None (native), 'markdown' or 'tabulate'.

</div></code></pre>
<h3 id="explode">Explode</h3>
<pre class="hljs"><code><div>--explode_infos

Explode VCF INFO/Tag into 'variants' table columns.

</div></code></pre>
<pre class="hljs"><code><div>--explode_infos_prefix=&lt;explode infos prefix&gt;

Explode VCF INFO/Tag with a specific prefix.

</div></code></pre>
<pre class="hljs"><code><div>--explode_infos_fields=&lt;explode infos list&gt; (default: *)

Explode VCF INFO/Tag specific fields/tags.
Keyword `*` specify all available fields, except those already specified.
Pattern (regex) can be used, such as `.*_score` for fields named with '_score' at the end.
Examples:
- 'HGVS,SIFT,Clinvar' (list of fields)
- 'HGVS,*,Clinvar' (list of fields with all other fields at the end)
- 'HGVS,.*_score,Clinvar' (list of 2 fields with all scores in the middle)
- 'HGVS,.*_score,*' (1 field, scores, all other fields)
- 'HGVS,*,.*_score' (1 field, all other fields, all scores)

</div></code></pre>
<h3 id="export">Export</h3>
<pre class="hljs"><code><div>--include_header

Include header (in VCF format) in output file.
Only for compatible formats (tab-delimiter format as TSV or BED).

</div></code></pre>
<pre class="hljs"><code><div>--order_by=&lt;order by&gt;

List of columns to sort the result-set in ascending or descending order.
Use SQL format, and keywords ASC (ascending) and DESC (descending).
If a column is not available, order will not be considered.
Order is enable only for compatible format (e.g. TSV, CSV, JSON).
Examples: 'ACMG_score DESC', 'PZFlag DESC, PZScore DESC'.

</div></code></pre>
<pre class="hljs"><code><div>--parquet_partitions=&lt;parquet partitions&gt;

Parquet partitioning using hive (available for any format).
This option is faster parallel writing, but memory consuming.
Use 'None' (string) for NO partition but split parquet files into a folder.
Examples: '#CHROM', '#CHROM,REF', 'None'.

</div></code></pre>
<h2 id="databases-tool">DATABASES tool</h2>
<p>Download databases and needed files for howard and associated tools</p>
<p>Usage examples:</p>
<blockquote>
<p>howard databases --assembly=hg19 --download-genomes=~/howard/databases/genomes/current --download-genomes-provider=UCSC --download-genomes-contig-regex='chr[0-9XYM]+$'</p>
</blockquote>
<blockquote>
<p>howard databases --assembly=hg19 --download-annovar=~/howard/databases/annovar/current --download-annovar-files='refGene,cosmic70,nci60'</p>
</blockquote>
<blockquote>
<p>howard databases --assembly=hg19 --download-snpeff=~/howard/databases/snpeff/current</p>
</blockquote>
<blockquote>
<p>howard databases --assembly=hg19 --download-refseq=~/howard/databases/refseq/current --download-refseq-format-file='ncbiRefSeq.txt'</p>
</blockquote>
<blockquote>
<p>howard databases --assembly=hg19 --download-dbnsfp=~/howard/databases/dbnsfp/current --download-dbnsfp-release='4.4a' --download-dbnsfp-subdatabases</p>
</blockquote>
<blockquote>
<p>howard databases --assembly=hg19 --download-alphamissense=~/howard/databases/alphamissense/current</p>
</blockquote>
<blockquote>
<p>howard databases --assembly=hg19 --download-exomiser=~/howard/databases/exomiser/current</p>
</blockquote>
<blockquote>
<p>howard databases --assembly=hg19 --download-dbsnp=~/howard/databases/dbsnp/current --download-dbsnp-vcf</p>
</blockquote>
<blockquote>
<p>cd ~/howard/databases &amp;&amp; howard databases --assembly=hg19 --download-genomes=genomes/current --download-genomes-provider=UCSC --download-genomes-contig-regex='chr[0-9XYM]+$' --download-annovar=annovar/current --download-annovar-files='refGene,cosmic70,nci60' --download-snpeff=snpeff/current --download-refseq=refseq/current --download-refseq-format-file='ncbiRefSeq.txt' --download-dbnsfp=dbnsfp/current --download-dbnsfp-release='4.4a' --download-dbnsfp-subdatabases --download-alphamissense=alphamissense/current --download-exomiser=exomiser/current --download-dbsnp=dbsnp/current --download-dbsnp-vcf --threads=8</p>
</blockquote>
<blockquote>
<p>howard databases --generate-param=/tmp/param.json --generate-param-description=/tmp/test.description.json --generate-param-formats=parquet</p>
</blockquote>
<blockquote>
<p>howard databases --input_annovar=tests/databases/others/hg19_nci60.txt --output_annovar=/tmp/nci60.from_annovar.vcf.gz --annovar_to_parquet=/tmp/nci60.from_annovar.parquet --annovar_code=nci60 --genome=~/howard/databases/genomes/current/hg19.fa</p>
</blockquote>
<p>Notes:</p>
<blockquote>
<ul>
<li>Downloading databases can take a while, depending on network, threads and memory</li>
</ul>
</blockquote>
<blockquote>
<ul>
<li>Proxy: Beware of network and proxy configuration</li>
</ul>
</blockquote>
<blockquote>
<ul>
<li>dbNSFP download: More threads, more memory usage (8 threads ~ 16Gb, 24 threads ~ 32Gb)</li>
</ul>
</blockquote>
<blockquote></blockquote>
<h3 id="main-options">Main options</h3>
<pre class="hljs"><code><div>--assembly=&lt;assembly&gt; (default: hg19)

Genome Assembly (e.g. 'hg19', 'hg38').

</div></code></pre>
<pre class="hljs"><code><div>--genomes-folder=&lt;genomes&gt; (default: /Users/lebechea/howard/databases/genomes/current)

Folder containing genomes.
(e.g. '/Users/lebechea/howard/databases/genomes/current'
</div></code></pre>
<pre class="hljs"><code><div>--genome=&lt;genome&gt; (default: ~/howard/databases/genomes/current/hg19/hg19.fa)

Genome file in fasta format (e.g. 'hg19.fa', 'hg38.fa').

</div></code></pre>
<pre class="hljs"><code><div>--param=&lt;param&gt; (default: {})

Parameters JSON file (or string) defines parameters to process 
annotations, calculations, prioritizations, convertions and queries.

</div></code></pre>
<h3 id="genomes">Genomes</h3>
<pre class="hljs"><code><div>--download-genomes=&lt;genomes&gt;

Path to genomes folder
with Fasta files, indexes,
and all files generated by pygenome module.
(e.g. '/Users/lebechea/howard/databases/genomes/current').

</div></code></pre>
<pre class="hljs"><code><div>--download-genomes-provider=&lt;genomes provider&gt; ['GENCODE', 'Ensembl', 'UCSC', 'NCBI'] (default: UCSC)

Download Genome from an external provider.
Available: GENCODE, Ensembl, UCSC, NCBI.

</div></code></pre>
<pre class="hljs"><code><div>--download-genomes-contig-regex=&lt;genomes contig regex&gt;

Regular expression to select specific chromosome
(e.g 'chr[0-9XYM]+$').

</div></code></pre>
<h3 id="snpeff">snpEff</h3>
<pre class="hljs"><code><div>--download-snpeff=&lt;snpEff&gt;

Download snpEff databases within snpEff folder
</div></code></pre>
<h3 id="annovar">Annovar</h3>
<pre class="hljs"><code><div>--download-annovar=&lt;Annovar&gt;

Path to Annovar databases
(e.g. '/Users/lebechea/howard/databases/annovar/current').

</div></code></pre>
<pre class="hljs"><code><div>--download-annovar-files=&lt;Annovar code&gt;

Download Annovar databases for a list of Annovar file code (see Annovar Doc).
Use None to donwload all available files,
or Annovar keyword (e.g. 'refGene', 'cosmic70', 'clinvar_202*').
Note that refGene will at least be downloaded,
and only files that not already exist or changed will be downloaded.

</div></code></pre>
<pre class="hljs"><code><div>--download-annovar-url=&lt;Annovar url&gt; (default: http://www.openbioinformatics.org/annovar/download)

Annovar databases URL (see Annovar Doc).

</div></code></pre>
<h3 id="refseq">refSeq</h3>
<pre class="hljs"><code><div>--download-refseq=&lt;refSeq&gt;

Path to refSeq databases
(e.g. '/Users/lebechea/howard/databases/refseq/current').

</div></code></pre>
<pre class="hljs"><code><div>--download-refseq-url=&lt;refSeq url&gt; (default: http://hgdownload.soe.ucsc.edu/goldenPath)

refSeq databases URL (see refSeq WebSite)
(e.g. 'http://hgdownload.soe.ucsc.edu/goldenPath')/n
</div></code></pre>
<pre class="hljs"><code><div>--download-refseq-prefix=&lt;refSeq prefix&gt; (default: ncbiRefSeq)

Check existing refSeq files in refSeq folder.

</div></code></pre>
<pre class="hljs"><code><div>--download-refseq-files=&lt;refSeq files&gt; (default: ncbiRefSeq.txt,ncbiRefSeqLink.txt)

List of refSeq files to download.

</div></code></pre>
<pre class="hljs"><code><div>--download-refseq-format-file=&lt;refSeq format file&gt;

Name of refSeq file to convert in BED format
(e.g. 'ncbiRefSeq.txt').
Process only if not None.

</div></code></pre>
<pre class="hljs"><code><div>--download-refseq-include-utr5

Formating BED refSeq file including 5'UTR.

</div></code></pre>
<pre class="hljs"><code><div>--download-refseq-include-utr3

Formating BED refSeq file including 3'UTR.

</div></code></pre>
<pre class="hljs"><code><div>--download-refseq-include-chrM

Formating BED refSeq file including Mitochondiral chromosome 'chrM' or 'chrMT'.

</div></code></pre>
<pre class="hljs"><code><div>--download-refseq-include-non-canonical-chr

Formating BED refSeq file including non canonical chromosomes.

</div></code></pre>
<pre class="hljs"><code><div>--download-refseq-include-non-coding-transcripts

Formating BED refSeq file including non coding transcripts.

</div></code></pre>
<pre class="hljs"><code><div>--download-refseq-include-transcript-version

Formating BED refSeq file including transcript version.

</div></code></pre>
<h3 id="dbnsfp">dbNSFP</h3>
<pre class="hljs"><code><div>--download-dbnsfp=&lt;dbNSFP&gt;

Download dbNSFP databases within dbNSFP folder(e.g. '/Users/lebechea/howard/databases').

</div></code></pre>
<pre class="hljs"><code><div>--download-dbnsfp-url=&lt;dbNSFP url&gt; (default: https://dbnsfp.s3.amazonaws.com)

Download dbNSFP databases URL (see dbNSFP website)
(e.g. https://dbnsfp.s3.amazonaws.com').

</div></code></pre>
<pre class="hljs"><code><div>--download-dbnsfp-release=&lt;dnNSFP release&gt; (default: 4.4a)

Release of dbNSFP to download (see dbNSFP website)
(e.g. '4.4a').

</div></code></pre>
<pre class="hljs"><code><div>--download-dbnsfp-parquet-size=&lt;dbNSFP parquet size&gt; (default: 100)

Maximum size (Mb) of data files in Parquet folder.
Parquet folder are partitioned (hive) by chromosome (sub-folder),
which contain N data files.

</div></code></pre>
<pre class="hljs"><code><div>--download-dbnsfp-subdatabases

Generate dbNSFP sub-databases.
dbNSFP provides multiple databases which are split onto multiple columns.
This option create a Parquet folder for each sub-database (based on columns names).

</div></code></pre>
<pre class="hljs"><code><div>--download-dbnsfp-parquet

Generate a Parquet file for each Parquet folder.

</div></code></pre>
<pre class="hljs"><code><div>--download-dbnsfp-vcf

Generate a VCF file for each Parquet folder.
Need genome FASTA file (see --download-genome).

</div></code></pre>
<pre class="hljs"><code><div>--download-dbnsfp-no-files-all

Not generate database Parquet/VCF file for the entire database ('ALL').
Only sub-databases files will be generated.
(see '--download-dbnsfp-subdatabases').

</div></code></pre>
<pre class="hljs"><code><div>--download-dbnsfp-add-info

Add INFO column (VCF format) in Parquet folder and file.
Useful for speed up full annotation (all available columns).
Increase memory and space during generation of files.

</div></code></pre>
<pre class="hljs"><code><div>--download-dbnsfp-row-group-size=&lt;dnNSFP row grooup size&gt; (default: 100000)

Minimum number of rows in a parquet row group (see duckDB doc).
Lower can reduce memory usage and slightly increase space during generation,
speed up highly selective queries, slow down whole file queries (e.g. aggregations).

</div></code></pre>
<h3 id="alphamissense">AlphaMissense</h3>
<pre class="hljs"><code><div>--download-alphamissense=&lt;AlphaMissense&gt;

Path to AlphaMissense databases
</div></code></pre>
<pre class="hljs"><code><div>--download-alphamissense-url=&lt;AlphaMissense url&gt; (default: https://storage.googleapis.com/dm_alphamissense)

Download AlphaMissense databases URL (see AlphaMissense website)
(e.g. 'https://storage.googleapis.com/dm_alphamissense').

</div></code></pre>
<h3 id="exomiser">Exomiser</h3>
<pre class="hljs"><code><div>--download-exomiser=&lt;Exomiser&gt;

Path to Exomiser databases
(e.g. /Users/lebechea/howard/databases/exomiser/current).

</div></code></pre>
<pre class="hljs"><code><div>--download-exomiser-application-properties=&lt;Exomiser application properties&gt;

Exomiser Application Properties configuration file (see Exomiser website).
This file contains configuration settings for the Exomiser tool.
If this parameter is not provided, the function will attempt to locate
the application properties file automatically based on the Exomiser.
Configuration information will be used to download expected releases (if no other parameters).
CADD and REMM will be downloaded only if 'path' are provided.

</div></code></pre>
<pre class="hljs"><code><div>--download-exomiser-url=&lt;Exomiser url&gt; (default: http://data.monarchinitiative.org/exomiser)

URL where Exomiser database files can be downloaded from
(e.g. 'http://data.monarchinitiative.org/exomiser').

</div></code></pre>
<pre class="hljs"><code><div>--download-exomiser-release=&lt;Exomiser release&gt;

Release of Exomiser data to download.
If &quot;default&quot;, &quot;auto&quot;, or &quot;config&quot;, retrieve from Application Properties file.
If not provided (None), from Application Properties file (Exomiser data-version) 
or default '2109'.

</div></code></pre>
<pre class="hljs"><code><div>--download-exomiser-phenotype-release=&lt;Exomiser phenoptye release&gt;

Release of Exomiser phenotype to download.
If not provided (None), from Application Properties file (Exomiser Phenotype data-version)
or Exomiser release.

</div></code></pre>
<pre class="hljs"><code><div>--download-exomiser-remm-release=&lt;Exomiser remm release&gt;

Release of ReMM (Regulatory Mendelian Mutation) database to download.
If &quot;default&quot;, &quot;auto&quot;, or &quot;config&quot;, retrieve from Application Properties file.

</div></code></pre>
<pre class="hljs"><code><div>--download-exomiser-remm-url=&lt;Exomiser remm url&gt; (default: https://kircherlab.bihealth.org/download/ReMM)

URL where ReMM (Regulatory Mendelian Mutation) database files can be downloaded from
(e.g. 'https://kircherlab.bihealth.org/download/ReMM').

</div></code></pre>
<pre class="hljs"><code><div>--download-exomiser-cadd-release=&lt;Exomiser cadd release&gt;

Release of CADD (Combined Annotation Dependent Depletion) database to download.
If &quot;default&quot;, &quot;auto&quot;, or &quot;config&quot;, retrieve from Application Properties file.

</div></code></pre>
<pre class="hljs"><code><div>--download-exomiser-cadd-url=&lt;Exomiser cadd url&gt; (default: https://kircherlab.bihealth.org/download/CADD)

URL where CADD (Combined Annotation Dependent Depletion) database files can be downloaded from
(e.g. 'https://kircherlab.bihealth.org/download/CADD').

</div></code></pre>
<pre class="hljs"><code><div>--download-exomiser-cadd-url-snv-file=&lt;Exomiser url snv file&gt; (default: whole_genome_SNVs.tsv.gz)

Name of the file containing the SNV (Single Nucleotide Variant) data
for the CADD (Combined Annotation Dependent Depletion) database.

</div></code></pre>
<pre class="hljs"><code><div>--download-exomiser-cadd-url-indel-file=&lt;Exomiser cadd url indel&gt; (default: InDels.tsv.gz)

Name of the file containing the INDEL (Insertion-Deletion) data
for the CADD (Combined Annotation Dependent Depletion) database.

</div></code></pre>
<h3 id="dbsnp">dbSNP</h3>
<pre class="hljs"><code><div>--download-dbsnp=&lt;dnSNP&gt;

Path to dbSNP databases
(e.g. '/Users/lebechea/howard/databases/exomiser/dbsnp').

</div></code></pre>
<pre class="hljs"><code><div>--download-dbsnp-releases=&lt;dnSNP releases&gt; (default: b156)

Release of dbSNP to download
(e.g. 'b152', 'b152,b156').

</div></code></pre>
<pre class="hljs"><code><div>--download-dbsnp-release-default=&lt;dnSNP release default&gt;

Default Release of dbSNP ('default' symlink)
(e.g. 'b156').
If None, first release to download will be assigned as default
only if it does not exists.

</div></code></pre>
<pre class="hljs"><code><div>--download-dbsnp-url=&lt;dbSNP url&gt; (default: https://ftp.ncbi.nih.gov/snp/archive)

URL where dbSNP database files can be downloaded from.
(e.g. 'https://ftp.ncbi.nih.gov/snp/archive').

</div></code></pre>
<pre class="hljs"><code><div>--download-dbsnp-url-files=&lt;dbSNP url files&gt;

Dictionary that maps assembly names to specific dbSNP URL files.
It allows you to provide custom dbSNP URL files for specific assemblies
instead of using the default file naming convention.

</div></code></pre>
<pre class="hljs"><code><div>--download-dbsnp-url-files-prefix=&lt;dbSNP url files prefix&gt; (default: GCF_000001405)

String that represents the prefix of the dbSNP file name for a specific assembly.
It is used to construct the full URL of the dbSNP file to be downloaded.

</div></code></pre>
<pre class="hljs"><code><div>--download-dbsnp-assemblies-map=&lt;dbSNP assemblies map&gt; (default: {'hg19': '25', 'hg38': '40'})

dictionary that maps assembly names to their corresponding dbSNP versions.
It is used to construct the dbSNP file name based on the assembly name.

</div></code></pre>
<pre class="hljs"><code><div>--download-dbsnp-vcf

Generate well-formatted VCF from downloaded file:
- Add and filter contigs associated to assembly
- Normalize by splitting multiallelics
- Need genome (see --download-genome)

</div></code></pre>
<pre class="hljs"><code><div>--download-dbsnp-parquet

Generate Parquet file from VCF.

</div></code></pre>
<h3 id="hgmd">HGMD</h3>
<pre class="hljs"><code><div>--convert-hgmd=&lt;HGMD&gt;

Convert HGMD databases.
Folder where the HGMD databases will be stored.
Fields in VCF, Parquet and TSV will be generated.
If the folder does not exist, it will be created.

</div></code></pre>
<pre class="hljs"><code><div>--convert-hgmd-file=&lt;HGMD file&gt;

File from HGMD.
Name format 'HGMD_Pro_&lt;release&gt;_&lt;assembly&gt;.vcf.gz'.

</div></code></pre>
<pre class="hljs"><code><div>--convert-hgmd-basename=&lt;HGMD basename&gt;

File output basename.
Generated files will be prefixed by basename
(e.g. 'HGMD_Pro_MY_RELEASE')
By default (None), input file name without '.vcf.gz'.

</div></code></pre>
<h3 id="fromannovar">from_Annovar</h3>
<pre class="hljs"><code><div>--input_annovar=&lt;input annovar&gt;

Input Annovar file path.
Format file must be a Annovar TXT file, associated with '.idx'.

</div></code></pre>
<pre class="hljs"><code><div>--output_annovar=&lt;output annovar&gt;

Output Annovar file path.
Format file must be either VCF compressesd file '.vcf.gz'.

</div></code></pre>
<pre class="hljs"><code><div>--annovar_code=&lt;Annovar code&gt;

Annovar code, or database name.
Usefull to name databases columns.

</div></code></pre>
<pre class="hljs"><code><div>--annovar_to_parquet=&lt;to parquet&gt;

Parquet file conversion.

</div></code></pre>
<pre class="hljs"><code><div>--annovar_reduce_memory=&lt;reduce memory&gt; ['auto', 'enable', 'disable'] (default: auto)

Reduce memory option for Annovar convert,
either 'auto' (auto-detection), 'enable' or 'disable'.

</div></code></pre>
<pre class="hljs"><code><div>--annovar_multi_variant=&lt;Annovar multi variant&gt; ['auto', 'enable', 'disable'] (default: auto)

Variant with multiple annotation lines on Annovar file.
Either 'auto' (auto-detection), 'enable' or 'disable'.

</div></code></pre>
<h3 id="fromextann">from_extann</h3>
<pre class="hljs"><code><div>--input_extann=&lt;input extann&gt;

Input Extann file path.
Format file must be a Extann TXT file or TSV file.
File need to have at least the genes column.

</div></code></pre>
<pre class="hljs"><code><div>--output_extann=&lt;output extann&gt;

Output Extann file path.
Output extann file, should be BED or BED.gz.

</div></code></pre>
<pre class="hljs"><code><div>--refgene=&lt;refGene&gt;

Path to refGene annotation file.

</div></code></pre>
<pre class="hljs"><code><div>--transcripts=&lt;transcripts&gt;

Transcripts TSV file,
with Transcript in first column, optional Gene in second column.

</div></code></pre>
<pre class="hljs"><code><div>--param_extann=&lt;param extann&gt;

Param extann file path.
Param containing configuration, options to replace chars and
bedlike header description, conf vcf specs.
(e.g. '~/howard/config/param.extann.json')

</div></code></pre>
<pre class="hljs"><code><div>--mode_extann=&lt;mode extann&gt; ['all', 'longest', 'chosen'] (default: longest)

Mode extann selection.
How to pick transcript from ncbi, keep all,
keep the longest, or keep the chosen one (transcript_extann).

</div></code></pre>
<h3 id="parameters">Parameters</h3>
<pre class="hljs"><code><div>--generate-param=&lt;param&gt;

Parameter file (JSON) with all databases found.
Databases folders scanned are defined in config file.
Structure of databases follow this structure (see doc):
.../&lt;database&gt;/&lt;release&gt;/&lt;assembly&gt;/*.[parquet|vcf.gz|...]

</div></code></pre>
<pre class="hljs"><code><div>--generate-param-description=&lt;param description&gt;

Description file (JSON) with all databases found.
Contains all databases with description of format, assembly, fields...

</div></code></pre>
<pre class="hljs"><code><div>--generate-param-releases=&lt;param release&gt; (default: current)

List of database folder releases to check
(e.g. 'current', 'latest').

</div></code></pre>
<pre class="hljs"><code><div>--generate-param-formats=&lt;param formats&gt; (default: parquet)

List of database formats to check
(e.g. 'parquet', 'parquet,vcf,bed,tsv').

</div></code></pre>
<pre class="hljs"><code><div>--generate-param-bcftools

Generate parameter JSON file with BCFTools annotation for allowed formats
(i.e. 'vcf', 'bed').

</div></code></pre>
<h2 id="gui-tool">GUI tool</h2>
<p>Graphical User Interface tools</p>
<p>Usage examples:</p>
<blockquote>
<p>howard gui</p>
</blockquote>
<h2 id="help-tool">HELP tool</h2>
<p>Help tools</p>
<p>Usage examples:</p>
<blockquote>
<p>howard help --help_md=docs/help.md --help_html=docs/html/help.html --help_pdf=docs/pdf/help.pdf</p>
</blockquote>
<blockquote>
<p>howard help --help_json_input=docs/json/help.config.json --help_json_input_title='HOWARD Configuration' --help_md=docs/help.config.md --help_html=docs/html/help.config.html --help_pdf=docs/pdf/help.config.pdf --code_type='json'</p>
</blockquote>
<blockquote>
<p>howard help --help_json_input=docs/json/help.param.json --help_json_input_title='HOWARD Parameters' --help_md=docs/help.param.md --help_html=docs/html/help.param.html --help_pdf=docs/pdf/help.param.pdf --code_type='json'</p>
</blockquote>
<blockquote>
<p>howard help --help_json_input=docs/json/help.param.databases.json --help_json_input_title='HOWARD Parameters Databases' --help_md=docs/help.param.databases.md --help_html=docs/html/help.param.databases.html --help_pdf=docs/pdf/help.param.databases.pdf --code_type='json'</p>
</blockquote>
<blockquote></blockquote>
<h3 id="main-options">Main options</h3>
<pre class="hljs"><code><div>--help_md=&lt;help markdown&gt;

Help Output file in MarkDown format.

</div></code></pre>
<pre class="hljs"><code><div>--help_html=&lt;help html&gt;

Help Output file in HTML format.

</div></code></pre>
<pre class="hljs"><code><div>--help_pdf=&lt;help pdf&gt;

Help Output file in PDF format.

</div></code></pre>
<pre class="hljs"><code><div>--help_json_input=&lt;help JSON input&gt;

Help input file in JSON format.

</div></code></pre>
<pre class="hljs"><code><div>--help_json_input_title=&lt;help JSON input title&gt; (default: Help)

Help JSON input title.

</div></code></pre>
<pre class="hljs"><code><div>--code_type=&lt;example code type&gt;

Help example code type for input JSON format
(e.g. 'json', 'bash').

</div></code></pre>
<h2 id="genebe-tool">GENEBE tool</h2>
<p>GeneBe annotation using REST API (see https://genebe.net/).</p>
<p>Usage examples:</p>
<blockquote>
<p>howard genebe --input=tests/data/example.vcf.gz --output=/tmp/example.genebe.vcf.gz --genebe_use_refseq</p>
</blockquote>
<blockquote></blockquote>
<h3 id="main-options">Main options</h3>
<pre class="hljs"><code><div>--input=&lt;input&gt; | required

Input file path.
Format file must be either VCF, Parquet, TSV, CSV, PSV or duckDB.
Files can be compressesd (e.g. vcf.gz, tsv.gz).

</div></code></pre>
<pre class="hljs"><code><div>--output=&lt;output&gt; | required

Output file path.
Format file must be either VCF, Parquet, TSV, CSV, PSV or duckDB.
Files can be compressesd (e.g. vcf.gz, tsv.gz).

</div></code></pre>
<pre class="hljs"><code><div>--param=&lt;param&gt; (default: {})

Parameters JSON file (or string) defines parameters to process 
annotations, calculations, prioritizations, convertions and queries.

</div></code></pre>
<pre class="hljs"><code><div>--assembly=&lt;assembly&gt; (default: hg19)

Genome Assembly (e.g. 'hg19', 'hg38').

</div></code></pre>
<h3 id="genebe">GeneBe</h3>
<pre class="hljs"><code><div>--genebe_use_refseq

Use refSeq to annotate (default).

</div></code></pre>
<pre class="hljs"><code><div>--genebe_use_ensembl

Use Ensembl to annotate.

</div></code></pre>
<pre class="hljs"><code><div>--not_flatten_consequences

Use exploded annotation informations.

</div></code></pre>
<h3 id="explode">Explode</h3>
<pre class="hljs"><code><div>--explode_infos

Explode VCF INFO/Tag into 'variants' table columns.

</div></code></pre>
<pre class="hljs"><code><div>--explode_infos_prefix=&lt;explode infos prefix&gt;

Explode VCF INFO/Tag with a specific prefix.

</div></code></pre>
<pre class="hljs"><code><div>--explode_infos_fields=&lt;explode infos list&gt; (default: *)

Explode VCF INFO/Tag specific fields/tags.
Keyword `*` specify all available fields, except those already specified.
Pattern (regex) can be used, such as `.*_score` for fields named with '_score' at the end.
Examples:
- 'HGVS,SIFT,Clinvar' (list of fields)
- 'HGVS,*,Clinvar' (list of fields with all other fields at the end)
- 'HGVS,.*_score,Clinvar' (list of 2 fields with all scores in the middle)
- 'HGVS,.*_score,*' (1 field, scores, all other fields)
- 'HGVS,*,.*_score' (1 field, all other fields, all scores)

</div></code></pre>
<h3 id="export">Export</h3>
<pre class="hljs"><code><div>--include_header

Include header (in VCF format) in output file.
Only for compatible formats (tab-delimiter format as TSV or BED).

</div></code></pre>
<pre class="hljs"><code><div>--order_by=&lt;order by&gt;

List of columns to sort the result-set in ascending or descending order.
Use SQL format, and keywords ASC (ascending) and DESC (descending).
If a column is not available, order will not be considered.
Order is enable only for compatible format (e.g. TSV, CSV, JSON).
Examples: 'ACMG_score DESC', 'PZFlag DESC, PZScore DESC'.

</div></code></pre>
<pre class="hljs"><code><div>--parquet_partitions=&lt;parquet partitions&gt;

Parquet partitioning using hive (available for any format).
This option is faster parallel writing, but memory consuming.
Use 'None' (string) for NO partition but split parquet files into a folder.
Examples: '#CHROM', '#CHROM,REF', 'None'.

</div></code></pre>
<h2 id="minimalize-tool">MINIMALIZE tool</h2>
<p>Minimalize a VCF file consists in put missing value ('.') on INFO/Tags, ID, QUAL or FILTER fields. Options can also minimalize samples (keep only GT) or remove all samples. INFO/tags can by exploded before minimalize to keep tags into separated columns (useful for Parquet or TSV format to constitute a database).</p>
<p>Usage examples:</p>
<blockquote>
<p>howard minimalize --input=tests/data/example.vcf.gz --output=/tmp/example.minimal.vcf.gz --minimalize_info --minimalize_filter --minimalize_qual --minimalize_id --minimalize_samples</p>
</blockquote>
<blockquote>
<p>howard minimalize --input=tests/data/example.vcf.gz --output=/tmp/example.minimal.tsv --remove_samples --explode_infos --minimalize_info</p>
</blockquote>
<blockquote></blockquote>
<h3 id="main-options">Main options</h3>
<pre class="hljs"><code><div>--input=&lt;input&gt; | required

Input file path.
Format file must be either VCF, Parquet, TSV, CSV, PSV or duckDB.
Files can be compressesd (e.g. vcf.gz, tsv.gz).

</div></code></pre>
<pre class="hljs"><code><div>--output=&lt;output&gt; | required

Output file path.
Format file must be either VCF, Parquet, TSV, CSV, PSV or duckDB.
Files can be compressesd (e.g. vcf.gz, tsv.gz).

</div></code></pre>
<pre class="hljs"><code><div>--param=&lt;param&gt; (default: {})

Parameters JSON file (or string) defines parameters to process 
annotations, calculations, prioritizations, convertions and queries.

</div></code></pre>
<h3 id="minimalize">Minimalize</h3>
<pre class="hljs"><code><div>--minimalize_info

Minimalize INFO field (e.g. '.' value).

</div></code></pre>
<pre class="hljs"><code><div>--minimalize_id

Minimalize ID field (e.g. '.' value).

</div></code></pre>
<pre class="hljs"><code><div>--minimalize_qual

Minimalize QUAL field (e.g. '.' value).

</div></code></pre>
<pre class="hljs"><code><div>--minimalize_filter

Minimalize FILTER field (e.g. '.' value).

</div></code></pre>
<pre class="hljs"><code><div>--minimalize_samples

Minimalize samples to keep only genotypes (i.e. 'GT').

</div></code></pre>
<pre class="hljs"><code><div>--remove_samples

Remove all samples to keep only variants.

</div></code></pre>
<h3 id="explode">Explode</h3>
<pre class="hljs"><code><div>--explode_infos

Explode VCF INFO/Tag into 'variants' table columns.

</div></code></pre>
<pre class="hljs"><code><div>--explode_infos_prefix=&lt;explode infos prefix&gt;

Explode VCF INFO/Tag with a specific prefix.

</div></code></pre>
<pre class="hljs"><code><div>--explode_infos_fields=&lt;explode infos list&gt; (default: *)

Explode VCF INFO/Tag specific fields/tags.
Keyword `*` specify all available fields, except those already specified.
Pattern (regex) can be used, such as `.*_score` for fields named with '_score' at the end.
Examples:
- 'HGVS,SIFT,Clinvar' (list of fields)
- 'HGVS,*,Clinvar' (list of fields with all other fields at the end)
- 'HGVS,.*_score,Clinvar' (list of 2 fields with all scores in the middle)
- 'HGVS,.*_score,*' (1 field, scores, all other fields)
- 'HGVS,*,.*_score' (1 field, all other fields, all scores)

</div></code></pre>
<h3 id="export">Export</h3>
<pre class="hljs"><code><div>--include_header

Include header (in VCF format) in output file.
Only for compatible formats (tab-delimiter format as TSV or BED).

</div></code></pre>
<pre class="hljs"><code><div>--order_by=&lt;order by&gt;

List of columns to sort the result-set in ascending or descending order.
Use SQL format, and keywords ASC (ascending) and DESC (descending).
If a column is not available, order will not be considered.
Order is enable only for compatible format (e.g. TSV, CSV, JSON).
Examples: 'ACMG_score DESC', 'PZFlag DESC, PZScore DESC'.

</div></code></pre>
<pre class="hljs"><code><div>--parquet_partitions=&lt;parquet partitions&gt;

Parquet partitioning using hive (available for any format).
This option is faster parallel writing, but memory consuming.
Use 'None' (string) for NO partition but split parquet files into a folder.
Examples: '#CHROM', '#CHROM,REF', 'None'.

</div></code></pre>
<h2 id="shared-arguments">Shared arguments</h2>
<pre class="hljs"><code><div>--config=&lt;config&gt; (default: {})

Configuration JSON file defined default configuration regarding 
resources (e.g. threads, memory),
settings (e.g. verbosity, temporary files),
default folders (e.g. for databases)
and paths to external tools.

</div></code></pre>
<pre class="hljs"><code><div>--threads=&lt;threads&gt; (default: -1)

Specify the number of threads to use for processing HOWARD.
It determines the level of parallelism,
either on python scripts, duckdb engine and external tools.
It and can help speed up the process/tool.
Use -1 to use all available CPU/cores.
Either non valid value is 1 CPU/core.

</div></code></pre>
<pre class="hljs"><code><div>--memory=&lt;memory&gt;

Specify the memory to use in format FLOAT[kMG]
(e.g. '8G', '12.42G', '1024M').
It determines the amount of memory for duckDB engine and external tools
(especially for JAR programs).
It can help to prevent 'out of memory' failures.
By default (None) is 80%% of RAM (for duckDB).

</div></code></pre>
<pre class="hljs"><code><div>--chunk_size=&lt;chunk size&gt; (default: 1000000)

Number of records in batch to export output file.
The lower the chunk size, the less memory consumption.
For Parquet partitioning, files size will depend on the chunk size.

</div></code></pre>
<pre class="hljs"><code><div>--tmp=&lt;Temporary folder&gt;

Temporary folder (e.g. '/tmp').
By default, '.tmp' for duckDB (see doc),external tools and python scripts.

</div></code></pre>
<pre class="hljs"><code><div>--duckdb_settings=&lt;duckDB settings&gt;

DuckDB settings (see duckDB doc) as JSON (string or file).
These settings have priority (see options 'threads', 'tmp'...).
Examples: '{&quot;TimeZone&quot;: &quot;GMT&quot;, &quot;temp_directory&quot;: &quot;/tmp/duckdb&quot;, &quot;threads&quot;: 8}'.

</div></code></pre>
<pre class="hljs"><code><div>--verbosity=&lt;verbosity&gt; ['CRITICAL', 'ERROR', 'WARNING', 'INFO', 'DEBUG', 'NOTSET'] (default: INFO)

Verbosity level
Available: CRITICAL, ERROR, WARNING, INFO, DEBUG or NOTSET
- DEBUG: Detailed information, typically of interest only when diagnosing problems.
- INFO: Confirmation that things are working as expected.
- WARNING: An indication that something unexpected happened.
- ERROR: Due to a more serious problem.
- CRITICAL: A serious error.
- NOTSET: All messages.

</div></code></pre>
<pre class="hljs"><code><div>--log=&lt;log&gt;

Logs file
(e.g. 'my.log').

</div></code></pre>
<pre class="hljs"><code><div>--quiet

==SUPPRESS==
</div></code></pre>
<pre class="hljs"><code><div>--verbose

==SUPPRESS==
</div></code></pre>
<pre class="hljs"><code><div>--debug

==SUPPRESS==
</div></code></pre>

</body>
</html>
