<H1 id='HOWARD Configuration'>HOWARD Configuration</H1>HOWARD Configuration JSON file defined default configuration regarding resources (e.g. threads, memory), settings (e.g. verbosity, temporary files), default folders (e.g. for databases) and paths to external tools.<br>
                    <H2>Table of contents</H1>
                    - <a href='#HOWARD Configuration'>HOWARD Configuration</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#folders'>folders</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#folders::databases'>databases</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#folders::databases::genomes'>genomes</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#folders::databases::annotations'>annotations</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#folders::databases::parquet'>parquet</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#folders::databases::bcftools'>bcftools</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#folders::databases::annovar'>annovar</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#folders::databases::snpeff'>snpeff</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#folders::databases::exomiser'>exomiser</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#folders::databases::refseq'>refseq</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#tools'>tools</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#tools::bcftools'>bcftools</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#tools::bgzip'>bgzip</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#tools::java'>java</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#tools::snpeff'>snpeff</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#tools::annovar'>annovar</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#tools::exomiser'>exomiser</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#threads'>threads</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#memory'>memory</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#assembly'>assembly</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#verbosity'>verbosity</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#tmp'>tmp</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#access'>access</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#duckdb_settings'>duckdb_settings</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#chunk_size'>chunk_size</a><BR>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- <a href='#log'>log</a><BR>
                    <br>
                    Examples: {'Example of a configuration JSON file': ['{', '  "threads": 8,', '  "memory": null,', '  "verbosity": "WARNING",', '  "folders": {', '    "databases": {', '      "genomes": "~/howard/databases/genomes/current",', '      "annotations": [', '        "~/howard/databases/annotations/current",', '        "~/howard/databases/dbnsfp/current",', '        "~/howard/databases/dbsnp/current"', '      ],', '      "parquet": [', '        "~/howard/databases/annotations/current"', '      ],', '      "bcftools": [', '        "~/howard/databases/annotations/current"', '      ],', '      "annovar": [', '        "~/howard/databases/annovar/current"', '      ],', '      "snpeff": "~/howard/databases/snpeff/current",', '      "varank": "~/howard/databases/varank/current"', '    }', '  },', '  "tools": {', '    "bcftools": "bcftools",', '    "bgzip": "bgzip",', '    "java": "java",', '    "snpeff": "~/howard/tools/snpeff/current/bin/snpEff.jar",', '    "annovar": "~/howard/tools/annovar/current/bin/table_annovar.pl",', '    "exomiser": "~/howard/tools/exomiser/current/bin/exomiser-cli-13.2.0.jar"', '  }', '}']}<br><H2 id='folders'>folders</H2>Folders configuration such as for databases.<br><H3 id='folders::databases'>folders::databases</H3>Default folders for databases that follows the specific database HOWARD format. These folders will be used in HOWARD tools to autodetect databases by their name and using assembly. Within database folders, multiple releases can be provides (e.g. 'b152' and 'b156' for dbSNP and 'hg19' assembly within folder '~/howard/databases/dbsnp/current/hg19', in 2 subfolders resp.)<br>Format: /path/to/databases/<db_name>/<db_release>/<assembly>/<database_files><br>Examples: <xmp># Example of a configuration for databases folders
"databases": {
   "genomes": "~/howard/databases/genomes/current",
   "annotations": [
      "~/howard/databases/annotations/current",
      "~/howard/databases/dbnsfp/current",
      "~/howard/databases/dbsnp/current"
   ],
   "parquet": [
      "~/howard/databases/annotations/current"
   ],
   "bcftools": [
      "~/howard/databases/bcftools/current"
   ],
   "annovar": "~/howard/databases/annovar/current",
   "snpeff": "~/howard/databases/snpeff/current",
   "exomiser": "~/howard/databases/exomiser/current"
}</xmp><br><H4 id='folders::databases::genomes'>folders::databases::genomes</H4>Genome folder with, for each assembly, FASTA files, indexes, and all files generated by pygenome module.<br>Type: Path<br>Format: A folder path (without assembly)<br>Default: ~/howard/databases/genomes/current<br>Examples: <xmp># Path to genomes folder
"genomes": "~/howard/databases/genomes/current"</xmp><br><H4 id='folders::databases::annotations'>folders::databases::annotations</H4>Annotation databases folders that contains databases in various format such as Parquet, VCF, duckDB and TSV.<br>Type: Path<br>Format: a list of folder path (without assembly)<br>Default: ["~/howard/databases/annotations/current"]<br>Examples: <xmp># Uniq folder with multiple annotations for Parquet annotation method, or other External tools
"annotations": [
   "~/howard/databases/annotations/current"
]
# Combinason of 2 folders with multiple annotations for Parquet annotation method, or other External tools
"annotations": [
   "~/howard/databases/annotations/current",
   "~/howard/databases/dejavu/current",
   "~/howard/databases/dbnsfp/current"
]</xmp><br><H4 id='folders::databases::parquet'>folders::databases::parquet</H4>Annotation databases folders that contains databases in Parquet format.<br>Format: a list of folder path (without assembly)<br>Default: ["~/howard/databases/annotations/current"]<br>Examples: <xmp># Uniq folder with multiple annotations for Parquet annotation method
"annotations": [
   "~/howard/databases/annotations/current"
]
# Combinason of 2 folders with multiple annotations for Parquet annotation method
"annotations": [
   "~/howard/databases/annotations/current",
   "~/howard/databases/dejavu/current",
   "~/howard/databases/dbnsfp/current"
]</xmp><br><H4 id='folders::databases::bcftools'>folders::databases::bcftools</H4>Annotation databases folders for BCFTools annotation.<br>Format: a list of folder path (without assembly)<br>Default: ["~/howard/databases/bcftools/current"]<br>Examples: <xmp># Uniq folder with multiple VCF and BED files for BCFTools annotation
"bcftools": [
   "~/howard/databases/bcftools/current"
]
# Combinason of 2 folders with multiple VCF and BED files for BCFTools annotation
"bcftools": [
   "~/howard/databases/bcftools/current",
   "~/howard/databases/dejavu/current"
]</xmp><br><H4 id='folders::databases::annovar'>folders::databases::annovar</H4>Annotation databases folders for Annovar annotation.<br>Format: a list of folder path (without assembly)<br>Default: ["~/howard/databases/annovar/current"]<br>Examples: <xmp># Uniq folder with multiple Annovar TXT files for Annovar annotation
"annovar": [
   "~/howard/databases/annovar/current/"
]
# Combinason of 2 folders with multiple Annovar TXT files for Annovar annotation
"annovar": [
   "~/howard/databases/annovar/current/",
   "~/howard/databases/annovar/homemade/"
]</xmp><br><H4 id='folders::databases::snpeff'>folders::databases::snpeff</H4>Annotation databases folders for snpEff annotation.<br>Format: A folder path (without assembly)<br>Default: ~/howard/databases/snpeff/current<br>Examples: <xmp># Path to snpEff database folder
"snpeff": "~/howard/databases/snpeff/current/"</xmp><br><H4 id='folders::databases::exomiser'>folders::databases::exomiser</H4>Annotation databases folders for Exomiser annotation.<br>Format: A folder path (without assembly)<br>Default: ~/howard/databases/exomiser/current<br>Examples: <xmp># Path to Exomiser database folder
"exomiser": "~/howard/databases/exomiser/current/"</xmp><br><H4 id='folders::databases::refseq'>folders::databases::refseq</H4>Annotation databases folders for refSeq annotation.<br>Format: A folder path (without assembly)<br>Default: ~/howard/databases/refseq/current<br>Examples: <xmp># Path to refSeq files folder
"refseq": "~/howard/databases/refseq/current/"</xmp><br><H2 id='tools'>tools</H2>External tools paths that can be defined as path to a binary or a dict including the binary type (such as "bin", "jar", "perl").<br>Examples: <xmp># Example of a configuration for tools, with env $PATH, full path and path with type
"tools": {
   "bcftools": "bcftools",
   "bgzip": "bgzip",
   "java": "/usr/bin/java",
   "snpeff": "~/howard/tools/snpeff/current/bin/snpEff.jar",
   "annovar": {"jar": "~/howard/tools/annovar/current/bin/table_annovar.pl"},
   "exomiser": {"jar": "~/howard/tools/exomiser/current/bin/exomiser-cli-13.2.0.jar}"
}</xmp><br><H3 id='tools::bcftools'>tools::bcftools</H3>BCFTools binary (see https://samtools.github.io/bcftools/).<br>Default: bcftools<br>Examples: <xmp># Path to binary in $PATH env variable
"bcftools": "bcftools"
# Path to binary as a dict with binary type 'bin'
"bcftools": {"bin": "~/howard/tools/bcftools/current/bin/bcftools"}</xmp><br><H3 id='tools::bgzip'>tools::bgzip</H3>BGZip binary (see https://samtools.github.io/bcftools/).<br>Default: bgzip<br>Examples: <xmp># Path to binary in $PATH env variable
"bgzip": "bgzip"
# Path to binary as a dict with binary type 'bin'
"bgzip": {"bin": "~/howard/tools/htslib/current/bin/bgzip"}</xmp><br><H3 id='tools::java'>tools::java</H3>Java binary (see https://www.java.com).<br>Default: java<br>Examples: <xmp># Path to binary in $PATH env variable
"java": "java"
# Path to binary as a dict with binary type 'bin'
"java": {"bin": "/usr/bin/java"}</xmp><br><H3 id='tools::snpeff'>tools::snpeff</H3>snpEff binary (see https://pcingola.github.io/SnpEff/).<br>Default: ~/howard/tools/snpeff/current/bin/snpEff.jar<br>Examples: <xmp># Path to binary as a dict without binary type
"snpeff": "~/howard/tools/snpeff/current/bin/snpEff.jar"
# Path to binary as a dict with binary type 'jar'
"snpeff": {"jar": "~/howard/tools/snpeff/current/bin/snpEff.jar"}</xmp><br><H3 id='tools::annovar'>tools::annovar</H3>ANNOVAR perl script (see https://annovar.openbioinformatics.org/).<br>Default: ~/howard/tools/annovar/current/bin/table_annovar.pl<br>Examples: <xmp># Path to binary as a dict without binary type
"annovar": "~/howard/tools/annovar/current/bin/table_annovar.pl"
# Path to binary as a dict with binary type 'perl'
"annovar": {"jar": "~/howard/tools/annovar/current/bin/table_annovar.pl"}</xmp><br><H3 id='tools::exomiser'>tools::exomiser</H3>Exomiser binary (see https://www.sanger.ac.uk/tool/exomiser/).<br>Default: ~/howard/tools/exomiser/current/bin/exomiser-cli-13.2.0.jar<br>Examples: <xmp># Path to binary as a dict without binary type
"snpeff": "~/howard/tools/exomiser/current/bin/exomiser-cli-13.2.0.jar"
# Path to binary as a dict with binary type 'jar'
"snpeff": {"jar": "~/howard/tools/exomiser/current/bin/exomiser-cli-13.2.0.jar"}</xmp><br><H2 id='threads'>threads</H2>Specify the number of threads to use for processing HOWARD. It determines the level of parallelism, either on python scripts, duckdb engine and external tools. It and can help speed up the process/tool. Use -1 to use all available CPU/cores. Either non valid value is 1 CPU/core. <br>Type: int<br>Default: -1<br>Examples: {'# Automatically detect all available CPU/cores': '"threads": -1', '# Define 8 CPU/cores': '"threads": 8'}<br><H2 id='memory'>memory</H2>Specify the memory to use in format FLOAT[kMG] (e.g. '8G', '12.42G', '1024M'). It determines the amount of memory for duckDB engine and external tools (especially for JAR programs). It can help to prevent 'out of memory' failures. By default (None) is 80%% of RAM (for duckDB). <br>Type: str<br>Format: FLOAT[kMG]<br>Default: None<br>Examples: {'# Automatically detect all available CPU/cores': '"threads": -1', '# Define 8 CPU/cores': '"threads": 8'}<br><H2 id='assembly'>assembly</H2>Genome Assembly (e.g. 'hg19', 'hg38'). <br>Type: str<br>Default: hg19<br>Examples: {'Default assembly for all analysis tools': '"assembly": \'hg19\'', 'List of assemblies for databases download tool': '"assembly": \'hg19,hg38\''}<br><H2 id='verbosity'>verbosity</H2>Verbosity level Available: CRITICAL, ERROR, WARNING, INFO, DEBUG or NOTSET<br>- DEBUG: Detailed information, typically of interest only when diagnosing problems.<br>- INFO: Confirmation that things are working as expected.<br>- WARNING: An indication that something unexpected happened.<br>- ERROR: Due to a more serious problem.<br>- CRITICAL: A serious error.<br>- NOTSET: All messages. <br>Type: str<br>Choices: <br>CRITICAL<br>ERROR<br>WARNING<br>INFO<br>DEBUG<br>NOTSET<br>Default: INFO<br>Examples: {'Default verbosity': '"verbosity": "INFO"', 'ERROR level (quiet mode)': '"verbosity": "ERROR"', 'For debug': '"verbosity": "DEBUG"'}<br><H2 id='tmp'>tmp</H2>Temporary folder (e.g. '/tmp'). By default, '.tmp' for duckDB (see doc),external tools and python scripts. <br>Type: Path<br>Default: None<br>Examples: {'# System temporary folder': '"tmp": "/tmp"', '# HOWARD work directory': '"tmp": "~/howard/tmp"', '# Current work directory': '"tmp": ".tmp"'}<br><H2 id='access'>access</H2>Access mode to variants file or database. Either 'RW' for Read and Write, or 'RO' for Read Only. <br>Type: str<br>Choices: <br>RW<br>RO<br>Default: RW<br>Examples: {'Read and Write mode': '"access": "RW"', 'Read only mode': '"access": "RO"'}<br><H2 id='duckdb_settings'>duckdb_settings</H2>DuckDB settings (see duckDB doc) as JSON (string or file). These settings have priority (see options 'threads', 'tmp'...). Examples: '{"TimeZone": "GMT", "temp_directory": "/tmp/duckdb", "threads": 8}'. <br>Type: Path<br>Default: None<br>Examples: {'DuckDB settings JSON file': '"duckdb_settings": "/path/to/duckdb_config.json"', 'JSON string for Time zone, temporary directory and threads for duckDB': '"duckdb_settings": {\n   "TimeZone": "GMT",\n   "temp_directory": "/tmp/duckdb",\n   "threads": 8\n}'}<br><H2 id='chunk_size'>chunk_size</H2>Number of records in batch to export output file. The lower the chunk size, the less memory consumption. For Parquet partitioning, files size will depend on the chunk size. <br>Type: int<br>Default: 1000000<br>Examples: {'Chunk size of 1.000.000 by default': '"chunk_size": 1000000', 'Smaller chunk size to reduce Parquet file size and memory usage': '"chunk_size": 100000'}<br><H2 id='log'>log</H2>Logs file (e.g. 'my.log'). <br>Type: Path<br>Default: None<br>Examples: {'Relative path to log file': '"log": "my.log"', '# HOWARD work directory': '"log": "~/howard/log"', 'Full path to log file': '"log": "/tmp/my.log"'}<br>