<!DOCTYPE html>
<html>
<head>
<title>user_guide.md</title>
<meta http-equiv="Content-type" content="text/html;charset=UTF-8">

<style>
/* https://github.com/microsoft/vscode/blob/master/extensions/markdown-language-features/media/markdown.css */
/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Microsoft Corporation. All rights reserved.
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/

body {
	font-family: var(--vscode-markdown-font-family, -apple-system, BlinkMacSystemFont, "Segoe WPC", "Segoe UI", "Ubuntu", "Droid Sans", sans-serif);
	font-size: var(--vscode-markdown-font-size, 14px);
	padding: 0 26px;
	line-height: var(--vscode-markdown-line-height, 22px);
	word-wrap: break-word;
}

#code-csp-warning {
	position: fixed;
	top: 0;
	right: 0;
	color: white;
	margin: 16px;
	text-align: center;
	font-size: 12px;
	font-family: sans-serif;
	background-color:#444444;
	cursor: pointer;
	padding: 6px;
	box-shadow: 1px 1px 1px rgba(0,0,0,.25);
}

#code-csp-warning:hover {
	text-decoration: none;
	background-color:#007acc;
	box-shadow: 2px 2px 2px rgba(0,0,0,.25);
}

body.scrollBeyondLastLine {
	margin-bottom: calc(100vh - 22px);
}

body.showEditorSelection .code-line {
	position: relative;
}

body.showEditorSelection .code-active-line:before,
body.showEditorSelection .code-line:hover:before {
	content: "";
	display: block;
	position: absolute;
	top: 0;
	left: -12px;
	height: 100%;
}

body.showEditorSelection li.code-active-line:before,
body.showEditorSelection li.code-line:hover:before {
	left: -30px;
}

.vscode-light.showEditorSelection .code-active-line:before {
	border-left: 3px solid rgba(0, 0, 0, 0.15);
}

.vscode-light.showEditorSelection .code-line:hover:before {
	border-left: 3px solid rgba(0, 0, 0, 0.40);
}

.vscode-light.showEditorSelection .code-line .code-line:hover:before {
	border-left: none;
}

.vscode-dark.showEditorSelection .code-active-line:before {
	border-left: 3px solid rgba(255, 255, 255, 0.4);
}

.vscode-dark.showEditorSelection .code-line:hover:before {
	border-left: 3px solid rgba(255, 255, 255, 0.60);
}

.vscode-dark.showEditorSelection .code-line .code-line:hover:before {
	border-left: none;
}

.vscode-high-contrast.showEditorSelection .code-active-line:before {
	border-left: 3px solid rgba(255, 160, 0, 0.7);
}

.vscode-high-contrast.showEditorSelection .code-line:hover:before {
	border-left: 3px solid rgba(255, 160, 0, 1);
}

.vscode-high-contrast.showEditorSelection .code-line .code-line:hover:before {
	border-left: none;
}

img {
	max-width: 100%;
	max-height: 100%;
}

a {
	text-decoration: none;
}

a:hover {
	text-decoration: underline;
}

a:focus,
input:focus,
select:focus,
textarea:focus {
	outline: 1px solid -webkit-focus-ring-color;
	outline-offset: -1px;
}

hr {
	border: 0;
	height: 2px;
	border-bottom: 2px solid;
}

h1 {
	padding-bottom: 0.3em;
	line-height: 1.2;
	border-bottom-width: 1px;
	border-bottom-style: solid;
}

h1, h2, h3 {
	font-weight: normal;
}

table {
	border-collapse: collapse;
}

table > thead > tr > th {
	text-align: left;
	border-bottom: 1px solid;
}

table > thead > tr > th,
table > thead > tr > td,
table > tbody > tr > th,
table > tbody > tr > td {
	padding: 5px 10px;
}

table > tbody > tr + tr > td {
	border-top: 1px solid;
}

blockquote {
	margin: 0 7px 0 5px;
	padding: 0 16px 0 10px;
	border-left-width: 5px;
	border-left-style: solid;
}

code {
	font-family: Menlo, Monaco, Consolas, "Droid Sans Mono", "Courier New", monospace, "Droid Sans Fallback";
	font-size: 1em;
	line-height: 1.357em;
}

body.wordWrap pre {
	white-space: pre-wrap;
}

pre:not(.hljs),
pre.hljs code > div {
	padding: 16px;
	border-radius: 3px;
	overflow: auto;
}

pre code {
	color: var(--vscode-editor-foreground);
	tab-size: 4;
}

/** Theming */

.vscode-light pre {
	background-color: rgba(220, 220, 220, 0.4);
}

.vscode-dark pre {
	background-color: rgba(10, 10, 10, 0.4);
}

.vscode-high-contrast pre {
	background-color: rgb(0, 0, 0);
}

.vscode-high-contrast h1 {
	border-color: rgb(0, 0, 0);
}

.vscode-light table > thead > tr > th {
	border-color: rgba(0, 0, 0, 0.69);
}

.vscode-dark table > thead > tr > th {
	border-color: rgba(255, 255, 255, 0.69);
}

.vscode-light h1,
.vscode-light hr,
.vscode-light table > tbody > tr + tr > td {
	border-color: rgba(0, 0, 0, 0.18);
}

.vscode-dark h1,
.vscode-dark hr,
.vscode-dark table > tbody > tr + tr > td {
	border-color: rgba(255, 255, 255, 0.18);
}

</style>

<style>
/*

github.com style (c) Vasily Polovnyov <vast@whiteants.net>

*/

.hljs {
  display: block;
  overflow-x: auto;
  padding: 0.5em;
  color: #333;
  background: #f8f8f8;
}

.hljs-comment,
.hljs-quote {
  color: #998;
  font-style: italic;
}

.hljs-keyword,
.hljs-selector-tag,
.hljs-subst {
  color: #333;
  font-weight: bold;
}

.hljs-number,
.hljs-literal,
.hljs-variable,
.hljs-template-variable,
.hljs-tag .hljs-attr {
  color: #008080;
}

.hljs-string,
.hljs-doctag {
  color: #d14;
}

.hljs-title,
.hljs-section,
.hljs-selector-id {
  color: #900;
  font-weight: bold;
}

.hljs-subst {
  font-weight: normal;
}

.hljs-type,
.hljs-class .hljs-title {
  color: #458;
  font-weight: bold;
}

.hljs-tag,
.hljs-name,
.hljs-attribute {
  color: #000080;
  font-weight: normal;
}

.hljs-regexp,
.hljs-link {
  color: #009926;
}

.hljs-symbol,
.hljs-bullet {
  color: #990073;
}

.hljs-built_in,
.hljs-builtin-name {
  color: #0086b3;
}

.hljs-meta {
  color: #999;
  font-weight: bold;
}

.hljs-deletion {
  background: #fdd;
}

.hljs-addition {
  background: #dfd;
}

.hljs-emphasis {
  font-style: italic;
}

.hljs-strong {
  font-weight: bold;
}

</style>

<style>
/*
 * Markdown PDF CSS
 */

 body {
	font-family: -apple-system, BlinkMacSystemFont, "Segoe WPC", "Segoe UI", "Ubuntu", "Droid Sans", sans-serif, "Meiryo";
	padding: 0 12px;
}

pre {
	background-color: #f8f8f8;
	border: 1px solid #cccccc;
	border-radius: 3px;
	overflow-x: auto;
	white-space: pre-wrap;
	overflow-wrap: break-word;
}

pre:not(.hljs) {
	padding: 23px;
	line-height: 19px;
}

blockquote {
	background: rgba(127, 127, 127, 0.1);
	border-color: rgba(0, 122, 204, 0.5);
}

.emoji {
	height: 1.4em;
}

code {
	font-size: 14px;
	line-height: 19px;
}

/* for inline code */
:not(pre):not(.hljs) > code {
	color: #C9AE75; /* Change the old color so it seems less like an error */
	font-size: inherit;
}

/* Page Break : use <div class="page"/> to insert page break
-------------------------------------------------------- */
.page {
	page-break-after: always;
}

</style>

<script src="https://unpkg.com/mermaid/dist/mermaid.min.js"></script>
</head>
<body>
  <script>
    mermaid.initialize({
      startOnLoad: true,
      theme: document.body.classList.contains('vscode-dark') || document.body.classList.contains('vscode-high-contrast')
          ? 'dark'
          : 'default'
    });
  </script>
<h1 id="howard-user-guide">HOWARD User Guide</h1>
<p><img src="../images/icon.png" alt="HOWARD Graphical User Interface" title="HOWARD Graphical User Interface"></p>
<p>Highly Open Workflow for Annotation &amp; Ranking toward genomic variant Discovery</p>
<p>HOWARD annotates and prioritizes genetic variations, calculates and normalizes annotations, translates files in multiple formats (e.g. vcf, tsv, parquet) and generates variants statistics.</p>
<p>HOWARD annotation is mainly based on a build-in Parquet annotation method, and external tools such as BCFTOOLS, ANNOVAR, snpEff, Exomiser and Splice (see docs, automatically downloaded if needed). Parquet annotation uses annotation database in VCF or BED format, in mutliple file format: Parquet/duckdb, VCF, BED, TSV, CSV, TBL, JSON.</p>
<p>HOWARD calculation processes variants information to calculate new information, such as: harmonizes allele frequency (VAF), extracts Nomen (transcript, cNomen, pNomen...) from HGVS fields with an optional list of personalized transcripts, generates VaRank format barcode.</p>
<p>HOWARD prioritization algorithm uses profiles to flag variants (as passed or filtered), calculate a prioritization score, and automatically generate a comment for each variants (example: 'polymorphism identified in dbSNP. associated to Lung Cancer. Found in ClinVar database').Prioritization profiles are defined in a configuration file. A profile is defined as a list of annotation/value, using wildcards and comparison options (contains, lower than, greater than, equal...). Annotations fields may be quality values (usually from callers, such as 'GQ', 'DP') or other annotations fields provided by annotations tools, such as HOWARD itself (example: COSMIC, Clinvar, 1000genomes, PolyPhen, SIFT). Multiple profiles can be used simultaneously, which is useful to define multiple validation/prioritization levels (example: 'standard', 'stringent', 'rare variants', 'low allele frequency').</p>
<p>HOWARD translates VCF format into multiple formats (e.g. VCF, TSV, Parquet), by sorting variants using specific fields (example : 'prioritization score', 'allele frequency', 'gene symbol'), including/excluding annotations/fields, including/excluding variants, adding fixed columns.</p>
<p>HOWARD generates statistics files with a specific algorithm, snpEff and BCFTOOLS.</p>
<p>HOWARD is multithreaded through the number of variants and by database (data-scaling).</p>
<h2 id="table-of-contents">Table of Contents</h2>
<ul>
<li><a href="#installation">Installation</a>
<ul>
<li><a href="#python">Python</a>
<ul>
<li><a href="#quick-install">Quick install</a></li>
<li><a href="#quick-install">GUI install</a></li>
<li><a href="#configuration">Configuration</a></li>
</ul>
</li>
<li><a href="#docker">Docker</a>
<ul>
<li><a href="#quick-start">Quick start</a></li>
<li><a href="#setup-container">Setup container</a></li>
<li><a href="#command-line-interface">Command Line Interface</a></li>
<li><a href="#tests">Tests</a></li>
</ul>
</li>
</ul>
</li>
<li><a href="#databases">Databases</a>
<ul>
<li><a href="#databases-tool">Databases tool</a></li>
<li><a href="#home-made-databases">Home-made databases</a></li>
<li><a href="#databases-from-annovar-and-extann">Databases from Annovar and Extann</a></li>
</ul>
</li>
<li><a href="#tools">Tools</a>
<ul>
<li><a href="#parameters">Parameters</a></li>
<li><a href="#stats">Stats</a></li>
<li><a href="#convert">Convert</a>
<ul>
<li><a href="#csv-tsv-and-json-convert">CSV, TSV and JSON convert</a></li>
<li><a href="#parquet-and-duckdb-convert">Parquet and duckDB convert</a></li>
<li><a href="#bed-convert">BED convert</a></li>
<li><a href="#partitioning">Partitioning</a></li>
<li><a href="#explode-info-tags">Explode INFO Tags</a></li>
</ul>
</li>
<li><a href="#query">Query</a>
<ul>
<li><a href="#variants-file">Variants file</a>
<ul>
<li><a href="#loading-data">Loading data</a></li>
<li><a href="#external-file">External file</a></li>
</ul>
</li>
<li><a href="#other-files">Other files</a>
<ul>
<li><a href="#parquet-format">Parquet format</a></li>
<li><a href="#csv-format">CSV format</a></li>
</ul>
</li>
</ul>
</li>
<li><a href="#annotation">Annotation</a>
<ul>
<li><a href="#quick-annotation">Quick Annotation</a>
<ul>
<li><a href="#parquet-annotation-method">Parquet annotation method</a></li>
<li><a href="#external-tools-annotation">External tools annotation</a>
<ul>
<li><a href="#annovar-annotation">Annovar annotation</a></li>
<li><a href="#snpeff-annotation">snpEff annotation</a></li>
<li><a href="#exomiser-annotation">Exomiser annotation</a></li>
<li><a href="#splice-annotation">Splice annotation</a></li>
<li><a href="#bcftools-annotation">BCFTools annotation</a></li>
</ul>
</li>
<li><a href="#annotation-combination">Annotation combination</a></li>
</ul>
</li>
<li><a href="#annotation-parameters">Annotation parameters</a></li>
</ul>
</li>
<li><a href="#calculation">Calculation</a>
<ul>
<li><a href="#available-calculations">Available calculations</a></li>
<li><a href="#calculation-configuration-json-file">Calculation configu JSON file</a></li>
<li><a href="#build-in-calculations-examples">Build-in calculation examples</a>
<ul>
<li><a href="#variant-type">Variant type</a></li>
<li><a href="#hgvs-and-nomen-from-snpeff">HGVS and NOMEN from snpEff</a></li>
</ul>
</li>
</ul>
</li>
<li><a href="#prioritization">Prioritization</a>
<ul>
<li><a href="#prioritization-options">Prioritization options</a></li>
<li><a href="#prioritization-query">Prioritization query</a></li>
</ul>
</li>
<li><a href="#hgvs-annotation">HGVS annotation</a></li>
<li><a href="#process">Process</a>
<ul>
<li><a href="#process-with-options">Process with options</a></li>
<li><a href="#process-with-parameters-json-file">Process with Parameters JSON file</a></li>
</ul>
</li>
</ul>
</li>
</ul>
<h1 id="installation">Installation</h1>
<h2 id="python">Python</h2>
<h3 id="quick-install">Quick install</h3>
<p>Install HOWARD using Python Pip tool:</p>
<pre class="hljs"><code><div>python -m pip install -e .
</div></code></pre>
<p>Run HOWARD for help options:</p>
<pre class="hljs"><code><div>howard --help
</div></code></pre>
<h3 id="gui-install">GUI install</h3>
<p>Install HOWARD Graphical User Interface using Python Pip tool with supplementary packages:</p>
<pre class="hljs"><code><div>python -m pip install -r requirements-gui.txt
</div></code></pre>
<p>Run HOWARD Graphical User Interface as a tool:</p>
<pre class="hljs"><code><div>howard gui
</div></code></pre>
<p><img src="../images/howard-gui.png" alt="HOWARD Graphical User Interface" title="HOWARD Graphical User Interface"></p>
<h3 id="configuration">Configuration</h3>
<p>HOWARD Configuration JSON file defined default configuration regarding resources (e.g. threads, memory), settings (e.g. verbosity, temporary files), default folders (e.g. for databases) and paths to external tools.</p>
<p>See <a href="help.config.md">HOWARD Configuration JSON</a> for more information.</p>
<p>Configuration file example:</p>
<pre class="hljs"><code><div>{
  <span class="hljs-attr">"threads"</span>: <span class="hljs-number">8</span>,
  <span class="hljs-attr">"memory"</span>: <span class="hljs-literal">null</span>,
  <span class="hljs-attr">"verbosity"</span>: <span class="hljs-string">"warning"</span>,
  <span class="hljs-attr">"folders"</span>: {
    <span class="hljs-attr">"databases"</span>: {
      <span class="hljs-attr">"genomes"</span>: <span class="hljs-string">"~/howard/databases/genomes/current"</span>,
      <span class="hljs-attr">"annotations"</span>: [
        <span class="hljs-string">"~/howard/databases/annotations/current"</span>,
        <span class="hljs-string">"~/howard/databases/dbnsfp/current"</span>,
        <span class="hljs-string">"~/howard/databases/dbsnp/current"</span>
      ],
      <span class="hljs-attr">"parquet"</span>: [
        <span class="hljs-string">"~/howard/databases/annotations/current"</span>
        ],
      <span class="hljs-attr">"bcftools"</span>: [
        <span class="hljs-string">"~/howard/databases/annotations/current"</span>
        ],
      <span class="hljs-attr">"annovar"</span>: [
        <span class="hljs-string">"~/howard/databases/annovar/current"</span>
      ],
      <span class="hljs-attr">"snpeff"</span>: <span class="hljs-string">"~/howard/databases/snpeff/current"</span>,
      <span class="hljs-attr">"exomiser"</span>: <span class="hljs-string">"~/howard/databases/exomiser/current"</span>
    }
  },
  <span class="hljs-attr">"tools"</span>: {
    <span class="hljs-attr">"bcftools"</span>: <span class="hljs-string">"bcftools"</span>,
    <span class="hljs-attr">"bgzip"</span>: <span class="hljs-string">"bgzip"</span>,
    <span class="hljs-attr">"java"</span>: <span class="hljs-string">"java"</span>,
    <span class="hljs-attr">"snpeff"</span>: <span class="hljs-string">"~/howard/tools/snpeff/current/bin/snpEff.jar"</span>,
    <span class="hljs-attr">"snpsift"</span>: <span class="hljs-string">"~/howard/tools/snpeff/current/bin/SnpSift.jar"</span>,
    <span class="hljs-attr">"annovar"</span>: <span class="hljs-string">"~/howard/tools/annovar/current/bin/table_annovar.pl"</span>,
    <span class="hljs-attr">"exomiser"</span>: <span class="hljs-string">"~/howard/tools/exomiser/current/bin/exomiser-cli-14.0.0.jar"</span>,
    <span class="hljs-attr">"splice"</span>: {
      <span class="hljs-attr">"docker"</span>: {
        <span class="hljs-attr">"image"</span>: <span class="hljs-string">"bioinfochrustrasbourg/splice:0.2.1"</span>,
        <span class="hljs-attr">"entrypoint"</span>: <span class="hljs-string">"/bin/bash"</span>,
        <span class="hljs-attr">"options"</span>: <span class="hljs-literal">null</span>,
        <span class="hljs-attr">"command"</span>: <span class="hljs-literal">null</span>
      }
    },
    <span class="hljs-attr">"docker"</span>: <span class="hljs-string">"docker"</span>
  }
}

</div></code></pre>
<h3 id="external-tools">External tools</h3>
<p>In order to use external tools, mainly for annotation (e.g. Annovar, snpEff, Exomiser, Splice), they need to be installed (see doc of each tools). For Splice tool, a Docker image is automatically downloaded using configuration file.</p>
<h2 id="docker">Docker</h2>
<h3 id="quick-start">Quick Start</h3>
<p>In order to build images, launch default setup and create a persitent CLI (Command Line Inferface container), docker-compose command build images and launch services as containers.</p>
<pre class="hljs"><code><div>docker-compose up -d
</div></code></pre>
<p>The persitent CLI contains external tools, such as:</p>
<table>
<thead>
<tr>
<th>External tool</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="https://samtools.github.io/bcftools/">BCFTools</a></td>
<td>Utilities for variant calling and manipulating VCFs and BCFs</td>
</tr>
<tr>
<td><a href="https://pcingola.github.io/SnpEff/">snpEff</a></td>
<td>Genomic variant annotations, and functional effect prediction toolbox</td>
</tr>
<tr>
<td><a href="https://annovar.openbioinformatics.org/">Annovar</a></td>
<td>Efficient software tool to utilize update-to-date information to functionally annotate genetic variants</td>
</tr>
<tr>
<td><a href="https://www.sanger.ac.uk/tool/exomiser/">Exomiser</a></td>
<td>Program that finds potential disease-causing variants from whole-exome or whole-genome sequencing data</td>
</tr>
<tr>
<td><a href="https://hub.docker.com/r/bioinfochrustrasbourg/splice">Splice</a></td>
<td>Image to run SPiP and SpliceAI tools in an nextflow pipeline.</td>
</tr>
</tbody>
</table>
<h3 id="setup-container">Setup container</h3>
<p>Docker service HOWARD-setup creates HOWARD image and download useful databases to start with HOWARD tools.</p>
<p>List of databases downloaded in HOWARD setup for hg19 assembly (see <a href="#databases">Databases section</a> for more information):</p>
<ul>
<li>Genome</li>
<li>Annovar (<a href="https://genome.ucsc.edu/">refGene</a>, <a href="https://cancer.sanger.ac.uk/cosmic">COSMIC</a>)</li>
<li>snpEff</li>
<li>refSeq</li>
<li>dbNSFP</li>
<li>AlphaMissense</li>
<li>dnSNP</li>
</ul>
<p>To avoid databases download (see <a href="#databases">Databases section</a> to download manually), just start <a href="#command-line-interface">Command Line Interface</a></p>
<h3 id="command-line-interface">Command Line Interface</h3>
<p>A Command Line Interface container (HOWARD-CLI) is started with host data and databases folders mounted (by default both in ${HOME}/HOWARD folder). To manually start CLI container:</p>
<pre class="hljs"><code><div>docker-compose up -d HOWARD-CLI
</div></code></pre>
<p>To use HOWARD tools within HOWARD-CLI container:</p>
<pre class="hljs"><code><div>docker exec -ti HOWARD-CLI bash
howard --help
</div></code></pre>
<p>To run a command into HOWARD-CLI container:</p>
<pre class="hljs"><code><div>docker exec HOWARD-CLI &lt;howard command&gt;
</div></code></pre>
<p>Docker HOWARD-CLI container (Command Line Interface) can be used to execute commands.</p>
<blockquote>
<p>Example: Query of an existing VCF</p>
<pre class="hljs"><code><div>docker exec HOWARD-CLI \
   howard query \
      --input=/tool/tests/data/example.vcf.gz \
      --query='SELECT * FROM variants'
</div></code></pre>
</blockquote>
<blockquote>
<p>Example: VCF annotation using HOWARD-CLI (snpEff and ANNOVAR databases will be automatically downloaded), and query list of genes with HGVS</p>
<pre class="hljs"><code><div>docker exec --workdir=/tool HOWARD-CLI \
   howard process \
      --config=config/config.json \
      --param=config/param.json \
      --input=tests/data/example.vcf.gz \
      --output=/tmp/example.process.tsv \
      --explode_infos \
      --query='SELECT &quot;NOMEN&quot;, &quot;PZFlag&quot;, &quot;PZScore&quot;, &quot;PZComment&quot; FROM variants ORDER BY &quot;PZScore&quot; DESC'
</div></code></pre>
</blockquote>
<p>See <a href="help.md">HOWARD Help</a> for more options.</p>
<p>Let's play within Docker HOWARD-CLI service!</p>
<h3 id="tests">Tests</h3>
<p>In order to test HOWARD within Docker, use this command:</p>
<pre class="hljs"><code><div>docker exec -ti HOWARD-CLI bash
cd /tool
# Basic test
coverage run -m pytest .
# Debug test
coverage run -m pytest . -x -v --log-cli-level=DEBUG --capture=tee-sys
</div></code></pre>
<h1 id="databases">Databases</h1>
<h2 id="databases-tool">Databases tool</h2>
<p>Multiple databases can be automatically downloaded with databases tool, such as:</p>
<table>
<thead>
<tr>
<th>database</th>
<th>description</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="https://genome.ucsc.edu/cgi-bin/hgGateway">Genome</a></td>
<td>Genome Reference Consortium Human</td>
</tr>
<tr>
<td><a href="https://annovar.openbioinformatics.org/en/latest/">Annovar</a></td>
<td>ANNOVAR is an efficient software tool to utilize update-to-date information to functionally annotate genetic variants detected from diverse genomes</td>
</tr>
<tr>
<td><a href="https://pcingola.github.io/SnpEff/">snpEff</a></td>
<td>Genetic variant annotation, and functional effect prediction toolbox</td>
</tr>
<tr>
<td><a href="https://www.ncbi.nlm.nih.gov/refseq/">refSeq</a></td>
<td>A comprehensive, integrated, non-redundant, well-annotated set of reference sequences including genomic, transcript, and protein</td>
</tr>
<tr>
<td><a href="https://www.ncbi.nlm.nih.gov/snp/">dbSNP</a></td>
<td>dbSNP contains human single nucleotide variations, microsatellites, and small-scale insertions and deletions along with publication, population frequency, molecular consequence, and genomic and RefSeq mapping information for both common variations and clinical mutations</td>
</tr>
<tr>
<td><a href="https://sites.google.com/site/jpopgen/dbNSFP">dbNSFP</a></td>
<td>dbNSFP is a database developed for functional prediction and annotation of all potential non-synonymous single-nucleotide variants (nsSNVs) in the human genome</td>
</tr>
<tr>
<td><a href="https://github.com/google-deepmind/alphamissense">AlphaMissense</a></td>
<td>AlphaMissense model implementation</td>
</tr>
<tr>
<td><a href="https://www.sanger.ac.uk/tool/exomiser/">Exomiser</a></td>
<td>The Exomiser is a Java program that finds potential disease-causing variants from whole-exome or whole-genome sequencing data</td>
</tr>
</tbody>
</table>
<blockquote>
<p>Example: Download Multiple databases in the same time for assembly 'hg19' (can take a while):</p>
<pre class="hljs"><code><div>howard databases \
   --assembly=hg19 \
   --download-genomes=~/howard/databases/genomes/current --download-genomes-provider=UCSC --download-genomes-contig-regex='chr[0-9XYM]+$' \
   --download-annovar=~/howard/databases/annovar/current --download-annovar-files='refGene,cosmic70,nci60' \
   --download-snpeff=~/howard/databases/snpeff/current \
   --download-refseq=~/howard/databases/refseq/current --download-refseq-format-file='ncbiRefSeq.txt' \
   --download-dbnsfp=~/howard/databases/dbnsfp/current --download-dbnsfp-release='4.4a' --download-dbnsfp-subdatabases \
   --download-alphamissense=~/howard/databases/alphamissense/current \
   --download-exomiser=~/howard/databases/exomiser/current \
   --download-dbsnp=~/howard/databases/dbsnp/current --download-dbsnp-vcf --threads=8
</div></code></pre>
</blockquote>
<p>See <a href="help.md#databases-tool">HOWARD Help Databases tool</a> for more information.</p>
<h2 id="home-made-databases">Home-made Databases</h2>
<p>Databases can be generated using an home-made existing annotation file and <a href="#convert">HOWARD convert</a> tool. The home-made annotation file need to contain specific fields (depending on the annotation type):</p>
<ul>
<li>variant annotation: '#CHROM', 'POS', 'ALT', 'REF'</li>
<li>region annotation: '#CHROM', 'START', 'STOP'</li>
</ul>
<p>An home-made existing annotation file can be converted into multiple formats (e.g. Parquet, VCF, TSV), but it's strongly suggested to use Parquet format.</p>
<p>After convertion, the database file is associated with a 'header' file ('.hdr'), in VCF header format, to describe annotations within the database. Use the 'header' file to describe annotation fields/columns present in the existing file. An Home-made annotation file in VCF format which is converted in another format will keep all annotation information from the initial VCF header.</p>
<p>Note that a VCF can be directly used as a database (annotation field information within the header of the VCF file). Also, an home-made existing annotation file can be used as a database, but will not be totaly compliant due to the lack of annotation information ('header' will be generated by default).</p>
<p>See <a href="help.md#convert-tool">HOWARD Help Convert tool</a> for more information.</p>
<h2 id="databases-from-annovar-and-extann">Databases from Annovar and Extann</h2>
<p>See <a href="help.md#databases">HOWARD Help Databases tool</a> tool for more information aboout the tool, and <a href="help.param.databases.md">HOWARD Parameters Databases</a> help for generate databases annotation file from Annovar databases (<a href="help.param.databases.md#databasesfrom_annovar">from Annovar</a>, under development) and from Gene annotation file (<a href="help.param.databases.md#databasesfrom_extann">from Extann</a>).</p>
<h1 id="tools">Tools</h1>
<p>HOWARD annotates and prioritizes genetic variations, calculates and normalizes annotations, convert on multiple formats, query variations and generates statistics. These tools require options or a <a href="help.param.md"> Parameters JSON</a> file.</p>
<h2 id="parameters">Parameters</h2>
<p>HOWARD Parameters JSON file defined parameters to process annotations, prioritization, calculations, convertions and queries. Use this parameters file to configure tools, instead of options or as a main configuration (options will replace parameters in JSON file).</p>
<p>See <a href="help.param.md">HOWARD Parameters JSON</a> for more information.</p>
<blockquote>
<p>Example: Use parameters JSON file with query tool</p>
<pre class="hljs"><code><div>howard query \
   --input=tests/data/example.vcf.gz \
   --param=config/param.json
</div></code></pre>
<pre class="hljs"><code><div>  #CHROM       POS REF ALT                   INFO
<span class="hljs-number">0</span>   chr1     <span class="hljs-number">28736</span>   A   C      CLNSIG=pathogenic
<span class="hljs-number">1</span>   chr1     <span class="hljs-number">35144</span>   A   C  CLNSIG=non-pathogenic
<span class="hljs-number">2</span>   chr1     <span class="hljs-number">69101</span>   A   G                  DP=<span class="hljs-number">50</span>
<span class="hljs-number">3</span>   chr1    <span class="hljs-number">768251</span>   A   G                   None
<span class="hljs-number">4</span>   chr1    <span class="hljs-number">768252</span>   A   G                   None
<span class="hljs-number">5</span>   chr1    <span class="hljs-number">768253</span>   A   G                   None
<span class="hljs-number">6</span>   chr7  <span class="hljs-number">55249063</span>   G   A                 DP=<span class="hljs-number">125</span>
</div></code></pre>
</blockquote>
<blockquote>
<p>Example: Use parameters JSON file with query tool, and add an option to change the query (list of chromosomes)</p>
<pre class="hljs"><code><div>howard query \
   --input=tests/data/example.vcf.gz \
   --param=config/param.json \
   --query=&quot;SELECT distinct(\&quot;#CHROM\&quot;) as 'chromosomes' FROM variants&quot;
</div></code></pre>
<pre class="hljs"><code><div>  chromosomes
<span class="hljs-number">0</span>        chr7
<span class="hljs-number">1</span>        chr1
</div></code></pre>
</blockquote>
<blockquote>
<p>Example: Parameters JSON file with multiple options for tools</p>
<pre class="hljs"><code><div>{
  <span class="hljs-attr">"annotation"</span>: {
    <span class="hljs-attr">"parquet"</span>: {
      <span class="hljs-attr">"annotations"</span>: {
        <span class="hljs-attr">"tests/databases/annotations/current/hg19/avsnp150.parquet"</span>: {
          <span class="hljs-attr">"INFO"</span>: <span class="hljs-literal">null</span>
        },
        <span class="hljs-attr">"tests/databases/annotations/current/hg19/dbnsfp42a.parquet"</span>: {
          <span class="hljs-attr">"INFO"</span>: <span class="hljs-literal">null</span>
        },
        <span class="hljs-attr">"tests/databases/annotations/current/hg19/gnomad211_genome.parquet"</span>: {
          <span class="hljs-attr">"INFO"</span>: <span class="hljs-literal">null</span>
        }
      }
    },
    <span class="hljs-attr">"bcftools"</span>: {
      <span class="hljs-attr">"annotations"</span>: {
        <span class="hljs-attr">"tests/databases/annotations/current/hg19/cosmic70.vcf.gz"</span>: {
          <span class="hljs-attr">"INFO"</span>: <span class="hljs-literal">null</span>
        }
      }
    },
    <span class="hljs-attr">"snpeff"</span>: {
      <span class="hljs-attr">"options"</span>: <span class="hljs-string">"-lof -hgvs -oicr -noShiftHgvs -spliceSiteSize 3 "</span>
    },
    <span class="hljs-attr">"snpsift"</span>: {
      <span class="hljs-attr">"annotations"</span>: {
        <span class="hljs-attr">"tests/databases/annotations/current/hg19/cosmic70.vcf.gz"</span>: {
          <span class="hljs-attr">"INFO"</span>: <span class="hljs-literal">null</span>
        }
      }
    },
    <span class="hljs-attr">"annovar"</span>: {
      <span class="hljs-attr">"annotations"</span>: {
        <span class="hljs-attr">"refGene"</span>: {
          <span class="hljs-attr">"INFO"</span>: <span class="hljs-literal">null</span>
        }
      },
      <span class="hljs-attr">"options"</span>: {
        <span class="hljs-attr">"genebase"</span>: <span class="hljs-string">"-hgvs -splicing_threshold 3 "</span>,
        <span class="hljs-attr">"intronhgvs"</span>: <span class="hljs-number">10</span>
      }
    }
  },
  <span class="hljs-attr">"calculation"</span>: {
    <span class="hljs-attr">"calculations"</span>: {
      <span class="hljs-attr">"vartype"</span>: <span class="hljs-literal">null</span>,
      <span class="hljs-attr">"snpeff_hgvs"</span>: <span class="hljs-literal">null</span>,
      <span class="hljs-attr">"VAF"</span>: <span class="hljs-string">""</span>,
      <span class="hljs-attr">"NOMEN"</span>: {
        <span class="hljs-attr">"options"</span>: {
          <span class="hljs-attr">"hgvs_field"</span>: <span class="hljs-string">"snpeff_hgvs"</span>,
          <span class="hljs-attr">"transcripts"</span>: <span class="hljs-string">"tests/data/transcripts.tsv"</span>
        }
      }
    },
    <span class="hljs-attr">"config/calculations_config.json"</span>: <span class="hljs-string">"config/calculations_config.json"</span>
  },
  <span class="hljs-attr">"prioritization"</span>: {
    <span class="hljs-attr">"profiles"</span>: [<span class="hljs-string">"default"</span>, <span class="hljs-string">"GERMLINE"</span>],
    <span class="hljs-attr">"prioritization_config"</span>: <span class="hljs-string">"config/prioritization_profiles.json"</span>,
    <span class="hljs-attr">"pzfields"</span>: [<span class="hljs-string">"PZScore"</span>, <span class="hljs-string">"PZFlag"</span>, <span class="hljs-string">"PZComment"</span>],
    <span class="hljs-attr">"prioritization_score_mode"</span>: <span class="hljs-string">"VaRank"</span>
  },
  <span class="hljs-attr">"hgvs"</span>: {
    <span class="hljs-attr">"full_format"</span>: <span class="hljs-literal">true</span>,
    <span class="hljs-attr">"use_exon"</span>: <span class="hljs-literal">true</span>
  },
  <span class="hljs-attr">"stats"</span>: {
    <span class="hljs-attr">"stats_md"</span>: <span class="hljs-literal">null</span>,
    <span class="hljs-attr">"stats_json"</span>: <span class="hljs-literal">null</span>
  },
  <span class="hljs-attr">"query"</span>: {
    <span class="hljs-attr">"query"</span>: <span class="hljs-string">"SELECT \"#CHROM\", POS, REF, ALT, INFO FROM variants"</span>,
    <span class="hljs-attr">"query_limit"</span>: <span class="hljs-number">10</span>,
    <span class="hljs-attr">"query_print_mode"</span>: <span class="hljs-string">"default"</span>
  },
  <span class="hljs-attr">"explode_infos"</span>: {
    <span class="hljs-attr">"explode_infos"</span>: <span class="hljs-literal">false</span>,
    <span class="hljs-attr">"explode_infos_prefix"</span>: <span class="hljs-string">""</span>,
    <span class="hljs-attr">"explode_infos_fields"</span>: <span class="hljs-literal">null</span>
  },
  <span class="hljs-attr">"export"</span>: {
    <span class="hljs-attr">"header_in_output"</span>: <span class="hljs-literal">false</span>,
    <span class="hljs-attr">"parquet_partitions"</span>: <span class="hljs-literal">null</span>,
    <span class="hljs-attr">"order_by"</span>: <span class="hljs-literal">null</span>
  },
  <span class="hljs-attr">"threads"</span>: <span class="hljs-number">8</span>
}
</div></code></pre>
</blockquote>
<p>Moreover, a transcripts file can be defined, especially to select NOMEN from a list of HGVS annotation (see <a href="#calculation">Calculation</a> and <a href="#hgvs-and-nomen-from-snpeff">HGVS and NOMEN from snpEff</a>). This file is a tab-delimited with 'transcript' as first column and 'gene' a second column. For a gene, transcripts of reference are ordered (first is priority, e.g. 'NM_001346897' has prior over 'NM_005228').</p>
<blockquote>
<p>Example: Transcripts file in tab-delimited format with column 'transcript' and column 'gene'</p>
<pre class="hljs"><code><div>NR_024540	    WASH7P
NR_036266	    MIR1302<span class="hljs-number">-9</span>
NM_001346897	EGFR
NM_005228	    EGFR
</div></code></pre>
</blockquote>
<h2 id="stats">Stats</h2>
<p>Generates statistics on genetic variations, such as number of variants, number of samples, statistics by chromosome, genotypes by samples, annotations. Theses statsitics can be applied to VCF files from all database annotation file formats. Statistics can be wrote into files in Markdown and JSON format (resp. <code>--stats_md</code> and <code>--stats_json</code> parameter).</p>
<p>See <a href="help.md#stats-tool">HOWARD Help Stats tool</a> for more information.</p>
<blockquote>
<p>Example: Show example VCF statistics and brief overview</p>
<pre class="hljs"><code><div>howard stats \
   --input=tests/data/example.vcf.gz
</div></code></pre>
</blockquote>
<blockquote>
<p>Example: Show example VCF statistics and generate a file in JSON and Markdown formats (extract)</p>
<pre class="hljs"><code><div>howard stats \
   --input=tests/data/example.vcf.gz \
   --stats_json=/tmp/stats.json \
   --stats_md=/tmp/stats.md

cat /tmp/stats.json /tmp/stats.md
</div></code></pre>
<pre class="hljs"><code><div>{
    <span class="hljs-attr">"Infos"</span>: {
        <span class="hljs-attr">"Input file"</span>: <span class="hljs-string">"tests/data/example.vcf.gz"</span>,
        <span class="hljs-attr">"Number of variants"</span>: <span class="hljs-number">7</span>,
        <span class="hljs-attr">"Number of samples"</span>: <span class="hljs-number">4</span>,
        <span class="hljs-attr">"Number of INFO fields"</span>: <span class="hljs-number">5</span>,
        <span class="hljs-attr">"Number of FORMAT fields"</span>: <span class="hljs-number">7</span>
    },
    <span class="hljs-attr">"Variants"</span>: {
        <span class="hljs-attr">"Number of variants by chromosome"</span>: {
            <span class="hljs-attr">"1"</span>: {
                <span class="hljs-attr">"CHROM"</span>: <span class="hljs-string">"chr1"</span>,
                <span class="hljs-attr">"count"</span>: <span class="hljs-number">6</span>,
                <span class="hljs-attr">"percent"</span>: <span class="hljs-number">0.8571428571428571</span>
            },
            <span class="hljs-attr">"0"</span>: {
                <span class="hljs-attr">"CHROM"</span>: <span class="hljs-string">"chr7"</span>,
                <span class="hljs-attr">"count"</span>: <span class="hljs-number">1</span>,
                <span class="hljs-attr">"percent"</span>: <span class="hljs-number">0.14285714285714285</span>
            }
        }
    }
}
</div></code></pre>
<pre class="hljs"><code><div>...
<span class="hljs-section">## Variants</span>
<span class="hljs-section">### Number of variants by chromosome</span>
| CHROM   |   count |   percent |
|:--------|--------:|----------:|
| chr1    |       6 |  0.857143 |
| chr7    |       1 |  0.142857 |
<span class="hljs-section">### Counts</span>
| Type   |   count |
|:-------|--------:|
| Total  |       7 |
| SNV    |       7 |
| MNV    |       0 |
| InDel  |       0 |
...
</div></code></pre>
</blockquote>
<blockquote>
<p>Example of statistics in Markdown output</p>
</blockquote>
<ul>
<li>Input file: tests/data/example.vcf.gz</li>
<li>Number of variants: 7</li>
<li>Number of samples: 4</li>
<li>Number of INFO fields: 5</li>
<li>Number of FORMAT fields: 7</li>
</ul>
<table>
<thead>
<tr>
<th style="text-align:left">CHROM</th>
<th style="text-align:right">count</th>
<th style="text-align:right">percent</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left">chr1</td>
<td style="text-align:right">6</td>
<td style="text-align:right">0.857143</td>
</tr>
<tr>
<td style="text-align:left">chr7</td>
<td style="text-align:right">1</td>
<td style="text-align:right">0.142857</td>
</tr>
</tbody>
</table>
<table>
<thead>
<tr>
<th style="text-align:left">Type</th>
<th style="text-align:right">count</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left">Total</td>
<td style="text-align:right">7</td>
</tr>
<tr>
<td style="text-align:left">SNV</td>
<td style="text-align:right">7</td>
</tr>
<tr>
<td style="text-align:left">MNV</td>
<td style="text-align:right">0</td>
</tr>
<tr>
<td style="text-align:left">InDel</td>
<td style="text-align:right">0</td>
</tr>
</tbody>
</table>
<h2 id="convert">Convert</h2>
<p>Convert genetic variations file to another format. Multiple format are available, such as usual and official VCF format, but also other formats such as TSV, CSV, TBL, JSON and Parquet/duckDB. These formats need a header '.hdr' file to take advantage of the power of howard (especially through INFO/tag definition), and using howard convert tool automatically generate header file fo futher use (otherwise, an default '.hdr' file is generated).</p>
<p>Multiple options are available, such as explode VCF INFO/tags (parameter <code>--explode_infos</code>, see <a href="help.md#explode-infos">HOWARD Help query - Explode infos</a>), order by columns, include header within file (only TSV and CSV format), or use partitioning into multiple files within a folder. See <a href="help.md#convert-tool">HOWARD Help Convert tool</a> for more information.</p>
<h3 id="csv-tsv-and-json-convert">CSV, TSV and JSON convert</h3>
<p>To convert a file (multiple formats) into another flat file, such as CSV (tab-delimiter) and TSV (comma-delimiter), or JSON format, simply name output file with desired extension. Use <code>.gz</code> extension to compress file.</p>
<blockquote>
<p>Example: Convert VCF into TSV and show output file</p>
<pre class="hljs"><code><div>howard convert \
   --input=tests/data/example.vcf.gz \
   --output=/tmp/example.tsv
cat /tmp/example.tsv
</div></code></pre>
</blockquote>
<blockquote>
<p>Example: Convert TSV into VCF and show output file</p>
<pre class="hljs"><code><div>howard convert \
   --input=tests/data/example.tsv \
   --output=/tmp/example.vcf
cat /tmp/example.vcf
</div></code></pre>
</blockquote>
<blockquote>
<p>Example: Convert VCF into CSV and compress file</p>
<pre class="hljs"><code><div>howard convert \
   --input=tests/data/example.vcf.gz \
   --output=/tmp/example.csv.gz
</div></code></pre>
</blockquote>
<blockquote>
<p>Example: Convert VCF into JSON and compress file</p>
<pre class="hljs"><code><div>howard convert \
   --input=tests/data/example.vcf.gz \
   --output=/tmp/example.json.gz
</div></code></pre>
</blockquote>
<h3 id="parquet-and-duckdb-convert">Parquet and duckDB convert</h3>
<p>Files can be format into Parquet and duckDB format. For duckDB format, a duckDB database will be created with a <code>variants</code> table.</p>
<blockquote>
<p>Example: Convert VCF into parquet</p>
<pre class="hljs"><code><div>howard convert \
   --input=tests/data/example.vcf.gz \
   --output=/tmp/example.parquet
</div></code></pre>
</blockquote>
<blockquote>
<p>Example: Convert VCF into duckDB</p>
<pre class="hljs"><code><div>howard convert \
   --input=tests/data/example.vcf.gz \
   --output=/tmp/example.duckdb
</div></code></pre>
</blockquote>
<h3 id="bed-convert">BED convert</h3>
<p>To convert into BED format, input file needs mandatory columns chrommosome &quot;#CHROM&quot;, and positions columns &quot;START&quot;/&quot;END&quot; or &quot;POS&quot; (corresponfing to regions with a uniq nucleotide). Header will be automatically included to decribe all columns.</p>
<blockquote>
<p>Example: Convert BED in TSV format into BED format</p>
<pre class="hljs"><code><div>howard convert \
   --input=tests/data/example.bed.tsv \
   --output=/tmp/example.bed
</div></code></pre>
</blockquote>
<blockquote>
<p>Example: Convert BED in Parquet format into BED format</p>
<pre class="hljs"><code><div>howard convert \
   --input=tests/data/example.bed.parquet \
   --output=/tmp/example.bed
</div></code></pre>
</blockquote>
<p>HOWARD input file format does not allow BED format. To read a BED file and export into a CSV or Parquet file format, see <a href="#bed-format">Query BED format</a> section.</p>
<h3 id="partitioning">Partitioning</h3>
<p>Partitioning (or Hive partitioning) is a partitioning strategy that is used to split a table into multiple files based on partition keys. The files are organized into folders. Within each folder, the partition key has a value that is determined by the name of the folder (see <a href="https://duckdb.org/docs/data/partitioning/hive_partitioning.html">duckDB hive partitioning</a>).</p>
<p>Simply list columns as keys to process partitioning. Use 'None' (string) for NO partition but split parquet files into a folder. The partitioning is available for all format (e.g. Parquet, TSV, JSON, except duckDB format), by naming output file with desired extension. This option is faster parallel writing, but memory consuming, and also is faster reading.</p>
<blockquote>
<p>Example: Convert VCF into partitioned Parquet and show tree structure</p>
<pre class="hljs"><code><div>howard convert \
   --input=tests/data/example.vcf.gz \
   --output=/tmp/example.partitioned.parquet \
   --parquet_partitions=&quot;#CHROM&quot;

tree /tmp/example.partitioned.parquet
</div></code></pre>
<pre class="hljs"><code><div>/tmp/example.partitioned.parquet
├── #CHROM=chr1
│   └── fe61bf182de640f8840270a527aa1582-0.parquet
└── #CHROM=chr7
    └── fe61bf182de640f8840270a527aa1582-0.parquet
</div></code></pre>
</blockquote>
<blockquote>
<p>Example: Convert VCF into partitioned TSV (compressed) and show tree structure (files are named with <code>.csv</code> extension, but are tab-delimited and compressed)</p>
<pre class="hljs"><code><div>howard convert \
   --input=tests/data/example.vcf.gz \
   --output=/tmp/example.partitioned.tsv.gz \
   --parquet_partitions=&quot;#CHROM,REF&quot;

tree /tmp/example.partitioned.tsv.gz
</div></code></pre>
<pre class="hljs"><code><div>/tmp/example.partitioned.tsv
├── #CHROM=chr1
│   └── REF=A
│       └── data_0.csv
└── #CHROM=chr7
    └── REF=G
        └── data_0.csv
</div></code></pre>
</blockquote>
<h3 id="explode-info-tags">Explode INFO tags</h3>
<p>Use <code>--explode_infos</code> parameter to extract all INFO tags (i.e. annotations) into columns (see <a href="help.md#explode-infos">HOWARD Help query - Explode infos</a>).</p>
<blockquote>
<p>Example: Convert VCF into TSV, export INFO/tags into columns, and show output file</p>
<pre class="hljs"><code><div>howard convert \
   --input=tests/data/example.vcf.gz \
   --explode_infos \
   --output=/tmp/example.tsv

cut /tmp/example.tsv -f1-4,7,15
</div></code></pre>
<pre class="hljs"><code><div><span class="hljs-meta">#CHROM  POS       REF  ALT  FILTER  CLNSIG</span>
chr1    <span class="hljs-number">28736</span>     A    C    PASS    pathogenic
chr1    <span class="hljs-number">35144</span>     A    C    PASS    non-pathogenic
chr1    <span class="hljs-number">69101</span>     A    G    PASS
chr1    <span class="hljs-number">768251</span>    A    G    PASS
...
</div></code></pre>
</blockquote>
<h2 id="query">Query</h2>
<p>Query tool provides a simple way to query genetic variations in SQL format. Data are loaded into 'variants' table from various formats (e.g. VCF, TSV, Parquet...). Using <code>--explode_infos</code> allows querying on INFO/tag annotations. SQL query can also use external file within the request, such as a Parquet file(s), or TSV files.</p>
<p>See <a href="help.md#query-tool">HOWARD Help Query tool</a> for more information.</p>
<h3 id="variants-file">Variants file</h3>
<h4 id="loading-data">Loading data</h4>
<p>Query tool is able to read variants (i.e. VCF) or regions files (i.e. BED) files, in various format (e.g. VCF, BED, Parquet, TSV, JSON), using <code>--input</code> parameter. This allows to load data to perfom actions, such as explode VCF INFO/tags (parameter <code>--explode_infos</code>, see <a href="help.md#explode-infos">HOWARD Help query - Explode infos</a>) in columns to be easier querying. Each columns format (e.g. string, integer) are automatically detected to be used in a SQL query.</p>
<blockquote>
<p>Example: Select variants in VCF with REF and POS fields filter</p>
<pre class="hljs"><code><div>howard query \
   --input=tests/data/example.vcf.gz \
   --query=&quot;SELECT * FROM variants WHERE REF = 'A' AND POS &lt; 100000&quot;
</div></code></pre>
</blockquote>
<blockquote>
<p>Example: Select variants in VCF with INFO Tags criteria filters</p>
<pre class="hljs"><code><div>howard query \
   --input=tests/data/example.vcf.gz \
   --explode_infos \
   --query='SELECT &quot;#CHROM&quot;, POS, REF, ALT, DP, CLNSIG, sample2, sample3 
            FROM variants 
            WHERE DP &gt;= 50 OR CLNSIG NOT NULL 
            ORDER BY CLNSIG DESC, DP DESC'
</div></code></pre>
</blockquote>
<blockquote>
<p>Example: Select variants in VCF and generate VCF output with variants</p>
<pre class="hljs"><code><div>howard query \
   --input=tests/data/example.vcf.gz \
   --output=/tmp/example.filtered.vcf \
   --explode_infos \
   --query='SELECT &quot;#CHROM&quot;, POS, REF, ALT, QUAL, FILTER, INFO 
            FROM variants 
            WHERE DP &gt;= 50 OR CLNSIG NOT NULL'
</div></code></pre>
</blockquote>
<h4 id="external-file">External file</h4>
<p>Variants files can be used directly within the query, espacially if they already contain variants information (e.g. &quot;#CHROM&quot;, &quot;POS&quot;, &quot;REF&quot;, &quot;ALT&quot;) and annotations as columns.</p>
<blockquote>
<p>Example: Query a Parquet file with specific columns (e.g. from VCF convertion to Parquet)</p>
<pre class="hljs"><code><div>howard query \
   --query=&quot;SELECT * \
            FROM 'tests/databases/annotations/current/hg19/dbnsfp42a.parquet' \
            WHERE \&quot;INFO/Interpro_domain\&quot; NOT NULL \
            ORDER BY \&quot;INFO/SiPhy_29way_logOdds_rankscore\&quot; DESC&quot;
</div></code></pre>
</blockquote>
<blockquote>
<p>Example: Query multiple Parquet files, merge INFO columns, and extract as TSV (in VCF format)</p>
<pre class="hljs"><code><div>howard query \
   --query=&quot;
      SELECT \
         \&quot;#CHROM\&quot; AS \&quot;#CHROM\&quot;, \
         POS AS POS, \
         '' AS ID, \
         REF AS REF, \
         ALT AS ALT, \
         '' AS QUAL, \
         '' AS FILTER, \
         STRING_AGG(INFO, ';') AS INFO \
      FROM 'tests/databases/annotations/current/hg19/*.parquet' \
      GROUP BY \&quot;#CHROM\&quot;, POS, REF, ALT&quot; \
   --output=/tmp/full_annotation.tsv \
   --include_header
</div></code></pre>
</blockquote>
<h3 id="other-files">Other files</h3>
<p>Whatever the external file format, if it is compatible with duckDB, query tool is able to query data (see <a href="https://duckdb.org/docs/sql/statements/select">duckDB Select Statement</a>).</p>
<h4 id="parquet-format">Parquet format</h4>
<p>Simply use Parquet file path within the query (as descibe above).</p>
<blockquote>
<p>Example: Query a Parquet file with specific columns (e.g. from VCF convertion to Parquet)</p>
<pre class="hljs"><code><div>howard query \
   --query=&quot;SELECT * \
            FROM 'tests/databases/annotations/current/hg19/dbnsfp42a.parquet' \
            WHERE \&quot;INFO/Interpro_domain\&quot; NOT NULL \
            ORDER BY \&quot;INFO/SiPhy_29way_logOdds_rankscore\&quot; DESC&quot;
</div></code></pre>
</blockquote>
<h4 id="csv-format">CSV format</h4>
<p>Use duckDB function <code>read_csv_auto</code> to read a TSV file format as a table. See <a href="https://duckdb.org/docs/data/csv/overview.html">duckDB CSV import</a> for more information.</p>
<blockquote>
<p>Example: Query a TSV file</p>
<pre class="hljs"><code><div>howard query \
   --query=&quot;SELECT * FROM read_csv_auto('tests/data/transcripts.tsv')&quot;
</div></code></pre>
</blockquote>
<blockquote>
<p>Example: Query a TSV file with columns struct</p>
<pre class="hljs"><code><div>howard query \
   --query=&quot;SELECT * FROM read_csv_auto('tests/data/transcripts.tsv', columns={'transcript': 'VARCHAR','gene': 'VARCHAR'})&quot;
</div></code></pre>
</blockquote>
<h4 id="bed-format">BED format</h4>
<p>In order to read a BED file, create a query (using appropiate columns), and export file into a desired format.</p>
<blockquote>
<p>Example: Read a BED file and export in Parquet format</p>
<pre class="hljs"><code><div>howard query \
   --query=&quot;SELECT * FROM read_csv_auto('tests/data/example.bed', columns={'#CHROM': 'VARCHAR', 'START': 'INTEGER', 'END': 'INTEGER'})&quot; \
   --output=/tmp/example.bed.parquet
</div></code></pre>
</blockquote>
<blockquote>
<p>Example: Convert a BED file in a Parquet format into a BED file format</p>
<pre class="hljs"><code><div>howard convert \
   --input=tests/data/example.bed.parquet \
   --output=/tmp/example.bed

cat /tmp/example.bed
</div></code></pre>
<pre class="hljs"><code><div><span class="hljs-meta">#CHROM	START	END</span>
chr1	<span class="hljs-number">28735</span>	<span class="hljs-number">69101</span>
chr1	<span class="hljs-number">768250</span>	<span class="hljs-number">768253</span>
chr7	<span class="hljs-number">55249060</span>	<span class="hljs-number">55249069</span>
</div></code></pre>
</blockquote>
<p>A BED file can be filtered using positions or other columns such as gene names, or transcripts.</p>
<blockquote>
<p>Example: Filter a BED using positions</p>
<pre class="hljs"><code><div>howard query \
   --input=tests/data/example.bed.parquet \
   --query=&quot;SELECT * \
            FROM variants \
            WHERE \
               \&quot;#CHROM\&quot; = 'chr1' and \
               ( \
                  (\&quot;START\&quot;&gt;28000 and \&quot;END\&quot;&lt;70000) or \
                  (\&quot;START\&quot;&gt;760000 and \&quot;END\&quot;&lt;770000) \
               )&quot;
</div></code></pre>
<pre class="hljs"><code><div>  <span class="hljs-meta">#CHROM   START     END</span>
<span class="hljs-number">0</span>   chr1   <span class="hljs-number">28735</span>   <span class="hljs-number">69101</span>
<span class="hljs-number">1</span>   chr1  <span class="hljs-number">768250</span>  <span class="hljs-number">768253</span>
</div></code></pre>
</blockquote>
<h3 id="extract-variants">Extract variants</h3>
<p>In order to extract variants from a VCF file, without annotations and samples, use a query to construct the VCF.</p>
<blockquote>
<p>Example: Extract variants onnly</p>
<pre class="hljs"><code><div>howard query \
   --input=tests/data/example.vcf.gz \
   --output=/tmp/example.vcf.gz \
   --query=&quot;SELECT \&quot;#CHROM\&quot;, POS, ID, REF, ALT, QUAL, FILTER, '.' AS INFO \
            FROM variants&quot;
</div></code></pre>
</blockquote>
<h2 id="annotation">Annotation</h2>
<p>Annotation is mainly based on a build-in Parquet annotation method, using annotation database file (in multiple format such as Parquet, duckdb, VCF, BED, TSV, JSON).</p>
<p>See <a href="help.md#annotation-tool">HOWARD Help Annotation tool</a> for more information.</p>
<p>These annotation databases can be automatically downloaded using <a href="help.md#databases-tool">HOWARD Databases tool</a> and manually generated using existing annotation files and <a href="help.md#convert-tool">HOWARD Convert tool</a>. Annotation databases need a header file (<code>.hdr</code>) to describe each annotation in the database. However, a default header will be generated if no header file is associated to the annotation database file.</p>
<p>Moreover, some external annotation tools are integrated into HOWARD to extend annotation with their own options and databases.</p>
<p>HOWARD annotation tool can use annotation databases files in 2 differents ways: <a href="#quick-annotation">Quick annotation</a> and <a href="#annotation-parameters">Annotation Parameters</a> JSON file.</p>
<h3 id="quick-annotation">Quick annotation</h3>
<h4 id="parquet-annotation-method">Parquet annotation method</h4>
<p>Quick annotation allows to annotates by simply listing annotation databases (in multiple format).</p>
<h5 id="parquet-annotation-with-path">Parquet annotation with path</h5>
<p>These annotation databases are defined with their full path (e.g. <code>/full/path/to/my.database.parquet</code>) or relative path (e.g. <code>databases/my.database.parquet</code>). A list (separator <code>:</code> or <code>+</code>) of annotation databases files can be used.</p>
<blockquote>
<p>Example: VCF annotation with full path Parquet databases</p>
<pre class="hljs"><code><div>howard annotation \
   --input=tests/data/example.vcf.gz \
   --annotations='/tool/tests/databases/annotations/current/hg19/dbnsfp42a.parquet' \
   --output=/tmp/example.howard.vcf.gz
</div></code></pre>
</blockquote>
<blockquote>
<p>Example: VCF annotation with relative path VCF databases</p>
<pre class="hljs"><code><div>cd /tool
howard annotation \
   --input=tests/data/example.vcf.gz \
   --annotations='tests/databases/annotations/current/hg19/cosmic70.vcf.gz' \
   --output=/tmp/example.howard.vcf.gz
</div></code></pre>
</blockquote>
<blockquote>
<p>Example: VCF annotation with relative path BED databases</p>
<pre class="hljs"><code><div>cd /tool
howard annotation \
   --input=tests/data/example.vcf.gz \
   --annotations='tests/databases/annotations/current/hg19/refGene.bed.gz' \
   --output=/tmp/example.howard.vcf.gz
</div></code></pre>
</blockquote>
<blockquote>
<p>Example: VCF annotation with 3 annotation databases files</p>
<pre class="hljs"><code><div>howard annotation \
   --input=tests/data/example.vcf.gz \
   --annotations='/tool/tests/databases/annotations/current/hg19/dbnsfp42a.parquet+tests/databases/annotations/current/hg19/cosmic70.vcf.gz+tests/databases/annotations/current/hg19/refGene.bed.gz' \
   --output=/tmp/example.howard.vcf.gz
</div></code></pre>
</blockquote>
<h5 id="parquet-annotation-with-annotation-folder">Parquet annotation with annotation folder</h5>
<p>If annotation folder is configured in <a href="help.config.md">HOWARD Configuration JSON</a>, just mention the annotation database basename file. Annotation database file will be found (depending of the assembly).</p>
<blockquote>
<p>Example: VCF annotation with Parquet and VCF databases, with annotation database defined in JSON configuration (as a string)</p>
<pre class="hljs"><code><div>howard annotation \
   --input=tests/data/example.vcf.gz \
   --annotations='dbnsfp42a.parquet,cosmic70.vcf.gz' \
   --config='{&quot;folders&quot;: {&quot;databases&quot;: {&quot;annotations&quot;: [&quot;/tool/tests/databases/annotations/current&quot;]}}}' \
   --output=/tmp/example.howard.vcf.gz 
</div></code></pre>
</blockquote>
<h5 id="full-annotation">Full annotation</h5>
<p>In order to annotate with all available annotation databases, the keyword <code>ALL</code> will auto-detect files in the databases annotation folder. The option <code>format</code> (defaut <code>parquet</code>) can filter annotation databases by listing (separator <code>+</code>) desired formats (such as <code>parquet</code>, <code>vcf</code>). The option <code>release</code> (default <code>current</code>) is able to scan annotation databases in one or more specific releases in a list (separator <code>+</code>). See <a href="help.config.md#foldersdatabases">HOWARD Configuration JSON - Folders - Databases</a> for more information about databases structure.</p>
<blockquote>
<p>Example: VCF annotation with all available database annotation files in Parquet format (within the database annotation folder in configuration):</p>
<pre class="hljs"><code><div>howard annotation \
   --input=tests/data/example.vcf.gz \
   --assembly='hg19' \
   --annotations='ALL:format=parquet+vcf:release=current' \
   --config='{&quot;folders&quot;: {&quot;databases&quot;: {&quot;annotations&quot;: [&quot;/tool/tests/databases/annotations/current&quot;]}}}' \
   --output=/tmp/example.howard.tsv
</div></code></pre>
</blockquote>
<h4 id="external-tools-annotation">External tools annotation</h4>
<p>External annotation tools are also available, such as BCFTOOLS, Annovar, snpEff, Exomiser and Splice. Annovar, snpEff and Exomiser databases are automatically downloaded (see <a href="help.md#databases-tool">HOWARD Help Databases tool</a>). Quick annotation allows to annotates by simply defining external tools keywords.</p>
<h5 id="bcftools-annotation">BCFTools annotation</h5>
<p>For BCFTools, use HOWARD keyword <code>bcftools</code> and list (separator <code>:</code> or <code>+</code>) annotation databases with format such as VCF or BED (compressed). More options are available using <a href="help.param.md">HOWARD Parameters JSON</a> file.</p>
<blockquote>
<p>Example: VCF annotation with Cosmic VCF databases and refGene BED database</p>
<pre class="hljs"><code><div>howard annotation \
   --input=tests/data/example.vcf.gz \
   --annotations='bcftools:tests/databases/annotations/current/hg19/cosmic70.vcf.gz+tests/databases/annotations/current/hg19/refGene.bed.gz' \
   --output=/tmp/example.howard.vcf.gz
</div></code></pre>
</blockquote>
<h5 id="annovar-annotation">Annovar annotation</h5>
<p>For Annovar tool, use HOWARD keyword <code>annovar</code> and mention specific Annovar database keywords (separator <code>:</code>). More options are available using <a href="help.param.md">HOWARD Parameters JSON</a> file.</p>
<blockquote>
<p>Example: VCF annotation with Annovar refGene and cosmic70</p>
<pre class="hljs"><code><div>howard annotation \
   --input=tests/data/example.vcf.gz \
   --annotations='annovar:refGene:cosmic70' \
   --output=/tmp/example.howard.tsv
</div></code></pre>
</blockquote>
<h5 id="snpeff-annotation">snpEff annotation</h5>
<p>For snpEff tool, use HOWARD keyword <code>snpeff</code>. Options are available for quick annotation with snpEff, see <a href="help.param.md#snpeff">HOWARD Parameters JSON - snpEff</a> for more options.</p>
<blockquote>
<p>Example: VCF annotation with snpEff</p>
<pre class="hljs"><code><div>howard annotation \
   --input=tests/data/example.vcf.gz \
   --annotations='snpeff' \
   --output=/tmp/example.howard.tsv
</div></code></pre>
</blockquote>
<h5 id="exomiser-annotation">Exomiser Annotation</h5>
<p>For Exomiser tool, use HOWARD keyword <code>exomiser</code>. A list of options can be provided as key-value format, such as exomiser release, a preset (pre-configured options), source of transcripts (e.g. 'refseq', 'ucsc'), a list of HPO terms (do not use ':' separator, e.g. '0001156', 'HP0001156', 'hpo0001156'). More options are available using <a href="help.param.md">HOWARD Parameters JSON</a> file.</p>
<blockquote>
<p>Example: VCF annotation with Exomiser (exome preset, list of HPO terms, transcript as refseq and release 2109)</p>
<pre class="hljs"><code><div>howard annotation \
   --input=tests/data/example.vcf.gz \
   --annotations='exomiser:preset=exome:hpo=0001156+0001363+0011304+0010055:transcript_source=refseq:release=2109' \
   --output=/tmp/example.howard.tsv
</div></code></pre>
</blockquote>
<h5 id="splice-annotation">Splice Annotation</h5>
<p>For Splice tool, use HOWARD keyword <code>splice</code>. A list of options can be provided as key-value format, such as split mode, spliceAI distance, spliceAI mask. More options are available using <a href="help.param.md">HOWARD Parameters JSON</a> file.</p>
<blockquote>
<p>Example: VCF annotation with Splice (split mode, spliceAI distance, spliceAI mask)</p>
<pre class="hljs"><code><div>howard annotation \
   --input=tests/data/example.vcf.gz \
   --annotations='splice:split_mode=one:spliceai_distance=500:spliceai_mask=1' \
   --output=/tmp/example.howard.tsv
</div></code></pre>
</blockquote>
<h4 id="annotation-combination">Annotation combination</h4>
<p>Quick annotation allows to combine annotations, from build-in Parquet method and external tools. Simply use a list with a comma separator.</p>
<blockquote>
<p>Example: VCF annotation with build-in Parquet method and external tools (Annovar, snpEff and Exomiser)</p>
<pre class="hljs"><code><div>howard annotation \
   --input=tests/data/example.vcf.gz \
   --annotations='tests/databases/annotations/current/hg19/dbnsfp42a.parquet,bcftools:tests/databases/annotations/current/hg19/cosmic70.vcf.gz,annovar:refGene:cosmic70,snpeff,exomiser:preset=exome:hpo=0001156+0001363+0011304+0010055' \
   --output=/tmp/example.howard.tsv
</div></code></pre>
</blockquote>
<p>See <a href="help.md#annotation-tool">HOWARD Help Annotation tool</a> tool for more information.</p>
<h3 id="annotation-parameters">Annotation parameters</h3>
<p>All annotation parameters can be defined in <a href="help.param.md">HOWARD Parameters JSON</a> file. All annotations can be combined (bild-in parquet method and external tools annotation), and options can be detailed in a full JSON file format, including selection of annotation database columns (and rename them) and specific options for external tools.</p>
<h2 id="calculation">Calculation</h2>
<p>Calculation processes variants annotations to generate new annotation, such as: identify variation type (VarType), harmonizes allele frequency (VAF) and calculate statistics (VAF_stats), extracts Nomen (transcript, cNomen, pNomen...) from an HGVS field (e.g. snpEff, Annovar) with an optional list of personalized transcripts, generates VaRank format barcode, identify trio inheritance. These calculations are based on existing annotations of variants (and genotypes).</p>
<p>Calculations are either provided by HOWARD within code, or configured into a JSON file. Calculations are either an inner HOWARD Python code, or a SQL query.</p>
<p>See <a href="help.md#calculation-tool">HOWARD Help Calculation tool</a> tool for more information.</p>
<p>To process a calculation, use is keyword with the <code>--calculations</code> parameter.</p>
<blockquote>
<p>Example: calculation of the variant type with <code>vartype</code> keyword</p>
<pre class="hljs"><code><div>howard calculation \
   --input=tests/data/example.full.vcf \
   --calculations='vartype' \
   --output=/tmp/example.calculation.tsv
</div></code></pre>
</blockquote>
<h3 id="available-calculations">Available calculations</h3>
<p>To list all available calculations, from HOWARD default configuration or with a homemade Calculation configuration JSON file, use the <code>--show_calculations</code> parameter.</p>
<blockquote>
<p>Example: List of build-in calculation</p>
<pre class="hljs"><code><div>howard calculation \
   --show_calculations
</div></code></pre>
<pre class="hljs"><code><div>#[INFO] Start
#[INFO] Available calculation operations:
#[INFO]    BARCODE: BARCODE as VaRank tool
#[INFO]    BARCODEFAMILY: BARCODEFAMILY as VaRank tool
#[INFO]    DP_STATS: Depth (DP) statistics
#[INFO]    FINDBYPIPELINE: Number of pipeline that identify the variant (for multi pipeline VCF)
#[INFO]    FINDBYSAMPLE: Number of sample that have a genotype for the variant (for multi sample VCF)
#[INFO]    GENOTYPECONCORDANCE: Concordance of genotype for multi caller VCF
#[INFO]    NOMEN: NOMEN information (e.g. NOMEN, CNOMEN, PNOMEN...) from HGVS nomenclature field
#[INFO]    SNPEFF_ANN_EXPLODE: Explode snpEff annotations
#[INFO]    SNPEFF_ANN_EXPLODE_JSON: Explode snpEff annotations in JSON format
#[INFO]    SNPEFF_ANN_EXPLODE_UNIQUIFY: Explode snpEff annotations with uniquify values
#[INFO]    SNPEFF_HGVS: HGVS nomenclatures from snpEff annotation
#[INFO]    TRANSCRIPTS_ANN: Add transcripts annotations in structured format (field 'transcripts_ann')
#[INFO]    TRANSCRIPTS_ANNOTATIONS: Add transcripts annotations in JSON and/or structured format (see param JSON file)
#[INFO]    TRANSCRIPTS_JSON: Add transcripts annotations in JSON format (field 'transcripts_json')
#[INFO]    TRANSCRIPTS_PRIORITIZATION: Prioritize transcripts with a prioritization profile (using param.json)
#[INFO]    TRIO: Inheritance for a trio family
#[INFO]    VAF: Variant Allele Frequency (VAF) harmonization
#[INFO]    VAF_STATS: Variant Allele Frequency (VAF) statistics
#[INFO]    VARIANT_ID: Variant ID generated from variant position and type
#[INFO]    VARTYPE: Variant type (e.g. SNV, INDEL, MNV, BND...)
#[INFO] End
</div></code></pre>
</blockquote>
<h3 id="calculation-configuration-json-file">Calculation configuration JSON file</h3>
<p>All calculations are configured in a JSON file. A default configuration is provided with default calculations.</p>
<p>Basically, a calculation is defined by:</p>
<ul>
<li>Type: either 'sql' for a SQL query or 'python' for a Python function</li>
<li>Name/keyword: a keyword that is used with <code>--show_calculations</code> parameter (case unsensitive)</li>
<li>Description: a description of the calculation</li>
<li>Output column information: Name, type and decription of the new annotation calculated</li>
<li>Query and fields: an SQL query (for 'sql' type) with parameters such as mandatory INFO fields</li>
<li>Function name and parameters: a existing Python function and parameters (for 'python' type)</li>
</ul>
<blockquote>
<p>Example: Configuration with calculation of variant type using an SQL query and calculation of variant id using an existing Python function <code>calculation_variant_id</code></p>
<pre class="hljs"><code><div>{
  <span class="hljs-attr">"VARTYPE"</span>: {
    <span class="hljs-attr">"type"</span>: <span class="hljs-string">"sql"</span>,
    <span class="hljs-attr">"name"</span>: <span class="hljs-string">"VARTYPE"</span>,
    <span class="hljs-attr">"description"</span>: <span class="hljs-string">"Variant type (e.g. SNV, INDEL, MNV, BND...)"</span>,
    <span class="hljs-attr">"available"</span>: <span class="hljs-literal">true</span>,
    <span class="hljs-attr">"output_column_name"</span>: <span class="hljs-string">"VARTYPE"</span>,
    <span class="hljs-attr">"output_column_type"</span>: <span class="hljs-string">"String"</span>,
    <span class="hljs-attr">"output_column_description"</span>: <span class="hljs-string">"Variant type: SNV if X&gt;Y, MOSAIC if X&gt;Y,Z or X,Y&gt;Z, INDEL if XY&gt;Z or X&gt;YZ"</span>,
    <span class="hljs-attr">"operation_query"</span>: [
      <span class="hljs-string">"CASE"</span>,
      <span class="hljs-string">"WHEN \"SVTYPE\" NOT NULL THEN \"SVTYPE\""</span>,
      <span class="hljs-string">"WHEN LENGTH(REF) = 1 AND LENGTH(ALT) = 1 THEN 'SNV'"</span>,
      <span class="hljs-string">"WHEN REF LIKE '%,%' OR ALT LIKE '%,%' THEN 'MOSAIC'"</span>,
      <span class="hljs-string">"WHEN LENGTH(REF) == LENGTH(ALT) AND LENGTH(REF) &gt; 1 THEN 'MNV'"</span>,
      <span class="hljs-string">"WHEN LENGTH(REF) &lt;&gt; LENGTH(ALT) THEN 'INDEL'"</span>,
      <span class="hljs-string">"ELSE 'UNDEFINED'"</span>,
      <span class="hljs-string">"END"</span>
    ],
    <span class="hljs-attr">"info_fields"</span>: [<span class="hljs-string">"SVTYPE"</span>],
    <span class="hljs-attr">"operation_info"</span>: <span class="hljs-literal">true</span>
  },
  <span class="hljs-attr">"variant_id"</span>: {
    <span class="hljs-attr">"type"</span>: <span class="hljs-string">"python"</span>,
    <span class="hljs-attr">"name"</span>: <span class="hljs-string">"variant_id"</span>,
    <span class="hljs-attr">"description"</span>: <span class="hljs-string">"Variant ID generated from variant position and type"</span>,
    <span class="hljs-attr">"available"</span>: <span class="hljs-literal">true</span>,
    <span class="hljs-attr">"function_name"</span>: <span class="hljs-string">"calculation_variant_id"</span>,
    <span class="hljs-attr">"function_params"</span>: []
  }
}
</div></code></pre>
</blockquote>
<p>See <a href="../config/calculations_config.json">Calculation configuration JSON file example</a>.</p>
<p>See  <a href="calculations.config.md">Calculation configuration JSON file</a> for more information.</p>
<h3 id="build-in-calculations-examples">Build-in calculations examples</h3>
<h4 id="variant-type">Variant type</h4>
<p>Variant type calculation <code>vartype</code> detect the type of variant (e.g. SNV, INDEL, MNV). Variant type are claculated with these criteria: SNV if X&gt;Y, MOSAIC if X&gt;Y,Z or X,Y&gt;Z, INDEL if XY&gt;Z or X&gt;YZ.</p>
<blockquote>
<p>Example: Identify variant types and generate a table of variant type count</p>
<pre class="hljs"><code><div>howard calculation \
   --input=tests/data/example.full.vcf \
   --calculations='vartype' \
   --output=/tmp/example.calculation.tsv

howard query \
   --input=/tmp/example.calculation.tsv \
   --explode_infos \
   --query='SELECT
               &quot;VARTYPE&quot; AS 'VariantType',
               count(*) AS 'Count'
            FROM variants
            GROUP BY &quot;VARTYPE&quot;
            ORDER BY count DESC'
</div></code></pre>
<pre class="hljs"><code><div>  VariantType  Count
0         BND      7
1         DUP      6
2         INS      5
3         SNV      4
4         CNV      3
5         DEL      3
6         INV      3
7      MOSAIC      2
8       INDEL      2
9         MNV      1
</div></code></pre>
</blockquote>
<h4 id="hgvs-and-nomen-from-snpeff">HGVS and NOMEN from snpEff</h4>
<p>NOMEN can be extracted from snpEff annotation (see <a href="help.param.md#snpeff">HOWARD Parameters JSON - snpEff</a>). The first calculation extract list of HGVS annotations from snpEff annotation (<code>snpeff_hgvs</code> keyword), the second calculation choose the NOMEN from snpEff HGVS annotations using a list of reference transcripts (<code>NOMEN</code> keyword, <code>--hgvs_field</code> and <code>--transcripts</code> parameters). More options are available (see <a href="help.param.md">HOWARD Parameters JSON</a>). See <a href="#parameters">Parameters</a> for more information about list of reference transcripts</p>
<blockquote>
<p>Example: Calculate NOMEN by extracting hgvs from snpEff annotation and identifying transcripts from a list</p>
<pre class="hljs"><code><div>howard calculation \
   --input=tests/data/example.ann.vcf.gz \
   --calculations='snpeff_hgvs,NOMEN' \
   --hgvs_field='snpeff_hgvs' \
   --transcripts=tests/data/transcripts.tsv \
   --output=/tmp/example.NOMEN.vcf.gz

howard query \
   --input=/tmp/example.NOMEN.vcf.gz \
   --explode_infos \
   --query=&quot;SELECT GNOMEN, NOMEN, snpeff_hgvs \
            FROM variants \
            WHERE GNOMEN='EGFR'&quot;
</div></code></pre>
<pre class="hljs"><code><div>  GNOMEN  NOMEN                                           snpeff_hgvs
0   EGFR  EGFR:NM_001346897:exon19:c.2226G&gt;A:p.Gln742Gln  EGFR:NM_005228.5:exon20:c.2361G&gt;A:p.Gln787Gln,...
</div></code></pre>
</blockquote>
<h2 id="prioritization">Prioritization</h2>
<p>See <a href="help.md#prioritization-tool">HOWARD Help Prioritization tool</a> tool for more information.</p>
<p>Prioritization algorithm uses profiles to flag variants (as passed or filtered), calculate a prioritization score, and automatically generate a comment for each variants (example: 'polymorphism identified in dbSNP. associated to Lung Cancer. Found in ClinVar database'). Prioritization profiles are defined in a configuration file in JSON format. A profile is defined as a list of annotation/value, using wildcards and comparison options (contains, lower than, greater than, equal...). Annotations fields may be quality values (usually from callers, such as 'DP') or other annotations fields provided by annotations tools, such as HOWARD itself (example: COSMIC, Clinvar, 1000genomes, PolyPhen, SIFT).</p>
<h3 id="prioritization-options">Prioritization options</h3>
<p>Multiple profiles can be used simultaneously (<code>--prioritizations</code> option), which is useful to define multiple validation/prioritization levels (e.g. 'standard', 'stringent', 'rare variants', 'low allele frequency', 'GERMLINE'). By default, all profiles will be processed. A default profile can be defined with <code>--default_profile</code> option (by default, the first profile in list of profiles is selected).</p>
<p>Prioritization score can be calculated following multiple mode. The <code>HOWARD</code> mode will increment scores of all passing filters (default). The <code>VaRank</code> mode will select the maximum score from all passing filters.</p>
<p>Prioritization fields can be selected from:</p>
<ul>
<li>PZScore: calculated score from all passing filters, depending of the mode</li>
<li>PZFlag: final flag ('PASS' or 'FILTERED'), with strategy that consider a variant is filtered as soon as at least one filter do not pass. By default, the variant is considered as 'PASS' (no filter pass)</li>
<li>PZComment: concatenation of all passing filter comments</li>
<li>PZTags: combinason of score, flags and comments in a tags format (e.g. 'PZFlag#PASS|PZScore#15|PZComment#Described on ...')</li>
<li>PZInfos: information about passing filter criteria</li>
</ul>
<blockquote>
<p>Example: Prioritize variants from criteria on INFO annotations for profiles 'default' and 'GERMLINE' (from 'prioritization_profiles.json' profiles configuration), export prioritization tags, and query variants passing filters</p>
<pre class="hljs"><code><div>howard prioritization \
   --input=tests/data/example.vcf.gz \
   --prioritizations='default,GERMLINE' \
   --prioritization_config=config/prioritization_profiles.json \
   --default_profile='default' \
   --pzfields='PZFlag,PZScore,PZComment,PZTags,PZInfos' \
   --prioritization_score_mode='HOWARD' \
   --output=/tmp/example.prioritized.vcf.gz

howard query \
   --input=/tmp/example.prioritized.vcf.gz \
   --explode_infos \
   --query=&quot;SELECT \&quot;#CHROM\&quot;, POS, ALT, REF, \&quot;PZFlag\&quot;, \&quot;PZScore\&quot;, \&quot;PZTags\&quot;, \&quot;DP\&quot;, \&quot;CLNSIG\&quot; \
            FROM variants \
            WHERE \&quot;PZScore\&quot; &gt; 0 \
              AND \&quot;PZFlag\&quot; == 'PASS' \
            ORDER BY \&quot;PZScore\&quot; DESC&quot;
</div></code></pre>
<pre class="hljs"><code><div>  #CHROM       POS ALT REF    PZFlag  PZScore                                             PZTags     DP          CLNSIG
<span class="hljs-number">0</span>   chr1     <span class="hljs-number">28736</span>   C   A      PASS       <span class="hljs-number">15</span>  PZFlag#PASS|PZScore#<span class="hljs-number">15</span>|PZComment#Described on ...    <span class="hljs-literal">NaN</span>      pathogenic
<span class="hljs-number">1</span>   chr1     <span class="hljs-number">35144</span>   C   A  FILTERED     <span class="hljs-number">-100</span>  PZFlag#FILTERED|PZScore#<span class="hljs-number">-100</span>|PZComment#Describ...    <span class="hljs-literal">NaN</span>  non-pathogenic
<span class="hljs-number">2</span>   chr1     <span class="hljs-number">69101</span>   G   A      PASS        <span class="hljs-number">5</span>  PZFlag#PASS|PZScore#<span class="hljs-number">5</span>|PZComment#DP higher than...   <span class="hljs-number">50.0</span>            None
<span class="hljs-number">3</span>   chr1    <span class="hljs-number">768251</span>   G   A      PASS        <span class="hljs-number">0</span>          PZFlag#PASS|PZScore#<span class="hljs-number">0</span>|PZComment#|PZInfos#    <span class="hljs-literal">NaN</span>            None
<span class="hljs-number">4</span>   chr1    <span class="hljs-number">768252</span>   G   A      PASS        <span class="hljs-number">0</span>          PZFlag#PASS|PZScore#<span class="hljs-number">0</span>|PZComment#|PZInfos#    <span class="hljs-literal">NaN</span>            None
<span class="hljs-number">5</span>   chr1    <span class="hljs-number">768253</span>   G   A      PASS        <span class="hljs-number">0</span>          PZFlag#PASS|PZScore#<span class="hljs-number">0</span>|PZComment#|PZInfos#    <span class="hljs-literal">NaN</span>            None
<span class="hljs-number">6</span>   chr7  <span class="hljs-number">55249063</span>   A   G      PASS        <span class="hljs-number">5</span>  PZFlag#PASS|PZScore#<span class="hljs-number">5</span>|PZComment#DP higher than...  <span class="hljs-number">125.0</span>            None
</div></code></pre>
</blockquote>
<h3 id="prioritization-query">Prioritization query</h3>
<p>Prioritization fields can be then easily querying, by filtering on fields and order by fields.</p>
<blockquote>
<p>Example: Query variants using prioritization fields</p>
<pre class="hljs"><code><div>howard query \
   --input=/tmp/example.prioritized.vcf.gz \
   --explode_infos \
   --query=&quot;SELECT \&quot;#CHROM\&quot;, POS, ALT, REF, \&quot;PZFlag\&quot;, \&quot;PZScore\&quot;, \&quot;PZTags\&quot;, \&quot;DP\&quot;, \&quot;CLNSIG\&quot; \
            FROM variants \
            WHERE \&quot;PZScore\&quot; &gt; 0 \
              AND \&quot;PZFlag\&quot; == 'PASS' \
            ORDER BY \&quot;PZScore\&quot; DESC&quot;
</div></code></pre>
<pre class="hljs"><code><div>  #CHROM       POS ALT REF PZFlag  PZScore                                             PZTags     DP      CLNSIG
<span class="hljs-number">0</span>   chr1     <span class="hljs-number">28736</span>   C   A   PASS       <span class="hljs-number">15</span>  PZFlag#PASS|PZScore#<span class="hljs-number">15</span>|PZComment#Described on ...    <span class="hljs-literal">NaN</span>  pathogenic
<span class="hljs-number">1</span>   chr1     <span class="hljs-number">69101</span>   G   A   PASS        <span class="hljs-number">5</span>  PZFlag#PASS|PZScore#<span class="hljs-number">5</span>|PZComment#DP higher than <span class="hljs-number">50</span>   <span class="hljs-number">50.0</span>        None
<span class="hljs-number">2</span>   chr7  <span class="hljs-number">55249063</span>   A   G   PASS        <span class="hljs-number">5</span>  PZFlag#PASS|PZScore#<span class="hljs-number">5</span>|PZComment#DP higher than <span class="hljs-number">50</span>  <span class="hljs-number">125.0</span>        None
</div></code></pre>
</blockquote>
<blockquote>
<p>Example: Query variants with different prioritization flag between profiles</p>
<pre class="hljs"><code><div>howard query \
   --input=/tmp/example.prioritized.vcf.gz \
   --explode_infos \
   --query=&quot;SELECT \&quot;#CHROM\&quot;, POS, ALT, REF, \&quot;PZFlag_default\&quot;, \&quot;PZFlag_GERMLINE\&quot; \
            FROM variants \
            WHERE \&quot;PZFlag_default\&quot; != \&quot;PZFlag_GERMLINE\&quot; \
            ORDER BY \&quot;PZScore\&quot; DESC&quot;
</div></code></pre>
<pre class="hljs"><code><div>  #CHROM    POS ALT REF PZFlag_default PZFlag_GERMLINE
<span class="hljs-number">0</span>   chr1  <span class="hljs-number">35144</span>   C   A       FILTERED            PASS
</div></code></pre>
</blockquote>
<blockquote>
<p>Example: Showing propritization comments of variants, with flags and scores</p>
<pre class="hljs"><code><div>howard query \
   --input=/tmp/example.prioritized.vcf.gz \
   --explode_infos \
   --query=&quot;SELECT \&quot;#CHROM\&quot;, POS, ALT, REF, \&quot;PZComment\&quot;, \&quot;PZFlag\&quot;, \&quot;PZScore\&quot; \
            FROM variants WHERE \&quot;PZComment\&quot; IS NOT NULL \
            ORDER BY \&quot;PZScore\&quot; DESC&quot;
</div></code></pre>
<pre class="hljs"><code><div>  #CHROM       POS ALT REF                                        PZComment    PZFlag  PZScore
<span class="hljs-number">0</span>   chr1     <span class="hljs-number">28736</span>   C   A      Described on CLINVAR database <span class="hljs-keyword">as</span> pathogenic      PASS       <span class="hljs-number">15</span>
<span class="hljs-number">1</span>   chr1     <span class="hljs-number">69101</span>   G   A                                DP higher than <span class="hljs-number">50</span>      PASS        <span class="hljs-number">5</span>
<span class="hljs-number">2</span>   chr7  <span class="hljs-number">55249063</span>   A   G                                DP higher than <span class="hljs-number">50</span>      PASS        <span class="hljs-number">5</span>
<span class="hljs-number">3</span>   chr1     <span class="hljs-number">35144</span>   C   A  Described on CLINVAR database <span class="hljs-keyword">as</span> non-pathogenic  FILTERED     <span class="hljs-number">-100</span>
</div></code></pre>
</blockquote>
<p>Prioritization profiles are defined in a JSON configuration file. Each profiles are defined as a list of annotation fields with associated filters (type of comparison and threshold, with related score, flag and comment).</p>
<blockquote>
<p>Example: Profiles with 2 filters on annotation field 'DP' (threashold 50) and 2 filters on annotation field 'CLNSIG' (&quot;pathogenic&quot; or &quot;non-pathogenic&quot;)</p>
<pre class="hljs"><code><div>{
    <span class="hljs-attr">"default"</span>: {
        <span class="hljs-attr">"DP"</span>: [
            {
                <span class="hljs-attr">"type"</span>: <span class="hljs-string">"gte"</span>,
                <span class="hljs-attr">"value"</span>: <span class="hljs-string">"50"</span>,
                <span class="hljs-attr">"score"</span>: <span class="hljs-number">5</span>,
                <span class="hljs-attr">"flag"</span>: <span class="hljs-string">"PASS"</span>,
                <span class="hljs-attr">"comment"</span>: [
                    <span class="hljs-string">"DP higher than 50"</span>
                ]
            },
            {
                <span class="hljs-attr">"type"</span>: <span class="hljs-string">"lt"</span>,
                <span class="hljs-attr">"value"</span>: <span class="hljs-string">"50"</span>,
                <span class="hljs-attr">"score"</span>: <span class="hljs-number">0</span>,
                <span class="hljs-attr">"flag"</span>: <span class="hljs-string">"FILTERED"</span>,
                <span class="hljs-attr">"comment"</span>: [
                    <span class="hljs-string">"DP lower than 50"</span>
                ]
            }
        ],
        <span class="hljs-attr">"CLNSIG"</span>: [
            {
                <span class="hljs-attr">"type"</span>: <span class="hljs-string">"equals"</span>,
                <span class="hljs-attr">"value"</span>: <span class="hljs-string">"pathogenic"</span>,
                <span class="hljs-attr">"score"</span>: <span class="hljs-number">15</span>,
                <span class="hljs-attr">"flag"</span>: <span class="hljs-string">"PASS"</span>,
                <span class="hljs-attr">"comment"</span>: [
                    <span class="hljs-string">"Described on CLINVAR database as pathogenic"</span>
                ]
            },
            {
                <span class="hljs-attr">"type"</span>: <span class="hljs-string">"equals"</span>,
                <span class="hljs-attr">"value"</span>: <span class="hljs-string">"non-pathogenic"</span>,
                <span class="hljs-attr">"score"</span>: <span class="hljs-number">-100</span>,
                <span class="hljs-attr">"flag"</span>: <span class="hljs-string">"FILTERED"</span>,
                <span class="hljs-attr">"comment"</span>: [
                    <span class="hljs-string">"Described on CLINVAR database as non-pathogenic"</span>
                ]
            }
        ]
    }
}
</div></code></pre>
</blockquote>
<p>See <a href="help.config.prioritization.md">HOWARD Help Prioritization Profiles</a> for more options.</p>
<h2 id="hgvs-annotation">HGVS annotation</h2>
<p>HOWARD annotates variants with HGVS annotation using HUGO HGVS internation Sequence Variant Nomenclature (http://varnomen.hgvs.org/). Annotation refere to refGene and genome to generate HGVS nomenclature for all available transcripts. This annotation add 'hgvs' field into VCF INFO column of a VCF file.</p>
<p>To enhance the functionality of HGVS tool, several options available. The <code>--use_gene</code> option enables the utilization of gene information for the generation of HGVS annotation, providing a concise representation such as 'NM_152232(TAS1R2):c.231T&gt;C'. Alternatively, the <code>--use_exon</code> option incorporates exon details into the annotation, as seen in the example 'NM_152232(exon2):c.231T&gt;C', but this is only applicable if 'use_gene' is not enabled. For a more detailed HGVS annotation, activate the <code>--full_format</code> option, which generates a comprehensive HGVS annotation, including all available information, such as 'TAS1R2:NM_152232:NP_689418:c.231T&gt;C:p.Cys77Arg'. This is non-standard format that ensures exhaustive data representation. Furthermore, specify the format of amino acid codons with the <code>--codon_type</code> option, choosing between single-character ('1'), three-character ('3'), or full-name formats ('FULL'). Additionally, protein-level information can be integrated using the <code>--use_protein</code> option, enabling annotations like 'NP_689418:p.Cys77Arg', which can be combined with DNA-level annotations using the <code>--add_protein</code> option (e.g. 'NM_152232:c.231T&gt;C,NP_689418:p.Cys77Arg').</p>
<p>All these options can be combined into one quick option <code>--hgvs</code>. Simply concatenate options (separator ',') with there value (e.g <code>--hgvs=use_gene:True</code> or <code>--hgvs=use_gene:False,codon_type:FULL</code>). By default, value option is 'True' (e.g. <code>--hgvs=use_gene</code> is equal to <code>--hgvs=use_gene:True</code>)</p>
<p>For database management, specify refSeq annotation files with <code>--refgene</code> and <code>--refseqlink</code> options (or set custom folders with <code>--refseq-folder</code>), and genomes folder using and <code>--genomes-folder</code> (folders are set by default, see <a href="help.config.md">HOWARD Configuration JSON</a> file).</p>
<p>See <a href="help.md#prioritization-tool">HOWARD Help Prioritization tool</a> for more options.</p>
<blockquote>
<p>Example: HGVS annotation with quick options</p>
<pre class="hljs"><code><div>howard hgvs \
   --input=tests/data/example.vcf.gz \
   --output=/tmp/example.process.tsv \
   --hgvs=full_format,use_exon

howard query \
   --input=/tmp/example.process.tsv \
   --explode_infos \
   --query=&quot;SELECT hgvs \
            FROM variants &quot;
</div></code></pre>
</blockquote>
<blockquote>
<p>Example: HGVS annotation with separated options</p>
<pre class="hljs"><code><div>howard hgvs \
   --input=tests/data/example.vcf.gz \
   --output=/tmp/example.process.tsv \
   --full_format \
   --use_exon

howard query \
   --input=/tmp/example.process.tsv \
   --explode_infos \
   --query=&quot;SELECT hgvs \
            FROM variants &quot;
</div></code></pre>
<pre class="hljs"><code><div>                                                hgvs
0                     WASH7P:NR_024540.1:n.50+585T&gt;G
1     FAM138A:NR_026818.1:exon3:n.597T&gt;G:p.Tyr199Asp
2  OR4F5:NM_001005484.2:NP_001005484.2:exon3:c.74...
3  LINC01128:NR_047526.1:n.287+3767A&gt;G,LINC01128:...
4  LINC01128:NR_047526.1:n.287+3768A&gt;G,LINC01128:...
5  LINC01128:NR_047526.1:n.287+3769A&gt;G,LINC01128:...
6  EGFR:NM_001346897.2:NP_001333826.1:exon19:c.22...
</div></code></pre>
</blockquote>
<h2 id="process">Process</h2>
<p>HOWARD process tool manage genetic variations to:</p>
<ul>
<li>annotates genetic variants with multiple annotation databases/files and tools</li>
<li>calculates and normalizes annotations</li>
<li>prioritizes variants with profiles (list of citeria) to calculate scores and flags</li>
<li>annotates genetic variants with HGVS nomenclature</li>
<li>translates into various formats</li>
<li>query genetic variants and annotations</li>
<li>generates variants statistics</li>
</ul>
<p>This process tool combines all other tools to pipe them in a uniq command, through available options or a parameters file in JSON format (see <a href="docs/help.param.md">HOWARD Parameters JSON</a> file).</p>
<p>See <a href="help.md#process-tool">HOWARD Help Process tool</a> tool for more information (under development).</p>
<h3 id="process-with-options">Process with options</h3>
<p>Process tool uses quick options for annotation, calculation and prioritization to enrich variant file, and to query variants annotations.</p>
<blockquote>
<p>Example: Process command with options (HGVS, annotation, calculation)</p>
<pre class="hljs"><code><div>howard process \
   --input=tests/data/example.vcf.gz \
   --output=/tmp/example.process.tsv \
   --hgvs=full_format,use_exon \
   --annotations='tests/databases/annotations/current/hg19/avsnp150.parquet,tests/databases/annotations/current/hg19/dbnsfp42a.parquet,tests/databases/annotations/current/hg19/gnomad211_genome.parquet' \
   --calculations=&quot;NOMEN&quot; \
   --explode_infos \
   --query=&quot;SELECT NOMEN, REVEL_score, SIFT_score, AF AS 'gnomad_AF', ClinPred_score, ClinPred_pred \
            FROM variants&quot;
</div></code></pre>
<pre class="hljs"><code><div>                                            NOMEN REVEL_score SIFT_score gnomad_AF ClinPred_score ClinPred_pred
0                    WASH7P:NR_024540:n.50+585T&gt;G        None       None      None           None          None
1    FAM138A:NR_026818:exon3:n.597T&gt;G:p.Tyr199Asp        None       None      None           None          None
2     OR4F5:NP_001005484:exon3:c.74A&gt;G:p.Glu25Gly       0.076      0.005      None          0.688             D
3               LINC01128:NR_047526:n.287+3767A&gt;G        None       None      None           None          None
4               LINC01128:NR_047526:n.287+3768A&gt;G        None       None      None           None          None
5               LINC01128:NR_047526:n.287+3769A&gt;G        None       None      None           None          None
6  EGFR:NM_001346897:exon19:c.2226G&gt;A:p.Gln742Gln        None       None    0.5029           None          None
</div></code></pre>
</blockquote>
<blockquote>
<p>Example: Full process command with options (HGVS, annotation parquet, snpEff and Annovar, calculation and prioritization)</p>
<pre class="hljs"><code><div>howard process \
   --input=tests/data/example.vcf.gz \
   --output=/tmp/example.process.tsv \
   --hgvs=full_format,use_exon \
   --annotations=&quot;tests/databases/annotations/current/hg19/avsnp150.parquet,tests/databases/annotations/current/hg19/dbnsfp42a.parquet,tests/databases/annotations/current/hg19/gnomad211_genome.parquet,bcftools:tests/databases/annotations/current/hg19/cosmic70.vcf.gz,snpeff,annovar:refGene&quot; \
   --calculations=&quot;vartype,snpeff_hgvs,VAF,NOMEN&quot; \
   --prioritizations=&quot;default,GERMLINE&quot; \
   --prioritization_config=&quot;config/prioritization_profiles.json&quot; \
   --explode_infos \
   --query=&quot;SELECT NOMEN, PZFlag, PZScore, snpeff_hgvs \
            FROM variants \
            ORDER BY PZScore DESC&quot;
</div></code></pre>
<pre class="hljs"><code><div>                                            NOMEN    PZFlag  PZScore                                        snpeff_hgvs
0                    WASH7P:NR_024540:n.50+585T&gt;G      PASS       15  MIR1302-2:NR_036051.1:n.-1630A&gt;C,MIR1302-9:NR_...
1     OR4F5:NP_001005484:exon3:c.74A&gt;G:p.Glu25Gly      PASS        5       OR4F5:NM_001005484.1:exon1:c.11A&gt;G:p.Glu4Gly
2  EGFR:NM_001346897:exon19:c.2226G&gt;A:p.Gln742Gln      PASS        5  EGFR:NM_005228.5:exon20:c.2361G&gt;A:p.Gln787Gln,...
3               LINC01128:NR_047526:n.287+3767A&gt;G      PASS        0  LINC01128:NR_047519.1:exon2:n.287+3767A&gt;G,LINC...
4               LINC01128:NR_047526:n.287+3768A&gt;G      PASS        0  LINC01128:NR_047519.1:exon2:n.287+3768A&gt;G,LINC...
5               LINC01128:NR_047526:n.287+3769A&gt;G      PASS        0  LINC01128:NR_047519.1:exon2:n.287+3769A&gt;G,LINC...
6    FAM138A:NR_026818:exon3:n.597T&gt;G:p.Tyr199Asp  FILTERED     -100  MIR1302-2:NR_036051.1:n.*4641A&gt;C,MIR1302-9:NR_...
</div></code></pre>
</blockquote>
<h3 id="process-with-parameters-json-file">Process with parameters JSON file</h3>
<p>In order to fine tune process, all tools can be defined in a <a href="docs/help.param.md">HOWARD Parameters JSON</a>. This allows to add specific options, such as selecting specific fields (and rename them) for annotation, defining options for external tools, specifying a list of transcripts of preference for NOMEN calculation. This Parameters JSON file can be combine with options.</p>
<blockquote>
<p>Example: Full process command with <a href="#parameters">Parameters JSON file example</a> and a query as option</p>
<pre class="hljs"><code><div>howard process \
   --input=tests/data/example.vcf.gz \
   --output=/tmp/example.process.tsv \
   --param=config/param.json \
   --explode_infos \
   --query=&quot;SELECT NOMEN, PZFlag, PZScore, PZComment \
            FROM variants \
            ORDER BY PZScore DESC&quot;
</div></code></pre>
<pre class="hljs"><code><div>                                            NOMEN    PZFlag  PZScore                                        PZComment
0              WASH7P:NR_024540:exon1:n.50+585T&gt;G      PASS       15.0      Described on CLINVAR database as pathogenic
1      OR4F5:NM_001005484:exon1:c.11A&gt;G:p.Glu4Gly      PASS        5.0                                DP higher than 50
2  EGFR:NM_001346897:exon19:c.2226G&gt;A:p.Gln742Gln      PASS        5.0                                DP higher than 50
3         LINC01128:NR_047519:exon2:n.287+3767A&gt;G      PASS        0.0                                              NaN
4         LINC01128:NR_047519:exon2:n.287+3768A&gt;G      PASS        0.0                                              NaN
5         LINC01128:NR_047519:exon2:n.287+3769A&gt;G      PASS        0.0                                              NaN
6                  MIR1302-9:NR_036266:n.*4641A&gt;C  FILTERED        NaN  Described on CLINVAR database as non-pathogenic
</div></code></pre>
<pre class="hljs"><code><div>column -t  /tmp/example.process.tsv 
</div></code></pre>
<pre class="hljs"><code><div>NOMEN                                           PZFlag    PZScore    PZComment
WASH7P:NR_024540:exon1:n.50+585T&gt;G              PASS      15         Described  on       CLINVAR   database  as              pathogenic
OR4F5:NM_001005484:exon1:c.11A&gt;G:p.Glu4Gly      PASS      5          DP         higher   than      50
EGFR:NM_001346897:exon19:c.2226G&gt;A:p.Gln742Gln  PASS      5          DP         higher   than      50
LINC01128:NR_047519:exon2:n.287+3767A&gt;G         PASS      0
LINC01128:NR_047519:exon2:n.287+3768A&gt;G         PASS      0
LINC01128:NR_047519:exon2:n.287+3769A&gt;G         PASS      0
MIR1302-9:NR_036266:n.*4641A&gt;C                  FILTERED  Described  on         CLINVAR  database  as        non-pathogenic
</div></code></pre>
</blockquote>
<blockquote>
<p>Example: Full process command with <a href="#parameters">Parameters JSON file example</a> and switch off query and generate a VCF file</p>
<pre class="hljs"><code><div>howard process \
   --input=tests/data/example.vcf.gz \
   --output=/tmp/example.process.vcf \
   --param=config/param.json \
   --explode_infos \
   --query=&quot;&quot;

howard query \
   --input=/tmp/example.process.vcf \
   --explode_infos \
   --query=&quot;SELECT \&quot;#CHROM\&quot;, POS, ALT, REF, NOMEN, PZComment, PZFlag, PZScore \
            FROM variants \
            ORDER BY PZScore DESC&quot;
</div></code></pre>
<pre class="hljs"><code><div>  #CHROM       POS ALT REF                                           NOMEN                                        PZComment    PZFlag  PZScore
<span class="hljs-number">0</span>   chr1     <span class="hljs-number">28736</span>   C   A              WASH7P:NR_024540:exon1:n<span class="hljs-number">.50</span>+<span class="hljs-number">585</span>T&gt;G      Described on CLINVAR database <span class="hljs-keyword">as</span> pathogenic      PASS     <span class="hljs-number">15.0</span>
<span class="hljs-number">1</span>   chr1     <span class="hljs-number">69101</span>   G   A      OR4F5:NM_001005484:exon1:c<span class="hljs-number">.11</span>A&gt;G:p.Glu4Gly                                DP higher than <span class="hljs-number">50</span>      PASS      <span class="hljs-number">5.0</span>
<span class="hljs-number">2</span>   chr7  <span class="hljs-number">55249063</span>   A   G  EGFR:NM_001346897:exon19:c<span class="hljs-number">.2226</span>G&gt;A:p.Gln742Gln                                DP higher than <span class="hljs-number">50</span>      PASS      <span class="hljs-number">5.0</span>
<span class="hljs-number">3</span>   chr1    <span class="hljs-number">768251</span>   G   A         LINC01128:NR_047519:exon2:n<span class="hljs-number">.287</span>+<span class="hljs-number">3767</span>A&gt;G                                             None      PASS      <span class="hljs-number">0.0</span>
<span class="hljs-number">4</span>   chr1    <span class="hljs-number">768252</span>   G   A         LINC01128:NR_047519:exon2:n<span class="hljs-number">.287</span>+<span class="hljs-number">3768</span>A&gt;G                                             None      PASS      <span class="hljs-number">0.0</span>
<span class="hljs-number">5</span>   chr1    <span class="hljs-number">768253</span>   G   A         LINC01128:NR_047519:exon2:n<span class="hljs-number">.287</span>+<span class="hljs-number">3769</span>A&gt;G                                             None      PASS      <span class="hljs-number">0.0</span>
<span class="hljs-number">6</span>   chr1     <span class="hljs-number">35144</span>   C   A                  MIR1302<span class="hljs-number">-9</span>:NR_036266:n.*<span class="hljs-number">4641</span>A&gt;C  Described on CLINVAR database <span class="hljs-keyword">as</span> non-pathogenic  FILTERED      <span class="hljs-literal">NaN</span>
</div></code></pre>
</blockquote>

</body>
</html>
