<!doctype html>
<html lang="en">
<head>
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, minimum-scale=1" />
<meta name="generator" content="pdoc 0.10.0" />
<title>howard.functions.utils API documentation</title>
<meta name="description" content="Helper functions." />
<link rel="preload stylesheet" as="style" href="https://cdnjs.cloudflare.com/ajax/libs/10up-sanitize.css/11.0.1/sanitize.min.css" integrity="sha256-PK9q560IAAa6WVRRh76LtCaI8pjTJ2z11v0miyNNjrs=" crossorigin>
<link rel="preload stylesheet" as="style" href="https://cdnjs.cloudflare.com/ajax/libs/10up-sanitize.css/11.0.1/typography.min.css" integrity="sha256-7l/o7C8jubJiy74VsKTidCy1yBkRtiUGbVkYBylBqUg=" crossorigin>
<link rel="stylesheet preload" as="style" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.1.1/styles/github.min.css" crossorigin>
<style>:root{--highlight-color:#fe9}.flex{display:flex !important}body{line-height:1.5em}#content{padding:20px}#sidebar{padding:30px;overflow:hidden}#sidebar > *:last-child{margin-bottom:2cm}.http-server-breadcrumbs{font-size:130%;margin:0 0 15px 0}#footer{font-size:.75em;padding:5px 30px;border-top:1px solid #ddd;text-align:right}#footer p{margin:0 0 0 1em;display:inline-block}#footer p:last-child{margin-right:30px}h1,h2,h3,h4,h5{font-weight:300}h1{font-size:2.5em;line-height:1.1em}h2{font-size:1.75em;margin:1em 0 .50em 0}h3{font-size:1.4em;margin:25px 0 10px 0}h4{margin:0;font-size:105%}h1:target,h2:target,h3:target,h4:target,h5:target,h6:target{background:var(--highlight-color);padding:.2em 0}a{color:#058;text-decoration:none;transition:color .3s ease-in-out}a:hover{color:#e82}.title code{font-weight:bold}h2[id^="header-"]{margin-top:2em}.ident{color:#900}pre code{background:#f8f8f8;font-size:.8em;line-height:1.4em}code{background:#f2f2f1;padding:1px 4px;overflow-wrap:break-word}h1 code{background:transparent}pre{background:#f8f8f8;border:0;border-top:1px solid #ccc;border-bottom:1px solid #ccc;margin:1em 0;padding:1ex}#http-server-module-list{display:flex;flex-flow:column}#http-server-module-list div{display:flex}#http-server-module-list dt{min-width:10%}#http-server-module-list p{margin-top:0}.toc ul,#index{list-style-type:none;margin:0;padding:0}#index code{background:transparent}#index h3{border-bottom:1px solid #ddd}#index ul{padding:0}#index h4{margin-top:.6em;font-weight:bold}@media (min-width:200ex){#index .two-column{column-count:2}}@media (min-width:300ex){#index .two-column{column-count:3}}dl{margin-bottom:2em}dl dl:last-child{margin-bottom:4em}dd{margin:0 0 1em 3em}#header-classes + dl > dd{margin-bottom:3em}dd dd{margin-left:2em}dd p{margin:10px 0}.name{background:#eee;font-weight:bold;font-size:.85em;padding:5px 10px;display:inline-block;min-width:40%}.name:hover{background:#e0e0e0}dt:target .name{background:var(--highlight-color)}.name > span:first-child{white-space:nowrap}.name.class > span:nth-child(2){margin-left:.4em}.inherited{color:#999;border-left:5px solid #eee;padding-left:1em}.inheritance em{font-style:normal;font-weight:bold}.desc h2{font-weight:400;font-size:1.25em}.desc h3{font-size:1em}.desc dt code{background:inherit}.source summary,.git-link-div{color:#666;text-align:right;font-weight:400;font-size:.8em;text-transform:uppercase}.source summary > *{white-space:nowrap;cursor:pointer}.git-link{color:inherit;margin-left:1em}.source pre{max-height:500px;overflow:auto;margin:0}.source pre code{font-size:12px;overflow:visible}.hlist{list-style:none}.hlist li{display:inline}.hlist li:after{content:',\2002'}.hlist li:last-child:after{content:none}.hlist .hlist{display:inline;padding-left:1em}img{max-width:100%}td{padding:0 .5em}.admonition{padding:.1em .5em;margin-bottom:1em}.admonition-title{font-weight:bold}.admonition.note,.admonition.info,.admonition.important{background:#aef}.admonition.todo,.admonition.versionadded,.admonition.tip,.admonition.hint{background:#dfd}.admonition.warning,.admonition.versionchanged,.admonition.deprecated{background:#fd4}.admonition.error,.admonition.danger,.admonition.caution{background:lightpink}</style>
<style media="screen and (min-width: 700px)">@media screen and (min-width:700px){#sidebar{width:30%;height:100vh;overflow:auto;position:sticky;top:0}#content{width:70%;max-width:100ch;padding:3em 4em;border-left:1px solid #ddd}pre code{font-size:1em}.item .name{font-size:1em}main{display:flex;flex-direction:row-reverse;justify-content:flex-end}.toc ul ul,#index ul{padding-left:1.5em}.toc > ul > li{margin-top:.5em}}</style>
<style media="print">@media print{#sidebar h1{page-break-before:always}.source{display:none}}@media print{*{background:transparent !important;color:#000 !important;box-shadow:none !important;text-shadow:none !important}a[href]:after{content:" (" attr(href) ")";font-size:90%}a[href][title]:after{content:none}abbr[title]:after{content:" (" attr(title) ")"}.ir a:after,a[href^="javascript:"]:after,a[href^="#"]:after{content:""}pre,blockquote{border:1px solid #999;page-break-inside:avoid}thead{display:table-header-group}tr,img{page-break-inside:avoid}img{max-width:100% !important}@page{margin:0.5cm}p,h2,h3{orphans:3;widows:3}h1,h2,h3,h4,h5,h6{page-break-after:avoid}}</style>
<script defer src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.1.1/highlight.min.js" integrity="sha256-Uv3H6lx7dJmRfRvH8TH6kJD1TSK1aFcwgx+mdg3epi8=" crossorigin></script>
<script>window.addEventListener('DOMContentLoaded', () => hljs.initHighlighting())</script>
</head>
<body>
<main>
<article id="content">
<header>
<h1 class="title">Module <code>howard.functions.utils</code></h1>
</header>
<section id="section-intro">
<p>Helper functions.</p>
</section>
<section>
</section>
<section>
</section>
<section>
<h2 class="section-title" id="header-functions">Functions</h2>
<dl>
<dt id="howard.functions.utils.cdna_to_protein"><code class="name flex">
<span>def <span class="ident">cdna_to_protein</span></span>(<span>hgvs, offset, genome, chrom, transcript, ref, alt, mutation_type, codon_type: str = '3')</span>
</code></dt>
<dd>
<div class="desc"><p>The function <code><a title="howard.functions.utils.cdna_to_protein" href="#howard.functions.utils.cdna_to_protein">cdna_to_protein()</a></code> takes in various parameters related to a genetic mutation and returns
an updated HGVS object with additional protein information.</p>
<p>:param hgvs: The parameter <code>hgvs</code> is an object that represents a variant in the Human Genome
Variation Society (HGVS) format. It contains information about the variant, such as the cDNA start
and end positions
:param offset: The offset is a numerical value that represents the starting position of the genomic
sequence in the reference genome. It is used to calculate the genomic position of the mutation
:param genome: The <code>genome</code> parameter is a dictionary that represents the genomic sequence. It
contains the chromosome as the key and the corresponding DNA sequence as the value
:param chrom: The <code>chrom</code> parameter represents the chromosome on which the mutation occurs
:param transcript: The <code>transcript</code> parameter is a string that represents the transcript ID or name.
It is used to identify the specific transcript in the genome
:param ref: The parameter "ref" is a string that represents the reference nucleotide sequence. It is
used to determine the codons in the DNA sequence
:param alt: The <code>alt</code> parameter in the <code><a title="howard.functions.utils.cdna_to_protein" href="#howard.functions.utils.cdna_to_protein">cdna_to_protein()</a></code> function is a string that represents the
alternate nucleotide sequence for a mutation
:param mutation_type: The <code>mutation_type</code> parameter is a string that represents the type of
mutation. It can have the following values:
:param codon_type: The <code>codon_type</code> parameter is a string that specifies the type of codon
translation to be used. It can have one of the following values:, defaults to 3
:type codon_type: str (optional)
:return: the updated <code>hgvs</code> object.</p></div>
</dd>
<dt id="howard.functions.utils.create_refseq_table"><code class="name flex">
<span>def <span class="ident">create_refseq_table</span></span>(<span>conn, refseq_table: str = 'refseq', refseq_file: str = None) ‑> str</span>
</code></dt>
<dd>
<div class="desc"><p>The function <code><a title="howard.functions.utils.create_refseq_table" href="#howard.functions.utils.create_refseq_table">create_refseq_table()</a></code> creates a table in a database with the specified name and
structure, either using a file or without a file.</p>
<p>:param conn: The <code>conn</code> parameter is a connection object that represents a connection to a database.
It is used to execute SQL queries and interact with the database
:param refseq_table: The <code>refseq_table</code> parameter is a string that specifies the name of the table
that will be created in the database to store the RefGene data, defaults to refseq
:type refseq_table: str (optional)
:param refseq_file: The <code>refseq_file</code> parameter is a string that specifies the path to a file
containing the data for the refGene table. If this parameter is provided, the function will create
the refGene table in the database using the data from the file. If this parameter is not provided,
the function will
:type refseq_file: str
:return: the name of the refseq table that was created or used.</p></div>
</dd>
<dt id="howard.functions.utils.format_hgvs_name"><code class="name flex">
<span>def <span class="ident">format_hgvs_name</span></span>(<span>chrom, offset, ref, alt, genome, transcript, transcript_protein=None, exon=None, use_prefix=True, use_gene=True, use_protein=False, use_counsyl=False, max_allele_length=4, full_format=False, use_version=False, codon_type: str = '3')</span>
</code></dt>
<dd>
<div class="desc"><p>The <code><a title="howard.functions.utils.format_hgvs_name" href="#howard.functions.utils.format_hgvs_name">format_hgvs_name()</a></code> function generates a HGVS name from a genomic coordinate.</p>
<p>:param chrom: The <code>chrom</code> parameter represents the chromosome name. It is a string that specifies
the chromosome on which the variant occurs
:param offset: The <code>offset</code> parameter represents the genomic offset of the allele, which is the
position of the variant on the chromosome. It is used to generate the HGVS name based on the genomic
coordinate
:param ref: The <code>ref</code> parameter represents the reference allele. In genetics, a variant or mutation
can occur at a specific position in the genome, resulting in a change from the reference allele to
an alternate allele. The <code>ref</code> parameter specifies the sequence of the reference allele at that
position
:param alt: The <code>alt</code> parameter represents the alternate allele. In genetics, a variant or mutation
can occur at a specific position in the genome, resulting in a change from the reference allele to
an alternate allele. The <code>alt</code> parameter specifies the sequence of the alternate allele at that
position
:param genome: A pygr compatible genome object, which is used to retrieve genomic sequences and
annotations. It provides methods to access genomic information such as chromosome names, sequences,
and gene annotations
:param transcript: The <code>transcript</code> parameter is the transcript corresponding to the allele. It is
used to generate the HGVS name based on the genomic coordinate
:param transcript_protein: The <code>transcript_protein</code> parameter is an optional argument that
represents the protein transcript corresponding to the cDNA transcript. It is used to generate the
protein HGVS name if it exists
:param exon: The <code>exon</code> parameter is used to specify the exon number in the HGVS name. It is an
optional parameter and is used to generate a more specific HGVS name when needed
:param use_prefix: A boolean indicating whether to include a transcript/gene/chromosome prefix in
the HGVS name. If set to True, the prefix will be included; if set to False, the prefix will be
excluded, defaults to True (optional)
:param use_gene: A boolean parameter that determines whether to include the gene name in the HGVS
prefix. If set to True, the gene name will be included; if set to False, the gene name will be
excluded, defaults to True (optional)
:param use_protein: A boolean parameter that determines whether to include protein HGVS notation in
the generated HGVS name. If set to True, the protein HGVS notation will be included if it exists. If
set to False, only the genomic and transcript HGVS notation will be included, defaults to False
(optional)
:param use_counsyl: The <code>use_counsyl</code> parameter is a boolean parameter that determines whether to
use Counsyl-specific formatting for the HGVS name. If set to True, the HGVS name will be formatted
according to Counsyl's specific guidelines. If set to False, the HGVS name will be, defaults to
False (optional)
:param max_allele_length: The <code>max_allele_length</code> parameter is used to determine the maximum length
of the allele. If the length of the allele is greater than the specified <code>max_allele_length</code>, then
the allele length will be used in the HGVS name instead of the actual allele sequence. By default,
the `, defaults to 4 (optional)
:param full_format: A boolean parameter that determines whether to use the full HGVS format or not.
If set to True, the HGVS name will include the gene name, transcript name, exon number (if
provided), and the amino acid change (if protein information is available). If set to False, the
HGVS, defaults to False (optional)
:param use_version: A boolean parameter that determines whether to include the version number of the
transcript in the HGVS name. If set to True, the version number will be included; if set to False,
the version number will be excluded, defaults to False (optional)
:param codon_type: The <code>codon_type</code> parameter is a string that specifies the type of codon numbering
to be used in the HGVS name. It can have one of the following values:, defaults to 3
:type codon_type: str (optional)
:return: a formatted HGVS name generated from a genomic coordinate.</p></div>
</dd>
<dt id="howard.functions.utils.get_alt_from_sequence"><code class="name flex">
<span>def <span class="ident">get_alt_from_sequence</span></span>(<span>hgvs, genome, transcript)</span>
</code></dt>
<dd>
<div class="desc"><p>The function "get_alt_from_sequence" returns a genomic sequence from a given HGVS notation, genome,
and transcript.</p>
<p>:param hgvs: The <code>hgvs</code> parameter is an object that provides methods for working with Human Genome
Variation Society (HGVS) nomenclature. It likely has a method called <code>get_raw_coords()</code> that takes a
transcript as input and returns the chromosome, start position, and end position of the
corresponding genomic sequence
:param genome: The genome parameter refers to the genomic sequence from which the alternative allele
will be extracted
:param transcript: The transcript parameter is a string that represents the transcript ID or name
:return: the genomic sequence from the specified region in the genome.</p></div>
</dd>
<dt id="howard.functions.utils.get_genomic_sequence"><code class="name flex">
<span>def <span class="ident">get_genomic_sequence</span></span>(<span>genome, chrom, start, end)</span>
</code></dt>
<dd>
<div class="desc"><p>The function <code><a title="howard.functions.utils.get_genomic_sequence" href="#howard.functions.utils.get_genomic_sequence">get_genomic_sequence()</a></code> returns a sequence for a given genomic region.</p>
<p>:param genome: A dictionary containing genomic sequences for different chromosomes. The keys of the
dictionary are chromosome names (e.g., 'chr1', 'chr2', etc.), and the values are the corresponding
genomic sequences
:param chrom: The chrom parameter represents the chromosome or genomic region from which you want to
extract the sequence
:param start: The start parameter is the 1-based coordinate of the beginning of the genomic region
:param end: The <code>end</code> parameter is the end coordinate of the genomic region. It is a 1-based,
end-inclusive coordinate, meaning that the base at the <code>end</code> position is included in the returned
sequence
:return: a sequence for the specified genomic region.</p></div>
</dd>
<dt id="howard.functions.utils.get_refseq_table"><code class="name flex">
<span>def <span class="ident">get_refseq_table</span></span>(<span>conn, refseq_table: str = 'refseq', refseq_file: str = None) ‑> str</span>
</code></dt>
<dd>
<div class="desc"><p>The function <code><a title="howard.functions.utils.get_refseq_table" href="#howard.functions.utils.get_refseq_table">get_refseq_table()</a></code> checks if a table named <code>refseq</code> exists in a database, and if not,
creates it using the <code><a title="howard.functions.utils.create_refseq_table" href="#howard.functions.utils.create_refseq_table">create_refseq_table()</a></code> function.</p>
<p>:param conn: The parameter <code>conn</code> is expected to be a connection object that allows you to interact
with a database. It could be an instance of a database connector class, such as <code>pymysql.connect()</code>
for MySQL or <code>psycopg2.connect()</code> for PostgreSQL
:param refseq_table: The parameter "refseq_table" is a string that specifies the name of the table
in the database where the refGene data will be stored. If this table already exists in the database,
the function will return the name of the existing table. If the table does not exist, the function
will create, defaults to refseq
:type refseq_table: str (optional)
:param refseq_file: The <code>refseq_file</code> parameter is the name or path of the file that contains the
refGene data. This file is used to populate the refGene table in the database
:type refseq_file: str
:return: the name of the refseq_table.</p></div>
</dd>
<dt id="howard.functions.utils.get_transcript"><code class="name flex">
<span>def <span class="ident">get_transcript</span></span>(<span>transcripts: dict, transcript_name: str) ‑> howard.objects.transcript.Transcript</span>
</code></dt>
<dd>
<div class="desc"><p>The function <code><a title="howard.functions.utils.get_transcript" href="#howard.functions.utils.get_transcript">get_transcript()</a></code> takes a dictionary of transcripts and a name as input, and returns the
transcript associated with that name.</p>
<p>:param transcripts: A dictionary containing transcripts as values, with names as keys
:type transcripts: dict
:param name: The name parameter is a string that represents the name of the transcript that you want
to retrieve from the transcripts dictionary
:type name: str
:return: the value associated with the given name key in the transcripts dictionary.</p></div>
</dd>
<dt id="howard.functions.utils.get_vcf_allele"><code class="name flex">
<span>def <span class="ident">get_vcf_allele</span></span>(<span>hgvs, genome, transcript=None)</span>
</code></dt>
<dd>
<div class="desc"><p>The function <code><a title="howard.functions.utils.get_vcf_allele" href="#howard.functions.utils.get_vcf_allele">get_vcf_allele()</a></code> takes a HGVS name, a genome, and an optional transcript, and returns a
VCF-style allele.</p>
<p>:param hgvs: The <code>hgvs</code> parameter is an object of type <code>HGVSName</code>. It likely contains information
about a genetic variant, such as the chromosome, start and end positions, and the type of mutation
(e.g., substitution, deletion, insertion, etc.)
:param genome: The <code>genome</code> parameter is the genomic sequence from which the allele will be
extracted. It is a string representing the entire genome sequence
:param transcript: The <code>transcript</code> parameter is an optional argument that represents a transcript.
It is used to retrieve the VCF-style allele from the given HGVSName and genome. If a transcript is
provided, the function will use it to get the VCF coordinates and the reference and alternate
alleles. If no
:return: the chromosome, start position, end position, reference allele, and alternate allele.</p></div>
</dd>
<dt id="howard.functions.utils.hgvs_justify_dup"><code class="name flex">
<span>def <span class="ident">hgvs_justify_dup</span></span>(<span>chrom, offset, ref, alt, genome)</span>
</code></dt>
<dd>
<div class="desc"><p>The function <code><a title="howard.functions.utils.hgvs_justify_dup" href="#howard.functions.utils.hgvs_justify_dup">hgvs_justify_dup()</a></code> determines if an allele is a duplication and justifies it by
returning the duplicated region if applicable.</p>
<p>:param chrom: The chromosome name where the allele is located
:param offset: The offset parameter is the 1-index genomic coordinate, which represents the position
of the variant on the chromosome
:param ref: The "ref" parameter represents the reference allele, which is the allele that is present
in the reference genome at the given genomic coordinate
:param alt: The <code>alt</code> parameter represents the alternate allele, which is the allele that differs
from the reference allele at a specific genomic position
:param genome: The <code>genome</code> parameter is a pygr compatible genome object. It is an object that
represents a reference genome and provides methods to access genomic sequences
:return: a tuple containing the chromosome name, offset, reference allele, alternate allele, and
mutation type.</p></div>
</dd>
<dt id="howard.functions.utils.hgvs_justify_indel"><code class="name flex">
<span>def <span class="ident">hgvs_justify_indel</span></span>(<span>chrom, offset, ref, alt, strand, genome)</span>
</code></dt>
<dd>
<div class="desc"><p>The function <code><a title="howard.functions.utils.hgvs_justify_indel" href="#howard.functions.utils.hgvs_justify_indel">hgvs_justify_indel()</a></code> justifies an indel (insertion or deletion) according to the HGVS
standard by determining the genomic sequence around the lesion, identifying the actual lesion
sequence, and 3' justifying the offset.</p>
<p>:param chrom: The chromosome where the indel is located
:param offset: The offset parameter represents the position of the indel (insertion or deletion)
within the chromosome or genomic sequence
:param ref: The <code>ref</code> parameter represents the reference allele of the variant. It is a string that
contains the nucleotide sequence of the reference allele
:param alt: The <code>alt</code> parameter in the <code><a title="howard.functions.utils.hgvs_justify_indel" href="#howard.functions.utils.hgvs_justify_indel">hgvs_justify_indel()</a></code> function represents the alternate allele
sequence for an indel variant. It is the sequence that replaces the reference allele sequence
(<code>ref</code>) at the specified <code>offset</code> position on the <code>chrom</code> chromosome
:param strand: The parameter "strand" represents the orientation of the DNA strand where the indel
is located. It can have two possible values: "+" or "-". The "+" strand refers to the forward
strand, while the "-" strand refers to the reverse complement strand
:param genome: The <code>genome</code> parameter is a dictionary that contains the genomic sequence for each
chromosome. The keys of the dictionary are the chromosome names (e.g., "chr1", "chr2", etc.), and
the values are the corresponding genomic sequences
:return: the variables chrom, offset, ref, and alt.</p></div>
</dd>
<dt id="howard.functions.utils.hgvs_normalize_variant"><code class="name flex">
<span>def <span class="ident">hgvs_normalize_variant</span></span>(<span>chrom, offset, ref, alt, genome, transcript=None)</span>
</code></dt>
<dd>
<div class="desc"><p>The function <code><a title="howard.functions.utils.hgvs_normalize_variant" href="#howard.functions.utils.hgvs_normalize_variant">hgvs_normalize_variant()</a></code> converts a variant in VCF-style to HGVS-style by adjusting the
offset, reference and alternate alleles, and determining the mutation type.</p>
<p>:param chrom: The chromosome where the variant is located
:param offset: The offset parameter represents the position of the variant within the chromosome. It
is an integer value
:param ref: The <code>ref</code> parameter represents the reference allele in a variant
:param alt: The <code>alt</code> parameter represents the alternate allele in a variant. It is a string that
represents the alternative nucleotide(s) or sequence(s) at a specific position in the genome
:param genome: The <code>genome</code> parameter is the reference genome sequence. It is used to perform
certain operations on the variant, such as justifying indels and representing duplications
:param transcript: The <code>transcript</code> parameter is an optional argument that represents the transcript
or gene in which the variant occurs. It is used to determine the strand of the gene and to perform
certain operations on the variant. If no transcript is provided, the default value is <code>None</code>
:return: the following values: chrom, offset, ref, alt, and mutation_type.</p></div>
</dd>
<dt id="howard.functions.utils.json_perfect_exons_to_cdna_match"><code class="name flex">
<span>def <span class="ident">json_perfect_exons_to_cdna_match</span></span>(<span>ordered_exons, single=False)</span>
</code></dt>
<dd>
<div class="desc"><p>The function <code><a title="howard.functions.utils.json_perfect_exons_to_cdna_match" href="#howard.functions.utils.json_perfect_exons_to_cdna_match">json_perfect_exons_to_cdna_match()</a></code> converts a list of ordered exons into a list of cDNA
matches, where each match consists of the start and end positions of the exon, the start and end
positions of the corresponding cDNA sequence, and an optional gap list.</p>
<p>Perfectly matched exons are basically a no-gap case of cDNA match
single - use a single cDNA match (deletions for introns) - this is currently broken do not use</p>
<p>:param ordered_exons: A list of tuples representing the start and end positions of exons in a gene
sequence. The exons should be ordered based on their position in the gene
:param single: The <code>single</code> parameter is a boolean flag that determines whether to use a single cDNA
match or not. If <code>single</code> is set to <code>True</code>, the function will create a single cDNA match by
considering deletions for introns. If <code>single</code> is set to <code>False</code> (, defaults to False (optional)
:return: a list of lists, where each inner list represents a cDNA match. Each inner list contains
the start and end positions of the exon, the start and end positions of the corresponding cDNA
match, and a string representing any gaps (intron lengths) between exons.</p></div>
</dd>
<dt id="howard.functions.utils.make_transcript"><code class="name flex">
<span>def <span class="ident">make_transcript</span></span>(<span>transcript_json)</span>
</code></dt>
<dd>
<div class="desc"><p>The function <code><a title="howard.functions.utils.make_transcript" href="#howard.functions.utils.make_transcript">make_transcript()</a></code> takes a JSON object representing a transcript and creates a
Transcript object from it.</p>
<p>:param transcript_json: The <code>transcript_json</code> parameter is a JSON object that contains information
about a transcript. It should have the following keys:
:return: a Transcript object.</p></div>
</dd>
<dt id="howard.functions.utils.matches_ref_allele"><code class="name flex">
<span>def <span class="ident">matches_ref_allele</span></span>(<span>hgvs, genome, transcript=None)</span>
</code></dt>
<dd>
<div class="desc"><p>The function <code><a title="howard.functions.utils.matches_ref_allele" href="#howard.functions.utils.matches_ref_allele">matches_ref_allele()</a></code> checks if the reference allele in a given HGVS notation matches
the corresponding genomic sequence.</p>
<p>:param hgvs: The <code>hgvs</code> parameter is an object that represents a variant in the Human Genome
Variation Society (HGVS) format. It contains information about the variant's reference allele,
alternative allele, and genomic coordinates
:param genome: The <code>genome</code> parameter is the genomic sequence from which the reference allele is
extracted
:param transcript: The transcript parameter is an object that represents a transcript. It has a
property called tx_position which provides information about the position of the transcript on the
genome, including whether it is on the forward or reverse strand
:return: True if the reference allele matches the genomic sequence, and False otherwise.</p></div>
</dd>
<dt id="howard.functions.utils.parse_hgvs_name"><code class="name flex">
<span>def <span class="ident">parse_hgvs_name</span></span>(<span>hgvs_name, genome, transcript=None, get_transcript=&lt;function &lt;lambda&gt;&gt;, flank_length=30, normalize=True, lazy=False, indels_start_with_same_base=True)</span>
</code></dt>
<dd>
<div class="desc"><p>The function <code><a title="howard.functions.utils.parse_hgvs_name" href="#howard.functions.utils.parse_hgvs_name">parse_hgvs_name()</a></code> takes an HGVS name, a genome object, and optional parameters, and
returns the chromosome, start position, reference allele, and alternate allele of the variant
described by the HGVS name.</p>
<p>:param hgvs_name: The HGVS name to parse
:param genome: A pygr compatible genome object. This object represents the reference genome and
provides methods to access genomic sequences and annotations
:param transcript: The transcript parameter is an optional argument that represents the transcript
corresponding to the HGVS name. It is used to determine the reference sequence for the variant. If
not provided, the get_transcript function is used to retrieve the transcript based on the HGVS name.
If neither transcript nor get_transcript is
:param get_transcript: A function that takes a transcript name as input and returns the
corresponding transcript object. If not provided, the default behavior is to return None
:param flank_length: The <code>flank_length</code> parameter is an integer that specifies the length of the
flanking sequence to include when normalizing the variant allele. This is used in the
<code>normalize_variant</code> function to determine the reference allele and normalize the variant allele
according to the VCF standard, defaults to 30 (optional)
:param normalize: A boolean parameter that determines whether the allele should be normalized
according to the VCF standard. If set to True, the allele will be normalized; if set to False, the
allele will not be normalized, defaults to True (optional)
:param lazy: The <code>lazy</code> parameter is a boolean flag that determines whether or not to discard
version information from the incoming transcript or gene. If <code>lazy</code> is set to <code>True</code>, the version
information will be discarded. If <code>lazy</code> is set to <code>False</code>, the version information will be included
in the, defaults to False (optional)
:param indels_start_with_same_base: The parameter "indels_start_with_same_base" is a boolean flag
that determines whether or not to strip the common prefix from indels when normalizing alleles. If
set to True, the common prefix will not be stripped, defaults to True (optional)
:return: The function <code><a title="howard.functions.utils.parse_hgvs_name" href="#howard.functions.utils.parse_hgvs_name">parse_hgvs_name()</a></code> returns a tuple containing the chromosome, start position,
reference allele, and alternate allele of the parsed HGVS name.</p></div>
</dd>
<dt id="howard.functions.utils.read_genepred"><code class="name flex">
<span>def <span class="ident">read_genepred</span></span>(<span>infile, skip_first_column=False)</span>
</code></dt>
<dd>
<div class="desc"><p>The function <code><a title="howard.functions.utils.read_genepred" href="#howard.functions.utils.read_genepred">read_genepred()</a></code> reads a file in GenePred extension format and yields a dictionary for
each line, containing information about a gene.</p>
<p>:param infile: The <code>infile</code> parameter is the input file object that contains the gene annotation
data in the GenePred format. It is used to read the lines of the file and extract the necessary
information
:param skip_first_column: The <code>skip_first_column</code> parameter is a boolean flag that determines
whether to skip the first column of the input file when parsing the genePred format. By default, it
is set to <code>False</code>, which means the first column (usually the transcript ID) will be included in the
output. If you, defaults to False (optional)</p>
<p>GenePred extension format:
<a href="http://genome.ucsc.edu/FAQ/FAQformat.html#GenePredExt">http://genome.ucsc.edu/FAQ/FAQformat.html#GenePredExt</a></p>
<p>Column definitions:
0. string name;
"Name of gene (usually transcript_id from GTF)"
1. string chrom;
"Chromosome name"
2. char[1] strand;
"+ or - for strand"
3. uint txStart;
"Transcription start position"
4. uint txEnd;
"Transcription end position"
5. uint cdsStart;
"Coding region start"
6. uint cdsEnd;
"Coding region end"
7. uint exonCount;
"Number of exons"
8. uint[exonCount] exonStarts;
"Exon start positions"
9. uint[exonCount] exonEnds;
"Exon end positions"
10. uint id;
"Unique identifier"
11. string name2;
"Alternate name (e.g. gene_id from GTF)"</p></div>
</dd>
<dt id="howard.functions.utils.read_refgene"><code class="name flex">
<span>def <span class="ident">read_refgene</span></span>(<span>infile)</span>
</code></dt>
<dd>
<div class="desc"><p>The function "read_refgene" reads a genePred file with an extra column at the front and returns the
genePred data.</p>
<p>refGene = genePred with extra column at front (and ignored ones after)</p>
<p>:param infile: The input file containing the refGene data
:return: the result of calling the function <code><a title="howard.functions.utils.read_genepred" href="#howard.functions.utils.read_genepred">read_genepred()</a></code> with the argument <code>infile</code> and the
keyword argument <code>skip_first_column</code> set to <code>True</code>.</p></div>
</dd>
<dt id="howard.functions.utils.read_transcripts"><code class="name flex">
<span>def <span class="ident">read_transcripts</span></span>(<span>refgene_file)</span>
</code></dt>
<dd>
<div class="desc"><p>The function <code><a title="howard.functions.utils.read_transcripts" href="#howard.functions.utils.read_transcripts">read_transcripts()</a></code> reads all transcripts in a RefGene file and returns them as a
dictionary.</p>
<p>:param refgene_file: The <code>refgene_file</code> parameter is the file path to a RefGene file. This file
contains information about gene transcripts, such as their names, full names, and other relevant
details. The <code><a title="howard.functions.utils.read_transcripts" href="#howard.functions.utils.read_transcripts">read_transcripts()</a></code> function reads this file and returns a dictionary of transcripts,
where the keys are the
:return: a dictionary of transcripts.</p></div>
</dd>
<dt id="howard.functions.utils.variant_to_hgvs_name"><code class="name flex">
<span>def <span class="ident">variant_to_hgvs_name</span></span>(<span>chrom, offset, ref, alt, genome, transcript, transcript_protein=None, exon=None, max_allele_length=4, use_counsyl=False, codon_type: str = '3')</span>
</code></dt>
<dd>
<div class="desc"><p>The function <code><a title="howard.functions.utils.variant_to_hgvs_name" href="#howard.functions.utils.variant_to_hgvs_name">variant_to_hgvs_name()</a></code> takes in genomic coordinates, alleles, and other parameters, and
returns a HGVS-style name for the variant.</p>
<p>:param chrom: The chromosome name where the variant is located
:param offset: The <code>offset</code> parameter represents the genomic offset of the allele. It is the
position of the variant on the chromosome
:param ref: The reference allele at the given genomic coordinate
:param alt: The <code>alt</code> parameter is the alternate allele. In genetics, a variant or mutation can
occur at a specific position in the genome, and the <code>alt</code> allele represents the alternative
nucleotide or sequence at that position compared to the reference genome
:param genome: A pygr compatible genome object, which represents the reference genome sequence
:param transcript: The <code>transcript</code> parameter is the transcript corresponding to the allele. It is
used to determine the type of coordinates to use in the HGVS name (either genomic coordinates or
cDNA coordinates). If the transcript is not available, the function will use genomic coordinates
:param transcript_protein: The <code>transcript_protein</code> parameter is an optional argument that
represents the protein sequence corresponding to the transcript. It is used to populate the
<code>transcript_protein</code> attribute of the <code>HGVSName</code> object
:param exon: The <code>exon</code> parameter is an optional argument that represents the exon number or
identifier associated with the variant. It is used to populate the <code>exon</code> attribute of the
<code>HGVSName</code> object. If provided, it will be included in the final HGVS name generated by the function
:param max_allele_length: The <code>max_allele_length</code> parameter is used to determine whether to
represent the alleles as their actual sequence or as the length of the sequence. If the length of
the reference allele or alternate allele is greater than <code>max_allele_length</code>, then the length of the
allele is used instead of the actual, defaults to 4 (optional)
:param use_counsyl: A boolean flag indicating whether to use Counsyl-specific rules for single-base
indels, defaults to False (optional)
:param codon_type: The parameter <code>codon_type</code> is a string that specifies the type of codon numbering
to be used in the HGVS name. It is used in the <code><a title="howard.functions.utils.cdna_to_protein" href="#howard.functions.utils.cdna_to_protein">cdna_to_protein()</a></code> function to determine the type of
codon numbering to be used in the protein-level HGVS name. The, defaults to 3
:type codon_type: str (optional)
:return: an object of type HGVSName.</p></div>
</dd>
</dl>
</section>
<section>
</section>
</article>
<nav id="sidebar">
<h1>Index</h1>
<div class="toc">
<ul></ul>
</div>
<ul id="index">
<li><h3><a href="#header-functions">Functions</a></h3>
<ul class="">
<li><code><a title="howard.functions.utils.cdna_to_protein" href="#howard.functions.utils.cdna_to_protein">cdna_to_protein</a></code></li>
<li><code><a title="howard.functions.utils.create_refseq_table" href="#howard.functions.utils.create_refseq_table">create_refseq_table</a></code></li>
<li><code><a title="howard.functions.utils.format_hgvs_name" href="#howard.functions.utils.format_hgvs_name">format_hgvs_name</a></code></li>
<li><code><a title="howard.functions.utils.get_alt_from_sequence" href="#howard.functions.utils.get_alt_from_sequence">get_alt_from_sequence</a></code></li>
<li><code><a title="howard.functions.utils.get_genomic_sequence" href="#howard.functions.utils.get_genomic_sequence">get_genomic_sequence</a></code></li>
<li><code><a title="howard.functions.utils.get_refseq_table" href="#howard.functions.utils.get_refseq_table">get_refseq_table</a></code></li>
<li><code><a title="howard.functions.utils.get_transcript" href="#howard.functions.utils.get_transcript">get_transcript</a></code></li>
<li><code><a title="howard.functions.utils.get_vcf_allele" href="#howard.functions.utils.get_vcf_allele">get_vcf_allele</a></code></li>
<li><code><a title="howard.functions.utils.hgvs_justify_dup" href="#howard.functions.utils.hgvs_justify_dup">hgvs_justify_dup</a></code></li>
<li><code><a title="howard.functions.utils.hgvs_justify_indel" href="#howard.functions.utils.hgvs_justify_indel">hgvs_justify_indel</a></code></li>
<li><code><a title="howard.functions.utils.hgvs_normalize_variant" href="#howard.functions.utils.hgvs_normalize_variant">hgvs_normalize_variant</a></code></li>
<li><code><a title="howard.functions.utils.json_perfect_exons_to_cdna_match" href="#howard.functions.utils.json_perfect_exons_to_cdna_match">json_perfect_exons_to_cdna_match</a></code></li>
<li><code><a title="howard.functions.utils.make_transcript" href="#howard.functions.utils.make_transcript">make_transcript</a></code></li>
<li><code><a title="howard.functions.utils.matches_ref_allele" href="#howard.functions.utils.matches_ref_allele">matches_ref_allele</a></code></li>
<li><code><a title="howard.functions.utils.parse_hgvs_name" href="#howard.functions.utils.parse_hgvs_name">parse_hgvs_name</a></code></li>
<li><code><a title="howard.functions.utils.read_genepred" href="#howard.functions.utils.read_genepred">read_genepred</a></code></li>
<li><code><a title="howard.functions.utils.read_refgene" href="#howard.functions.utils.read_refgene">read_refgene</a></code></li>
<li><code><a title="howard.functions.utils.read_transcripts" href="#howard.functions.utils.read_transcripts">read_transcripts</a></code></li>
<li><code><a title="howard.functions.utils.variant_to_hgvs_name" href="#howard.functions.utils.variant_to_hgvs_name">variant_to_hgvs_name</a></code></li>
</ul>
</li>
</ul>
</nav>
</main>
<footer id="footer">
<p>Generated by <a href="https://pdoc3.github.io/pdoc" title="pdoc: Python API documentation generator"><cite>pdoc</cite> 0.10.0</a>.</p>
</footer>
</body>
</html>